<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Francesc Castro-Giner" />

<meta name="date" content="2025-10-31" />

<title>Clonal dynamics using mock CRISPR screen</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.34.0/datatables.js"></script>
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<script src="site_libs/jszip-1.13.6/jszip.min.js"></script>
<link href="site_libs/dt-ext-buttons-1.13.6/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.13.6/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.6/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.6/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.6/js/buttons.print.min.js"></script>
<link href="site_libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet" />
<script src="site_libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="site_libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
<script src="site_libs/selectize-0.12.0/selectize.min.js"></script>
<link href="site_libs/crosstalk-1.2.2/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.2/js/crosstalk.min.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">StealTHY</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="about.html">
    <span class="fa fa-info"></span>
     
    About
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-bar-chart"></span>
     
    Analyses
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Data generation</li>
    <li>
      <a href="0_differential_expression_gsea_gsva.html">Run differential gene expression analysis, GSEA and GSVA</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Main results</li>
    <li>
      <a href="patients_ctc_counts_distribution.html">CTC counts distribution in breast cancer patients</a>
    </li>
    <li>
      <a href="br16_pca.html">PCA analysis of CTC gene expression in NSG-CDX-BR16 mice</a>
    </li>
    <li>
      <a href="br16_dge.html">Differential gene expression report for NSG-CDX-BR16 data</a>
    </li>
    <li>
      <a href="gsea_across_models.html">Report of GSEA and GSVA analysis in NSG-CDX-BR16, NSG-LM2, Patient and NSG-LM2 time kinetics</a>
    </li>
    <li>
      <a href="core_gene_sets.html">Analysis of genes from candidate pathways</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="https://github.com/TheAcetoLab/saini-stealTHY">
    <span class="fa fa-github"></span>
     
  </a>
</li>
<li>
  <a href="https://www.theacetolab.com">
    <span class="fa fa-flask"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Clonal dynamics using mock CRISPR
screen</h1>
<h3 class="subtitle">Mouse libray (n = 2215)</h3>
<h4 class="author">Francesc Castro-Giner</h4>
<h4 class="date">October 31, 2025</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-10-31
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>saini-stealTHY/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240517code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240517)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240517code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240517)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomTheAcetoLabsainistealTHYtree455fa9121fcca51363cf639f7ee170fb108e34c1targetblank455fa91a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/tree/455fa9121fcca51363cf639f7ee170fb108e34c1" target="_blank">455fa91</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomTheAcetoLabsainistealTHYtree455fa9121fcca51363cf639f7ee170fb108e34c1targetblank455fa91a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/TheAcetoLab/saini-stealTHY/tree/455fa9121fcca51363cf639f7ee170fb108e34c1" target="_blank">455fa91</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/templates/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    code/raw_data_processing/rnaseq/p27851_o32062/pipelines/
    Ignored:    configuration/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/crispr/
    Ignored:    data/resources/
    Ignored:    data/rnaseq/
    Ignored:    output/.DS_Store
    Ignored:    output/clinical/
    Ignored:    output/crispr/
    Ignored:    output/rnaseq/

Untracked files:
    Untracked:  analysis/index.md
    Untracked:  analysis/style.css
    Untracked:  code/R-functions/subchunkify.R

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/_site.yml
    Deleted:    analysis/crispr-hsapiens_2180_sgRNA_r1.Rmd
    Deleted:    analysis/crispr-mm_2215_sgRNA-clonality_r1.Rmd
    Deleted:    analysis/crispr-mm_2215_sgRNA-r2oC.Rmd
    Deleted:    analysis/crispr-mm_2215_sgRNA-r2oD.Rmd
    Deleted:    analysis/crispr-mm_2215_sgRNA-r2oD_r1.Rmd
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_ko_culture--over--amhr2_control_culture_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_ko_culture--over--amhr2_control_culture_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_ko_notgfbeta--over--amhr2_ko_culture_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_ko_notgfbeta--over--amhr2_ko_culture_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_overexpression--over--amhr2_control_culture_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_overexpression--over--amhr2_control_culture_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_overexpression--over--amhr2_ko_culture_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_overexpression--over--amhr2_ko_culture_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_overexpression--over--amhr2_ko_notgfbeta_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_overexpression--over--amhr2_ko_notgfbeta_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/figure/gse-dotplot-invivo_amhr2_ko-+-amhr2_ko_culture--over--invivo_control_sgrna-+-amhr2_control_culture_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-invivo_amhr2_ko-+-amhr2_ko_culture--over--invivo_control_sgrna-+-amhr2_control_culture_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/figure/gse-dotplot-invivo_amhr2_ko--over--invivo_control_sgrna_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-invivo_amhr2_ko--over--invivo_control_sgrna_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/file/tmp/dge.xlsx
    Deleted:    analysis/templates/docs/file/tmp/gse_gsea_msigdb.h.xlsx
    Deleted:    analysis/templates/docs/file/tmp/gse_gsego_bp.xlsx
    Deleted:    analysis/templates/docs/file/tmp/gse_gsego_mf.xlsx
    Modified:   analysis/templates/rnaseq-deg-edger.Rmd
    Modified:   code/R-functions/gse_report.r
    Modified:   configuration/rmarkdown/color_palettes.R
    Modified:   configuration/rmarkdown/ggplot_theme.R
    Modified:   update_workflowr.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/crispr-mm_2215_sgRNA-clonality.Rmd</code>) and HTML
(<code>docs/crispr-mm_2215_sgRNA-clonality.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/455fa9121fcca51363cf639f7ee170fb108e34c1/analysis/crispr-mm_2215_sgRNA-clonality.Rmd" target="_blank">455fa91</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-10-31
</td>
<td>
Prepare files for publication
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheAcetoLab/saini-stealTHY/a6972e9baca3624749a8db52f17092d7d2e8c2e2/docs/crispr-mm_2215_sgRNA-clonality.html" target="_blank">a6972e9</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-07-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/b6bc77088b14bf2712a35dc41be83f6032057e27/analysis/crispr-mm_2215_sgRNA-clonality.Rmd" target="_blank">b6bc770</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-07-01
</td>
<td>
update crispr analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheAcetoLab/saini-stealTHY/f982e709689944cbcd3283f0040d2811c625c7d4/docs/crispr-mm_2215_sgRNA-clonality.html" target="_blank">f982e70</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-05-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/f2497227bea4d0c618e871f06ff31d6181a6e3d2/analysis/crispr-mm_2215_sgRNA-clonality.Rmd" target="_blank">f249722</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-05-13
</td>
<td>
mod clonality figuress
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheAcetoLab/saini-stealTHY/0e8d1db849f537c8abb88e943b7f6fd049d838fc/docs/crispr-mm_2215_sgRNA-clonality.html" target="_blank">0e8d1db</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-05-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/748478ddf826aace4d4ea19c1a8a6bf5c7f1ba6e/analysis/crispr-mm_2215_sgRNA-clonality.Rmd" target="_blank">748478d</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-05-13
</td>
<td>
mod clonality figuress
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheAcetoLab/saini-stealTHY/e4cd8b5b49c70c2dac6df9053e5582afcfcd89c5/docs/crispr-mm_2215_sgRNA-clonality.html" target="_blank">e4cd8b5</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-05-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/ab1136cc72c79ad56e6d477f4078b442e134505d/analysis/crispr-mm_2215_sgRNA-clonality.Rmd" target="_blank">ab1136c</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-05-13
</td>
<td>
update clonality
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/fa3898f6c69fe480547f4cc28cc5a78fce514b67/analysis/crispr-mm_2215_sgRNA-clonality.Rmd" target="_blank">fa3898f</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-05-12
</td>
<td>
updated revisions
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/e513b9973e91772c074941bf5bcfb1a1eaf0380e/analysis/crispr-mm_2215_sgRNA-clonality.Rmd" target="_blank">e513b99</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-05-08
</td>
<td>
update clonality figures
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/2dda789fcb78485deaf0274d2ef9be581c62e90c/analysis/crispr-mm_2215_sgRNA-clonality.Rmd" target="_blank">2dda789</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-05-07
</td>
<td>
update clonality figures
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/382cde0b92a2a809b4848d543a398e0d89ffc454/analysis/crispr-mm_2215_sgRNA-clonality.Rmd" target="_blank">382cde0</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-05-07
</td>
<td>
update clonality figures
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/c2a5c1c6d68eaf53b7e3abc93f74bd519068b3c2/analysis/crispr-mm_2215_sgRNA-clonality.Rmd" target="_blank">c2a5c1c</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-05-06
</td>
<td>
update r1od
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/b6d9355f3511e9503b4f6da1b3b2571277a4a223/analysis/crispr-mm_2215_sgRNA-clonality.Rmd" target="_blank">b6d9355</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-05-01
</td>
<td>
mod r2oD
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/5fab19d765a869233e820f2aed94595a89d5fe9b/analysis/crispr-mm_2215_sgRNA-clonality.Rmd" target="_blank">5fab19d</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2024-05-28
</td>
<td>
clean code
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/1920712c9ede8023880e174fc87f3cd238c3d612/analysis/crispr-mm_2215_sgRNA-clonality.Rmd" target="_blank">1920712</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2024-05-24
</td>
<td>
added F5
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/628f8ec8600fd6671cb7c7f57c8e717199ff7f75/analysis/crispr-mm_2215_sgRNA-clonality.Rmd" target="_blank">628f8ec</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2024-05-24
</td>
<td>
added F4
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/abf695c6a645a3aa42ea9e2fb622b5f22e723b11/analysis/crispr-mm_2215_sgRNA-clonality.Rmd" target="_blank">abf695c</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2024-05-23
</td>
<td>
done f2 anf f3
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries-additional-functions-and-data"
class="section level2">
<h2>Load libraries, additional functions and data</h2>
<p>Setup environment</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">results=</span><span class="st">&#39;asis&#39;</span>, <span class="at">echo=</span><span class="cn">TRUE</span>, <span class="at">message=</span><span class="cn">FALSE</span>, <span class="at">warning=</span><span class="cn">FALSE</span>, <span class="at">error=</span><span class="cn">FALSE</span>, <span class="at">fig.align =</span> <span class="st">&#39;center&#39;</span>, <span class="at">fig.width =</span> <span class="fl">3.5</span>, <span class="at">fig.asp =</span> <span class="fl">0.618</span>, <span class="at">dpi =</span> <span class="dv">600</span>, <span class="at">dev =</span> <span class="fu">c</span>(<span class="st">&quot;png&quot;</span>, <span class="st">&quot;pdf&quot;</span>), <span class="at">fig.showtext =</span> <span class="cn">FALSE</span>, <span class="at">engine.opts =</span> <span class="fu">list</span>(<span class="at">bash =</span> <span class="st">&quot;-l&quot;</span>))</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">options</span>(<span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>use_seed <span class="ot">&lt;-</span> <span class="dv">1100101</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="fu">set.seed</span>(use_seed)</span></code></pre></div>
<p>Load packages</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="fu">library</span>(foreach)</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="fu">library</span>(magrittr)</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="fu">library</span>(DT)</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="fu">library</span>(diptest)</span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a><span class="fu">library</span>(ggridges)</span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a><span class="fu">library</span>(ggh4x)</span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a><span class="fu">library</span>(colorblindr)</span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a><span class="fu">library</span>(ggbeeswarm)</span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span></code></pre></div>
<p>Load ggplot theme</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;./configuration/rmarkdown/ggplot_theme.R&quot;</span>)</span></code></pre></div>
<p>Load ggplot theme</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;./configuration/rmarkdown/color_palettes.R&quot;</span>)</span></code></pre></div>
<p>Load Summarized Experiment object</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">file.path</span>(params<span class="sc">$</span>data_dir, <span class="st">&#39;se.rds&#39;</span>))</span></code></pre></div>
<p>Filter samples used for this analysis</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>use_cols <span class="ot">&lt;-</span> <span class="fu">colData</span>(se) <span class="sc">%&gt;%</span> </span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>  data.frame <span class="sc">%&gt;%</span> </span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>  <span class="fu">filter</span>(effect_type_short <span class="sc">==</span> <span class="st">&#39;noKO&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>  <span class="fu">filter</span>(vector_short <span class="sc">!=</span> <span class="st">&quot;Thy1_EF1alpha_dCas9&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a>  rownames</span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a></span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a>se <span class="ot">&lt;-</span> se[,use_cols]</span></code></pre></div>
<p>Data wrangling</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>chunk_colData <span class="ot">&lt;-</span> <span class="fu">colData</span>(se) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>    <span class="at">vector_short =</span> <span class="fu">ifelse</span>(vector_short <span class="sc">==</span> <span class="st">&#39;Puro_EF1alpha_SpCas9&#39;</span>, </span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>                          <span class="st">&#39;Cas9+Puro&#39;</span>, vector_short),</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>    <span class="at">vector_short =</span> <span class="fu">ifelse</span>(vector_short <span class="sc">==</span> <span class="st">&#39;Puro+ dCas9&#39;</span>, </span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>                          <span class="st">&#39;Cas9+Puro&#39;</span>, vector_short),</span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>    <span class="at">vector_short =</span> <span class="fu">factor</span>(vector_short, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&#39;Thy1&#39;</span>, <span class="st">&#39;Puro&#39;</span>, <span class="st">&#39;Cas9+Puro&#39;</span>)),</span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a>    <span class="at">mouse_model_short =</span> <span class="fu">case_match</span>(mouse_model_global,</span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a>                                   <span class="st">&#39;immunodeficient&#39;</span> <span class="sc">~</span> <span class="st">&#39;NSG&#39;</span>,</span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a>                                   <span class="st">&#39;immunocompetent&#39;</span> <span class="sc">~</span> <span class="st">&#39;Syngeneic&#39;</span>,</span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a>                                   <span class="st">&#39;C57BL6&#39;</span> <span class="sc">~</span> <span class="st">&#39;Syngeneic&#39;</span></span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a>                                   ),</span>
<span id="cb7-13"><a href="#cb7-13" tabindex="-1"></a>    <span class="at">mouse_model_short =</span> <span class="fu">factor</span>(mouse_model_short, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&#39;NSG&#39;</span>, <span class="st">&#39;Syngeneic&#39;</span>)),</span>
<span id="cb7-14"><a href="#cb7-14" tabindex="-1"></a>    <span class="at">vector_mhost =</span> <span class="fu">paste0</span>(vector_short, <span class="st">&#39;, &#39;</span>, mouse_model_short),</span>
<span id="cb7-15"><a href="#cb7-15" tabindex="-1"></a>    <span class="at">model_vector_mhost =</span> <span class="fu">paste0</span>(donor,<span class="st">&#39;, &#39;</span>, vector_short, <span class="st">&#39;, &#39;</span>, mouse_model_short),</span>
<span id="cb7-16"><a href="#cb7-16" tabindex="-1"></a>    <span class="at">sample_type =</span> <span class="fu">ifelse</span>(cancer_type <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&#39;Breast&#39;</span>, <span class="st">&#39;CRC&#39;</span>),</span>
<span id="cb7-17"><a href="#cb7-17" tabindex="-1"></a>                         <span class="fu">case_match</span>(sample_type, </span>
<span id="cb7-18"><a href="#cb7-18" tabindex="-1"></a>                             <span class="st">&#39;whole_tumor&#39;</span> <span class="sc">~</span> <span class="st">&#39;Primary tumor&#39;</span>, </span>
<span id="cb7-19"><a href="#cb7-19" tabindex="-1"></a>                             <span class="st">&#39;whole_lung&#39;</span> <span class="sc">~</span> <span class="st">&#39;Lung mets&#39;</span>,</span>
<span id="cb7-20"><a href="#cb7-20" tabindex="-1"></a>                             <span class="st">&#39;bulk&#39;</span> <span class="sc">~</span> <span class="st">&#39;Bulk&#39;</span>,</span>
<span id="cb7-21"><a href="#cb7-21" tabindex="-1"></a>                             <span class="st">&#39;cell_culture&#39;</span> <span class="sc">~</span> <span class="st">&#39;Cell culture&#39;</span>),</span>
<span id="cb7-22"><a href="#cb7-22" tabindex="-1"></a>                         sample_type</span>
<span id="cb7-23"><a href="#cb7-23" tabindex="-1"></a>                         ),</span>
<span id="cb7-24"><a href="#cb7-24" tabindex="-1"></a>        <span class="at">sample_type =</span> <span class="fu">ifelse</span>(cancer_type <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&#39;NSCLC&#39;</span>),</span>
<span id="cb7-25"><a href="#cb7-25" tabindex="-1"></a>                         <span class="fu">case_match</span>(sample_type, </span>
<span id="cb7-26"><a href="#cb7-26" tabindex="-1"></a>                             <span class="st">&#39;met_liver&#39;</span> <span class="sc">~</span> <span class="st">&#39;Liver mets&#39;</span>, </span>
<span id="cb7-27"><a href="#cb7-27" tabindex="-1"></a>                             <span class="st">&#39;whole_lung&#39;</span> <span class="sc">~</span> <span class="st">&#39;Primary tumor&#39;</span>,</span>
<span id="cb7-28"><a href="#cb7-28" tabindex="-1"></a>                             <span class="st">&#39;bulk&#39;</span> <span class="sc">~</span> <span class="st">&#39;Bulk&#39;</span>,</span>
<span id="cb7-29"><a href="#cb7-29" tabindex="-1"></a>                             <span class="st">&#39;cell_culture&#39;</span> <span class="sc">~</span> <span class="st">&#39;Cell culture&#39;</span>),</span>
<span id="cb7-30"><a href="#cb7-30" tabindex="-1"></a>                         sample_type</span>
<span id="cb7-31"><a href="#cb7-31" tabindex="-1"></a>                         ),</span>
<span id="cb7-32"><a href="#cb7-32" tabindex="-1"></a>    </span>
<span id="cb7-33"><a href="#cb7-33" tabindex="-1"></a>    <span class="at">sample_type =</span> <span class="fu">factor</span>(sample_type, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&#39;Primary tumor&#39;</span>, <span class="st">&#39;Lung mets&#39;</span>, <span class="st">&#39;Liver mets&#39;</span>, <span class="st">&#39;Bulk&#39;</span>, <span class="st">&#39;Cell culture&#39;</span>))</span>
<span id="cb7-34"><a href="#cb7-34" tabindex="-1"></a>    </span>
<span id="cb7-35"><a href="#cb7-35" tabindex="-1"></a>  )</span>
<span id="cb7-36"><a href="#cb7-36" tabindex="-1"></a></span>
<span id="cb7-37"><a href="#cb7-37" tabindex="-1"></a><span class="fu">colData</span>(se) <span class="ot">&lt;-</span> chunk_colData <span class="sc">%&gt;%</span> DataFrame</span></code></pre></div>
</div>
<div id="configure-analyses" class="section level2">
<h2>Configure analyses</h2>
<p>Define comparisons for modality and Kolmogorov-Smirnov test</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">colData</span>(se) <span class="sc">%&gt;%</span> data.frame</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>group_list_modality <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>  <span class="st">`</span><span class="at">Syngeneic Thy1</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>    <span class="fu">filter</span>(vector_mmodel <span class="sc">==</span> <span class="st">&#39;4T1-noKO-Thy1-BALB&#39;</span> <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Primary tumor&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a>    <span class="fu">pull</span>(sample_alias),</span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a>  </span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a>  <span class="st">`</span><span class="at">Syngeneic Puro</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a>    <span class="fu">filter</span>(vector_mmodel <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&#39;4T1-noKO-Puro-BALB&#39;</span>, <span class="st">&#39;MVT1-noKO-Puro-FVB&#39;</span>, <span class="st">&#39;Py2T-noKO-Puro-FVB&#39;</span>) <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Primary tumor&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a>    <span class="fu">pull</span>(sample_alias),</span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a>  </span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a>  <span class="st">`</span><span class="at">Syngeneic dCas9 + Puro</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a>    <span class="fu">filter</span>(vector_mmodel <span class="sc">==</span> <span class="st">&#39;4T1-noKO-Puro_EF1alpha_dCas9-BALB&#39;</span> <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Primary tumor&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-14"><a href="#cb8-14" tabindex="-1"></a>    <span class="fu">pull</span>(sample_alias),</span>
<span id="cb8-15"><a href="#cb8-15" tabindex="-1"></a>  </span>
<span id="cb8-16"><a href="#cb8-16" tabindex="-1"></a>  <span class="st">`</span><span class="at">NSG Thy1</span><span class="st">`</span> <span class="ot">=</span>  x <span class="sc">%&gt;%</span> </span>
<span id="cb8-17"><a href="#cb8-17" tabindex="-1"></a>    <span class="fu">filter</span>(vector_mmodel <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&#39;4T1-noKO-Thy1-NSG&#39;</span>, <span class="st">&#39;MVT1-noKO-Thy1-NSG&#39;</span>, <span class="st">&#39;Py2T-noKO-Thy1-NSG&#39;</span>) <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Primary tumor&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb8-18"><a href="#cb8-18" tabindex="-1"></a>    <span class="fu">pull</span>(sample_alias),</span>
<span id="cb8-19"><a href="#cb8-19" tabindex="-1"></a>  </span>
<span id="cb8-20"><a href="#cb8-20" tabindex="-1"></a>  <span class="st">`</span><span class="at">NSG Puro</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb8-21"><a href="#cb8-21" tabindex="-1"></a>    <span class="fu">filter</span>(vector_mmodel <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&#39;4T1-noKO-Puro-NSG&#39;</span>, <span class="st">&#39;MVT1-noKO-Puro-NSG&#39;</span>, <span class="st">&#39;Py2T-noKO-Puro-NSG&#39;</span>) <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Primary tumor&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb8-22"><a href="#cb8-22" tabindex="-1"></a>    <span class="fu">pull</span>(sample_alias),</span>
<span id="cb8-23"><a href="#cb8-23" tabindex="-1"></a>  </span>
<span id="cb8-24"><a href="#cb8-24" tabindex="-1"></a>  <span class="st">`</span><span class="at">NSG dCas9 + Puro</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb8-25"><a href="#cb8-25" tabindex="-1"></a>    <span class="fu">filter</span>(vector_mmodel <span class="sc">==</span> <span class="st">&#39;4T1-noKO-Puro_EF1alpha_dCas9-NSG&#39;</span> <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Primary tumor&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-26"><a href="#cb8-26" tabindex="-1"></a>    <span class="fu">pull</span>(sample_alias)</span>
<span id="cb8-27"><a href="#cb8-27" tabindex="-1"></a>)</span>
<span id="cb8-28"><a href="#cb8-28" tabindex="-1"></a></span>
<span id="cb8-29"><a href="#cb8-29" tabindex="-1"></a>comp_list_modality <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb8-30"><a href="#cb8-30" tabindex="-1"></a>  <span class="st">`</span><span class="at">Syngeneic Thy1 vs. dCas9 + Puro</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb8-31"><a href="#cb8-31" tabindex="-1"></a>    <span class="st">`</span><span class="at">dCas9 + Puro</span><span class="st">`</span> <span class="ot">=</span> group_list_modality<span class="sc">$</span><span class="st">`</span><span class="at">Syngeneic dCas9 + Puro</span><span class="st">`</span>,</span>
<span id="cb8-32"><a href="#cb8-32" tabindex="-1"></a>    <span class="st">`</span><span class="at">Thy1</span><span class="st">`</span> <span class="ot">=</span> group_list_modality<span class="sc">$</span><span class="st">`</span><span class="at">Syngeneic Thy1</span><span class="st">`</span></span>
<span id="cb8-33"><a href="#cb8-33" tabindex="-1"></a>  ),</span>
<span id="cb8-34"><a href="#cb8-34" tabindex="-1"></a>  <span class="st">`</span><span class="at">Syngeneic Thy1 vs. Puro</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb8-35"><a href="#cb8-35" tabindex="-1"></a>    <span class="st">`</span><span class="at">Puro</span><span class="st">`</span> <span class="ot">=</span> group_list_modality<span class="sc">$</span><span class="st">`</span><span class="at">Syngeneic Puro</span><span class="st">`</span>,</span>
<span id="cb8-36"><a href="#cb8-36" tabindex="-1"></a>    <span class="st">`</span><span class="at">Thy1</span><span class="st">`</span> <span class="ot">=</span> group_list_modality<span class="sc">$</span><span class="st">`</span><span class="at">Syngeneic Thy1</span><span class="st">`</span></span>
<span id="cb8-37"><a href="#cb8-37" tabindex="-1"></a>  )</span>
<span id="cb8-38"><a href="#cb8-38" tabindex="-1"></a>)</span>
<span id="cb8-39"><a href="#cb8-39" tabindex="-1"></a></span>
<span id="cb8-40"><a href="#cb8-40" tabindex="-1"></a>group_df_modality <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i =</span> <span class="fu">names</span>(group_list_modality), <span class="at">.combine =</span> rbind) <span class="sc">%do%</span> {</span>
<span id="cb8-41"><a href="#cb8-41" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb8-42"><a href="#cb8-42" tabindex="-1"></a>    <span class="at">sample_alias =</span> group_list_modality[[i]],</span>
<span id="cb8-43"><a href="#cb8-43" tabindex="-1"></a>    <span class="at">modality_group =</span> i</span>
<span id="cb8-44"><a href="#cb8-44" tabindex="-1"></a>  )</span>
<span id="cb8-45"><a href="#cb8-45" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="sgrna-distribution-in-primary-tumor" class="section level2">
<h2>sgRNA distribution in Primary tumor</h2>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>use_sample_type <span class="ot">&lt;-</span> <span class="st">&#39;Primary tumor&#39;</span></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>use_colData <span class="ot">&lt;-</span> <span class="fu">colData</span>(se) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span></span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>  <span class="fu">filter</span>(sample_type <span class="sc">==</span> use_sample_type)</span></code></pre></div>
<div
id="figure-2b-sgrna-distribution-in-mice-transplanted-with-4t1-cells"
class="section level3">
<h3>Figure 2B: sgRNA distribution in mice transplanted with 4T1
cells</h3>
<p>Density plots showing unique sgRNAs in 4T1 tumors (n=5); frequency =
read counts per million (cpm+1).</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>selected_panels <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>  <span class="st">`</span><span class="at">Thy1, NSG</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;4T_TumoT_N4&#39;</span>, <span class="st">&#39;4T_TumoT_N5&#39;</span>, <span class="st">&#39;4T_TumoT_N2&#39;</span>),</span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>  <span class="st">`</span><span class="at">Thy1, Syngeneic</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;4T_TumoT_B1&#39;</span>, <span class="st">&#39;4T_TumoT_B3&#39;</span>, <span class="st">&#39;4T_TumoT_B4&#39;</span>),</span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a> </span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a>  <span class="st">`</span><span class="at">Puro, NSG</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;4T_TumoP_N5&#39;</span>, <span class="st">&#39;4T_TumoP_N4&#39;</span>, <span class="st">&#39;4T_TumoP_N2&#39;</span>),</span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a>  <span class="st">`</span><span class="at">Puro, Syngeneic</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;4T_TumoP_B3&#39;</span>, <span class="st">&#39;4T_TumoP_B2&#39;</span>, <span class="st">&#39;4T_TumoP_B1&#39;</span>),</span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a>  </span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a>  <span class="st">`</span><span class="at">dCas9+Puro, NSG</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;PlatePositionE1&#39;</span>, <span class="st">&#39;PlatePositionC1&#39;</span>, <span class="st">&#39;PlatePositionA2&#39;</span>),</span>
<span id="cb10-9"><a href="#cb10-9" tabindex="-1"></a>  <span class="st">`</span><span class="at">dCas9+Puro, Syngeneic</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;PlatePositionE2&#39;</span>, <span class="st">&#39;PlatePositionG2&#39;</span>, <span class="st">&#39;PlatePositionC2&#39;</span>)</span>
<span id="cb10-10"><a href="#cb10-10" tabindex="-1"></a>  </span>
<span id="cb10-11"><a href="#cb10-11" tabindex="-1"></a>)</span>
<span id="cb10-12"><a href="#cb10-12" tabindex="-1"></a></span>
<span id="cb10-13"><a href="#cb10-13" tabindex="-1"></a>selected_panels_df <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i =</span> <span class="fu">names</span>(selected_panels), <span class="at">.combine =</span> rbind) <span class="sc">%do%</span> {</span>
<span id="cb10-14"><a href="#cb10-14" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">group =</span> i, <span class="at">sample_alias =</span> selected_panels[[i]])</span>
<span id="cb10-15"><a href="#cb10-15" tabindex="-1"></a>}</span>
<span id="cb10-16"><a href="#cb10-16" tabindex="-1"></a></span>
<span id="cb10-17"><a href="#cb10-17" tabindex="-1"></a>use_samples <span class="ot">&lt;-</span> <span class="fu">unlist</span>(selected_panels)</span>
<span id="cb10-18"><a href="#cb10-18" tabindex="-1"></a>use_assay <span class="ot">&lt;-</span> <span class="fu">assay</span>(se[,use_samples], <span class="st">&#39;cpm&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-19"><a href="#cb10-19" tabindex="-1"></a>  <span class="fu">as.data.frame</span>(<span class="at">check.names =</span> F) <span class="sc">%&gt;%</span> </span>
<span id="cb10-20"><a href="#cb10-20" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&#39;guide&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-21"><a href="#cb10-21" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>guide, <span class="at">names_to =</span> <span class="st">&#39;sample_alias&#39;</span>, <span class="at">values_to =</span> <span class="st">&#39;cpm&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-22"><a href="#cb10-22" tabindex="-1"></a>  <span class="fu">left_join</span>(use_colData) <span class="sc">%&gt;%</span> </span>
<span id="cb10-23"><a href="#cb10-23" tabindex="-1"></a>  <span class="fu">left_join</span>(selected_panels_df) <span class="sc">%&gt;%</span> </span>
<span id="cb10-24"><a href="#cb10-24" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb10-25"><a href="#cb10-25" tabindex="-1"></a>    <span class="at">sample_alias =</span> <span class="fu">factor</span>(sample_alias, <span class="at">levels =</span> use_samples),</span>
<span id="cb10-26"><a href="#cb10-26" tabindex="-1"></a>    <span class="at">group =</span> <span class="fu">factor</span>(group, <span class="at">levels =</span> <span class="fu">names</span>(selected_panels))</span>
<span id="cb10-27"><a href="#cb10-27" tabindex="-1"></a>  )</span>
<span id="cb10-28"><a href="#cb10-28" tabindex="-1"></a></span>
<span id="cb10-29"><a href="#cb10-29" tabindex="-1"></a>use_assay <span class="sc">%&gt;%</span> </span>
<span id="cb10-30"><a href="#cb10-30" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>cpm <span class="sc">+</span> <span class="dv">1</span>, <span class="at">y =</span> sample_alias, </span>
<span id="cb10-31"><a href="#cb10-31" tabindex="-1"></a>             <span class="at">fill =</span> vector_short, <span class="at">color =</span> vector_short, </span>
<span id="cb10-32"><a href="#cb10-32" tabindex="-1"></a>             <span class="at">height =</span> <span class="fu">after_stat</span>(density))) <span class="sc">+</span></span>
<span id="cb10-33"><a href="#cb10-33" tabindex="-1"></a>  <span class="fu">geom_density_ridges</span>(<span class="at">size =</span> one_pt<span class="sc">/</span><span class="dv">4</span>, </span>
<span id="cb10-34"><a href="#cb10-34" tabindex="-1"></a>                      <span class="at">alpha =</span> <span class="fl">0.8</span>, </span>
<span id="cb10-35"><a href="#cb10-35" tabindex="-1"></a>                      <span class="at">scale =</span> <span class="dv">4</span>,</span>
<span id="cb10-36"><a href="#cb10-36" tabindex="-1"></a>                      <span class="at">rel_min_height =</span> <span class="fl">0.001</span> <span class="co"># set the `rel_min_height` argument to remove tails,</span></span>
<span id="cb10-37"><a href="#cb10-37" tabindex="-1"></a>                      ) <span class="sc">+</span></span>
<span id="cb10-38"><a href="#cb10-38" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> palette_vector) <span class="sc">+</span></span>
<span id="cb10-39"><a href="#cb10-39" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> palette_vector_line) <span class="sc">+</span></span>
<span id="cb10-40"><a href="#cb10-40" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>))) <span class="sc">+</span></span>
<span id="cb10-41"><a href="#cb10-41" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="fl">0.01</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb10-42"><a href="#cb10-42" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&#39;&#39;</span>) <span class="sc">+</span></span>
<span id="cb10-43"><a href="#cb10-43" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill =</span> <span class="st">&#39;none&#39;</span>, <span class="at">color =</span> <span class="st">&#39;none&#39;</span>) <span class="sc">+</span></span>
<span id="cb10-44"><a href="#cb10-44" tabindex="-1"></a>  <span class="fu">facet_wrap2</span>(<span class="fu">vars</span>(group), <span class="at">ncol =</span> <span class="dv">2</span>,</span>
<span id="cb10-45"><a href="#cb10-45" tabindex="-1"></a>                     <span class="at">scales =</span> <span class="st">&#39;free_y&#39;</span>,</span>
<span id="cb10-46"><a href="#cb10-46" tabindex="-1"></a>                     <span class="at">axes =</span> <span class="st">&quot;x&quot;</span>) <span class="sc">+</span></span>
<span id="cb10-47"><a href="#cb10-47" tabindex="-1"></a>  <span class="fu">theme_ridges</span>(<span class="at">font_size =</span> <span class="dv">3</span>, <span class="at">grid =</span> F) <span class="sc">+</span></span>
<span id="cb10-48"><a href="#cb10-48" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb10-49"><a href="#cb10-49" tabindex="-1"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb10-50"><a href="#cb10-50" tabindex="-1"></a>    <span class="at">axis.line.x =</span> <span class="fu">element_line</span>(<span class="at">linewidth =</span> one_pt<span class="sc">/</span><span class="dv">4</span>, <span class="at">color =</span> <span class="st">&#39;black&#39;</span>),</span>
<span id="cb10-51"><a href="#cb10-51" tabindex="-1"></a>    <span class="at">axis.ticks.x =</span> <span class="fu">element_line</span>(<span class="at">linewidth =</span> one_pt<span class="sc">/</span><span class="dv">4</span>, <span class="at">color =</span> <span class="st">&#39;black&#39;</span>),</span>
<span id="cb10-52"><a href="#cb10-52" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb10-53"><a href="#cb10-53" tabindex="-1"></a>    <span class="at">strip.text.x =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">4</span>, <span class="at">hjust =</span> <span class="dv">0</span>)</span>
<span id="cb10-54"><a href="#cb10-54" tabindex="-1"></a>  )</span></code></pre></div>
<p><img src="figure/crispr-mm_2215_sgRNA-clonality.Rmd/test-dist-tumor-distribution-ridges_plot-4t1-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-test-dist-tumor-distribution-ridges_plot-4t1-1">
Past versions of test-dist-tumor-distribution-ridges_plot-4t1-1.png
</button>
</p>
<div id="fig-test-dist-tumor-distribution-ridges_plot-4t1-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/e4cd8b5b49c70c2dac6df9053e5582afcfcd89c5/docs/figure/crispr-mm_2215_sgRNA-clonality.Rmd/test-dist-tumor-distribution-ridges_plot-4t1-1.png" target="_blank">e4cd8b5</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-05-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="supplementary-figure-2b-sgrna-distribution-in-mice-transplanted-with-py2t-and-mvt1-cells"
class="section level3">
<h3>Supplementary Figure 2B: sgRNA distribution in mice transplanted
with Py2T and MVT1 cells</h3>
<p>Density plots showing the distribution of unique sgRNAs retrieved in
primary tumors from the indicated mice transplanted with Py2T and MVT1
cells, previously transduced as indicated; frequency=read counts per
million (cpm+1); each plot is representative of a group of n=5.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>selected_panels <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>  <span class="st">`</span><span class="at">Py2T, Thy1, NSG</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;PY_TumoT_N5&#39;</span>, <span class="st">&#39;PY_TumoT_N4&#39;</span>, <span class="st">&#39;PY_TumoT_N3&#39;</span>),</span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a>  <span class="st">`</span><span class="at">Py2T, Thy1, Syngeneic</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;PY_TumoT_F3&#39;</span>, <span class="st">&#39;PY_TumoT_F4&#39;</span>, <span class="st">&#39;PY_TumoT_F5&#39;</span>), </span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a>  </span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a>  <span class="st">`</span><span class="at">Py2T, Puro, NSG</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;PY_TumoP_N1&#39;</span>, <span class="st">&#39;PY_TumoP_N2&#39;</span>, <span class="st">&#39;PY_TumoP_N3&#39;</span>),</span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a>  <span class="st">`</span><span class="at">Py2T, Puro, Syngeneic</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;PY_TumoP_F1&#39;</span>, <span class="st">&#39;PY_TumoP_F5&#39;</span>, <span class="st">&#39;PY_TumoP_F2&#39;</span>),</span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a>  </span>
<span id="cb11-8"><a href="#cb11-8" tabindex="-1"></a>  <span class="st">`</span><span class="at">MVT1, Thy1, NSG</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;MVT_TumoT_N3&#39;</span>, <span class="st">&#39;MVT_TumoT_N4&#39;</span>, <span class="st">&#39;MVT_TumoT_N5&#39;</span>),</span>
<span id="cb11-9"><a href="#cb11-9" tabindex="-1"></a>  <span class="st">`</span><span class="at">MVT1, Thy1, Syngeneic</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;MVT_TumoT_F1&#39;</span>, <span class="st">&#39;MVT_TumoT_F4&#39;</span>, <span class="st">&#39;MVT_TumoT_F3&#39;</span>),</span>
<span id="cb11-10"><a href="#cb11-10" tabindex="-1"></a>  </span>
<span id="cb11-11"><a href="#cb11-11" tabindex="-1"></a>  <span class="st">`</span><span class="at">MVT1, Puro, NSG</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;MVT_TumoP_N3&#39;</span>, <span class="st">&#39;MVT_TumoP_N4&#39;</span>, <span class="st">&#39;MVT_TumoP_N5&#39;</span>),</span>
<span id="cb11-12"><a href="#cb11-12" tabindex="-1"></a>  <span class="st">`</span><span class="at">MVT1, Puro, Syngeneic</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;MVT_TumoP_F3&#39;</span>, <span class="st">&#39;MVT_TumoP_F4&#39;</span>, <span class="st">&#39;MVT_TumoP_F5&#39;</span>)</span>
<span id="cb11-13"><a href="#cb11-13" tabindex="-1"></a>)</span>
<span id="cb11-14"><a href="#cb11-14" tabindex="-1"></a></span>
<span id="cb11-15"><a href="#cb11-15" tabindex="-1"></a>selected_panels_df <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i =</span> <span class="fu">names</span>(selected_panels), <span class="at">.combine =</span> rbind) <span class="sc">%do%</span> {</span>
<span id="cb11-16"><a href="#cb11-16" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">group =</span> i, <span class="at">sample_alias =</span> selected_panels[[i]])</span>
<span id="cb11-17"><a href="#cb11-17" tabindex="-1"></a>}</span>
<span id="cb11-18"><a href="#cb11-18" tabindex="-1"></a></span>
<span id="cb11-19"><a href="#cb11-19" tabindex="-1"></a>use_samples <span class="ot">&lt;-</span> <span class="fu">unlist</span>(selected_panels)</span>
<span id="cb11-20"><a href="#cb11-20" tabindex="-1"></a>use_assay <span class="ot">&lt;-</span> <span class="fu">assay</span>(se[,use_samples], <span class="st">&#39;cpm&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-21"><a href="#cb11-21" tabindex="-1"></a>  <span class="fu">as.data.frame</span>(<span class="at">check.names =</span> F) <span class="sc">%&gt;%</span> </span>
<span id="cb11-22"><a href="#cb11-22" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&#39;guide&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-23"><a href="#cb11-23" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>guide, <span class="at">names_to =</span> <span class="st">&#39;sample_alias&#39;</span>, <span class="at">values_to =</span> <span class="st">&#39;cpm&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-24"><a href="#cb11-24" tabindex="-1"></a>  <span class="fu">left_join</span>(use_colData) <span class="sc">%&gt;%</span> </span>
<span id="cb11-25"><a href="#cb11-25" tabindex="-1"></a>  <span class="fu">left_join</span>(selected_panels_df) <span class="sc">%&gt;%</span> </span>
<span id="cb11-26"><a href="#cb11-26" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb11-27"><a href="#cb11-27" tabindex="-1"></a>    <span class="at">sample_alias =</span> <span class="fu">factor</span>(sample_alias, <span class="at">levels =</span> use_samples),</span>
<span id="cb11-28"><a href="#cb11-28" tabindex="-1"></a>    <span class="at">group =</span> <span class="fu">factor</span>(group, <span class="at">levels =</span> <span class="fu">names</span>(selected_panels))</span>
<span id="cb11-29"><a href="#cb11-29" tabindex="-1"></a>  )</span>
<span id="cb11-30"><a href="#cb11-30" tabindex="-1"></a></span>
<span id="cb11-31"><a href="#cb11-31" tabindex="-1"></a>use_assay <span class="sc">%&gt;%</span> </span>
<span id="cb11-32"><a href="#cb11-32" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>cpm <span class="sc">+</span> <span class="dv">1</span>, <span class="at">y =</span> sample_alias, </span>
<span id="cb11-33"><a href="#cb11-33" tabindex="-1"></a>             <span class="at">fill =</span> vector_short, </span>
<span id="cb11-34"><a href="#cb11-34" tabindex="-1"></a>             <span class="at">color =</span> vector_short, </span>
<span id="cb11-35"><a href="#cb11-35" tabindex="-1"></a>             <span class="at">height =</span> <span class="fu">after_stat</span>(density))) <span class="sc">+</span></span>
<span id="cb11-36"><a href="#cb11-36" tabindex="-1"></a>  <span class="fu">geom_density_ridges</span>(<span class="at">size =</span> one_pt<span class="sc">/</span><span class="dv">4</span>, </span>
<span id="cb11-37"><a href="#cb11-37" tabindex="-1"></a>                      <span class="at">alpha =</span> <span class="fl">0.8</span>, </span>
<span id="cb11-38"><a href="#cb11-38" tabindex="-1"></a>                      <span class="at">scale =</span> <span class="dv">4</span>,</span>
<span id="cb11-39"><a href="#cb11-39" tabindex="-1"></a>                      <span class="at">rel_min_height =</span> <span class="fl">0.001</span> <span class="co"># set the `rel_min_height` argument to remove tails,</span></span>
<span id="cb11-40"><a href="#cb11-40" tabindex="-1"></a>                      ) <span class="sc">+</span></span>
<span id="cb11-41"><a href="#cb11-41" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> palette_vector) <span class="sc">+</span></span>
<span id="cb11-42"><a href="#cb11-42" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> palette_vector_line) <span class="sc">+</span></span>
<span id="cb11-43"><a href="#cb11-43" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>))) <span class="sc">+</span></span>
<span id="cb11-44"><a href="#cb11-44" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="fl">0.01</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb11-45"><a href="#cb11-45" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&#39;&#39;</span>) <span class="sc">+</span></span>
<span id="cb11-46"><a href="#cb11-46" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill =</span> <span class="st">&#39;none&#39;</span>, <span class="at">color =</span> <span class="st">&#39;none&#39;</span>) <span class="sc">+</span></span>
<span id="cb11-47"><a href="#cb11-47" tabindex="-1"></a>  <span class="fu">facet_wrap2</span>(<span class="fu">vars</span>(group), <span class="at">ncol =</span> <span class="dv">2</span>,</span>
<span id="cb11-48"><a href="#cb11-48" tabindex="-1"></a>                     <span class="at">scales =</span> <span class="st">&#39;free_y&#39;</span>,</span>
<span id="cb11-49"><a href="#cb11-49" tabindex="-1"></a>                     <span class="at">axes =</span> <span class="st">&quot;x&quot;</span>) <span class="sc">+</span></span>
<span id="cb11-50"><a href="#cb11-50" tabindex="-1"></a>  <span class="fu">theme_ridges</span>(<span class="at">font_size =</span> <span class="dv">3</span>, <span class="at">grid =</span> F) <span class="sc">+</span></span>
<span id="cb11-51"><a href="#cb11-51" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb11-52"><a href="#cb11-52" tabindex="-1"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb11-53"><a href="#cb11-53" tabindex="-1"></a>    <span class="at">axis.line.x =</span> <span class="fu">element_line</span>(<span class="at">linewidth =</span> one_pt<span class="sc">/</span><span class="dv">4</span>, <span class="at">color =</span> <span class="st">&#39;black&#39;</span>),</span>
<span id="cb11-54"><a href="#cb11-54" tabindex="-1"></a>    <span class="at">axis.ticks.x =</span> <span class="fu">element_line</span>(<span class="at">linewidth =</span> one_pt<span class="sc">/</span><span class="dv">4</span>, <span class="at">color =</span> <span class="st">&#39;black&#39;</span>),</span>
<span id="cb11-55"><a href="#cb11-55" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb11-56"><a href="#cb11-56" tabindex="-1"></a>    <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span><span class="cn">NULL</span>,</span>
<span id="cb11-57"><a href="#cb11-57" tabindex="-1"></a>                                <span class="at">color =</span> <span class="st">&quot;black&quot;</span>,</span>
<span id="cb11-58"><a href="#cb11-58" tabindex="-1"></a>                                <span class="at">linewidth =</span> <span class="dv">2</span><span class="sc">*</span>one_pt)</span>
<span id="cb11-59"><a href="#cb11-59" tabindex="-1"></a>  )</span></code></pre></div>
<p><img src="figure/crispr-mm_2215_sgRNA-clonality.Rmd/test-dist-tumor-distribution-ridges_plot-py2t-mvt1-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-test-dist-tumor-distribution-ridges_plot-py2t-mvt1-1">
Past versions of
test-dist-tumor-distribution-ridges_plot-py2t-mvt1-1.png
</button>
</p>
<div id="fig-test-dist-tumor-distribution-ridges_plot-py2t-mvt1-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/e4cd8b5b49c70c2dac6df9053e5582afcfcd89c5/docs/figure/crispr-mm_2215_sgRNA-clonality.Rmd/test-dist-tumor-distribution-ridges_plot-py2t-mvt1-1.png" target="_blank">e4cd8b5</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-05-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="kolmogorov-smirnov-test-and-ecdf-plots" class="section level2">
<h2>Kolmogorov-Smirnov test and ECDF plots</h2>
<p>Run Kolmogorov-Smirnov test to compare distributions merged by group
and create Empirical Cumulative Density Function (ECDF) by
comparison.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a>i <span class="ot">&lt;-</span> <span class="fu">names</span>(comp_list_modality)[<span class="dv">1</span>]</span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a>ks_test_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a>ecdf_plots <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">names</span>(comp_list_modality)) {</span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a>  use_samples <span class="ot">&lt;-</span> comp_list_modality[[i]]</span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a>  use_assay_c <span class="ot">&lt;-</span> <span class="fu">assay</span>(se[,use_samples[[<span class="dv">1</span>]]], <span class="st">&#39;cpm&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">check.names =</span> F) <span class="sc">%&gt;%</span> </span>
<span id="cb12-8"><a href="#cb12-8" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>(<span class="st">&#39;guide&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb12-9"><a href="#cb12-9" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="sc">-</span>guide, <span class="at">values_to =</span> <span class="st">&#39;cpm&#39;</span>,<span class="at">names_to =</span> <span class="st">&#39;sample_alias&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb12-10"><a href="#cb12-10" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">group =</span> <span class="st">&#39;case&#39;</span>, <span class="at">condition =</span> <span class="fu">names</span>(use_samples)[<span class="dv">1</span>])</span>
<span id="cb12-11"><a href="#cb12-11" tabindex="-1"></a>  use_assay_r <span class="ot">&lt;-</span> <span class="fu">assay</span>(se[,use_samples[[<span class="dv">2</span>]]], <span class="st">&#39;cpm&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb12-12"><a href="#cb12-12" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">check.names =</span> F) <span class="sc">%&gt;%</span> </span>
<span id="cb12-13"><a href="#cb12-13" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>(<span class="st">&#39;guide&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb12-14"><a href="#cb12-14" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="sc">-</span>guide, <span class="at">values_to =</span> <span class="st">&#39;cpm&#39;</span>,<span class="at">names_to =</span> <span class="st">&#39;sample_alias&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb12-15"><a href="#cb12-15" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">group =</span> <span class="st">&#39;referene&#39;</span>, <span class="at">condition =</span> <span class="fu">names</span>(use_samples)[<span class="dv">2</span>])</span>
<span id="cb12-16"><a href="#cb12-16" tabindex="-1"></a>  use_assay <span class="ot">&lt;-</span> <span class="fu">rbind</span>(use_assay_c, use_assay_r)</span>
<span id="cb12-17"><a href="#cb12-17" tabindex="-1"></a>  </span>
<span id="cb12-18"><a href="#cb12-18" tabindex="-1"></a>  <span class="co"># Perform the Kolmogorov-Smirnov test</span></span>
<span id="cb12-19"><a href="#cb12-19" tabindex="-1"></a>  ks.res <span class="ot">&lt;-</span> <span class="fu">ks.test</span>(use_assay_c<span class="sc">$</span>cpm, use_assay_r<span class="sc">$</span>cpm)</span>
<span id="cb12-20"><a href="#cb12-20" tabindex="-1"></a>  ks_test_df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(</span>
<span id="cb12-21"><a href="#cb12-21" tabindex="-1"></a>    ks_test_df,</span>
<span id="cb12-22"><a href="#cb12-22" tabindex="-1"></a>      <span class="fu">data.frame</span>(</span>
<span id="cb12-23"><a href="#cb12-23" tabindex="-1"></a>        <span class="at">comparison =</span> i,</span>
<span id="cb12-24"><a href="#cb12-24" tabindex="-1"></a>        <span class="at">case =</span> <span class="fu">names</span>(use_samples)[<span class="dv">1</span>],</span>
<span id="cb12-25"><a href="#cb12-25" tabindex="-1"></a>        <span class="at">reference =</span> <span class="fu">names</span>(use_samples)[<span class="dv">2</span>],</span>
<span id="cb12-26"><a href="#cb12-26" tabindex="-1"></a>        <span class="at">D =</span> ks.res<span class="sc">$</span>statistic,</span>
<span id="cb12-27"><a href="#cb12-27" tabindex="-1"></a>        <span class="at">P =</span> ks.res<span class="sc">$</span>p.value</span>
<span id="cb12-28"><a href="#cb12-28" tabindex="-1"></a>      )</span>
<span id="cb12-29"><a href="#cb12-29" tabindex="-1"></a>    )</span>
<span id="cb12-30"><a href="#cb12-30" tabindex="-1"></a></span>
<span id="cb12-31"><a href="#cb12-31" tabindex="-1"></a>  <span class="co"># Create ECDFs plot Empirical Cumulative Density Function</span></span>
<span id="cb12-32"><a href="#cb12-32" tabindex="-1"></a>  <span class="co"># use_colors &lt;- c(&#39;red&#39;, &#39;blue&#39;) %&gt;% set_names(names(use_samples))</span></span>
<span id="cb12-33"><a href="#cb12-33" tabindex="-1"></a>  ecdf_plots[[i]] <span class="ot">&lt;-</span> use_assay <span class="sc">%&gt;%</span> </span>
<span id="cb12-34"><a href="#cb12-34" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">log</span>(<span class="dv">1</span><span class="sc">+</span>cpm), <span class="at">color =</span> condition, <span class="at">group =</span> sample_alias)) <span class="sc">+</span></span>
<span id="cb12-35"><a href="#cb12-35" tabindex="-1"></a>    <span class="fu">stat_ecdf</span>(<span class="at">geom =</span> <span class="st">&#39;step&#39;</span>, <span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">linewidth =</span> one_pt<span class="sc">/</span><span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb12-36"><a href="#cb12-36" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> palette_vector) <span class="sc">+</span></span>
<span id="cb12-37"><a href="#cb12-37" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;log(1+cpm)&quot;</span>,</span>
<span id="cb12-38"><a href="#cb12-38" tabindex="-1"></a>         <span class="at">color =</span> <span class="st">&#39;&#39;</span>,</span>
<span id="cb12-39"><a href="#cb12-39" tabindex="-1"></a>         <span class="at">sub =</span> i</span>
<span id="cb12-40"><a href="#cb12-40" tabindex="-1"></a>         ) <span class="sc">+</span></span>
<span id="cb12-41"><a href="#cb12-41" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb12-42"><a href="#cb12-42" tabindex="-1"></a>      <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">2</span>),</span>
<span id="cb12-43"><a href="#cb12-43" tabindex="-1"></a>      <span class="at">axis.text.y =</span>  <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">3</span>),</span>
<span id="cb12-44"><a href="#cb12-44" tabindex="-1"></a>      <span class="at">axis.text.x =</span>  <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">3</span>),</span>
<span id="cb12-45"><a href="#cb12-45" tabindex="-1"></a>      <span class="at">legend.text =</span>  <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">2</span>),</span>
<span id="cb12-46"><a href="#cb12-46" tabindex="-1"></a>      <span class="at">legend.title =</span>  <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">2</span>),</span>
<span id="cb12-47"><a href="#cb12-47" tabindex="-1"></a>      <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">3</span>),</span>
<span id="cb12-48"><a href="#cb12-48" tabindex="-1"></a>      <span class="at">plot.caption =</span>  <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">2</span>)</span>
<span id="cb12-49"><a href="#cb12-49" tabindex="-1"></a>      </span>
<span id="cb12-50"><a href="#cb12-50" tabindex="-1"></a>    )</span>
<span id="cb12-51"><a href="#cb12-51" tabindex="-1"></a></span>
<span id="cb12-52"><a href="#cb12-52" tabindex="-1"></a>}</span></code></pre></div>
<div id="supplementary-figure-2c-left-k-s-test-table"
class="section level3">
<h3>Supplementary Figure 2C left : K-S test table</h3>
<p>Table depicting the results of the Kolmogorov-Smirnov test for
comparing sgRNA distributions from the datasets indicated, encompassing
three mammary carcinoma models (MVT1, 4T1, Py2T). The datasets are
obtained from primary tumors shown in Figure 2B and Supplementary Figure
2B.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a>ks_test_df <span class="sc">%&gt;%</span> </span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>case, <span class="sc">-</span>reference) <span class="sc">%&gt;%</span> </span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">P =</span> <span class="fu">format.pval</span>(P)) <span class="sc">%&gt;%</span> </span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a>  <span class="fu">datatable</span>(., </span>
<span id="cb13-5"><a href="#cb13-5" tabindex="-1"></a>            <span class="at">rownames =</span> <span class="cn">FALSE</span>, </span>
<span id="cb13-6"><a href="#cb13-6" tabindex="-1"></a>            <span class="at">filter =</span> <span class="st">&#39;top&#39;</span>, </span>
<span id="cb13-7"><a href="#cb13-7" tabindex="-1"></a>            <span class="at">caption =</span> <span class="st">&#39;Results of Kolmogorov-Smirnov test&#39;</span>,</span>
<span id="cb13-8"><a href="#cb13-8" tabindex="-1"></a>            <span class="at">extensions =</span> <span class="st">&#39;Buttons&#39;</span>, </span>
<span id="cb13-9"><a href="#cb13-9" tabindex="-1"></a>            <span class="at">options =</span> <span class="fu">list</span>(</span>
<span id="cb13-10"><a href="#cb13-10" tabindex="-1"></a>              <span class="at">dom =</span> <span class="st">&#39;Blfrtip&#39;</span>,</span>
<span id="cb13-11"><a href="#cb13-11" tabindex="-1"></a>              <span class="at">buttons =</span> <span class="fu">c</span>(<span class="st">&#39;csv&#39;</span>, <span class="st">&#39;excel&#39;</span>),</span>
<span id="cb13-12"><a href="#cb13-12" tabindex="-1"></a>              <span class="at">columnDefs =</span> <span class="fu">list</span>(</span>
<span id="cb13-13"><a href="#cb13-13" tabindex="-1"></a>                <span class="fu">list</span>(<span class="at">width =</span> <span class="dv">120</span>, <span class="at">targets =</span> <span class="dv">1</span>)</span>
<span id="cb13-14"><a href="#cb13-14" tabindex="-1"></a>              )</span>
<span id="cb13-15"><a href="#cb13-15" tabindex="-1"></a>            )</span>
<span id="cb13-16"><a href="#cb13-16" tabindex="-1"></a>            ) <span class="sc">%&gt;%</span> </span>
<span id="cb13-17"><a href="#cb13-17" tabindex="-1"></a>    <span class="fu">formatRound</span>(<span class="at">columns =</span> <span class="fu">c</span>(<span class="st">&#39;D&#39;</span>), <span class="at">digits =</span> <span class="dv">3</span>)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-0e41329c1963262c826a" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-0e41329c1963262c826a">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.375289691497238\" data-max=\"0.732731376975173\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\" disabled=\"\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption>Results of Kolmogorov-Smirnov test<\/caption>","data":[["Syngeneic Thy1 vs. dCas9 + Puro","Syngeneic Thy1 vs. Puro"],[0.7327313769751728,0.3752896914972385],["&lt; 2.22e-16","&lt; 2.22e-16"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>comparison<\/th>\n      <th>D<\/th>\n      <th>P<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["csv","excel"],"columnDefs":[{"targets":1,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\", null);\n  }"},{"width":120,"targets":1},{"className":"dt-right","targets":1},{"name":"comparison","targets":0},{"name":"D","targets":1},{"name":"P","targets":2}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true}},"evals":["options.columnDefs.0.render"],"jsHooks":[]}</script>
</div>
<div
id="supplementary-figure-2c-right-empirical-cumulative-density-function-ecdf-plots"
class="section level3">
<h3>Supplementary Figure 2C right : Empirical Cumulative Density
Function (ECDF) plots</h3>
<p>Cumulative curves depicting the results of the Kolmogorov-Smirnov
test for comparing sgRNA distributions from the datasets indicated,
encompassing three mammary carcinoma models (MVT1, 4T1, Py2T). The
datasets are obtained from primary tumors shown in Figure 2B and
Supplementary Figure 2B.</p>
<p>Empirical Cumulative Density Function (ECDF) plot by comparison</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a>use_index <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Syngeneic Thy1 vs. Puro&quot;</span>, <span class="st">&quot;Syngeneic Thy1 vs. dCas9 + Puro&quot;</span>)</span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a><span class="fu">plot_grid</span>(<span class="at">plotlist =</span> ecdf_plots[use_index], <span class="at">ncol =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="figure/crispr-mm_2215_sgRNA-clonality.Rmd/modality-comparisons-ks-ecdf-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-modality-comparisons-ks-ecdf-1">
Past versions of modality-comparisons-ks-ecdf-1.png
</button>
</p>
<div id="fig-modality-comparisons-ks-ecdf-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/e4cd8b5b49c70c2dac6df9053e5582afcfcd89c5/docs/figure/crispr-mm_2215_sgRNA-clonality.Rmd/modality-comparisons-ks-ecdf-1.png" target="_blank">e4cd8b5</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-05-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div
id="figure-2c-representative-müller-plots-of-4t1-in-syngeneic-recipients"
class="section level2">
<h2>Figure 2C: Representative Müller plots of 4T1 in syngeneic
recipients</h2>
<p>Define selected comparisons for muller plots</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a>comp_list_muller_syngeneic_representative <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a> <span class="st">`</span><span class="at">Thy1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;4T_TumoT_B3&#39;</span>, <span class="st">&#39;4T_LungsT_B3&#39;</span>),</span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a> <span class="st">`</span><span class="at">Puro</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;4T_TumoP_B1&#39;</span>, <span class="st">&#39;4T_LungsP_B1&#39;</span>),</span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a> <span class="st">`</span><span class="at">dCas9 + Puro</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;PlatePositionA3&#39;</span>, <span class="st">&#39;PlatePositionB3&#39;</span>)</span>
<span id="cb15-5"><a href="#cb15-5" tabindex="-1"></a>)</span></code></pre></div>
<p>Generate muller plots</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a>cpm_threshold <span class="ot">&lt;-</span> <span class="dv">10000</span></span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a>comp_list <span class="ot">&lt;-</span> comp_list_muller_syngeneic_representative</span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a></span>
<span id="cb16-4"><a href="#cb16-4" tabindex="-1"></a>use_df <span class="ot">&lt;-</span> <span class="fu">assay</span>(se, <span class="st">&#39;cpm&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>(<span class="at">check.names =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-5"><a href="#cb16-5" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&#39;guide&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-6"><a href="#cb16-6" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb16-7"><a href="#cb16-7" tabindex="-1"></a>    <span class="at">gene =</span> <span class="fu">rowData</span>(se)<span class="sc">$</span>Gene,</span>
<span id="cb16-8"><a href="#cb16-8" tabindex="-1"></a>    <span class="at">gene =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&#39;Non_Target&#39;</span>, gene), <span class="st">&#39;Non_Target&#39;</span>, gene)</span>
<span id="cb16-9"><a href="#cb16-9" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb16-10"><a href="#cb16-10" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span><span class="fu">c</span>(guide, gene), <span class="at">names_to =</span> <span class="st">&#39;sample_alias&#39;</span>, <span class="at">values_to =</span> <span class="st">&#39;cpm&#39;</span>)</span>
<span id="cb16-11"><a href="#cb16-11" tabindex="-1"></a></span>
<span id="cb16-12"><a href="#cb16-12" tabindex="-1"></a>j <span class="ot">&lt;-</span>  <span class="fu">names</span>(comp_list)[<span class="dv">1</span>]</span>
<span id="cb16-13"><a href="#cb16-13" tabindex="-1"></a>muller_plots <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">j =</span> <span class="fu">names</span>(comp_list)) <span class="sc">%do%</span> {</span>
<span id="cb16-14"><a href="#cb16-14" tabindex="-1"></a>  i <span class="ot">&lt;-</span> comp_list[[j]]</span>
<span id="cb16-15"><a href="#cb16-15" tabindex="-1"></a>  x <span class="ot">&lt;-</span> use_df <span class="sc">%&gt;%</span> </span>
<span id="cb16-16"><a href="#cb16-16" tabindex="-1"></a>    <span class="fu">filter</span>(sample_alias <span class="sc">%in%</span> i) <span class="sc">%&gt;%</span> </span>
<span id="cb16-17"><a href="#cb16-17" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb16-18"><a href="#cb16-18" tabindex="-1"></a>      <span class="at">sample_type =</span> <span class="fu">ifelse</span>(sample_alias <span class="sc">==</span> i[<span class="dv">1</span>], <span class="st">&#39;Primary tumor&#39;</span>, <span class="st">&#39;Lungs&#39;</span>),</span>
<span id="cb16-19"><a href="#cb16-19" tabindex="-1"></a>      <span class="at">sample_type =</span> <span class="fu">factor</span>(sample_type, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&#39;Primary tumor&#39;</span>, <span class="st">&#39;Lungs&#39;</span>))</span>
<span id="cb16-20"><a href="#cb16-20" tabindex="-1"></a>    )</span>
<span id="cb16-21"><a href="#cb16-21" tabindex="-1"></a>  </span>
<span id="cb16-22"><a href="#cb16-22" tabindex="-1"></a>  <span class="co"># Select guides with cpm &gt; cpm_threshold</span></span>
<span id="cb16-23"><a href="#cb16-23" tabindex="-1"></a>  x_guides_keep <span class="ot">&lt;-</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb16-24"><a href="#cb16-24" tabindex="-1"></a>    <span class="fu">group_by</span>(sample_type, guide) <span class="sc">%&gt;%</span> </span>
<span id="cb16-25"><a href="#cb16-25" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">cpm =</span> <span class="fu">max</span>(cpm)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-26"><a href="#cb16-26" tabindex="-1"></a>    <span class="fu">filter</span>(cpm <span class="sc">&gt;</span> cpm_threshold) <span class="sc">%&gt;%</span> </span>
<span id="cb16-27"><a href="#cb16-27" tabindex="-1"></a>    <span class="fu">pull</span>(guide)</span>
<span id="cb16-28"><a href="#cb16-28" tabindex="-1"></a>  </span>
<span id="cb16-29"><a href="#cb16-29" tabindex="-1"></a>  <span class="co"># Group guides with cpm &lt; cpm_threshold into the low abundant category, and sum counts for low abundant</span></span>
<span id="cb16-30"><a href="#cb16-30" tabindex="-1"></a>  x <span class="sc">%&lt;&gt;%</span> </span>
<span id="cb16-31"><a href="#cb16-31" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb16-32"><a href="#cb16-32" tabindex="-1"></a>      <span class="at">guide =</span> <span class="fu">ifelse</span>(guide <span class="sc">%in%</span>  x_guides_keep, guide, <span class="st">&#39;low abundant&#39;</span>)</span>
<span id="cb16-33"><a href="#cb16-33" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb16-34"><a href="#cb16-34" tabindex="-1"></a>    <span class="fu">group_by</span>(sample_type, guide) <span class="sc">%&gt;%</span></span>
<span id="cb16-35"><a href="#cb16-35" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">cpm =</span> <span class="fu">sum</span>(cpm)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-36"><a href="#cb16-36" tabindex="-1"></a>    <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb16-37"><a href="#cb16-37" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb16-38"><a href="#cb16-38" tabindex="-1"></a>      <span class="at">guide =</span> <span class="fu">fct_reorder</span>(guide, cpm),</span>
<span id="cb16-39"><a href="#cb16-39" tabindex="-1"></a>      <span class="at">guide =</span> <span class="fu">relevel</span>(guide, <span class="at">ref =</span> <span class="st">&#39;low abundant&#39;</span>)</span>
<span id="cb16-40"><a href="#cb16-40" tabindex="-1"></a>    )</span>
<span id="cb16-41"><a href="#cb16-41" tabindex="-1"></a>  </span>
<span id="cb16-42"><a href="#cb16-42" tabindex="-1"></a>  use_cols <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb16-43"><a href="#cb16-43" tabindex="-1"></a>    <span class="st">&#39;low abundant&#39;</span> <span class="ot">=</span> <span class="st">&#39;grey80&#39;</span>,</span>
<span id="cb16-44"><a href="#cb16-44" tabindex="-1"></a>    <span class="fu">colorRampPalette</span>(<span class="fu">rev</span>(palette_OkabeIto))(<span class="fu">nlevels</span>(x<span class="sc">$</span>guide) <span class="sc">-</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="fu">set_names</span>(<span class="fu">levels</span>(x<span class="sc">$</span>guide)[<span class="sc">-</span><span class="dv">1</span>])</span>
<span id="cb16-45"><a href="#cb16-45" tabindex="-1"></a>  )</span>
<span id="cb16-46"><a href="#cb16-46" tabindex="-1"></a>  </span>
<span id="cb16-47"><a href="#cb16-47" tabindex="-1"></a>  x <span class="sc">%&gt;%</span> </span>
<span id="cb16-48"><a href="#cb16-48" tabindex="-1"></a>    <span class="fu">ggplot</span>( <span class="fu">aes</span>(<span class="at">x =</span> sample_type, <span class="at">y =</span> cpm<span class="sc">/</span><span class="dv">10000</span>, <span class="at">group =</span> guide, <span class="at">fill =</span> guide)) <span class="sc">+</span> </span>
<span id="cb16-49"><a href="#cb16-49" tabindex="-1"></a>    <span class="fu">geom_area</span>(<span class="at">colour =</span> <span class="fu">alpha</span>(<span class="st">&quot;white&quot;</span>, <span class="fl">0.1</span>), <span class="at">linewidth =</span> <span class="fl">0.08</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb16-50"><a href="#cb16-50" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> use_cols, <span class="at">guide =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb16-51"><a href="#cb16-51" tabindex="-1"></a>    <span class="fu">labs</span>(</span>
<span id="cb16-52"><a href="#cb16-52" tabindex="-1"></a>      <span class="at">x =</span> <span class="st">&#39;&#39;</span>,</span>
<span id="cb16-53"><a href="#cb16-53" tabindex="-1"></a>      <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">&quot;CPM x 10&quot;</span><span class="sc">^-</span><span class="dv">4</span>),</span>
<span id="cb16-54"><a href="#cb16-54" tabindex="-1"></a>      <span class="at">title =</span> j,</span>
<span id="cb16-55"><a href="#cb16-55" tabindex="-1"></a>      <span class="at">fill =</span> <span class="st">&#39;&#39;</span></span>
<span id="cb16-56"><a href="#cb16-56" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb16-57"><a href="#cb16-57" tabindex="-1"></a>    <span class="fu">scale_x_discrete</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb16-58"><a href="#cb16-58" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1000000</span><span class="sc">/</span><span class="dv">10000</span>)) <span class="sc">+</span></span>
<span id="cb16-59"><a href="#cb16-59" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">fill =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb16-60"><a href="#cb16-60" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb16-61"><a href="#cb16-61" tabindex="-1"></a>      <span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">&quot;grey90&quot;</span>),</span>
<span id="cb16-62"><a href="#cb16-62" tabindex="-1"></a>      <span class="co"># plot.margin = margin(0.5, 1.5, 0.5, 0, &quot;cm&quot;),</span></span>
<span id="cb16-63"><a href="#cb16-63" tabindex="-1"></a>      <span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="dv">0</span>, <span class="fl">0.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="st">&quot;cm&quot;</span>),</span>
<span id="cb16-64"><a href="#cb16-64" tabindex="-1"></a>      <span class="at">axis.text =</span>  <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">3</span>),</span>
<span id="cb16-65"><a href="#cb16-65" tabindex="-1"></a>      <span class="at">axis.title =</span>  <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">3</span>),</span>
<span id="cb16-66"><a href="#cb16-66" tabindex="-1"></a>      <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">3</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>)</span>
<span id="cb16-67"><a href="#cb16-67" tabindex="-1"></a>    )</span>
<span id="cb16-68"><a href="#cb16-68" tabindex="-1"></a></span>
<span id="cb16-69"><a href="#cb16-69" tabindex="-1"></a>}</span>
<span id="cb16-70"><a href="#cb16-70" tabindex="-1"></a><span class="fu">names</span>(muller_plots) <span class="ot">&lt;-</span> <span class="fu">names</span>(comp_list)</span>
<span id="cb16-71"><a href="#cb16-71" tabindex="-1"></a>muller_syngeneic_representative <span class="ot">&lt;-</span> muller_plots</span></code></pre></div>
<p>Muller plots comparing sgRNAs frequency in matched primary tumors
versus lung metastases from syngeneic 4T1 transplants (n=5;
representative shown).</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a><span class="fu">wrap_plots</span>(muller_plots, <span class="at">ncol =</span> <span class="dv">1</span>)</span></code></pre></div>
<p><img src="figure/crispr-mm_2215_sgRNA-clonality.Rmd/muller-syngeneic-representative-plots-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-muller-syngeneic-representative-plots-1">
Past versions of muller-syngeneic-representative-plots-1.png
</button>
</p>
<div id="fig-muller-syngeneic-representative-plots-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/e4cd8b5b49c70c2dac6df9053e5582afcfcd89c5/docs/figure/crispr-mm_2215_sgRNA-clonality.Rmd/muller-syngeneic-representative-plots-1.png" target="_blank">e4cd8b5</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-05-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="hartigans-dip-test" class="section level2">
<h2>Hartigans’ dip test</h2>
<p>Hartigans’ dip test for unimodality. If the p-value is less than
0.05, there’s enough evidence to claim that the data are not unimodal
(multimodal, at least bimodal).</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a>use_assay <span class="ot">&lt;-</span> <span class="fu">assay</span>(se, <span class="st">&#39;cpm&#39;</span>)</span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a>i <span class="ot">&lt;-</span> <span class="fu">colnames</span>(use_assay)[<span class="dv">1</span>]</span>
<span id="cb18-3"><a href="#cb18-3" tabindex="-1"></a>modality_test_df <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i=</span> <span class="fu">colnames</span>(use_assay), <span class="at">.combine =</span> rbind) <span class="sc">%do%</span>{</span>
<span id="cb18-4"><a href="#cb18-4" tabindex="-1"></a>  x <span class="ot">&lt;-</span> use_assay[,i]</span>
<span id="cb18-5"><a href="#cb18-5" tabindex="-1"></a>  dip.res <span class="ot">&lt;-</span> <span class="fu">dip.test</span>(x, <span class="at">simulate.p.value =</span> <span class="cn">FALSE</span>, <span class="at">B =</span> <span class="dv">2000</span>)</span>
<span id="cb18-6"><a href="#cb18-6" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb18-7"><a href="#cb18-7" tabindex="-1"></a>    <span class="at">sample_alias =</span> i,</span>
<span id="cb18-8"><a href="#cb18-8" tabindex="-1"></a>    <span class="at">dip.D =</span> dip.res<span class="sc">$</span>statistic,</span>
<span id="cb18-9"><a href="#cb18-9" tabindex="-1"></a>    <span class="at">dip.p =</span> dip.res<span class="sc">$</span>p.value,</span>
<span id="cb18-10"><a href="#cb18-10" tabindex="-1"></a>    <span class="at">dip.result =</span> <span class="fu">ifelse</span>(dip.res<span class="sc">$</span>p.value <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="st">&#39;multimodal&#39;</span>, <span class="st">&#39;unimodal&#39;</span>)</span>
<span id="cb18-11"><a href="#cb18-11" tabindex="-1"></a>    )</span>
<span id="cb18-12"><a href="#cb18-12" tabindex="-1"></a>}</span>
<span id="cb18-13"><a href="#cb18-13" tabindex="-1"></a></span>
<span id="cb18-14"><a href="#cb18-14" tabindex="-1"></a>modality_test_df <span class="sc">%&lt;&gt;%</span> </span>
<span id="cb18-15"><a href="#cb18-15" tabindex="-1"></a>  <span class="fu">left_join</span>(group_df_modality) <span class="sc">%&gt;%</span> </span>
<span id="cb18-16"><a href="#cb18-16" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">colData</span>(se) <span class="sc">%&gt;%</span> data.frame)</span></code></pre></div>
<div id="dip-test-table-for-syngeneic-models" class="section level3">
<h3>Dip test table for Syngeneic models</h3>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a>modality_test_df <span class="sc">%&gt;%</span> </span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(modality_group)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-3"><a href="#cb19-3" tabindex="-1"></a>  <span class="fu">filter</span>(mouse_model_global <span class="sc">==</span> <span class="st">&#39;immunocompetent&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-4"><a href="#cb19-4" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb19-5"><a href="#cb19-5" tabindex="-1"></a>    <span class="at">modality_group =</span> <span class="fu">factor</span>(modality_group, <span class="at">levels =</span> <span class="fu">names</span>(group_list_modality))</span>
<span id="cb19-6"><a href="#cb19-6" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb19-7"><a href="#cb19-7" tabindex="-1"></a>  <span class="fu">group_by</span>(modality_group) <span class="sc">%&gt;%</span></span>
<span id="cb19-8"><a href="#cb19-8" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb19-9"><a href="#cb19-9" tabindex="-1"></a>    <span class="st">`</span><span class="at">Median D</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">median</span>(dip.D) <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="dv">3</span>),</span>
<span id="cb19-10"><a href="#cb19-10" tabindex="-1"></a>    <span class="at">n =</span> <span class="fu">n</span>(),</span>
<span id="cb19-11"><a href="#cb19-11" tabindex="-1"></a>    <span class="st">`</span><span class="at">Multimodal samples (n)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(dip.result <span class="sc">==</span> <span class="st">&#39;multimodal&#39;</span>),</span>
<span id="cb19-12"><a href="#cb19-12" tabindex="-1"></a>    <span class="st">`</span><span class="at">Unimodal samples (n)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(dip.result <span class="sc">==</span> <span class="st">&#39;unimodal&#39;</span>),</span>
<span id="cb19-13"><a href="#cb19-13" tabindex="-1"></a>    <span class="st">`</span><span class="at">Multimodal samples (%)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">round</span>(<span class="dv">100</span><span class="sc">*</span><span class="st">`</span><span class="at">Multimodal samples (n)</span><span class="st">`</span><span class="sc">/</span>n, <span class="dv">1</span>),</span>
<span id="cb19-14"><a href="#cb19-14" tabindex="-1"></a>    <span class="st">`</span><span class="at">Unimodal samples (%)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">round</span>(<span class="dv">100</span><span class="sc">*</span><span class="st">`</span><span class="at">Unimodal samples (n)</span><span class="st">`</span><span class="sc">/</span>n, <span class="dv">1</span>)</span>
<span id="cb19-15"><a href="#cb19-15" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> </span>
<span id="cb19-16"><a href="#cb19-16" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">check.names =</span> F) <span class="sc">%&gt;%</span> </span>
<span id="cb19-17"><a href="#cb19-17" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">caption =</span> <span class="st">&#39;Dip test results for Syngeneic models&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-18"><a href="#cb19-18" tabindex="-1"></a>  <span class="fu">kable_paper</span>(<span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">&quot;striped&quot;</span>, <span class="st">&quot;hover&quot;</span>, <span class="st">&quot;condensed&quot;</span>), <span class="at">full_width =</span> F)</span></code></pre></div>
<table class=" lightable-paper" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
Dip test results for Syngeneic models
</caption>
<thead>
<tr>
<th style="text-align:left;">
modality_group
</th>
<th style="text-align:right;">
Median D
</th>
<th style="text-align:right;">
n
</th>
<th style="text-align:right;">
Multimodal samples (n)
</th>
<th style="text-align:right;">
Unimodal samples (n)
</th>
<th style="text-align:right;">
Multimodal samples (%)
</th>
<th style="text-align:right;">
Unimodal samples (%)
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Syngeneic Thy1
</td>
<td style="text-align:right;">
0.005
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
20.0
</td>
<td style="text-align:right;">
80.0
</td>
</tr>
<tr>
<td style="text-align:left;">
Syngeneic Puro
</td>
<td style="text-align:right;">
0.054
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
93.3
</td>
<td style="text-align:right;">
6.7
</td>
</tr>
<tr>
<td style="text-align:left;">
Syngeneic dCas9 + Puro
</td>
<td style="text-align:right;">
0.077
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
100.0
</td>
<td style="text-align:right;">
0.0
</td>
</tr>
</tbody>
</table>
</div>
<div id="dip-test-table-for-nsg-models" class="section level3">
<h3>Dip test table for NSG models</h3>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a>modality_test_df <span class="sc">%&gt;%</span> </span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(modality_group)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-3"><a href="#cb20-3" tabindex="-1"></a>  <span class="fu">filter</span>(mouse_model_global <span class="sc">==</span> <span class="st">&#39;immunodeficient&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb20-4"><a href="#cb20-4" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb20-5"><a href="#cb20-5" tabindex="-1"></a>    <span class="at">modality_group =</span> <span class="fu">factor</span>(modality_group, <span class="at">levels =</span> <span class="fu">names</span>(group_list_modality))</span>
<span id="cb20-6"><a href="#cb20-6" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb20-7"><a href="#cb20-7" tabindex="-1"></a>  <span class="fu">group_by</span>(modality_group) <span class="sc">%&gt;%</span></span>
<span id="cb20-8"><a href="#cb20-8" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb20-9"><a href="#cb20-9" tabindex="-1"></a>    <span class="st">`</span><span class="at">Median D</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">median</span>(dip.D) <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="dv">3</span>),</span>
<span id="cb20-10"><a href="#cb20-10" tabindex="-1"></a>    <span class="at">n =</span> <span class="fu">n</span>(),</span>
<span id="cb20-11"><a href="#cb20-11" tabindex="-1"></a>    <span class="st">`</span><span class="at">Multimodal samples (n)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(dip.result <span class="sc">==</span> <span class="st">&#39;multimodal&#39;</span>),</span>
<span id="cb20-12"><a href="#cb20-12" tabindex="-1"></a>    <span class="st">`</span><span class="at">Unimodal samples (n)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(dip.result <span class="sc">==</span> <span class="st">&#39;unimodal&#39;</span>),</span>
<span id="cb20-13"><a href="#cb20-13" tabindex="-1"></a>    <span class="st">`</span><span class="at">Multimodal samples (%)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">round</span>(<span class="dv">100</span><span class="sc">*</span><span class="st">`</span><span class="at">Multimodal samples (n)</span><span class="st">`</span><span class="sc">/</span>n, <span class="dv">1</span>),</span>
<span id="cb20-14"><a href="#cb20-14" tabindex="-1"></a>    <span class="st">`</span><span class="at">Unimodal samples (%)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">round</span>(<span class="dv">100</span><span class="sc">*</span><span class="st">`</span><span class="at">Unimodal samples (n)</span><span class="st">`</span><span class="sc">/</span>n, <span class="dv">1</span>)</span>
<span id="cb20-15"><a href="#cb20-15" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> </span>
<span id="cb20-16"><a href="#cb20-16" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">check.names =</span> F) <span class="sc">%&gt;%</span> </span>
<span id="cb20-17"><a href="#cb20-17" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">caption =</span> <span class="st">&#39;Dip test results for NSG models&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb20-18"><a href="#cb20-18" tabindex="-1"></a>  <span class="fu">kable_paper</span>(<span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">&quot;striped&quot;</span>, <span class="st">&quot;hover&quot;</span>, <span class="st">&quot;condensed&quot;</span>), <span class="at">full_width =</span> F)</span></code></pre></div>
<table class=" lightable-paper" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
Dip test results for NSG models
</caption>
<thead>
<tr>
<th style="text-align:left;">
modality_group
</th>
<th style="text-align:right;">
Median D
</th>
<th style="text-align:right;">
n
</th>
<th style="text-align:right;">
Multimodal samples (n)
</th>
<th style="text-align:right;">
Unimodal samples (n)
</th>
<th style="text-align:right;">
Multimodal samples (%)
</th>
<th style="text-align:right;">
Unimodal samples (%)
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
NSG Thy1
</td>
<td style="text-align:right;">
0.005
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
13.3
</td>
<td style="text-align:right;">
86.7
</td>
</tr>
<tr>
<td style="text-align:left;">
NSG Puro
</td>
<td style="text-align:right;">
0.005
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
0.0
</td>
<td style="text-align:right;">
100.0
</td>
</tr>
<tr>
<td style="text-align:left;">
NSG dCas9 + Puro
</td>
<td style="text-align:right;">
0.007
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0.0
</td>
<td style="text-align:right;">
100.0
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="supplementary-figure-4c" class="section level2">
<h2>Supplementary Figure 4C</h2>
<p>Muller plots depicting the relative frequency of the predominant
sgRNAs in matched primary tumors (mammary fat pad) versus synchronous
metastasis samples (lungs) from syngeneic recipients transplanted with
the indicated cell model, which received StealTHY KO of the mouse
interactome library; n=5 each (representative samples are shown).</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">file.path</span>(params<span class="sc">$</span>data_dir, <span class="st">&#39;se.rds&#39;</span>))</span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a>comp_list <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb21-3"><a href="#cb21-3" tabindex="-1"></a> <span class="st">`</span><span class="at">Condition KO Thy1_all immunocompetent TUMORS--vs--Condition KO Thy1_all immunocompetent Lungs mets</span><span class="st">`</span><span class="ot">=</span><span class="fu">c</span>(<span class="st">&#39;CRISPRKO_Clone_14&#39;</span>, <span class="st">&#39;CRISPRKO_Clone_19&#39;</span>)</span>
<span id="cb21-4"><a href="#cb21-4" tabindex="-1"></a>)</span>
<span id="cb21-5"><a href="#cb21-5" tabindex="-1"></a></span>
<span id="cb21-6"><a href="#cb21-6" tabindex="-1"></a>cpm_threshold <span class="ot">&lt;-</span> <span class="dv">10000</span></span>
<span id="cb21-7"><a href="#cb21-7" tabindex="-1"></a></span>
<span id="cb21-8"><a href="#cb21-8" tabindex="-1"></a>use_df <span class="ot">&lt;-</span> <span class="fu">assay</span>(se, <span class="st">&#39;cpm&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>(<span class="at">check.names =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb21-9"><a href="#cb21-9" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&#39;guide&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb21-10"><a href="#cb21-10" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb21-11"><a href="#cb21-11" tabindex="-1"></a>    <span class="at">gene =</span> <span class="fu">rowData</span>(se)<span class="sc">$</span>Gene,</span>
<span id="cb21-12"><a href="#cb21-12" tabindex="-1"></a>    <span class="at">gene =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&#39;Non_Target&#39;</span>, gene), <span class="st">&#39;Non_Target&#39;</span>, gene)</span>
<span id="cb21-13"><a href="#cb21-13" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb21-14"><a href="#cb21-14" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span><span class="fu">c</span>(guide, gene), <span class="at">names_to =</span> <span class="st">&#39;sample_alias&#39;</span>, <span class="at">values_to =</span> <span class="st">&#39;cpm&#39;</span>)</span>
<span id="cb21-15"><a href="#cb21-15" tabindex="-1"></a></span>
<span id="cb21-16"><a href="#cb21-16" tabindex="-1"></a>j <span class="ot">&lt;-</span>  <span class="fu">names</span>(comp_list)[<span class="dv">1</span>]</span>
<span id="cb21-17"><a href="#cb21-17" tabindex="-1"></a>muller_plots <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">j =</span> <span class="fu">names</span>(comp_list)) <span class="sc">%do%</span> {</span>
<span id="cb21-18"><a href="#cb21-18" tabindex="-1"></a>  i <span class="ot">&lt;-</span> comp_list[[j]]</span>
<span id="cb21-19"><a href="#cb21-19" tabindex="-1"></a>  x <span class="ot">&lt;-</span> use_df <span class="sc">%&gt;%</span> </span>
<span id="cb21-20"><a href="#cb21-20" tabindex="-1"></a>    <span class="fu">filter</span>(sample_alias <span class="sc">%in%</span> i) <span class="sc">%&gt;%</span> </span>
<span id="cb21-21"><a href="#cb21-21" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb21-22"><a href="#cb21-22" tabindex="-1"></a>      <span class="at">sample_type =</span> <span class="fu">ifelse</span>(sample_alias <span class="sc">==</span> i[<span class="dv">1</span>], <span class="st">&#39;Primary tumor&#39;</span>, <span class="st">&#39;Lungs&#39;</span>),</span>
<span id="cb21-23"><a href="#cb21-23" tabindex="-1"></a>      <span class="at">sample_type =</span> <span class="fu">factor</span>(sample_type, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&#39;Primary tumor&#39;</span>, <span class="st">&#39;Lungs&#39;</span>))</span>
<span id="cb21-24"><a href="#cb21-24" tabindex="-1"></a>    )</span>
<span id="cb21-25"><a href="#cb21-25" tabindex="-1"></a>  </span>
<span id="cb21-26"><a href="#cb21-26" tabindex="-1"></a>  <span class="co"># Select guides with cpm &gt; cpm_threshold</span></span>
<span id="cb21-27"><a href="#cb21-27" tabindex="-1"></a>  x_guides_keep <span class="ot">&lt;-</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb21-28"><a href="#cb21-28" tabindex="-1"></a>    <span class="fu">group_by</span>(sample_type, guide) <span class="sc">%&gt;%</span> </span>
<span id="cb21-29"><a href="#cb21-29" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">cpm =</span> <span class="fu">max</span>(cpm)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-30"><a href="#cb21-30" tabindex="-1"></a>    <span class="fu">filter</span>(cpm <span class="sc">&gt;</span> cpm_threshold) <span class="sc">%&gt;%</span> </span>
<span id="cb21-31"><a href="#cb21-31" tabindex="-1"></a>    <span class="fu">pull</span>(guide)</span>
<span id="cb21-32"><a href="#cb21-32" tabindex="-1"></a>  </span>
<span id="cb21-33"><a href="#cb21-33" tabindex="-1"></a>  <span class="co"># Group guides with cpm &lt; cpm_threshold into the low abundant category, and sum counts for low abundant</span></span>
<span id="cb21-34"><a href="#cb21-34" tabindex="-1"></a>  x <span class="sc">%&lt;&gt;%</span> </span>
<span id="cb21-35"><a href="#cb21-35" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb21-36"><a href="#cb21-36" tabindex="-1"></a>      <span class="at">guide =</span> <span class="fu">ifelse</span>(guide <span class="sc">%in%</span>  x_guides_keep, guide, <span class="st">&#39;low abundant&#39;</span>)</span>
<span id="cb21-37"><a href="#cb21-37" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb21-38"><a href="#cb21-38" tabindex="-1"></a>    <span class="fu">group_by</span>(sample_type, guide) <span class="sc">%&gt;%</span></span>
<span id="cb21-39"><a href="#cb21-39" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">cpm =</span> <span class="fu">sum</span>(cpm)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-40"><a href="#cb21-40" tabindex="-1"></a>    <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb21-41"><a href="#cb21-41" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb21-42"><a href="#cb21-42" tabindex="-1"></a>      <span class="at">guide =</span> <span class="fu">fct_reorder</span>(guide, cpm),</span>
<span id="cb21-43"><a href="#cb21-43" tabindex="-1"></a>      <span class="at">guide =</span> <span class="fu">relevel</span>(guide, <span class="at">ref =</span> <span class="st">&#39;low abundant&#39;</span>)</span>
<span id="cb21-44"><a href="#cb21-44" tabindex="-1"></a>    )</span>
<span id="cb21-45"><a href="#cb21-45" tabindex="-1"></a>  </span>
<span id="cb21-46"><a href="#cb21-46" tabindex="-1"></a>  use_cols <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb21-47"><a href="#cb21-47" tabindex="-1"></a>    <span class="st">&#39;low abundant&#39;</span> <span class="ot">=</span> <span class="st">&#39;grey80&#39;</span>,</span>
<span id="cb21-48"><a href="#cb21-48" tabindex="-1"></a>    <span class="fu">colorRampPalette</span>(<span class="fu">rev</span>(palette_OkabeIto))(<span class="fu">nlevels</span>(x<span class="sc">$</span>guide) <span class="sc">-</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="fu">set_names</span>(<span class="fu">levels</span>(x<span class="sc">$</span>guide)[<span class="sc">-</span><span class="dv">1</span>])</span>
<span id="cb21-49"><a href="#cb21-49" tabindex="-1"></a>  )</span>
<span id="cb21-50"><a href="#cb21-50" tabindex="-1"></a>  </span>
<span id="cb21-51"><a href="#cb21-51" tabindex="-1"></a>  x <span class="sc">%&gt;%</span> </span>
<span id="cb21-52"><a href="#cb21-52" tabindex="-1"></a>    <span class="fu">ggplot</span>( <span class="fu">aes</span>(<span class="at">x =</span> sample_type, <span class="at">y =</span> cpm<span class="sc">/</span><span class="dv">10000</span>, <span class="at">group =</span> guide, <span class="at">fill =</span> guide)) <span class="sc">+</span> </span>
<span id="cb21-53"><a href="#cb21-53" tabindex="-1"></a>    <span class="fu">geom_area</span>(<span class="at">colour =</span> <span class="fu">alpha</span>(<span class="st">&quot;white&quot;</span>, <span class="fl">0.1</span>), <span class="at">linewidth =</span> <span class="fl">0.08</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb21-54"><a href="#cb21-54" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> use_cols, <span class="at">guide =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb21-55"><a href="#cb21-55" tabindex="-1"></a>    <span class="fu">labs</span>(</span>
<span id="cb21-56"><a href="#cb21-56" tabindex="-1"></a>      <span class="at">x =</span> <span class="st">&#39;&#39;</span>,</span>
<span id="cb21-57"><a href="#cb21-57" tabindex="-1"></a>      <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">&quot;CPM x 10&quot;</span><span class="sc">^-</span><span class="dv">4</span>),</span>
<span id="cb21-58"><a href="#cb21-58" tabindex="-1"></a>      <span class="at">title =</span> j,</span>
<span id="cb21-59"><a href="#cb21-59" tabindex="-1"></a>      <span class="at">fill =</span> <span class="st">&#39;&#39;</span></span>
<span id="cb21-60"><a href="#cb21-60" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb21-61"><a href="#cb21-61" tabindex="-1"></a>    <span class="fu">scale_x_discrete</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb21-62"><a href="#cb21-62" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1000000</span><span class="sc">/</span><span class="dv">10000</span>)) <span class="sc">+</span></span>
<span id="cb21-63"><a href="#cb21-63" tabindex="-1"></a>    <span class="co"># guides(fill = FALSE) +</span></span>
<span id="cb21-64"><a href="#cb21-64" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb21-65"><a href="#cb21-65" tabindex="-1"></a>        <span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">&quot;grey90&quot;</span>),</span>
<span id="cb21-66"><a href="#cb21-66" tabindex="-1"></a>        <span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="dv">0</span>, <span class="fl">0.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="st">&quot;cm&quot;</span>),</span>
<span id="cb21-67"><a href="#cb21-67" tabindex="-1"></a>        <span class="at">axis.text =</span>  <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">3</span>),</span>
<span id="cb21-68"><a href="#cb21-68" tabindex="-1"></a>        <span class="at">axis.title =</span>  <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">3</span>),</span>
<span id="cb21-69"><a href="#cb21-69" tabindex="-1"></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">3</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>),</span>
<span id="cb21-70"><a href="#cb21-70" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">3</span>),</span>
<span id="cb21-71"><a href="#cb21-71" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">3</span>),</span>
<span id="cb21-72"><a href="#cb21-72" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">&#39;bottom&#39;</span>,</span>
<span id="cb21-73"><a href="#cb21-73" tabindex="-1"></a>        <span class="at">legend.key.size =</span> <span class="fu">unit</span>(<span class="fl">0.1</span>, <span class="st">&#39;cm&#39;</span>),</span>
<span id="cb21-74"><a href="#cb21-74" tabindex="-1"></a>        <span class="at">legend.key.spacing.y =</span> <span class="fu">unit</span>(<span class="fl">0.1</span>, <span class="st">&#39;mm&#39;</span>),</span>
<span id="cb21-75"><a href="#cb21-75" tabindex="-1"></a>        <span class="at">legend.key.spacing.x =</span> <span class="fu">unit</span>(<span class="fl">0.1</span>, <span class="st">&#39;mm&#39;</span>)</span>
<span id="cb21-76"><a href="#cb21-76" tabindex="-1"></a>    )</span>
<span id="cb21-77"><a href="#cb21-77" tabindex="-1"></a></span>
<span id="cb21-78"><a href="#cb21-78" tabindex="-1"></a>}</span>
<span id="cb21-79"><a href="#cb21-79" tabindex="-1"></a></span>
<span id="cb21-80"><a href="#cb21-80" tabindex="-1"></a><span class="fu">print</span>(muller_plots)</span></code></pre></div>
<p>[[1]]
<img src="figure/crispr-mm_2215_sgRNA-clonality.Rmd/muller-sf4-c-1.png" style="display: block; margin: auto;" /></p>
</div>
<div id="supplementary-figure-2d-lower-panels" class="section level2">
<h2>Supplementary Figure 2D Lower Panels</h2>
<p>Muller plots depicting the relative frequency of unique sgRNAs in
matched primary tumors (mammary or lung orthotopic) versus synchronous
metastasis samples (lung metastasis for mammary tumors, liver metastasis
for lung tumors) from syngeneic and NSG recipients transplanted with the
indicated models previously transduced as indicated; n=5 per group
(representative samples are shown).</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">file.path</span>(params<span class="sc">$</span>data_dir, <span class="st">&#39;se.rds&#39;</span>))</span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a>comp_list <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb22-3"><a href="#cb22-3" tabindex="-1"></a>  <span class="st">`</span><span class="at">Condition Puro_all immunoDEFICIENT TUMORS--vs--Condition Puro_all immunoDEFICIENT Lungs mets</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb22-4"><a href="#cb22-4" tabindex="-1"></a>    <span class="at">MVT1 =</span> <span class="fu">c</span>(<span class="st">&#39;MVT_TumoP_N1&#39;</span>, <span class="st">&#39;MVT_LungsP_N1&#39;</span>), <span class="co">#muller-plots-67.png</span></span>
<span id="cb22-5"><a href="#cb22-5" tabindex="-1"></a>    <span class="at">Py2T =</span> <span class="fu">c</span>(<span class="st">&#39;PY_TumoP_N1&#39;</span>, <span class="st">&#39;PY_LungsP_N1&#39;</span>)<span class="co">#muller-plots-77.png</span></span>
<span id="cb22-6"><a href="#cb22-6" tabindex="-1"></a>    ),</span>
<span id="cb22-7"><a href="#cb22-7" tabindex="-1"></a>  <span class="st">`</span><span class="at">Condition Puro_all immunocompetent TUMORS--vs--Condition Puro_all immunocompetent Lungs mets</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb22-8"><a href="#cb22-8" tabindex="-1"></a>    <span class="at">MVT1 =</span> <span class="fu">c</span>(<span class="st">&#39;MVT_TumoP_F5&#39;</span>, <span class="st">&#39;MVT_LungsP_F5&#39;</span>), <span class="co">#muller-plots-19.png</span></span>
<span id="cb22-9"><a href="#cb22-9" tabindex="-1"></a>    <span class="at">Py2T =</span> <span class="fu">c</span>(<span class="st">&#39;PY_TumoP_F5&#39;</span>, <span class="st">&#39;PY_LungsP_F5&#39;</span>) <span class="co">#muller-plots-25.png</span></span>
<span id="cb22-10"><a href="#cb22-10" tabindex="-1"></a>    ),</span>
<span id="cb22-11"><a href="#cb22-11" tabindex="-1"></a>  <span class="st">`</span><span class="at">Condition Thy1_all immunoDEFICIENT TUMORS--vs--Condition Thy1_all immunoDEFICIENT Lungs mets</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb22-12"><a href="#cb22-12" tabindex="-1"></a>    <span class="at">MVT1 =</span> <span class="fu">c</span>(<span class="st">&#39;MVT_TumoT_N1&#39;</span>, <span class="st">&#39;MVT_LungsT_N1&#39;</span>), <span class="co">#muller-plots-127.png</span></span>
<span id="cb22-13"><a href="#cb22-13" tabindex="-1"></a>    <span class="at">Py2T =</span> <span class="fu">c</span>(<span class="st">&#39;PY_TumoT_N1&#39;</span>, <span class="st">&#39;PY_LungsT_N1&#39;</span>) <span class="co">#muller-plots-137.png</span></span>
<span id="cb22-14"><a href="#cb22-14" tabindex="-1"></a>  ),</span>
<span id="cb22-15"><a href="#cb22-15" tabindex="-1"></a>  <span class="st">`</span><span class="at">Condition Thy_all immunocompetent TUMORS--vs--Condition Thy_all immunocompetent Lungs mets</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb22-16"><a href="#cb22-16" tabindex="-1"></a>    <span class="at">MVT1 =</span> <span class="fu">c</span>(<span class="st">&#39;MVT_TumoT_F2&#39;</span>, <span class="st">&#39;MVT_LungsT_F2&#39;</span>), <span class="co">#muller-plots-39.png</span></span>
<span id="cb22-17"><a href="#cb22-17" tabindex="-1"></a>    <span class="at">Py2T =</span> <span class="fu">c</span>(<span class="st">&#39;PY_TumoT_F3&#39;</span>, <span class="st">&#39;PY_LungsT_F3&#39;</span>) <span class="co">#muller-plots-51.png</span></span>
<span id="cb22-18"><a href="#cb22-18" tabindex="-1"></a>  )</span>
<span id="cb22-19"><a href="#cb22-19" tabindex="-1"></a>  </span>
<span id="cb22-20"><a href="#cb22-20" tabindex="-1"></a>)</span>
<span id="cb22-21"><a href="#cb22-21" tabindex="-1"></a></span>
<span id="cb22-22"><a href="#cb22-22" tabindex="-1"></a>cpm_threshold <span class="ot">&lt;-</span> <span class="dv">10000</span></span>
<span id="cb22-23"><a href="#cb22-23" tabindex="-1"></a></span>
<span id="cb22-24"><a href="#cb22-24" tabindex="-1"></a></span>
<span id="cb22-25"><a href="#cb22-25" tabindex="-1"></a>use_df <span class="ot">&lt;-</span> <span class="fu">assay</span>(se, <span class="st">&#39;cpm&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>(<span class="at">check.names =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb22-26"><a href="#cb22-26" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&#39;guide&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb22-27"><a href="#cb22-27" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb22-28"><a href="#cb22-28" tabindex="-1"></a>    <span class="at">gene =</span> <span class="fu">rowData</span>(se)<span class="sc">$</span>Gene,</span>
<span id="cb22-29"><a href="#cb22-29" tabindex="-1"></a>    <span class="at">gene =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&#39;Non_Target&#39;</span>, gene), <span class="st">&#39;Non_Target&#39;</span>, gene)</span>
<span id="cb22-30"><a href="#cb22-30" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb22-31"><a href="#cb22-31" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span><span class="fu">c</span>(guide, gene), <span class="at">names_to =</span> <span class="st">&#39;sample_alias&#39;</span>, <span class="at">values_to =</span> <span class="st">&#39;cpm&#39;</span>)</span>
<span id="cb22-32"><a href="#cb22-32" tabindex="-1"></a></span>
<span id="cb22-33"><a href="#cb22-33" tabindex="-1"></a>i <span class="ot">&lt;-</span>  <span class="fu">names</span>(comp_list)[<span class="dv">1</span>]</span>
<span id="cb22-34"><a href="#cb22-34" tabindex="-1"></a>muller_plots <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i =</span> <span class="fu">names</span>(comp_list)) <span class="sc">%do%</span> {</span>
<span id="cb22-35"><a href="#cb22-35" tabindex="-1"></a>  j <span class="ot">&lt;-</span> comp_list[[i]][[<span class="dv">1</span>]]</span>
<span id="cb22-36"><a href="#cb22-36" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">j =</span> comp_list[[i]]) <span class="sc">%do%</span> {</span>
<span id="cb22-37"><a href="#cb22-37" tabindex="-1"></a>    x <span class="ot">&lt;-</span> use_df <span class="sc">%&gt;%</span> </span>
<span id="cb22-38"><a href="#cb22-38" tabindex="-1"></a>      <span class="fu">filter</span>(sample_alias <span class="sc">%in%</span> j) <span class="sc">%&gt;%</span> </span>
<span id="cb22-39"><a href="#cb22-39" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">sample_alias =</span> <span class="fu">factor</span>(sample_alias, j))</span>
<span id="cb22-40"><a href="#cb22-40" tabindex="-1"></a>    </span>
<span id="cb22-41"><a href="#cb22-41" tabindex="-1"></a>    <span class="co"># Select guides with cpm &gt; cpm_threshold</span></span>
<span id="cb22-42"><a href="#cb22-42" tabindex="-1"></a>    x_guides_keep <span class="ot">&lt;-</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb22-43"><a href="#cb22-43" tabindex="-1"></a>      <span class="fu">group_by</span>(sample_alias, guide) <span class="sc">%&gt;%</span> </span>
<span id="cb22-44"><a href="#cb22-44" tabindex="-1"></a>      <span class="fu">summarise</span>(<span class="at">cpm =</span> <span class="fu">max</span>(cpm)) <span class="sc">%&gt;%</span> </span>
<span id="cb22-45"><a href="#cb22-45" tabindex="-1"></a>      <span class="fu">filter</span>(cpm <span class="sc">&gt;</span> cpm_threshold) <span class="sc">%&gt;%</span> </span>
<span id="cb22-46"><a href="#cb22-46" tabindex="-1"></a>      <span class="fu">pull</span>(guide)</span>
<span id="cb22-47"><a href="#cb22-47" tabindex="-1"></a>    </span>
<span id="cb22-48"><a href="#cb22-48" tabindex="-1"></a>    <span class="co"># Group guides with cpm &lt; cpm_threshold into the low abundant category, and sum counts for low abundant</span></span>
<span id="cb22-49"><a href="#cb22-49" tabindex="-1"></a>    x <span class="sc">%&lt;&gt;%</span> </span>
<span id="cb22-50"><a href="#cb22-50" tabindex="-1"></a>      <span class="fu">mutate</span>(</span>
<span id="cb22-51"><a href="#cb22-51" tabindex="-1"></a>         <span class="at">guide =</span> <span class="fu">ifelse</span>(guide <span class="sc">%in%</span>  x_guides_keep, guide, <span class="st">&#39;low abundant&#39;</span>)</span>
<span id="cb22-52"><a href="#cb22-52" tabindex="-1"></a>      ) <span class="sc">%&gt;%</span> </span>
<span id="cb22-53"><a href="#cb22-53" tabindex="-1"></a>      <span class="fu">group_by</span>(sample_alias, guide) <span class="sc">%&gt;%</span></span>
<span id="cb22-54"><a href="#cb22-54" tabindex="-1"></a>      <span class="fu">summarise</span>(<span class="at">cpm =</span> <span class="fu">sum</span>(cpm)) <span class="sc">%&gt;%</span> </span>
<span id="cb22-55"><a href="#cb22-55" tabindex="-1"></a>      <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb22-56"><a href="#cb22-56" tabindex="-1"></a>      <span class="fu">mutate</span>(</span>
<span id="cb22-57"><a href="#cb22-57" tabindex="-1"></a>        <span class="at">guide =</span> <span class="fu">gsub</span>(<span class="st">&#39;_guide&#39;</span>, <span class="st">&#39; sg&#39;</span>, guide),</span>
<span id="cb22-58"><a href="#cb22-58" tabindex="-1"></a>        <span class="at">guide =</span> <span class="fu">fct_reorder</span>(guide, cpm),</span>
<span id="cb22-59"><a href="#cb22-59" tabindex="-1"></a>        <span class="at">guide =</span> <span class="fu">relevel</span>(guide, <span class="at">ref =</span> <span class="st">&#39;low abundant&#39;</span>)</span>
<span id="cb22-60"><a href="#cb22-60" tabindex="-1"></a>      )</span>
<span id="cb22-61"><a href="#cb22-61" tabindex="-1"></a>    </span>
<span id="cb22-62"><a href="#cb22-62" tabindex="-1"></a>    use_cols <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb22-63"><a href="#cb22-63" tabindex="-1"></a>      <span class="st">&#39;low abundant&#39;</span> <span class="ot">=</span> <span class="st">&#39;grey80&#39;</span>,</span>
<span id="cb22-64"><a href="#cb22-64" tabindex="-1"></a>      <span class="fu">colorRampPalette</span>(<span class="fu">rev</span>(palette_OkabeIto))(<span class="fu">nlevels</span>(x<span class="sc">$</span>guide) <span class="sc">-</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="fu">set_names</span>(<span class="fu">levels</span>(x<span class="sc">$</span>guide)[<span class="sc">-</span><span class="dv">1</span>])</span>
<span id="cb22-65"><a href="#cb22-65" tabindex="-1"></a>    )</span>
<span id="cb22-66"><a href="#cb22-66" tabindex="-1"></a>    </span>
<span id="cb22-67"><a href="#cb22-67" tabindex="-1"></a>    x <span class="sc">%&gt;%</span> </span>
<span id="cb22-68"><a href="#cb22-68" tabindex="-1"></a>      <span class="fu">ggplot</span>( <span class="fu">aes</span>(<span class="at">x =</span> sample_alias, <span class="at">y =</span> cpm, <span class="at">group =</span> guide, <span class="at">fill =</span> guide)) <span class="sc">+</span> </span>
<span id="cb22-69"><a href="#cb22-69" tabindex="-1"></a>      <span class="fu">geom_area</span>(<span class="at">colour =</span> <span class="fu">alpha</span>(<span class="st">&quot;white&quot;</span>, <span class="fl">0.1</span>), <span class="at">linewidth =</span> <span class="fl">0.08</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb22-70"><a href="#cb22-70" tabindex="-1"></a>      <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> use_cols, <span class="at">guide =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb22-71"><a href="#cb22-71" tabindex="-1"></a>      <span class="fu">labs</span>(</span>
<span id="cb22-72"><a href="#cb22-72" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">&#39;&#39;</span>,</span>
<span id="cb22-73"><a href="#cb22-73" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">&#39;cpm&#39;</span>,</span>
<span id="cb22-74"><a href="#cb22-74" tabindex="-1"></a>        <span class="at">fill =</span> <span class="st">&#39;&#39;</span></span>
<span id="cb22-75"><a href="#cb22-75" tabindex="-1"></a>      ) <span class="sc">+</span></span>
<span id="cb22-76"><a href="#cb22-76" tabindex="-1"></a>      <span class="fu">theme</span>(</span>
<span id="cb22-77"><a href="#cb22-77" tabindex="-1"></a>        <span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">&quot;grey90&quot;</span>),</span>
<span id="cb22-78"><a href="#cb22-78" tabindex="-1"></a>        <span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="fl">0.5</span>, <span class="fl">1.5</span>, <span class="fl">0.5</span>, <span class="dv">0</span>, <span class="st">&quot;cm&quot;</span>)</span>
<span id="cb22-79"><a href="#cb22-79" tabindex="-1"></a>        ) <span class="sc">+</span></span>
<span id="cb22-80"><a href="#cb22-80" tabindex="-1"></a>      <span class="fu">scale_x_discrete</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb22-81"><a href="#cb22-81" tabindex="-1"></a>      <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1000000</span>))</span>
<span id="cb22-82"><a href="#cb22-82" tabindex="-1"></a>  }</span>
<span id="cb22-83"><a href="#cb22-83" tabindex="-1"></a>  <span class="fu">names</span>(res) <span class="ot">&lt;-</span> <span class="fu">names</span>(comp_list[[i]])</span>
<span id="cb22-84"><a href="#cb22-84" tabindex="-1"></a>  res</span>
<span id="cb22-85"><a href="#cb22-85" tabindex="-1"></a>}</span>
<span id="cb22-86"><a href="#cb22-86" tabindex="-1"></a><span class="fu">names</span>(muller_plots) <span class="ot">&lt;-</span> <span class="fu">names</span>(comp_list)</span></code></pre></div>
<div id="supplementary-figure-2d-left-lower" class="section level3">
<h3>Supplementary Figure 2D (left lower)</h3>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a>plot_list <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a>  muller_plots[[<span class="st">&#39;Condition Puro_all immunoDEFICIENT TUMORS--vs--Condition Puro_all immunoDEFICIENT Lungs mets&#39;</span>]][[<span class="st">&#39;MVT1&#39;</span>]] <span class="sc">+</span> <span class="fu">guides</span>(<span class="at">fill =</span> <span class="st">&#39;none&#39;</span>),</span>
<span id="cb23-3"><a href="#cb23-3" tabindex="-1"></a>  muller_plots[[<span class="st">&#39;Condition Puro_all immunocompetent TUMORS--vs--Condition Puro_all immunocompetent Lungs mets&#39;</span>]][[<span class="st">&#39;MVT1&#39;</span>]]<span class="sc">+</span>  <span class="fu">guides</span>(<span class="at">fill =</span> <span class="st">&#39;none&#39;</span>),</span>
<span id="cb23-4"><a href="#cb23-4" tabindex="-1"></a>  muller_plots[[<span class="st">&#39;Condition Puro_all immunoDEFICIENT TUMORS--vs--Condition Puro_all immunoDEFICIENT Lungs mets&#39;</span>]][[<span class="st">&#39;Py2T&#39;</span>]] <span class="sc">+</span> <span class="fu">guides</span>(<span class="at">fill =</span> <span class="st">&#39;none&#39;</span>),</span>
<span id="cb23-5"><a href="#cb23-5" tabindex="-1"></a>  muller_plots[[<span class="st">&#39;Condition Puro_all immunocompetent TUMORS--vs--Condition Puro_all immunocompetent Lungs mets&#39;</span>]][[<span class="st">&#39;Py2T&#39;</span>]]<span class="sc">+</span>  <span class="fu">guides</span>(<span class="at">fill =</span> <span class="st">&#39;none&#39;</span>)</span>
<span id="cb23-6"><a href="#cb23-6" tabindex="-1"></a>)</span>
<span id="cb23-7"><a href="#cb23-7" tabindex="-1"></a></span>
<span id="cb23-8"><a href="#cb23-8" tabindex="-1"></a><span class="fu">wrap_plots</span>(plot_list, <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">byrow =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> print</span></code></pre></div>
<p><img src="figure/crispr-mm_2215_sgRNA-clonality.Rmd/muller-sfig2-d-left-lower-1.png" style="display: block; margin: auto;" /></p>
</div>
<div id="supplementary-figure-2d-right-lower" class="section level3">
<h3>Supplementary Figure 2D (right lower)</h3>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a>plot_list <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb24-2"><a href="#cb24-2" tabindex="-1"></a>  muller_plots[[<span class="st">&#39;Condition Thy1_all immunoDEFICIENT TUMORS--vs--Condition Thy1_all immunoDEFICIENT Lungs mets&#39;</span>]][[<span class="st">&#39;MVT1&#39;</span>]] <span class="sc">+</span> <span class="fu">guides</span>(<span class="at">fill =</span> <span class="st">&#39;none&#39;</span>),</span>
<span id="cb24-3"><a href="#cb24-3" tabindex="-1"></a>  muller_plots[[<span class="st">&#39;Condition Thy_all immunocompetent TUMORS--vs--Condition Thy_all immunocompetent Lungs mets&#39;</span>]][[<span class="st">&#39;MVT1&#39;</span>]]<span class="sc">+</span>  <span class="fu">guides</span>(<span class="at">fill =</span> <span class="st">&#39;none&#39;</span>),</span>
<span id="cb24-4"><a href="#cb24-4" tabindex="-1"></a>  muller_plots[[<span class="st">&#39;Condition Thy1_all immunoDEFICIENT TUMORS--vs--Condition Thy1_all immunoDEFICIENT Lungs mets&#39;</span>]][[<span class="st">&#39;Py2T&#39;</span>]] <span class="sc">+</span> <span class="fu">guides</span>(<span class="at">fill =</span> <span class="st">&#39;none&#39;</span>),</span>
<span id="cb24-5"><a href="#cb24-5" tabindex="-1"></a>  muller_plots[[<span class="st">&#39;Condition Thy_all immunocompetent TUMORS--vs--Condition Thy_all immunocompetent Lungs mets&#39;</span>]][[<span class="st">&#39;Py2T&#39;</span>]]<span class="sc">+</span>  <span class="fu">guides</span>(<span class="at">fill =</span> <span class="st">&#39;none&#39;</span>)</span>
<span id="cb24-6"><a href="#cb24-6" tabindex="-1"></a>)</span>
<span id="cb24-7"><a href="#cb24-7" tabindex="-1"></a></span>
<span id="cb24-8"><a href="#cb24-8" tabindex="-1"></a><span class="fu">wrap_plots</span>(plot_list, <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">byrow =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> print</span></code></pre></div>
<p><img src="figure/crispr-mm_2215_sgRNA-clonality.Rmd/muller-sfig2-d-right-lower-1.png" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="other-müller-plots" class="section level2">
<h2>Other Müller plots</h2>
<p>Define selected comparisons for muller plots</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">colData</span>(se) <span class="sc">%&gt;%</span> data.frame</span>
<span id="cb25-2"><a href="#cb25-2" tabindex="-1"></a></span>
<span id="cb25-3"><a href="#cb25-3" tabindex="-1"></a>comp_list_muller <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb25-4"><a href="#cb25-4" tabindex="-1"></a>  <span class="st">`</span><span class="at">Comparison 1: CMT167 Syngeneic Thy1.1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb25-5"><a href="#cb25-5" tabindex="-1"></a>    <span class="st">`</span><span class="at">CMT167 Syngeneic Thy1.1 1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;C167_StealTHY_Tum_1&#39;</span>,  <span class="st">&#39;C167_StealTHY_Met_1&#39;</span>),</span>
<span id="cb25-6"><a href="#cb25-6" tabindex="-1"></a>    <span class="st">`</span><span class="at">CMT167 Syngeneic Thy1.1 3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;C167_StealTHY_Tum_3&#39;</span>,  <span class="st">&#39;C167_StealTHY_Met_3&#39;</span>),</span>
<span id="cb25-7"><a href="#cb25-7" tabindex="-1"></a>    <span class="st">`</span><span class="at">CMT167 Syngeneic Thy1.1 4</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;C167_StealTHY_Tum_4&#39;</span>,  <span class="st">&#39;C167_StealTHY_Met_4&#39;</span>),</span>
<span id="cb25-8"><a href="#cb25-8" tabindex="-1"></a>    <span class="st">`</span><span class="at">CMT167 Syngeneic Thy1.1 5</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;C167_StealTHY_Tum_5&#39;</span>,  <span class="st">&#39;C167_StealTHY_Met_5&#39;</span>)</span>
<span id="cb25-9"><a href="#cb25-9" tabindex="-1"></a>  ),</span>
<span id="cb25-10"><a href="#cb25-10" tabindex="-1"></a>  <span class="st">`</span><span class="at">Comparison 2: LLC1 Syngeneic Thy1.1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>( </span>
<span id="cb25-11"><a href="#cb25-11" tabindex="-1"></a>   <span class="st">`</span><span class="at">LLC1 Syngeneic Thy1.1 1</span><span class="st">`</span> <span class="ot">=</span>  <span class="fu">c</span>(<span class="st">&#39;LLC1_StealTHY_Tum_1&#39;</span>,    <span class="st">&#39;LLC1_StealTHY_Met_1&#39;</span>),</span>
<span id="cb25-12"><a href="#cb25-12" tabindex="-1"></a>   <span class="st">`</span><span class="at">LLC1 Syngeneic Thy1.1 2</span><span class="st">`</span> <span class="ot">=</span>  <span class="fu">c</span>(<span class="st">&#39;LLC1_StealTHY_Tum_2&#39;</span>,    <span class="st">&#39;LLC1_StealTHY_Met_2&#39;</span>),</span>
<span id="cb25-13"><a href="#cb25-13" tabindex="-1"></a>   <span class="st">`</span><span class="at">LLC1 Syngeneic Thy1.1 3</span><span class="st">`</span> <span class="ot">=</span>  <span class="fu">c</span>(<span class="st">&#39;LLC1_StealTHY_Tum_3&#39;</span>,    <span class="st">&#39;LLC1_StealTHY_Met_3&#39;</span>),</span>
<span id="cb25-14"><a href="#cb25-14" tabindex="-1"></a>   <span class="st">`</span><span class="at">LLC1 Syngeneic Thy1.1 4</span><span class="st">`</span> <span class="ot">=</span>  <span class="fu">c</span>(<span class="st">&#39;LLC1_StealTHY_Tum_4&#39;</span>,    <span class="st">&#39;LLC1_StealTHY_Met_4&#39;</span>),</span>
<span id="cb25-15"><a href="#cb25-15" tabindex="-1"></a>   <span class="st">`</span><span class="at">LLC1 Syngeneic Thy1.1 5</span><span class="st">`</span> <span class="ot">=</span>  <span class="fu">c</span>(<span class="st">&#39;LLC1_StealTHY_Tum_5&#39;</span>,    <span class="st">&#39;LLC1_StealTHY_Met_5&#39;</span>)</span>
<span id="cb25-16"><a href="#cb25-16" tabindex="-1"></a>  ),</span>
<span id="cb25-17"><a href="#cb25-17" tabindex="-1"></a>  <span class="st">`</span><span class="at">Comparison 3: CT26 Syngeneic Thy1.1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>( </span>
<span id="cb25-18"><a href="#cb25-18" tabindex="-1"></a>    <span class="st">`</span><span class="at">CT26 Syngeneic Thy1.1 1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;ColoCT26_StealTHY_Tum_1&#39;</span>,    <span class="st">&#39;ColoCT26_StealTHY_Met_1&#39;</span>),</span>
<span id="cb25-19"><a href="#cb25-19" tabindex="-1"></a>    <span class="st">`</span><span class="at">CT26 Syngeneic Thy1.1 2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;ColoCT26_StealTHY_Tum_2&#39;</span>,    <span class="st">&#39;ColoCT26_StealTHY_Met_2&#39;</span>),</span>
<span id="cb25-20"><a href="#cb25-20" tabindex="-1"></a>    <span class="st">`</span><span class="at">CT26 Syngeneic Thy1.1 3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;ColoCT26_StealTHY_Tum_3&#39;</span>,    <span class="st">&#39;Replacement_1&#39;</span>),</span>
<span id="cb25-21"><a href="#cb25-21" tabindex="-1"></a>    <span class="st">`</span><span class="at">CT26 Syngeneic Thy1.1 4</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;Replacement_2&#39;</span>,  <span class="st">&#39;Replacement_3&#39;</span>),</span>
<span id="cb25-22"><a href="#cb25-22" tabindex="-1"></a>    <span class="st">`</span><span class="at">CT26 Syngeneic Thy1.1 5</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;Replacement_4&#39;</span>,  <span class="st">&#39;Replacement_4bis&#39;</span>)</span>
<span id="cb25-23"><a href="#cb25-23" tabindex="-1"></a>  ),</span>
<span id="cb25-24"><a href="#cb25-24" tabindex="-1"></a>  <span class="st">`</span><span class="at">Comparison 4: CMT167 Syngeneic dCas9 + Puro</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb25-25"><a href="#cb25-25" tabindex="-1"></a>    <span class="st">`</span><span class="at">CMT167 Syngeneic dCas9 + Puro 3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;dC9BL6_LungIV_C167_3&#39;</span>,   <span class="st">&#39;dC9BL6_LiverIV_C167_3&#39;</span>),</span>
<span id="cb25-26"><a href="#cb25-26" tabindex="-1"></a>    <span class="st">`</span><span class="at">CMT167 Syngeneic dCas9 + Puro 5</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;dC9BL6_LungIV_C167_5&#39;</span>,   <span class="st">&#39;dC9BL6_LiverIV_C167_5&#39;</span>)</span>
<span id="cb25-27"><a href="#cb25-27" tabindex="-1"></a>  ),</span>
<span id="cb25-28"><a href="#cb25-28" tabindex="-1"></a>  </span>
<span id="cb25-29"><a href="#cb25-29" tabindex="-1"></a>  <span class="st">`</span><span class="at">Comparison 5: LLC1 Syngeneic dCas9 + Puro</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb25-30"><a href="#cb25-30" tabindex="-1"></a>    <span class="st">`</span><span class="at">LLC1 Syngeneic dCas9 + Puro 1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;dC9BL6_LungIV_LLC1_1&#39;</span>, <span class="st">&#39;dC9BL6_LiverIV_LLC1_1&#39;</span>),</span>
<span id="cb25-31"><a href="#cb25-31" tabindex="-1"></a>    <span class="st">`</span><span class="at">LLC1 Syngeneic dCas9 + Puro 3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;dC9BL6_LungIV_LLC1_3&#39;</span>, <span class="st">&#39;dC9BL6_LiverIV_LLC1_3&#39;</span>),</span>
<span id="cb25-32"><a href="#cb25-32" tabindex="-1"></a>    <span class="st">`</span><span class="at">LLC1 Syngeneic dCas9 + Puro 4</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;dC9BL6_LungIV_LLC1_4&#39;</span>, <span class="st">&#39;dC9BL6_LiverIV_LLC1_4&#39;</span>),</span>
<span id="cb25-33"><a href="#cb25-33" tabindex="-1"></a>    <span class="st">`</span><span class="at">LLC1 Syngeneic dCas9 + Puro 5</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;dC9BL6_LungIV_LLC1_5&#39;</span>, <span class="st">&#39;dC9BL6_LiverIV_LLC1_5&#39;</span>)</span>
<span id="cb25-34"><a href="#cb25-34" tabindex="-1"></a>  ),</span>
<span id="cb25-35"><a href="#cb25-35" tabindex="-1"></a>  </span>
<span id="cb25-36"><a href="#cb25-36" tabindex="-1"></a>  <span class="st">`</span><span class="at">Comparison 6: CMT167 NSG dCas9 + Puro</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb25-37"><a href="#cb25-37" tabindex="-1"></a>    <span class="st">`</span><span class="at">CMT167 NSG dCas9 + Puro 1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;dC9NSG_LungIV_C167_1&#39;</span>, <span class="st">&#39;dC9NSG_LiverIV_C167_1&#39;</span>),</span>
<span id="cb25-38"><a href="#cb25-38" tabindex="-1"></a>    <span class="st">`</span><span class="at">CMT167 NSG dCas9 + Puro 2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;dC9NSG_LungIV_C167_2&#39;</span>, <span class="st">&#39;dC9NSG_LiverIV_C167_2&#39;</span>),</span>
<span id="cb25-39"><a href="#cb25-39" tabindex="-1"></a>    <span class="st">`</span><span class="at">CMT167 NSG dCas9 + Puro 3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;dC9NSG_LungIV_C167_3&#39;</span>, <span class="st">&#39;dC9NSG_LiverIV_C167_3&#39;</span>),</span>
<span id="cb25-40"><a href="#cb25-40" tabindex="-1"></a>    <span class="st">`</span><span class="at">CMT167 NSG dCas9 + Puro 4</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;dC9NSG_LungIV_C167_4&#39;</span>, <span class="st">&#39;dC9NSG_LiverIV_C167_4&#39;</span>),</span>
<span id="cb25-41"><a href="#cb25-41" tabindex="-1"></a>    <span class="st">`</span><span class="at">CMT167 NSG dCas9 + Puro 5</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;dC9NSG_LungIV_C167_5&#39;</span>, <span class="st">&#39;dC9NSG_LiverIV_C167_5&#39;</span>)</span>
<span id="cb25-42"><a href="#cb25-42" tabindex="-1"></a>  ),</span>
<span id="cb25-43"><a href="#cb25-43" tabindex="-1"></a>  </span>
<span id="cb25-44"><a href="#cb25-44" tabindex="-1"></a>  <span class="st">`</span><span class="at">Comparison 7: LLC1 NSG dCas9 + Puro</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb25-45"><a href="#cb25-45" tabindex="-1"></a>    <span class="st">`</span><span class="at">LLC1 NSG dCas9 + Puro 1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;dC9NSG_LungIV_LLC1_1&#39;</span>,   <span class="st">&#39;dC9NSG_LiverIV_LLC1_1&#39;</span>),</span>
<span id="cb25-46"><a href="#cb25-46" tabindex="-1"></a>    <span class="st">`</span><span class="at">LLC1 NSG dCas9 + Puro 2</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;dC9NSG_LungIV_LLC1_2&#39;</span>,   <span class="st">&#39;dC9NSG_LiverIV_LLC1_2&#39;</span>),</span>
<span id="cb25-47"><a href="#cb25-47" tabindex="-1"></a>    <span class="st">`</span><span class="at">LLC1 NSG dCas9 + Puro 3</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;dC9NSG_LungIV_LLC1_3&#39;</span>,   <span class="st">&#39;dC9NSG_LiverIV_LLC1_3&#39;</span>),</span>
<span id="cb25-48"><a href="#cb25-48" tabindex="-1"></a>    <span class="st">`</span><span class="at">LLC1 NSG dCas9 + Puro 4</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;dC9NSG_LungIV_LLC1_4&#39;</span>,   <span class="st">&#39;dC9NSG_LiverIV_LLC1_4&#39;</span>),</span>
<span id="cb25-49"><a href="#cb25-49" tabindex="-1"></a>    <span class="st">`</span><span class="at">LLC1 NSG dCas9 + Puro 5</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;dC9NSG_LungIV_LLC1_5&#39;</span>,   <span class="st">&#39;dC9NSG_LiverIV_LLC1_5&#39;</span>)</span>
<span id="cb25-50"><a href="#cb25-50" tabindex="-1"></a>  ),</span>
<span id="cb25-51"><a href="#cb25-51" tabindex="-1"></a>  <span class="st">`</span><span class="at">Comparison 10: Thy1.1 in BL6</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb25-52"><a href="#cb25-52" tabindex="-1"></a>    <span class="st">`</span><span class="at">CMT167 BL6 Thy1.1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;Replacement_9&#39;</span>,    <span class="st">&#39;Replacement_1bis&#39;</span>),</span>
<span id="cb25-53"><a href="#cb25-53" tabindex="-1"></a>    <span class="st">`</span><span class="at">LLC1 BL6 Thy1.1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;Replacement_5&#39;</span>,  <span class="st">&#39;Replacement_6&#39;</span>)</span>
<span id="cb25-54"><a href="#cb25-54" tabindex="-1"></a>  ),</span>
<span id="cb25-55"><a href="#cb25-55" tabindex="-1"></a>  <span class="st">`</span><span class="at">Comparison 11: Thy1.1 in NSG</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb25-56"><a href="#cb25-56" tabindex="-1"></a>    <span class="st">`</span><span class="at">CMT167 NSG Thy1.1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;Replacement_2bis&#39;</span>, <span class="st">&#39;Replacement_3bis&#39;</span>),</span>
<span id="cb25-57"><a href="#cb25-57" tabindex="-1"></a>    <span class="st">`</span><span class="at">LLC1 NSG Thy1.1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;Replacement_7&#39;</span>,  <span class="st">&#39;Replacement_8&#39;</span>)</span>
<span id="cb25-58"><a href="#cb25-58" tabindex="-1"></a>  )</span>
<span id="cb25-59"><a href="#cb25-59" tabindex="-1"></a>)</span></code></pre></div>
<p>Generate muller plots</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a>cpm_threshold <span class="ot">&lt;-</span> <span class="dv">10000</span></span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a>comp_list <span class="ot">&lt;-</span> comp_list_muller</span>
<span id="cb26-3"><a href="#cb26-3" tabindex="-1"></a></span>
<span id="cb26-4"><a href="#cb26-4" tabindex="-1"></a>use_df <span class="ot">&lt;-</span> <span class="fu">assay</span>(se, <span class="st">&#39;cpm&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>(<span class="at">check.names =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-5"><a href="#cb26-5" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&#39;guide&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-6"><a href="#cb26-6" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb26-7"><a href="#cb26-7" tabindex="-1"></a>    <span class="at">gene =</span> <span class="fu">rowData</span>(se)<span class="sc">$</span>Gene,</span>
<span id="cb26-8"><a href="#cb26-8" tabindex="-1"></a>    <span class="at">gene =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&#39;Non_Target&#39;</span>, gene), <span class="st">&#39;Non_Target&#39;</span>, gene)</span>
<span id="cb26-9"><a href="#cb26-9" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb26-10"><a href="#cb26-10" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span><span class="fu">c</span>(guide, gene), <span class="at">names_to =</span> <span class="st">&#39;sample_alias&#39;</span>, <span class="at">values_to =</span> <span class="st">&#39;cpm&#39;</span>)</span>
<span id="cb26-11"><a href="#cb26-11" tabindex="-1"></a></span>
<span id="cb26-12"><a href="#cb26-12" tabindex="-1"></a>j <span class="ot">&lt;-</span>  <span class="fu">names</span>(comp_list)[<span class="dv">1</span>]</span>
<span id="cb26-13"><a href="#cb26-13" tabindex="-1"></a>muller_plots <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">j =</span> <span class="fu">names</span>(comp_list)) <span class="sc">%do%</span> {</span>
<span id="cb26-14"><a href="#cb26-14" tabindex="-1"></a>  i <span class="ot">&lt;-</span> comp_list[[j]][[<span class="dv">1</span>]]</span>
<span id="cb26-15"><a href="#cb26-15" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">iname =</span> <span class="fu">names</span>(comp_list[[j]])) <span class="sc">%do%</span> {</span>
<span id="cb26-16"><a href="#cb26-16" tabindex="-1"></a>    i <span class="ot">&lt;-</span> comp_list[[j]][[iname]]</span>
<span id="cb26-17"><a href="#cb26-17" tabindex="-1"></a>    x <span class="ot">&lt;-</span> use_df <span class="sc">%&gt;%</span> </span>
<span id="cb26-18"><a href="#cb26-18" tabindex="-1"></a>      <span class="fu">filter</span>(sample_alias <span class="sc">%in%</span> i) <span class="sc">%&gt;%</span> </span>
<span id="cb26-19"><a href="#cb26-19" tabindex="-1"></a>      <span class="fu">mutate</span>(</span>
<span id="cb26-20"><a href="#cb26-20" tabindex="-1"></a>        <span class="at">sample_type =</span> <span class="fu">ifelse</span>(sample_alias <span class="sc">==</span> i[<span class="dv">1</span>], <span class="st">&#39;Primary tumor&#39;</span>, <span class="st">&#39;Metastasis&#39;</span>),</span>
<span id="cb26-21"><a href="#cb26-21" tabindex="-1"></a>        <span class="at">sample_type =</span> <span class="fu">factor</span>(sample_type, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&#39;Primary tumor&#39;</span>, <span class="st">&#39;Metastasis&#39;</span>))</span>
<span id="cb26-22"><a href="#cb26-22" tabindex="-1"></a>      )</span>
<span id="cb26-23"><a href="#cb26-23" tabindex="-1"></a>    </span>
<span id="cb26-24"><a href="#cb26-24" tabindex="-1"></a>    <span class="co"># Select guides with cpm &gt; cpm_threshold</span></span>
<span id="cb26-25"><a href="#cb26-25" tabindex="-1"></a>    x_guides_keep <span class="ot">&lt;-</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb26-26"><a href="#cb26-26" tabindex="-1"></a>      <span class="fu">group_by</span>(sample_type, guide) <span class="sc">%&gt;%</span> </span>
<span id="cb26-27"><a href="#cb26-27" tabindex="-1"></a>      <span class="fu">summarise</span>(<span class="at">cpm =</span> <span class="fu">max</span>(cpm)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-28"><a href="#cb26-28" tabindex="-1"></a>      <span class="fu">filter</span>(cpm <span class="sc">&gt;</span> cpm_threshold) <span class="sc">%&gt;%</span> </span>
<span id="cb26-29"><a href="#cb26-29" tabindex="-1"></a>      <span class="fu">pull</span>(guide)</span>
<span id="cb26-30"><a href="#cb26-30" tabindex="-1"></a>    </span>
<span id="cb26-31"><a href="#cb26-31" tabindex="-1"></a>    <span class="co"># Group guides with cpm &lt; cpm_threshold into the low abundant category, and sum counts for low abundant</span></span>
<span id="cb26-32"><a href="#cb26-32" tabindex="-1"></a>    x <span class="sc">%&lt;&gt;%</span> </span>
<span id="cb26-33"><a href="#cb26-33" tabindex="-1"></a>      <span class="fu">mutate</span>(</span>
<span id="cb26-34"><a href="#cb26-34" tabindex="-1"></a>        <span class="at">guide =</span> <span class="fu">ifelse</span>(guide <span class="sc">%in%</span>  x_guides_keep, guide, <span class="st">&#39;low abundant&#39;</span>)</span>
<span id="cb26-35"><a href="#cb26-35" tabindex="-1"></a>      ) <span class="sc">%&gt;%</span> </span>
<span id="cb26-36"><a href="#cb26-36" tabindex="-1"></a>      <span class="fu">group_by</span>(sample_type, guide) <span class="sc">%&gt;%</span></span>
<span id="cb26-37"><a href="#cb26-37" tabindex="-1"></a>      <span class="fu">summarise</span>(<span class="at">cpm =</span> <span class="fu">sum</span>(cpm)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-38"><a href="#cb26-38" tabindex="-1"></a>      <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb26-39"><a href="#cb26-39" tabindex="-1"></a>      <span class="fu">mutate</span>(</span>
<span id="cb26-40"><a href="#cb26-40" tabindex="-1"></a>        <span class="at">guide =</span> <span class="fu">fct_reorder</span>(guide, cpm),</span>
<span id="cb26-41"><a href="#cb26-41" tabindex="-1"></a>        <span class="at">guide =</span> <span class="fu">relevel</span>(guide, <span class="at">ref =</span> <span class="st">&#39;low abundant&#39;</span>)</span>
<span id="cb26-42"><a href="#cb26-42" tabindex="-1"></a>      )</span>
<span id="cb26-43"><a href="#cb26-43" tabindex="-1"></a>    </span>
<span id="cb26-44"><a href="#cb26-44" tabindex="-1"></a>    use_cols <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb26-45"><a href="#cb26-45" tabindex="-1"></a>      <span class="st">&#39;low abundant&#39;</span> <span class="ot">=</span> <span class="st">&#39;grey80&#39;</span>,</span>
<span id="cb26-46"><a href="#cb26-46" tabindex="-1"></a>      <span class="fu">colorRampPalette</span>(<span class="fu">rev</span>(palette_OkabeIto))(<span class="fu">nlevels</span>(x<span class="sc">$</span>guide) <span class="sc">-</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="fu">set_names</span>(<span class="fu">levels</span>(x<span class="sc">$</span>guide)[<span class="sc">-</span><span class="dv">1</span>])</span>
<span id="cb26-47"><a href="#cb26-47" tabindex="-1"></a>    )</span>
<span id="cb26-48"><a href="#cb26-48" tabindex="-1"></a>    </span>
<span id="cb26-49"><a href="#cb26-49" tabindex="-1"></a>    x <span class="sc">%&gt;%</span> </span>
<span id="cb26-50"><a href="#cb26-50" tabindex="-1"></a>      <span class="fu">ggplot</span>( <span class="fu">aes</span>(<span class="at">x =</span> sample_type, <span class="at">y =</span> cpm<span class="sc">/</span><span class="dv">10000</span>, <span class="at">group =</span> guide, <span class="at">fill =</span> guide)) <span class="sc">+</span> </span>
<span id="cb26-51"><a href="#cb26-51" tabindex="-1"></a>      <span class="fu">geom_area</span>(<span class="at">colour =</span> <span class="fu">alpha</span>(<span class="st">&quot;white&quot;</span>, <span class="fl">0.1</span>), <span class="at">linewidth =</span> <span class="fl">0.08</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb26-52"><a href="#cb26-52" tabindex="-1"></a>      <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> use_cols, <span class="at">guide =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb26-53"><a href="#cb26-53" tabindex="-1"></a>      <span class="fu">labs</span>(</span>
<span id="cb26-54"><a href="#cb26-54" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">&#39;&#39;</span>,</span>
<span id="cb26-55"><a href="#cb26-55" tabindex="-1"></a>        <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">&quot;CPM x 10&quot;</span><span class="sc">^-</span><span class="dv">4</span>),</span>
<span id="cb26-56"><a href="#cb26-56" tabindex="-1"></a>        <span class="at">title =</span> iname,</span>
<span id="cb26-57"><a href="#cb26-57" tabindex="-1"></a>        <span class="at">fill =</span> <span class="st">&#39;&#39;</span></span>
<span id="cb26-58"><a href="#cb26-58" tabindex="-1"></a>      ) <span class="sc">+</span></span>
<span id="cb26-59"><a href="#cb26-59" tabindex="-1"></a>      <span class="fu">scale_x_discrete</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb26-60"><a href="#cb26-60" tabindex="-1"></a>      <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1000000</span><span class="sc">/</span><span class="dv">10000</span>)) <span class="sc">+</span></span>
<span id="cb26-61"><a href="#cb26-61" tabindex="-1"></a>      <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">8</span>)) <span class="sc">+</span></span>
<span id="cb26-62"><a href="#cb26-62" tabindex="-1"></a>      <span class="fu">theme</span>(</span>
<span id="cb26-63"><a href="#cb26-63" tabindex="-1"></a>        <span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">&quot;grey90&quot;</span>),</span>
<span id="cb26-64"><a href="#cb26-64" tabindex="-1"></a>        <span class="co"># plot.margin = margin(0.5, 1.5, 0.5, 0, &quot;cm&quot;),</span></span>
<span id="cb26-65"><a href="#cb26-65" tabindex="-1"></a>        <span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="dv">0</span>, <span class="fl">0.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="st">&quot;cm&quot;</span>),</span>
<span id="cb26-66"><a href="#cb26-66" tabindex="-1"></a>        <span class="at">axis.text =</span>  <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">3</span>),</span>
<span id="cb26-67"><a href="#cb26-67" tabindex="-1"></a>        <span class="at">axis.title =</span>  <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">3</span>),</span>
<span id="cb26-68"><a href="#cb26-68" tabindex="-1"></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">3</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>),</span>
<span id="cb26-69"><a href="#cb26-69" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">3</span>),</span>
<span id="cb26-70"><a href="#cb26-70" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">3</span>),</span>
<span id="cb26-71"><a href="#cb26-71" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">&#39;bottom&#39;</span>,</span>
<span id="cb26-72"><a href="#cb26-72" tabindex="-1"></a>        <span class="at">legend.key.size =</span> <span class="fu">unit</span>(<span class="fl">0.1</span>, <span class="st">&#39;cm&#39;</span>),</span>
<span id="cb26-73"><a href="#cb26-73" tabindex="-1"></a>        <span class="at">legend.key.spacing.y =</span> <span class="fu">unit</span>(<span class="fl">0.1</span>, <span class="st">&#39;mm&#39;</span>),</span>
<span id="cb26-74"><a href="#cb26-74" tabindex="-1"></a>        <span class="at">legend.key.spacing.x =</span> <span class="fu">unit</span>(<span class="fl">0.1</span>, <span class="st">&#39;mm&#39;</span>)</span>
<span id="cb26-75"><a href="#cb26-75" tabindex="-1"></a>      )</span>
<span id="cb26-76"><a href="#cb26-76" tabindex="-1"></a>  }</span>
<span id="cb26-77"><a href="#cb26-77" tabindex="-1"></a>  <span class="fu">names</span>(res) <span class="ot">&lt;-</span> <span class="fu">names</span>(comp_list[[j]])</span>
<span id="cb26-78"><a href="#cb26-78" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb26-79"><a href="#cb26-79" tabindex="-1"></a></span>
<span id="cb26-80"><a href="#cb26-80" tabindex="-1"></a>}</span>
<span id="cb26-81"><a href="#cb26-81" tabindex="-1"></a><span class="fu">names</span>(muller_plots) <span class="ot">&lt;-</span> <span class="fu">names</span>(comp_list)</span>
<span id="cb26-82"><a href="#cb26-82" tabindex="-1"></a><span class="co"># muller_syngeneic_representative &lt;- muller_plots</span></span></code></pre></div>
<div id="supplementary-figure-2d-left-upper" class="section level3">
<h3>Supplementary Figure 2D (left upper)</h3>
<p>Muller plots depicting the relative frequency of unique sgRNAs in
matched primary tumors (mammary or lung orthotopic) versus synchronous
metastasis samples (lung metastasis for mammary tumors, liver metastasis
for lung tumors) from syngeneic and NSG recipients transplanted with the
indicated models previously transduced as indicated; n=5 per group
(representative samples are shown).</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a>plot_list <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb27-2"><a href="#cb27-2" tabindex="-1"></a>  muller_plots[[<span class="st">&#39;Comparison 7: LLC1 NSG dCas9 + Puro&#39;</span>]][[<span class="st">&#39;LLC1 NSG dCas9 + Puro 2&#39;</span>]] <span class="sc">+</span> <span class="fu">guides</span>(<span class="at">fill =</span> <span class="st">&#39;none&#39;</span>),</span>
<span id="cb27-3"><a href="#cb27-3" tabindex="-1"></a>  muller_plots[[<span class="st">&#39;Comparison 5: LLC1 Syngeneic dCas9 + Puro&#39;</span>]][[<span class="st">&#39;LLC1 Syngeneic dCas9 + Puro 1&#39;</span>]]<span class="sc">+</span>  <span class="fu">guides</span>(<span class="at">fill =</span> <span class="st">&#39;none&#39;</span>),</span>
<span id="cb27-4"><a href="#cb27-4" tabindex="-1"></a>  muller_plots[[<span class="st">&#39;Comparison 6: CMT167 NSG dCas9 + Puro&#39;</span>]][[<span class="st">&#39;CMT167 NSG dCas9 + Puro 4&#39;</span>]] <span class="sc">+</span> <span class="fu">guides</span>(<span class="at">fill =</span> <span class="st">&#39;none&#39;</span>) ,</span>
<span id="cb27-5"><a href="#cb27-5" tabindex="-1"></a>  muller_plots[[<span class="st">&#39;Comparison 4: CMT167 Syngeneic dCas9 + Puro&#39;</span>]][[<span class="st">&#39;CMT167 Syngeneic dCas9 + Puro 3&#39;</span>]] <span class="sc">+</span> <span class="fu">guides</span>(<span class="at">fill =</span> <span class="st">&#39;none&#39;</span>)</span>
<span id="cb27-6"><a href="#cb27-6" tabindex="-1"></a>)</span>
<span id="cb27-7"><a href="#cb27-7" tabindex="-1"></a></span>
<span id="cb27-8"><a href="#cb27-8" tabindex="-1"></a><span class="fu">wrap_plots</span>(plot_list, <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">byrow =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> print</span></code></pre></div>
<p><img src="figure/crispr-mm_2215_sgRNA-clonality.Rmd/muller-sfig2-d-left-upper-1.png" style="display: block; margin: auto;" /></p>
</div>
<div id="supplementary-figure-2d-right-upper" class="section level3">
<h3>Supplementary Figure 2D (right upper)</h3>
<p>Muller plots depicting the relative frequency of unique sgRNAs in
matched primary tumors (mammary or lung orthotopic) versus synchronous
metastasis samples (lung metastasis for mammary tumors, liver metastasis
for lung tumors) from syngeneic and NSG recipients transplanted with the
indicated models previously transduced as indicated; n=5 per group
(representative samples are shown).</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a>plot_list <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb28-2"><a href="#cb28-2" tabindex="-1"></a>  muller_plots[[<span class="st">&#39;Comparison 11: Thy1.1 in NSG&#39;</span>]][[<span class="st">&#39;LLC1 NSG Thy1.1&#39;</span>]]<span class="sc">+</span> <span class="fu">guides</span>(<span class="at">fill =</span> <span class="st">&#39;none&#39;</span>),</span>
<span id="cb28-3"><a href="#cb28-3" tabindex="-1"></a>  muller_plots[[<span class="st">&#39;Comparison 10: Thy1.1 in BL6&#39;</span>]][[<span class="st">&#39;LLC1 BL6 Thy1.1&#39;</span>]]<span class="sc">+</span> <span class="fu">guides</span>(<span class="at">fill =</span> <span class="st">&#39;none&#39;</span>),</span>
<span id="cb28-4"><a href="#cb28-4" tabindex="-1"></a>  muller_plots[[<span class="st">&#39;Comparison 11: Thy1.1 in NSG&#39;</span>]][[<span class="st">&#39;CMT167 NSG Thy1.1&#39;</span>]]<span class="sc">+</span> <span class="fu">guides</span>(<span class="at">fill =</span> <span class="st">&#39;none&#39;</span>),</span>
<span id="cb28-5"><a href="#cb28-5" tabindex="-1"></a>  muller_plots[[<span class="st">&#39;Comparison 10: Thy1.1 in BL6&#39;</span>]][[<span class="st">&#39;CMT167 BL6 Thy1.1&#39;</span>]]<span class="sc">+</span> <span class="fu">guides</span>(<span class="at">fill =</span> <span class="st">&#39;none&#39;</span>)</span>
<span id="cb28-6"><a href="#cb28-6" tabindex="-1"></a>)</span>
<span id="cb28-7"><a href="#cb28-7" tabindex="-1"></a></span>
<span id="cb28-8"><a href="#cb28-8" tabindex="-1"></a><span class="fu">wrap_plots</span>(plot_list, <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">byrow =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> print</span></code></pre></div>
<p><img src="figure/crispr-mm_2215_sgRNA-clonality.Rmd/muller-sfig2-d-right-upper-1.png" style="display: block; margin: auto;" /></p>
</div>
<div id="supplementary-figure-2f" class="section level3">
<h3>Supplementary Figure 2F</h3>
<p>Muller plot depicting the relative frequency of the predominant
sgRNAs in matched primary tumors versus synchronous metastasis samples
(lungs) from syngeneic recipients transplanted with CT26 cells,
previously transduced with Thy1 as indicated; n=5 each (representative
samples are shown).</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" tabindex="-1"></a><span class="fu">print</span>(muller_plots[[<span class="st">&#39;Comparison 3: CT26 Syngeneic Thy1.1&#39;</span>]][[<span class="st">&#39;CT26 Syngeneic Thy1.1 3&#39;</span>]] <span class="sc">+</span> <span class="fu">guides</span>(<span class="at">fill =</span> <span class="st">&#39;none&#39;</span>))</span></code></pre></div>
<p><img src="figure/crispr-mm_2215_sgRNA-clonality.Rmd/muller-sfig2-f-1.png" style="display: block; margin: auto;" /></p>
</div>
<div id="supplementary-figure-4l" class="section level3">
<h3>Supplementary Figure 4L</h3>
<p>Application of StealTHY to syngeneic models of lung and colon
carcinoma. Left: schematic representation of the in vivo CRISPR screen
strategy to identify metastatic genes in immunocompetent mice, showing
the experimental pipeline followed for each cancer type. Right: Muller
plots depicting the relative frequency of the predominant sgRNAs in
matched primary tumors (lung or colon orthotopic) versus synchronous
metastasis samples (liver metastasis for lung tumors, lung metastasis
for colon tumors) from syngeneic recipients transplanted with the
indicated cell models, which received StealTHY KO of the mouse
interactome library; n=5 each (representative samples are shown).</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a>plot_list <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb30-2"><a href="#cb30-2" tabindex="-1"></a>  muller_plots[[<span class="st">&#39;Comparison 2: LLC1 Syngeneic Thy1.1&#39;</span>]][[<span class="st">&#39;LLC1 Syngeneic Thy1.1 4&#39;</span>]],</span>
<span id="cb30-3"><a href="#cb30-3" tabindex="-1"></a>  muller_plots[[<span class="st">&#39;Comparison 3: CT26 Syngeneic Thy1.1&#39;</span>]][[<span class="st">&#39;CT26 Syngeneic Thy1.1 1&#39;</span>]]</span>
<span id="cb30-4"><a href="#cb30-4" tabindex="-1"></a>)</span>
<span id="cb30-5"><a href="#cb30-5" tabindex="-1"></a></span>
<span id="cb30-6"><a href="#cb30-6" tabindex="-1"></a><span class="fu">wrap_plots</span>(plot_list, <span class="at">ncol =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> print</span></code></pre></div>
<p><img src="figure/crispr-mm_2215_sgRNA-clonality.Rmd/muller-sfig4-l-1.png" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<p>R version 4.4.3 (2025-02-28) Platform: aarch64-apple-darwin20 Running
under: macOS 26.0.1</p>
<p>Matrix products: default BLAS:
/Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib
LAPACK:
/Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;
LAPACK version 3.12.0</p>
<p>locale: [1]
en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</p>
<p>time zone: Europe/Zurich tzcode source: internal</p>
<p>attached base packages: [1] stats4 stats graphics grDevices utils
datasets methods<br />
[8] base</p>
<p>other attached packages: [1] cowplot_1.2.0 ggpubr_0.6.1<br />
[3] ggbeeswarm_0.7.2 colorblindr_0.1.0<br />
[5] colorspace_2.1-1 patchwork_1.3.1<br />
[7] ggh4x_0.3.1 ggridges_0.5.6<br />
[9] SummarizedExperiment_1.36.0 Biobase_2.66.0<br />
[11] GenomicRanges_1.58.0 GenomeInfoDb_1.42.3<br />
[13] IRanges_2.40.1 S4Vectors_0.44.0<br />
[15] BiocGenerics_0.52.0 MatrixGenerics_1.18.1<br />
[17] matrixStats_1.5.0 diptest_0.77-1<br />
[19] kableExtra_1.4.0 DT_0.34.0<br />
[21] magrittr_2.0.3 foreach_1.5.2<br />
[23] knitr_1.50 lubridate_1.9.4<br />
[25] forcats_1.0.0 stringr_1.5.1<br />
[27] dplyr_1.1.4 purrr_1.1.0<br />
[29] readr_2.1.5 tidyr_1.3.1<br />
[31] tibble_3.3.0 ggplot2_3.5.2<br />
[33] tidyverse_2.0.0 workflowr_1.7.1</p>
<p>loaded via a namespace (and not attached): [1] rlang_1.1.6
git2r_0.36.2 compiler_4.4.3<br />
[4] getPass_0.2-4 systemfonts_1.2.3 callr_3.7.6<br />
[7] vctrs_0.6.5 pkgconfig_2.0.3 crayon_1.5.3<br />
[10] fastmap_1.2.0 backports_1.5.0 XVector_0.46.0<br />
[13] labeling_0.4.3 promises_1.3.3 rmarkdown_2.29<br />
[16] tzdb_0.5.0 UCSC.utils_1.2.0 ps_1.9.1<br />
[19] xfun_0.53 zlibbioc_1.52.0 cachem_1.1.0<br />
[22] jsonlite_2.0.0 later_1.4.4 DelayedArray_0.32.0<br />
[25] broom_1.0.9 R6_2.6.1 bslib_0.9.0<br />
[28] stringi_1.8.7 RColorBrewer_1.1-3 car_3.1-3<br />
[31] jquerylib_0.1.4 Rcpp_1.1.0 iterators_1.0.14<br />
[34] httpuv_1.6.16 Matrix_1.7-3 timechange_0.3.0<br />
[37] tidyselect_1.2.1 rstudioapi_0.17.1 dichromat_2.0-0.1<br />
[40] abind_1.4-8 yaml_2.3.10 codetools_0.2-20<br />
[43] processx_3.8.6 lattice_0.22-7 withr_3.0.2<br />
[46] evaluate_1.0.5 xml2_1.3.8 pillar_1.11.0<br />
[49] carData_3.0-5 whisker_0.4.1 generics_0.1.4<br />
[52] rprojroot_2.1.0 hms_1.1.3 scales_1.4.0<br />
[55] glue_1.8.0 tools_4.4.3 ggsignif_0.6.4<br />
[58] fs_1.6.6 grid_4.4.3 crosstalk_1.2.2<br />
[61] GenomeInfoDbData_1.2.13 beeswarm_0.4.0 vipor_0.4.7<br />
[64] Formula_1.2-5 cli_3.6.5 textshaping_1.0.1<br />
[67] S4Arrays_1.6.0 viridisLite_0.4.2 svglite_2.2.1<br />
[70] gtable_0.3.6 rstatix_0.7.2 sass_0.4.10<br />
[73] digest_0.6.37 SparseArray_1.6.2 htmlwidgets_1.6.4<br />
[76] farver_2.1.2 htmltools_0.5.8.1 lifecycle_1.0.4<br />
[79] httr_1.4.7</p>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
