<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Francesc Castro-Giner" />

<meta name="date" content="2025-10-31" />

<title>Clonal dynamics in humanized NSG mice (huNSG)</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">StealTHY</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="about.html">
    <span class="fa fa-info"></span>
     
    About
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-bar-chart"></span>
     
    Analyses
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Data generation</li>
    <li>
      <a href="0_differential_expression_gsea_gsva.html">Run differential gene expression analysis, GSEA and GSVA</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Main results</li>
    <li>
      <a href="patients_ctc_counts_distribution.html">CTC counts distribution in breast cancer patients</a>
    </li>
    <li>
      <a href="br16_pca.html">PCA analysis of CTC gene expression in NSG-CDX-BR16 mice</a>
    </li>
    <li>
      <a href="br16_dge.html">Differential gene expression report for NSG-CDX-BR16 data</a>
    </li>
    <li>
      <a href="gsea_across_models.html">Report of GSEA and GSVA analysis in NSG-CDX-BR16, NSG-LM2, Patient and NSG-LM2 time kinetics</a>
    </li>
    <li>
      <a href="core_gene_sets.html">Analysis of genes from candidate pathways</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="https://github.com/TheAcetoLab/saini-stealTHY">
    <span class="fa fa-github"></span>
     
  </a>
</li>
<li>
  <a href="https://www.theacetolab.com">
    <span class="fa fa-flask"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Clonal dynamics in humanized NSG mice
(huNSG)</h1>
<h3 class="subtitle">Human libray (n = 2180)</h3>
<h4 class="author">Francesc Castro-Giner</h4>
<h4 class="date">October 31, 2025</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-10-31
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>saini-stealTHY/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240517code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240517)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240517code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240517)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomTheAcetoLabsainistealTHYtree455fa9121fcca51363cf639f7ee170fb108e34c1targetblank455fa91a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/tree/455fa9121fcca51363cf639f7ee170fb108e34c1" target="_blank">455fa91</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomTheAcetoLabsainistealTHYtree455fa9121fcca51363cf639f7ee170fb108e34c1targetblank455fa91a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/TheAcetoLab/saini-stealTHY/tree/455fa9121fcca51363cf639f7ee170fb108e34c1" target="_blank">455fa91</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/templates/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    code/raw_data_processing/rnaseq/p27851_o32062/pipelines/
    Ignored:    configuration/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/crispr/
    Ignored:    data/resources/
    Ignored:    data/rnaseq/
    Ignored:    output/.DS_Store
    Ignored:    output/clinical/
    Ignored:    output/crispr/
    Ignored:    output/rnaseq/

Untracked files:
    Untracked:  analysis/crispr-mm_2215_sgRNA-CMT167-LLC1-CT26.md
    Untracked:  analysis/crispr-mm_2215_sgRNA-StealTHY.md
    Untracked:  analysis/crispr-mm_2215_sgRNA-clonality.md
    Untracked:  analysis/index.md
    Untracked:  analysis/rnaseq-tumor-bulk.md
    Untracked:  analysis/style.css
    Untracked:  code/R-functions/subchunkify.R

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/_site.yml
    Deleted:    analysis/crispr-hsapiens_2180_sgRNA_r1.Rmd
    Deleted:    analysis/crispr-mm_2215_sgRNA-clonality_r1.Rmd
    Deleted:    analysis/crispr-mm_2215_sgRNA-r2oC.Rmd
    Deleted:    analysis/crispr-mm_2215_sgRNA-r2oD.Rmd
    Deleted:    analysis/crispr-mm_2215_sgRNA-r2oD_r1.Rmd
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_ko_culture--over--amhr2_control_culture_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_ko_culture--over--amhr2_control_culture_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_ko_notgfbeta--over--amhr2_ko_culture_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_ko_notgfbeta--over--amhr2_ko_culture_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_overexpression--over--amhr2_control_culture_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_overexpression--over--amhr2_control_culture_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_overexpression--over--amhr2_ko_culture_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_overexpression--over--amhr2_ko_culture_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_overexpression--over--amhr2_ko_notgfbeta_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_overexpression--over--amhr2_ko_notgfbeta_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/figure/gse-dotplot-invivo_amhr2_ko-+-amhr2_ko_culture--over--invivo_control_sgrna-+-amhr2_control_culture_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-invivo_amhr2_ko-+-amhr2_ko_culture--over--invivo_control_sgrna-+-amhr2_control_culture_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/figure/gse-dotplot-invivo_amhr2_ko--over--invivo_control_sgrna_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-invivo_amhr2_ko--over--invivo_control_sgrna_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/file/tmp/dge.xlsx
    Deleted:    analysis/templates/docs/file/tmp/gse_gsea_msigdb.h.xlsx
    Deleted:    analysis/templates/docs/file/tmp/gse_gsego_bp.xlsx
    Deleted:    analysis/templates/docs/file/tmp/gse_gsego_mf.xlsx
    Modified:   analysis/templates/rnaseq-deg-edger.Rmd
    Modified:   code/R-functions/gse_report.r
    Modified:   configuration/rmarkdown/color_palettes.R
    Modified:   configuration/rmarkdown/ggplot_theme.R
    Modified:   update_workflowr.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/crispr-hsapiens_2180_sgRNA.Rmd</code>) and HTML
(<code>docs/crispr-hsapiens_2180_sgRNA.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/455fa9121fcca51363cf639f7ee170fb108e34c1/analysis/crispr-hsapiens_2180_sgRNA.Rmd" target="_blank">455fa91</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-10-31
</td>
<td>
Prepare files for publication
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/5fab19d765a869233e820f2aed94595a89d5fe9b/analysis/crispr-hsapiens_2180_sgRNA.Rmd" target="_blank">5fab19d</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2024-05-28
</td>
<td>
clean code
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries-additional-functions-and-data"
class="section level2">
<h2>Load libraries, additional functions and data</h2>
<p>Setup environment</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">results=</span><span class="st">&#39;asis&#39;</span>, <span class="at">echo=</span><span class="cn">TRUE</span>, <span class="at">message=</span><span class="cn">FALSE</span>, <span class="at">warning=</span><span class="cn">FALSE</span>, <span class="at">error=</span><span class="cn">FALSE</span>, <span class="at">fig.align =</span> <span class="st">&#39;center&#39;</span>, <span class="at">fig.width =</span> <span class="fl">3.5</span>, <span class="at">fig.asp =</span> <span class="fl">0.618</span>, <span class="at">dpi =</span> <span class="dv">600</span>, <span class="at">dev =</span> <span class="fu">c</span>(<span class="st">&quot;png&quot;</span>, <span class="st">&quot;pdf&quot;</span>), <span class="at">fig.showtext =</span> <span class="cn">FALSE</span>, <span class="at">engine.opts =</span> <span class="fu">list</span>(<span class="at">bash =</span> <span class="st">&quot;-l&quot;</span>))</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">options</span>(<span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>use_seed <span class="ot">&lt;-</span> <span class="dv">1100101</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="fu">set.seed</span>(use_seed)</span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a><span class="co"># dir.create(params$output_dir, recursive = TRUE, showWarnings = FALSE)</span></span></code></pre></div>
<p>Load packages</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="fu">library</span>(foreach)</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="fu">library</span>(magrittr)</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="fu">library</span>(DT)</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="fu">library</span>(diptest)</span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a><span class="fu">library</span>(ggridges)</span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a><span class="fu">library</span>(ggh4x)</span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a><span class="fu">library</span>(colorblindr)</span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a><span class="fu">library</span>(ggbeeswarm)</span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb2-17"><a href="#cb2-17" tabindex="-1"></a><span class="fu">library</span>(grid)</span>
<span id="cb2-18"><a href="#cb2-18" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb2-19"><a href="#cb2-19" tabindex="-1"></a><span class="fu">library</span>(MAGeCKFlute)</span></code></pre></div>
<p>Load ggplot theme</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;./configuration/rmarkdown/ggplot_theme.R&quot;</span>)</span></code></pre></div>
<p>Load ggplot theme</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;./configuration/rmarkdown/color_palettes.R&quot;</span>)</span></code></pre></div>
<p>Load custom functions</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&#39;./code/R-functions/geom_split_violin.R&#39;</span>)</span></code></pre></div>
<p>Load Summarized Experiment object</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">file.path</span>(params<span class="sc">$</span>data_dir, <span class="st">&#39;se.rds&#39;</span>))</span></code></pre></div>
<p>Filter samples used for this analysis</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>use_cols <span class="ot">&lt;-</span> <span class="fu">colData</span>(se) <span class="sc">%&gt;%</span> </span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>  data.frame <span class="sc">%&gt;%</span> </span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>  <span class="fu">filter</span>(effect_type_short <span class="sc">==</span> <span class="st">&#39;noKO&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>  <span class="fu">filter</span>(vector_short <span class="sc">!=</span> <span class="st">&quot;Thy1_EF1alpha_dCas9&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>  rownames</span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a></span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>se <span class="ot">&lt;-</span> se[,use_cols]</span></code></pre></div>
<p>Data wrangling</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>chunk_colData <span class="ot">&lt;-</span> <span class="fu">colData</span>(se) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>    <span class="at">immunogen_mmmodel =</span> <span class="fu">paste</span>(immunogen, mm_model_type, <span class="at">sep =</span> <span class="st">&#39;_&#39;</span>),</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>    <span class="at">immunogen_burden =</span> <span class="fu">ifelse</span>(</span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>      immunogen <span class="sc">==</span> <span class="st">&#39;no_immunogen&#39;</span> <span class="sc">&amp;</span> mm_model_type <span class="sc">==</span> <span class="st">&#39;Imm.competent&#39;</span>,</span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a>      <span class="st">&#39;HuTHY1&#39;</span>, <span class="cn">NA</span>),</span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a>    <span class="at">immunogen_burden =</span> <span class="fu">ifelse</span>(</span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a>      immunogen <span class="sc">==</span> <span class="st">&#39;dCas9-GFP&#39;</span> <span class="sc">&amp;</span> mm_model_type <span class="sc">==</span> <span class="st">&#39;Imm.deficient&#39;</span>,</span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a>      <span class="st">&#39;GFP&#39;</span>, immunogen_burden),</span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a>    <span class="at">immunogen_burden =</span> <span class="fu">ifelse</span>(</span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a>      immunogen <span class="sc">==</span> <span class="st">&#39;dCas9-THY&#39;</span> <span class="sc">&amp;</span> mm_model_type <span class="sc">==</span> <span class="st">&#39;Imm.competent&#39;</span>,</span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a>      <span class="st">&#39;Cas9-HuTHY1&#39;</span>, immunogen_burden),</span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a>    <span class="at">immunogen_burden =</span> <span class="fu">ifelse</span>(</span>
<span id="cb8-14"><a href="#cb8-14" tabindex="-1"></a>      immunogen <span class="sc">==</span> <span class="st">&#39;dCas9-GFP&#39;</span> <span class="sc">&amp;</span> mm_model_type <span class="sc">==</span> <span class="st">&#39;Imm.competent&#39;</span>,</span>
<span id="cb8-15"><a href="#cb8-15" tabindex="-1"></a>      <span class="st">&#39;Cas9-GFP&#39;</span>, immunogen_burden),</span>
<span id="cb8-16"><a href="#cb8-16" tabindex="-1"></a>    <span class="at">immunogen_burden =</span> <span class="fu">factor</span>(</span>
<span id="cb8-17"><a href="#cb8-17" tabindex="-1"></a>      immunogen_burden,</span>
<span id="cb8-18"><a href="#cb8-18" tabindex="-1"></a>      <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&#39;HuTHY1&#39;</span>, <span class="st">&#39;GFP&#39;</span>, <span class="st">&#39;Cas9-HuTHY1&#39;</span>, <span class="st">&#39;Cas9-GFP&#39;</span>)),</span>
<span id="cb8-19"><a href="#cb8-19" tabindex="-1"></a>    <span class="at">sample_type =</span> <span class="fu">case_match</span>(</span>
<span id="cb8-20"><a href="#cb8-20" tabindex="-1"></a>        sample_type,</span>
<span id="cb8-21"><a href="#cb8-21" tabindex="-1"></a>        <span class="st">&quot;In vitro&quot;</span> <span class="sc">~</span> <span class="st">&#39;Input&#39;</span>,</span>
<span id="cb8-22"><a href="#cb8-22" tabindex="-1"></a>        <span class="st">&quot;whole_tumor&quot;</span> <span class="sc">~</span> <span class="st">&#39;Primary tumor&#39;</span>,</span>
<span id="cb8-23"><a href="#cb8-23" tabindex="-1"></a>        <span class="st">&quot;whole_lung&quot;</span> <span class="sc">~</span> <span class="st">&#39;Lungs&#39;</span>,</span>
<span id="cb8-24"><a href="#cb8-24" tabindex="-1"></a>        <span class="st">&quot;met_lymph&quot;</span> <span class="sc">~</span> <span class="st">&#39;Lymph nodes&#39;</span>,</span>
<span id="cb8-25"><a href="#cb8-25" tabindex="-1"></a>        <span class="st">&quot;met_liver&quot;</span> <span class="sc">~</span> <span class="st">&#39;Liver&#39;</span></span>
<span id="cb8-26"><a href="#cb8-26" tabindex="-1"></a>      ),</span>
<span id="cb8-27"><a href="#cb8-27" tabindex="-1"></a>    <span class="at">sample_type =</span> <span class="fu">factor</span>(</span>
<span id="cb8-28"><a href="#cb8-28" tabindex="-1"></a>      sample_type,</span>
<span id="cb8-29"><a href="#cb8-29" tabindex="-1"></a>      <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&#39;Input&#39;</span>, <span class="st">&#39;Primary tumor&#39;</span>, <span class="st">&#39;Lungs&#39;</span>, <span class="st">&#39;Liver&#39;</span>, <span class="st">&#39;Lymph nodes&#39;</span>)               </span>
<span id="cb8-30"><a href="#cb8-30" tabindex="-1"></a>    ),</span>
<span id="cb8-31"><a href="#cb8-31" tabindex="-1"></a>    </span>
<span id="cb8-32"><a href="#cb8-32" tabindex="-1"></a>    <span class="at">mouse_model_short =</span> <span class="fu">ifelse</span>(mm_model_type <span class="sc">==</span> <span class="st">&#39;Imm.competent&#39;</span>,</span>
<span id="cb8-33"><a href="#cb8-33" tabindex="-1"></a>                               <span class="st">&#39;huNSG&#39;</span>, <span class="cn">NA</span>),</span>
<span id="cb8-34"><a href="#cb8-34" tabindex="-1"></a>    <span class="at">mouse_model_short =</span> <span class="fu">ifelse</span>(mm_model_type <span class="sc">==</span> <span class="st">&#39;Imm.deficient&#39;</span>,</span>
<span id="cb8-35"><a href="#cb8-35" tabindex="-1"></a>                               <span class="st">&#39;NSG&#39;</span>, mouse_model_short),</span>
<span id="cb8-36"><a href="#cb8-36" tabindex="-1"></a>    <span class="at">mouse_model_short =</span> <span class="fu">ifelse</span>(mm_model_type <span class="sc">==</span> <span class="st">&#39;In_vitro&#39;</span>,</span>
<span id="cb8-37"><a href="#cb8-37" tabindex="-1"></a>                               <span class="st">&#39;In vitro&#39;</span>, mouse_model_short)</span>
<span id="cb8-38"><a href="#cb8-38" tabindex="-1"></a>    <span class="co"># </span></span>
<span id="cb8-39"><a href="#cb8-39" tabindex="-1"></a>    <span class="co"># vector_short = ifelse(vector_short == &#39;Puro_EF1alpha_dCas9&#39;, </span></span>
<span id="cb8-40"><a href="#cb8-40" tabindex="-1"></a>    <span class="co">#                       &#39;dCas9+Puro&#39;, vector_short),</span></span>
<span id="cb8-41"><a href="#cb8-41" tabindex="-1"></a>    <span class="co"># vector_short = factor(vector_short, levels = c(&#39;Thy1&#39;, &#39;Puro&#39;, &#39;dCas9+Puro&#39;)),</span></span>
<span id="cb8-42"><a href="#cb8-42" tabindex="-1"></a>    <span class="co"># mouse_model_short = case_match(mouse_model_global,</span></span>
<span id="cb8-43"><a href="#cb8-43" tabindex="-1"></a>    <span class="co">#                                &#39;immunodeficient&#39; ~ &#39;NSG&#39;,</span></span>
<span id="cb8-44"><a href="#cb8-44" tabindex="-1"></a>    <span class="co">#                                &#39;immunocompetent&#39; ~ &#39;Syngeneic&#39;</span></span>
<span id="cb8-45"><a href="#cb8-45" tabindex="-1"></a>    <span class="co">#                                ),</span></span>
<span id="cb8-46"><a href="#cb8-46" tabindex="-1"></a>    <span class="co"># mouse_model_short = factor(mouse_model_short, levels = c(&#39;NSG&#39;, &#39;Syngeneic&#39;)),</span></span>
<span id="cb8-47"><a href="#cb8-47" tabindex="-1"></a>    <span class="co"># vector_mhost = paste0(vector_short, &#39;, &#39;, mouse_model_short),</span></span>
<span id="cb8-48"><a href="#cb8-48" tabindex="-1"></a>    <span class="co"># model_vector_mhost = paste0(donor,&#39;, &#39;, vector_short, &#39;, &#39;, mouse_model_short),</span></span>
<span id="cb8-49"><a href="#cb8-49" tabindex="-1"></a>    <span class="co"># </span></span>
<span id="cb8-50"><a href="#cb8-50" tabindex="-1"></a>    <span class="co"># sample_type = case_match(sample_type, </span></span>
<span id="cb8-51"><a href="#cb8-51" tabindex="-1"></a>    <span class="co">#                          &#39;whole_tumor&#39; ~ &#39;Primary tumor&#39;, </span></span>
<span id="cb8-52"><a href="#cb8-52" tabindex="-1"></a>    <span class="co">#                          &#39;whole_lung&#39; ~ &#39;Lung mets&#39;),</span></span>
<span id="cb8-53"><a href="#cb8-53" tabindex="-1"></a>    <span class="co"># sample_type = factor(sample_type, levels = c(&#39;Primary tumor&#39;, &#39;Lung mets&#39;))</span></span>
<span id="cb8-54"><a href="#cb8-54" tabindex="-1"></a>  )</span>
<span id="cb8-55"><a href="#cb8-55" tabindex="-1"></a></span>
<span id="cb8-56"><a href="#cb8-56" tabindex="-1"></a><span class="fu">colData</span>(se) <span class="ot">&lt;-</span> chunk_colData <span class="sc">%&gt;%</span> DataFrame</span></code></pre></div>
</div>
<div id="configure-analysis" class="section level2">
<h2>Configure analysis</h2>
<div id="mageck-analysis" class="section level3">
<h3>MAGeCK analysis</h3>
<div id="list-of-comparisons" class="section level4">
<h4>List of comparisons</h4>
<p>List of comparisons for differential abundance using MAGeCK</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">colData</span>(se) <span class="sc">%&gt;%</span> data.frame</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>mageck_comp_list <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a>  <span class="st">`</span><span class="at">StealTHY huNSG</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a>    <span class="st">`</span><span class="at">Primary tumor</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(immunogen <span class="sc">==</span> <span class="st">&#39;StealTHY&#39;</span> <span class="sc">&amp;</span> mm_model_type <span class="sc">==</span> <span class="st">&#39;Imm.competent&#39;</span> <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Primary tumor&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a>      <span class="fu">arrange</span>(replicate) <span class="sc">%&gt;%</span> </span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a>      <span class="fu">pull</span>(sample_alias),</span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a>    <span class="st">`</span><span class="at">Lung mets</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(immunogen <span class="sc">==</span> <span class="st">&#39;StealTHY&#39;</span> <span class="sc">&amp;</span> mm_model_type <span class="sc">==</span> <span class="st">&#39;Imm.competent&#39;</span> <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Lungs&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a>      <span class="fu">filter</span>(replicate <span class="sc">%in%</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb9-10"><a href="#cb9-10" tabindex="-1"></a>      <span class="fu">arrange</span>(replicate) <span class="sc">%&gt;%</span> </span>
<span id="cb9-11"><a href="#cb9-11" tabindex="-1"></a>      <span class="fu">pull</span>(sample_alias),</span>
<span id="cb9-12"><a href="#cb9-12" tabindex="-1"></a>    <span class="at">paired =</span> <span class="cn">TRUE</span></span>
<span id="cb9-13"><a href="#cb9-13" tabindex="-1"></a>    ),</span>
<span id="cb9-14"><a href="#cb9-14" tabindex="-1"></a>  <span class="st">`</span><span class="at">StealTHY NSG</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb9-15"><a href="#cb9-15" tabindex="-1"></a>    <span class="at">Tumor =</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(immunogen <span class="sc">==</span> <span class="st">&#39;StealTHY&#39;</span> <span class="sc">&amp;</span> mm_model_type <span class="sc">==</span> <span class="st">&#39;Imm.deficient&#39;</span> <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Primary tumor&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb9-16"><a href="#cb9-16" tabindex="-1"></a>      <span class="fu">arrange</span>(replicate) <span class="sc">%&gt;%</span> </span>
<span id="cb9-17"><a href="#cb9-17" tabindex="-1"></a>      <span class="fu">pull</span>(sample_alias),</span>
<span id="cb9-18"><a href="#cb9-18" tabindex="-1"></a>    <span class="at">Lung =</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(immunogen <span class="sc">==</span> <span class="st">&#39;StealTHY&#39;</span> <span class="sc">&amp;</span> mm_model_type <span class="sc">==</span> <span class="st">&#39;Imm.deficient&#39;</span> <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Lungs&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb9-19"><a href="#cb9-19" tabindex="-1"></a>      <span class="fu">arrange</span>(replicate) <span class="sc">%&gt;%</span> </span>
<span id="cb9-20"><a href="#cb9-20" tabindex="-1"></a>      <span class="fu">pull</span>(sample_alias),</span>
<span id="cb9-21"><a href="#cb9-21" tabindex="-1"></a>    <span class="at">paired =</span> <span class="cn">TRUE</span></span>
<span id="cb9-22"><a href="#cb9-22" tabindex="-1"></a>    )</span>
<span id="cb9-23"><a href="#cb9-23" tabindex="-1"></a>)</span>
<span id="cb9-24"><a href="#cb9-24" tabindex="-1"></a></span>
<span id="cb9-25"><a href="#cb9-25" tabindex="-1"></a></span>
<span id="cb9-26"><a href="#cb9-26" tabindex="-1"></a>mageck_comp_list_names <span class="ot">&lt;-</span> <span class="fu">names</span>(mageck_comp_list) </span>
<span id="cb9-27"><a href="#cb9-27" tabindex="-1"></a><span class="fu">names</span>(mageck_comp_list_names) <span class="ot">&lt;-</span> <span class="fu">names</span>(mageck_comp_list) <span class="sc">%&gt;%</span> <span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>, <span class="st">&quot;_&quot;</span>,.) <span class="sc">%&gt;%</span> tolower</span></code></pre></div>
</div>
<div id="run-mageck" class="section level4">
<h4>Run MAGeCK</h4>
<p>Generate MAGeCK test scripts. To generate the script files change
<code>eval = FALSE</code> to <code>eval = TRUE</code> in the chunk
options.</p>
<p>The scripts must be run in the terminal, inside the path
./output/crispr/hsa_2180_sgRNA/mageck_rra. The script will activate the
mageck environment and run the test for each comparison. The results
will be saved in the same directory.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>out_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(params<span class="sc">$</span>output_dir, params<span class="sc">$</span>mageck_dir)</span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a><span class="fu">dir.create</span>(out_dir, <span class="at">recursive =</span> <span class="cn">TRUE</span>, <span class="at">showWarnings =</span> <span class="cn">FALSE</span>)</span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a><span class="co"># Generate count matrix</span></span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a>count_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(se, <span class="st">&#39;counts&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>(<span class="at">check.names =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&#39;sgRNA&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Gene =</span> <span class="fu">rowData</span>(se)<span class="sc">$</span>Gene) <span class="sc">%&gt;%</span> </span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sgRNA, Gene, <span class="fu">everything</span>())</span>
<span id="cb10-9"><a href="#cb10-9" tabindex="-1"></a><span class="fu">write_tsv</span>(count_mat, <span class="at">file =</span> <span class="fu">file.path</span>(out_dir, <span class="st">&#39;counts.txt&#39;</span>))</span>
<span id="cb10-10"><a href="#cb10-10" tabindex="-1"></a></span>
<span id="cb10-11"><a href="#cb10-11" tabindex="-1"></a><span class="co"># Generate list of control sgRNA IDs</span></span>
<span id="cb10-12"><a href="#cb10-12" tabindex="-1"></a>non_target_list <span class="ot">&lt;-</span> <span class="fu">rownames</span>(se) <span class="sc">%&gt;%</span> <span class="fu">grep</span>(<span class="st">&#39;Non_Target&#39;</span>, ., <span class="at">value =</span> T) <span class="sc">%&gt;%</span> data.frame</span>
<span id="cb10-13"><a href="#cb10-13" tabindex="-1"></a><span class="fu">write_tsv</span>(non_target_list, <span class="at">file =</span> <span class="fu">file.path</span>(out_dir, <span class="st">&#39;control_sgrna.txt&#39;</span>), <span class="at">col_names =</span> F)</span>
<span id="cb10-14"><a href="#cb10-14" tabindex="-1"></a></span>
<span id="cb10-15"><a href="#cb10-15" tabindex="-1"></a><span class="co"># Generate mageck commands</span></span>
<span id="cb10-16"><a href="#cb10-16" tabindex="-1"></a>res_cmd <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="st">&#39;source activate mageckenv&#39;</span>)</span>
<span id="cb10-17"><a href="#cb10-17" tabindex="-1"></a>i<span class="ot">=</span><span class="fu">names</span>(mageck_comp_list)[<span class="dv">1</span>]</span>
<span id="cb10-18"><a href="#cb10-18" tabindex="-1"></a>res_cmd <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i=</span><span class="fu">names</span>(mageck_comp_list), <span class="at">.combine =</span> rbind) <span class="sc">%do%</span> {</span>
<span id="cb10-19"><a href="#cb10-19" tabindex="-1"></a>  <span class="cf">if</span>(mageck_comp_list[[i]]<span class="sc">$</span>paired) {</span>
<span id="cb10-20"><a href="#cb10-20" tabindex="-1"></a>    <span class="fu">paste</span>(<span class="st">&#39;mageck test -k counts.txt --control-sgrna control_sgrna.txt --norm-method control --paired -t&#39;</span>, <span class="fu">paste</span>(mageck_comp_list[[i]][[<span class="dv">2</span>]], <span class="at">collapse =</span> <span class="st">&#39;,&#39;</span>),<span class="st">&#39;-c&#39;</span>,  <span class="fu">paste</span>(mageck_comp_list[[i]][[<span class="dv">1</span>]], <span class="at">collapse =</span> <span class="st">&#39;,&#39;</span>), <span class="st">&#39;-n&#39;</span>, <span class="fu">tolower</span>(<span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>, <span class="st">&quot;_&quot;</span>, i)), <span class="st">&#39;</span><span class="sc">\n</span><span class="st">&#39;</span>)} <span class="cf">else</span> {</span>
<span id="cb10-21"><a href="#cb10-21" tabindex="-1"></a>    <span class="fu">paste</span>(<span class="st">&#39;mageck test -k counts.txt --control-sgrna control_sgrna.txt --norm-method control -t&#39;</span>, <span class="fu">paste</span>(mageck_comp_list[[i]][[<span class="dv">2</span>]], <span class="at">collapse =</span> <span class="st">&#39;,&#39;</span>),<span class="st">&#39;-c&#39;</span>,  <span class="fu">paste</span>(mageck_comp_list[[i]][[<span class="dv">1</span>]], <span class="at">collapse =</span> <span class="st">&#39;,&#39;</span>), <span class="st">&#39;-n&#39;</span>, <span class="fu">tolower</span>(<span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>, <span class="st">&quot;_&quot;</span>, i)), <span class="st">&#39;</span><span class="sc">\n</span><span class="st">&#39;</span>)</span>
<span id="cb10-22"><a href="#cb10-22" tabindex="-1"></a>  }</span>
<span id="cb10-23"><a href="#cb10-23" tabindex="-1"></a>} <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>()</span>
<span id="cb10-24"><a href="#cb10-24" tabindex="-1"></a></span>
<span id="cb10-25"><a href="#cb10-25" tabindex="-1"></a>res_cmd <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="st">&#39;conda activate mageckenv&#39;</span>, res_cmd, <span class="st">&#39;conda deactivate&#39;</span>)</span>
<span id="cb10-26"><a href="#cb10-26" tabindex="-1"></a><span class="fu">write_tsv</span>(res_cmd, <span class="fu">file.path</span>(out_dir, <span class="st">&#39;run_mageck_test.sh&#39;</span>), <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<div id="load-mageck-results" class="section level4">
<h4>Load MAGeCK results</h4>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>gene_summ_files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> <span class="fu">file.path</span>(params<span class="sc">$</span>output_dir, params<span class="sc">$</span>mageck_dir), <span class="at">pattern =</span> <span class="st">&#39;gene_summary.txt&#39;</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>analysis_prefix <span class="ot">&lt;-</span> <span class="fu">basename</span>(gene_summ_files) <span class="sc">%&gt;%</span> <span class="fu">gsub</span>(<span class="st">&quot;.gene_summary.txt&quot;</span>, <span class="st">&quot;&quot;</span>, .)</span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a>gene_summ <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i =</span> gene_summ_files) <span class="sc">%do%</span> <span class="fu">read.delim</span>(i, <span class="at">check.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a><span class="fu">names</span>(gene_summ) <span class="ot">&lt;-</span> analysis_prefix</span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a></span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a>sgrna_summ_files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> <span class="fu">file.path</span>(params<span class="sc">$</span>output_dir, params<span class="sc">$</span>mageck_dir), <span class="at">pattern =</span> <span class="st">&#39;sgrna_summary.txt&#39;</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a>analysis_prefix <span class="ot">&lt;-</span> <span class="fu">basename</span>(sgrna_summ_files) <span class="sc">%&gt;%</span> <span class="fu">gsub</span>(<span class="st">&quot;.sgrna_summary.txt&quot;</span>, <span class="st">&quot;&quot;</span>, .)</span>
<span id="cb11-8"><a href="#cb11-8" tabindex="-1"></a>sgrna_summ <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i =</span> sgrna_summ_files) <span class="sc">%do%</span> <span class="fu">read.delim</span>(i, <span class="at">check.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb11-9"><a href="#cb11-9" tabindex="-1"></a><span class="fu">names</span>(sgrna_summ) <span class="ot">&lt;-</span> analysis_prefix</span></code></pre></div>
</div>
</div>
</div>
<div id="muller-plots" class="section level2">
<h2>Muller plots</h2>
<p>Define muller plots groups</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">colData</span>(se) <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>() </span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a></span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a>muller_groups_selected <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a>  <span class="st">`</span><span class="at">HuTHY1</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a>    <span class="fu">filter</span>(immunogen_burden <span class="sc">==</span> <span class="st">&#39;HuTHY1&#39;</span> <span class="sc">&amp;</span> replicate <span class="sc">==</span> <span class="st">&#39;4&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a>    <span class="fu">pull</span>(sample_alias),</span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a>  <span class="st">`</span><span class="at">GFP</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb12-8"><a href="#cb12-8" tabindex="-1"></a>    <span class="fu">filter</span>(immunogen_burden <span class="sc">==</span> <span class="st">&#39;GFP&#39;</span> <span class="sc">&amp;</span> replicate <span class="sc">==</span> <span class="st">&#39;4&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb12-9"><a href="#cb12-9" tabindex="-1"></a>    <span class="fu">pull</span>(sample_alias),</span>
<span id="cb12-10"><a href="#cb12-10" tabindex="-1"></a>  <span class="st">`</span><span class="at">Cas9-HuTHY1</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb12-11"><a href="#cb12-11" tabindex="-1"></a>    <span class="fu">filter</span>(immunogen_burden <span class="sc">==</span> <span class="st">&#39;Cas9-HuTHY1&#39;</span> <span class="sc">&amp;</span> replicate <span class="sc">==</span> <span class="st">&#39;1&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb12-12"><a href="#cb12-12" tabindex="-1"></a>    <span class="fu">pull</span>(sample_alias),</span>
<span id="cb12-13"><a href="#cb12-13" tabindex="-1"></a>  <span class="st">`</span><span class="at">Cas9-GFP</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb12-14"><a href="#cb12-14" tabindex="-1"></a>    <span class="fu">filter</span>(immunogen_burden <span class="sc">==</span> <span class="st">&#39;Cas9-GFP&#39;</span> <span class="sc">&amp;</span> replicate <span class="sc">==</span> <span class="st">&#39;1&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb12-15"><a href="#cb12-15" tabindex="-1"></a>    <span class="fu">pull</span>(sample_alias)</span>
<span id="cb12-16"><a href="#cb12-16" tabindex="-1"></a>)</span>
<span id="cb12-17"><a href="#cb12-17" tabindex="-1"></a></span>
<span id="cb12-18"><a href="#cb12-18" tabindex="-1"></a></span>
<span id="cb12-19"><a href="#cb12-19" tabindex="-1"></a><span class="co"># Add loading control</span></span>
<span id="cb12-20"><a href="#cb12-20" tabindex="-1"></a>loading_control <span class="ot">&lt;-</span> <span class="st">&#39;Replacement_Sample&#39;</span></span>
<span id="cb12-21"><a href="#cb12-21" tabindex="-1"></a>muller_groups_selected <span class="ot">&lt;-</span> <span class="fu">map</span>(muller_groups_selected, <span class="cf">function</span>(x) <span class="fu">c</span>(loading_control, x))</span></code></pre></div>
<p>Generate muller plots</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a>cpm_threshold <span class="ot">&lt;-</span> <span class="dv">50000</span></span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a>use_df <span class="ot">&lt;-</span> <span class="fu">assay</span>(se, <span class="st">&#39;cpm&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>(<span class="at">check.names =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&#39;guide&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-5"><a href="#cb13-5" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb13-6"><a href="#cb13-6" tabindex="-1"></a>    <span class="at">gene =</span> <span class="fu">rowData</span>(se)<span class="sc">$</span>Gene,</span>
<span id="cb13-7"><a href="#cb13-7" tabindex="-1"></a>    <span class="at">gene =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&#39;Non_Target&#39;</span>, gene), <span class="st">&#39;Non_Target&#39;</span>, gene)</span>
<span id="cb13-8"><a href="#cb13-8" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb13-9"><a href="#cb13-9" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span><span class="fu">c</span>(guide, gene), <span class="at">names_to =</span> <span class="st">&#39;sample_alias&#39;</span>, <span class="at">values_to =</span> <span class="st">&#39;cpm&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-10"><a href="#cb13-10" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">colData</span>(se) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(sample_alias, sample_type)) <span class="sc">%&gt;%</span> </span>
<span id="cb13-11"><a href="#cb13-11" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb13-12"><a href="#cb13-12" tabindex="-1"></a>    <span class="at">sample_type =</span> <span class="fu">recode_factor</span>(</span>
<span id="cb13-13"><a href="#cb13-13" tabindex="-1"></a>        sample_type,</span>
<span id="cb13-14"><a href="#cb13-14" tabindex="-1"></a>        <span class="st">`</span><span class="at">Primary tumor</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;Primary</span><span class="sc">\n</span><span class="st">tumor&#39;</span>,</span>
<span id="cb13-15"><a href="#cb13-15" tabindex="-1"></a>        <span class="st">`</span><span class="at">Lymph nodes</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;Lymph</span><span class="sc">\n</span><span class="st">nodes&#39;</span></span>
<span id="cb13-16"><a href="#cb13-16" tabindex="-1"></a>      ),</span>
<span id="cb13-17"><a href="#cb13-17" tabindex="-1"></a>    <span class="at">sample_type =</span> <span class="fu">factor</span>(</span>
<span id="cb13-18"><a href="#cb13-18" tabindex="-1"></a>      <span class="fu">as.character</span>(sample_type),</span>
<span id="cb13-19"><a href="#cb13-19" tabindex="-1"></a>      <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&#39;Input&#39;</span>, <span class="st">&#39;Primary</span><span class="sc">\n</span><span class="st">tumor&#39;</span>, <span class="st">&#39;Lungs&#39;</span>, <span class="st">&#39;Lymph</span><span class="sc">\n</span><span class="st">nodes&#39;</span>, <span class="st">&#39;Liver&#39;</span>)</span>
<span id="cb13-20"><a href="#cb13-20" tabindex="-1"></a>    )</span>
<span id="cb13-21"><a href="#cb13-21" tabindex="-1"></a>  )</span>
<span id="cb13-22"><a href="#cb13-22" tabindex="-1"></a></span>
<span id="cb13-23"><a href="#cb13-23" tabindex="-1"></a>i <span class="ot">&lt;-</span> <span class="fu">names</span>(muller_groups_selected)[<span class="dv">1</span>]</span>
<span id="cb13-24"><a href="#cb13-24" tabindex="-1"></a>muller_plots <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i =</span> <span class="fu">names</span>(muller_groups_selected)) <span class="sc">%do%</span> {</span>
<span id="cb13-25"><a href="#cb13-25" tabindex="-1"></a>  j <span class="ot">&lt;-</span>  muller_groups_selected[[i]]</span>
<span id="cb13-26"><a href="#cb13-26" tabindex="-1"></a>  </span>
<span id="cb13-27"><a href="#cb13-27" tabindex="-1"></a>  x <span class="ot">&lt;-</span> use_df <span class="sc">%&gt;%</span> </span>
<span id="cb13-28"><a href="#cb13-28" tabindex="-1"></a>    <span class="fu">filter</span>(sample_alias <span class="sc">%in%</span> j) <span class="sc">%&gt;%</span> </span>
<span id="cb13-29"><a href="#cb13-29" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb13-30"><a href="#cb13-30" tabindex="-1"></a>      <span class="at">sample_alias =</span> <span class="fu">factor</span>(sample_alias, j)</span>
<span id="cb13-31"><a href="#cb13-31" tabindex="-1"></a>    )</span>
<span id="cb13-32"><a href="#cb13-32" tabindex="-1"></a>  x_guides_keep <span class="ot">&lt;-</span> x <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample_alias, sample_type, guide) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">cpm =</span> <span class="fu">max</span>(cpm)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(cpm <span class="sc">&gt;</span> cpm_threshold) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(guide)</span>
<span id="cb13-33"><a href="#cb13-33" tabindex="-1"></a>  </span>
<span id="cb13-34"><a href="#cb13-34" tabindex="-1"></a>  low_abundant_tag <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&#39;&lt;&#39;</span>, cpm_threshold, <span class="st">&#39;cpm&#39;</span>)</span>
<span id="cb13-35"><a href="#cb13-35" tabindex="-1"></a>  x <span class="sc">%&lt;&gt;%</span> </span>
<span id="cb13-36"><a href="#cb13-36" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb13-37"><a href="#cb13-37" tabindex="-1"></a>      <span class="at">guide =</span> <span class="fu">ifelse</span>(guide <span class="sc">%in%</span>  x_guides_keep, guide, low_abundant_tag)</span>
<span id="cb13-38"><a href="#cb13-38" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb13-39"><a href="#cb13-39" tabindex="-1"></a>    <span class="fu">group_by</span>(sample_alias, sample_type, guide) <span class="sc">%&gt;%</span></span>
<span id="cb13-40"><a href="#cb13-40" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">cpm =</span> <span class="fu">sum</span>(cpm)) <span class="sc">%&gt;%</span> </span>
<span id="cb13-41"><a href="#cb13-41" tabindex="-1"></a>    <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb13-42"><a href="#cb13-42" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb13-43"><a href="#cb13-43" tabindex="-1"></a>      <span class="at">guide =</span> <span class="fu">fct_reorder</span>(guide, cpm),</span>
<span id="cb13-44"><a href="#cb13-44" tabindex="-1"></a>      <span class="at">guide =</span> <span class="fu">relevel</span>(guide, <span class="at">ref =</span> low_abundant_tag)</span>
<span id="cb13-45"><a href="#cb13-45" tabindex="-1"></a>    )</span>
<span id="cb13-46"><a href="#cb13-46" tabindex="-1"></a>  </span>
<span id="cb13-47"><a href="#cb13-47" tabindex="-1"></a>  use_cols <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb13-48"><a href="#cb13-48" tabindex="-1"></a>    <span class="st">&#39;grey80&#39;</span>,</span>
<span id="cb13-49"><a href="#cb13-49" tabindex="-1"></a>    <span class="fu">colorRampPalette</span>(<span class="fu">rev</span>(palette_OkabeIto[<span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>]))(<span class="fu">nlevels</span>(x<span class="sc">$</span>guide) <span class="sc">-</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="fu">set_names</span>(<span class="fu">levels</span>(x<span class="sc">$</span>guide)[<span class="sc">-</span><span class="dv">1</span>])</span>
<span id="cb13-50"><a href="#cb13-50" tabindex="-1"></a>  )</span>
<span id="cb13-51"><a href="#cb13-51" tabindex="-1"></a>  <span class="fu">names</span>(use_cols)[<span class="dv">1</span>] <span class="ot">&lt;-</span> low_abundant_tag</span>
<span id="cb13-52"><a href="#cb13-52" tabindex="-1"></a>  </span>
<span id="cb13-53"><a href="#cb13-53" tabindex="-1"></a>  x <span class="sc">%&gt;%</span> </span>
<span id="cb13-54"><a href="#cb13-54" tabindex="-1"></a>    <span class="fu">ggplot</span>( <span class="fu">aes</span>(<span class="at">x =</span> sample_type, <span class="at">y =</span> cpm<span class="sc">/</span><span class="dv">10000</span>, <span class="at">group =</span> guide, <span class="at">fill =</span> guide)) <span class="sc">+</span> </span>
<span id="cb13-55"><a href="#cb13-55" tabindex="-1"></a>    <span class="fu">geom_area</span>(<span class="at">colour =</span> <span class="fu">alpha</span>(<span class="st">&quot;white&quot;</span>, <span class="fl">0.1</span>), <span class="at">linewidth =</span> <span class="fl">0.08</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb13-56"><a href="#cb13-56" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> use_cols, <span class="at">guide =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb13-57"><a href="#cb13-57" tabindex="-1"></a>    <span class="fu">labs</span>(</span>
<span id="cb13-58"><a href="#cb13-58" tabindex="-1"></a>      <span class="at">x =</span> <span class="st">&#39;&#39;</span>,</span>
<span id="cb13-59"><a href="#cb13-59" tabindex="-1"></a>      <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">&quot;CPM x 10&quot;</span><span class="sc">^-</span><span class="dv">4</span>),</span>
<span id="cb13-60"><a href="#cb13-60" tabindex="-1"></a>      <span class="at">fill =</span> <span class="st">&#39;&#39;</span>,</span>
<span id="cb13-61"><a href="#cb13-61" tabindex="-1"></a>      <span class="at">title =</span> i</span>
<span id="cb13-62"><a href="#cb13-62" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb13-63"><a href="#cb13-63" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb13-64"><a href="#cb13-64" tabindex="-1"></a>      <span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="dv">0</span>, <span class="fl">0.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="st">&quot;cm&quot;</span>),</span>
<span id="cb13-65"><a href="#cb13-65" tabindex="-1"></a>      <span class="at">axis.text =</span>  <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">6</span>),</span>
<span id="cb13-66"><a href="#cb13-66" tabindex="-1"></a>      <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">6</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>)</span>
<span id="cb13-67"><a href="#cb13-67" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb13-68"><a href="#cb13-68" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">fill =</span> <span class="st">&#39;none&#39;</span>) <span class="sc">+</span></span>
<span id="cb13-69"><a href="#cb13-69" tabindex="-1"></a>    <span class="fu">scale_x_discrete</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">guide =</span> <span class="fu">guide_axis</span>(<span class="at">angle =</span> <span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb13-70"><a href="#cb13-70" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>))</span>
<span id="cb13-71"><a href="#cb13-71" tabindex="-1"></a>}</span>
<span id="cb13-72"><a href="#cb13-72" tabindex="-1"></a><span class="fu">names</span>(muller_plots) <span class="ot">&lt;-</span> <span class="fu">names</span>(muller_groups_selected)</span></code></pre></div>
<div id="figure-5h-muller-plots-for-hunsgs" class="section level3">
<h3>Figure 5H: Muller plots for huNSGs</h3>
<p>Muller plots showing the relative frequency (CPM) of unique sgRNAs in
the indicated samples. Guides with CPM ≥ 5^{4} are colored. Individual
guides with CPM &lt; 5^{4} are grouped in a grey area.</p>
<p>Muller plots for four huNSGs, representative of data groups (G),
showing the relative frequency of unique sgRNAs in matched primary
tumors versus synchronous metastasis-relevant sites; cpm, read counts
per million.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="fu">wrap_plots</span>(muller_plots, <span class="at">nrow =</span> <span class="dv">1</span>)</span></code></pre></div>
<p><img src="figure/crispr-hsapiens_2180_sgRNA.Rmd/muller-plots-1.png" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="sgrna-distribution-by-condition-primary-tumor"
class="section level2">
<h2>sgRNA distribution by condition Primary tumor</h2>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a>use_sample_type <span class="ot">&lt;-</span> <span class="st">&#39;Primary tumor&#39;</span></span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a>use_colData <span class="ot">&lt;-</span> <span class="fu">colData</span>(se) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span></span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a>  <span class="fu">filter</span>(sample_type <span class="sc">==</span> use_sample_type)</span></code></pre></div>
<div
id="figure-6b-sgrna-distribution-of-in-mice-transplanted-with-cells-subjected-to-stealthy-ko"
class="section level3">
<h3>Figure 6B: sgRNA distribution of in mice transplanted with cells
subjected to StealTHY KO</h3>
<p>Density plots of unique sgRNAs retrieved in MDA-MB-231 primary tumors
(n≥6) under StealTHY KO; frequency(cpm+1)=read counts per million.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a>sample_ord <span class="ot">&lt;-</span> <span class="fu">rev</span>(<span class="fu">c</span>(</span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a>  <span class="st">&#39;C1Position_HumanPlate2&#39;</span>,</span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a>  <span class="st">&#39;C2Position_HumanPlate2&#39;</span>,</span>
<span id="cb16-4"><a href="#cb16-4" tabindex="-1"></a>  <span class="st">&#39;E2Position_HumanPlate2&#39;</span>,</span>
<span id="cb16-5"><a href="#cb16-5" tabindex="-1"></a>  <span class="st">&#39;E1Position_HumanPlate2&#39;</span>,</span>
<span id="cb16-6"><a href="#cb16-6" tabindex="-1"></a>  <span class="st">&#39;G1Position_HumanPlate2&#39;</span>,</span>
<span id="cb16-7"><a href="#cb16-7" tabindex="-1"></a>  </span>
<span id="cb16-8"><a href="#cb16-8" tabindex="-1"></a>  <span class="st">&#39;C4Position_HumanPlate2&#39;</span>,</span>
<span id="cb16-9"><a href="#cb16-9" tabindex="-1"></a>  <span class="st">&#39;C3Position_HumanPlate2&#39;</span>,</span>
<span id="cb16-10"><a href="#cb16-10" tabindex="-1"></a>  <span class="st">&#39;G3Position_HumanPlate2&#39;</span>,</span>
<span id="cb16-11"><a href="#cb16-11" tabindex="-1"></a>  <span class="st">&#39;E3Position_HumanPlate2&#39;</span>,</span>
<span id="cb16-12"><a href="#cb16-12" tabindex="-1"></a>  <span class="st">&#39;A4Position_HumanPlate2&#39;</span></span>
<span id="cb16-13"><a href="#cb16-13" tabindex="-1"></a>  ))</span>
<span id="cb16-14"><a href="#cb16-14" tabindex="-1"></a></span>
<span id="cb16-15"><a href="#cb16-15" tabindex="-1"></a>use_assay <span class="ot">&lt;-</span> <span class="fu">assay</span>(se[,sample_ord], <span class="st">&#39;cpm&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-16"><a href="#cb16-16" tabindex="-1"></a>  <span class="fu">as.data.frame</span>(<span class="at">check.names =</span> F) <span class="sc">%&gt;%</span> </span>
<span id="cb16-17"><a href="#cb16-17" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&#39;guide&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-18"><a href="#cb16-18" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>guide, <span class="at">names_to =</span> <span class="st">&#39;sample_alias&#39;</span>, <span class="at">values_to =</span> <span class="st">&#39;cpm&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-19"><a href="#cb16-19" tabindex="-1"></a>  <span class="fu">left_join</span>(use_colData) <span class="sc">%&gt;%</span> </span>
<span id="cb16-20"><a href="#cb16-20" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb16-21"><a href="#cb16-21" tabindex="-1"></a>    <span class="at">sample_alias =</span> <span class="fu">factor</span>(sample_alias, <span class="at">levels =</span> sample_ord)</span>
<span id="cb16-22"><a href="#cb16-22" tabindex="-1"></a>    )</span>
<span id="cb16-23"><a href="#cb16-23" tabindex="-1"></a></span>
<span id="cb16-24"><a href="#cb16-24" tabindex="-1"></a></span>
<span id="cb16-25"><a href="#cb16-25" tabindex="-1"></a>use_assay <span class="sc">%&gt;%</span> </span>
<span id="cb16-26"><a href="#cb16-26" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>cpm <span class="sc">+</span> <span class="dv">1</span>, <span class="at">y =</span> sample_alias, </span>
<span id="cb16-27"><a href="#cb16-27" tabindex="-1"></a>             <span class="at">fill =</span> mouse_model_short, </span>
<span id="cb16-28"><a href="#cb16-28" tabindex="-1"></a>             <span class="at">height =</span> <span class="fu">after_stat</span>(density))) <span class="sc">+</span></span>
<span id="cb16-29"><a href="#cb16-29" tabindex="-1"></a>  <span class="fu">geom_density_ridges</span>(<span class="at">color =</span> <span class="st">&#39;black&#39;</span>,</span>
<span id="cb16-30"><a href="#cb16-30" tabindex="-1"></a>                      <span class="at">linewidth =</span> default_linewidth, </span>
<span id="cb16-31"><a href="#cb16-31" tabindex="-1"></a>                      <span class="at">alpha =</span> <span class="fl">0.8</span>, </span>
<span id="cb16-32"><a href="#cb16-32" tabindex="-1"></a>                      <span class="at">scale =</span> <span class="dv">4</span>,</span>
<span id="cb16-33"><a href="#cb16-33" tabindex="-1"></a>                      <span class="at">rel_min_height =</span> <span class="fl">0.001</span> <span class="co"># set the `rel_min_height` argument to remove tails,</span></span>
<span id="cb16-34"><a href="#cb16-34" tabindex="-1"></a>                      ) <span class="sc">+</span></span>
<span id="cb16-35"><a href="#cb16-35" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> palette_hu_stealthy_ko) <span class="sc">+</span></span>
<span id="cb16-36"><a href="#cb16-36" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>))) <span class="sc">+</span></span>
<span id="cb16-37"><a href="#cb16-37" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="fl">0.01</span>, <span class="fl">0.05</span>))) <span class="sc">+</span></span>
<span id="cb16-38"><a href="#cb16-38" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&#39;&#39;</span>) <span class="sc">+</span></span>
<span id="cb16-39"><a href="#cb16-39" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill =</span> <span class="st">&#39;none&#39;</span>, <span class="at">color =</span> <span class="st">&#39;none&#39;</span>) <span class="sc">+</span></span>
<span id="cb16-40"><a href="#cb16-40" tabindex="-1"></a>  <span class="fu">theme_ridges</span>(<span class="at">font_size =</span> <span class="dv">6</span>, <span class="at">grid =</span> F) <span class="sc">+</span></span>
<span id="cb16-41"><a href="#cb16-41" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb16-42"><a href="#cb16-42" tabindex="-1"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb16-43"><a href="#cb16-43" tabindex="-1"></a>    <span class="at">axis.line.x =</span> <span class="fu">element_line</span>(<span class="at">linewidth =</span> default_linewidth, <span class="at">color =</span> <span class="st">&#39;black&#39;</span>),</span>
<span id="cb16-44"><a href="#cb16-44" tabindex="-1"></a>    <span class="at">axis.ticks.x =</span> <span class="fu">element_line</span>(<span class="at">linewidth =</span> default_linewidth, <span class="at">color =</span> <span class="st">&#39;black&#39;</span>),</span>
<span id="cb16-45"><a href="#cb16-45" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>()</span>
<span id="cb16-46"><a href="#cb16-46" tabindex="-1"></a>  )</span></code></pre></div>
<p><img src="figure/crispr-hsapiens_2180_sgRNA.Rmd/test-dist-tumor-distribution-ridges_plot--ic-id-1.png" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="differential-abundance-analysis" class="section level2">
<h2>Differential abundance analysis</h2>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a>gene_info <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(<span class="st">&#39;./data/resources/selected_gene_list/gene_list_for_figure_7d.xlsx&#39;</span>)</span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a></span>
<span id="cb17-3"><a href="#cb17-3" tabindex="-1"></a>use_colors <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb17-4"><a href="#cb17-4" tabindex="-1"></a>  <span class="st">`</span><span class="at">Immune system cytokines and chemokines</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;grey&#39;</span>,</span>
<span id="cb17-5"><a href="#cb17-5" tabindex="-1"></a>  <span class="st">`</span><span class="at">Immune system cytokines</span><span class="sc">\n</span><span class="at">and chemokines</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;grey&#39;</span>,</span>
<span id="cb17-6"><a href="#cb17-6" tabindex="-1"></a>  </span>
<span id="cb17-7"><a href="#cb17-7" tabindex="-1"></a>  <span class="st">`</span><span class="at">Interferone, TNF and danger/death-inducing signals</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;black&#39;</span>,</span>
<span id="cb17-8"><a href="#cb17-8" tabindex="-1"></a>  <span class="st">`</span><span class="at">Interferon, TNF</span><span class="sc">\n</span><span class="at">and danger/death-inducing</span><span class="sc">\n</span><span class="at">signals</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;black&#39;</span>,</span>
<span id="cb17-9"><a href="#cb17-9" tabindex="-1"></a>  <span class="st">`</span><span class="at">Interferon, TNF and</span><span class="sc">\n</span><span class="at">danger/death-inducing</span><span class="sc">\n</span><span class="at">signals</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;black&#39;</span>,</span>
<span id="cb17-10"><a href="#cb17-10" tabindex="-1"></a>  <span class="st">`</span><span class="at">TGF-β family related</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;red&#39;</span>,</span>
<span id="cb17-11"><a href="#cb17-11" tabindex="-1"></a>  </span>
<span id="cb17-12"><a href="#cb17-12" tabindex="-1"></a>  <span class="st">`</span><span class="at">RTK signaling</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;blue&#39;</span>,</span>
<span id="cb17-13"><a href="#cb17-13" tabindex="-1"></a>  <span class="st">`</span><span class="at">Positive controls</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;black&#39;</span></span>
<span id="cb17-14"><a href="#cb17-14" tabindex="-1"></a>)</span>
<span id="cb17-15"><a href="#cb17-15" tabindex="-1"></a></span>
<span id="cb17-16"><a href="#cb17-16" tabindex="-1"></a>use_fill <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb17-17"><a href="#cb17-17" tabindex="-1"></a>  <span class="st">`</span><span class="at">Immune system cytokines and chemokines</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;grey&#39;</span>,</span>
<span id="cb17-18"><a href="#cb17-18" tabindex="-1"></a>  <span class="st">`</span><span class="at">Immune system cytokines</span><span class="sc">\n</span><span class="at">and chemokines</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;grey&#39;</span>,</span>
<span id="cb17-19"><a href="#cb17-19" tabindex="-1"></a>  </span>
<span id="cb17-20"><a href="#cb17-20" tabindex="-1"></a>  <span class="st">`</span><span class="at">Interferone, TNF and danger/death-inducing signals</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;black&#39;</span>,</span>
<span id="cb17-21"><a href="#cb17-21" tabindex="-1"></a>  <span class="st">`</span><span class="at">Interferon, TNF</span><span class="sc">\n</span><span class="at">and danger/death-inducing</span><span class="sc">\n</span><span class="at">signals</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;black&#39;</span>,</span>
<span id="cb17-22"><a href="#cb17-22" tabindex="-1"></a>  <span class="st">`</span><span class="at">Interferon, TNF and</span><span class="sc">\n</span><span class="at">danger/death-inducing</span><span class="sc">\n</span><span class="at">signals</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;black&#39;</span>,</span>
<span id="cb17-23"><a href="#cb17-23" tabindex="-1"></a>  <span class="st">`</span><span class="at">TGF-β family related</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;red&#39;</span>,</span>
<span id="cb17-24"><a href="#cb17-24" tabindex="-1"></a>  </span>
<span id="cb17-25"><a href="#cb17-25" tabindex="-1"></a>  <span class="st">`</span><span class="at">RTK signaling</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;blue&#39;</span>,</span>
<span id="cb17-26"><a href="#cb17-26" tabindex="-1"></a>  <span class="st">`</span><span class="at">Positive controls</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;white&#39;</span></span>
<span id="cb17-27"><a href="#cb17-27" tabindex="-1"></a>)</span>
<span id="cb17-28"><a href="#cb17-28" tabindex="-1"></a></span>
<span id="cb17-29"><a href="#cb17-29" tabindex="-1"></a>use_shape <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb17-30"><a href="#cb17-30" tabindex="-1"></a>  <span class="st">`</span><span class="at">Immune system cytokines and chemokines</span><span class="st">`</span> <span class="ot">=</span> <span class="dv">16</span>,</span>
<span id="cb17-31"><a href="#cb17-31" tabindex="-1"></a>  <span class="st">`</span><span class="at">Immune system cytokines</span><span class="sc">\n</span><span class="at">and chemokines</span><span class="st">`</span> <span class="ot">=</span> <span class="dv">16</span>,</span>
<span id="cb17-32"><a href="#cb17-32" tabindex="-1"></a>  </span>
<span id="cb17-33"><a href="#cb17-33" tabindex="-1"></a>  <span class="st">`</span><span class="at">Interferone, TNF and danger/death-inducing signals</span><span class="st">`</span> <span class="ot">=</span> <span class="dv">16</span>,</span>
<span id="cb17-34"><a href="#cb17-34" tabindex="-1"></a>  <span class="st">`</span><span class="at">Interferon, TNF</span><span class="sc">\n</span><span class="at">and danger/death-inducing</span><span class="sc">\n</span><span class="at">signals</span><span class="st">`</span> <span class="ot">=</span> <span class="dv">16</span>,</span>
<span id="cb17-35"><a href="#cb17-35" tabindex="-1"></a>  <span class="st">`</span><span class="at">Interferon, TNF and</span><span class="sc">\n</span><span class="at">danger/death-inducing</span><span class="sc">\n</span><span class="at">signals</span><span class="st">`</span> <span class="ot">=</span> <span class="dv">16</span>,</span>
<span id="cb17-36"><a href="#cb17-36" tabindex="-1"></a>  <span class="st">`</span><span class="at">TGF-β family related</span><span class="st">`</span> <span class="ot">=</span> <span class="dv">16</span>,</span>
<span id="cb17-37"><a href="#cb17-37" tabindex="-1"></a>  </span>
<span id="cb17-38"><a href="#cb17-38" tabindex="-1"></a>  <span class="st">`</span><span class="at">RTK signaling</span><span class="st">`</span> <span class="ot">=</span> <span class="dv">16</span>,</span>
<span id="cb17-39"><a href="#cb17-39" tabindex="-1"></a>  <span class="st">`</span><span class="at">Positive controls</span><span class="st">`</span> <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb17-40"><a href="#cb17-40" tabindex="-1"></a>)</span>
<span id="cb17-41"><a href="#cb17-41" tabindex="-1"></a></span>
<span id="cb17-42"><a href="#cb17-42" tabindex="-1"></a><span class="co"># get gene order based on LFC in StealTHY_IC</span></span>
<span id="cb17-43"><a href="#cb17-43" tabindex="-1"></a>i <span class="ot">&lt;-</span> <span class="st">&#39;stealthy_hunsg&#39;</span></span>
<span id="cb17-44"><a href="#cb17-44" tabindex="-1"></a>gene_levels <span class="ot">&lt;-</span> gene_summ[[i]] <span class="sc">%&gt;%</span> </span>
<span id="cb17-45"><a href="#cb17-45" tabindex="-1"></a>  <span class="fu">ReadRRA</span>(<span class="at">score =</span> <span class="st">&#39;lfc&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-46"><a href="#cb17-46" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">LFC =</span> Score) <span class="sc">%&gt;%</span> </span>
<span id="cb17-47"><a href="#cb17-47" tabindex="-1"></a>  <span class="fu">left_join</span>(gene_info, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&#39;id&#39;</span> <span class="ot">=</span> <span class="st">&#39;id&#39;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-48"><a href="#cb17-48" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(gene_set)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-49"><a href="#cb17-49" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">gene_name =</span> <span class="fu">fct_reorder</span>(gene_name, LFC)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-50"><a href="#cb17-50" tabindex="-1"></a>  <span class="fu">pull</span>(gene_name) <span class="sc">%&gt;%</span> </span>
<span id="cb17-51"><a href="#cb17-51" tabindex="-1"></a>  levels</span></code></pre></div>
<div
id="figure-6c-volcano-plots-showing-genes-lungs-over-primary-tumor-in-stealthy-ko"
class="section level3">
<h3>Figure 6C: Volcano plots showing genes Lungs over Primary Tumor in
StealTHY KO</h3>
<p>Volcano plot of the StealTHY KO screen in huNSGs (n=10 mice, n=1
human HLA-matched donor). Genes referenced in the main text and/or
analyzed in Figure 6E–F are highlighted.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a>lfc_colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">none =</span> <span class="st">&quot;gray80&quot;</span>, <span class="at">up =</span> <span class="st">&quot;#e41a1c&quot;</span>, <span class="at">down =</span> <span class="st">&quot;#377eb8&quot;</span>)</span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a>i <span class="ot">&lt;-</span> <span class="st">&#39;stealthy_hunsg&#39;</span></span>
<span id="cb18-3"><a href="#cb18-3" tabindex="-1"></a></span>
<span id="cb18-4"><a href="#cb18-4" tabindex="-1"></a>x <span class="ot">&lt;-</span> gene_summ[[i]] <span class="sc">%&gt;%</span> </span>
<span id="cb18-5"><a href="#cb18-5" tabindex="-1"></a>  <span class="fu">ReadRRA</span>(<span class="at">score =</span> <span class="st">&#39;lfc&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb18-6"><a href="#cb18-6" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">LFC =</span> Score) <span class="sc">%&gt;%</span> </span>
<span id="cb18-7"><a href="#cb18-7" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb18-8"><a href="#cb18-8" tabindex="-1"></a>    <span class="at">direction =</span> <span class="fu">ifelse</span>(FDR <span class="sc">&lt;=</span> <span class="fl">0.10</span> <span class="sc">&amp;</span> LFC <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">&#39;up&#39;</span>, <span class="st">&#39;none&#39;</span>),</span>
<span id="cb18-9"><a href="#cb18-9" tabindex="-1"></a>    <span class="at">direction =</span> <span class="fu">ifelse</span>(FDR <span class="sc">&lt;=</span> <span class="fl">0.10</span> <span class="sc">&amp;</span> LFC <span class="sc">&lt;</span> <span class="dv">0</span>, <span class="st">&#39;down&#39;</span>, direction),</span>
<span id="cb18-10"><a href="#cb18-10" tabindex="-1"></a>    <span class="at">is.significant =</span> FDR <span class="sc">&lt;=</span> <span class="fl">0.10</span>,</span>
<span id="cb18-11"><a href="#cb18-11" tabindex="-1"></a>    <span class="at">id =</span> <span class="fu">ifelse</span>(id <span class="sc">==</span> <span class="st">&#39;TNFRSF10B&#39;</span>, <span class="st">&#39;TRAIL-R2&#39;</span>, id)</span>
<span id="cb18-12"><a href="#cb18-12" tabindex="-1"></a>  )</span>
<span id="cb18-13"><a href="#cb18-13" tabindex="-1"></a>use_xlim <span class="ot">&lt;-</span> <span class="fu">xlim</span>(<span class="sc">-</span><span class="fu">max</span>(<span class="fu">abs</span>(x<span class="sc">$</span>LFC), <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="fu">max</span>(<span class="fu">abs</span>(x<span class="sc">$</span>LFC), <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb18-14"><a href="#cb18-14" tabindex="-1"></a></span>
<span id="cb18-15"><a href="#cb18-15" tabindex="-1"></a>genes_to_highlight <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;CCR10&#39;</span>, <span class="st">&#39;TRAIL-R2&#39;</span>, <span class="st">&#39;CCL27&#39;</span>, <span class="st">&#39;IGFBP1&#39;</span>, <span class="st">&#39;ANGPT2&#39;</span>, <span class="st">&#39;AMH&#39;</span>, <span class="st">&#39;IL34&#39;</span>,</span>
<span id="cb18-16"><a href="#cb18-16" tabindex="-1"></a>                   <span class="st">&#39;IFNA8&#39;</span>, <span class="st">&#39;TGFB2&#39;</span>, <span class="st">&#39;IL13RA2&#39;</span>, <span class="st">&#39;IFNAR1&#39;</span>, <span class="st">&#39;IL24&#39;</span>, <span class="st">&#39;IFNLR1&#39;</span>, <span class="st">&#39;PTEN&#39;</span>, <span class="st">&#39;GDF9&#39;</span>)</span>
<span id="cb18-17"><a href="#cb18-17" tabindex="-1"></a>data_text <span class="ot">&lt;-</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb18-18"><a href="#cb18-18" tabindex="-1"></a>  <span class="fu">filter</span>(id <span class="sc">%in%</span> genes_to_highlight)</span>
<span id="cb18-19"><a href="#cb18-19" tabindex="-1"></a></span>
<span id="cb18-20"><a href="#cb18-20" tabindex="-1"></a><span class="fu">ggplot</span>(x, <span class="fu">aes</span>(LFC, <span class="sc">-</span><span class="fu">log10</span>(FDR), <span class="at">color =</span> direction)) <span class="sc">+</span></span>
<span id="cb18-21"><a href="#cb18-21" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.10</span>), <span class="at">linetype =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb18-22"><a href="#cb18-22" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb18-23"><a href="#cb18-23" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">16</span>, <span class="at">alpha =</span> <span class="fl">0.6</span>, <span class="at">size =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb18-24"><a href="#cb18-24" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> lfc_colors) <span class="sc">+</span></span>
<span id="cb18-25"><a href="#cb18-25" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(</span>
<span id="cb18-26"><a href="#cb18-26" tabindex="-1"></a>    <span class="at">data =</span> data_text,</span>
<span id="cb18-27"><a href="#cb18-27" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">label =</span> id),</span>
<span id="cb18-28"><a href="#cb18-28" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">&#39;black&#39;</span>,</span>
<span id="cb18-29"><a href="#cb18-29" tabindex="-1"></a>    <span class="at">size =</span> <span class="fl">0.4</span><span class="sc">*</span>geom_text_size,</span>
<span id="cb18-30"><a href="#cb18-30" tabindex="-1"></a>    <span class="at">min.segment.length =</span> <span class="dv">0</span>,</span>
<span id="cb18-31"><a href="#cb18-31" tabindex="-1"></a>    <span class="at">segment.size  =</span> <span class="fl">0.2</span>,</span>
<span id="cb18-32"><a href="#cb18-32" tabindex="-1"></a>    <span class="at">box.padding =</span> <span class="fl">0.1</span>,</span>
<span id="cb18-33"><a href="#cb18-33" tabindex="-1"></a>    <span class="at">max.overlaps =</span> <span class="cn">Inf</span>, </span>
<span id="cb18-34"><a href="#cb18-34" tabindex="-1"></a>    <span class="at">fontface=</span><span class="st">&quot;italic&quot;</span></span>
<span id="cb18-35"><a href="#cb18-35" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb18-36"><a href="#cb18-36" tabindex="-1"></a>  use_xlim <span class="sc">+</span></span>
<span id="cb18-37"><a href="#cb18-37" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb18-38"><a href="#cb18-38" tabindex="-1"></a>    <span class="at">x =</span> <span class="fu">expression</span>(<span class="fu">paste</span>(<span class="st">&quot;lo&quot;</span>, g[<span class="dv">2</span>],<span class="st">&quot;(Fold change)&quot;</span>)), </span>
<span id="cb18-39"><a href="#cb18-39" tabindex="-1"></a>    <span class="at">y =</span> <span class="fu">expression</span>(<span class="fu">paste</span>(<span class="st">&quot;-lo&quot;</span>, g[<span class="dv">10</span>],<span class="st">&quot;(FDR)&quot;</span>))</span>
<span id="cb18-40"><a href="#cb18-40" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb18-41"><a href="#cb18-41" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color =</span> <span class="st">&#39;none&#39;</span>, <span class="at">size =</span> <span class="st">&#39;none&#39;</span>) <span class="sc">+</span></span>
<span id="cb18-42"><a href="#cb18-42" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb18-43"><a href="#cb18-43" tabindex="-1"></a>      <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">6</span>), </span>
<span id="cb18-44"><a href="#cb18-44" tabindex="-1"></a>      <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">6</span>)</span>
<span id="cb18-45"><a href="#cb18-45" tabindex="-1"></a>  )</span></code></pre></div>
<p><img src="figure/crispr-hsapiens_2180_sgRNA.Rmd/volcano-plot-stealthy-ko-pt-over-lung-1.png" style="display: block; margin: auto;" /></p>
</div>
<div
id="figure-6d-distribution-of-the-fold-change-of-lungs-over-primary-tumor-in-hunsg-and-nsg-host"
class="section level3">
<h3>Figure 6D: Distribution of the fold-change of Lungs over Primary
Tumor in huNSG and NSG host</h3>
<p>Violin plots showing the fold-change distribution of sgRNAs targeting
metastasis-relevant genes identified upon orthotopic transplantation of
MDA-MB-231 in huNSGs (n=10) and NSGs (n=6).</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a>i <span class="ot">&lt;-</span> <span class="fu">names</span>(gene_summ)[<span class="dv">1</span>]</span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a></span>
<span id="cb19-3"><a href="#cb19-3" tabindex="-1"></a>curate_gene_sets <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb19-4"><a href="#cb19-4" tabindex="-1"></a>  <span class="st">&#39;Immune system cytokines and chemokines&#39;</span>,</span>
<span id="cb19-5"><a href="#cb19-5" tabindex="-1"></a>  <span class="st">&#39;TGF-β family related&#39;</span>,</span>
<span id="cb19-6"><a href="#cb19-6" tabindex="-1"></a>  <span class="st">&#39;Positive controls&#39;</span></span>
<span id="cb19-7"><a href="#cb19-7" tabindex="-1"></a>) </span>
<span id="cb19-8"><a href="#cb19-8" tabindex="-1"></a></span>
<span id="cb19-9"><a href="#cb19-9" tabindex="-1"></a>levels_gene_sets <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb19-10"><a href="#cb19-10" tabindex="-1"></a>  <span class="st">&#39;Immune system cytokines</span><span class="sc">\n</span><span class="st">and chemokines&#39;</span>,</span>
<span id="cb19-11"><a href="#cb19-11" tabindex="-1"></a>  <span class="st">&#39;TGF-β family related&#39;</span>,</span>
<span id="cb19-12"><a href="#cb19-12" tabindex="-1"></a>  <span class="st">&#39;Positive controls&#39;</span></span>
<span id="cb19-13"><a href="#cb19-13" tabindex="-1"></a>) </span>
<span id="cb19-14"><a href="#cb19-14" tabindex="-1"></a></span>
<span id="cb19-15"><a href="#cb19-15" tabindex="-1"></a>curated_gene_list <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;IL24&#39;</span>, <span class="st">&#39;IL13RA2&#39;</span>, <span class="st">&#39;IL33&#39;</span>, <span class="st">&#39;IL17C&#39;</span>, <span class="st">&#39;IL17R&#39;</span>, <span class="st">&#39;CCL27&#39;</span>, </span>
<span id="cb19-16"><a href="#cb19-16" tabindex="-1"></a>                       <span class="st">&#39;GDF9&#39;</span>, <span class="st">&#39;TGFB2&#39;</span>, <span class="st">&#39;AMH&#39;</span>, </span>
<span id="cb19-17"><a href="#cb19-17" tabindex="-1"></a>                       <span class="st">&#39;PTEN&#39;</span>, <span class="st">&#39;LKB1&#39;</span>, <span class="st">&#39;MET&#39;</span>, <span class="st">&#39;EGFR&#39;</span>)</span>
<span id="cb19-18"><a href="#cb19-18" tabindex="-1"></a></span>
<span id="cb19-19"><a href="#cb19-19" tabindex="-1"></a>curated_gene_info <span class="ot">&lt;-</span> gene_info <span class="sc">%&gt;%</span> </span>
<span id="cb19-20"><a href="#cb19-20" tabindex="-1"></a>  <span class="fu">filter</span>(gene_set <span class="sc">%in%</span> curate_gene_sets) <span class="sc">%&gt;%</span> </span>
<span id="cb19-21"><a href="#cb19-21" tabindex="-1"></a>  <span class="fu">filter</span>(gene_name <span class="sc">%in%</span> curated_gene_list) </span>
<span id="cb19-22"><a href="#cb19-22" tabindex="-1"></a></span>
<span id="cb19-23"><a href="#cb19-23" tabindex="-1"></a></span>
<span id="cb19-24"><a href="#cb19-24" tabindex="-1"></a>plot_data <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i=</span><span class="fu">names</span>(gene_summ), <span class="at">.combine =</span> rbind) <span class="sc">%do%</span> {</span>
<span id="cb19-25"><a href="#cb19-25" tabindex="-1"></a>  xsg <span class="ot">&lt;-</span> sgrna_summ[[i]] <span class="sc">%&gt;%</span> </span>
<span id="cb19-26"><a href="#cb19-26" tabindex="-1"></a>     <span class="fu">select</span>(Gene, sgrna, LFC, FDR) <span class="sc">%&gt;%</span> </span>
<span id="cb19-27"><a href="#cb19-27" tabindex="-1"></a>     dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">sgLFC =</span> LFC, <span class="at">sgFDR =</span> FDR)</span>
<span id="cb19-28"><a href="#cb19-28" tabindex="-1"></a></span>
<span id="cb19-29"><a href="#cb19-29" tabindex="-1"></a>  gene_summ[[i]] <span class="sc">%&gt;%</span> </span>
<span id="cb19-30"><a href="#cb19-30" tabindex="-1"></a>    <span class="fu">ReadRRA</span>(<span class="at">score =</span> <span class="st">&#39;lfc&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-31"><a href="#cb19-31" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">LFC =</span> Score) <span class="sc">%&gt;%</span> </span>
<span id="cb19-32"><a href="#cb19-32" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb19-33"><a href="#cb19-33" tabindex="-1"></a>      <span class="at">direction =</span> <span class="fu">ifelse</span>(FDR <span class="sc">&lt;=</span> <span class="fl">0.10</span> <span class="sc">&amp;</span> LFC <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">&#39;up&#39;</span>, <span class="st">&#39;none&#39;</span>),</span>
<span id="cb19-34"><a href="#cb19-34" tabindex="-1"></a>      <span class="at">direction =</span> <span class="fu">ifelse</span>(FDR <span class="sc">&lt;=</span> <span class="fl">0.10</span> <span class="sc">&amp;</span> LFC <span class="sc">&lt;</span> <span class="dv">0</span>, <span class="st">&#39;down&#39;</span>, direction),</span>
<span id="cb19-35"><a href="#cb19-35" tabindex="-1"></a>      <span class="at">is.significant =</span> FDR <span class="sc">&lt;=</span> <span class="fl">0.10</span>,</span>
<span id="cb19-36"><a href="#cb19-36" tabindex="-1"></a>      <span class="at">comparison =</span> i</span>
<span id="cb19-37"><a href="#cb19-37" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb19-38"><a href="#cb19-38" tabindex="-1"></a>    <span class="fu">left_join</span>(curated_gene_info, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&#39;id&#39;</span> <span class="ot">=</span> <span class="st">&#39;id&#39;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-39"><a href="#cb19-39" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(gene_set)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-40"><a href="#cb19-40" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb19-41"><a href="#cb19-41" tabindex="-1"></a>      <span class="at">gene_name =</span> <span class="fu">factor</span>(gene_name, <span class="at">levels =</span> curated_gene_list),</span>
<span id="cb19-42"><a href="#cb19-42" tabindex="-1"></a>      <span class="at">gene_set =</span> <span class="fu">ifelse</span>(</span>
<span id="cb19-43"><a href="#cb19-43" tabindex="-1"></a>        gene_set <span class="sc">==</span> <span class="st">&#39;Immune system cytokines and chemokines&#39;</span>,</span>
<span id="cb19-44"><a href="#cb19-44" tabindex="-1"></a>        <span class="st">&#39;Immune system cytokines</span><span class="sc">\n</span><span class="st">and chemokines&#39;</span>,</span>
<span id="cb19-45"><a href="#cb19-45" tabindex="-1"></a>        gene_set</span>
<span id="cb19-46"><a href="#cb19-46" tabindex="-1"></a>      ),</span>
<span id="cb19-47"><a href="#cb19-47" tabindex="-1"></a>      <span class="at">gene_set =</span> <span class="fu">factor</span>(gene_set, levels_gene_sets)</span>
<span id="cb19-48"><a href="#cb19-48" tabindex="-1"></a>      ) <span class="sc">%&gt;%</span> </span>
<span id="cb19-49"><a href="#cb19-49" tabindex="-1"></a>    <span class="fu">left_join</span>(xsg, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&#39;id&#39;</span> <span class="ot">=</span> <span class="st">&#39;Gene&#39;</span>), <span class="at">relationship =</span> <span class="st">&quot;many-to-many&quot;</span>)</span>
<span id="cb19-50"><a href="#cb19-50" tabindex="-1"></a>  </span>
<span id="cb19-51"><a href="#cb19-51" tabindex="-1"></a>}</span>
<span id="cb19-52"><a href="#cb19-52" tabindex="-1"></a></span>
<span id="cb19-53"><a href="#cb19-53" tabindex="-1"></a>plot_data <span class="sc">%&lt;&gt;%</span> </span>
<span id="cb19-54"><a href="#cb19-54" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb19-55"><a href="#cb19-55" tabindex="-1"></a>    <span class="at">mouse_model_short =</span> <span class="fu">case_match</span>(comparison, </span>
<span id="cb19-56"><a href="#cb19-56" tabindex="-1"></a>                            <span class="st">&#39;stealthy_hunsg&#39;</span> <span class="sc">~</span> <span class="st">&#39;huNSG&#39;</span>, </span>
<span id="cb19-57"><a href="#cb19-57" tabindex="-1"></a>                            <span class="st">&#39;stealthy_nsg&#39;</span> <span class="sc">~</span> <span class="st">&#39;NSG&#39;</span>),</span>
<span id="cb19-58"><a href="#cb19-58" tabindex="-1"></a>    <span class="at">mouse_model_short =</span> <span class="fu">factor</span>(mouse_model_short,</span>
<span id="cb19-59"><a href="#cb19-59" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&#39;NSG&#39;</span>, <span class="st">&#39;huNSG&#39;</span>))</span>
<span id="cb19-60"><a href="#cb19-60" tabindex="-1"></a>  )</span>
<span id="cb19-61"><a href="#cb19-61" tabindex="-1"></a></span>
<span id="cb19-62"><a href="#cb19-62" tabindex="-1"></a>xlim <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fu">ceiling</span>(<span class="fu">max</span>(<span class="fu">abs</span>(plot_data<span class="sc">$</span>LFC), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)), </span>
<span id="cb19-63"><a href="#cb19-63" tabindex="-1"></a>          <span class="fu">ceiling</span>(<span class="fu">max</span>(<span class="fu">abs</span>(plot_data<span class="sc">$</span>LFC), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)))</span>
<span id="cb19-64"><a href="#cb19-64" tabindex="-1"></a></span>
<span id="cb19-65"><a href="#cb19-65" tabindex="-1"></a>plot_data <span class="sc">%&gt;%</span> </span>
<span id="cb19-66"><a href="#cb19-66" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y =</span> sgLFC, <span class="at">x =</span> gene_name, <span class="at">fill =</span> mouse_model_short)) <span class="sc">+</span></span>
<span id="cb19-67"><a href="#cb19-67" tabindex="-1"></a>  <span class="fu">geom_split_violin</span>(</span>
<span id="cb19-68"><a href="#cb19-68" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">y =</span> sgLFC, <span class="at">x =</span> gene_name, <span class="at">fill =</span> mouse_model_short),</span>
<span id="cb19-69"><a href="#cb19-69" tabindex="-1"></a>    <span class="at">alpha =</span> <span class="fl">0.8</span>,</span>
<span id="cb19-70"><a href="#cb19-70" tabindex="-1"></a>    <span class="at">draw_quantiles =</span> <span class="fl">0.5</span>,</span>
<span id="cb19-71"><a href="#cb19-71" tabindex="-1"></a>    <span class="at">linewidth =</span> one_pt<span class="sc">/</span><span class="dv">2</span>,</span>
<span id="cb19-72"><a href="#cb19-72" tabindex="-1"></a>    <span class="at">scale =</span> <span class="st">&#39;width&#39;</span></span>
<span id="cb19-73"><a href="#cb19-73" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb19-74"><a href="#cb19-74" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb19-75"><a href="#cb19-75" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="dv">3</span>,<span class="at">linewidth =</span> one_pt<span class="sc">/</span><span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb19-76"><a href="#cb19-76" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">add =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>))) <span class="sc">+</span></span>
<span id="cb19-77"><a href="#cb19-77" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> palette_hu_stealthy_ko) <span class="sc">+</span></span>
<span id="cb19-78"><a href="#cb19-78" tabindex="-1"></a>  <span class="fu">ylim</span>(xlim) <span class="sc">+</span></span>
<span id="cb19-79"><a href="#cb19-79" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="at">rows =</span> <span class="fu">vars</span>(gene_set), <span class="at">scales =</span> <span class="st">&#39;free_y&#39;</span>, <span class="at">space =</span> <span class="st">&#39;free_y&#39;</span>)<span class="sc">+</span></span>
<span id="cb19-80"><a href="#cb19-80" tabindex="-1"></a>  theme_facet <span class="sc">+</span></span>
<span id="cb19-81"><a href="#cb19-81" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb19-82"><a href="#cb19-82" tabindex="-1"></a>    <span class="co"># axis.line = element_line(linewidth = one_pt/2, color = &#39;black&#39;),</span></span>
<span id="cb19-83"><a href="#cb19-83" tabindex="-1"></a>      <span class="at">axis.ticks =</span> <span class="fu">element_line</span>(<span class="at">linewidth =</span> one_pt<span class="sc">/</span><span class="dv">2</span>, <span class="at">color =</span> <span class="st">&#39;black&#39;</span>),</span>
<span id="cb19-84"><a href="#cb19-84" tabindex="-1"></a>      <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">linewidth =</span> one_pt<span class="sc">/</span><span class="dv">2</span>, <span class="at">color =</span> <span class="st">&#39;black&#39;</span>),</span>
<span id="cb19-85"><a href="#cb19-85" tabindex="-1"></a>      <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">8</span>), </span>
<span id="cb19-86"><a href="#cb19-86" tabindex="-1"></a>      <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>),</span>
<span id="cb19-87"><a href="#cb19-87" tabindex="-1"></a>      <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">8</span>),</span>
<span id="cb19-88"><a href="#cb19-88" tabindex="-1"></a>      <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">8</span>),</span>
<span id="cb19-89"><a href="#cb19-89" tabindex="-1"></a>      <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">4</span>)</span>
<span id="cb19-90"><a href="#cb19-90" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb19-91"><a href="#cb19-91" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb19-92"><a href="#cb19-92" tabindex="-1"></a>    <span class="at">y =</span> <span class="fu">expression</span>(<span class="fu">paste</span>(<span class="st">&quot;lo&quot;</span>, g[<span class="dv">2</span>],<span class="st">&quot;(Fold change)&quot;</span>)),</span>
<span id="cb19-93"><a href="#cb19-93" tabindex="-1"></a>    <span class="at">x =</span> <span class="cn">NULL</span></span>
<span id="cb19-94"><a href="#cb19-94" tabindex="-1"></a>  )<span class="sc">+</span></span>
<span id="cb19-95"><a href="#cb19-95" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color =</span> <span class="st">&#39;none&#39;</span>, <span class="at">fill =</span> <span class="st">&#39;none&#39;</span>, <span class="at">size =</span> <span class="st">&#39;none&#39;</span>)</span></code></pre></div>
<p><img src="figure/crispr-hsapiens_2180_sgRNA.Rmd/split-violinplot-selected-genes-ic-vs-id-curated-1.png" style="display: block; margin: auto;" /></p>
</div>
<div
id="supplementary-figure-6d-fold-change-of-lungs-over-primary-tumor-in-hunsg-and-nsg-host"
class="section level3">
<h3>Supplementary Figure 6D: Fold-change of Lungs over Primary Tumor in
huNSG and NSG host</h3>
<p>Dot plots highlighting screen hits identified upon transplantation in
the indicated hosts of MDA-MB-231 previously subjected to StealTHY KO as
depicted in Figure 6A; representative genes from six recurrently
identified gene families are shown on top, followed below by genes
previously known to impact tumorigenicity and metastasis, presented as
positive controls. The fold-change values indicate differential sgRNA
abundance between primary tumors and synchronous lung metastases; n≥6
each; RTK, receptor tyrosine kinase, TNF, tumor necrosis factor; IFN,
interferon; FDR, false discovery rate.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a>i <span class="ot">&lt;-</span> <span class="fu">names</span>(gene_summ)[<span class="dv">1</span>]</span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a>plot_data <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i=</span><span class="fu">names</span>(gene_summ), <span class="at">.combine =</span> rbind) <span class="sc">%do%</span> {</span>
<span id="cb20-3"><a href="#cb20-3" tabindex="-1"></a>  gene_summ[[i]] <span class="sc">%&gt;%</span> </span>
<span id="cb20-4"><a href="#cb20-4" tabindex="-1"></a>    <span class="fu">ReadRRA</span>(<span class="at">score =</span> <span class="st">&#39;lfc&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb20-5"><a href="#cb20-5" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">LFC =</span> Score) <span class="sc">%&gt;%</span> </span>
<span id="cb20-6"><a href="#cb20-6" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb20-7"><a href="#cb20-7" tabindex="-1"></a>      <span class="at">direction =</span> <span class="fu">ifelse</span>(FDR <span class="sc">&lt;=</span> <span class="fl">0.10</span> <span class="sc">&amp;</span> LFC <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">&#39;up&#39;</span>, <span class="st">&#39;none&#39;</span>),</span>
<span id="cb20-8"><a href="#cb20-8" tabindex="-1"></a>      <span class="at">direction =</span> <span class="fu">ifelse</span>(FDR <span class="sc">&lt;=</span> <span class="fl">0.10</span> <span class="sc">&amp;</span> LFC <span class="sc">&lt;</span> <span class="dv">0</span>, <span class="st">&#39;down&#39;</span>, direction),</span>
<span id="cb20-9"><a href="#cb20-9" tabindex="-1"></a>      <span class="at">is.significant =</span> FDR <span class="sc">&lt;=</span> <span class="fl">0.10</span>,</span>
<span id="cb20-10"><a href="#cb20-10" tabindex="-1"></a>      <span class="at">comparison =</span> i</span>
<span id="cb20-11"><a href="#cb20-11" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb20-12"><a href="#cb20-12" tabindex="-1"></a>    <span class="fu">left_join</span>(gene_info, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&#39;id&#39;</span> <span class="ot">=</span> <span class="st">&#39;id&#39;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-13"><a href="#cb20-13" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(gene_set)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-14"><a href="#cb20-14" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb20-15"><a href="#cb20-15" tabindex="-1"></a>      <span class="at">gene_name =</span> <span class="fu">factor</span>(gene_name, <span class="at">levels =</span> gene_levels),</span>
<span id="cb20-16"><a href="#cb20-16" tabindex="-1"></a>      <span class="at">gene_set =</span> <span class="fu">ifelse</span>(</span>
<span id="cb20-17"><a href="#cb20-17" tabindex="-1"></a>        gene_set <span class="sc">==</span> <span class="st">&#39;Immune system cytokines and chemokines&#39;</span>,</span>
<span id="cb20-18"><a href="#cb20-18" tabindex="-1"></a>        <span class="st">&#39;Immune system cytokines</span><span class="sc">\n</span><span class="st">and chemokines&#39;</span>,</span>
<span id="cb20-19"><a href="#cb20-19" tabindex="-1"></a>        gene_set</span>
<span id="cb20-20"><a href="#cb20-20" tabindex="-1"></a>      ),</span>
<span id="cb20-21"><a href="#cb20-21" tabindex="-1"></a>      <span class="at">gene_set =</span> <span class="fu">ifelse</span>(</span>
<span id="cb20-22"><a href="#cb20-22" tabindex="-1"></a>        gene_set <span class="sc">==</span> <span class="st">&#39;Interferone, TNF and danger/death-inducing signals&#39;</span>,</span>
<span id="cb20-23"><a href="#cb20-23" tabindex="-1"></a>        <span class="st">&#39;Interferon, TNF and</span><span class="sc">\n</span><span class="st">danger/death-inducing</span><span class="sc">\n</span><span class="st">signals&#39;</span>,</span>
<span id="cb20-24"><a href="#cb20-24" tabindex="-1"></a>        gene_set</span>
<span id="cb20-25"><a href="#cb20-25" tabindex="-1"></a>      ),</span>
<span id="cb20-26"><a href="#cb20-26" tabindex="-1"></a>      <span class="at">gene_set =</span> <span class="fu">factor</span>(gene_set, <span class="fu">names</span>(use_colors))</span>
<span id="cb20-27"><a href="#cb20-27" tabindex="-1"></a>      )</span>
<span id="cb20-28"><a href="#cb20-28" tabindex="-1"></a>}</span>
<span id="cb20-29"><a href="#cb20-29" tabindex="-1"></a></span>
<span id="cb20-30"><a href="#cb20-30" tabindex="-1"></a>plot_data <span class="sc">%&lt;&gt;%</span> </span>
<span id="cb20-31"><a href="#cb20-31" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb20-32"><a href="#cb20-32" tabindex="-1"></a>    <span class="at">mouse_model_short =</span> <span class="fu">case_match</span>(comparison, </span>
<span id="cb20-33"><a href="#cb20-33" tabindex="-1"></a>                            <span class="st">&#39;stealthy_hunsg&#39;</span> <span class="sc">~</span> <span class="st">&#39;huNSG&#39;</span>, </span>
<span id="cb20-34"><a href="#cb20-34" tabindex="-1"></a>                            <span class="st">&#39;stealthy_nsg&#39;</span> <span class="sc">~</span> <span class="st">&#39;NSG&#39;</span>),</span>
<span id="cb20-35"><a href="#cb20-35" tabindex="-1"></a>    <span class="at">mouse_model_short =</span> <span class="fu">factor</span>(mouse_model_short,</span>
<span id="cb20-36"><a href="#cb20-36" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&#39;huNSG&#39;</span>, <span class="st">&#39;NSG&#39;</span>))</span>
<span id="cb20-37"><a href="#cb20-37" tabindex="-1"></a>  )</span>
<span id="cb20-38"><a href="#cb20-38" tabindex="-1"></a></span>
<span id="cb20-39"><a href="#cb20-39" tabindex="-1"></a>xlim <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fu">ceiling</span>(<span class="fu">max</span>(<span class="fu">abs</span>(plot_data<span class="sc">$</span>LFC), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)), </span>
<span id="cb20-40"><a href="#cb20-40" tabindex="-1"></a>            <span class="fu">ceiling</span>(<span class="fu">max</span>(<span class="fu">abs</span>(plot_data<span class="sc">$</span>LFC), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)))</span>
<span id="cb20-41"><a href="#cb20-41" tabindex="-1"></a></span>
<span id="cb20-42"><a href="#cb20-42" tabindex="-1"></a>plot_data <span class="sc">%&gt;%</span> </span>
<span id="cb20-43"><a href="#cb20-43" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(LFC, gene_name, <span class="at">color =</span> gene_set, <span class="at">size =</span> <span class="sc">-</span><span class="fu">log10</span>(FDR))) <span class="sc">+</span></span>
<span id="cb20-44"><a href="#cb20-44" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">shape =</span> gene_set)) <span class="sc">+</span></span>
<span id="cb20-45"><a href="#cb20-45" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb20-46"><a href="#cb20-46" tabindex="-1"></a>    <span class="fu">scale_y_discrete</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">add =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>))) <span class="sc">+</span> <span class="co"># add fixed units around each facet to avoid lines over points</span></span>
<span id="cb20-47"><a href="#cb20-47" tabindex="-1"></a>    <span class="fu">scale_x_continuous</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">add =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>))) <span class="sc">+</span> <span class="co"># add fixed units around each facet to avoid lines over points</span></span>
<span id="cb20-48"><a href="#cb20-48" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> use_colors) <span class="sc">+</span></span>
<span id="cb20-49"><a href="#cb20-49" tabindex="-1"></a>    <span class="fu">scale_size</span>(<span class="at">range =</span> <span class="fu">c</span>(<span class="fl">0.5</span>, <span class="fl">2.5</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">4</span>)) <span class="sc">+</span></span>
<span id="cb20-50"><a href="#cb20-50" tabindex="-1"></a>    <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> use_shape) <span class="sc">+</span></span>
<span id="cb20-51"><a href="#cb20-51" tabindex="-1"></a>    <span class="fu">xlim</span>(xlim) <span class="sc">+</span></span>
<span id="cb20-52"><a href="#cb20-52" tabindex="-1"></a>    <span class="fu">facet_grid</span>(<span class="at">cols =</span> <span class="fu">vars</span>(mouse_model_short), <span class="at">rows =</span> <span class="fu">vars</span>(gene_set), <span class="at">scales =</span> <span class="st">&#39;free_y&#39;</span>, <span class="at">space =</span> <span class="st">&#39;free_y&#39;</span>) <span class="sc">+</span></span>
<span id="cb20-53"><a href="#cb20-53" tabindex="-1"></a>    theme_facet <span class="sc">+</span></span>
<span id="cb20-54"><a href="#cb20-54" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb20-55"><a href="#cb20-55" tabindex="-1"></a>      <span class="at">axis.ticks =</span> <span class="fu">element_line</span>(<span class="at">linewidth =</span> one_pt<span class="sc">/</span><span class="dv">2</span>, <span class="at">color =</span> <span class="st">&#39;black&#39;</span>),</span>
<span id="cb20-56"><a href="#cb20-56" tabindex="-1"></a>      <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">linewidth =</span> one_pt<span class="sc">/</span><span class="dv">2</span>, <span class="at">color =</span> <span class="st">&#39;black&#39;</span>),</span>
<span id="cb20-57"><a href="#cb20-57" tabindex="-1"></a>      <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">6</span>), </span>
<span id="cb20-58"><a href="#cb20-58" tabindex="-1"></a>      <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">6</span>),</span>
<span id="cb20-59"><a href="#cb20-59" tabindex="-1"></a>      <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">6</span>),</span>
<span id="cb20-60"><a href="#cb20-60" tabindex="-1"></a>      <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">6</span>),</span>
<span id="cb20-61"><a href="#cb20-61" tabindex="-1"></a>      <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">4</span>)</span>
<span id="cb20-62"><a href="#cb20-62" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb20-63"><a href="#cb20-63" tabindex="-1"></a>    <span class="fu">labs</span>(</span>
<span id="cb20-64"><a href="#cb20-64" tabindex="-1"></a>      <span class="at">x =</span> <span class="fu">expression</span>(<span class="fu">paste</span>(<span class="st">&quot;lo&quot;</span>, g[<span class="dv">2</span>],<span class="st">&quot;(Fold change)&quot;</span>)),</span>
<span id="cb20-65"><a href="#cb20-65" tabindex="-1"></a>      <span class="at">y =</span> <span class="cn">NULL</span>,</span>
<span id="cb20-66"><a href="#cb20-66" tabindex="-1"></a>      <span class="at">size =</span> <span class="fu">expression</span>(<span class="fu">paste</span>(<span class="st">&quot;-lo&quot;</span>, g[<span class="dv">10</span>],<span class="st">&quot;(FDR)&quot;</span>))</span>
<span id="cb20-67"><a href="#cb20-67" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb20-68"><a href="#cb20-68" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">color =</span> <span class="st">&#39;none&#39;</span>, <span class="at">shape =</span> <span class="st">&#39;none&#39;</span>)</span></code></pre></div>
<p><img src="figure/crispr-hsapiens_2180_sgRNA.Rmd/dotplot-selected-genes-allconditions-1.png" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<p>R version 4.4.3 (2025-02-28) Platform: aarch64-apple-darwin20 Running
under: macOS 26.0.1</p>
<p>Matrix products: default BLAS:
/Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib
LAPACK:
/Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;
LAPACK version 3.12.0</p>
<p>locale: [1]
en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</p>
<p>time zone: Europe/Zurich tzcode source: internal</p>
<p>attached base packages: [1] grid stats4 stats graphics grDevices
utils datasets [8] methods base</p>
<p>other attached packages: [1] MAGeCKFlute_2.9.0 ggrepel_0.9.6<br />
[3] cowplot_1.2.0 ggpubr_0.6.1<br />
[5] ggbeeswarm_0.7.2 colorblindr_0.1.0<br />
[7] colorspace_2.1-1 patchwork_1.3.1<br />
[9] ggh4x_0.3.1 ggridges_0.5.6<br />
[11] SummarizedExperiment_1.36.0 Biobase_2.66.0<br />
[13] GenomicRanges_1.58.0 GenomeInfoDb_1.42.3<br />
[15] IRanges_2.40.1 S4Vectors_0.44.0<br />
[17] BiocGenerics_0.52.0 MatrixGenerics_1.18.1<br />
[19] matrixStats_1.5.0 diptest_0.77-1<br />
[21] kableExtra_1.4.0 DT_0.34.0<br />
[23] magrittr_2.0.3 foreach_1.5.2<br />
[25] knitr_1.50 lubridate_1.9.4<br />
[27] forcats_1.0.0 stringr_1.5.1<br />
[29] dplyr_1.1.4 purrr_1.1.0<br />
[31] readr_2.1.5 tidyr_1.3.1<br />
[33] tibble_3.3.0 ggplot2_3.5.2<br />
[35] tidyverse_2.0.0 workflowr_1.7.1</p>
<p>loaded via a namespace (and not attached): [1] splines_4.4.3
later_1.4.4 ggplotify_0.1.2<br />
[4] bitops_1.0-9 filelock_1.0.3 cellranger_1.1.0<br />
[7] R.oo_1.27.1 graph_1.84.1 XML_3.99-0.19<br />
[10] lifecycle_1.0.4 httr2_1.2.1 rstatix_0.7.2<br />
[13] rprojroot_2.1.0 processx_3.8.6 lattice_0.22-7<br />
[16] backports_1.5.0 sass_0.4.10 rmarkdown_2.29<br />
[19] jquerylib_0.1.4 yaml_2.3.10 ggtangle_0.0.7<br />
[22] httpuv_1.6.16 depmap_1.20.0 DBI_1.2.3<br />
[25] RColorBrewer_1.1-3 abind_1.4-8 zlibbioc_1.52.0<br />
[28] R.utils_2.13.0 RCurl_1.98-1.17 yulab.utils_0.2.0<br />
[31] rappdirs_0.3.3 git2r_0.36.2 GenomeInfoDbData_1.2.13 [34]
enrichplot_1.26.6 tidytree_0.4.6 svglite_2.2.1<br />
[37] codetools_0.2-20 DelayedArray_0.32.0 DOSE_4.0.1<br />
[40] xml2_1.3.8 tidyselect_1.2.1 aplot_0.2.8<br />
[43] UCSC.utils_1.2.0 farver_2.1.2 BiocFileCache_2.14.0<br />
[46] pathview_1.46.0 jsonlite_2.0.0 Formula_1.2-5<br />
[49] iterators_1.0.14 systemfonts_1.2.3 tools_4.4.3<br />
[52] treeio_1.30.0 Rcpp_1.1.0 glue_1.8.0<br />
[55] gridExtra_2.3 SparseArray_1.6.2 xfun_0.53<br />
[58] qvalue_2.38.0 withr_3.0.2 BiocManager_1.30.26<br />
[61] fastmap_1.2.0 callr_3.7.6 digest_0.6.37<br />
[64] gridGraphics_0.5-1 timechange_0.3.0 R6_2.6.1<br />
[67] textshaping_1.0.1 GO.db_3.20.0 dichromat_2.0-0.1<br />
[70] RSQLite_2.4.3 R.methodsS3_1.8.2 generics_0.1.4<br />
[73] data.table_1.17.8 httr_1.4.7 htmlwidgets_1.6.4<br />
[76] S4Arrays_1.6.0 whisker_0.4.1 pkgconfig_2.0.3<br />
[79] gtable_0.3.6 blob_1.2.4 XVector_0.46.0<br />
[82] clusterProfiler_4.14.6 htmltools_0.5.8.1 carData_3.0-5<br />
[85] fgsea_1.32.4 scales_1.4.0 png_0.1-8<br />
[88] ggfun_0.2.0 rstudioapi_0.17.1 tzdb_0.5.0<br />
[91] reshape2_1.4.4 nlme_3.1-168 curl_7.0.0<br />
[94] org.Hs.eg.db_3.20.0 cachem_1.1.0 BiocVersion_3.20.0<br />
[97] parallel_4.4.3 vipor_0.4.7 AnnotationDbi_1.68.0<br />
[100] pillar_1.11.0 vctrs_0.6.5 promises_1.3.3<br />
[103] car_3.1-3 dbplyr_2.5.0 beeswarm_0.4.0<br />
[106] Rgraphviz_2.50.0 evaluate_1.0.5 KEGGgraph_1.66.0<br />
[109] cli_3.6.5 compiler_4.4.3 rlang_1.1.6<br />
[112] crayon_1.5.3 ggsignif_0.6.4 labeling_0.4.3<br />
[115] ps_1.9.1 getPass_0.2-4 plyr_1.8.9<br />
[118] fs_1.6.6 stringi_1.8.7 viridisLite_0.4.2<br />
[121] BiocParallel_1.40.2 Biostrings_2.74.1 lazyeval_0.2.2<br />
[124] GOSemSim_2.32.0 Matrix_1.7-3 ExperimentHub_2.14.0<br />
[127] hms_1.1.3 bit64_4.6.0-1 KEGGREST_1.46.0<br />
[130] AnnotationHub_3.14.0 igraph_2.1.4 broom_1.0.9<br />
[133] memoise_2.0.1 bslib_0.9.0 ggtree_3.14.0<br />
[136] fastmatch_1.1-6 bit_4.6.0 readxl_1.4.5<br />
[139] gson_0.1.0 ape_5.8-1</p>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
