<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Francesc Castro-Giner" />

<meta name="date" content="2024-05-18" />

<title>Visualize and compare sgRNA abundance distributions</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.30/datatables.js"></script>
<link href="site_libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<script src="site_libs/jszip-1.13.4/jszip.min.js"></script>
<link href="site_libs/dt-ext-buttons-1.13.4/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.13.4/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.4/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.4/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.4/js/buttons.print.min.js"></script>
<link href="site_libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet" />
<script src="site_libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="site_libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
<script src="site_libs/selectize-0.12.0/selectize.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">crispr-immunorejection</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/TheAcetoLab/saini-stealTHY">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Visualize and compare sgRNA abundance
distributions</h1>
<h3 class="subtitle">Mouse libray (n = 2215)</h3>
<h4 class="author">Francesc Castro-Giner</h4>
<h4 class="date">May 18, 2024</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-05-18
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>saini-stealTHY/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240517code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240517)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240517code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240517)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomTheAcetoLabsainistealTHYtree48661cdcf16148181af7fefb40fc5a72c2f085e6targetblank48661cda">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/tree/48661cdcf16148181af7fefb40fc5a72c2f085e6" target="_blank">48661cd</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomTheAcetoLabsainistealTHYtree48661cdcf16148181af7fefb40fc5a72c2f085e6targetblank48661cda"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/TheAcetoLab/saini-stealTHY/tree/48661cdcf16148181af7fefb40fc5a72c2f085e6" target="_blank">48661cd</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/raw_data_processing/rnaseq/p27851_o32062/pipelines/
    Ignored:    configuration/.DS_Store

Untracked files:
    Untracked:  data/crispr/
    Untracked:  update_workflowr.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/crispr-mm_2215_sgRNA-distribution.Rmd</code>) and HTML
(<code>docs/crispr-mm_2215_sgRNA-distribution.html</code>) files. If
you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/48661cdcf16148181af7fefb40fc5a72c2f085e6/analysis/crispr-mm_2215_sgRNA-distribution.Rmd" target="_blank">48661cd</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2024-05-18
</td>
<td>
Initial commit of analysis/crispr-mm_2215_sgRNA-distribution.Rmd
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries-additional-functions-and-data"
class="section level2" number="0.1">
<h2><span class="header-section-number">0.1</span> Load libraries,
additional functions and data</h2>
<p>Setup environment</p>
<pre class="r"><code>knitr::opts_chunk$set(results=&#39;asis&#39;, echo=TRUE, message=FALSE, warning=FALSE, error=FALSE, fig.align = &#39;center&#39;, fig.width = 3.5, fig.asp = 0.618, dpi = 600, dev = c(&quot;png&quot;, &quot;pdf&quot;), engine.opts = list(bash = &quot;-l&quot;))

options(stringsAsFactors = FALSE)

use_seed &lt;- 1100101

set.seed(use_seed)

# dir.create(params$output_dir, recursive = TRUE, showWarnings = FALSE)</code></pre>
<p>Load packages</p>
<pre class="r"><code>library(tidyverse)
library(knitr)
library(foreach)
library(magrittr)
library(DT)
library(kableExtra)
library(diptest)
library(SummarizedExperiment)

library(ggridges)
library(ggh4x)
library(patchwork)
library(colorblindr)
# library(showtext)

# library(DT)
# library(colorblindr)
# library(ggdendro)
# library(RColorBrewer)
# library(circlize)
# library(Hmisc)
# library(ComplexHeatmap)
# library(ineq)

# library(kableExtra)
# library(magrittr)
# library(ggrepel)
# library(ggpubr)
# library(ggbeeswarm)
# library(ggridges)</code></pre>
<p>Load ggplot theme</p>
<pre class="r"><code>source(&quot;./configuration/rmarkdown/ggplot_theme.R&quot;)</code></pre>
<p>Load Summarized Experiment object</p>
<pre class="r"><code>se &lt;- readRDS(file.path(params$data_dir, &#39;se.rds&#39;))</code></pre>
</div>
<div id="configure-analyses" class="section level2" number="0.2">
<h2><span class="header-section-number">0.2</span> Configure
analyses</h2>
<p>Define comparisons for modality and Kolmogorov-Smirnov test</p>
<pre class="r"><code>x &lt;- colData(se) %&gt;% data.frame

group_list_modality &lt;- list(
  `Syngeneic Thy1` = x %&gt;% 
    filter(vector_mmodel == &#39;4T1-noKO-Thy1-BALB&#39; &amp; sample_type == &#39;whole_tumor&#39;) %&gt;% 
    pull(sample_alias),
  
  `Syngeneic Puro` = x %&gt;% 
    filter(vector_mmodel %in% c(&#39;4T1-noKO-Puro-BALB&#39;, &#39;MVT1-noKO-Puro-FVB&#39;, &#39;Py2T-noKO-Puro-FVB&#39;) &amp; sample_type == &#39;whole_tumor&#39;) %&gt;%
    pull(sample_alias),
  
  `Syngeneic dCas9 + Puro` = x %&gt;% 
    filter(vector_mmodel == &#39;4T1-noKO-Puro_EF1alpha_dCas9-BALB&#39; &amp; sample_type == &#39;whole_tumor&#39;) %&gt;% 
    pull(sample_alias),
  
  `NSG Thy1` =  x %&gt;% 
    filter(vector_mmodel %in% c(&#39;4T1-noKO-Thy1-NSG&#39;, &#39;MVT1-noKO-Thy1-NSG&#39;, &#39;Py2T-noKO-Thy1-NSG&#39;) &amp; sample_type == &#39;whole_tumor&#39;) %&gt;%
    pull(sample_alias),
  
  `NSG Puro` = x %&gt;% 
    filter(vector_mmodel %in% c(&#39;4T1-noKO-Puro-NSG&#39;, &#39;MVT1-noKO-Puro-NSG&#39;, &#39;Py2T-noKO-Puro-NSG&#39;) &amp; sample_type == &#39;whole_tumor&#39;) %&gt;%
    pull(sample_alias),
  
  `NSG dCas9 + Puro` = x %&gt;% 
    filter(vector_mmodel == &#39;4T1-noKO-Puro_EF1alpha_dCas9-NSG&#39; &amp; sample_type == &#39;whole_tumor&#39;) %&gt;% 
    pull(sample_alias)
)

comp_list_modality &lt;- list(
  `Syngeneic Thy1 vs. dCas9 + Puro` = list(
    `dCas9 + Puro` = group_list_modality$`Syngeneic dCas9 + Puro`,
    `Thy1` = group_list_modality$`Syngeneic Thy1`
  ),
  `Syngeneic Thy1 vs. Puro` = list(
    `Puro` = group_list_modality$`Syngeneic Puro`,
    `Thy1` = group_list_modality$`Syngeneic Thy1`
  )
)

group_df_modality &lt;- foreach(i = names(group_list_modality), .combine = rbind) %do% {
  data.frame(
    sample_alias = group_list_modality[[i]],
    modality_group = i
  )
}</code></pre>
</div>
<div id="primary-tumor" class="section level2" number="0.3">
<h2><span class="header-section-number">0.3</span> Primary tumor</h2>
<pre class="r"><code>use_sample_type &lt;- &#39;whole_tumor&#39;
# comp_list &lt;- comp_list_by_sample_type
# comp_list_unlisted &lt;- foreach(x = comp_list, .combine = c) %do% x
# comp_list_df_double &lt;- foreach(x = comp_list_unlisted, .combine = rbind) %do% {rbind(x, rev(x))} %&gt;% 
#   data.frame %&gt;% 
#   set_names(c(&#39;group1&#39;, &#39;group2&#39;))
use_colData &lt;- colData(se) %&gt;% data.frame %&gt;%
  filter(sample_type == use_sample_type) %&gt;% 
  mutate(
    vector_short = ifelse(vector_short == &#39;Puro_EF1alpha_dCas9&#39;, 
                          &#39;dCas9+Puro&#39;, vector_short),
    vector_short = factor(vector_short, levels = c(&#39;Thy1&#39;, &#39;Puro&#39;, &#39;dCas9+Puro&#39;)),
    mouse_model_short = case_match(mouse_model_global,
                                   &#39;immunodeficient&#39; ~ &#39;NSG&#39;,
                                   &#39;immunocompetent&#39; ~ &#39;Syngeneic&#39;
                                   ),
    mouse_model_short = factor(mouse_model_short, levels = c(&#39;NSG&#39;, &#39;Syngeneic&#39;))
  )</code></pre>
<div
id="figure-2b-sgrna-distribution-in-mice-transplanted-with-4t1-cells"
class="section level3" number="0.3.1">
<h3><span class="header-section-number">0.3.1</span> Figure 2B: sgRNA
distribution in mice transplanted with 4T1 cells</h3>
<p>Density plots showing the distribution of unique sgRNAs retrieved in
orthotopic primary tumors from the indicated mice transplanted with 4T1
cells, previously transduced as indicated; frequency is shown as read
counts per million (cpm) in log10 scale adding a pseudocunt of 1; Each
plot is representative of a group of n=5 mice.</p>
<pre class="r"><code>selected_panels &lt;- list(
  `Thy1, NSG` = c(&#39;4T_TumoT_N4&#39;, &#39;4T_TumoT_N5&#39;, &#39;4T_TumoT_N2&#39;),
  `Thy1, Syngeneic` = c(&#39;4T_TumoT_B1&#39;, &#39;4T_TumoT_B3&#39;, &#39;4T_TumoT_B4&#39;),
 
  `Puro, NSG` = c(&#39;4T_TumoP_N5&#39;, &#39;4T_TumoP_N4&#39;, &#39;4T_TumoP_N2&#39;),
  `Puro, Syngeneic` = c(&#39;4T_TumoP_B3&#39;, &#39;4T_TumoP_B2&#39;, &#39;4T_TumoP_B1&#39;),
  
  `dCas9+Puro, NSG` = c(&#39;PlatePositionE1&#39;, &#39;PlatePositionC1&#39;, &#39;PlatePositionA2&#39;),
  `dCas9+Puro, Syngeneic` = c(&#39;PlatePositionE2&#39;, &#39;PlatePositionG2&#39;, &#39;PlatePositionC2&#39;)
  
)

selected_panels_df &lt;- foreach(i = names(selected_panels), .combine = rbind) %do% {
  data.frame(group = i, sample_alias = selected_panels[[i]])
}

use_samples &lt;- unlist(selected_panels)
use_assay &lt;- assay(se[,use_samples], &#39;cpm&#39;) %&gt;% 
  as.data.frame(check.names = F) %&gt;% 
  rownames_to_column(&#39;guide&#39;) %&gt;% 
  pivot_longer(-guide, names_to = &#39;sample_alias&#39;, values_to = &#39;cpm&#39;) %&gt;% 
  left_join(use_colData) %&gt;% 
  left_join(selected_panels_df) %&gt;% 
  mutate(
    sample_alias = factor(sample_alias, levels = use_samples),
    group = factor(group, levels = names(selected_panels))
  )
  

use_colors &lt;- c(
  `Thy1, Syngeneic` = &#39;#F7EC13&#39;, 
  `Thy1, NSG`  = &#39;#F7EC13&#39;,
  `Puro, Syngeneic` = &#39;#58595B&#39;, 
  `Puro, NSG`  = &#39;#58595B&#39;,
  `dCas9+Puro, Syngeneic` = &#39;black&#39;,
  `dCas9+Puro, NSG` = &#39;black&#39;
  )

use_line_colors &lt;- c(
  `Thy1, Syngeneic` = &#39;black&#39;, 
  `Thy1, NSG`  = &#39;black&#39;,
  `Puro, Syngeneic` = &#39;white&#39;, 
  `Puro, NSG`  = &#39;white&#39;,
  `dCas9+Puro, Syngeneic` = &#39;white&#39;,
  `dCas9+Puro, NSG` = &#39;white&#39;
  )

use_assay %&gt;% 
  ggplot(aes(x=cpm + 1, y = sample_alias, fill = group, color = group, height = after_stat(density))) +
  geom_density_ridges(size = one_pt/4, 
                      alpha = 0.8, 
                      scale = 4,
                      rel_min_height = 0.001 # set the `rel_min_height` argument to remove tails,
                      ) +
  scale_fill_manual(values = use_colors) +
  scale_color_manual(values = use_line_colors) +
  scale_x_log10(expand = expansion(mult = c(0, 0))) +
  scale_y_discrete(expand = c(0.01, 0)) +
  labs(y = &#39;&#39;) +
  guides(fill = &#39;none&#39;, color = &#39;none&#39;) +
  facet_wrap2(vars(group), ncol = 2,
                     scales = &#39;free_y&#39;,
                     axes = &quot;x&quot;) +
  theme_ridges(font_size = 8, grid = F) +
  theme(
    strip.background = element_blank(),
    axis.line.x = element_line(linewidth = one_pt/4, color = &#39;black&#39;),
    axis.ticks.x = element_line(linewidth = one_pt/4, color = &#39;black&#39;),
    axis.text.y = element_blank(),
    strip.text.x = element_text(size=4, hjust = 0)
  )</code></pre>
<p><img src="figure/crispr-mm_2215_sgRNA-distribution.Rmd/test-dist-tumor-distribution-ridges_plot-4t1-1.png" width="1800" style="display: block; margin: auto;" /></p>
</div>
<div
id="supplementary-figure-2b-sgrna-distribution-in-mice-transplanted-with-py2t-and-mvt1-cells"
class="section level3" number="0.3.2">
<h3><span class="header-section-number">0.3.2</span> Supplementary
Figure 2B: sgRNA distribution in mice transplanted with Py2T and MVT1
cells</h3>
<pre class="r"><code>selected_panels &lt;- list(
  `Py2T, Thy1, NSG` = c(&#39;PY_TumoT_N5&#39;, &#39;PY_TumoT_N4&#39;, &#39;PY_TumoT_N3&#39;),
  `Py2T, Thy1, Syngeneic` = c(&#39;PY_TumoT_F3&#39;, &#39;PY_TumoT_F4&#39;, &#39;PY_TumoT_F5&#39;), 
  
  `Py2T, Puro, NSG` = c(&#39;PY_TumoP_N1&#39;, &#39;PY_TumoP_N2&#39;, &#39;PY_TumoP_N3&#39;),
  `Py2T, Puro, Syngeneic` = c(&#39;PY_TumoP_F1&#39;, &#39;PY_TumoP_F5&#39;, &#39;PY_TumoP_F2&#39;),
  
  `MVT1, Thy1, NSG` = c(&#39;MVT_TumoT_N3&#39;, &#39;MVT_TumoT_N4&#39;, &#39;MVT_TumoT_N5&#39;),
  `MVT1, Thy1, Syngeneic` = c(&#39;MVT_TumoT_F1&#39;, &#39;MVT_TumoT_F4&#39;, &#39;MVT_TumoT_F3&#39;),
  
  `MVT1, Puro, NSG` = c(&#39;MVT_TumoP_N3&#39;, &#39;MVT_TumoP_N4&#39;, &#39;MVT_TumoP_N5&#39;),
  `MVT1, Puro, Syngeneic` = c(&#39;MVT_TumoP_F3&#39;, &#39;MVT_TumoP_F4&#39;, &#39;MVT_TumoP_F5&#39;)
)

selected_panels_df &lt;- foreach(i = names(selected_panels), .combine = rbind) %do% {
  data.frame(group = i, sample_alias = selected_panels[[i]])
}

use_samples &lt;- unlist(selected_panels)
use_assay &lt;- assay(se[,use_samples], &#39;cpm&#39;) %&gt;% 
  as.data.frame(check.names = F) %&gt;% 
  rownames_to_column(&#39;guide&#39;) %&gt;% 
  pivot_longer(-guide, names_to = &#39;sample_alias&#39;, values_to = &#39;cpm&#39;) %&gt;% 
  left_join(use_colData) %&gt;% 
  left_join(selected_panels_df) %&gt;% 
  mutate(
    sample_alias = factor(sample_alias, levels = use_samples),
    group = factor(group, levels = names(selected_panels))
  )
  

use_colors &lt;- c(
  `Py2T, Thy1, Syngeneic` = &#39;#F7EC13&#39;, 
  `Py2T, Thy1, NSG`  = &#39;#F7EC13&#39;,
  `Py2T, Puro, Syngeneic` = &#39;#58595B&#39;, 
  `Py2T, Puro, NSG`  = &#39;#58595B&#39;,
  
  `MVT1, Thy1, Syngeneic` = &#39;#F7EC13&#39;, 
  `MVT1, Thy1, NSG`  = &#39;#F7EC13&#39;,
  `MVT1, Puro, Syngeneic` = &#39;#58595B&#39;, 
  `MVT1, Puro, NSG`  = &#39;#58595B&#39;
  )

use_line_colors &lt;- c(
  `Py2T, Thy1, Syngeneic` = &#39;black&#39;, 
  `Py2T, Thy1, NSG`  = &#39;black&#39;,
  `Py2T, Puro, Syngeneic` = &#39;white&#39;, 
  `Py2T, Puro, NSG`  = &#39;white&#39;,
  
  `MVT1, Thy1, Syngeneic` = &#39;black&#39;, 
  `MVT1, Thy1, NSG`  = &#39;black&#39;,
  `MVT1, Puro, Syngeneic` = &#39;white&#39;, 
  `MVT1, Puro, NSG`  = &#39;white&#39;
  )

use_assay %&gt;% 
  ggplot(aes(x=cpm + 1, y = sample_alias, fill = group, color = group, height = after_stat(density))) +
  geom_density_ridges(size = one_pt/4, 
                      alpha = 0.8, 
                      scale = 4,
                      rel_min_height = 0.001 # set the `rel_min_height` argument to remove tails,
                      ) +
  scale_fill_manual(values = use_colors) +
  scale_color_manual(values = use_line_colors) +
  scale_x_log10(expand = expansion(mult = c(0, 0))) +
  scale_y_discrete(expand = c(0.01, 0)) +
  labs(y = &#39;&#39;) +
  guides(fill = &#39;none&#39;, color = &#39;none&#39;) +
  facet_wrap2(vars(group), ncol = 2,
                     scales = &#39;free_y&#39;,
                     axes = &quot;x&quot;) +
  theme_ridges(font_size = 3, grid = F) +
  theme(
    strip.background = element_blank(),
    axis.line.x = element_line(linewidth = one_pt/4, color = &#39;black&#39;),
    axis.ticks.x = element_line(linewidth = one_pt/4, color = &#39;black&#39;),
    axis.text.y = element_blank(),
    panel.border = element_rect(fill =NULL,
                                color = &quot;black&quot;,
                                linewidth = 2*one_pt)
  )</code></pre>
<p><img src="figure/crispr-mm_2215_sgRNA-distribution.Rmd/test-dist-tumor-distribution-ridges_plot-py2t-mvt1-1.png" width="1800" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="kolmogorov-smirnov-test-and-ecdf-plots" class="section level2"
number="0.4">
<h2><span class="header-section-number">0.4</span> Kolmogorov-Smirnov
test and ECDF plots</h2>
<p>Run Kolmogorov-Smirnov test to compare distributions merged by group
and create Empirical Cumulative Density Function (ECDF) by
comparison.</p>
<pre class="r"><code>use_colors &lt;- c(
  `Thy1` = &#39;#F7EC13&#39;,
  `Puro`  = &#39;#58595B&#39;,
  `dCas9 + Puro` = &#39;black&#39;
  )

i &lt;- names(comp_list_modality)[1]
ks_test_df &lt;- data.frame()
ecdf_plots &lt;- list()
for(i in names(comp_list_modality)) {
  use_samples &lt;- comp_list_modality[[i]]
  use_assay_c &lt;- assay(se[,use_samples[[1]]], &#39;cpm&#39;) %&gt;% 
    data.frame(check.names = F) %&gt;% 
    rownames_to_column(&#39;guide&#39;) %&gt;% 
    pivot_longer(-guide, values_to = &#39;cpm&#39;,names_to = &#39;sample_alias&#39;) %&gt;% 
    mutate(group = &#39;case&#39;, condition = names(use_samples)[1])
  use_assay_r &lt;- assay(se[,use_samples[[2]]], &#39;cpm&#39;) %&gt;% 
    data.frame(check.names = F) %&gt;% 
    rownames_to_column(&#39;guide&#39;) %&gt;% 
    pivot_longer(-guide, values_to = &#39;cpm&#39;,names_to = &#39;sample_alias&#39;) %&gt;% 
    mutate(group = &#39;referene&#39;, condition = names(use_samples)[2])
  use_assay &lt;- rbind(use_assay_c, use_assay_r)
  
  # Perform the Kolmogorov-Smirnov test
  ks.res &lt;- ks.test(use_assay_c$cpm, use_assay_r$cpm)
  ks_test_df &lt;- rbind(
    ks_test_df,
      data.frame(
        comparison = i,
        case = names(use_samples)[1],
        reference = names(use_samples)[2],
        D = ks.res$statistic,
        P = ks.res$p.value
      )
    )

  # Create ECDFs plot Empirical Cumulative Density Function
  # use_colors &lt;- c(&#39;red&#39;, &#39;blue&#39;) %&gt;% set_names(names(use_samples))
  ecdf_plots[[i]] &lt;- use_assay %&gt;% 
    ggplot(aes(x = log(1+cpm), color = condition, group = sample_alias)) +
    stat_ecdf(geom = &#39;step&#39;, alpha = 1, linewidth = one_pt/4) +
    scale_color_manual(values = use_colors) +
    labs(x = &quot;log(1+cpm)&quot;,
         color = &#39;&#39;,
         sub = i#,
         # caption =paste(&quot;Kolmogorov-Smirnov test p-value:&quot;, round(ks.res$p.value, 3), &quot;\n&quot;,
                        # &quot;Kolmogorov-Smirnov test D:&quot;, round(ks.res$statistic, 3))
         ) +
    theme(
      text = element_text(size=2),
      axis.line = element_line(linewidth = one_pt/4, color = &#39;black&#39;),
      axis.ticks = element_line(linewidth = one_pt/4, color = &#39;black&#39;),
      axis.text.y =  element_text(size=3),
      axis.text.x =  element_text(size=3),
      legend.text =  element_text(size=2),
      legend.title =  element_text(size=2),
      plot.subtitle = element_text(size=3),
      plot.caption =  element_text(size=2)
      
    )

}</code></pre>
<div id="supplementary-figure-2c-left-k-s-test-table"
class="section level3" number="0.4.1">
<h3><span class="header-section-number">0.4.1</span> Supplementary
Figure 2C left : K-S test table</h3>
<pre class="r"><code>ks_test_df %&gt;% 
  dplyr::select(-case, -reference) %&gt;% 
  mutate(P = format.pval(P)) %&gt;% 
  datatable(., 
            rownames = FALSE, 
            filter = &#39;top&#39;, 
            caption = &#39;Results of Kolmogorov-Smirnov test&#39;,
            extensions = &#39;Buttons&#39;, 
            options = list(
              dom = &#39;Blfrtip&#39;,
              buttons = c(&#39;csv&#39;, &#39;excel&#39;),
              columnDefs = list(
                list(width = 120, targets = 1)
              )
            )
            ) %&gt;% 
    formatRound(columns = c(&#39;D&#39;), digits = 3)</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-0e41329c1963262c826a" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-0e41329c1963262c826a">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.375289691497366\" data-max=\"0.73273137697517\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\" disabled=\"\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption>Results of Kolmogorov-Smirnov test<\/caption>","data":[["Syngeneic Thy1 vs. dCas9 + Puro","Syngeneic Thy1 vs. Puro"],[0.7327313769751692,0.3752896914973664],["&lt; 2.22e-16","&lt; 2.22e-16"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>comparison<\/th>\n      <th>D<\/th>\n      <th>P<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["csv","excel"],"columnDefs":[{"targets":1,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\", null);\n  }"},{"width":120,"targets":1},{"className":"dt-right","targets":1}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true}},"evals":["options.columnDefs.0.render"],"jsHooks":[]}</script>
</div>
<div
id="supplementary-figure-2c-right-empirical-cumulative-density-function-ecdf-plots"
class="section level3" number="0.4.2">
<h3><span class="header-section-number">0.4.2</span> Supplementary
Figure 2C right : Empirical Cumulative Density Function (ECDF)
plots</h3>
<p>Empirical Cumulative Density Function (ECDF) plot by comparison</p>
<pre class="r"><code># Dim for an individual plot : fig.width = 1.5, fig.asp = 0.66
# for(i in names(ecdf_plots)) {
#   cat(&quot;#####&quot;, i, &quot;\n&quot;)
#   print(ecdf_plots[[i]])
#   cat(&quot;\n\n&quot;)
# }
use_index &lt;- c(&quot;Syngeneic Thy1 vs. Puro&quot;, &quot;Syngeneic Thy1 vs. dCas9 + Puro&quot;)
plot_grid(plotlist = ecdf_plots[use_index], ncol = 2)</code></pre>
<p><img src="figure/crispr-mm_2215_sgRNA-distribution.Rmd/modality-comparisons-ks-ecdf-1.png" width="1800" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="hartigans-dip-test" class="section level2" number="0.5">
<h2><span class="header-section-number">0.5</span> Hartigans’ dip
test</h2>
<p>Hartigans’ dip test for unimodality. If the p-value is less than
0.05, there’s enough evidence to claim that the data are not unimodal
(multimodal, at least bimodal).</p>
<pre class="r"><code>use_assay &lt;- assay(se, &#39;cpm&#39;)
i &lt;- colnames(use_assay)[1]
modality_test_df &lt;- foreach(i= colnames(use_assay), .combine = rbind) %do%{
  x &lt;- use_assay[,i]
  dip.res &lt;- dip.test(x, simulate.p.value = FALSE, B = 2000)
  data.frame(
    sample_alias = i,
    dip.D = dip.res$statistic,
    dip.p = dip.res$p.value,
    dip.result = ifelse(dip.res$p.value &lt; 0.05, &#39;multimodal&#39;, &#39;unimodal&#39;)
    )
}

modality_test_df %&lt;&gt;% 
  left_join(group_df_modality) %&gt;% 
  left_join(colData(se) %&gt;% data.frame)</code></pre>
<div
id="supplementary-figure-2d-left-dip-test-table-for-syngeneic-models"
class="section level3" number="0.5.1">
<h3><span class="header-section-number">0.5.1</span> Supplementary
Figure 2D left: Dip test table for Syngeneic models</h3>
<pre class="r"><code>modality_test_df %&gt;% 
  filter(!is.na(modality_group)) %&gt;% 
  filter(mouse_model_global == &#39;immunocompetent&#39;) %&gt;% 
  mutate(
    modality_group = factor(modality_group, levels = names(group_list_modality))
  ) %&gt;%
  group_by(modality_group) %&gt;%
  summarise(
    `Median D` = median(dip.D) %&gt;% round(3),
    n = n(),
    `Multimodal samples (n)` = sum(dip.result == &#39;multimodal&#39;),
    `Unimodal samples (n)` = sum(dip.result == &#39;unimodal&#39;),
    `Multimodal samples (%)` = round(100*`Multimodal samples (n)`/n, 1),
    `Unimodal samples (%)` = round(100*`Unimodal samples (n)`/n, 1)
  ) %&gt;% 
  data.frame(check.names = F)</code></pre>
<pre><code>      modality_group Median D  n Multimodal samples (n)</code></pre>
<p>1 Syngeneic Thy1 0.005 5 1 2 Syngeneic Puro 0.054 15 14 3 Syngeneic
dCas9 + Puro 0.077 5 5 Unimodal samples (n) Multimodal samples (%)
Unimodal samples (%) 1 4 20.0 80.0 2 1 93.3 6.7 3 0 100.0 0.0</p>
</div>
<div id="supplementary-figure-2d-right-dip-test-table-for-nsg-models"
class="section level3" number="0.5.2">
<h3><span class="header-section-number">0.5.2</span> Supplementary
Figure 2D right: Dip test table for NSG models</h3>
<pre class="r"><code>modality_test_df %&gt;% 
  filter(!is.na(modality_group)) %&gt;% 
  filter(mouse_model_global == &#39;immunodeficient&#39;) %&gt;% 
  mutate(
    modality_group = factor(modality_group, levels = names(group_list_modality))
  ) %&gt;%
  group_by(modality_group) %&gt;%
  summarise(
    `Median D` = median(dip.D) %&gt;% round(3),
    n = n(),
    `Multimodal samples (n)` = sum(dip.result == &#39;multimodal&#39;),
    `Unimodal samples (n)` = sum(dip.result == &#39;unimodal&#39;),
    `Multimodal samples (%)` = round(100*`Multimodal samples (n)`/n, 1),
    `Unimodal samples (%)` = round(100*`Unimodal samples (n)`/n, 1)
  ) %&gt;% 
  data.frame(check.names = F)</code></pre>
<pre><code>modality_group Median D  n Multimodal samples (n) Unimodal samples (n)</code></pre>
<p>1 NSG Thy1 0.005 15 2 13 2 NSG Puro 0.005 15 0 15 3 NSG dCas9 + Puro
0.007 5 0 5 Multimodal samples (%) Unimodal samples (%) 1 13.3 86.7 2
0.0 100.0 3 0.0 100.0</p>
</div>
</div>
<div id="muller-plots" class="section level2" number="0.6">
<h2><span class="header-section-number">0.6</span> Muller plots</h2>
<p>Define selected comparisons for muller plots</p>
<pre class="r"><code>comp_list_muller_syngeneic_representative &lt;- list(
 `Thy1` = c(&#39;4T_TumoT_B3&#39;, &#39;4T_LungsT_B3&#39;),
 `Puro` = c(&#39;4T_TumoP_B1&#39;, &#39;4T_LungsP_B1&#39;),
 `dCas9 + Puro` = c(&#39;PlatePositionA3&#39;, &#39;PlatePositionB3&#39;)
)</code></pre>
<p>Generate muller plots</p>
<pre class="r"><code>cpm_threshold &lt;- 10000
comp_list &lt;- comp_list_muller_syngeneic_representative

use_df &lt;- assay(se, &#39;cpm&#39;) %&gt;% data.frame(check.names = FALSE) %&gt;% 
  rownames_to_column(&#39;guide&#39;) %&gt;% 
  mutate(
    gene = rowData(se)$Gene,
    gene = ifelse(grepl(&#39;Non_Target&#39;, gene), &#39;Non_Target&#39;, gene)
  ) %&gt;%
  pivot_longer(-c(guide, gene), names_to = &#39;sample_alias&#39;, values_to = &#39;cpm&#39;)

j &lt;-  names(comp_list)[1]
muller_plots &lt;- foreach(j = names(comp_list)) %do% {
  i &lt;- comp_list[[j]]
  x &lt;- use_df %&gt;% 
    filter(sample_alias %in% i) %&gt;% 
    mutate(
      sample_type = ifelse(sample_alias == i[1], &#39;Primary tumor&#39;, &#39;Lungs&#39;),
      sample_type = factor(sample_type, levels = c(&#39;Primary tumor&#39;, &#39;Lungs&#39;))
    )
  
  # Select guides with cpm &gt; cpm_threshold
  x_guides_keep &lt;- x %&gt;% 
    group_by(sample_type, guide) %&gt;% 
    summarise(cpm = max(cpm)) %&gt;% 
    filter(cpm &gt; cpm_threshold) %&gt;% 
    pull(guide)
  
  # Group guides with cpm &lt; cpm_threshold into the low abundant category, and sum counts for low abundant
  x %&lt;&gt;% 
    mutate(
      guide = ifelse(guide %in%  x_guides_keep, guide, &#39;low abundant&#39;)
    ) %&gt;% 
    group_by(sample_type, guide) %&gt;%
    summarise(cpm = sum(cpm)) %&gt;% 
    ungroup() %&gt;%
    mutate(
      guide = fct_reorder(guide, cpm),
      guide = relevel(guide, ref = &#39;low abundant&#39;)
    )
  
  use_cols &lt;- c(
    &#39;low abundant&#39; = &#39;grey80&#39;,
    colorRampPalette(rev(palette_OkabeIto))(nlevels(x$guide) - 1) %&gt;% set_names(levels(x$guide)[-1])
  )
  
  x %&gt;% 
    ggplot( aes(x = sample_type, y = cpm/10000, group = guide, fill = guide)) + 
    geom_area(colour = alpha(&quot;white&quot;, 0.1), linewidth = 0.08, alpha = 0.8) +
    scale_fill_manual(values = use_cols, guide = guide_legend(ncol = 3)) +
    labs(
      x = &#39;&#39;,
      y = expression(&quot;CPM x 10&quot;^-4),
      title = j,
      fill = &#39;&#39;
    ) +
    scale_x_discrete(expand = c(0, 0)) +
    scale_y_continuous(expand = c(0, 0), limits = c(0, 1000000/10000)) +
    guides(fill = FALSE) +
    theme(
      panel.background = element_rect(fill = &quot;grey90&quot;),
      # plot.margin = margin(0.5, 1.5, 0.5, 0, &quot;cm&quot;),
      plot.margin = margin(0, 0.5, 0, 0, &quot;cm&quot;),
      axis.line = element_line(linewidth = one_pt/4, color = &#39;black&#39;),
      axis.ticks = element_line(linewidth = one_pt/4, color = &#39;black&#39;),
      axis.text =  element_text(size=3),
      axis.title =  element_text(size=3),
      plot.title = element_text(size=3, hjust = 0.5)
    )

}
names(muller_plots) &lt;- names(comp_list)
muller_syngeneic_representative &lt;- muller_plots</code></pre>
<div
id="figure-2c-representative-müller-plots-of-4t1-in-syngeneic-recipients"
class="section level3" number="0.6.1">
<h3><span class="header-section-number">0.6.1</span> Figure 2C:
Representative Müller plots of 4T1 in syngeneic recipients</h3>
<pre class="r"><code>wrap_plots(muller_plots, ncol = 1)</code></pre>
<p><img src="figure/crispr-mm_2215_sgRNA-distribution.Rmd/muller-syngeneic-representative-plots-1.png" width="900" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="stop-knitr" class="section level2" number="0.7">
<h2><span class="header-section-number">0.7</span> Stop knitr</h2>
<pre class="r"><code>knitr::knit_exit()</code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4,h5",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
