<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Francesc Castro-Giner" />

<meta name="date" content="2025-10-31" />

<title>StealTHY CRISPR screen</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.34.0/datatables.js"></script>
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<script src="site_libs/jszip-1.13.6/jszip.min.js"></script>
<link href="site_libs/dt-ext-buttons-1.13.6/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.13.6/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.6/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.6/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.6/js/buttons.print.min.js"></script>
<link href="site_libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet" />
<script src="site_libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="site_libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
<script src="site_libs/selectize-0.12.0/selectize.min.js"></script>
<link href="site_libs/crosstalk-1.2.2/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.2/js/crosstalk.min.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">StealTHY</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="about.html">
    <span class="fa fa-info"></span>
     
    About
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-bar-chart"></span>
     
    Analyses
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Data generation</li>
    <li>
      <a href="0_differential_expression_gsea_gsva.html">Run differential gene expression analysis, GSEA and GSVA</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Main results</li>
    <li>
      <a href="patients_ctc_counts_distribution.html">CTC counts distribution in breast cancer patients</a>
    </li>
    <li>
      <a href="br16_pca.html">PCA analysis of CTC gene expression in NSG-CDX-BR16 mice</a>
    </li>
    <li>
      <a href="br16_dge.html">Differential gene expression report for NSG-CDX-BR16 data</a>
    </li>
    <li>
      <a href="gsea_across_models.html">Report of GSEA and GSVA analysis in NSG-CDX-BR16, NSG-LM2, Patient and NSG-LM2 time kinetics</a>
    </li>
    <li>
      <a href="core_gene_sets.html">Analysis of genes from candidate pathways</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="https://github.com/TheAcetoLab/saini-stealTHY">
    <span class="fa fa-github"></span>
     
  </a>
</li>
<li>
  <a href="https://www.theacetolab.com">
    <span class="fa fa-flask"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">StealTHY CRISPR screen</h1>
<h3 class="subtitle">Mouse libray (n = 2215)</h3>
<h4 class="author">Francesc Castro-Giner</h4>
<h4 class="date">October 31, 2025</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-10-31
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>saini-stealTHY/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240517code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240517)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240517code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240517)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomTheAcetoLabsainistealTHYtree455fa9121fcca51363cf639f7ee170fb108e34c1targetblank455fa91a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/tree/455fa9121fcca51363cf639f7ee170fb108e34c1" target="_blank">455fa91</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomTheAcetoLabsainistealTHYtree455fa9121fcca51363cf639f7ee170fb108e34c1targetblank455fa91a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/TheAcetoLab/saini-stealTHY/tree/455fa9121fcca51363cf639f7ee170fb108e34c1" target="_blank">455fa91</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/templates/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    code/raw_data_processing/rnaseq/p27851_o32062/pipelines/
    Ignored:    configuration/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/crispr/
    Ignored:    data/resources/
    Ignored:    data/rnaseq/
    Ignored:    output/.DS_Store
    Ignored:    output/clinical/
    Ignored:    output/crispr/
    Ignored:    output/rnaseq/

Untracked files:
    Untracked:  analysis/crispr-mm_2215_sgRNA-clonality.md
    Untracked:  analysis/index.md
    Untracked:  analysis/rnaseq-tumor-bulk.md
    Untracked:  analysis/style.css
    Untracked:  code/R-functions/subchunkify.R

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/_site.yml
    Deleted:    analysis/crispr-hsapiens_2180_sgRNA_r1.Rmd
    Deleted:    analysis/crispr-mm_2215_sgRNA-clonality_r1.Rmd
    Deleted:    analysis/crispr-mm_2215_sgRNA-r2oC.Rmd
    Deleted:    analysis/crispr-mm_2215_sgRNA-r2oD.Rmd
    Deleted:    analysis/crispr-mm_2215_sgRNA-r2oD_r1.Rmd
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_ko_culture--over--amhr2_control_culture_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_ko_culture--over--amhr2_control_culture_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_ko_notgfbeta--over--amhr2_ko_culture_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_ko_notgfbeta--over--amhr2_ko_culture_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_overexpression--over--amhr2_control_culture_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_overexpression--over--amhr2_control_culture_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_overexpression--over--amhr2_ko_culture_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_overexpression--over--amhr2_ko_culture_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_overexpression--over--amhr2_ko_notgfbeta_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_overexpression--over--amhr2_ko_notgfbeta_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/figure/gse-dotplot-invivo_amhr2_ko-+-amhr2_ko_culture--over--invivo_control_sgrna-+-amhr2_control_culture_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-invivo_amhr2_ko-+-amhr2_ko_culture--over--invivo_control_sgrna-+-amhr2_control_culture_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/figure/gse-dotplot-invivo_amhr2_ko--over--invivo_control_sgrna_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-invivo_amhr2_ko--over--invivo_control_sgrna_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/file/tmp/dge.xlsx
    Deleted:    analysis/templates/docs/file/tmp/gse_gsea_msigdb.h.xlsx
    Deleted:    analysis/templates/docs/file/tmp/gse_gsego_bp.xlsx
    Deleted:    analysis/templates/docs/file/tmp/gse_gsego_mf.xlsx
    Modified:   analysis/templates/rnaseq-deg-edger.Rmd
    Modified:   code/R-functions/gse_report.r
    Modified:   configuration/rmarkdown/color_palettes.R
    Modified:   configuration/rmarkdown/ggplot_theme.R
    Modified:   update_workflowr.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/crispr-mm_2215_sgRNA-StealTHY.Rmd</code>) and HTML
(<code>docs/crispr-mm_2215_sgRNA-StealTHY.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/455fa9121fcca51363cf639f7ee170fb108e34c1/analysis/crispr-mm_2215_sgRNA-StealTHY.Rmd" target="_blank">455fa91</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-10-31
</td>
<td>
Prepare files for publication
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheAcetoLab/saini-stealTHY/32aca023d6d8209f1aa3df3f177fc383c7569844/docs/crispr-mm_2215_sgRNA-StealTHY.html" target="_blank">32aca02</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-07-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/e391b94d2101a3ebbb4ef002cf4887c5d14738d0/analysis/crispr-mm_2215_sgRNA-StealTHY.Rmd" target="_blank">e391b94</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-07-01
</td>
<td>
update crispr plots
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/00ae28bcb2715ec6f4030d74d92f51359fa43ce4/analysis/crispr-mm_2215_sgRNA-StealTHY.Rmd" target="_blank">00ae28b</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-05-05
</td>
<td>
updated r1oa
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/1c45f78451a8507c8de37da9a7d78c2e7cd92a3a/analysis/crispr-mm_2215_sgRNA-StealTHY.Rmd" target="_blank">1c45f78</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-04-30
</td>
<td>
rm temporary files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/5fab19d765a869233e820f2aed94595a89d5fe9b/analysis/crispr-mm_2215_sgRNA-StealTHY.Rmd" target="_blank">5fab19d</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2024-05-28
</td>
<td>
clean code
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/1920712c9ede8023880e174fc87f3cd238c3d612/analysis/crispr-mm_2215_sgRNA-StealTHY.Rmd" target="_blank">1920712</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2024-05-24
</td>
<td>
added F5
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries-additional-functions-and-data"
class="section level2">
<h2>Load libraries, additional functions and data</h2>
<p>Setup environment</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">results=</span><span class="st">&#39;asis&#39;</span>, <span class="at">echo=</span><span class="cn">TRUE</span>, <span class="at">message=</span><span class="cn">FALSE</span>, <span class="at">warning=</span><span class="cn">FALSE</span>, <span class="at">error=</span><span class="cn">FALSE</span>, <span class="at">fig.align =</span> <span class="st">&#39;center&#39;</span>, <span class="at">fig.width =</span> <span class="fl">3.5</span>, <span class="at">fig.asp =</span> <span class="fl">0.618</span>, <span class="at">dpi =</span> <span class="dv">600</span>, <span class="at">dev =</span> <span class="fu">c</span>(<span class="st">&quot;png&quot;</span>, <span class="st">&quot;pdf&quot;</span>), <span class="at">fig.showtext =</span> <span class="cn">FALSE</span>, <span class="at">engine.opts =</span> <span class="fu">list</span>(<span class="at">bash =</span> <span class="st">&quot;-l&quot;</span>))</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">options</span>(<span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>use_seed <span class="ot">&lt;-</span> <span class="dv">1100101</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="fu">set.seed</span>(use_seed)</span></code></pre></div>
<p>Load packages</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="fu">library</span>(foreach)</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="fu">library</span>(magrittr)</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="fu">library</span>(DT)</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="fu">library</span>(diptest)</span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a><span class="fu">library</span>(ggridges)</span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a><span class="fu">library</span>(ggh4x)</span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a><span class="fu">library</span>(colorblindr)</span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a><span class="fu">library</span>(ggbeeswarm)</span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a></span>
<span id="cb2-17"><a href="#cb2-17" tabindex="-1"></a><span class="fu">library</span>(MAGeCKFlute)</span></code></pre></div>
<p>Load ggplot theme</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;./configuration/rmarkdown/ggplot_theme.R&quot;</span>)</span></code></pre></div>
<p>Load ggplot theme</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;./configuration/rmarkdown/color_palettes.R&quot;</span>)</span></code></pre></div>
<p>Load Summarized Experiment object</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">file.path</span>(params<span class="sc">$</span>data_dir, <span class="st">&#39;se.rds&#39;</span>))</span></code></pre></div>
<p>Filter samples used for this analysis</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>use_cols <span class="ot">&lt;-</span> <span class="fu">colData</span>(se) <span class="sc">%&gt;%</span> </span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>  data.frame <span class="sc">%&gt;%</span> </span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>  rownames</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a></span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a>se <span class="ot">&lt;-</span> se[,use_cols]</span></code></pre></div>
<p>Data wrangling</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>chunk_colData <span class="ot">&lt;-</span> <span class="fu">colData</span>(se) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>    <span class="at">vector_short =</span> <span class="fu">ifelse</span>(vector_short <span class="sc">==</span> <span class="st">&#39;Puro_EF1alpha_SpCas9&#39;</span>, </span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>                          <span class="st">&#39;Cas9+Puro&#39;</span>, vector_short),</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>    <span class="at">vector_short =</span> <span class="fu">factor</span>(vector_short, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&#39;Thy1&#39;</span>, <span class="st">&#39;Puro&#39;</span>, <span class="st">&#39;Cas9+Puro&#39;</span>)),</span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>    <span class="at">mouse_model_short =</span> <span class="fu">case_match</span>(mouse_model_global,</span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>                                   <span class="st">&#39;immunodeficient&#39;</span> <span class="sc">~</span> <span class="st">&#39;NSG&#39;</span>,</span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a>                                   <span class="st">&#39;immunocompetent&#39;</span> <span class="sc">~</span> <span class="st">&#39;Syngeneic&#39;</span></span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a>                                   ),</span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a>    <span class="at">mouse_model_short =</span> <span class="fu">factor</span>(mouse_model_short, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&#39;NSG&#39;</span>, <span class="st">&#39;Syngeneic&#39;</span>)),</span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a>    <span class="at">vector_mhost =</span> <span class="fu">paste0</span>(vector_short, <span class="st">&#39;, &#39;</span>, mouse_model_short),</span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a>    <span class="at">model_vector_mhost =</span> <span class="fu">paste0</span>(donor,<span class="st">&#39;, &#39;</span>, vector_short, <span class="st">&#39;, &#39;</span>, mouse_model_short),</span>
<span id="cb7-13"><a href="#cb7-13" tabindex="-1"></a>    </span>
<span id="cb7-14"><a href="#cb7-14" tabindex="-1"></a>    <span class="at">sample_type =</span> <span class="fu">case_match</span>(sample_type, </span>
<span id="cb7-15"><a href="#cb7-15" tabindex="-1"></a>                             <span class="st">&#39;whole_tumor&#39;</span> <span class="sc">~</span> <span class="st">&#39;Primary tumor&#39;</span>, </span>
<span id="cb7-16"><a href="#cb7-16" tabindex="-1"></a>                             <span class="st">&#39;whole_lung&#39;</span> <span class="sc">~</span> <span class="st">&#39;Lung mets&#39;</span>),</span>
<span id="cb7-17"><a href="#cb7-17" tabindex="-1"></a>    <span class="at">sample_type =</span> <span class="fu">factor</span>(sample_type, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&#39;Primary tumor&#39;</span>, <span class="st">&#39;Lung mets&#39;</span>))</span>
<span id="cb7-18"><a href="#cb7-18" tabindex="-1"></a>  )</span>
<span id="cb7-19"><a href="#cb7-19" tabindex="-1"></a></span>
<span id="cb7-20"><a href="#cb7-20" tabindex="-1"></a><span class="fu">colData</span>(se) <span class="ot">&lt;-</span> chunk_colData <span class="sc">%&gt;%</span> DataFrame</span></code></pre></div>
</div>
<div id="configure-analyses" class="section level2">
<h2>Configure analyses</h2>
<div id="sgrna-distribution" class="section level3">
<h3>sgRNA distribution</h3>
<p>Define comparisons for modality and Kolmogorov-Smirnov test</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">colData</span>(se) <span class="sc">%&gt;%</span> data.frame</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>group_list_modality <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>  <span class="st">`</span><span class="at">StealTHY syngeneic</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>    <span class="fu">filter</span>(vector_mmodel <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&#39;4T1-KO-Thy1-BALB&#39;</span>, <span class="st">&#39;MVT1-KO-Thy1-FVB&#39;</span>, <span class="st">&#39;Py2T-KO-Thy1-FVB&#39;</span>) <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Primary tumor&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a>    <span class="fu">pull</span>(sample_alias),</span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a>  </span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a>  <span class="st">`</span><span class="at">Cas9+Puro syngeneic</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a>    <span class="fu">filter</span>(vector_mmodel <span class="sc">==</span> <span class="st">&#39;4T1-KO-Puro_EF1alpha_SpCas9-BALB&#39;</span> <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Primary tumor&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a>    <span class="fu">pull</span>(sample_alias),</span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a>  </span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a>  <span class="st">`</span><span class="at">StealTHY NSG</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a>    <span class="fu">filter</span>(vector_mmodel <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&#39;4T1-KO-Thy1-NSG&#39;</span>, <span class="st">&#39;MVT1-KO-Thy1-NSG&#39;</span>, <span class="st">&#39;Py2T-KO-Thy1-NSG&#39;</span>) <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Primary tumor&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb8-14"><a href="#cb8-14" tabindex="-1"></a>    <span class="fu">pull</span>(sample_alias),</span>
<span id="cb8-15"><a href="#cb8-15" tabindex="-1"></a>  </span>
<span id="cb8-16"><a href="#cb8-16" tabindex="-1"></a>  <span class="st">`</span><span class="at">Cas9+Puro NSG</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb8-17"><a href="#cb8-17" tabindex="-1"></a>    <span class="fu">filter</span>(vector_mmodel <span class="sc">==</span> <span class="st">&#39;4T1-KO-Puro_EF1alpha_SpCas9-NSG&#39;</span> <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Primary tumor&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-18"><a href="#cb8-18" tabindex="-1"></a>    <span class="fu">pull</span>(sample_alias),</span>
<span id="cb8-19"><a href="#cb8-19" tabindex="-1"></a>  </span>
<span id="cb8-20"><a href="#cb8-20" tabindex="-1"></a>  <span class="st">`</span><span class="at">StealTHY syngeneic 4T1</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb8-21"><a href="#cb8-21" tabindex="-1"></a>    <span class="fu">filter</span>(vector_mmodel <span class="sc">==</span> <span class="st">&#39;4T1-KO-Thy1-BALB&#39;</span> <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Primary tumor&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb8-22"><a href="#cb8-22" tabindex="-1"></a>    <span class="fu">pull</span>(sample_alias)</span>
<span id="cb8-23"><a href="#cb8-23" tabindex="-1"></a>  </span>
<span id="cb8-24"><a href="#cb8-24" tabindex="-1"></a>)</span>
<span id="cb8-25"><a href="#cb8-25" tabindex="-1"></a></span>
<span id="cb8-26"><a href="#cb8-26" tabindex="-1"></a></span>
<span id="cb8-27"><a href="#cb8-27" tabindex="-1"></a>comp_list_modality <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb8-28"><a href="#cb8-28" tabindex="-1"></a>  <span class="st">`</span><span class="at">StealTHY syngeneic vs. StealTHY NSG</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb8-29"><a href="#cb8-29" tabindex="-1"></a>    <span class="st">`</span><span class="at">StealTHY syngeneic</span><span class="st">`</span> <span class="ot">=</span> group_list_modality<span class="sc">$</span><span class="st">`</span><span class="at">StealTHY syngeneic</span><span class="st">`</span>,</span>
<span id="cb8-30"><a href="#cb8-30" tabindex="-1"></a>    <span class="st">`</span><span class="at">StealTHY NSG</span><span class="st">`</span> <span class="ot">=</span> group_list_modality<span class="sc">$</span><span class="st">`</span><span class="at">StealTHY NSG</span><span class="st">`</span></span>
<span id="cb8-31"><a href="#cb8-31" tabindex="-1"></a>  ),</span>
<span id="cb8-32"><a href="#cb8-32" tabindex="-1"></a>  <span class="st">`</span><span class="at">StealTHY syngeneic vs. Cas9+Puro</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb8-33"><a href="#cb8-33" tabindex="-1"></a>    <span class="st">`</span><span class="at">StealTHY syngeneic</span><span class="st">`</span> <span class="ot">=</span> group_list_modality<span class="sc">$</span><span class="st">`</span><span class="at">StealTHY syngeneic 4T1</span><span class="st">`</span>,</span>
<span id="cb8-34"><a href="#cb8-34" tabindex="-1"></a>    <span class="st">`</span><span class="at">Cas9+Puro syngeneic</span><span class="st">`</span> <span class="ot">=</span> group_list_modality<span class="sc">$</span><span class="st">`</span><span class="at">Cas9+Puro syngeneic</span><span class="st">`</span></span>
<span id="cb8-35"><a href="#cb8-35" tabindex="-1"></a>  )</span>
<span id="cb8-36"><a href="#cb8-36" tabindex="-1"></a>)</span>
<span id="cb8-37"><a href="#cb8-37" tabindex="-1"></a></span>
<span id="cb8-38"><a href="#cb8-38" tabindex="-1"></a>group_df_modality <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i =</span> <span class="fu">names</span>(group_list_modality), <span class="at">.combine =</span> rbind) <span class="sc">%do%</span> {</span>
<span id="cb8-39"><a href="#cb8-39" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb8-40"><a href="#cb8-40" tabindex="-1"></a>    <span class="at">sample_alias =</span> group_list_modality[[i]],</span>
<span id="cb8-41"><a href="#cb8-41" tabindex="-1"></a>    <span class="at">modality_group =</span> i</span>
<span id="cb8-42"><a href="#cb8-42" tabindex="-1"></a>  )</span>
<span id="cb8-43"><a href="#cb8-43" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="mageck-analysis" class="section level3">
<h3>MAGeCK analysis</h3>
<div id="list-of-comparisons" class="section level4">
<h4>List of comparisons</h4>
<p>List of comparisons for differential abundance using MAGeCK</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">colData</span>(se) <span class="sc">%&gt;%</span> data.frame</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>mageck_comp_list <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a>  <span class="st">`</span><span class="at">StealTHY KO Syngeneic</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a>    <span class="st">`</span><span class="at">Primary tumor</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(vector_mmodel <span class="sc">%in%</span> <span class="fu">list</span>(<span class="st">&#39;4T1-KO-Thy1-BALB&#39;</span>, <span class="st">&#39;MVT1-KO-Thy1-FVB&#39;</span>, <span class="st">&#39;Py2T-KO-Thy1-FVB&#39;</span>) <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Primary tumor&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias),</span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a>    <span class="st">`</span><span class="at">Lung mets</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(vector_mmodel <span class="sc">%in%</span> <span class="fu">list</span>(<span class="st">&#39;4T1-KO-Thy1-BALB&#39;</span>, <span class="st">&#39;MVT1-KO-Thy1-FVB&#39;</span>, <span class="st">&#39;Py2T-KO-Thy1-FVB&#39;</span>) <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Lung mets&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias)</span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a>    ),</span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a>  <span class="st">`</span><span class="at">StealTHY KO NSG</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a>    <span class="st">`</span><span class="at">Primary tumor</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(global_vector_mmodel <span class="sc">==</span> <span class="st">&#39;whole_tumor-KO-Thy1-immunodeficient&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias),</span>
<span id="cb9-10"><a href="#cb9-10" tabindex="-1"></a>    <span class="st">`</span><span class="at">Lung mets</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(global_vector_mmodel <span class="sc">==</span> <span class="st">&#39;whole_lung-KO-Thy1-immunodeficient&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias)</span>
<span id="cb9-11"><a href="#cb9-11" tabindex="-1"></a>  ),</span>
<span id="cb9-12"><a href="#cb9-12" tabindex="-1"></a>  <span class="st">`</span><span class="at">Cas9+Puro Syngeneic</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb9-13"><a href="#cb9-13" tabindex="-1"></a>    <span class="st">`</span><span class="at">Primary tumor</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(vector_mmodel <span class="sc">==</span> <span class="st">&#39;4T1-KO-Puro_EF1alpha_SpCas9-BALB&#39;</span> <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Primary tumor&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias),</span>
<span id="cb9-14"><a href="#cb9-14" tabindex="-1"></a>    <span class="st">`</span><span class="at">Lung mets</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(vector_mmodel <span class="sc">==</span> <span class="st">&#39;4T1-KO-Puro_EF1alpha_SpCas9-BALB&#39;</span> <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Lung mets&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias)</span>
<span id="cb9-15"><a href="#cb9-15" tabindex="-1"></a>  ),</span>
<span id="cb9-16"><a href="#cb9-16" tabindex="-1"></a>  </span>
<span id="cb9-17"><a href="#cb9-17" tabindex="-1"></a>  <span class="st">`</span><span class="at">StealTHY KO over dCas9 In vitro</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb9-18"><a href="#cb9-18" tabindex="-1"></a>    <span class="st">`</span><span class="at">StealTHY</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(global_vector_mmodel_invitro_cas9transfection <span class="sc">==</span> <span class="st">&#39;bulk-KO-Thy1-Invitro-pre-Cas9&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias),</span>
<span id="cb9-19"><a href="#cb9-19" tabindex="-1"></a>    <span class="st">`</span><span class="at">dCas9</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(global_vector_mmodel_invitro_cas9transfection_grouped <span class="sc">==</span> <span class="st">&#39;KO-Thy1-Invitro-post-Cas9&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias)</span>
<span id="cb9-20"><a href="#cb9-20" tabindex="-1"></a>    ),</span>
<span id="cb9-21"><a href="#cb9-21" tabindex="-1"></a>  </span>
<span id="cb9-22"><a href="#cb9-22" tabindex="-1"></a>  <span class="st">`</span><span class="at">StealTHY KO over dCas9 In vivo</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb9-23"><a href="#cb9-23" tabindex="-1"></a>    <span class="st">`</span><span class="at">StealTHY</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(vector_mmodel <span class="sc">%in%</span> <span class="fu">list</span>(<span class="st">&#39;4T1-KO-Thy1-BALB&#39;</span>, <span class="st">&#39;MVT1-KO-Thy1-FVB&#39;</span>, <span class="st">&#39;Py2T-KO-Thy1-FVB&#39;</span>) <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Primary tumor&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias),</span>
<span id="cb9-24"><a href="#cb9-24" tabindex="-1"></a>    <span class="st">`</span><span class="at">dCas9 mets</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(vector_mmodel <span class="sc">%in%</span> <span class="fu">list</span>(<span class="st">&#39;4T1-noKO-Thy1-BALB&#39;</span>, <span class="st">&#39;MVT1-noKO-Thy1-FVB&#39;</span>, <span class="st">&#39;Py2T-noKO-Thy1-FVB&#39;</span>) <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Primary tumor&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias)</span>
<span id="cb9-25"><a href="#cb9-25" tabindex="-1"></a>    )</span>
<span id="cb9-26"><a href="#cb9-26" tabindex="-1"></a>)</span>
<span id="cb9-27"><a href="#cb9-27" tabindex="-1"></a></span>
<span id="cb9-28"><a href="#cb9-28" tabindex="-1"></a></span>
<span id="cb9-29"><a href="#cb9-29" tabindex="-1"></a>mageck_comp_list_names <span class="ot">&lt;-</span> <span class="fu">names</span>(mageck_comp_list) </span>
<span id="cb9-30"><a href="#cb9-30" tabindex="-1"></a><span class="fu">names</span>(mageck_comp_list_names) <span class="ot">&lt;-</span> <span class="fu">names</span>(mageck_comp_list) <span class="sc">%&gt;%</span> <span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>, <span class="st">&quot;_&quot;</span>,.) <span class="sc">%&gt;%</span> tolower</span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>data_comp <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i =</span> <span class="fu">names</span>(mageck_comp_list), <span class="at">.combine =</span> rbind) <span class="sc">%do%</span> {</span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>  <span class="fu">c</span>(</span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>    <span class="at">Comparison =</span> i,</span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a>    <span class="at">Case =</span> <span class="fu">paste</span>(mageck_comp_list[[i]][[<span class="dv">1</span>]], <span class="at">collapse =</span> <span class="st">&quot; &quot;</span>),</span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a>    <span class="at">Control =</span> <span class="fu">paste</span>(mageck_comp_list[[i]][[<span class="dv">2</span>]], <span class="at">collapse =</span> <span class="st">&quot; &quot;</span>)</span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a>  )</span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a>}</span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a></span>
<span id="cb10-9"><a href="#cb10-9" tabindex="-1"></a>data_comp <span class="sc">%&gt;%</span> </span>
<span id="cb10-10"><a href="#cb10-10" tabindex="-1"></a>  <span class="fu">datatable</span>(., </span>
<span id="cb10-11"><a href="#cb10-11" tabindex="-1"></a>            <span class="at">rownames =</span> <span class="cn">FALSE</span>, </span>
<span id="cb10-12"><a href="#cb10-12" tabindex="-1"></a>            <span class="at">filter =</span> <span class="st">&#39;top&#39;</span>, </span>
<span id="cb10-13"><a href="#cb10-13" tabindex="-1"></a>            <span class="at">caption =</span> <span class="st">&#39;List of comparisons for MAGeCK analysis&#39;</span>,</span>
<span id="cb10-14"><a href="#cb10-14" tabindex="-1"></a>            <span class="at">extensions =</span> <span class="st">&#39;Buttons&#39;</span>, </span>
<span id="cb10-15"><a href="#cb10-15" tabindex="-1"></a>            <span class="at">options =</span> <span class="fu">list</span>(</span>
<span id="cb10-16"><a href="#cb10-16" tabindex="-1"></a>              <span class="at">dom =</span> <span class="st">&#39;Blfrtip&#39;</span>,</span>
<span id="cb10-17"><a href="#cb10-17" tabindex="-1"></a>              <span class="at">buttons =</span> <span class="fu">c</span>(<span class="st">&#39;csv&#39;</span>, <span class="st">&#39;excel&#39;</span>),</span>
<span id="cb10-18"><a href="#cb10-18" tabindex="-1"></a>              <span class="at">columnDefs =</span> <span class="fu">list</span>(</span>
<span id="cb10-19"><a href="#cb10-19" tabindex="-1"></a>                <span class="fu">list</span>(<span class="at">width =</span> <span class="dv">120</span>, <span class="at">targets =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>)</span>
<span id="cb10-20"><a href="#cb10-20" tabindex="-1"></a>              )</span>
<span id="cb10-21"><a href="#cb10-21" tabindex="-1"></a>            )</span>
<span id="cb10-22"><a href="#cb10-22" tabindex="-1"></a>            )</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-0e41329c1963262c826a" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-0e41329c1963262c826a">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption>List of comparisons for MAGeCK analysis<\/caption>","data":[["StealTHY KO Syngeneic","StealTHY KO NSG","Cas9+Puro Syngeneic","StealTHY KO over dCas9 In vitro","StealTHY KO over dCas9 In vivo"],["CRISPRKO_Clone_1 CRISPRKO_Clone_10 CRISPRKO_Clone_11 CRISPRKO_Clone_12 CRISPRKO_Clone_13 CRISPRKO_Clone_14 CRISPRKO_Clone_2 CRISPRKO_Clone_20 CRISPRKO_Clone_21 CRISPRKO_Clone_22 CRISPRKO_Clone_23 CRISPRKO_Clone_24 CRISPRKO_Clone_3 CRISPRKO_Clone_4 CRISPRKO","CRISPRKO_Clone_45 CRISPRKO_Clone_46 CRISPRKO_Clone_47 CRISPRKO_Clone_48 CRISPRKO_Clone_49 CRISPRKO_Clone_5 CRISPRKO_Clone_50 CRISPRKO_Clone_51 CRISPRKO_Clone_52 CRISPRKO_Clone_53 CRISPRKO_Clone_54 CRISPRKO_Clone_6 CRISPRKO_Clone_7 CRISPRKO_Clone_8 CRISPRKO_Clone_9","PlatePositionA5 PlatePositionA6 PlatePositionC5 PlatePositionC6 PlatePositionE5 PlatePositionE6 PlatePositionG4 PlatePositionG5 PlatePositionG6","CRISPRKO_Clone_61 CRISPRKO_Clone_64 CRISPRKO_Clone_67","CRISPRKO_Clone_1 CRISPRKO_Clone_10 CRISPRKO_Clone_11 CRISPRKO_Clone_12 CRISPRKO_Clone_13 CRISPRKO_Clone_14 CRISPRKO_Clone_2 CRISPRKO_Clone_20 CRISPRKO_Clone_21 CRISPRKO_Clone_22 CRISPRKO_Clone_23 CRISPRKO_Clone_24 CRISPRKO_Clone_3 CRISPRKO_Clone_4 CRISPRKO"],["CRISPRKO_Clone_15 CRISPRKO_Clone_16 CRISPRKO_Clone_17 CRISPRKO_Clone_18 CRISPRKO_Clone_19 CRISPRKO_Clone_30 CRISPRKO_Clone_31 CRISPRKO_Clone_32 CRISPRKO_Clone_33 CRISPRKO_Clone_34 CRISPRKO_Clone_35 CRISPRKO_Clone_36 CRISPRKO_Clone_37 CRISPRKO_Clone_38 CRISPRKO_Clone_39","CRISPRKO_Clone_25 CRISPRKO_Clone_26 CRISPRKO_Clone_27 CRISPRKO_Clone_28 CRISPRKO_Clone_29 CRISPRKO_Clone_40 CRISPRKO_Clone_41 CRISPRKO_Clone_42 CRISPRKO_Clone_43 CRISPRKO_Clone_44 CRISPRKO_Clone_55 CRISPRKO_Clone_56 CRISPRKO_Clone_57 CRISPRKO_Clone_58 CRISPRKO_Clone_59","PlatePositionB5 PlatePositionD5 PlatePositionD6 PlatePositionF3 PlatePositionF5 PlatePositionF6 PlatePositionH4 PlatePositionH5 PlatePositionH6","CRISPRKO_Clone_60 CRISPRKO_Clone_65 CRISPRKO_Clone_68 PlatePositionA11 PlatePositionB11 PlatePositionC11 PlatePositionD11 PlatePositionG10 PlatePositionH10","4T_TumoT_B1 4T_TumoT_B2 4T_TumoT_B3 4T_TumoT_B4 4T_TumoT_B5 MVT_TumoT_F1 MVT_TumoT_F2 MVT_TumoT_F3 MVT_TumoT_F4 MVT_TumoT_F5 PY_TumoT_F1 PY_TumoT_F2 PY_TumoT_F3 PY_TumoT_F4 PY_TumoT_F5"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Comparison<\/th>\n      <th>Case<\/th>\n      <th>Control<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["csv","excel"],"columnDefs":[{"width":120,"targets":[1,2]},{"name":"Comparison","targets":0},{"name":"Case","targets":1},{"name":"Control","targets":2}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="genes-to-remove" class="section level4">
<h4>Genes to remove</h4>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>min_counts <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>min_prop <span class="ot">&lt;-</span> <span class="fl">0.30</span></span></code></pre></div>
<p>Remove genes not expressed in the RNA-seq experiment: we keep genes
that have counts above 1 counts in 30 % of the samples from at least one
group. The group is defined by the cell model. We consider only non-NSG
samples for this analysis. additionally we also remove
<em>Gt(ROSA)26Sor</em> and <em>Cd4</em> (they cannot be expressed by
cancer cells).</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="co"># Read RNA-seq data</span></span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a>se_rna <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(params<span class="sc">$</span>se_rnaseq_bulk)</span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a><span class="fu">assay</span>(se_rna, <span class="st">&#39;cpm&#39;</span>) <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">cpm</span>(se_rna)</span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a></span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a><span class="co"># CRISPR library genes</span></span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a>crispr_genes <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">rowData</span>(se)<span class="sc">$</span>Gene) <span class="sc">%&gt;%</span> <span class="fu">grep</span>(<span class="st">&#39;Non_Target&#39;</span>, ., <span class="at">invert =</span> T, <span class="at">value =</span> T)</span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a></span>
<span id="cb12-8"><a href="#cb12-8" tabindex="-1"></a><span class="co"># RNA-seq : Select genes and non-NSG samples</span></span>
<span id="cb12-9"><a href="#cb12-9" tabindex="-1"></a>use_cols <span class="ot">&lt;-</span> <span class="fu">colData</span>(se_rna)<span class="sc">$</span>mouse_model <span class="sc">!=</span> <span class="st">&#39;NSG&#39;</span></span>
<span id="cb12-10"><a href="#cb12-10" tabindex="-1"></a>use_se <span class="ot">&lt;-</span> se_rna[<span class="fu">rowData</span>(se_rna)<span class="sc">$</span>gene_name <span class="sc">%in%</span> crispr_genes, use_cols]</span>
<span id="cb12-11"><a href="#cb12-11" tabindex="-1"></a></span>
<span id="cb12-12"><a href="#cb12-12" tabindex="-1"></a><span class="co"># Proportion of samples with min_counts at each group</span></span>
<span id="cb12-13"><a href="#cb12-13" tabindex="-1"></a>use_group <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(use_se<span class="sc">$</span>donor)</span>
<span id="cb12-14"><a href="#cb12-14" tabindex="-1"></a>counts_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(use_se, <span class="st">&#39;counts&#39;</span>)</span>
<span id="cb12-15"><a href="#cb12-15" tabindex="-1"></a>counts_stats <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i=</span><span class="fu">levels</span>(use_group), <span class="at">.combine =</span> cbind) <span class="sc">%do%</span> {</span>
<span id="cb12-16"><a href="#cb12-16" tabindex="-1"></a>  use_cols <span class="ot">&lt;-</span> use_se<span class="sc">$</span>donor <span class="sc">==</span> i</span>
<span id="cb12-17"><a href="#cb12-17" tabindex="-1"></a>  <span class="fu">rowSums</span>(counts_mat[,use_cols] <span class="sc">&gt;=</span> min_counts)  <span class="sc">/</span> <span class="fu">sum</span>(use_cols)</span>
<span id="cb12-18"><a href="#cb12-18" tabindex="-1"></a>} <span class="sc">%&gt;%</span> data.frame</span>
<span id="cb12-19"><a href="#cb12-19" tabindex="-1"></a><span class="fu">colnames</span>(counts_stats) <span class="ot">&lt;-</span> <span class="fu">levels</span>(use_group)</span>
<span id="cb12-20"><a href="#cb12-20" tabindex="-1"></a>counts_stats<span class="sc">$</span>all <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(counts_mat <span class="sc">&gt;=</span> min_counts)  <span class="sc">/</span> <span class="fu">ncol</span>(counts_mat)</span>
<span id="cb12-21"><a href="#cb12-21" tabindex="-1"></a>  </span>
<span id="cb12-22"><a href="#cb12-22" tabindex="-1"></a><span class="co"># Keep genes with min_counts threshold in &gt;=min_prop of samples in at least 1 group</span></span>
<span id="cb12-23"><a href="#cb12-23" tabindex="-1"></a>rows_to_keep <span class="ot">&lt;-</span> <span class="fu">rownames</span>(counts_stats)[<span class="fu">rowSums</span>(counts_stats <span class="sc">&gt;=</span> min_prop) <span class="sc">&gt;</span> <span class="dv">0</span>]</span>
<span id="cb12-24"><a href="#cb12-24" tabindex="-1"></a></span>
<span id="cb12-25"><a href="#cb12-25" tabindex="-1"></a><span class="co"># Change to gene names</span></span>
<span id="cb12-26"><a href="#cb12-26" tabindex="-1"></a>genes_to_keep <span class="ot">&lt;-</span> <span class="fu">rowData</span>(use_se[rows_to_keep,])<span class="sc">$</span>gene_name <span class="sc">%&gt;%</span> </span>
<span id="cb12-27"><a href="#cb12-27" tabindex="-1"></a>  unique</span>
<span id="cb12-28"><a href="#cb12-28" tabindex="-1"></a></span>
<span id="cb12-29"><a href="#cb12-29" tabindex="-1"></a><span class="co"># Remove Rosa26 and Cd4 : cannot be expressed by cancer cells</span></span>
<span id="cb12-30"><a href="#cb12-30" tabindex="-1"></a>genes_to_keep <span class="ot">&lt;-</span> genes_to_keep[<span class="sc">!</span>genes_to_keep <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&#39;Gt(ROSA)26Sor&#39;</span>, <span class="st">&#39;Cd4&#39;</span>)]</span>
<span id="cb12-31"><a href="#cb12-31" tabindex="-1"></a></span>
<span id="cb12-32"><a href="#cb12-32" tabindex="-1"></a><span class="co"># Define genes to remove</span></span>
<span id="cb12-33"><a href="#cb12-33" tabindex="-1"></a>genes_to_rm <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(crispr_genes, genes_to_keep)</span>
<span id="cb12-34"><a href="#cb12-34" tabindex="-1"></a></span>
<span id="cb12-35"><a href="#cb12-35" tabindex="-1"></a><span class="co"># Show stats for removed genes</span></span>
<span id="cb12-36"><a href="#cb12-36" tabindex="-1"></a>counts_stats<span class="sc">$</span>gene_name <span class="ot">&lt;-</span> <span class="fu">rowData</span>(use_se[<span class="fu">rownames</span>(counts_stats),])<span class="sc">$</span>gene_name</span>
<span id="cb12-37"><a href="#cb12-37" tabindex="-1"></a>counts_stats <span class="sc">%&lt;&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_name, <span class="fu">everything</span>()) </span>
<span id="cb12-38"><a href="#cb12-38" tabindex="-1"></a></span>
<span id="cb12-39"><a href="#cb12-39" tabindex="-1"></a>counts_stats_filterByExpr <span class="ot">&lt;-</span> counts_stats</span>
<span id="cb12-40"><a href="#cb12-40" tabindex="-1"></a></span>
<span id="cb12-41"><a href="#cb12-41" tabindex="-1"></a><span class="co"># Filter Crispr Data</span></span>
<span id="cb12-42"><a href="#cb12-42" tabindex="-1"></a>keep_gene_guides <span class="ot">&lt;-</span> <span class="fu">rownames</span>(se)[<span class="fu">rowData</span>(se)<span class="sc">$</span>Gene <span class="sc">%in%</span> genes_to_keep]</span>
<span id="cb12-43"><a href="#cb12-43" tabindex="-1"></a>keep_ctrl_guides <span class="ot">&lt;-</span> <span class="fu">rownames</span>(se)[<span class="fu">grepl</span>(<span class="st">&#39;Non_Target&#39;</span>, <span class="fu">rowData</span>(se)<span class="sc">$</span>Gene)]</span></code></pre></div>
<p>The table below shows the proportion of samples with counts &gt;= 1in
each group for genes removed (n = 66`).</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a>counts_stats_filterByExpr <span class="sc">%&gt;%</span> </span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>  <span class="fu">filter</span>(gene_name <span class="sc">%in%</span> genes_to_rm) <span class="sc">%&gt;%</span> </span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(all)) <span class="sc">%&gt;%</span> </span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a>  <span class="fu">datatable</span>(., </span>
<span id="cb13-5"><a href="#cb13-5" tabindex="-1"></a>            <span class="at">rownames =</span> <span class="cn">FALSE</span>, </span>
<span id="cb13-6"><a href="#cb13-6" tabindex="-1"></a>            <span class="at">filter =</span> <span class="st">&#39;top&#39;</span>, </span>
<span id="cb13-7"><a href="#cb13-7" tabindex="-1"></a>            <span class="at">caption =</span> <span class="st">&#39;Statistics&#39;</span>,</span>
<span id="cb13-8"><a href="#cb13-8" tabindex="-1"></a>            <span class="at">extensions =</span> <span class="st">&#39;Buttons&#39;</span>, </span>
<span id="cb13-9"><a href="#cb13-9" tabindex="-1"></a>            <span class="at">options =</span> <span class="fu">list</span>(</span>
<span id="cb13-10"><a href="#cb13-10" tabindex="-1"></a>              <span class="at">dom =</span> <span class="st">&#39;Blfrtip&#39;</span>,</span>
<span id="cb13-11"><a href="#cb13-11" tabindex="-1"></a>              <span class="at">buttons =</span> <span class="fu">c</span>(<span class="st">&#39;csv&#39;</span>, <span class="st">&#39;excel&#39;</span>)</span>
<span id="cb13-12"><a href="#cb13-12" tabindex="-1"></a>              )) <span class="sc">%&gt;%</span> </span>
<span id="cb13-13"><a href="#cb13-13" tabindex="-1"></a>  <span class="fu">formatRound</span>(<span class="at">columns =</span> <span class="dv">2</span><span class="sc">:</span><span class="dv">5</span>)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-cf74d9caecbce7389320" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-cf74d9caecbce7389320">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0\" data-max=\"1\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0\" data-max=\"0.916666666666667\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0\" data-max=\"0.416666666666667\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0\" data-max=\"0.777777777777778\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption>Statistics<\/caption>","data":[["Cd4","Bmp5","Acvr1c","Il12a","Il4","Mpl","Fgf23","Gdf2","Cxcl15","Il31","Lep","Ccr1l1","Nrg3","Cd40lg","Il21","Angptl3","Il5ra","Fgf3","Il13","Fgf14","Tnfrsf17","Cd70","Fgf8","Cxcr1","Ifnk","Ifna12","Ifna2","Fgf6","Fgf4","Nodal","Il9","Il17a","Mstn","Erbb4","Il1f10","Ccl21b","Ccl27b","Il11ra2","Ifna15","Ifna14","Ifna9","Ifna7","Ifna11","Ifna6","Ifna5","Ifna4","Ifna1","Ccl26","Ifnl2","Ifnl3","Klk1b4","Inhbc","Igfbp1","Il3","Prl","Il25","Fgf12","Dkk1","Fgf16"],[1,0.25,0.25,0.25,0.25,0.08333333333333333,0.25,0.08333333333333333,0.1666666666666667,0.1666666666666667,0,0,0,0.1666666666666667,0.1666666666666667,0,0.08333333333333333,0,0.1666666666666667,0,0.1666666666666667,0,0.08333333333333333,0,0.08333333333333333,0,0,0,0,0,0.08333333333333333,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0.9166666666666666,0.25,0.08333333333333333,0.1666666666666667,0.08333333333333333,0,0.08333333333333333,0.25,0,0,0.08333333333333333,0.1666666666666667,0,0.08333333333333333,0,0.1666666666666667,0.08333333333333333,0,0,0.08333333333333333,0,0.1666666666666667,0,0.08333333333333333,0,0.08333333333333333,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0.4166666666666667,0.1666666666666667,0.08333333333333333,0,0.08333333333333333,0.25,0,0,0.08333333333333333,0.08333333333333333,0.1666666666666667,0.08333333333333333,0.25,0,0,0,0,0.1666666666666667,0,0.08333333333333333,0,0,0.08333333333333333,0,0,0,0.08333333333333333,0.08333333333333333,0.08333333333333333,0.08333333333333333,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0.7777777777777778,0.2222222222222222,0.1388888888888889,0.1388888888888889,0.1388888888888889,0.1111111111111111,0.1111111111111111,0.1111111111111111,0.08333333333333333,0.08333333333333333,0.08333333333333333,0.08333333333333333,0.08333333333333333,0.08333333333333333,0.05555555555555555,0.05555555555555555,0.05555555555555555,0.05555555555555555,0.05555555555555555,0.05555555555555555,0.05555555555555555,0.05555555555555555,0.05555555555555555,0.02777777777777778,0.02777777777777778,0.02777777777777778,0.02777777777777778,0.02777777777777778,0.02777777777777778,0.02777777777777778,0.02777777777777778,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>gene_name<\/th>\n      <th>4T1<\/th>\n      <th>MVT1<\/th>\n      <th>Py2T<\/th>\n      <th>all<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["csv","excel"],"columnDefs":[{"targets":1,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"targets":2,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"targets":3,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"targets":4,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"className":"dt-right","targets":[1,2,3,4]},{"name":"gene_name","targets":0},{"name":"4T1","targets":1},{"name":"MVT1","targets":2},{"name":"Py2T","targets":3},{"name":"all","targets":4}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render","options.columnDefs.3.render"],"jsHooks":[]}</script>
</div>
<div id="run-mageck" class="section level4">
<h4>Run MAGeCK</h4>
<p>Generate MAGeCK test scripts. To generate the script files change
<code>eval = FALSE</code> to <code>eval = TRUE</code> in the chunk
options.</p>
<p>The scripts must be run in the terminal, inside the path
./output/crispr/mm_2215_sgRNA/mageck_rra_expr. The script will activate
the mageck environment and run the test for each comparison. The results
will be saved in the same directory.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a>chunk_se <span class="ot">&lt;-</span> se[<span class="fu">c</span>(keep_gene_guides, keep_ctrl_guides),]</span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a>out_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(params<span class="sc">$</span>output_dir, params<span class="sc">$</span>mageck_dir)</span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a><span class="fu">dir.create</span>(out_dir, <span class="at">recursive =</span> <span class="cn">TRUE</span>, <span class="at">showWarnings =</span> <span class="cn">FALSE</span>)</span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a></span>
<span id="cb14-5"><a href="#cb14-5" tabindex="-1"></a><span class="co"># Generate count matrix</span></span>
<span id="cb14-6"><a href="#cb14-6" tabindex="-1"></a>count_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(chunk_se, <span class="st">&#39;counts&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>(<span class="at">check.names =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb14-7"><a href="#cb14-7" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&#39;sgRNA&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb14-8"><a href="#cb14-8" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Gene =</span> <span class="fu">rowData</span>(chunk_se)<span class="sc">$</span>Gene) <span class="sc">%&gt;%</span> </span>
<span id="cb14-9"><a href="#cb14-9" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sgRNA, Gene, <span class="fu">everything</span>())</span>
<span id="cb14-10"><a href="#cb14-10" tabindex="-1"></a><span class="fu">write_tsv</span>(count_mat, <span class="at">file =</span> <span class="fu">file.path</span>(out_dir, <span class="st">&#39;counts.txt&#39;</span>))</span>
<span id="cb14-11"><a href="#cb14-11" tabindex="-1"></a></span>
<span id="cb14-12"><a href="#cb14-12" tabindex="-1"></a><span class="co"># Generate list of control sgRNA IDs</span></span>
<span id="cb14-13"><a href="#cb14-13" tabindex="-1"></a>non_target_list <span class="ot">&lt;-</span> <span class="fu">rownames</span>(chunk_se) <span class="sc">%&gt;%</span> <span class="fu">grep</span>(<span class="st">&#39;Non_Target&#39;</span>, ., <span class="at">value =</span> T) <span class="sc">%&gt;%</span> data.frame</span>
<span id="cb14-14"><a href="#cb14-14" tabindex="-1"></a><span class="fu">write_tsv</span>(non_target_list, <span class="at">file =</span> <span class="fu">file.path</span>(out_dir, <span class="st">&#39;control_sgrna.txt&#39;</span>), <span class="at">col_names =</span> F)</span>
<span id="cb14-15"><a href="#cb14-15" tabindex="-1"></a></span>
<span id="cb14-16"><a href="#cb14-16" tabindex="-1"></a><span class="co"># Generate mageck commands</span></span>
<span id="cb14-17"><a href="#cb14-17" tabindex="-1"></a>res_cmd <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="st">&#39;source activate mageckenv&#39;</span>)</span>
<span id="cb14-18"><a href="#cb14-18" tabindex="-1"></a>res_cmd <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i=</span><span class="fu">names</span>(mageck_comp_list), <span class="at">.combine =</span> rbind) <span class="sc">%do%</span> {</span>
<span id="cb14-19"><a href="#cb14-19" tabindex="-1"></a>  <span class="fu">paste</span>(<span class="st">&#39;mageck test -k counts.txt --control-sgrna control_sgrna.txt --norm-method control -t&#39;</span>, <span class="fu">paste</span>(mageck_comp_list[[i]][[<span class="dv">1</span>]], <span class="at">collapse =</span> <span class="st">&#39;,&#39;</span>),<span class="st">&#39;-c&#39;</span>,  <span class="fu">paste</span>(mageck_comp_list[[i]][[<span class="dv">2</span>]], <span class="at">collapse =</span> <span class="st">&#39;,&#39;</span>), <span class="st">&#39;-n&#39;</span>, <span class="fu">tolower</span>(<span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>, <span class="st">&quot;_&quot;</span>, i)), <span class="st">&#39;</span><span class="sc">\n</span><span class="st">&#39;</span>)</span>
<span id="cb14-20"><a href="#cb14-20" tabindex="-1"></a>} <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>()</span>
<span id="cb14-21"><a href="#cb14-21" tabindex="-1"></a></span>
<span id="cb14-22"><a href="#cb14-22" tabindex="-1"></a>res_cmd <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="st">&#39;conda activate mageckenv&#39;</span>, res_cmd, <span class="st">&#39;conda deactivate&#39;</span>)</span>
<span id="cb14-23"><a href="#cb14-23" tabindex="-1"></a><span class="fu">write_tsv</span>(res_cmd, <span class="fu">file.path</span>(out_dir, <span class="st">&#39;run_mageck_test.sh&#39;</span>), <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<div id="load-mageck-results" class="section level4">
<h4>Load MAGeCK results</h4>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a>gene_summ_files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> <span class="fu">file.path</span>(params<span class="sc">$</span>output_dir, params<span class="sc">$</span>mageck_dir), <span class="at">pattern =</span> <span class="st">&#39;gene_summary.txt&#39;</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a>analysis_prefix <span class="ot">&lt;-</span> <span class="fu">basename</span>(gene_summ_files) <span class="sc">%&gt;%</span> <span class="fu">gsub</span>(<span class="st">&quot;.gene_summary.txt&quot;</span>, <span class="st">&quot;&quot;</span>, .)</span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a>gene_summ <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i =</span> gene_summ_files) <span class="sc">%do%</span> <span class="fu">read.delim</span>(i, <span class="at">check.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a><span class="fu">names</span>(gene_summ) <span class="ot">&lt;-</span> analysis_prefix</span>
<span id="cb15-5"><a href="#cb15-5" tabindex="-1"></a></span>
<span id="cb15-6"><a href="#cb15-6" tabindex="-1"></a>sgrna_summ_files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> <span class="fu">file.path</span>(params<span class="sc">$</span>output_dir, params<span class="sc">$</span>mageck_dir), <span class="at">pattern =</span> <span class="st">&#39;sgrna_summary.txt&#39;</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-7"><a href="#cb15-7" tabindex="-1"></a>analysis_prefix <span class="ot">&lt;-</span> <span class="fu">basename</span>(sgrna_summ_files) <span class="sc">%&gt;%</span> <span class="fu">gsub</span>(<span class="st">&quot;.sgrna_summary.txt&quot;</span>, <span class="st">&quot;&quot;</span>, .)</span>
<span id="cb15-8"><a href="#cb15-8" tabindex="-1"></a>sgrna_summ <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i =</span> sgrna_summ_files) <span class="sc">%do%</span> <span class="fu">read.delim</span>(i, <span class="at">check.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb15-9"><a href="#cb15-9" tabindex="-1"></a><span class="fu">names</span>(sgrna_summ) <span class="ot">&lt;-</span> analysis_prefix</span></code></pre></div>
</div>
</div>
<div id="mageck-analysis-with-non-expressed-genes"
class="section level3">
<h3>MAGeCK analysis with non-expressed genes</h3>
<div id="list-of-comparisons-1" class="section level4">
<h4>List of comparisons</h4>
<p>List of comparisons for differential abundance using MAGeCK</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">colData</span>(se) <span class="sc">%&gt;%</span> data.frame</span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a></span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a>ne_mageck_comp_list <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb16-4"><a href="#cb16-4" tabindex="-1"></a>  </span>
<span id="cb16-5"><a href="#cb16-5" tabindex="-1"></a>  <span class="st">`</span><span class="at">StealTHY</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb16-6"><a href="#cb16-6" tabindex="-1"></a>    <span class="st">`</span><span class="at">Primary tumor</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(vector_mmodel <span class="sc">%in%</span> <span class="fu">list</span>(<span class="st">&#39;4T1-KO-Thy1-BALB&#39;</span>, <span class="st">&#39;MVT1-KO-Thy1-FVB&#39;</span>, <span class="st">&#39;Py2T-KO-Thy1-FVB&#39;</span>) <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Primary tumor&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias),</span>
<span id="cb16-7"><a href="#cb16-7" tabindex="-1"></a>    <span class="st">`</span><span class="at">Lung mets</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(vector_mmodel <span class="sc">%in%</span> <span class="fu">list</span>(<span class="st">&#39;4T1-KO-Thy1-BALB&#39;</span>, <span class="st">&#39;MVT1-KO-Thy1-FVB&#39;</span>, <span class="st">&#39;Py2T-KO-Thy1-FVB&#39;</span>) <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Lung mets&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias)</span>
<span id="cb16-8"><a href="#cb16-8" tabindex="-1"></a>    ),</span>
<span id="cb16-9"><a href="#cb16-9" tabindex="-1"></a>  </span>
<span id="cb16-10"><a href="#cb16-10" tabindex="-1"></a>  <span class="st">`</span><span class="at">Puro</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb16-11"><a href="#cb16-11" tabindex="-1"></a>    <span class="st">`</span><span class="at">Primary tumor</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(global_vector_mmodel <span class="sc">==</span> <span class="st">&#39;whole_tumor-KO-Puro-immunocompetent&#39;</span> ) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias),</span>
<span id="cb16-12"><a href="#cb16-12" tabindex="-1"></a>    <span class="st">`</span><span class="at">Lung mets</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(global_vector_mmodel <span class="sc">==</span> <span class="st">&#39;whole_lung-KO-Puro-immunocompetent&#39;</span> ) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias)</span>
<span id="cb16-13"><a href="#cb16-13" tabindex="-1"></a>  ),</span>
<span id="cb16-14"><a href="#cb16-14" tabindex="-1"></a>  </span>
<span id="cb16-15"><a href="#cb16-15" tabindex="-1"></a>  <span class="st">`</span><span class="at">Cas9+Puro</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb16-16"><a href="#cb16-16" tabindex="-1"></a>    <span class="st">`</span><span class="at">Primary tumor</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(vector_mmodel <span class="sc">==</span> <span class="st">&#39;4T1-KO-Puro_EF1alpha_SpCas9-BALB&#39;</span> <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Primary tumor&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias),</span>
<span id="cb16-17"><a href="#cb16-17" tabindex="-1"></a>    <span class="st">`</span><span class="at">Lung mets</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(vector_mmodel <span class="sc">==</span> <span class="st">&#39;4T1-KO-Puro_EF1alpha_SpCas9-BALB&#39;</span> <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Lung mets&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias)</span>
<span id="cb16-18"><a href="#cb16-18" tabindex="-1"></a>  ),</span>
<span id="cb16-19"><a href="#cb16-19" tabindex="-1"></a>  </span>
<span id="cb16-20"><a href="#cb16-20" tabindex="-1"></a>  <span class="st">`</span><span class="at">StealTHY NSG</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb16-21"><a href="#cb16-21" tabindex="-1"></a>    <span class="st">`</span><span class="at">Primary tumor</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(global_vector_mmodel <span class="sc">==</span> <span class="st">&#39;whole_tumor-KO-Thy1-immunodeficient&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias),</span>
<span id="cb16-22"><a href="#cb16-22" tabindex="-1"></a>    <span class="st">`</span><span class="at">Lung mets</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(global_vector_mmodel <span class="sc">==</span> <span class="st">&#39;whole_lung-KO-Thy1-immunodeficient&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias)</span>
<span id="cb16-23"><a href="#cb16-23" tabindex="-1"></a>  ),</span>
<span id="cb16-24"><a href="#cb16-24" tabindex="-1"></a>  </span>
<span id="cb16-25"><a href="#cb16-25" tabindex="-1"></a>  <span class="do">## 4T1</span></span>
<span id="cb16-26"><a href="#cb16-26" tabindex="-1"></a>  <span class="st">`</span><span class="at">StealTHY 4T1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb16-27"><a href="#cb16-27" tabindex="-1"></a>    <span class="st">`</span><span class="at">Primary tumor</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(vector_mmodel <span class="sc">==</span> <span class="st">&#39;4T1-KO-Thy1-BALB&#39;</span> <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Primary tumor&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias),</span>
<span id="cb16-28"><a href="#cb16-28" tabindex="-1"></a>    <span class="st">`</span><span class="at">Lung mets</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(vector_mmodel <span class="sc">==</span> <span class="st">&#39;4T1-KO-Thy1-BALB&#39;</span> <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Lung mets&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias)</span>
<span id="cb16-29"><a href="#cb16-29" tabindex="-1"></a>    ),</span>
<span id="cb16-30"><a href="#cb16-30" tabindex="-1"></a>  </span>
<span id="cb16-31"><a href="#cb16-31" tabindex="-1"></a>  <span class="st">`</span><span class="at">Puro 4T1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb16-32"><a href="#cb16-32" tabindex="-1"></a>    <span class="st">`</span><span class="at">Primary tumor</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(vector_mmodel <span class="sc">==</span> <span class="st">&#39;4T1-KO-Puro-BALB&#39;</span> <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Primary tumor&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias),</span>
<span id="cb16-33"><a href="#cb16-33" tabindex="-1"></a>    <span class="st">`</span><span class="at">In vitro</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(vector_mmodel <span class="sc">==</span> <span class="st">&#39;4T1-KO-lentiguide-Puro-In vitro&#39;</span> <span class="sc">&amp;</span> invitro_type <span class="sc">==</span> <span class="st">&#39;Puro-resistant&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias)</span>
<span id="cb16-34"><a href="#cb16-34" tabindex="-1"></a>  ),</span>
<span id="cb16-35"><a href="#cb16-35" tabindex="-1"></a>  </span>
<span id="cb16-36"><a href="#cb16-36" tabindex="-1"></a>  <span class="st">`</span><span class="at">Cas9+Puro 4T1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb16-37"><a href="#cb16-37" tabindex="-1"></a>    <span class="st">`</span><span class="at">Primary tumor</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(vector_mmodel <span class="sc">==</span> <span class="st">&#39;4T1-KO-Puro_EF1alpha_SpCas9-BALB&#39;</span> <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Primary tumor&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias),</span>
<span id="cb16-38"><a href="#cb16-38" tabindex="-1"></a>    <span class="st">`</span><span class="at">Lung mets</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(vector_mmodel <span class="sc">==</span> <span class="st">&#39;4T1-KO-Puro_EF1alpha_SpCas9-BALB&#39;</span> <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Lung mets&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias)</span>
<span id="cb16-39"><a href="#cb16-39" tabindex="-1"></a>  ),</span>
<span id="cb16-40"><a href="#cb16-40" tabindex="-1"></a>  </span>
<span id="cb16-41"><a href="#cb16-41" tabindex="-1"></a>  <span class="st">`</span><span class="at">StealTHY NSG 4T1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb16-42"><a href="#cb16-42" tabindex="-1"></a>    <span class="st">`</span><span class="at">Primary tumor</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(vector_mmodel <span class="sc">==</span> <span class="st">&#39;4T1-KO-Thy1-NSG&#39;</span> <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Primary tumor&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias),</span>
<span id="cb16-43"><a href="#cb16-43" tabindex="-1"></a>    <span class="st">`</span><span class="at">Lung mets</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(vector_mmodel <span class="sc">==</span> <span class="st">&#39;4T1-KO-Thy1-NSG&#39;</span> <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Lung mets&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias)</span>
<span id="cb16-44"><a href="#cb16-44" tabindex="-1"></a>    ),</span>
<span id="cb16-45"><a href="#cb16-45" tabindex="-1"></a>  </span>
<span id="cb16-46"><a href="#cb16-46" tabindex="-1"></a>  <span class="do">## MVT1</span></span>
<span id="cb16-47"><a href="#cb16-47" tabindex="-1"></a>  <span class="st">`</span><span class="at">StealTHY MVT1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb16-48"><a href="#cb16-48" tabindex="-1"></a>    <span class="st">`</span><span class="at">Primary tumor</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(vector_mmodel <span class="sc">==</span> <span class="st">&#39;MVT1-KO-Thy1-FVB&#39;</span> <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Primary tumor&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias),</span>
<span id="cb16-49"><a href="#cb16-49" tabindex="-1"></a>    <span class="st">`</span><span class="at">Lung mets</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(vector_mmodel <span class="sc">==</span> <span class="st">&#39;MVT1-KO-Thy1-FVB&#39;</span> <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Lung mets&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias)</span>
<span id="cb16-50"><a href="#cb16-50" tabindex="-1"></a>    ),</span>
<span id="cb16-51"><a href="#cb16-51" tabindex="-1"></a>  </span>
<span id="cb16-52"><a href="#cb16-52" tabindex="-1"></a>  <span class="st">`</span><span class="at">Puro MVT1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb16-53"><a href="#cb16-53" tabindex="-1"></a>    <span class="st">`</span><span class="at">Primary tumor</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(vector_mmodel <span class="sc">==</span> <span class="st">&#39;MVT1-KO-Puro-FVB&#39;</span> <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Primary tumor&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias),</span>
<span id="cb16-54"><a href="#cb16-54" tabindex="-1"></a>    <span class="st">`</span><span class="at">In vitro</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(vector_mmodel <span class="sc">==</span> <span class="st">&#39;MVT1-KO-lentiguide-Puro-In vitro&#39;</span> <span class="sc">&amp;</span> invitro_type <span class="sc">==</span> <span class="st">&#39;Puro-resistant&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias)</span>
<span id="cb16-55"><a href="#cb16-55" tabindex="-1"></a>  ),</span>
<span id="cb16-56"><a href="#cb16-56" tabindex="-1"></a>  </span>
<span id="cb16-57"><a href="#cb16-57" tabindex="-1"></a>  <span class="st">`</span><span class="at">StealTHY NSG MVT1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb16-58"><a href="#cb16-58" tabindex="-1"></a>    <span class="st">`</span><span class="at">Primary tumor</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(vector_mmodel <span class="sc">==</span> <span class="st">&#39;MVT1-KO-Thy1-NSG&#39;</span> <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Primary tumor&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias),</span>
<span id="cb16-59"><a href="#cb16-59" tabindex="-1"></a>    <span class="st">`</span><span class="at">Lung mets</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(vector_mmodel <span class="sc">==</span> <span class="st">&#39;MVT1-KO-Thy1-NSG&#39;</span> <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Lung mets&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias)</span>
<span id="cb16-60"><a href="#cb16-60" tabindex="-1"></a>    ),</span>
<span id="cb16-61"><a href="#cb16-61" tabindex="-1"></a>  </span>
<span id="cb16-62"><a href="#cb16-62" tabindex="-1"></a>  <span class="do">## Py2T</span></span>
<span id="cb16-63"><a href="#cb16-63" tabindex="-1"></a>  <span class="st">`</span><span class="at">StealTHY Py2T</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb16-64"><a href="#cb16-64" tabindex="-1"></a>    <span class="st">`</span><span class="at">Primary tumor</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(vector_mmodel <span class="sc">==</span> <span class="st">&#39;Py2T-KO-Thy1-FVB&#39;</span> <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Primary tumor&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias),</span>
<span id="cb16-65"><a href="#cb16-65" tabindex="-1"></a>    <span class="st">`</span><span class="at">Lung mets</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(vector_mmodel <span class="sc">==</span> <span class="st">&#39;Py2T-KO-Thy1-FVB&#39;</span> <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Lung mets&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias)</span>
<span id="cb16-66"><a href="#cb16-66" tabindex="-1"></a>    ),</span>
<span id="cb16-67"><a href="#cb16-67" tabindex="-1"></a>  </span>
<span id="cb16-68"><a href="#cb16-68" tabindex="-1"></a>  <span class="st">`</span><span class="at">Puro Py2T</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb16-69"><a href="#cb16-69" tabindex="-1"></a>    <span class="st">`</span><span class="at">Primary tumor</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(vector_mmodel <span class="sc">==</span> <span class="st">&#39;Py2T-KO-Puro-FVB&#39;</span> <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Primary tumor&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias),</span>
<span id="cb16-70"><a href="#cb16-70" tabindex="-1"></a>    <span class="st">`</span><span class="at">In vitro</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(vector_mmodel <span class="sc">==</span> <span class="st">&#39;Py2T-KO-lentiguide-Puro-In vitro&#39;</span> <span class="sc">&amp;</span> invitro_type <span class="sc">==</span> <span class="st">&#39;Puro-resistant&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias)</span>
<span id="cb16-71"><a href="#cb16-71" tabindex="-1"></a>  ),</span>
<span id="cb16-72"><a href="#cb16-72" tabindex="-1"></a>  </span>
<span id="cb16-73"><a href="#cb16-73" tabindex="-1"></a>  <span class="st">`</span><span class="at">StealTHY NSG Py2T</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb16-74"><a href="#cb16-74" tabindex="-1"></a>    <span class="st">`</span><span class="at">Primary tumor</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(vector_mmodel <span class="sc">==</span> <span class="st">&#39;Py2T-KO-Thy1-NSG&#39;</span> <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Primary tumor&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias),</span>
<span id="cb16-75"><a href="#cb16-75" tabindex="-1"></a>    <span class="st">`</span><span class="at">Lung mets</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(vector_mmodel <span class="sc">==</span> <span class="st">&#39;Py2T-KO-Thy1-NSG&#39;</span> <span class="sc">&amp;</span> sample_type <span class="sc">==</span> <span class="st">&#39;Lung mets&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample_alias)</span>
<span id="cb16-76"><a href="#cb16-76" tabindex="-1"></a>    )</span>
<span id="cb16-77"><a href="#cb16-77" tabindex="-1"></a>  </span>
<span id="cb16-78"><a href="#cb16-78" tabindex="-1"></a>  </span>
<span id="cb16-79"><a href="#cb16-79" tabindex="-1"></a>)</span>
<span id="cb16-80"><a href="#cb16-80" tabindex="-1"></a></span>
<span id="cb16-81"><a href="#cb16-81" tabindex="-1"></a></span>
<span id="cb16-82"><a href="#cb16-82" tabindex="-1"></a>ne_mageck_comp_list_names <span class="ot">&lt;-</span> <span class="fu">names</span>(ne_mageck_comp_list) </span>
<span id="cb16-83"><a href="#cb16-83" tabindex="-1"></a><span class="fu">names</span>(ne_mageck_comp_list_names) <span class="ot">&lt;-</span> <span class="fu">names</span>(ne_mageck_comp_list) <span class="sc">%&gt;%</span> <span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>, <span class="st">&quot;_&quot;</span>,.) <span class="sc">%&gt;%</span> tolower</span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a>data_comp <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i =</span> <span class="fu">names</span>(ne_mageck_comp_list), <span class="at">.combine =</span> rbind) <span class="sc">%do%</span> {</span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a>  <span class="fu">c</span>(</span>
<span id="cb17-3"><a href="#cb17-3" tabindex="-1"></a>    <span class="at">Comparison =</span> i,</span>
<span id="cb17-4"><a href="#cb17-4" tabindex="-1"></a>    <span class="at">Case =</span> <span class="fu">paste</span>(ne_mageck_comp_list[[i]][[<span class="dv">1</span>]], <span class="at">collapse =</span> <span class="st">&quot; &quot;</span>),</span>
<span id="cb17-5"><a href="#cb17-5" tabindex="-1"></a>    <span class="at">Control =</span> <span class="fu">paste</span>(ne_mageck_comp_list[[i]][[<span class="dv">2</span>]], <span class="at">collapse =</span> <span class="st">&quot; &quot;</span>)</span>
<span id="cb17-6"><a href="#cb17-6" tabindex="-1"></a>  )</span>
<span id="cb17-7"><a href="#cb17-7" tabindex="-1"></a>}</span>
<span id="cb17-8"><a href="#cb17-8" tabindex="-1"></a></span>
<span id="cb17-9"><a href="#cb17-9" tabindex="-1"></a>data_comp <span class="sc">%&gt;%</span> </span>
<span id="cb17-10"><a href="#cb17-10" tabindex="-1"></a>  <span class="fu">datatable</span>(., </span>
<span id="cb17-11"><a href="#cb17-11" tabindex="-1"></a>            <span class="at">rownames =</span> <span class="cn">FALSE</span>, </span>
<span id="cb17-12"><a href="#cb17-12" tabindex="-1"></a>            <span class="at">filter =</span> <span class="st">&#39;top&#39;</span>, </span>
<span id="cb17-13"><a href="#cb17-13" tabindex="-1"></a>            <span class="at">caption =</span> <span class="st">&#39;List of comparisons for MAGeCK analysis&#39;</span>,</span>
<span id="cb17-14"><a href="#cb17-14" tabindex="-1"></a>            <span class="at">extensions =</span> <span class="st">&#39;Buttons&#39;</span>, </span>
<span id="cb17-15"><a href="#cb17-15" tabindex="-1"></a>            <span class="at">options =</span> <span class="fu">list</span>(</span>
<span id="cb17-16"><a href="#cb17-16" tabindex="-1"></a>              <span class="at">dom =</span> <span class="st">&#39;Blfrtip&#39;</span>,</span>
<span id="cb17-17"><a href="#cb17-17" tabindex="-1"></a>              <span class="at">buttons =</span> <span class="fu">c</span>(<span class="st">&#39;csv&#39;</span>, <span class="st">&#39;excel&#39;</span>),</span>
<span id="cb17-18"><a href="#cb17-18" tabindex="-1"></a>              <span class="at">columnDefs =</span> <span class="fu">list</span>(</span>
<span id="cb17-19"><a href="#cb17-19" tabindex="-1"></a>                <span class="fu">list</span>(<span class="at">width =</span> <span class="dv">120</span>, <span class="at">targets =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>)</span>
<span id="cb17-20"><a href="#cb17-20" tabindex="-1"></a>              )</span>
<span id="cb17-21"><a href="#cb17-21" tabindex="-1"></a>            )</span>
<span id="cb17-22"><a href="#cb17-22" tabindex="-1"></a>            )</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-684d012f49e01d513276" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-684d012f49e01d513276">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption>List of comparisons for MAGeCK analysis<\/caption>","data":[["StealTHY","Puro","Cas9+Puro","StealTHY NSG","StealTHY 4T1","Puro 4T1","Cas9+Puro 4T1","StealTHY NSG 4T1","StealTHY MVT1","Puro MVT1","StealTHY NSG MVT1","StealTHY Py2T","Puro Py2T","StealTHY NSG Py2T"],["CRISPRKO_Clone_1 CRISPRKO_Clone_10 CRISPRKO_Clone_11 CRISPRKO_Clone_12 CRISPRKO_Clone_13 CRISPRKO_Clone_14 CRISPRKO_Clone_2 CRISPRKO_Clone_20 CRISPRKO_Clone_21 CRISPRKO_Clone_22 CRISPRKO_Clone_23 CRISPRKO_Clone_24 CRISPRKO_Clone_3 CRISPRKO_Clone_4 CRISPRKO","PlatePositionA7 PlatePositionA8 PlatePositionA9 PlatePositionB7 PlatePositionB8 PlatePositionB9 PlatePositionC7 PlatePositionC8 PlatePositionC9 PlatePositionD7 PlatePositionD9","PlatePositionA5 PlatePositionA6 PlatePositionC5 PlatePositionC6 PlatePositionE5 PlatePositionE6 PlatePositionG4 PlatePositionG5 PlatePositionG6","CRISPRKO_Clone_45 CRISPRKO_Clone_46 CRISPRKO_Clone_47 CRISPRKO_Clone_48 CRISPRKO_Clone_49 CRISPRKO_Clone_5 CRISPRKO_Clone_50 CRISPRKO_Clone_51 CRISPRKO_Clone_52 CRISPRKO_Clone_53 CRISPRKO_Clone_54 CRISPRKO_Clone_6 CRISPRKO_Clone_7 CRISPRKO_Clone_8 CRISPRKO_Clone_9","CRISPRKO_Clone_10 CRISPRKO_Clone_11 CRISPRKO_Clone_12 CRISPRKO_Clone_13 CRISPRKO_Clone_14","PlatePositionA7 PlatePositionB7 PlatePositionC7 PlatePositionD7","PlatePositionA5 PlatePositionA6 PlatePositionC5 PlatePositionC6 PlatePositionE5 PlatePositionE6 PlatePositionG4 PlatePositionG5 PlatePositionG6","CRISPRKO_Clone_45 CRISPRKO_Clone_46 CRISPRKO_Clone_47 CRISPRKO_Clone_48 CRISPRKO_Clone_49","CRISPRKO_Clone_20 CRISPRKO_Clone_21 CRISPRKO_Clone_22 CRISPRKO_Clone_23 CRISPRKO_Clone_24","PlatePositionA8 PlatePositionB8 PlatePositionC8","CRISPRKO_Clone_5 CRISPRKO_Clone_6 CRISPRKO_Clone_7 CRISPRKO_Clone_8 CRISPRKO_Clone_9","CRISPRKO_Clone_1 CRISPRKO_Clone_2 CRISPRKO_Clone_3 CRISPRKO_Clone_4 CRISPRKO","PlatePositionA9 PlatePositionB9 PlatePositionC9 PlatePositionD9","CRISPRKO_Clone_50 CRISPRKO_Clone_51 CRISPRKO_Clone_52 CRISPRKO_Clone_53 CRISPRKO_Clone_54"],["CRISPRKO_Clone_15 CRISPRKO_Clone_16 CRISPRKO_Clone_17 CRISPRKO_Clone_18 CRISPRKO_Clone_19 CRISPRKO_Clone_30 CRISPRKO_Clone_31 CRISPRKO_Clone_32 CRISPRKO_Clone_33 CRISPRKO_Clone_34 CRISPRKO_Clone_35 CRISPRKO_Clone_36 CRISPRKO_Clone_37 CRISPRKO_Clone_38 CRISPRKO_Clone_39","PlatePositionE7 PlatePositionE8 PlatePositionE9 PlatePositionF7 PlatePositionF8 PlatePositionF9 PlatePositionG7 PlatePositionG8 PlatePositionG9 PlatePositionH7 PlatePositionH9","PlatePositionB5 PlatePositionD5 PlatePositionD6 PlatePositionF3 PlatePositionF5 PlatePositionF6 PlatePositionH4 PlatePositionH5 PlatePositionH6","CRISPRKO_Clone_25 CRISPRKO_Clone_26 CRISPRKO_Clone_27 CRISPRKO_Clone_28 CRISPRKO_Clone_29 CRISPRKO_Clone_40 CRISPRKO_Clone_41 CRISPRKO_Clone_42 CRISPRKO_Clone_43 CRISPRKO_Clone_44 CRISPRKO_Clone_55 CRISPRKO_Clone_56 CRISPRKO_Clone_57 CRISPRKO_Clone_58 CRISPRKO_Clone_59","CRISPRKO_Clone_15 CRISPRKO_Clone_16 CRISPRKO_Clone_17 CRISPRKO_Clone_18 CRISPRKO_Clone_19","CRISPRKO_Clone_71","PlatePositionB5 PlatePositionD5 PlatePositionD6 PlatePositionF3 PlatePositionF5 PlatePositionF6 PlatePositionH4 PlatePositionH5 PlatePositionH6","CRISPRKO_Clone_40 CRISPRKO_Clone_41 CRISPRKO_Clone_42 CRISPRKO_Clone_43 CRISPRKO_Clone_44","CRISPRKO_Clone_35 CRISPRKO_Clone_36 CRISPRKO_Clone_37 CRISPRKO_Clone_38 CRISPRKO_Clone_39","CRISPRKO_Clone_70","CRISPRKO_Clone_55 CRISPRKO_Clone_56 CRISPRKO_Clone_57 CRISPRKO_Clone_58 CRISPRKO_Clone_59","CRISPRKO_Clone_30 CRISPRKO_Clone_31 CRISPRKO_Clone_32 CRISPRKO_Clone_33 CRISPRKO_Clone_34","CRISPRKO_Clone_63","CRISPRKO_Clone_25 CRISPRKO_Clone_26 CRISPRKO_Clone_27 CRISPRKO_Clone_28 CRISPRKO_Clone_29"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Comparison<\/th>\n      <th>Case<\/th>\n      <th>Control<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["csv","excel"],"columnDefs":[{"width":120,"targets":[1,2]},{"name":"Comparison","targets":0},{"name":"Case","targets":1},{"name":"Control","targets":2}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="run-mageck-1" class="section level4">
<h4>Run MAGeCK</h4>
<p>Generate MAGeCK test scripts. To generate the script files change
<code>eval = FALSE</code> to <code>eval = TRUE</code> in the chunk
options.</p>
<p>The scripts must be run in the terminal, inside the path
./output/crispr/mm_2215_sgRNA/mageck_rra. The script will activate the
mageck environment and run the test for each comparison. The results
will be saved in the same directory.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a>chunk_se <span class="ot">&lt;-</span> se</span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a>out_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(params<span class="sc">$</span>output_dir, params<span class="sc">$</span>ne_mageck_dir)</span>
<span id="cb18-3"><a href="#cb18-3" tabindex="-1"></a><span class="fu">dir.create</span>(out_dir, <span class="at">recursive =</span> <span class="cn">TRUE</span>, <span class="at">showWarnings =</span> <span class="cn">FALSE</span>)</span>
<span id="cb18-4"><a href="#cb18-4" tabindex="-1"></a></span>
<span id="cb18-5"><a href="#cb18-5" tabindex="-1"></a><span class="co"># Generate count matrix</span></span>
<span id="cb18-6"><a href="#cb18-6" tabindex="-1"></a>count_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(chunk_se, <span class="st">&#39;counts&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>(<span class="at">check.names =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb18-7"><a href="#cb18-7" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&#39;sgRNA&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb18-8"><a href="#cb18-8" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Gene =</span> <span class="fu">rowData</span>(chunk_se)<span class="sc">$</span>Gene) <span class="sc">%&gt;%</span> </span>
<span id="cb18-9"><a href="#cb18-9" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sgRNA, Gene, <span class="fu">everything</span>())</span>
<span id="cb18-10"><a href="#cb18-10" tabindex="-1"></a><span class="fu">write_tsv</span>(count_mat, <span class="at">file =</span> <span class="fu">file.path</span>(out_dir, <span class="st">&#39;counts.txt&#39;</span>))</span>
<span id="cb18-11"><a href="#cb18-11" tabindex="-1"></a></span>
<span id="cb18-12"><a href="#cb18-12" tabindex="-1"></a><span class="co"># Generate list of control sgRNA IDs</span></span>
<span id="cb18-13"><a href="#cb18-13" tabindex="-1"></a>non_target_list <span class="ot">&lt;-</span> <span class="fu">rownames</span>(chunk_se) <span class="sc">%&gt;%</span> <span class="fu">grep</span>(<span class="st">&#39;Non_Target&#39;</span>, ., <span class="at">value =</span> T) <span class="sc">%&gt;%</span> data.frame</span>
<span id="cb18-14"><a href="#cb18-14" tabindex="-1"></a><span class="fu">write_tsv</span>(non_target_list, <span class="at">file =</span> <span class="fu">file.path</span>(out_dir, <span class="st">&#39;control_sgrna.txt&#39;</span>), <span class="at">col_names =</span> F)</span>
<span id="cb18-15"><a href="#cb18-15" tabindex="-1"></a></span>
<span id="cb18-16"><a href="#cb18-16" tabindex="-1"></a><span class="co"># Generate mageck commands</span></span>
<span id="cb18-17"><a href="#cb18-17" tabindex="-1"></a>res_cmd <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="st">&#39;source activate mageckenv&#39;</span>)</span>
<span id="cb18-18"><a href="#cb18-18" tabindex="-1"></a>res_cmd <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i=</span><span class="fu">names</span>(ne_mageck_comp_list), <span class="at">.combine =</span> rbind) <span class="sc">%do%</span> {</span>
<span id="cb18-19"><a href="#cb18-19" tabindex="-1"></a>  <span class="fu">paste</span>(<span class="st">&#39;mageck test -k counts.txt --control-sgrna control_sgrna.txt --norm-method control -t&#39;</span>, <span class="fu">paste</span>(ne_mageck_comp_list[[i]][[<span class="dv">1</span>]], <span class="at">collapse =</span> <span class="st">&#39;,&#39;</span>),<span class="st">&#39;-c&#39;</span>,  <span class="fu">paste</span>(ne_mageck_comp_list[[i]][[<span class="dv">2</span>]], <span class="at">collapse =</span> <span class="st">&#39;,&#39;</span>), <span class="st">&#39;-n&#39;</span>, <span class="fu">tolower</span>(<span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>, <span class="st">&quot;_&quot;</span>, i)), <span class="st">&#39;</span><span class="sc">\n</span><span class="st">&#39;</span>)</span>
<span id="cb18-20"><a href="#cb18-20" tabindex="-1"></a>} <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>()</span>
<span id="cb18-21"><a href="#cb18-21" tabindex="-1"></a></span>
<span id="cb18-22"><a href="#cb18-22" tabindex="-1"></a>res_cmd <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="st">&#39;conda activate mageckenv&#39;</span>, res_cmd, <span class="st">&#39;conda deactivate&#39;</span>)</span>
<span id="cb18-23"><a href="#cb18-23" tabindex="-1"></a><span class="fu">write_tsv</span>(res_cmd, <span class="fu">file.path</span>(out_dir, <span class="st">&#39;run_mageck_test.sh&#39;</span>), <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<div id="load-mageck-results-1" class="section level4">
<h4>Load MAGeCK results</h4>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a>gene_summ_files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> <span class="fu">file.path</span>(params<span class="sc">$</span>output_dir, params<span class="sc">$</span>ne_mageck_dir), <span class="at">pattern =</span> <span class="st">&#39;gene_summary.txt&#39;</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a>analysis_prefix <span class="ot">&lt;-</span> <span class="fu">basename</span>(gene_summ_files) <span class="sc">%&gt;%</span> <span class="fu">gsub</span>(<span class="st">&quot;.gene_summary.txt&quot;</span>, <span class="st">&quot;&quot;</span>, .)</span>
<span id="cb19-3"><a href="#cb19-3" tabindex="-1"></a>ne_gene_summ <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i =</span> gene_summ_files) <span class="sc">%do%</span> <span class="fu">read.delim</span>(i, <span class="at">check.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb19-4"><a href="#cb19-4" tabindex="-1"></a><span class="fu">names</span>(ne_gene_summ) <span class="ot">&lt;-</span> analysis_prefix</span>
<span id="cb19-5"><a href="#cb19-5" tabindex="-1"></a></span>
<span id="cb19-6"><a href="#cb19-6" tabindex="-1"></a>sgrna_summ_files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> <span class="fu">file.path</span>(params<span class="sc">$</span>output_dir, params<span class="sc">$</span>ne_mageck_dir), <span class="at">pattern =</span> <span class="st">&#39;sgrna_summary.txt&#39;</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb19-7"><a href="#cb19-7" tabindex="-1"></a>analysis_prefix <span class="ot">&lt;-</span> <span class="fu">basename</span>(sgrna_summ_files) <span class="sc">%&gt;%</span> <span class="fu">gsub</span>(<span class="st">&quot;.sgrna_summary.txt&quot;</span>, <span class="st">&quot;&quot;</span>, .)</span>
<span id="cb19-8"><a href="#cb19-8" tabindex="-1"></a>ne_sgrna_summ <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i =</span> sgrna_summ_files) <span class="sc">%do%</span> <span class="fu">read.delim</span>(i, <span class="at">check.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb19-9"><a href="#cb19-9" tabindex="-1"></a><span class="fu">names</span>(ne_sgrna_summ) <span class="ot">&lt;-</span> analysis_prefix</span></code></pre></div>
</div>
</div>
</div>
<div id="sgrna-distribution-1" class="section level2">
<h2>sgRNA distribution</h2>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a>selected_panels <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a>  <span class="at">MVT1 =</span> <span class="fu">c</span>(</span>
<span id="cb20-3"><a href="#cb20-3" tabindex="-1"></a>    <span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">&#39;CRISPRKO_Clone_6&#39;</span>, <span class="st">&#39;CRISPRKO_Clone_5&#39;</span>, <span class="st">&#39;CRISPRKO_Clone_7&#39;</span>, <span class="st">&#39;CRISPRKO_Clone_8&#39;</span>, <span class="st">&#39;CRISPRKO_Clone_9&#39;</span>)),</span>
<span id="cb20-4"><a href="#cb20-4" tabindex="-1"></a>    <span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">&#39;CRISPRKO_Clone_23&#39;</span>, <span class="st">&#39;CRISPRKO_Clone_20&#39;</span>, <span class="st">&#39;CRISPRKO_Clone_21&#39;</span>, <span class="st">&#39;CRISPRKO_Clone_24&#39;</span>, <span class="st">&#39;CRISPRKO_Clone_22&#39;</span>))</span>
<span id="cb20-5"><a href="#cb20-5" tabindex="-1"></a>    ),</span>
<span id="cb20-6"><a href="#cb20-6" tabindex="-1"></a>  <span class="st">`</span><span class="at">4T1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb20-7"><a href="#cb20-7" tabindex="-1"></a>    <span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">&#39;CRISPRKO_Clone_46&#39;</span>, <span class="st">&#39;CRISPRKO_Clone_47&#39;</span>, <span class="st">&#39;CRISPRKO_Clone_49&#39;</span>, <span class="st">&#39;CRISPRKO_Clone_45&#39;</span>, <span class="st">&#39;CRISPRKO_Clone_48&#39;</span>)),</span>
<span id="cb20-8"><a href="#cb20-8" tabindex="-1"></a>    <span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">&#39;CRISPRKO_Clone_12&#39;</span>, <span class="st">&#39;CRISPRKO_Clone_10&#39;</span>, <span class="st">&#39;CRISPRKO_Clone_11&#39;</span>, <span class="st">&#39;CRISPRKO_Clone_14&#39;</span>, <span class="st">&#39;CRISPRKO_Clone_13&#39;</span>))</span>
<span id="cb20-9"><a href="#cb20-9" tabindex="-1"></a>    ),</span>
<span id="cb20-10"><a href="#cb20-10" tabindex="-1"></a>  <span class="st">`</span><span class="at">Py2T</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb20-11"><a href="#cb20-11" tabindex="-1"></a>    <span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">&#39;CRISPRKO_Clone_50&#39;</span>, <span class="st">&#39;CRISPRKO_Clone_51&#39;</span>, <span class="st">&#39;CRISPRKO_Clone_52&#39;</span>, <span class="st">&#39;CRISPRKO_Clone_53&#39;</span>, <span class="st">&#39;CRISPRKO_Clone_54&#39;</span>)),</span>
<span id="cb20-12"><a href="#cb20-12" tabindex="-1"></a>    <span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">&#39;CRISPRKO&#39;</span>, <span class="st">&#39;CRISPRKO_Clone_1&#39;</span>, <span class="st">&#39;CRISPRKO_Clone_2&#39;</span>, <span class="st">&#39;CRISPRKO_Clone_3&#39;</span>, <span class="st">&#39;CRISPRKO_Clone_4&#39;</span>))</span>
<span id="cb20-13"><a href="#cb20-13" tabindex="-1"></a>    )</span>
<span id="cb20-14"><a href="#cb20-14" tabindex="-1"></a>)</span>
<span id="cb20-15"><a href="#cb20-15" tabindex="-1"></a></span>
<span id="cb20-16"><a href="#cb20-16" tabindex="-1"></a>selected_panels_df <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i =</span> <span class="fu">names</span>(selected_panels), <span class="at">.combine =</span> rbind) <span class="sc">%do%</span> {</span>
<span id="cb20-17"><a href="#cb20-17" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">group =</span> i, <span class="at">sample_alias =</span> selected_panels[[i]])</span>
<span id="cb20-18"><a href="#cb20-18" tabindex="-1"></a>}</span>
<span id="cb20-19"><a href="#cb20-19" tabindex="-1"></a></span>
<span id="cb20-20"><a href="#cb20-20" tabindex="-1"></a>use_samples <span class="ot">&lt;-</span> <span class="fu">unlist</span>(selected_panels)</span>
<span id="cb20-21"><a href="#cb20-21" tabindex="-1"></a>use_assay <span class="ot">&lt;-</span> <span class="fu">assay</span>(se[,use_samples], <span class="st">&#39;cpm&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb20-22"><a href="#cb20-22" tabindex="-1"></a>  <span class="fu">as.data.frame</span>(<span class="at">check.names =</span> F) <span class="sc">%&gt;%</span> </span>
<span id="cb20-23"><a href="#cb20-23" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&#39;guide&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb20-24"><a href="#cb20-24" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>guide, <span class="at">names_to =</span> <span class="st">&#39;sample_alias&#39;</span>, <span class="at">values_to =</span> <span class="st">&#39;cpm&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb20-25"><a href="#cb20-25" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">colData</span>(se)<span class="sc">%&gt;%</span> data.frame) <span class="sc">%&gt;%</span> </span>
<span id="cb20-26"><a href="#cb20-26" tabindex="-1"></a>  <span class="fu">left_join</span>(selected_panels_df) <span class="sc">%&gt;%</span> </span>
<span id="cb20-27"><a href="#cb20-27" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb20-28"><a href="#cb20-28" tabindex="-1"></a>    <span class="at">sample_alias =</span> <span class="fu">factor</span>(sample_alias, <span class="at">levels =</span> use_samples),</span>
<span id="cb20-29"><a href="#cb20-29" tabindex="-1"></a>    <span class="at">group =</span> <span class="fu">factor</span>(group, <span class="at">levels =</span> <span class="fu">names</span>(selected_panels))</span>
<span id="cb20-30"><a href="#cb20-30" tabindex="-1"></a>  )</span></code></pre></div>
<div
id="figure-4b-sgrna-distribution-in-mice-transplanted-with-4t1-cells-previously-subjected-to-stealthy-ko"
class="section level3">
<h3>Figure 4B: sgRNA distribution in mice transplanted with 4T1 cells
previously subjected to StealTHY KO</h3>
<p>Density plots showing unique sgRNAs from 4T1 orthotopic tumors
(n=5-7); frequency (cpm) = read counts per million.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a>use_samples <span class="ot">&lt;-</span> selected_panels_df <span class="sc">%&gt;%</span> </span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a>  <span class="fu">filter</span>(group <span class="sc">==</span> <span class="st">&#39;4T1&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb21-3"><a href="#cb21-3" tabindex="-1"></a>  <span class="fu">pull</span>(sample_alias)</span>
<span id="cb21-4"><a href="#cb21-4" tabindex="-1"></a></span>
<span id="cb21-5"><a href="#cb21-5" tabindex="-1"></a>use_assay <span class="ot">&lt;-</span> <span class="fu">assay</span>(se[,use_samples], <span class="st">&#39;cpm&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb21-6"><a href="#cb21-6" tabindex="-1"></a>  <span class="fu">as.data.frame</span>(<span class="at">check.names =</span> F) <span class="sc">%&gt;%</span> </span>
<span id="cb21-7"><a href="#cb21-7" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&#39;guide&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb21-8"><a href="#cb21-8" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>guide, <span class="at">names_to =</span> <span class="st">&#39;sample_alias&#39;</span>, <span class="at">values_to =</span> <span class="st">&#39;cpm&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb21-9"><a href="#cb21-9" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">colData</span>(se)<span class="sc">%&gt;%</span> data.frame) <span class="sc">%&gt;%</span> </span>
<span id="cb21-10"><a href="#cb21-10" tabindex="-1"></a>  <span class="fu">left_join</span>(selected_panels_df) <span class="sc">%&gt;%</span> </span>
<span id="cb21-11"><a href="#cb21-11" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb21-12"><a href="#cb21-12" tabindex="-1"></a>    <span class="at">sample_alias =</span> <span class="fu">factor</span>(sample_alias, <span class="at">levels =</span> use_samples),</span>
<span id="cb21-13"><a href="#cb21-13" tabindex="-1"></a>    <span class="at">group =</span> <span class="fu">factor</span>(group, <span class="at">levels =</span> <span class="fu">names</span>(selected_panels))</span>
<span id="cb21-14"><a href="#cb21-14" tabindex="-1"></a>  )</span>
<span id="cb21-15"><a href="#cb21-15" tabindex="-1"></a></span>
<span id="cb21-16"><a href="#cb21-16" tabindex="-1"></a></span>
<span id="cb21-17"><a href="#cb21-17" tabindex="-1"></a>use_assay <span class="sc">%&gt;%</span> </span>
<span id="cb21-18"><a href="#cb21-18" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>cpm <span class="sc">+</span> <span class="dv">1</span>, <span class="at">y =</span> sample_alias, </span>
<span id="cb21-19"><a href="#cb21-19" tabindex="-1"></a>             <span class="at">fill =</span> mouse_model_short, </span>
<span id="cb21-20"><a href="#cb21-20" tabindex="-1"></a>             <span class="at">color =</span> mouse_model_short, </span>
<span id="cb21-21"><a href="#cb21-21" tabindex="-1"></a>             <span class="at">height =</span> <span class="fu">after_stat</span>(density))) <span class="sc">+</span></span>
<span id="cb21-22"><a href="#cb21-22" tabindex="-1"></a>  <span class="fu">geom_density_ridges</span>(<span class="at">linewidth =</span> default_linewidth, </span>
<span id="cb21-23"><a href="#cb21-23" tabindex="-1"></a>                      <span class="at">alpha =</span> <span class="fl">0.8</span>, </span>
<span id="cb21-24"><a href="#cb21-24" tabindex="-1"></a>                      <span class="at">scale =</span> <span class="dv">4</span>,</span>
<span id="cb21-25"><a href="#cb21-25" tabindex="-1"></a>                      <span class="at">rel_min_height =</span> <span class="fl">0.001</span> <span class="co"># set the `rel_min_height` argument to remove tails,</span></span>
<span id="cb21-26"><a href="#cb21-26" tabindex="-1"></a>                      <span class="co"># stat = &quot;density&quot;</span></span>
<span id="cb21-27"><a href="#cb21-27" tabindex="-1"></a>                      ) <span class="sc">+</span></span>
<span id="cb21-28"><a href="#cb21-28" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> palette_mmodel) <span class="sc">+</span></span>
<span id="cb21-29"><a href="#cb21-29" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> palette_mmodel_line) <span class="sc">+</span></span>
<span id="cb21-30"><a href="#cb21-30" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)),</span>
<span id="cb21-31"><a href="#cb21-31" tabindex="-1"></a>                <span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1000</span>, <span class="dv">100000</span>)</span>
<span id="cb21-32"><a href="#cb21-32" tabindex="-1"></a>                ) <span class="sc">+</span></span>
<span id="cb21-33"><a href="#cb21-33" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="fl">0.01</span>, <span class="fl">0.2</span>))) <span class="sc">+</span></span>
<span id="cb21-34"><a href="#cb21-34" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&#39;&#39;</span>) <span class="sc">+</span></span>
<span id="cb21-35"><a href="#cb21-35" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill =</span> <span class="st">&#39;none&#39;</span>, <span class="at">color =</span> <span class="st">&#39;none&#39;</span>) <span class="sc">+</span></span>
<span id="cb21-36"><a href="#cb21-36" tabindex="-1"></a>  <span class="fu">theme_ridges</span>(<span class="at">font_size =</span> <span class="dv">8</span>, <span class="at">grid =</span> F) <span class="sc">+</span></span>
<span id="cb21-37"><a href="#cb21-37" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb21-38"><a href="#cb21-38" tabindex="-1"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb21-39"><a href="#cb21-39" tabindex="-1"></a>    <span class="at">axis.line.x =</span> <span class="fu">element_line</span>(<span class="at">linewidth =</span> default_linewidth, <span class="at">color =</span> <span class="st">&#39;black&#39;</span>),</span>
<span id="cb21-40"><a href="#cb21-40" tabindex="-1"></a>    <span class="at">axis.ticks.x =</span> <span class="fu">element_line</span>(<span class="at">linewidth =</span> default_linewidth, <span class="at">color =</span> <span class="st">&#39;black&#39;</span>),</span>
<span id="cb21-41"><a href="#cb21-41" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb21-42"><a href="#cb21-42" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">4</span>),</span>
<span id="cb21-43"><a href="#cb21-43" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">4</span>),</span>
<span id="cb21-44"><a href="#cb21-44" tabindex="-1"></a>    <span class="at">strip.text.x =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">4</span>, <span class="at">hjust =</span> <span class="dv">0</span>)</span>
<span id="cb21-45"><a href="#cb21-45" tabindex="-1"></a>  )</span></code></pre></div>
<p><img src="figure/crispr-mm_2215_sgRNA-StealTHY.Rmd/test-dist-stealthy-ko-ridges_plot-4t1-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-test-dist-stealthy-ko-ridges_plot-4t1-1">
Past versions of test-dist-stealthy-ko-ridges_plot-4t1-1.png
</button>
</p>
<div id="fig-test-dist-stealthy-ko-ridges_plot-4t1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/32aca023d6d8209f1aa3df3f177fc383c7569844/docs/figure/crispr-mm_2215_sgRNA-StealTHY.Rmd/test-dist-stealthy-ko-ridges_plot-4t1-1.png" target="_blank">32aca02</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-07-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="supplementary-figure-4b-sgrna-distribution-in-mice-transplanted-with-4t1-cells-previously-subjected-to-stealthy-ko"
class="section level3">
<h3>Supplementary figure 4B: sgRNA distribution in mice transplanted
with 4T1 cells previously subjected to StealTHY KO</h3>
<p>Density plots showing the distribution of unique sgRNAs retrieved in
primary tumors from the indicated mice transplanted with MVT1 cells
(left) or Py2T cells (right) previously subjected to Stealthy KO;
relative frequency is shown as read counts per million (cpm+1); n=5-7
each; representative tumors are shown.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a>use_samples <span class="ot">&lt;-</span> selected_panels_df <span class="sc">%&gt;%</span> </span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a>  <span class="fu">filter</span>(group <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&#39;MVT1&#39;</span>, <span class="st">&#39;Py2T&#39;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb22-3"><a href="#cb22-3" tabindex="-1"></a>  <span class="fu">pull</span>(sample_alias)</span>
<span id="cb22-4"><a href="#cb22-4" tabindex="-1"></a></span>
<span id="cb22-5"><a href="#cb22-5" tabindex="-1"></a>use_assay <span class="ot">&lt;-</span> <span class="fu">assay</span>(se[,use_samples], <span class="st">&#39;cpm&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb22-6"><a href="#cb22-6" tabindex="-1"></a>  <span class="fu">as.data.frame</span>(<span class="at">check.names =</span> F) <span class="sc">%&gt;%</span> </span>
<span id="cb22-7"><a href="#cb22-7" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&#39;guide&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb22-8"><a href="#cb22-8" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>guide, <span class="at">names_to =</span> <span class="st">&#39;sample_alias&#39;</span>, <span class="at">values_to =</span> <span class="st">&#39;cpm&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb22-9"><a href="#cb22-9" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">colData</span>(se)<span class="sc">%&gt;%</span> data.frame) <span class="sc">%&gt;%</span> </span>
<span id="cb22-10"><a href="#cb22-10" tabindex="-1"></a>  <span class="fu">left_join</span>(selected_panels_df) <span class="sc">%&gt;%</span> </span>
<span id="cb22-11"><a href="#cb22-11" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb22-12"><a href="#cb22-12" tabindex="-1"></a>    <span class="at">sample_alias =</span> <span class="fu">factor</span>(sample_alias, <span class="at">levels =</span> use_samples),</span>
<span id="cb22-13"><a href="#cb22-13" tabindex="-1"></a>    <span class="at">group =</span> <span class="fu">factor</span>(group, <span class="at">levels =</span> <span class="fu">names</span>(selected_panels))</span>
<span id="cb22-14"><a href="#cb22-14" tabindex="-1"></a>  )</span>
<span id="cb22-15"><a href="#cb22-15" tabindex="-1"></a></span>
<span id="cb22-16"><a href="#cb22-16" tabindex="-1"></a></span>
<span id="cb22-17"><a href="#cb22-17" tabindex="-1"></a>use_assay <span class="sc">%&gt;%</span> </span>
<span id="cb22-18"><a href="#cb22-18" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>cpm <span class="sc">+</span> <span class="dv">1</span>, <span class="at">y =</span> sample_alias, </span>
<span id="cb22-19"><a href="#cb22-19" tabindex="-1"></a>             <span class="at">fill =</span> mouse_model_short, </span>
<span id="cb22-20"><a href="#cb22-20" tabindex="-1"></a>             <span class="at">color =</span> mouse_model_short, </span>
<span id="cb22-21"><a href="#cb22-21" tabindex="-1"></a>             <span class="at">height =</span> <span class="fu">after_stat</span>(density))) <span class="sc">+</span></span>
<span id="cb22-22"><a href="#cb22-22" tabindex="-1"></a>  <span class="fu">geom_density_ridges</span>(<span class="at">linewidth =</span> default_linewidth, </span>
<span id="cb22-23"><a href="#cb22-23" tabindex="-1"></a>                      <span class="at">alpha =</span> <span class="fl">0.8</span>, </span>
<span id="cb22-24"><a href="#cb22-24" tabindex="-1"></a>                      <span class="at">scale =</span> <span class="dv">4</span>,</span>
<span id="cb22-25"><a href="#cb22-25" tabindex="-1"></a>                      <span class="at">rel_min_height =</span> <span class="fl">0.001</span> <span class="co"># set the `rel_min_height` argument to remove tails,</span></span>
<span id="cb22-26"><a href="#cb22-26" tabindex="-1"></a>                      ) <span class="sc">+</span></span>
<span id="cb22-27"><a href="#cb22-27" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> palette_mmodel) <span class="sc">+</span></span>
<span id="cb22-28"><a href="#cb22-28" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> palette_mmodel_line) <span class="sc">+</span></span>
<span id="cb22-29"><a href="#cb22-29" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)),</span>
<span id="cb22-30"><a href="#cb22-30" tabindex="-1"></a>                <span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1000</span>, <span class="dv">100000</span>)</span>
<span id="cb22-31"><a href="#cb22-31" tabindex="-1"></a>                ) <span class="sc">+</span></span>
<span id="cb22-32"><a href="#cb22-32" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="fl">0.01</span>, <span class="fl">0.2</span>))) <span class="sc">+</span></span>
<span id="cb22-33"><a href="#cb22-33" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&#39;&#39;</span>) <span class="sc">+</span></span>
<span id="cb22-34"><a href="#cb22-34" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill =</span> <span class="st">&#39;none&#39;</span>, <span class="at">color =</span> <span class="st">&#39;none&#39;</span>) <span class="sc">+</span></span>
<span id="cb22-35"><a href="#cb22-35" tabindex="-1"></a>  ggh4x<span class="sc">::</span><span class="fu">facet_wrap2</span>(<span class="fu">vars</span>(group), <span class="at">nrow =</span> <span class="dv">1</span>,</span>
<span id="cb22-36"><a href="#cb22-36" tabindex="-1"></a>                     <span class="at">scales =</span> <span class="st">&#39;free_y&#39;</span>, </span>
<span id="cb22-37"><a href="#cb22-37" tabindex="-1"></a>                     <span class="at">axes =</span> <span class="st">&quot;x&quot;</span>) <span class="sc">+</span></span>
<span id="cb22-38"><a href="#cb22-38" tabindex="-1"></a>  <span class="fu">theme_ridges</span>(<span class="at">font_size =</span> <span class="dv">8</span>, <span class="at">grid =</span> F) <span class="sc">+</span></span>
<span id="cb22-39"><a href="#cb22-39" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb22-40"><a href="#cb22-40" tabindex="-1"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb22-41"><a href="#cb22-41" tabindex="-1"></a>    <span class="at">axis.line.x =</span> <span class="fu">element_line</span>(<span class="at">linewidth =</span> default_linewidth, <span class="at">color =</span> <span class="st">&#39;black&#39;</span>),</span>
<span id="cb22-42"><a href="#cb22-42" tabindex="-1"></a>    <span class="at">axis.ticks.x =</span> <span class="fu">element_line</span>(<span class="at">linewidth =</span> default_linewidth, <span class="at">color =</span> <span class="st">&#39;black&#39;</span>),</span>
<span id="cb22-43"><a href="#cb22-43" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb22-44"><a href="#cb22-44" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">4</span>),</span>
<span id="cb22-45"><a href="#cb22-45" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">4</span>),</span>
<span id="cb22-46"><a href="#cb22-46" tabindex="-1"></a>    <span class="at">strip.text.x =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">4</span>, <span class="at">hjust =</span> <span class="dv">0</span>)</span>
<span id="cb22-47"><a href="#cb22-47" tabindex="-1"></a>  )</span></code></pre></div>
<p><img src="figure/crispr-mm_2215_sgRNA-StealTHY.Rmd/test-dist-stealthy-ko-ridges_plot-mvt1-py2t-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-test-dist-stealthy-ko-ridges_plot-mvt1-py2t-1">
Past versions of test-dist-stealthy-ko-ridges_plot-mvt1-py2t-1.png
</button>
</p>
<div id="fig-test-dist-stealthy-ko-ridges_plot-mvt1-py2t-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/32aca023d6d8209f1aa3df3f177fc383c7569844/docs/figure/crispr-mm_2215_sgRNA-StealTHY.Rmd/test-dist-stealthy-ko-ridges_plot-mvt1-py2t-1.png" target="_blank">32aca02</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-07-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="kolmogorov-smirnov-test-and-ecdf-plots" class="section level2">
<h2>Kolmogorov-Smirnov test and ECDF plots</h2>
<p>Run Kolmogorov-Smirnov test to compare distributions merged by group
and create Empirical Cumulative Density Function (ECDF) by
comparison.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a>i <span class="ot">&lt;-</span> <span class="fu">names</span>(comp_list_modality)[<span class="dv">1</span>]</span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a>ks_test_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb23-3"><a href="#cb23-3" tabindex="-1"></a>ecdf_plots <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb23-4"><a href="#cb23-4" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">names</span>(comp_list_modality)) {</span>
<span id="cb23-5"><a href="#cb23-5" tabindex="-1"></a>  use_samples <span class="ot">&lt;-</span> comp_list_modality[[i]]</span>
<span id="cb23-6"><a href="#cb23-6" tabindex="-1"></a>  use_assay_c <span class="ot">&lt;-</span> <span class="fu">assay</span>(se[,use_samples[[<span class="dv">1</span>]]], <span class="st">&#39;cpm&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-7"><a href="#cb23-7" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">check.names =</span> F) <span class="sc">%&gt;%</span> </span>
<span id="cb23-8"><a href="#cb23-8" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>(<span class="st">&#39;guide&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-9"><a href="#cb23-9" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="sc">-</span>guide, <span class="at">values_to =</span> <span class="st">&#39;cpm&#39;</span>,<span class="at">names_to =</span> <span class="st">&#39;sample_alias&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-10"><a href="#cb23-10" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">group =</span> <span class="st">&#39;case&#39;</span>, <span class="at">condition =</span> <span class="fu">names</span>(use_samples)[<span class="dv">1</span>])</span>
<span id="cb23-11"><a href="#cb23-11" tabindex="-1"></a>  use_assay_r <span class="ot">&lt;-</span> <span class="fu">assay</span>(se[,use_samples[[<span class="dv">2</span>]]], <span class="st">&#39;cpm&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-12"><a href="#cb23-12" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">check.names =</span> F) <span class="sc">%&gt;%</span> </span>
<span id="cb23-13"><a href="#cb23-13" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>(<span class="st">&#39;guide&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-14"><a href="#cb23-14" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="sc">-</span>guide, <span class="at">values_to =</span> <span class="st">&#39;cpm&#39;</span>,<span class="at">names_to =</span> <span class="st">&#39;sample_alias&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-15"><a href="#cb23-15" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">group =</span> <span class="st">&#39;referene&#39;</span>, <span class="at">condition =</span> <span class="fu">names</span>(use_samples)[<span class="dv">2</span>])</span>
<span id="cb23-16"><a href="#cb23-16" tabindex="-1"></a>  use_assay <span class="ot">&lt;-</span> <span class="fu">rbind</span>(use_assay_c, use_assay_r)</span>
<span id="cb23-17"><a href="#cb23-17" tabindex="-1"></a>  </span>
<span id="cb23-18"><a href="#cb23-18" tabindex="-1"></a>  <span class="co"># Perform the Kolmogorov-Smirnov test</span></span>
<span id="cb23-19"><a href="#cb23-19" tabindex="-1"></a>  ks.res <span class="ot">&lt;-</span> <span class="fu">ks.test</span>(use_assay_c<span class="sc">$</span>cpm, use_assay_r<span class="sc">$</span>cpm)</span>
<span id="cb23-20"><a href="#cb23-20" tabindex="-1"></a>  ks_test_df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(</span>
<span id="cb23-21"><a href="#cb23-21" tabindex="-1"></a>    ks_test_df,</span>
<span id="cb23-22"><a href="#cb23-22" tabindex="-1"></a>      <span class="fu">data.frame</span>(</span>
<span id="cb23-23"><a href="#cb23-23" tabindex="-1"></a>        <span class="at">comparison =</span> i,</span>
<span id="cb23-24"><a href="#cb23-24" tabindex="-1"></a>        <span class="at">case =</span> <span class="fu">names</span>(use_samples)[<span class="dv">1</span>],</span>
<span id="cb23-25"><a href="#cb23-25" tabindex="-1"></a>        <span class="at">reference =</span> <span class="fu">names</span>(use_samples)[<span class="dv">2</span>],</span>
<span id="cb23-26"><a href="#cb23-26" tabindex="-1"></a>        <span class="at">D =</span> ks.res<span class="sc">$</span>statistic,</span>
<span id="cb23-27"><a href="#cb23-27" tabindex="-1"></a>        <span class="at">P =</span> ks.res<span class="sc">$</span>p.value</span>
<span id="cb23-28"><a href="#cb23-28" tabindex="-1"></a>      )</span>
<span id="cb23-29"><a href="#cb23-29" tabindex="-1"></a>    )</span>
<span id="cb23-30"><a href="#cb23-30" tabindex="-1"></a></span>
<span id="cb23-31"><a href="#cb23-31" tabindex="-1"></a>  <span class="co"># Create ECDFs plot Empirical Cumulative Density Function</span></span>
<span id="cb23-32"><a href="#cb23-32" tabindex="-1"></a>  <span class="co"># use_colors &lt;- c(&#39;red&#39;, &#39;blue&#39;) %&gt;% set_names(names(use_samples))</span></span>
<span id="cb23-33"><a href="#cb23-33" tabindex="-1"></a>  ecdf_plots[[i]] <span class="ot">&lt;-</span> use_assay <span class="sc">%&gt;%</span> </span>
<span id="cb23-34"><a href="#cb23-34" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">log</span>(<span class="dv">1</span><span class="sc">+</span>cpm), <span class="at">color =</span> condition, <span class="at">group =</span> sample_alias)) <span class="sc">+</span></span>
<span id="cb23-35"><a href="#cb23-35" tabindex="-1"></a>    <span class="fu">stat_ecdf</span>(<span class="at">geom =</span> <span class="st">&#39;step&#39;</span>, <span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">linewidth =</span> one_pt<span class="sc">/</span><span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb23-36"><a href="#cb23-36" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> palette_vector_mmodel) <span class="sc">+</span></span>
<span id="cb23-37"><a href="#cb23-37" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;log(1+cpm)&quot;</span>,</span>
<span id="cb23-38"><a href="#cb23-38" tabindex="-1"></a>         <span class="at">color =</span> <span class="st">&#39;&#39;</span>,</span>
<span id="cb23-39"><a href="#cb23-39" tabindex="-1"></a>         <span class="at">sub =</span> i</span>
<span id="cb23-40"><a href="#cb23-40" tabindex="-1"></a>         ) <span class="sc">+</span></span>
<span id="cb23-41"><a href="#cb23-41" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb23-42"><a href="#cb23-42" tabindex="-1"></a>      <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">2</span>),</span>
<span id="cb23-43"><a href="#cb23-43" tabindex="-1"></a>      <span class="at">axis.text.y =</span>  <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">3</span>),</span>
<span id="cb23-44"><a href="#cb23-44" tabindex="-1"></a>      <span class="at">axis.text.x =</span>  <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">3</span>),</span>
<span id="cb23-45"><a href="#cb23-45" tabindex="-1"></a>      <span class="at">legend.text =</span>  <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">2</span>),</span>
<span id="cb23-46"><a href="#cb23-46" tabindex="-1"></a>      <span class="at">legend.title =</span>  <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">2</span>),</span>
<span id="cb23-47"><a href="#cb23-47" tabindex="-1"></a>      <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">3</span>),</span>
<span id="cb23-48"><a href="#cb23-48" tabindex="-1"></a>      <span class="at">plot.caption =</span>  <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">2</span>)</span>
<span id="cb23-49"><a href="#cb23-49" tabindex="-1"></a>      </span>
<span id="cb23-50"><a href="#cb23-50" tabindex="-1"></a>    )</span>
<span id="cb23-51"><a href="#cb23-51" tabindex="-1"></a></span>
<span id="cb23-52"><a href="#cb23-52" tabindex="-1"></a>}</span></code></pre></div>
<div id="supplementary-figure-4d-k-s-test-table" class="section level3">
<h3>Supplementary Figure 4D : K-S test table</h3>
<p>Table depicting the results of Kolmogorov-Smirnov tests, comparing
sgRNA distributions from the CRISPR KO datasets indicated, encompassing
three mammary carcinoma models (MVT1, 4T1, Py2T). The datasets are
obtained from primary tumors harvested from the experiments shown in
Figure 4A and Supplementary Figure 4A; n=5-7 each.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a>ks_test_df <span class="sc">%&gt;%</span> </span>
<span id="cb24-2"><a href="#cb24-2" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>case, <span class="sc">-</span>reference) <span class="sc">%&gt;%</span> </span>
<span id="cb24-3"><a href="#cb24-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">P =</span> <span class="fu">format.pval</span>(P)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-4"><a href="#cb24-4" tabindex="-1"></a>  <span class="fu">datatable</span>(., </span>
<span id="cb24-5"><a href="#cb24-5" tabindex="-1"></a>            <span class="at">rownames =</span> <span class="cn">FALSE</span>, </span>
<span id="cb24-6"><a href="#cb24-6" tabindex="-1"></a>            <span class="at">filter =</span> <span class="st">&#39;top&#39;</span>, </span>
<span id="cb24-7"><a href="#cb24-7" tabindex="-1"></a>            <span class="at">caption =</span> <span class="st">&#39;Results of Kolmogorov-Smirnov test&#39;</span>,</span>
<span id="cb24-8"><a href="#cb24-8" tabindex="-1"></a>            <span class="at">extensions =</span> <span class="st">&#39;Buttons&#39;</span>, </span>
<span id="cb24-9"><a href="#cb24-9" tabindex="-1"></a>            <span class="at">options =</span> <span class="fu">list</span>(</span>
<span id="cb24-10"><a href="#cb24-10" tabindex="-1"></a>              <span class="at">dom =</span> <span class="st">&#39;Blfrtip&#39;</span>,</span>
<span id="cb24-11"><a href="#cb24-11" tabindex="-1"></a>              <span class="at">buttons =</span> <span class="fu">c</span>(<span class="st">&#39;csv&#39;</span>, <span class="st">&#39;excel&#39;</span>),</span>
<span id="cb24-12"><a href="#cb24-12" tabindex="-1"></a>              <span class="at">columnDefs =</span> <span class="fu">list</span>(</span>
<span id="cb24-13"><a href="#cb24-13" tabindex="-1"></a>                <span class="fu">list</span>(<span class="at">width =</span> <span class="dv">120</span>, <span class="at">targets =</span> <span class="dv">1</span>)</span>
<span id="cb24-14"><a href="#cb24-14" tabindex="-1"></a>              )</span>
<span id="cb24-15"><a href="#cb24-15" tabindex="-1"></a>            )</span>
<span id="cb24-16"><a href="#cb24-16" tabindex="-1"></a>            ) <span class="sc">%&gt;%</span> </span>
<span id="cb24-17"><a href="#cb24-17" tabindex="-1"></a>    <span class="fu">formatRound</span>(<span class="at">columns =</span> <span class="fu">c</span>(<span class="st">&#39;D&#39;</span>), <span class="at">digits =</span> <span class="dv">3</span>)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-832cd5f680003d8dd78c" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-832cd5f680003d8dd78c">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.21920240782541\" data-max=\"0.382583396037086\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\" disabled=\"\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption>Results of Kolmogorov-Smirnov test<\/caption>","data":[["StealTHY syngeneic vs. StealTHY NSG","StealTHY syngeneic vs. Cas9+Puro"],[0.2192024078254109,0.3825833960370851],["&lt; 2.22e-16","&lt; 2.22e-16"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>comparison<\/th>\n      <th>D<\/th>\n      <th>P<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["csv","excel"],"columnDefs":[{"targets":1,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, 3, \",\", \".\", null);\n  }"},{"width":120,"targets":1},{"className":"dt-right","targets":1},{"name":"comparison","targets":0},{"name":"D","targets":1},{"name":"P","targets":2}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true}},"evals":["options.columnDefs.0.render"],"jsHooks":[]}</script>
</div>
</div>
<div id="hartigans-dip-test" class="section level2">
<h2>Hartigans’ dip test</h2>
<p>Hartigans’ dip test for unimodality. If the p-value is less than
0.05, there’s enough evidence to claim that the data are not unimodal
(multimodal, at least bimodal).</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a>use_assay <span class="ot">&lt;-</span> <span class="fu">assay</span>(se, <span class="st">&#39;cpm&#39;</span>)</span>
<span id="cb25-2"><a href="#cb25-2" tabindex="-1"></a>i <span class="ot">&lt;-</span> <span class="fu">colnames</span>(use_assay)[<span class="dv">1</span>]</span>
<span id="cb25-3"><a href="#cb25-3" tabindex="-1"></a>modality_test_df <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i=</span> <span class="fu">colnames</span>(use_assay), <span class="at">.combine =</span> rbind) <span class="sc">%do%</span>{</span>
<span id="cb25-4"><a href="#cb25-4" tabindex="-1"></a>  x <span class="ot">&lt;-</span> use_assay[,i]</span>
<span id="cb25-5"><a href="#cb25-5" tabindex="-1"></a>  dip.res <span class="ot">&lt;-</span> <span class="fu">dip.test</span>(x, <span class="at">simulate.p.value =</span> <span class="cn">FALSE</span>, <span class="at">B =</span> <span class="dv">2000</span>)</span>
<span id="cb25-6"><a href="#cb25-6" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb25-7"><a href="#cb25-7" tabindex="-1"></a>    <span class="at">sample_alias =</span> i,</span>
<span id="cb25-8"><a href="#cb25-8" tabindex="-1"></a>    <span class="at">dip.D =</span> dip.res<span class="sc">$</span>statistic,</span>
<span id="cb25-9"><a href="#cb25-9" tabindex="-1"></a>    <span class="at">dip.p =</span> dip.res<span class="sc">$</span>p.value,</span>
<span id="cb25-10"><a href="#cb25-10" tabindex="-1"></a>    <span class="at">dip.result =</span> <span class="fu">ifelse</span>(dip.res<span class="sc">$</span>p.value <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="st">&#39;multimodal&#39;</span>, <span class="st">&#39;unimodal&#39;</span>)</span>
<span id="cb25-11"><a href="#cb25-11" tabindex="-1"></a>    )</span>
<span id="cb25-12"><a href="#cb25-12" tabindex="-1"></a>}</span>
<span id="cb25-13"><a href="#cb25-13" tabindex="-1"></a></span>
<span id="cb25-14"><a href="#cb25-14" tabindex="-1"></a>modality_test_df <span class="sc">%&lt;&gt;%</span> </span>
<span id="cb25-15"><a href="#cb25-15" tabindex="-1"></a>  <span class="fu">left_join</span>(group_df_modality) <span class="sc">%&gt;%</span> </span>
<span id="cb25-16"><a href="#cb25-16" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">colData</span>(se) <span class="sc">%&gt;%</span> data.frame)</span></code></pre></div>
<div id="dip-test-table-for-syngeneic-and-nsg-models"
class="section level3">
<h3>Dip test table for Syngeneic and NSG models</h3>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a>modality_test_df <span class="sc">%&gt;%</span> </span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(modality_group)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-3"><a href="#cb26-3" tabindex="-1"></a>  <span class="fu">filter</span>(modality_group <span class="sc">!=</span> <span class="st">&quot;StealTHY syngeneic&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb26-4"><a href="#cb26-4" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb26-5"><a href="#cb26-5" tabindex="-1"></a>    <span class="at">modality_group =</span> <span class="fu">factor</span>(</span>
<span id="cb26-6"><a href="#cb26-6" tabindex="-1"></a>      modality_group, </span>
<span id="cb26-7"><a href="#cb26-7" tabindex="-1"></a>      <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;StealTHY syngeneic 4T1&quot;</span>,</span>
<span id="cb26-8"><a href="#cb26-8" tabindex="-1"></a>                 <span class="st">&quot;Cas9+Puro syngeneic&quot;</span>,</span>
<span id="cb26-9"><a href="#cb26-9" tabindex="-1"></a>                 <span class="st">&quot;StealTHY NSG&quot;</span>,</span>
<span id="cb26-10"><a href="#cb26-10" tabindex="-1"></a>                 <span class="st">&quot;Cas9+Puro NSG&quot;</span>)</span>
<span id="cb26-11"><a href="#cb26-11" tabindex="-1"></a>      )</span>
<span id="cb26-12"><a href="#cb26-12" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb26-13"><a href="#cb26-13" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">modality_group =</span> <span class="fu">fct_recode</span>(</span>
<span id="cb26-14"><a href="#cb26-14" tabindex="-1"></a>    modality_group, </span>
<span id="cb26-15"><a href="#cb26-15" tabindex="-1"></a>    <span class="st">`</span><span class="at">StealTHY syngeneic</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;StealTHY syngeneic 4T1&quot;</span>)</span>
<span id="cb26-16"><a href="#cb26-16" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb26-17"><a href="#cb26-17" tabindex="-1"></a>  <span class="fu">group_by</span>(modality_group) <span class="sc">%&gt;%</span></span>
<span id="cb26-18"><a href="#cb26-18" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb26-19"><a href="#cb26-19" tabindex="-1"></a>    <span class="st">`</span><span class="at">Median D</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">median</span>(dip.D) <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="dv">3</span>),</span>
<span id="cb26-20"><a href="#cb26-20" tabindex="-1"></a>    <span class="at">n =</span> <span class="fu">n</span>(),</span>
<span id="cb26-21"><a href="#cb26-21" tabindex="-1"></a>    <span class="st">`</span><span class="at">Multimodal samples (n)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(dip.result <span class="sc">==</span> <span class="st">&#39;multimodal&#39;</span>),</span>
<span id="cb26-22"><a href="#cb26-22" tabindex="-1"></a>    <span class="st">`</span><span class="at">Unimodal samples (n)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(dip.result <span class="sc">==</span> <span class="st">&#39;unimodal&#39;</span>),</span>
<span id="cb26-23"><a href="#cb26-23" tabindex="-1"></a>    <span class="st">`</span><span class="at">Multimodal samples (%)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">round</span>(<span class="dv">100</span><span class="sc">*</span><span class="st">`</span><span class="at">Multimodal samples (n)</span><span class="st">`</span><span class="sc">/</span>n, <span class="dv">1</span>),</span>
<span id="cb26-24"><a href="#cb26-24" tabindex="-1"></a>    <span class="st">`</span><span class="at">Unimodal samples (%)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">round</span>(<span class="dv">100</span><span class="sc">*</span><span class="st">`</span><span class="at">Unimodal samples (n)</span><span class="st">`</span><span class="sc">/</span>n, <span class="dv">1</span>)</span>
<span id="cb26-25"><a href="#cb26-25" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> </span>
<span id="cb26-26"><a href="#cb26-26" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">check.names =</span> F) <span class="sc">%&gt;%</span> </span>
<span id="cb26-27"><a href="#cb26-27" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">caption =</span> <span class="st">&#39;Dip test results for Syngeneic and NSG models&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb26-28"><a href="#cb26-28" tabindex="-1"></a>  <span class="fu">kable_paper</span>(<span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">&quot;striped&quot;</span>, <span class="st">&quot;hover&quot;</span>, <span class="st">&quot;condensed&quot;</span>), <span class="at">full_width =</span> F)</span></code></pre></div>
<table class=" lightable-paper" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
Dip test results for Syngeneic and NSG models
</caption>
<thead>
<tr>
<th style="text-align:left;">
modality_group
</th>
<th style="text-align:right;">
Median D
</th>
<th style="text-align:right;">
n
</th>
<th style="text-align:right;">
Multimodal samples (n)
</th>
<th style="text-align:right;">
Unimodal samples (n)
</th>
<th style="text-align:right;">
Multimodal samples (%)
</th>
<th style="text-align:right;">
Unimodal samples (%)
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
StealTHY syngeneic
</td>
<td style="text-align:right;">
0.019
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
60.0
</td>
<td style="text-align:right;">
40.0
</td>
</tr>
<tr>
<td style="text-align:left;">
Cas9+Puro syngeneic
</td>
<td style="text-align:right;">
0.068
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
100.0
</td>
<td style="text-align:right;">
0.0
</td>
</tr>
<tr>
<td style="text-align:left;">
StealTHY NSG
</td>
<td style="text-align:right;">
0.031
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
86.7
</td>
<td style="text-align:right;">
13.3
</td>
</tr>
<tr>
<td style="text-align:left;">
Cas9+Puro NSG
</td>
<td style="text-align:right;">
0.021
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
80.0
</td>
<td style="text-align:right;">
20.0
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="differential-abundance-analysis" class="section level2">
<h2>Differential abundance analysis</h2>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a>gene_info <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(<span class="st">&#39;./data/resources/selected_gene_list/gene_list_for_figure_5c.xlsx&#39;</span>)</span>
<span id="cb27-2"><a href="#cb27-2" tabindex="-1"></a></span>
<span id="cb27-3"><a href="#cb27-3" tabindex="-1"></a>use_colors <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb27-4"><a href="#cb27-4" tabindex="-1"></a>  <span class="st">`</span><span class="at">Immune system cytokines and chemokines</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;grey&#39;</span>,</span>
<span id="cb27-5"><a href="#cb27-5" tabindex="-1"></a>  <span class="st">`</span><span class="at">Immune system cytokines</span><span class="sc">\n</span><span class="at">and chemokines</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;grey&#39;</span>,</span>
<span id="cb27-6"><a href="#cb27-6" tabindex="-1"></a>  </span>
<span id="cb27-7"><a href="#cb27-7" tabindex="-1"></a>  <span class="st">`</span><span class="at">Interferone, TNF and danger/death-inducing signals</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;black&#39;</span>,</span>
<span id="cb27-8"><a href="#cb27-8" tabindex="-1"></a>  <span class="st">`</span><span class="at">Interferon, TNF</span><span class="sc">\n</span><span class="at">and danger/death-inducing</span><span class="sc">\n</span><span class="at">signals</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;black&#39;</span>,</span>
<span id="cb27-9"><a href="#cb27-9" tabindex="-1"></a>  <span class="st">`</span><span class="at">Interferon, TNF and</span><span class="sc">\n</span><span class="at">danger/death-inducing</span><span class="sc">\n</span><span class="at">signals</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;black&#39;</span>,</span>
<span id="cb27-10"><a href="#cb27-10" tabindex="-1"></a>  </span>
<span id="cb27-11"><a href="#cb27-11" tabindex="-1"></a>  <span class="st">`</span><span class="at">TGF-β family related</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;red&#39;</span>,</span>
<span id="cb27-12"><a href="#cb27-12" tabindex="-1"></a>  </span>
<span id="cb27-13"><a href="#cb27-13" tabindex="-1"></a>  <span class="st">`</span><span class="at">RTK signaling</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;blue&#39;</span>,</span>
<span id="cb27-14"><a href="#cb27-14" tabindex="-1"></a>  <span class="st">`</span><span class="at">Positive controls</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;black&#39;</span></span>
<span id="cb27-15"><a href="#cb27-15" tabindex="-1"></a>)</span>
<span id="cb27-16"><a href="#cb27-16" tabindex="-1"></a></span>
<span id="cb27-17"><a href="#cb27-17" tabindex="-1"></a>use_fill <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb27-18"><a href="#cb27-18" tabindex="-1"></a>  <span class="st">`</span><span class="at">Immune system cytokines and chemokines</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;grey&#39;</span>,</span>
<span id="cb27-19"><a href="#cb27-19" tabindex="-1"></a>  <span class="st">`</span><span class="at">Immune system cytokines</span><span class="sc">\n</span><span class="at">and chemokines</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;grey&#39;</span>,</span>
<span id="cb27-20"><a href="#cb27-20" tabindex="-1"></a>  </span>
<span id="cb27-21"><a href="#cb27-21" tabindex="-1"></a>  <span class="st">`</span><span class="at">Interferone, TNF and danger/death-inducing signals</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;black&#39;</span>,</span>
<span id="cb27-22"><a href="#cb27-22" tabindex="-1"></a>  <span class="st">`</span><span class="at">Interferon, TNF</span><span class="sc">\n</span><span class="at">and danger/death-inducing</span><span class="sc">\n</span><span class="at">signals</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;black&#39;</span>,</span>
<span id="cb27-23"><a href="#cb27-23" tabindex="-1"></a>  <span class="st">`</span><span class="at">Interferon, TNF and</span><span class="sc">\n</span><span class="at">danger/death-inducing</span><span class="sc">\n</span><span class="at">signals</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;black&#39;</span>,</span>
<span id="cb27-24"><a href="#cb27-24" tabindex="-1"></a>  <span class="st">`</span><span class="at">TGF-β family related</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;red&#39;</span>,</span>
<span id="cb27-25"><a href="#cb27-25" tabindex="-1"></a>  </span>
<span id="cb27-26"><a href="#cb27-26" tabindex="-1"></a>  <span class="st">`</span><span class="at">RTK signaling</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;blue&#39;</span>,</span>
<span id="cb27-27"><a href="#cb27-27" tabindex="-1"></a>  <span class="st">`</span><span class="at">Positive controls</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;white&#39;</span></span>
<span id="cb27-28"><a href="#cb27-28" tabindex="-1"></a>)</span>
<span id="cb27-29"><a href="#cb27-29" tabindex="-1"></a></span>
<span id="cb27-30"><a href="#cb27-30" tabindex="-1"></a>use_shape <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb27-31"><a href="#cb27-31" tabindex="-1"></a>  <span class="st">`</span><span class="at">Immune system cytokines and chemokines</span><span class="st">`</span> <span class="ot">=</span> <span class="dv">16</span>,</span>
<span id="cb27-32"><a href="#cb27-32" tabindex="-1"></a>  <span class="st">`</span><span class="at">Immune system cytokines</span><span class="sc">\n</span><span class="at">and chemokines</span><span class="st">`</span> <span class="ot">=</span> <span class="dv">16</span>,</span>
<span id="cb27-33"><a href="#cb27-33" tabindex="-1"></a>  </span>
<span id="cb27-34"><a href="#cb27-34" tabindex="-1"></a>  <span class="st">`</span><span class="at">Interferone, TNF and danger/death-inducing signals</span><span class="st">`</span> <span class="ot">=</span> <span class="dv">16</span>,</span>
<span id="cb27-35"><a href="#cb27-35" tabindex="-1"></a>  <span class="st">`</span><span class="at">Interferon, TNF</span><span class="sc">\n</span><span class="at">and danger/death-inducing</span><span class="sc">\n</span><span class="at">signals</span><span class="st">`</span> <span class="ot">=</span> <span class="dv">16</span>,</span>
<span id="cb27-36"><a href="#cb27-36" tabindex="-1"></a>  <span class="st">`</span><span class="at">Interferon, TNF and</span><span class="sc">\n</span><span class="at">danger/death-inducing</span><span class="sc">\n</span><span class="at">signals</span><span class="st">`</span> <span class="ot">=</span> <span class="dv">16</span>,</span>
<span id="cb27-37"><a href="#cb27-37" tabindex="-1"></a>  <span class="st">`</span><span class="at">TGF-β family related</span><span class="st">`</span> <span class="ot">=</span> <span class="dv">16</span>,</span>
<span id="cb27-38"><a href="#cb27-38" tabindex="-1"></a>  </span>
<span id="cb27-39"><a href="#cb27-39" tabindex="-1"></a>  <span class="st">`</span><span class="at">RTK signaling</span><span class="st">`</span> <span class="ot">=</span> <span class="dv">16</span>,</span>
<span id="cb27-40"><a href="#cb27-40" tabindex="-1"></a>  <span class="st">`</span><span class="at">Positive controls</span><span class="st">`</span> <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb27-41"><a href="#cb27-41" tabindex="-1"></a>)</span>
<span id="cb27-42"><a href="#cb27-42" tabindex="-1"></a></span>
<span id="cb27-43"><a href="#cb27-43" tabindex="-1"></a><span class="co"># get gene order based on Comparison : `StealTHY KO Syngeneic` </span></span>
<span id="cb27-44"><a href="#cb27-44" tabindex="-1"></a>gene_levels <span class="ot">&lt;-</span> gene_summ[[<span class="st">&#39;stealthy_ko_syngeneic&#39;</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb27-45"><a href="#cb27-45" tabindex="-1"></a>  <span class="fu">ReadRRA</span>(<span class="at">score =</span> <span class="st">&#39;lfc&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-46"><a href="#cb27-46" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">LFC =</span> Score) <span class="sc">%&gt;%</span></span>
<span id="cb27-47"><a href="#cb27-47" tabindex="-1"></a>  <span class="fu">left_join</span>(gene_info, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&#39;id&#39;</span> <span class="ot">=</span> <span class="st">&#39;id&#39;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb27-48"><a href="#cb27-48" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(gene_set)) <span class="sc">%&gt;%</span></span>
<span id="cb27-49"><a href="#cb27-49" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">gene_name =</span> <span class="fu">fct_reorder</span>(gene_name, LFC, <span class="at">.desc =</span> <span class="cn">FALSE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb27-50"><a href="#cb27-50" tabindex="-1"></a>  <span class="fu">pull</span>(gene_name) <span class="sc">%&gt;%</span></span>
<span id="cb27-51"><a href="#cb27-51" tabindex="-1"></a>  levels</span></code></pre></div>
<div id="figure-4c-and-supplementary-figure-4f" class="section level3">
<h3>Figure 4C and Supplementary Figure 4F</h3>
<p>Dot plots highlighting hits identified following the strategy
depicted in Figure 4A upon transplantation in the indicated hosts of
metastatic mammary carcinoma cells (MVT1, 4T1, Py2T) previously
subjected to Stealthy KO or gold standard KO by Cas9+PuroR;
representative genes from four recurrently identified gene families are
shown in comparison across screening methods. The fold-change values
indicate differential sgRNA abundance between primary tumors and
synchronous lung metastases; n=5-7 each; RTK, receptor tyrosine kinase,
Tnf, tumor necrosis factor; Ifn, interferon; FDR, false discovery
rate.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a>use_comp <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;stealthy_ko_syngeneic&#39;</span>, <span class="st">&#39;stealthy_ko_nsg&#39;</span>, <span class="st">&#39;cas9+puro_syngeneic&#39;</span>)</span>
<span id="cb28-2"><a href="#cb28-2" tabindex="-1"></a></span>
<span id="cb28-3"><a href="#cb28-3" tabindex="-1"></a><span class="co"># FDR limits</span></span>
<span id="cb28-4"><a href="#cb28-4" tabindex="-1"></a>xfdr <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i=</span>gene_summ[use_comp], <span class="at">.combine =</span> rbind) <span class="sc">%do%</span> {</span>
<span id="cb28-5"><a href="#cb28-5" tabindex="-1"></a>  i <span class="sc">%&gt;%</span> </span>
<span id="cb28-6"><a href="#cb28-6" tabindex="-1"></a>    <span class="fu">ReadRRA</span>(<span class="at">score =</span> <span class="st">&#39;lfc&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb28-7"><a href="#cb28-7" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">LFC =</span> Score) <span class="sc">%&gt;%</span> </span>
<span id="cb28-8"><a href="#cb28-8" tabindex="-1"></a>    <span class="fu">left_join</span>(gene_info, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&#39;id&#39;</span> <span class="ot">=</span> <span class="st">&#39;id&#39;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb28-9"><a href="#cb28-9" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(gene_set)) <span class="sc">%&gt;%</span> </span>
<span id="cb28-10"><a href="#cb28-10" tabindex="-1"></a>    <span class="fu">pull</span>(FDR) <span class="sc">%&gt;%</span> </span>
<span id="cb28-11"><a href="#cb28-11" tabindex="-1"></a>    min</span>
<span id="cb28-12"><a href="#cb28-12" tabindex="-1"></a>}</span>
<span id="cb28-13"><a href="#cb28-13" tabindex="-1"></a>maxFDR <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(xfdr[,<span class="dv">1</span>]) <span class="sc">%&gt;%</span> max</span>
<span id="cb28-14"><a href="#cb28-14" tabindex="-1"></a> </span>
<span id="cb28-15"><a href="#cb28-15" tabindex="-1"></a><span class="co"># Set LFC (x-axis) for comparable plots to &quot;stealthy_ko_syngeneic&quot;: &quot;stealthy_ko_nsg&quot;</span></span>
<span id="cb28-16"><a href="#cb28-16" tabindex="-1"></a>stealthy_comp <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;stealthy_ko_syngeneic&quot;</span>, <span class="st">&quot;stealthy_ko_nsg&quot;</span>)</span>
<span id="cb28-17"><a href="#cb28-17" tabindex="-1"></a>xlfc <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i=</span>gene_summ[stealthy_comp], <span class="at">.combine =</span> rbind) <span class="sc">%do%</span> {</span>
<span id="cb28-18"><a href="#cb28-18" tabindex="-1"></a>  i <span class="sc">%&gt;%</span> </span>
<span id="cb28-19"><a href="#cb28-19" tabindex="-1"></a>    <span class="fu">ReadRRA</span>(<span class="at">score =</span> <span class="st">&#39;lfc&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb28-20"><a href="#cb28-20" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">LFC =</span> Score) <span class="sc">%&gt;%</span> </span>
<span id="cb28-21"><a href="#cb28-21" tabindex="-1"></a>    <span class="fu">left_join</span>(gene_info, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&#39;id&#39;</span> <span class="ot">=</span> <span class="st">&#39;id&#39;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb28-22"><a href="#cb28-22" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(gene_set)) <span class="sc">%&gt;%</span> </span>
<span id="cb28-23"><a href="#cb28-23" tabindex="-1"></a>    <span class="fu">pull</span>(LFC) <span class="sc">%&gt;%</span> </span>
<span id="cb28-24"><a href="#cb28-24" tabindex="-1"></a>    abs <span class="sc">%&gt;%</span> max</span>
<span id="cb28-25"><a href="#cb28-25" tabindex="-1"></a>}</span>
<span id="cb28-26"><a href="#cb28-26" tabindex="-1"></a>stealthy_maxLFC <span class="ot">&lt;-</span> <span class="fu">max</span>(xlfc[,<span class="dv">1</span>])</span>
<span id="cb28-27"><a href="#cb28-27" tabindex="-1"></a>stealthy_xlim <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="sc">-</span>stealthy_maxLFC, stealthy_maxLFC)</span>
<span id="cb28-28"><a href="#cb28-28" tabindex="-1"></a>stealthy_breaks <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">2.5</span>, <span class="sc">-</span><span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="fl">2.5</span>)</span>
<span id="cb28-29"><a href="#cb28-29" tabindex="-1"></a>stealthy_labels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;-2.5&#39;</span>, <span class="st">&#39;-1&#39;</span>, <span class="st">&#39;0&#39;</span>, <span class="st">&#39;1&#39;</span>, <span class="st">&#39;2.5&#39;</span>)</span>
<span id="cb28-30"><a href="#cb28-30" tabindex="-1"></a></span>
<span id="cb28-31"><a href="#cb28-31" tabindex="-1"></a>i <span class="ot">&lt;-</span> use_comp[<span class="dv">1</span>]</span>
<span id="cb28-32"><a href="#cb28-32" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> use_comp)  {</span>
<span id="cb28-33"><a href="#cb28-33" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;####&quot;</span>, mageck_comp_list_names[i], <span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb28-34"><a href="#cb28-34" tabindex="-1"></a>  </span>
<span id="cb28-35"><a href="#cb28-35" tabindex="-1"></a>  x <span class="ot">&lt;-</span> gene_summ[[i]] <span class="sc">%&gt;%</span> </span>
<span id="cb28-36"><a href="#cb28-36" tabindex="-1"></a>    <span class="fu">ReadRRA</span>(<span class="at">score =</span> <span class="st">&#39;lfc&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb28-37"><a href="#cb28-37" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">LFC =</span> Score) <span class="sc">%&gt;%</span> </span>
<span id="cb28-38"><a href="#cb28-38" tabindex="-1"></a>    <span class="fu">left_join</span>(gene_info, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&#39;id&#39;</span> <span class="ot">=</span> <span class="st">&#39;id&#39;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb28-39"><a href="#cb28-39" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(gene_set)) <span class="sc">%&gt;%</span> </span>
<span id="cb28-40"><a href="#cb28-40" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb28-41"><a href="#cb28-41" tabindex="-1"></a>      <span class="at">gene_name =</span> <span class="fu">factor</span>(gene_name, <span class="at">levels =</span> gene_levels),</span>
<span id="cb28-42"><a href="#cb28-42" tabindex="-1"></a>      <span class="at">gene_set =</span> <span class="fu">ifelse</span>(</span>
<span id="cb28-43"><a href="#cb28-43" tabindex="-1"></a>        gene_set <span class="sc">==</span> <span class="st">&#39;Immune system cytokines and chemokines&#39;</span>,</span>
<span id="cb28-44"><a href="#cb28-44" tabindex="-1"></a>        <span class="st">&#39;Immune system cytokines</span><span class="sc">\n</span><span class="st">and chemokines&#39;</span>,</span>
<span id="cb28-45"><a href="#cb28-45" tabindex="-1"></a>        gene_set</span>
<span id="cb28-46"><a href="#cb28-46" tabindex="-1"></a>      ),</span>
<span id="cb28-47"><a href="#cb28-47" tabindex="-1"></a>      <span class="at">gene_set =</span> <span class="fu">ifelse</span>(</span>
<span id="cb28-48"><a href="#cb28-48" tabindex="-1"></a>        gene_set <span class="sc">==</span> <span class="st">&#39;Interferone, TNF and danger/death-inducing signals&#39;</span>,</span>
<span id="cb28-49"><a href="#cb28-49" tabindex="-1"></a>        <span class="st">&#39;Interferon, TNF and</span><span class="sc">\n</span><span class="st">danger/death-inducing</span><span class="sc">\n</span><span class="st">signals&#39;</span>,</span>
<span id="cb28-50"><a href="#cb28-50" tabindex="-1"></a>        gene_set</span>
<span id="cb28-51"><a href="#cb28-51" tabindex="-1"></a>      ),</span>
<span id="cb28-52"><a href="#cb28-52" tabindex="-1"></a>      <span class="at">gene_set =</span> <span class="fu">factor</span>(gene_set, <span class="fu">names</span>(use_colors))</span>
<span id="cb28-53"><a href="#cb28-53" tabindex="-1"></a>      )</span>
<span id="cb28-54"><a href="#cb28-54" tabindex="-1"></a>  xlim <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fu">ceiling</span>(<span class="fu">max</span>(<span class="fu">abs</span>(x<span class="sc">$</span>LFC), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)), </span>
<span id="cb28-55"><a href="#cb28-55" tabindex="-1"></a>            <span class="fu">ceiling</span>(<span class="fu">max</span>(<span class="fu">abs</span>(x<span class="sc">$</span>LFC), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)))</span>
<span id="cb28-56"><a href="#cb28-56" tabindex="-1"></a>  size_breaks <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.25</span>, <span class="fl">0.5</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="fu">ceiling</span>(maxFDR))</span>
<span id="cb28-57"><a href="#cb28-57" tabindex="-1"></a>  res <span class="ot">&lt;-</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb28-58"><a href="#cb28-58" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(LFC, gene_name, <span class="at">color =</span> gene_set, <span class="at">size =</span> <span class="sc">-</span><span class="fu">log10</span>(FDR))) <span class="sc">+</span></span>
<span id="cb28-59"><a href="#cb28-59" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">shape =</span> gene_set)) <span class="sc">+</span></span>
<span id="cb28-60"><a href="#cb28-60" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="dv">3</span>, <span class="at">linewidth =</span> one_pt<span class="sc">/</span><span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb28-61"><a href="#cb28-61" tabindex="-1"></a>    <span class="fu">scale_y_discrete</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">add =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>))) <span class="sc">+</span> <span class="co"># add fixed units around each facet to avoid lines over points</span></span>
<span id="cb28-62"><a href="#cb28-62" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> use_colors) <span class="sc">+</span></span>
<span id="cb28-63"><a href="#cb28-63" tabindex="-1"></a>    <span class="fu">scale_size</span>(<span class="at">range =</span> <span class="fu">c</span>(<span class="fl">0.5</span>, <span class="fl">2.5</span>), </span>
<span id="cb28-64"><a href="#cb28-64" tabindex="-1"></a>               <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fu">ceiling</span>(maxFDR)), </span>
<span id="cb28-65"><a href="#cb28-65" tabindex="-1"></a>               <span class="at">breaks =</span> size_breaks,</span>
<span id="cb28-66"><a href="#cb28-66" tabindex="-1"></a>               <span class="at">labels =</span> size_breaks <span class="sc">%&gt;%</span> as.character <span class="co"># remove trailing 0 from decimals</span></span>
<span id="cb28-67"><a href="#cb28-67" tabindex="-1"></a>               ) <span class="sc">+</span></span>
<span id="cb28-68"><a href="#cb28-68" tabindex="-1"></a>    <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> use_shape) <span class="sc">+</span></span>
<span id="cb28-69"><a href="#cb28-69" tabindex="-1"></a>    <span class="fu">scale_x_continuous</span>(</span>
<span id="cb28-70"><a href="#cb28-70" tabindex="-1"></a>        <span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">add =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>))</span>
<span id="cb28-71"><a href="#cb28-71" tabindex="-1"></a>        ) <span class="sc">+</span></span>
<span id="cb28-72"><a href="#cb28-72" tabindex="-1"></a>    <span class="fu">facet_grid</span>(<span class="at">rows =</span> <span class="fu">vars</span>(gene_set), <span class="at">scales =</span> <span class="st">&#39;free_y&#39;</span>, <span class="at">space =</span> <span class="st">&#39;free_y&#39;</span>) <span class="sc">+</span></span>
<span id="cb28-73"><a href="#cb28-73" tabindex="-1"></a>    theme_facet <span class="sc">+</span></span>
<span id="cb28-74"><a href="#cb28-74" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb28-75"><a href="#cb28-75" tabindex="-1"></a>      <span class="at">axis.ticks =</span> <span class="fu">element_line</span>(<span class="at">linewidth =</span> one_pt<span class="sc">/</span><span class="dv">2</span>, <span class="at">color =</span> <span class="st">&#39;black&#39;</span>),</span>
<span id="cb28-76"><a href="#cb28-76" tabindex="-1"></a>      <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">linewidth =</span> one_pt<span class="sc">/</span><span class="dv">2</span>, <span class="at">color =</span> <span class="st">&#39;black&#39;</span>),</span>
<span id="cb28-77"><a href="#cb28-77" tabindex="-1"></a>      <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">8</span>), </span>
<span id="cb28-78"><a href="#cb28-78" tabindex="-1"></a>      <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>),</span>
<span id="cb28-79"><a href="#cb28-79" tabindex="-1"></a>      <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">8</span>),</span>
<span id="cb28-80"><a href="#cb28-80" tabindex="-1"></a>      <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">8</span>),</span>
<span id="cb28-81"><a href="#cb28-81" tabindex="-1"></a>      <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">4</span>)</span>
<span id="cb28-82"><a href="#cb28-82" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb28-83"><a href="#cb28-83" tabindex="-1"></a>    <span class="fu">labs</span>(</span>
<span id="cb28-84"><a href="#cb28-84" tabindex="-1"></a>      <span class="at">x =</span> <span class="fu">expression</span>(<span class="fu">paste</span>(<span class="st">&quot;lo&quot;</span>, g[<span class="dv">2</span>],<span class="st">&quot;(Fold change)&quot;</span>)),</span>
<span id="cb28-85"><a href="#cb28-85" tabindex="-1"></a>      <span class="at">y =</span> <span class="cn">NULL</span>,</span>
<span id="cb28-86"><a href="#cb28-86" tabindex="-1"></a>      <span class="at">size =</span> <span class="fu">expression</span>(<span class="fu">paste</span>(<span class="st">&quot;-lo&quot;</span>, g[<span class="dv">10</span>],<span class="st">&quot;(FDR)&quot;</span>)),</span>
<span id="cb28-87"><a href="#cb28-87" tabindex="-1"></a>      <span class="at">title =</span> </span>
<span id="cb28-88"><a href="#cb28-88" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb28-89"><a href="#cb28-89" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">color =</span> <span class="st">&#39;none&#39;</span>, <span class="at">shape =</span> <span class="st">&#39;none&#39;</span>)</span>
<span id="cb28-90"><a href="#cb28-90" tabindex="-1"></a>  </span>
<span id="cb28-91"><a href="#cb28-91" tabindex="-1"></a>  <span class="cf">if</span>(i <span class="sc">%in%</span> stealthy_comp)</span>
<span id="cb28-92"><a href="#cb28-92" tabindex="-1"></a>    res <span class="ot">&lt;-</span> res <span class="sc">+</span></span>
<span id="cb28-93"><a href="#cb28-93" tabindex="-1"></a>      <span class="fu">scale_x_continuous</span>(</span>
<span id="cb28-94"><a href="#cb28-94" tabindex="-1"></a>        <span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">add =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>)), </span>
<span id="cb28-95"><a href="#cb28-95" tabindex="-1"></a>        <span class="at">limits =</span> stealthy_xlim,</span>
<span id="cb28-96"><a href="#cb28-96" tabindex="-1"></a>        <span class="at">breaks =</span> stealthy_breaks,</span>
<span id="cb28-97"><a href="#cb28-97" tabindex="-1"></a>        <span class="at">labels =</span> stealthy_labels</span>
<span id="cb28-98"><a href="#cb28-98" tabindex="-1"></a>      )</span>
<span id="cb28-99"><a href="#cb28-99" tabindex="-1"></a>  </span>
<span id="cb28-100"><a href="#cb28-100" tabindex="-1"></a>  <span class="fu">print</span>(res)</span>
<span id="cb28-101"><a href="#cb28-101" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&#39;</span><span class="sc">\n\n</span><span class="st">&#39;</span>)</span>
<span id="cb28-102"><a href="#cb28-102" tabindex="-1"></a>}</span></code></pre></div>
<div id="stealthy-ko-syngeneic" class="section level4">
<h4>StealTHY KO Syngeneic</h4>
<p><img src="figure/crispr-mm_2215_sgRNA-StealTHY.Rmd/da-selected-genes-dotplot-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-da-selected-genes-dotplot-1">
Past versions of da-selected-genes-dotplot-1.png
</button>
</p>
<div id="fig-da-selected-genes-dotplot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/32aca023d6d8209f1aa3df3f177fc383c7569844/docs/figure/crispr-mm_2215_sgRNA-StealTHY.Rmd/da-selected-genes-dotplot-1.png" target="_blank">32aca02</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-07-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="stealthy-ko-nsg" class="section level4">
<h4>StealTHY KO NSG</h4>
<p><img src="figure/crispr-mm_2215_sgRNA-StealTHY.Rmd/da-selected-genes-dotplot-2.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-da-selected-genes-dotplot-2">
Past versions of da-selected-genes-dotplot-2.png
</button>
</p>
<div id="fig-da-selected-genes-dotplot-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/32aca023d6d8209f1aa3df3f177fc383c7569844/docs/figure/crispr-mm_2215_sgRNA-StealTHY.Rmd/da-selected-genes-dotplot-2.png" target="_blank">32aca02</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-07-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cas9puro-syngeneic" class="section level4">
<h4>Cas9+Puro Syngeneic</h4>
<p><img src="figure/crispr-mm_2215_sgRNA-StealTHY.Rmd/da-selected-genes-dotplot-3.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-da-selected-genes-dotplot-3">
Past versions of da-selected-genes-dotplot-3.png
</button>
</p>
<div id="fig-da-selected-genes-dotplot-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/32aca023d6d8209f1aa3df3f177fc383c7569844/docs/figure/crispr-mm_2215_sgRNA-StealTHY.Rmd/da-selected-genes-dotplot-3.png" target="_blank">32aca02</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-07-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="supplementary-figure-4e" class="section level3">
<h3>Supplementary Figure 4E</h3>
<p>Dot plots presenting differential sgRNA abundance for selected
positive control genes in mammary carcinoma models (MVT1, 4T1, Py2T)
subjected to StealTHY KO with respect to a mock KO reference dataset. In
the mock KO condition, dCas9 transduction was performed instead of
Apo-Cas9 transfection in the same sgRNA-transduced cells. Top: dot plot
showing differential sgRNA abundance measured in vitro in cells prior to
transplantation, i.e. harvested 60 hours after Apo-Cas9 transfection or
dCas9 transduction; n=3 samples per group. Bottom: dot plot showing
differential sgRNA abundance measured in primary tumors in vivo,
harvested at the experimental endpoint from syngeneic hosts; n=5-7 mice
per group.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" tabindex="-1"></a>use_comp <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;stealthy_ko_over_dcas9_in_vitro&#39;</span>, <span class="st">&#39;stealthy_ko_over_dcas9_in_vivo&#39;</span>)</span>
<span id="cb29-2"><a href="#cb29-2" tabindex="-1"></a></span>
<span id="cb29-3"><a href="#cb29-3" tabindex="-1"></a><span class="co"># FDR limits</span></span>
<span id="cb29-4"><a href="#cb29-4" tabindex="-1"></a>xfdr <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i=</span>gene_summ[use_comp], <span class="at">.combine =</span> rbind) <span class="sc">%do%</span> {</span>
<span id="cb29-5"><a href="#cb29-5" tabindex="-1"></a>  i <span class="sc">%&gt;%</span> </span>
<span id="cb29-6"><a href="#cb29-6" tabindex="-1"></a>    <span class="fu">ReadRRA</span>(<span class="at">score =</span> <span class="st">&#39;lfc&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb29-7"><a href="#cb29-7" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">LFC =</span> Score) <span class="sc">%&gt;%</span> </span>
<span id="cb29-8"><a href="#cb29-8" tabindex="-1"></a>    <span class="fu">left_join</span>(gene_info, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&#39;id&#39;</span> <span class="ot">=</span> <span class="st">&#39;id&#39;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb29-9"><a href="#cb29-9" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(gene_set)) <span class="sc">%&gt;%</span> </span>
<span id="cb29-10"><a href="#cb29-10" tabindex="-1"></a>    <span class="fu">pull</span>(FDR) <span class="sc">%&gt;%</span> </span>
<span id="cb29-11"><a href="#cb29-11" tabindex="-1"></a>    min</span>
<span id="cb29-12"><a href="#cb29-12" tabindex="-1"></a>}</span>
<span id="cb29-13"><a href="#cb29-13" tabindex="-1"></a>maxFDR <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(xfdr[,<span class="dv">1</span>]) <span class="sc">%&gt;%</span> max</span>
<span id="cb29-14"><a href="#cb29-14" tabindex="-1"></a></span>
<span id="cb29-15"><a href="#cb29-15" tabindex="-1"></a>i <span class="ot">&lt;-</span> use_comp[<span class="dv">1</span>]</span>
<span id="cb29-16"><a href="#cb29-16" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> use_comp)  {</span>
<span id="cb29-17"><a href="#cb29-17" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;####&quot;</span>, mageck_comp_list_names[i], <span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb29-18"><a href="#cb29-18" tabindex="-1"></a>  </span>
<span id="cb29-19"><a href="#cb29-19" tabindex="-1"></a>  x <span class="ot">&lt;-</span> gene_summ[[i]] <span class="sc">%&gt;%</span> </span>
<span id="cb29-20"><a href="#cb29-20" tabindex="-1"></a>    <span class="fu">ReadRRA</span>(<span class="at">score =</span> <span class="st">&#39;lfc&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb29-21"><a href="#cb29-21" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">LFC =</span> Score) <span class="sc">%&gt;%</span> </span>
<span id="cb29-22"><a href="#cb29-22" tabindex="-1"></a>    <span class="fu">left_join</span>(gene_info <span class="sc">%&gt;%</span> <span class="fu">filter</span>(gene_set <span class="sc">==</span> <span class="st">&#39;Positive controls&#39;</span>), </span>
<span id="cb29-23"><a href="#cb29-23" tabindex="-1"></a>              <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&#39;id&#39;</span> <span class="ot">=</span> <span class="st">&#39;id&#39;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb29-24"><a href="#cb29-24" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(gene_set)) <span class="sc">%&gt;%</span> </span>
<span id="cb29-25"><a href="#cb29-25" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb29-26"><a href="#cb29-26" tabindex="-1"></a>      <span class="at">gene_name =</span> <span class="fu">factor</span>(gene_name, <span class="at">levels =</span> gene_levels),</span>
<span id="cb29-27"><a href="#cb29-27" tabindex="-1"></a>      <span class="at">gene_set =</span> <span class="fu">ifelse</span>(</span>
<span id="cb29-28"><a href="#cb29-28" tabindex="-1"></a>        gene_set <span class="sc">==</span> <span class="st">&#39;Immune system cytokines and chemokines&#39;</span>,</span>
<span id="cb29-29"><a href="#cb29-29" tabindex="-1"></a>        <span class="st">&#39;Immune system cytokines</span><span class="sc">\n</span><span class="st">and chemokines&#39;</span>,</span>
<span id="cb29-30"><a href="#cb29-30" tabindex="-1"></a>        gene_set</span>
<span id="cb29-31"><a href="#cb29-31" tabindex="-1"></a>      ),</span>
<span id="cb29-32"><a href="#cb29-32" tabindex="-1"></a>      <span class="at">gene_set =</span> <span class="fu">ifelse</span>(</span>
<span id="cb29-33"><a href="#cb29-33" tabindex="-1"></a>        gene_set <span class="sc">==</span> <span class="st">&#39;Interferone, TNF and danger/death-inducing signals&#39;</span>,</span>
<span id="cb29-34"><a href="#cb29-34" tabindex="-1"></a>        <span class="st">&#39;Interferon, TNF and</span><span class="sc">\n</span><span class="st">danger/death-inducing</span><span class="sc">\n</span><span class="st">signals&#39;</span>,</span>
<span id="cb29-35"><a href="#cb29-35" tabindex="-1"></a>        gene_set</span>
<span id="cb29-36"><a href="#cb29-36" tabindex="-1"></a>      ),</span>
<span id="cb29-37"><a href="#cb29-37" tabindex="-1"></a>      <span class="at">gene_set =</span> <span class="fu">factor</span>(gene_set, <span class="fu">names</span>(use_colors))</span>
<span id="cb29-38"><a href="#cb29-38" tabindex="-1"></a>      )</span>
<span id="cb29-39"><a href="#cb29-39" tabindex="-1"></a>  xlim <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fu">ceiling</span>(<span class="fu">max</span>(<span class="fu">abs</span>(x<span class="sc">$</span>LFC), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)), </span>
<span id="cb29-40"><a href="#cb29-40" tabindex="-1"></a>            <span class="fu">ceiling</span>(<span class="fu">max</span>(<span class="fu">abs</span>(x<span class="sc">$</span>LFC), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)))</span>
<span id="cb29-41"><a href="#cb29-41" tabindex="-1"></a>  size_breaks <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.25</span>, <span class="fl">0.5</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="fu">ceiling</span>(maxFDR))</span>
<span id="cb29-42"><a href="#cb29-42" tabindex="-1"></a>  res <span class="ot">&lt;-</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb29-43"><a href="#cb29-43" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(LFC, gene_name, <span class="at">size =</span> <span class="sc">-</span><span class="fu">log10</span>(FDR))) <span class="sc">+</span></span>
<span id="cb29-44"><a href="#cb29-44" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">1</span>, <span class="at">linewidth =</span> default_linewidth) <span class="sc">+</span></span>
<span id="cb29-45"><a href="#cb29-45" tabindex="-1"></a>    </span>
<span id="cb29-46"><a href="#cb29-46" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="dv">3</span>, <span class="at">linewidth =</span> one_pt<span class="sc">/</span><span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb29-47"><a href="#cb29-47" tabindex="-1"></a>    <span class="fu">scale_y_discrete</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">add =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>))) <span class="sc">+</span> <span class="co"># add fixed units around each facet to avoid lines over points</span></span>
<span id="cb29-48"><a href="#cb29-48" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> use_colors) <span class="sc">+</span></span>
<span id="cb29-49"><a href="#cb29-49" tabindex="-1"></a>    <span class="fu">scale_size</span>(<span class="at">range =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">4</span>), </span>
<span id="cb29-50"><a href="#cb29-50" tabindex="-1"></a>               <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fu">ceiling</span>(maxFDR)), </span>
<span id="cb29-51"><a href="#cb29-51" tabindex="-1"></a>               <span class="at">breaks =</span> size_breaks,</span>
<span id="cb29-52"><a href="#cb29-52" tabindex="-1"></a>               <span class="at">labels =</span> size_breaks <span class="sc">%&gt;%</span> as.character <span class="co"># remove trailing 0 from decimals</span></span>
<span id="cb29-53"><a href="#cb29-53" tabindex="-1"></a>               ) <span class="sc">+</span></span>
<span id="cb29-54"><a href="#cb29-54" tabindex="-1"></a>    <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> use_shape) <span class="sc">+</span></span>
<span id="cb29-55"><a href="#cb29-55" tabindex="-1"></a>    <span class="fu">scale_x_continuous</span>(</span>
<span id="cb29-56"><a href="#cb29-56" tabindex="-1"></a>        <span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">add =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>)),</span>
<span id="cb29-57"><a href="#cb29-57" tabindex="-1"></a>        <span class="at">limits =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">4.5</span>, <span class="fl">4.5</span>),</span>
<span id="cb29-58"><a href="#cb29-58" tabindex="-1"></a>        <span class="at">breaks =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">4</span>, <span class="dv">0</span>, <span class="dv">4</span>),</span>
<span id="cb29-59"><a href="#cb29-59" tabindex="-1"></a>        ) <span class="sc">+</span></span>
<span id="cb29-60"><a href="#cb29-60" tabindex="-1"></a>    <span class="fu">facet_grid</span>(<span class="at">rows =</span> <span class="fu">vars</span>(gene_set), <span class="at">scales =</span> <span class="st">&#39;free_y&#39;</span>, <span class="at">space =</span> <span class="st">&#39;free_y&#39;</span>) <span class="sc">+</span></span>
<span id="cb29-61"><a href="#cb29-61" tabindex="-1"></a>    theme_facet <span class="sc">+</span></span>
<span id="cb29-62"><a href="#cb29-62" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb29-63"><a href="#cb29-63" tabindex="-1"></a>      <span class="at">axis.ticks =</span> <span class="fu">element_line</span>(<span class="at">linewidth =</span> one_pt<span class="sc">/</span><span class="dv">2</span>, <span class="at">color =</span> <span class="st">&#39;black&#39;</span>),</span>
<span id="cb29-64"><a href="#cb29-64" tabindex="-1"></a>      <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">linewidth =</span> one_pt<span class="sc">/</span><span class="dv">2</span>, <span class="at">color =</span> <span class="st">&#39;black&#39;</span>),</span>
<span id="cb29-65"><a href="#cb29-65" tabindex="-1"></a>      <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">8</span>), </span>
<span id="cb29-66"><a href="#cb29-66" tabindex="-1"></a>      <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>),</span>
<span id="cb29-67"><a href="#cb29-67" tabindex="-1"></a>      <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">8</span>),</span>
<span id="cb29-68"><a href="#cb29-68" tabindex="-1"></a>      <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">8</span>),</span>
<span id="cb29-69"><a href="#cb29-69" tabindex="-1"></a>      <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">4</span>)</span>
<span id="cb29-70"><a href="#cb29-70" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb29-71"><a href="#cb29-71" tabindex="-1"></a>    <span class="fu">labs</span>(</span>
<span id="cb29-72"><a href="#cb29-72" tabindex="-1"></a>      <span class="at">x =</span> <span class="fu">expression</span>(<span class="fu">paste</span>(<span class="st">&quot;lo&quot;</span>, g[<span class="dv">2</span>],<span class="st">&quot;(Fold change)&quot;</span>)),</span>
<span id="cb29-73"><a href="#cb29-73" tabindex="-1"></a>      <span class="at">y =</span> <span class="cn">NULL</span>,</span>
<span id="cb29-74"><a href="#cb29-74" tabindex="-1"></a>      <span class="at">size =</span> <span class="fu">expression</span>(<span class="fu">paste</span>(<span class="st">&quot;-lo&quot;</span>, g[<span class="dv">10</span>],<span class="st">&quot;(FDR)&quot;</span>)),</span>
<span id="cb29-75"><a href="#cb29-75" tabindex="-1"></a>      <span class="at">title =</span> </span>
<span id="cb29-76"><a href="#cb29-76" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb29-77"><a href="#cb29-77" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">color =</span> <span class="st">&#39;none&#39;</span>, <span class="at">shape =</span> <span class="st">&#39;none&#39;</span>)</span>
<span id="cb29-78"><a href="#cb29-78" tabindex="-1"></a>  </span>
<span id="cb29-79"><a href="#cb29-79" tabindex="-1"></a>  <span class="fu">print</span>(res)</span>
<span id="cb29-80"><a href="#cb29-80" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&#39;</span><span class="sc">\n\n</span><span class="st">&#39;</span>)</span>
<span id="cb29-81"><a href="#cb29-81" tabindex="-1"></a>}</span></code></pre></div>
<div id="stealthy-ko-over-dcas9-in-vitro" class="section level4">
<h4>StealTHY KO over dCas9 In vitro</h4>
<p><img src="figure/crispr-mm_2215_sgRNA-StealTHY.Rmd/da-positive-controls-genes-dotplot-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-da-positive-controls-genes-dotplot-1">
Past versions of da-positive-controls-genes-dotplot-1.png
</button>
</p>
<div id="fig-da-positive-controls-genes-dotplot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/32aca023d6d8209f1aa3df3f177fc383c7569844/docs/figure/crispr-mm_2215_sgRNA-StealTHY.Rmd/da-positive-controls-genes-dotplot-1.png" target="_blank">32aca02</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-07-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="stealthy-ko-over-dcas9-in-vivo" class="section level4">
<h4>StealTHY KO over dCas9 In vivo</h4>
<p><img src="figure/crispr-mm_2215_sgRNA-StealTHY.Rmd/da-positive-controls-genes-dotplot-2.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-da-positive-controls-genes-dotplot-2">
Past versions of da-positive-controls-genes-dotplot-2.png
</button>
</p>
<div id="fig-da-positive-controls-genes-dotplot-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/32aca023d6d8209f1aa3df3f177fc383c7569844/docs/figure/crispr-mm_2215_sgRNA-StealTHY.Rmd/da-positive-controls-genes-dotplot-2.png" target="_blank">32aca02</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-07-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="non-expressed-hits" class="section level2">
<h2>Non-expressed hits</h2>
<ul>
<li><p>Definition of <strong>non-expressed genes</strong> : We keep
genes with ≥ 1 read count in 1/3 of the samples for each site (whole
tumor or whole lung). In this case, if only 3 samples for each site are
available, the gene must have at least 1 read count in 1 sample. Then,
the final list of non-expressed genes is defined as the intersection of
non-expressed genes in whole tumor and whole lung.</p></li>
<li><p>Definition of <strong>hits</strong> : Hits are defined as genes
with a log2 fold-change ≤ -2 or ≥ 2.</p></li>
</ul>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a><span class="co"># Color palette</span></span>
<span id="cb30-2"><a href="#cb30-2" tabindex="-1"></a>use_color_palette <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb30-3"><a href="#cb30-3" tabindex="-1"></a>  <span class="st">`</span><span class="at">StealTHY</span><span class="st">`</span>  <span class="ot">=</span> <span class="st">&#39;#F7EC13&#39;</span>,</span>
<span id="cb30-4"><a href="#cb30-4" tabindex="-1"></a>  <span class="st">`</span><span class="at">Puro</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;black&#39;</span>,</span>
<span id="cb30-5"><a href="#cb30-5" tabindex="-1"></a>  <span class="st">`</span><span class="at">Cas9+Puro</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;black&#39;</span>,</span>
<span id="cb30-6"><a href="#cb30-6" tabindex="-1"></a>  <span class="st">`</span><span class="at">StealTHY NSG</span><span class="st">`</span>  <span class="ot">=</span> <span class="st">&#39;#58595B&#39;</span></span>
<span id="cb30-7"><a href="#cb30-7" tabindex="-1"></a>)</span></code></pre></div>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" tabindex="-1"></a>min_counts <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb31-2"><a href="#cb31-2" tabindex="-1"></a>min_prop <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">3</span></span>
<span id="cb31-3"><a href="#cb31-3" tabindex="-1"></a></span>
<span id="cb31-4"><a href="#cb31-4" tabindex="-1"></a><span class="co"># Read RNA-seq data</span></span>
<span id="cb31-5"><a href="#cb31-5" tabindex="-1"></a>se_rna <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(params<span class="sc">$</span>se_rnaseq_facs)</span>
<span id="cb31-6"><a href="#cb31-6" tabindex="-1"></a></span>
<span id="cb31-7"><a href="#cb31-7" tabindex="-1"></a><span class="co"># Calculate CPM</span></span>
<span id="cb31-8"><a href="#cb31-8" tabindex="-1"></a><span class="fu">assay</span>(se_rna, <span class="st">&#39;cpm&#39;</span>) <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">cpm</span>(se_rna)</span>
<span id="cb31-9"><a href="#cb31-9" tabindex="-1"></a></span>
<span id="cb31-10"><a href="#cb31-10" tabindex="-1"></a><span class="co"># CRISPR library genes</span></span>
<span id="cb31-11"><a href="#cb31-11" tabindex="-1"></a>crispr_genes <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">rowData</span>(se)<span class="sc">$</span>Gene) <span class="sc">%&gt;%</span> <span class="fu">grep</span>(<span class="st">&#39;Non_Target&#39;</span>, ., <span class="at">invert =</span> T, <span class="at">value =</span> T)</span>
<span id="cb31-12"><a href="#cb31-12" tabindex="-1"></a></span>
<span id="cb31-13"><a href="#cb31-13" tabindex="-1"></a><span class="co"># Non-expressed genes</span></span>
<span id="cb31-14"><a href="#cb31-14" tabindex="-1"></a>non_expr_genes <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb31-15"><a href="#cb31-15" tabindex="-1"></a>expr_genes <span class="ot">&lt;-</span> <span class="fu">list</span>()</span></code></pre></div>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" tabindex="-1"></a>use_samples <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;4T1_In_vivo_Thy1&#39;</span>, <span class="st">&#39;MVT1_In_vivo_Thy1&#39;</span>, <span class="st">&#39;Py2T_In_vivo_Thy1&#39;</span>)</span>
<span id="cb32-2"><a href="#cb32-2" tabindex="-1"></a></span>
<span id="cb32-3"><a href="#cb32-3" tabindex="-1"></a><span class="co"># RNA-seq : Select genes and samples</span></span>
<span id="cb32-4"><a href="#cb32-4" tabindex="-1"></a>use_se <span class="ot">&lt;-</span> se_rna[<span class="fu">rowData</span>(se_rna)<span class="sc">$</span>gene_name <span class="sc">%in%</span> crispr_genes, use_samples]</span>
<span id="cb32-5"><a href="#cb32-5" tabindex="-1"></a></span>
<span id="cb32-6"><a href="#cb32-6" tabindex="-1"></a><span class="co"># Change rownames</span></span>
<span id="cb32-7"><a href="#cb32-7" tabindex="-1"></a><span class="fu">rownames</span>(use_se) <span class="ot">&lt;-</span> <span class="fu">rowData</span>(use_se)<span class="sc">$</span>gene_name</span>
<span id="cb32-8"><a href="#cb32-8" tabindex="-1"></a></span>
<span id="cb32-9"><a href="#cb32-9" tabindex="-1"></a><span class="co"># Proportion of samples with min_counts at each group</span></span>
<span id="cb32-10"><a href="#cb32-10" tabindex="-1"></a>use_group <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(use_se<span class="sc">$</span>donor)</span>
<span id="cb32-11"><a href="#cb32-11" tabindex="-1"></a>counts_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(use_se, <span class="st">&#39;counts&#39;</span>)</span>
<span id="cb32-12"><a href="#cb32-12" tabindex="-1"></a>counts_stats <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i=</span><span class="fu">levels</span>(use_group), <span class="at">.combine =</span> cbind) <span class="sc">%do%</span> {</span>
<span id="cb32-13"><a href="#cb32-13" tabindex="-1"></a>  use_cols <span class="ot">&lt;-</span> use_se<span class="sc">$</span>donor <span class="sc">==</span> i</span>
<span id="cb32-14"><a href="#cb32-14" tabindex="-1"></a>  <span class="fu">rowSums</span>(counts_mat[,use_cols, <span class="at">drop =</span> <span class="cn">FALSE</span>] <span class="sc">&gt;=</span> min_counts)  <span class="sc">/</span> <span class="fu">sum</span>(use_cols)</span>
<span id="cb32-15"><a href="#cb32-15" tabindex="-1"></a>} <span class="sc">%&gt;%</span> data.frame</span>
<span id="cb32-16"><a href="#cb32-16" tabindex="-1"></a><span class="fu">colnames</span>(counts_stats) <span class="ot">&lt;-</span> <span class="fu">levels</span>(use_group)</span>
<span id="cb32-17"><a href="#cb32-17" tabindex="-1"></a>counts_stats<span class="sc">$</span>all <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(counts_mat <span class="sc">&gt;=</span> min_counts)  <span class="sc">/</span> <span class="fu">ncol</span>(counts_mat)</span>
<span id="cb32-18"><a href="#cb32-18" tabindex="-1"></a>  </span>
<span id="cb32-19"><a href="#cb32-19" tabindex="-1"></a><span class="co"># Keep genes with min_counts threshold in &gt;=min_prop of samples in at least 1 group</span></span>
<span id="cb32-20"><a href="#cb32-20" tabindex="-1"></a>rows_to_keep <span class="ot">&lt;-</span> <span class="fu">rownames</span>(counts_stats)[counts_stats<span class="sc">$</span>all <span class="sc">&gt;=</span> min_prop]</span>
<span id="cb32-21"><a href="#cb32-21" tabindex="-1"></a>rows_to_remove <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(<span class="fu">rownames</span>(counts_stats), rows_to_keep)</span>
<span id="cb32-22"><a href="#cb32-22" tabindex="-1"></a></span>
<span id="cb32-23"><a href="#cb32-23" tabindex="-1"></a>expr_genes[[<span class="st">&#39;whole_tumor&#39;</span>]] <span class="ot">&lt;-</span> rows_to_keep</span>
<span id="cb32-24"><a href="#cb32-24" tabindex="-1"></a>non_expr_genes[[<span class="st">&#39;whole_tumor&#39;</span>]] <span class="ot">&lt;-</span> rows_to_remove</span></code></pre></div>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a>use_samples <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;4T1_In_vivo_Thy1_Lungs&#39;</span>, <span class="st">&#39;MVT1_In_vivo_Thy1_Lungs&#39;</span>, <span class="st">&#39;Py2T_In_vivo_Thy1_Lungs&#39;</span>)</span>
<span id="cb33-2"><a href="#cb33-2" tabindex="-1"></a></span>
<span id="cb33-3"><a href="#cb33-3" tabindex="-1"></a><span class="co"># RNA-seq : Select genes and samples</span></span>
<span id="cb33-4"><a href="#cb33-4" tabindex="-1"></a>use_se <span class="ot">&lt;-</span> se_rna[<span class="fu">rowData</span>(se_rna)<span class="sc">$</span>gene_name <span class="sc">%in%</span> crispr_genes, use_samples]</span>
<span id="cb33-5"><a href="#cb33-5" tabindex="-1"></a></span>
<span id="cb33-6"><a href="#cb33-6" tabindex="-1"></a><span class="co"># Change rownames</span></span>
<span id="cb33-7"><a href="#cb33-7" tabindex="-1"></a><span class="fu">rownames</span>(use_se) <span class="ot">&lt;-</span> <span class="fu">rowData</span>(use_se)<span class="sc">$</span>gene_name</span>
<span id="cb33-8"><a href="#cb33-8" tabindex="-1"></a></span>
<span id="cb33-9"><a href="#cb33-9" tabindex="-1"></a><span class="co"># Proportion of samples with min_counts at each group</span></span>
<span id="cb33-10"><a href="#cb33-10" tabindex="-1"></a>use_group <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(use_se<span class="sc">$</span>donor)</span>
<span id="cb33-11"><a href="#cb33-11" tabindex="-1"></a>counts_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(use_se, <span class="st">&#39;counts&#39;</span>)</span>
<span id="cb33-12"><a href="#cb33-12" tabindex="-1"></a>counts_stats <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i=</span><span class="fu">levels</span>(use_group), <span class="at">.combine =</span> cbind) <span class="sc">%do%</span> {</span>
<span id="cb33-13"><a href="#cb33-13" tabindex="-1"></a>  use_cols <span class="ot">&lt;-</span> use_se<span class="sc">$</span>donor <span class="sc">==</span> i</span>
<span id="cb33-14"><a href="#cb33-14" tabindex="-1"></a>  <span class="fu">rowSums</span>(counts_mat[,use_cols, <span class="at">drop =</span> <span class="cn">FALSE</span>] <span class="sc">&gt;=</span> min_counts)  <span class="sc">/</span> <span class="fu">sum</span>(use_cols)</span>
<span id="cb33-15"><a href="#cb33-15" tabindex="-1"></a>} <span class="sc">%&gt;%</span> data.frame</span>
<span id="cb33-16"><a href="#cb33-16" tabindex="-1"></a><span class="fu">colnames</span>(counts_stats) <span class="ot">&lt;-</span> <span class="fu">levels</span>(use_group)</span>
<span id="cb33-17"><a href="#cb33-17" tabindex="-1"></a>counts_stats<span class="sc">$</span>all <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(counts_mat <span class="sc">&gt;=</span> min_counts)  <span class="sc">/</span> <span class="fu">ncol</span>(counts_mat)</span>
<span id="cb33-18"><a href="#cb33-18" tabindex="-1"></a>  </span>
<span id="cb33-19"><a href="#cb33-19" tabindex="-1"></a><span class="co"># Keep genes with min_counts threshold in &gt;=min_prop of samples in at least 1 group</span></span>
<span id="cb33-20"><a href="#cb33-20" tabindex="-1"></a>rows_to_keep <span class="ot">&lt;-</span> <span class="fu">rownames</span>(counts_stats)[counts_stats<span class="sc">$</span>all <span class="sc">&gt;=</span> min_prop]</span>
<span id="cb33-21"><a href="#cb33-21" tabindex="-1"></a>rows_to_remove <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(<span class="fu">rownames</span>(counts_stats), rows_to_keep)</span>
<span id="cb33-22"><a href="#cb33-22" tabindex="-1"></a></span>
<span id="cb33-23"><a href="#cb33-23" tabindex="-1"></a>expr_genes[[<span class="st">&#39;whole_lung&#39;</span>]] <span class="ot">&lt;-</span> rows_to_keep</span>
<span id="cb33-24"><a href="#cb33-24" tabindex="-1"></a>non_expr_genes[[<span class="st">&#39;whole_lung&#39;</span>]] <span class="ot">&lt;-</span> rows_to_remove</span></code></pre></div>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" tabindex="-1"></a>use_annot <span class="ot">&lt;-</span> <span class="fu">colData</span>(se)</span>
<span id="cb34-2"><a href="#cb34-2" tabindex="-1"></a></span>
<span id="cb34-3"><a href="#cb34-3" tabindex="-1"></a>expr_genes_combined <span class="ot">&lt;-</span> expr_genes</span>
<span id="cb34-4"><a href="#cb34-4" tabindex="-1"></a>non_expr_genes_combined <span class="ot">&lt;-</span> non_expr_genes</span>
<span id="cb34-5"><a href="#cb34-5" tabindex="-1"></a></span>
<span id="cb34-6"><a href="#cb34-6" tabindex="-1"></a>use_comp <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;StealTHY&quot;</span>, <span class="st">&quot;Puro&quot;</span>, <span class="st">&quot;Cas9+Puro&quot;</span>, <span class="st">&quot;StealTHY NSG&quot;</span>)</span>
<span id="cb34-7"><a href="#cb34-7" tabindex="-1"></a>i <span class="ot">&lt;-</span> <span class="st">&#39;StealTHY&#39;</span></span>
<span id="cb34-8"><a href="#cb34-8" tabindex="-1"></a>hits_stats <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i =</span> use_comp, <span class="at">.combine =</span> rbind) <span class="sc">%do%</span> {</span>
<span id="cb34-9"><a href="#cb34-9" tabindex="-1"></a>  i_names <span class="ot">&lt;-</span> i <span class="sc">%&gt;%</span> <span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>, <span class="st">&quot;_&quot;</span>, .) <span class="sc">%&gt;%</span> tolower</span>
<span id="cb34-10"><a href="#cb34-10" tabindex="-1"></a>  st1 <span class="ot">&lt;-</span> <span class="fu">unique</span>(use_annot[ne_mageck_comp_list[[i]][[<span class="dv">1</span>]],]<span class="sc">$</span>sample_type)</span>
<span id="cb34-11"><a href="#cb34-11" tabindex="-1"></a>  st2 <span class="ot">&lt;-</span> <span class="fu">unique</span>(use_annot[ne_mageck_comp_list[[i]][[<span class="dv">2</span>]],]<span class="sc">$</span>sample_type)</span>
<span id="cb34-12"><a href="#cb34-12" tabindex="-1"></a>  </span>
<span id="cb34-13"><a href="#cb34-13" tabindex="-1"></a>  genes_expr <span class="ot">&lt;-</span> <span class="fu">intersect</span>(expr_genes[[st1]], expr_genes[[st2]])</span>
<span id="cb34-14"><a href="#cb34-14" tabindex="-1"></a> </span>
<span id="cb34-15"><a href="#cb34-15" tabindex="-1"></a>  da_res <span class="ot">&lt;-</span> ne_gene_summ[[i_names]]  <span class="sc">%&gt;%</span>  </span>
<span id="cb34-16"><a href="#cb34-16" tabindex="-1"></a>    <span class="fu">ReadRRA</span>(<span class="at">score =</span> <span class="st">&#39;lfc&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb34-17"><a href="#cb34-17" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">LFC =</span> Score) <span class="sc">%&gt;%</span> </span>
<span id="cb34-18"><a href="#cb34-18" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb34-19"><a href="#cb34-19" tabindex="-1"></a>      <span class="at">expressed =</span> id <span class="sc">%in%</span> genes_expr,</span>
<span id="cb34-20"><a href="#cb34-20" tabindex="-1"></a>      <span class="at">hit =</span> (<span class="fu">abs</span>(LFC) <span class="sc">&gt;=</span> <span class="dv">2</span>)</span>
<span id="cb34-21"><a href="#cb34-21" tabindex="-1"></a>      )</span>
<span id="cb34-22"><a href="#cb34-22" tabindex="-1"></a>  </span>
<span id="cb34-23"><a href="#cb34-23" tabindex="-1"></a>   prop_non_expr_genes <span class="ot">&lt;-</span> (<span class="fu">nrow</span>(da_res)<span class="sc">-</span><span class="fu">length</span>(genes_expr)) <span class="sc">/</span> <span class="fu">nrow</span>(da_res)</span>
<span id="cb34-24"><a href="#cb34-24" tabindex="-1"></a>   prop_non_expr_hits <span class="ot">&lt;-</span> <span class="fu">sum</span>(da_res<span class="sc">$</span>hit <span class="sc">&amp;</span> <span class="sc">!</span>da_res<span class="sc">$</span>expressed) <span class="sc">/</span> <span class="fu">sum</span>(da_res<span class="sc">$</span>hit)</span>
<span id="cb34-25"><a href="#cb34-25" tabindex="-1"></a>   <span class="fu">data.frame</span>(</span>
<span id="cb34-26"><a href="#cb34-26" tabindex="-1"></a>     <span class="at">comparison =</span> i,</span>
<span id="cb34-27"><a href="#cb34-27" tabindex="-1"></a>     <span class="at">total_genes =</span> <span class="fu">nrow</span>(da_res),</span>
<span id="cb34-28"><a href="#cb34-28" tabindex="-1"></a>     <span class="at">n_nonexpr_genes =</span> <span class="fu">nrow</span>(da_res)<span class="sc">-</span><span class="fu">length</span>(genes_expr),</span>
<span id="cb34-29"><a href="#cb34-29" tabindex="-1"></a>     <span class="at">prop_non_expr_genes =</span> (<span class="fu">nrow</span>(da_res)<span class="sc">-</span><span class="fu">length</span>(genes_expr)) <span class="sc">/</span> <span class="fu">nrow</span>(da_res),</span>
<span id="cb34-30"><a href="#cb34-30" tabindex="-1"></a>     <span class="at">n_hits =</span> <span class="fu">sum</span>(da_res<span class="sc">$</span>hit),</span>
<span id="cb34-31"><a href="#cb34-31" tabindex="-1"></a>     <span class="at">non_expr_hits =</span> <span class="fu">sum</span>(da_res<span class="sc">$</span>hit <span class="sc">&amp;</span> <span class="sc">!</span>da_res<span class="sc">$</span>expressed),</span>
<span id="cb34-32"><a href="#cb34-32" tabindex="-1"></a>     <span class="at">n_nohits =</span> <span class="fu">sum</span>(<span class="sc">!</span>da_res<span class="sc">$</span>hit)</span>
<span id="cb34-33"><a href="#cb34-33" tabindex="-1"></a>   ) <span class="sc">%&gt;%</span> </span>
<span id="cb34-34"><a href="#cb34-34" tabindex="-1"></a>     <span class="fu">mutate</span>(</span>
<span id="cb34-35"><a href="#cb34-35" tabindex="-1"></a>       <span class="at">prop_non_expr_hits =</span> non_expr_hits <span class="sc">/</span> n_hits,</span>
<span id="cb34-36"><a href="#cb34-36" tabindex="-1"></a>       <span class="at">lfc =</span> <span class="fu">log2</span>(prop_non_expr_hits <span class="sc">/</span> prop_non_expr_genes)</span>
<span id="cb34-37"><a href="#cb34-37" tabindex="-1"></a>     )</span>
<span id="cb34-38"><a href="#cb34-38" tabindex="-1"></a>  </span>
<span id="cb34-39"><a href="#cb34-39" tabindex="-1"></a>}</span>
<span id="cb34-40"><a href="#cb34-40" tabindex="-1"></a></span>
<span id="cb34-41"><a href="#cb34-41" tabindex="-1"></a>hits_stats <span class="ot">&lt;-</span> hits_stats <span class="sc">%&gt;%</span> </span>
<span id="cb34-42"><a href="#cb34-42" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb34-43"><a href="#cb34-43" tabindex="-1"></a>    <span class="at">comparison =</span> <span class="fu">factor</span>(comparison,<span class="at">levels =</span> <span class="fu">names</span>(ne_mageck_comp_list) <span class="sc">%&gt;%</span> rev)</span>
<span id="cb34-44"><a href="#cb34-44" tabindex="-1"></a>  )</span>
<span id="cb34-45"><a href="#cb34-45" tabindex="-1"></a></span>
<span id="cb34-46"><a href="#cb34-46" tabindex="-1"></a>hits_stats_global <span class="ot">&lt;-</span> hits_stats</span></code></pre></div>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" tabindex="-1"></a>min_counts <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb35-2"><a href="#cb35-2" tabindex="-1"></a>min_prop <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">3</span></span>
<span id="cb35-3"><a href="#cb35-3" tabindex="-1"></a>use_donor <span class="ot">&lt;-</span> <span class="st">&#39;4T1&#39;</span></span>
<span id="cb35-4"><a href="#cb35-4" tabindex="-1"></a>whole_tumor_sample <span class="ot">&lt;-</span> <span class="st">&#39;4T1_In_vivo_Thy1&#39;</span></span>
<span id="cb35-5"><a href="#cb35-5" tabindex="-1"></a>whole_lung_sample <span class="ot">&lt;-</span> <span class="st">&#39;4T1_In_vivo_Thy1_Lungs&#39;</span></span>
<span id="cb35-6"><a href="#cb35-6" tabindex="-1"></a></span>
<span id="cb35-7"><a href="#cb35-7" tabindex="-1"></a><span class="co"># Read RNA-seq data</span></span>
<span id="cb35-8"><a href="#cb35-8" tabindex="-1"></a>se_rna <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(params<span class="sc">$</span>se_rnaseq_facs)</span>
<span id="cb35-9"><a href="#cb35-9" tabindex="-1"></a></span>
<span id="cb35-10"><a href="#cb35-10" tabindex="-1"></a><span class="co"># Calculate CPM</span></span>
<span id="cb35-11"><a href="#cb35-11" tabindex="-1"></a><span class="fu">assay</span>(se_rna, <span class="st">&#39;cpm&#39;</span>) <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">cpm</span>(se_rna)</span>
<span id="cb35-12"><a href="#cb35-12" tabindex="-1"></a></span>
<span id="cb35-13"><a href="#cb35-13" tabindex="-1"></a><span class="co"># CRISPR library genes</span></span>
<span id="cb35-14"><a href="#cb35-14" tabindex="-1"></a>crispr_genes <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">rowData</span>(se)<span class="sc">$</span>Gene) <span class="sc">%&gt;%</span> <span class="fu">grep</span>(<span class="st">&#39;Non_Target&#39;</span>, ., <span class="at">invert =</span> T, <span class="at">value =</span> T)</span>
<span id="cb35-15"><a href="#cb35-15" tabindex="-1"></a></span>
<span id="cb35-16"><a href="#cb35-16" tabindex="-1"></a><span class="co"># Non-expressed genes</span></span>
<span id="cb35-17"><a href="#cb35-17" tabindex="-1"></a>non_expr_genes <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb35-18"><a href="#cb35-18" tabindex="-1"></a>expr_genes <span class="ot">&lt;-</span> <span class="fu">list</span>()</span></code></pre></div>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" tabindex="-1"></a>use_samples <span class="ot">&lt;-</span> whole_tumor_sample</span>
<span id="cb36-2"><a href="#cb36-2" tabindex="-1"></a></span>
<span id="cb36-3"><a href="#cb36-3" tabindex="-1"></a><span class="co"># RNA-seq : Select genes and samples</span></span>
<span id="cb36-4"><a href="#cb36-4" tabindex="-1"></a>use_se <span class="ot">&lt;-</span> se_rna[<span class="fu">rowData</span>(se_rna)<span class="sc">$</span>gene_name <span class="sc">%in%</span> crispr_genes, use_samples]</span>
<span id="cb36-5"><a href="#cb36-5" tabindex="-1"></a></span>
<span id="cb36-6"><a href="#cb36-6" tabindex="-1"></a><span class="co"># Change rownames</span></span>
<span id="cb36-7"><a href="#cb36-7" tabindex="-1"></a><span class="fu">rownames</span>(use_se) <span class="ot">&lt;-</span><span class="fu">make.names</span>(<span class="fu">rowData</span>(use_se)<span class="sc">$</span>gene_name, <span class="at">unique =</span> <span class="cn">TRUE</span>)</span>
<span id="cb36-8"><a href="#cb36-8" tabindex="-1"></a></span>
<span id="cb36-9"><a href="#cb36-9" tabindex="-1"></a><span class="co"># Proportion of samples with min_counts at each group</span></span>
<span id="cb36-10"><a href="#cb36-10" tabindex="-1"></a>use_group <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(use_se<span class="sc">$</span>donor)</span>
<span id="cb36-11"><a href="#cb36-11" tabindex="-1"></a>counts_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(use_se, <span class="st">&#39;counts&#39;</span>)</span>
<span id="cb36-12"><a href="#cb36-12" tabindex="-1"></a>counts_stats <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i=</span><span class="fu">levels</span>(use_group), <span class="at">.combine =</span> cbind) <span class="sc">%do%</span> {</span>
<span id="cb36-13"><a href="#cb36-13" tabindex="-1"></a>  use_cols <span class="ot">&lt;-</span> use_se<span class="sc">$</span>donor <span class="sc">==</span> i</span>
<span id="cb36-14"><a href="#cb36-14" tabindex="-1"></a>  <span class="fu">rowSums</span>(counts_mat[,use_cols, <span class="at">drop =</span> <span class="cn">FALSE</span>] <span class="sc">&gt;=</span> min_counts)  <span class="sc">/</span> <span class="fu">sum</span>(use_cols)</span>
<span id="cb36-15"><a href="#cb36-15" tabindex="-1"></a>} <span class="sc">%&gt;%</span> data.frame</span>
<span id="cb36-16"><a href="#cb36-16" tabindex="-1"></a><span class="fu">colnames</span>(counts_stats) <span class="ot">&lt;-</span> <span class="fu">levels</span>(use_group)</span>
<span id="cb36-17"><a href="#cb36-17" tabindex="-1"></a>counts_stats<span class="sc">$</span>all <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(counts_mat <span class="sc">&gt;=</span> min_counts)  <span class="sc">/</span> <span class="fu">ncol</span>(counts_mat)</span>
<span id="cb36-18"><a href="#cb36-18" tabindex="-1"></a>  </span>
<span id="cb36-19"><a href="#cb36-19" tabindex="-1"></a><span class="co"># Keep genes with min_counts threshold in &gt;=min_prop of samples in at least 1 group</span></span>
<span id="cb36-20"><a href="#cb36-20" tabindex="-1"></a>rows_to_keep <span class="ot">&lt;-</span> <span class="fu">rownames</span>(counts_stats)[counts_stats<span class="sc">$</span>all <span class="sc">&gt;=</span> min_prop]</span>
<span id="cb36-21"><a href="#cb36-21" tabindex="-1"></a>rows_to_remove <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(<span class="fu">rownames</span>(counts_stats), rows_to_keep)</span>
<span id="cb36-22"><a href="#cb36-22" tabindex="-1"></a></span>
<span id="cb36-23"><a href="#cb36-23" tabindex="-1"></a>expr_genes[[<span class="st">&#39;whole_tumor&#39;</span>]] <span class="ot">&lt;-</span> rows_to_keep</span>
<span id="cb36-24"><a href="#cb36-24" tabindex="-1"></a>non_expr_genes[[<span class="st">&#39;whole_tumor&#39;</span>]] <span class="ot">&lt;-</span> rows_to_remove</span></code></pre></div>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" tabindex="-1"></a>use_samples <span class="ot">&lt;-</span> whole_lung_sample</span>
<span id="cb37-2"><a href="#cb37-2" tabindex="-1"></a></span>
<span id="cb37-3"><a href="#cb37-3" tabindex="-1"></a><span class="co"># RNA-seq : Select genes and samples</span></span>
<span id="cb37-4"><a href="#cb37-4" tabindex="-1"></a>use_se <span class="ot">&lt;-</span> se_rna[<span class="fu">rowData</span>(se_rna)<span class="sc">$</span>gene_name <span class="sc">%in%</span> crispr_genes, use_samples]</span>
<span id="cb37-5"><a href="#cb37-5" tabindex="-1"></a></span>
<span id="cb37-6"><a href="#cb37-6" tabindex="-1"></a><span class="co"># Change rownames</span></span>
<span id="cb37-7"><a href="#cb37-7" tabindex="-1"></a><span class="fu">rownames</span>(use_se) <span class="ot">&lt;-</span><span class="fu">make.names</span>(<span class="fu">rowData</span>(use_se)<span class="sc">$</span>gene_name, <span class="at">unique =</span> <span class="cn">TRUE</span>)</span>
<span id="cb37-8"><a href="#cb37-8" tabindex="-1"></a></span>
<span id="cb37-9"><a href="#cb37-9" tabindex="-1"></a><span class="co"># Proportion of samples with min_counts at each group</span></span>
<span id="cb37-10"><a href="#cb37-10" tabindex="-1"></a>use_group <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(use_se<span class="sc">$</span>donor)</span>
<span id="cb37-11"><a href="#cb37-11" tabindex="-1"></a>counts_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(use_se, <span class="st">&#39;counts&#39;</span>)</span>
<span id="cb37-12"><a href="#cb37-12" tabindex="-1"></a>counts_stats <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i=</span><span class="fu">levels</span>(use_group), <span class="at">.combine =</span> cbind) <span class="sc">%do%</span> {</span>
<span id="cb37-13"><a href="#cb37-13" tabindex="-1"></a>  use_cols <span class="ot">&lt;-</span> use_se<span class="sc">$</span>donor <span class="sc">==</span> i</span>
<span id="cb37-14"><a href="#cb37-14" tabindex="-1"></a>  <span class="fu">rowSums</span>(counts_mat[,use_cols, <span class="at">drop =</span> <span class="cn">FALSE</span>] <span class="sc">&gt;=</span> min_counts)  <span class="sc">/</span> <span class="fu">sum</span>(use_cols)</span>
<span id="cb37-15"><a href="#cb37-15" tabindex="-1"></a>} <span class="sc">%&gt;%</span> data.frame</span>
<span id="cb37-16"><a href="#cb37-16" tabindex="-1"></a><span class="fu">colnames</span>(counts_stats) <span class="ot">&lt;-</span> <span class="fu">levels</span>(use_group)</span>
<span id="cb37-17"><a href="#cb37-17" tabindex="-1"></a>counts_stats<span class="sc">$</span>all <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(counts_mat <span class="sc">&gt;=</span> min_counts)  <span class="sc">/</span> <span class="fu">ncol</span>(counts_mat)</span>
<span id="cb37-18"><a href="#cb37-18" tabindex="-1"></a>  </span>
<span id="cb37-19"><a href="#cb37-19" tabindex="-1"></a><span class="co"># Keep genes with min_counts threshold in &gt;=min_prop of samples in at least 1 group</span></span>
<span id="cb37-20"><a href="#cb37-20" tabindex="-1"></a>rows_to_keep <span class="ot">&lt;-</span> <span class="fu">rownames</span>(counts_stats)[counts_stats<span class="sc">$</span>all <span class="sc">&gt;=</span> min_prop]</span>
<span id="cb37-21"><a href="#cb37-21" tabindex="-1"></a>rows_to_remove <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(<span class="fu">rownames</span>(counts_stats), rows_to_keep)</span>
<span id="cb37-22"><a href="#cb37-22" tabindex="-1"></a></span>
<span id="cb37-23"><a href="#cb37-23" tabindex="-1"></a>expr_genes[[<span class="st">&#39;whole_lung&#39;</span>]] <span class="ot">&lt;-</span> rows_to_keep</span>
<span id="cb37-24"><a href="#cb37-24" tabindex="-1"></a>non_expr_genes[[<span class="st">&#39;whole_lung&#39;</span>]] <span class="ot">&lt;-</span> rows_to_remove</span></code></pre></div>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" tabindex="-1"></a>use_annot <span class="ot">&lt;-</span> <span class="fu">colData</span>(se)</span>
<span id="cb38-2"><a href="#cb38-2" tabindex="-1"></a></span>
<span id="cb38-3"><a href="#cb38-3" tabindex="-1"></a>use_comp <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;StealTHY 4T1&quot;</span>, <span class="st">&quot;Puro 4T1&quot;</span>, <span class="st">&quot;Cas9+Puro 4T1&quot;</span>, <span class="st">&quot;StealTHY NSG 4T1&quot;</span>)</span>
<span id="cb38-4"><a href="#cb38-4" tabindex="-1"></a></span>
<span id="cb38-5"><a href="#cb38-5" tabindex="-1"></a>i <span class="ot">&lt;-</span> use_comp[<span class="dv">1</span>]</span>
<span id="cb38-6"><a href="#cb38-6" tabindex="-1"></a>hits_stats <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i =</span> use_comp, <span class="at">.combine =</span> rbind) <span class="sc">%do%</span> {</span>
<span id="cb38-7"><a href="#cb38-7" tabindex="-1"></a>  i_names <span class="ot">&lt;-</span> i <span class="sc">%&gt;%</span> <span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>, <span class="st">&quot;_&quot;</span>, .) <span class="sc">%&gt;%</span> tolower</span>
<span id="cb38-8"><a href="#cb38-8" tabindex="-1"></a>  st1 <span class="ot">&lt;-</span> <span class="fu">unique</span>(use_annot[ne_mageck_comp_list[[i]][[<span class="dv">1</span>]],]<span class="sc">$</span>sample_type)</span>
<span id="cb38-9"><a href="#cb38-9" tabindex="-1"></a>  st2 <span class="ot">&lt;-</span> <span class="fu">unique</span>(use_annot[ne_mageck_comp_list[[i]][[<span class="dv">2</span>]],]<span class="sc">$</span>sample_type)</span>
<span id="cb38-10"><a href="#cb38-10" tabindex="-1"></a>  </span>
<span id="cb38-11"><a href="#cb38-11" tabindex="-1"></a>  genes_expr <span class="ot">&lt;-</span> <span class="fu">intersect</span>(expr_genes[[st1]], expr_genes[[st2]])</span>
<span id="cb38-12"><a href="#cb38-12" tabindex="-1"></a> </span>
<span id="cb38-13"><a href="#cb38-13" tabindex="-1"></a>  da_res <span class="ot">&lt;-</span> ne_gene_summ[[i_names]]  <span class="sc">%&gt;%</span>  </span>
<span id="cb38-14"><a href="#cb38-14" tabindex="-1"></a>    <span class="fu">ReadRRA</span>(<span class="at">score =</span> <span class="st">&#39;lfc&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb38-15"><a href="#cb38-15" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">LFC =</span> Score) <span class="sc">%&gt;%</span> </span>
<span id="cb38-16"><a href="#cb38-16" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb38-17"><a href="#cb38-17" tabindex="-1"></a>      <span class="at">expressed =</span> id <span class="sc">%in%</span> genes_expr,</span>
<span id="cb38-18"><a href="#cb38-18" tabindex="-1"></a>      <span class="at">hit =</span> (<span class="fu">abs</span>(LFC) <span class="sc">&gt;=</span> <span class="dv">2</span>)</span>
<span id="cb38-19"><a href="#cb38-19" tabindex="-1"></a>      )</span>
<span id="cb38-20"><a href="#cb38-20" tabindex="-1"></a>  </span>
<span id="cb38-21"><a href="#cb38-21" tabindex="-1"></a>   prop_non_expr_genes <span class="ot">&lt;-</span> (<span class="fu">nrow</span>(da_res)<span class="sc">-</span><span class="fu">length</span>(genes_expr)) <span class="sc">/</span> <span class="fu">nrow</span>(da_res)</span>
<span id="cb38-22"><a href="#cb38-22" tabindex="-1"></a>   prop_non_expr_hits <span class="ot">&lt;-</span> <span class="fu">sum</span>(da_res<span class="sc">$</span>hit <span class="sc">&amp;</span> <span class="sc">!</span>da_res<span class="sc">$</span>expressed) <span class="sc">/</span> <span class="fu">sum</span>(da_res<span class="sc">$</span>hit)</span>
<span id="cb38-23"><a href="#cb38-23" tabindex="-1"></a>   <span class="fu">data.frame</span>(</span>
<span id="cb38-24"><a href="#cb38-24" tabindex="-1"></a>     <span class="at">comparison =</span> i,</span>
<span id="cb38-25"><a href="#cb38-25" tabindex="-1"></a>     <span class="at">total_genes =</span> <span class="fu">nrow</span>(da_res),</span>
<span id="cb38-26"><a href="#cb38-26" tabindex="-1"></a>     <span class="at">n_nonexpr_genes =</span> <span class="fu">nrow</span>(da_res)<span class="sc">-</span><span class="fu">length</span>(genes_expr),</span>
<span id="cb38-27"><a href="#cb38-27" tabindex="-1"></a>     <span class="at">prop_non_expr_genes =</span> (<span class="fu">nrow</span>(da_res)<span class="sc">-</span><span class="fu">length</span>(genes_expr)) <span class="sc">/</span> <span class="fu">nrow</span>(da_res),</span>
<span id="cb38-28"><a href="#cb38-28" tabindex="-1"></a>     <span class="at">n_hits =</span> <span class="fu">sum</span>(da_res<span class="sc">$</span>hit),</span>
<span id="cb38-29"><a href="#cb38-29" tabindex="-1"></a>     <span class="at">non_expr_hits =</span> <span class="fu">sum</span>(da_res<span class="sc">$</span>hit <span class="sc">&amp;</span> <span class="sc">!</span>da_res<span class="sc">$</span>expressed),</span>
<span id="cb38-30"><a href="#cb38-30" tabindex="-1"></a>     <span class="at">n_nohits =</span> <span class="fu">sum</span>(<span class="sc">!</span>da_res<span class="sc">$</span>hit)</span>
<span id="cb38-31"><a href="#cb38-31" tabindex="-1"></a>   ) <span class="sc">%&gt;%</span> </span>
<span id="cb38-32"><a href="#cb38-32" tabindex="-1"></a>     <span class="fu">mutate</span>(</span>
<span id="cb38-33"><a href="#cb38-33" tabindex="-1"></a>       <span class="at">prop_non_expr_hits =</span> non_expr_hits <span class="sc">/</span> n_hits,</span>
<span id="cb38-34"><a href="#cb38-34" tabindex="-1"></a>       <span class="at">prop_non_expr_hits_over_non_expr =</span> non_expr_hits <span class="sc">/</span> prop_non_expr_genes,</span>
<span id="cb38-35"><a href="#cb38-35" tabindex="-1"></a>       <span class="at">lfc =</span> <span class="fu">log2</span>(prop_non_expr_hits <span class="sc">/</span> prop_non_expr_genes)</span>
<span id="cb38-36"><a href="#cb38-36" tabindex="-1"></a>     )</span>
<span id="cb38-37"><a href="#cb38-37" tabindex="-1"></a>  </span>
<span id="cb38-38"><a href="#cb38-38" tabindex="-1"></a>}</span>
<span id="cb38-39"><a href="#cb38-39" tabindex="-1"></a></span>
<span id="cb38-40"><a href="#cb38-40" tabindex="-1"></a>hits_stats <span class="ot">&lt;-</span> hits_stats <span class="sc">%&gt;%</span> </span>
<span id="cb38-41"><a href="#cb38-41" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb38-42"><a href="#cb38-42" tabindex="-1"></a>    <span class="at">comparison =</span> <span class="fu">factor</span>(comparison,<span class="at">levels =</span> <span class="fu">names</span>(ne_mageck_comp_list) <span class="sc">%&gt;%</span> rev)</span>
<span id="cb38-43"><a href="#cb38-43" tabindex="-1"></a>  )</span>
<span id="cb38-44"><a href="#cb38-44" tabindex="-1"></a></span>
<span id="cb38-45"><a href="#cb38-45" tabindex="-1"></a>hits_stats_4t1 <span class="ot">&lt;-</span> hits_stats <span class="sc">%&gt;%</span> </span>
<span id="cb38-46"><a href="#cb38-46" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb38-47"><a href="#cb38-47" tabindex="-1"></a>    <span class="at">donor =</span> <span class="st">&#39;4T1&#39;</span>,</span>
<span id="cb38-48"><a href="#cb38-48" tabindex="-1"></a>    <span class="at">comparison =</span> <span class="fu">gsub</span>(<span class="st">&quot; 4T1&quot;</span>, <span class="st">&quot;&quot;</span>, comparison)</span>
<span id="cb38-49"><a href="#cb38-49" tabindex="-1"></a>    )</span></code></pre></div>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" tabindex="-1"></a>use_annot <span class="ot">&lt;-</span> <span class="fu">colData</span>(se)</span>
<span id="cb39-2"><a href="#cb39-2" tabindex="-1"></a></span>
<span id="cb39-3"><a href="#cb39-3" tabindex="-1"></a>use_comp <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;StealTHY 4T1&quot;</span>, <span class="st">&quot;Puro 4T1&quot;</span>, <span class="st">&quot;Cas9+Puro 4T1&quot;</span>, <span class="st">&quot;StealTHY NSG 4T1&quot;</span>)</span>
<span id="cb39-4"><a href="#cb39-4" tabindex="-1"></a></span>
<span id="cb39-5"><a href="#cb39-5" tabindex="-1"></a>i <span class="ot">&lt;-</span> use_comp[<span class="dv">1</span>]</span>
<span id="cb39-6"><a href="#cb39-6" tabindex="-1"></a>hits_stats <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i =</span> use_comp, <span class="at">.combine =</span> rbind) <span class="sc">%do%</span> {</span>
<span id="cb39-7"><a href="#cb39-7" tabindex="-1"></a>  i_names <span class="ot">&lt;-</span> i <span class="sc">%&gt;%</span> <span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>, <span class="st">&quot;_&quot;</span>, .) <span class="sc">%&gt;%</span> tolower</span>
<span id="cb39-8"><a href="#cb39-8" tabindex="-1"></a>  st1 <span class="ot">&lt;-</span> <span class="fu">unique</span>(use_annot[ne_mageck_comp_list[[i]][[<span class="dv">1</span>]],]<span class="sc">$</span>sample_type)</span>
<span id="cb39-9"><a href="#cb39-9" tabindex="-1"></a>  st2 <span class="ot">&lt;-</span> <span class="fu">unique</span>(use_annot[ne_mageck_comp_list[[i]][[<span class="dv">2</span>]],]<span class="sc">$</span>sample_type)</span>
<span id="cb39-10"><a href="#cb39-10" tabindex="-1"></a>  </span>
<span id="cb39-11"><a href="#cb39-11" tabindex="-1"></a>  genes_expr <span class="ot">&lt;-</span> <span class="fu">intersect</span>(expr_genes_combined[[st1]], expr_genes_combined[[st2]])</span>
<span id="cb39-12"><a href="#cb39-12" tabindex="-1"></a> </span>
<span id="cb39-13"><a href="#cb39-13" tabindex="-1"></a>  da_res <span class="ot">&lt;-</span> ne_gene_summ[[i_names]]  <span class="sc">%&gt;%</span>  </span>
<span id="cb39-14"><a href="#cb39-14" tabindex="-1"></a>    <span class="fu">ReadRRA</span>(<span class="at">score =</span> <span class="st">&#39;lfc&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb39-15"><a href="#cb39-15" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">LFC =</span> Score) <span class="sc">%&gt;%</span> </span>
<span id="cb39-16"><a href="#cb39-16" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb39-17"><a href="#cb39-17" tabindex="-1"></a>      <span class="at">expressed =</span> id <span class="sc">%in%</span> genes_expr,</span>
<span id="cb39-18"><a href="#cb39-18" tabindex="-1"></a>      <span class="at">hit =</span> (<span class="fu">abs</span>(LFC) <span class="sc">&gt;=</span> <span class="dv">2</span>)</span>
<span id="cb39-19"><a href="#cb39-19" tabindex="-1"></a>      )</span>
<span id="cb39-20"><a href="#cb39-20" tabindex="-1"></a>  </span>
<span id="cb39-21"><a href="#cb39-21" tabindex="-1"></a>   prop_non_expr_genes <span class="ot">&lt;-</span> (<span class="fu">nrow</span>(da_res)<span class="sc">-</span><span class="fu">length</span>(genes_expr)) <span class="sc">/</span> <span class="fu">nrow</span>(da_res)</span>
<span id="cb39-22"><a href="#cb39-22" tabindex="-1"></a>   prop_non_expr_hits <span class="ot">&lt;-</span> <span class="fu">sum</span>(da_res<span class="sc">$</span>hit <span class="sc">&amp;</span> <span class="sc">!</span>da_res<span class="sc">$</span>expressed) <span class="sc">/</span> <span class="fu">sum</span>(da_res<span class="sc">$</span>hit)</span>
<span id="cb39-23"><a href="#cb39-23" tabindex="-1"></a>   <span class="fu">data.frame</span>(</span>
<span id="cb39-24"><a href="#cb39-24" tabindex="-1"></a>     <span class="at">comparison =</span> i,</span>
<span id="cb39-25"><a href="#cb39-25" tabindex="-1"></a>     <span class="at">total_genes =</span> <span class="fu">nrow</span>(da_res),</span>
<span id="cb39-26"><a href="#cb39-26" tabindex="-1"></a>     <span class="at">n_nonexpr_genes =</span> <span class="fu">nrow</span>(da_res)<span class="sc">-</span><span class="fu">length</span>(genes_expr),</span>
<span id="cb39-27"><a href="#cb39-27" tabindex="-1"></a>     <span class="at">prop_non_expr_genes =</span> (<span class="fu">nrow</span>(da_res)<span class="sc">-</span><span class="fu">length</span>(genes_expr)) <span class="sc">/</span> <span class="fu">nrow</span>(da_res),</span>
<span id="cb39-28"><a href="#cb39-28" tabindex="-1"></a>     <span class="at">n_hits =</span> <span class="fu">sum</span>(da_res<span class="sc">$</span>hit),</span>
<span id="cb39-29"><a href="#cb39-29" tabindex="-1"></a>     <span class="at">non_expr_hits =</span> <span class="fu">sum</span>(da_res<span class="sc">$</span>hit <span class="sc">&amp;</span> <span class="sc">!</span>da_res<span class="sc">$</span>expressed),</span>
<span id="cb39-30"><a href="#cb39-30" tabindex="-1"></a>     <span class="at">n_nohits =</span> <span class="fu">sum</span>(<span class="sc">!</span>da_res<span class="sc">$</span>hit)</span>
<span id="cb39-31"><a href="#cb39-31" tabindex="-1"></a>   ) <span class="sc">%&gt;%</span> </span>
<span id="cb39-32"><a href="#cb39-32" tabindex="-1"></a>     <span class="fu">mutate</span>(</span>
<span id="cb39-33"><a href="#cb39-33" tabindex="-1"></a>       <span class="at">prop_non_expr_hits =</span> non_expr_hits <span class="sc">/</span> n_hits,</span>
<span id="cb39-34"><a href="#cb39-34" tabindex="-1"></a>       <span class="at">prop_non_expr_hits_over_non_expr =</span> non_expr_hits <span class="sc">/</span> prop_non_expr_genes,</span>
<span id="cb39-35"><a href="#cb39-35" tabindex="-1"></a>       <span class="at">lfc =</span> <span class="fu">log2</span>(prop_non_expr_hits <span class="sc">/</span> prop_non_expr_genes)</span>
<span id="cb39-36"><a href="#cb39-36" tabindex="-1"></a>     )</span>
<span id="cb39-37"><a href="#cb39-37" tabindex="-1"></a>  </span>
<span id="cb39-38"><a href="#cb39-38" tabindex="-1"></a>}</span>
<span id="cb39-39"><a href="#cb39-39" tabindex="-1"></a></span>
<span id="cb39-40"><a href="#cb39-40" tabindex="-1"></a>hits_stats <span class="ot">&lt;-</span> hits_stats <span class="sc">%&gt;%</span> </span>
<span id="cb39-41"><a href="#cb39-41" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb39-42"><a href="#cb39-42" tabindex="-1"></a>    <span class="at">comparison =</span> <span class="fu">factor</span>(comparison,<span class="at">levels =</span> <span class="fu">names</span>(ne_mageck_comp_list) <span class="sc">%&gt;%</span> rev)</span>
<span id="cb39-43"><a href="#cb39-43" tabindex="-1"></a>  )</span>
<span id="cb39-44"><a href="#cb39-44" tabindex="-1"></a></span>
<span id="cb39-45"><a href="#cb39-45" tabindex="-1"></a>hits_stats_4t1_comb <span class="ot">&lt;-</span> hits_stats <span class="sc">%&gt;%</span> </span>
<span id="cb39-46"><a href="#cb39-46" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb39-47"><a href="#cb39-47" tabindex="-1"></a>    <span class="at">donor =</span> <span class="st">&#39;4T1&#39;</span>,</span>
<span id="cb39-48"><a href="#cb39-48" tabindex="-1"></a>    <span class="at">comparison =</span> <span class="fu">gsub</span>(<span class="st">&quot; 4T1&quot;</span>, <span class="st">&quot;&quot;</span>, comparison)</span>
<span id="cb39-49"><a href="#cb39-49" tabindex="-1"></a>    )</span></code></pre></div>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" tabindex="-1"></a>min_counts <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb40-2"><a href="#cb40-2" tabindex="-1"></a>min_prop <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">3</span></span>
<span id="cb40-3"><a href="#cb40-3" tabindex="-1"></a>use_donor <span class="ot">&lt;-</span> <span class="st">&#39;mvt1&#39;</span></span>
<span id="cb40-4"><a href="#cb40-4" tabindex="-1"></a>whole_tumor_sample <span class="ot">&lt;-</span> <span class="st">&#39;MVT1_In_vivo_Thy1&#39;</span></span>
<span id="cb40-5"><a href="#cb40-5" tabindex="-1"></a>whole_lung_sample <span class="ot">&lt;-</span> <span class="st">&#39;MVT1_In_vivo_Thy1_Lungs&#39;</span></span>
<span id="cb40-6"><a href="#cb40-6" tabindex="-1"></a></span>
<span id="cb40-7"><a href="#cb40-7" tabindex="-1"></a><span class="co"># Read RNA-seq data</span></span>
<span id="cb40-8"><a href="#cb40-8" tabindex="-1"></a>se_rna <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(params<span class="sc">$</span>se_rnaseq_facs)</span>
<span id="cb40-9"><a href="#cb40-9" tabindex="-1"></a></span>
<span id="cb40-10"><a href="#cb40-10" tabindex="-1"></a><span class="co"># Calculate CPM</span></span>
<span id="cb40-11"><a href="#cb40-11" tabindex="-1"></a><span class="fu">assay</span>(se_rna, <span class="st">&#39;cpm&#39;</span>) <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">cpm</span>(se_rna)</span>
<span id="cb40-12"><a href="#cb40-12" tabindex="-1"></a></span>
<span id="cb40-13"><a href="#cb40-13" tabindex="-1"></a><span class="co"># CRISPR library genes</span></span>
<span id="cb40-14"><a href="#cb40-14" tabindex="-1"></a>crispr_genes <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">rowData</span>(se)<span class="sc">$</span>Gene) <span class="sc">%&gt;%</span> <span class="fu">grep</span>(<span class="st">&#39;Non_Target&#39;</span>, ., <span class="at">invert =</span> T, <span class="at">value =</span> T)</span>
<span id="cb40-15"><a href="#cb40-15" tabindex="-1"></a></span>
<span id="cb40-16"><a href="#cb40-16" tabindex="-1"></a><span class="co"># Non-expressed genes</span></span>
<span id="cb40-17"><a href="#cb40-17" tabindex="-1"></a>non_expr_genes <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb40-18"><a href="#cb40-18" tabindex="-1"></a>expr_genes <span class="ot">&lt;-</span> <span class="fu">list</span>()</span></code></pre></div>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" tabindex="-1"></a>use_samples <span class="ot">&lt;-</span> whole_tumor_sample</span>
<span id="cb41-2"><a href="#cb41-2" tabindex="-1"></a></span>
<span id="cb41-3"><a href="#cb41-3" tabindex="-1"></a><span class="co"># RNA-seq : Select genes and samples</span></span>
<span id="cb41-4"><a href="#cb41-4" tabindex="-1"></a>use_se <span class="ot">&lt;-</span> se_rna[<span class="fu">rowData</span>(se_rna)<span class="sc">$</span>gene_name <span class="sc">%in%</span> crispr_genes, use_samples]</span>
<span id="cb41-5"><a href="#cb41-5" tabindex="-1"></a></span>
<span id="cb41-6"><a href="#cb41-6" tabindex="-1"></a><span class="co"># Change rownames</span></span>
<span id="cb41-7"><a href="#cb41-7" tabindex="-1"></a><span class="fu">rownames</span>(use_se) <span class="ot">&lt;-</span><span class="fu">make.names</span>(<span class="fu">rowData</span>(use_se)<span class="sc">$</span>gene_name, <span class="at">unique =</span> <span class="cn">TRUE</span>)</span>
<span id="cb41-8"><a href="#cb41-8" tabindex="-1"></a></span>
<span id="cb41-9"><a href="#cb41-9" tabindex="-1"></a><span class="co"># Proportion of samples with min_counts at each group</span></span>
<span id="cb41-10"><a href="#cb41-10" tabindex="-1"></a>use_group <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(use_se<span class="sc">$</span>donor)</span>
<span id="cb41-11"><a href="#cb41-11" tabindex="-1"></a>counts_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(use_se, <span class="st">&#39;counts&#39;</span>)</span>
<span id="cb41-12"><a href="#cb41-12" tabindex="-1"></a>counts_stats <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i=</span><span class="fu">levels</span>(use_group), <span class="at">.combine =</span> cbind) <span class="sc">%do%</span> {</span>
<span id="cb41-13"><a href="#cb41-13" tabindex="-1"></a>  use_cols <span class="ot">&lt;-</span> use_se<span class="sc">$</span>donor <span class="sc">==</span> i</span>
<span id="cb41-14"><a href="#cb41-14" tabindex="-1"></a>  <span class="fu">rowSums</span>(counts_mat[,use_cols, <span class="at">drop =</span> <span class="cn">FALSE</span>] <span class="sc">&gt;=</span> min_counts)  <span class="sc">/</span> <span class="fu">sum</span>(use_cols)</span>
<span id="cb41-15"><a href="#cb41-15" tabindex="-1"></a>} <span class="sc">%&gt;%</span> data.frame</span>
<span id="cb41-16"><a href="#cb41-16" tabindex="-1"></a><span class="fu">colnames</span>(counts_stats) <span class="ot">&lt;-</span> <span class="fu">levels</span>(use_group)</span>
<span id="cb41-17"><a href="#cb41-17" tabindex="-1"></a>counts_stats<span class="sc">$</span>all <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(counts_mat <span class="sc">&gt;=</span> min_counts)  <span class="sc">/</span> <span class="fu">ncol</span>(counts_mat)</span>
<span id="cb41-18"><a href="#cb41-18" tabindex="-1"></a>  </span>
<span id="cb41-19"><a href="#cb41-19" tabindex="-1"></a><span class="co"># Keep genes with min_counts threshold in &gt;=min_prop of samples in at least 1 group</span></span>
<span id="cb41-20"><a href="#cb41-20" tabindex="-1"></a>rows_to_keep <span class="ot">&lt;-</span> <span class="fu">rownames</span>(counts_stats)[counts_stats<span class="sc">$</span>all <span class="sc">&gt;=</span> min_prop]</span>
<span id="cb41-21"><a href="#cb41-21" tabindex="-1"></a>rows_to_remove <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(<span class="fu">rownames</span>(counts_stats), rows_to_keep)</span>
<span id="cb41-22"><a href="#cb41-22" tabindex="-1"></a></span>
<span id="cb41-23"><a href="#cb41-23" tabindex="-1"></a>expr_genes[[<span class="st">&#39;whole_tumor&#39;</span>]] <span class="ot">&lt;-</span> rows_to_keep</span>
<span id="cb41-24"><a href="#cb41-24" tabindex="-1"></a>non_expr_genes[[<span class="st">&#39;whole_tumor&#39;</span>]] <span class="ot">&lt;-</span> rows_to_remove</span></code></pre></div>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" tabindex="-1"></a>use_samples <span class="ot">&lt;-</span> whole_lung_sample</span>
<span id="cb42-2"><a href="#cb42-2" tabindex="-1"></a></span>
<span id="cb42-3"><a href="#cb42-3" tabindex="-1"></a><span class="co"># RNA-seq : Select genes and samples</span></span>
<span id="cb42-4"><a href="#cb42-4" tabindex="-1"></a>use_se <span class="ot">&lt;-</span> se_rna[<span class="fu">rowData</span>(se_rna)<span class="sc">$</span>gene_name <span class="sc">%in%</span> crispr_genes, use_samples]</span>
<span id="cb42-5"><a href="#cb42-5" tabindex="-1"></a></span>
<span id="cb42-6"><a href="#cb42-6" tabindex="-1"></a><span class="co"># Change rownames</span></span>
<span id="cb42-7"><a href="#cb42-7" tabindex="-1"></a><span class="fu">rownames</span>(use_se) <span class="ot">&lt;-</span><span class="fu">make.names</span>(<span class="fu">rowData</span>(use_se)<span class="sc">$</span>gene_name, <span class="at">unique =</span> <span class="cn">TRUE</span>)</span>
<span id="cb42-8"><a href="#cb42-8" tabindex="-1"></a></span>
<span id="cb42-9"><a href="#cb42-9" tabindex="-1"></a><span class="co"># Proportion of samples with min_counts at each group</span></span>
<span id="cb42-10"><a href="#cb42-10" tabindex="-1"></a>use_group <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(use_se<span class="sc">$</span>donor)</span>
<span id="cb42-11"><a href="#cb42-11" tabindex="-1"></a>counts_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(use_se, <span class="st">&#39;counts&#39;</span>)</span>
<span id="cb42-12"><a href="#cb42-12" tabindex="-1"></a>counts_stats <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i=</span><span class="fu">levels</span>(use_group), <span class="at">.combine =</span> cbind) <span class="sc">%do%</span> {</span>
<span id="cb42-13"><a href="#cb42-13" tabindex="-1"></a>  use_cols <span class="ot">&lt;-</span> use_se<span class="sc">$</span>donor <span class="sc">==</span> i</span>
<span id="cb42-14"><a href="#cb42-14" tabindex="-1"></a>  <span class="fu">rowSums</span>(counts_mat[,use_cols, <span class="at">drop =</span> <span class="cn">FALSE</span>] <span class="sc">&gt;=</span> min_counts)  <span class="sc">/</span> <span class="fu">sum</span>(use_cols)</span>
<span id="cb42-15"><a href="#cb42-15" tabindex="-1"></a>} <span class="sc">%&gt;%</span> data.frame</span>
<span id="cb42-16"><a href="#cb42-16" tabindex="-1"></a><span class="fu">colnames</span>(counts_stats) <span class="ot">&lt;-</span> <span class="fu">levels</span>(use_group)</span>
<span id="cb42-17"><a href="#cb42-17" tabindex="-1"></a>counts_stats<span class="sc">$</span>all <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(counts_mat <span class="sc">&gt;=</span> min_counts)  <span class="sc">/</span> <span class="fu">ncol</span>(counts_mat)</span>
<span id="cb42-18"><a href="#cb42-18" tabindex="-1"></a>  </span>
<span id="cb42-19"><a href="#cb42-19" tabindex="-1"></a><span class="co"># Keep genes with min_counts threshold in &gt;=min_prop of samples in at least 1 group</span></span>
<span id="cb42-20"><a href="#cb42-20" tabindex="-1"></a>rows_to_keep <span class="ot">&lt;-</span> <span class="fu">rownames</span>(counts_stats)[counts_stats<span class="sc">$</span>all <span class="sc">&gt;=</span> min_prop]</span>
<span id="cb42-21"><a href="#cb42-21" tabindex="-1"></a>rows_to_remove <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(<span class="fu">rownames</span>(counts_stats), rows_to_keep)</span>
<span id="cb42-22"><a href="#cb42-22" tabindex="-1"></a></span>
<span id="cb42-23"><a href="#cb42-23" tabindex="-1"></a>expr_genes[[<span class="st">&#39;whole_lung&#39;</span>]] <span class="ot">&lt;-</span> rows_to_keep</span>
<span id="cb42-24"><a href="#cb42-24" tabindex="-1"></a>non_expr_genes[[<span class="st">&#39;whole_lung&#39;</span>]] <span class="ot">&lt;-</span> rows_to_remove</span></code></pre></div>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" tabindex="-1"></a>use_annot <span class="ot">&lt;-</span> <span class="fu">colData</span>(se)</span>
<span id="cb43-2"><a href="#cb43-2" tabindex="-1"></a></span>
<span id="cb43-3"><a href="#cb43-3" tabindex="-1"></a>use_comp <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;StealTHY MVT1&quot;</span>, <span class="st">&quot;Puro MVT1&quot;</span>, <span class="st">&quot;StealTHY NSG MVT1&quot;</span>)</span>
<span id="cb43-4"><a href="#cb43-4" tabindex="-1"></a></span>
<span id="cb43-5"><a href="#cb43-5" tabindex="-1"></a>i <span class="ot">&lt;-</span> use_comp[<span class="dv">1</span>]</span>
<span id="cb43-6"><a href="#cb43-6" tabindex="-1"></a>hits_stats <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i =</span> use_comp, <span class="at">.combine =</span> rbind) <span class="sc">%do%</span> {</span>
<span id="cb43-7"><a href="#cb43-7" tabindex="-1"></a>  i_names <span class="ot">&lt;-</span> i <span class="sc">%&gt;%</span> <span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>, <span class="st">&quot;_&quot;</span>, .) <span class="sc">%&gt;%</span> tolower</span>
<span id="cb43-8"><a href="#cb43-8" tabindex="-1"></a>  st1 <span class="ot">&lt;-</span> <span class="fu">unique</span>(use_annot[ne_mageck_comp_list[[i]][[<span class="dv">1</span>]],]<span class="sc">$</span>sample_type)</span>
<span id="cb43-9"><a href="#cb43-9" tabindex="-1"></a>  st2 <span class="ot">&lt;-</span> <span class="fu">unique</span>(use_annot[ne_mageck_comp_list[[i]][[<span class="dv">2</span>]],]<span class="sc">$</span>sample_type)</span>
<span id="cb43-10"><a href="#cb43-10" tabindex="-1"></a>  </span>
<span id="cb43-11"><a href="#cb43-11" tabindex="-1"></a>  genes_expr <span class="ot">&lt;-</span> <span class="fu">intersect</span>(expr_genes[[st1]], expr_genes[[st2]])</span>
<span id="cb43-12"><a href="#cb43-12" tabindex="-1"></a> </span>
<span id="cb43-13"><a href="#cb43-13" tabindex="-1"></a>  da_res <span class="ot">&lt;-</span> ne_gene_summ[[i_names]]  <span class="sc">%&gt;%</span>  </span>
<span id="cb43-14"><a href="#cb43-14" tabindex="-1"></a>    <span class="fu">ReadRRA</span>(<span class="at">score =</span> <span class="st">&#39;lfc&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb43-15"><a href="#cb43-15" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">LFC =</span> Score) <span class="sc">%&gt;%</span> </span>
<span id="cb43-16"><a href="#cb43-16" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb43-17"><a href="#cb43-17" tabindex="-1"></a>      <span class="at">expressed =</span> id <span class="sc">%in%</span> genes_expr,</span>
<span id="cb43-18"><a href="#cb43-18" tabindex="-1"></a>      <span class="at">hit =</span> (<span class="fu">abs</span>(LFC) <span class="sc">&gt;=</span> <span class="dv">2</span>)</span>
<span id="cb43-19"><a href="#cb43-19" tabindex="-1"></a>      )</span>
<span id="cb43-20"><a href="#cb43-20" tabindex="-1"></a>  </span>
<span id="cb43-21"><a href="#cb43-21" tabindex="-1"></a>   prop_non_expr_genes <span class="ot">&lt;-</span> (<span class="fu">nrow</span>(da_res)<span class="sc">-</span><span class="fu">length</span>(genes_expr)) <span class="sc">/</span> <span class="fu">nrow</span>(da_res)</span>
<span id="cb43-22"><a href="#cb43-22" tabindex="-1"></a>   prop_non_expr_hits <span class="ot">&lt;-</span> <span class="fu">sum</span>(da_res<span class="sc">$</span>hit <span class="sc">&amp;</span> <span class="sc">!</span>da_res<span class="sc">$</span>expressed) <span class="sc">/</span> <span class="fu">sum</span>(da_res<span class="sc">$</span>hit)</span>
<span id="cb43-23"><a href="#cb43-23" tabindex="-1"></a>   <span class="fu">data.frame</span>(</span>
<span id="cb43-24"><a href="#cb43-24" tabindex="-1"></a>     <span class="at">comparison =</span> i,</span>
<span id="cb43-25"><a href="#cb43-25" tabindex="-1"></a>     <span class="at">total_genes =</span> <span class="fu">nrow</span>(da_res),</span>
<span id="cb43-26"><a href="#cb43-26" tabindex="-1"></a>     <span class="at">n_nonexpr_genes =</span> <span class="fu">nrow</span>(da_res)<span class="sc">-</span><span class="fu">length</span>(genes_expr),</span>
<span id="cb43-27"><a href="#cb43-27" tabindex="-1"></a>     <span class="at">prop_non_expr_genes =</span> (<span class="fu">nrow</span>(da_res)<span class="sc">-</span><span class="fu">length</span>(genes_expr)) <span class="sc">/</span> <span class="fu">nrow</span>(da_res),</span>
<span id="cb43-28"><a href="#cb43-28" tabindex="-1"></a>     <span class="at">n_hits =</span> <span class="fu">sum</span>(da_res<span class="sc">$</span>hit),</span>
<span id="cb43-29"><a href="#cb43-29" tabindex="-1"></a>     <span class="at">non_expr_hits =</span> <span class="fu">sum</span>(da_res<span class="sc">$</span>hit <span class="sc">&amp;</span> <span class="sc">!</span>da_res<span class="sc">$</span>expressed),</span>
<span id="cb43-30"><a href="#cb43-30" tabindex="-1"></a>     <span class="at">n_nohits =</span> <span class="fu">sum</span>(<span class="sc">!</span>da_res<span class="sc">$</span>hit)</span>
<span id="cb43-31"><a href="#cb43-31" tabindex="-1"></a>   ) <span class="sc">%&gt;%</span> </span>
<span id="cb43-32"><a href="#cb43-32" tabindex="-1"></a>     <span class="fu">mutate</span>(</span>
<span id="cb43-33"><a href="#cb43-33" tabindex="-1"></a>       <span class="at">prop_non_expr_hits =</span> non_expr_hits <span class="sc">/</span> n_hits,</span>
<span id="cb43-34"><a href="#cb43-34" tabindex="-1"></a>       <span class="at">prop_non_expr_hits_over_non_expr =</span> non_expr_hits <span class="sc">/</span> n_nohits,</span>
<span id="cb43-35"><a href="#cb43-35" tabindex="-1"></a>       <span class="at">lfc =</span> <span class="fu">log2</span>(prop_non_expr_hits <span class="sc">/</span> prop_non_expr_genes)</span>
<span id="cb43-36"><a href="#cb43-36" tabindex="-1"></a>     )</span>
<span id="cb43-37"><a href="#cb43-37" tabindex="-1"></a>  </span>
<span id="cb43-38"><a href="#cb43-38" tabindex="-1"></a>}</span>
<span id="cb43-39"><a href="#cb43-39" tabindex="-1"></a></span>
<span id="cb43-40"><a href="#cb43-40" tabindex="-1"></a>hits_stats <span class="ot">&lt;-</span> hits_stats <span class="sc">%&gt;%</span> </span>
<span id="cb43-41"><a href="#cb43-41" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb43-42"><a href="#cb43-42" tabindex="-1"></a>    <span class="at">comparison =</span> <span class="fu">factor</span>(comparison,<span class="at">levels =</span> <span class="fu">names</span>(ne_mageck_comp_list) <span class="sc">%&gt;%</span> rev)</span>
<span id="cb43-43"><a href="#cb43-43" tabindex="-1"></a>  )</span>
<span id="cb43-44"><a href="#cb43-44" tabindex="-1"></a></span>
<span id="cb43-45"><a href="#cb43-45" tabindex="-1"></a>hits_stats_mvt1 <span class="ot">&lt;-</span> hits_stats <span class="sc">%&gt;%</span> </span>
<span id="cb43-46"><a href="#cb43-46" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb43-47"><a href="#cb43-47" tabindex="-1"></a>    <span class="at">donor =</span> <span class="st">&#39;MVT1&#39;</span>,</span>
<span id="cb43-48"><a href="#cb43-48" tabindex="-1"></a>    <span class="at">comparison =</span> <span class="fu">gsub</span>(<span class="st">&quot; MVT1&quot;</span>, <span class="st">&quot;&quot;</span>, comparison)</span>
<span id="cb43-49"><a href="#cb43-49" tabindex="-1"></a>    )</span></code></pre></div>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" tabindex="-1"></a>use_annot <span class="ot">&lt;-</span> <span class="fu">colData</span>(se)</span>
<span id="cb44-2"><a href="#cb44-2" tabindex="-1"></a></span>
<span id="cb44-3"><a href="#cb44-3" tabindex="-1"></a>use_comp <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;StealTHY MVT1&quot;</span>, <span class="st">&quot;Puro MVT1&quot;</span>, <span class="st">&quot;StealTHY NSG MVT1&quot;</span>)</span>
<span id="cb44-4"><a href="#cb44-4" tabindex="-1"></a></span>
<span id="cb44-5"><a href="#cb44-5" tabindex="-1"></a>i <span class="ot">&lt;-</span> use_comp[<span class="dv">1</span>]</span>
<span id="cb44-6"><a href="#cb44-6" tabindex="-1"></a>hits_stats <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i =</span> use_comp, <span class="at">.combine =</span> rbind) <span class="sc">%do%</span> {</span>
<span id="cb44-7"><a href="#cb44-7" tabindex="-1"></a>  i_names <span class="ot">&lt;-</span> i <span class="sc">%&gt;%</span> <span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>, <span class="st">&quot;_&quot;</span>, .) <span class="sc">%&gt;%</span> tolower</span>
<span id="cb44-8"><a href="#cb44-8" tabindex="-1"></a>  st1 <span class="ot">&lt;-</span> <span class="fu">unique</span>(use_annot[ne_mageck_comp_list[[i]][[<span class="dv">1</span>]],]<span class="sc">$</span>sample_type)</span>
<span id="cb44-9"><a href="#cb44-9" tabindex="-1"></a>  st2 <span class="ot">&lt;-</span> <span class="fu">unique</span>(use_annot[ne_mageck_comp_list[[i]][[<span class="dv">2</span>]],]<span class="sc">$</span>sample_type)</span>
<span id="cb44-10"><a href="#cb44-10" tabindex="-1"></a>  </span>
<span id="cb44-11"><a href="#cb44-11" tabindex="-1"></a>  genes_expr <span class="ot">&lt;-</span> <span class="fu">intersect</span>(expr_genes_combined[[st1]], expr_genes_combined[[st2]])</span>
<span id="cb44-12"><a href="#cb44-12" tabindex="-1"></a> </span>
<span id="cb44-13"><a href="#cb44-13" tabindex="-1"></a>  da_res <span class="ot">&lt;-</span> ne_gene_summ[[i_names]]  <span class="sc">%&gt;%</span>  </span>
<span id="cb44-14"><a href="#cb44-14" tabindex="-1"></a>    <span class="fu">ReadRRA</span>(<span class="at">score =</span> <span class="st">&#39;lfc&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb44-15"><a href="#cb44-15" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">LFC =</span> Score) <span class="sc">%&gt;%</span> </span>
<span id="cb44-16"><a href="#cb44-16" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb44-17"><a href="#cb44-17" tabindex="-1"></a>      <span class="at">expressed =</span> id <span class="sc">%in%</span> genes_expr,</span>
<span id="cb44-18"><a href="#cb44-18" tabindex="-1"></a>      <span class="at">hit =</span> (<span class="fu">abs</span>(LFC) <span class="sc">&gt;=</span> <span class="dv">2</span>)</span>
<span id="cb44-19"><a href="#cb44-19" tabindex="-1"></a>      )</span>
<span id="cb44-20"><a href="#cb44-20" tabindex="-1"></a>  </span>
<span id="cb44-21"><a href="#cb44-21" tabindex="-1"></a>   prop_non_expr_genes <span class="ot">&lt;-</span> (<span class="fu">nrow</span>(da_res)<span class="sc">-</span><span class="fu">length</span>(genes_expr)) <span class="sc">/</span> <span class="fu">nrow</span>(da_res)</span>
<span id="cb44-22"><a href="#cb44-22" tabindex="-1"></a>   prop_non_expr_hits <span class="ot">&lt;-</span> <span class="fu">sum</span>(da_res<span class="sc">$</span>hit <span class="sc">&amp;</span> <span class="sc">!</span>da_res<span class="sc">$</span>expressed) <span class="sc">/</span> <span class="fu">sum</span>(da_res<span class="sc">$</span>hit)</span>
<span id="cb44-23"><a href="#cb44-23" tabindex="-1"></a>   <span class="fu">data.frame</span>(</span>
<span id="cb44-24"><a href="#cb44-24" tabindex="-1"></a>     <span class="at">comparison =</span> i,</span>
<span id="cb44-25"><a href="#cb44-25" tabindex="-1"></a>     <span class="at">total_genes =</span> <span class="fu">nrow</span>(da_res),</span>
<span id="cb44-26"><a href="#cb44-26" tabindex="-1"></a>     <span class="at">n_nonexpr_genes =</span> <span class="fu">nrow</span>(da_res)<span class="sc">-</span><span class="fu">length</span>(genes_expr),</span>
<span id="cb44-27"><a href="#cb44-27" tabindex="-1"></a>     <span class="at">prop_non_expr_genes =</span> (<span class="fu">nrow</span>(da_res)<span class="sc">-</span><span class="fu">length</span>(genes_expr)) <span class="sc">/</span> <span class="fu">nrow</span>(da_res),</span>
<span id="cb44-28"><a href="#cb44-28" tabindex="-1"></a>     <span class="at">n_hits =</span> <span class="fu">sum</span>(da_res<span class="sc">$</span>hit),</span>
<span id="cb44-29"><a href="#cb44-29" tabindex="-1"></a>     <span class="at">non_expr_hits =</span> <span class="fu">sum</span>(da_res<span class="sc">$</span>hit <span class="sc">&amp;</span> <span class="sc">!</span>da_res<span class="sc">$</span>expressed),</span>
<span id="cb44-30"><a href="#cb44-30" tabindex="-1"></a>     <span class="at">n_nohits =</span> <span class="fu">sum</span>(<span class="sc">!</span>da_res<span class="sc">$</span>hit)</span>
<span id="cb44-31"><a href="#cb44-31" tabindex="-1"></a>   ) <span class="sc">%&gt;%</span> </span>
<span id="cb44-32"><a href="#cb44-32" tabindex="-1"></a>     <span class="fu">mutate</span>(</span>
<span id="cb44-33"><a href="#cb44-33" tabindex="-1"></a>       <span class="at">prop_non_expr_hits =</span> non_expr_hits <span class="sc">/</span> n_hits,</span>
<span id="cb44-34"><a href="#cb44-34" tabindex="-1"></a>       <span class="at">prop_non_expr_hits_over_non_expr =</span> non_expr_hits <span class="sc">/</span> n_nohits,</span>
<span id="cb44-35"><a href="#cb44-35" tabindex="-1"></a>       <span class="at">lfc =</span> <span class="fu">log2</span>(prop_non_expr_hits <span class="sc">/</span> prop_non_expr_genes)</span>
<span id="cb44-36"><a href="#cb44-36" tabindex="-1"></a>     )</span>
<span id="cb44-37"><a href="#cb44-37" tabindex="-1"></a>  </span>
<span id="cb44-38"><a href="#cb44-38" tabindex="-1"></a>}</span>
<span id="cb44-39"><a href="#cb44-39" tabindex="-1"></a></span>
<span id="cb44-40"><a href="#cb44-40" tabindex="-1"></a>hits_stats <span class="ot">&lt;-</span> hits_stats <span class="sc">%&gt;%</span> </span>
<span id="cb44-41"><a href="#cb44-41" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb44-42"><a href="#cb44-42" tabindex="-1"></a>    <span class="at">comparison =</span> <span class="fu">factor</span>(comparison,<span class="at">levels =</span> <span class="fu">names</span>(ne_mageck_comp_list) <span class="sc">%&gt;%</span> rev)</span>
<span id="cb44-43"><a href="#cb44-43" tabindex="-1"></a>  )</span>
<span id="cb44-44"><a href="#cb44-44" tabindex="-1"></a></span>
<span id="cb44-45"><a href="#cb44-45" tabindex="-1"></a>hits_stats_mvt1_comb <span class="ot">&lt;-</span> hits_stats <span class="sc">%&gt;%</span> </span>
<span id="cb44-46"><a href="#cb44-46" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb44-47"><a href="#cb44-47" tabindex="-1"></a>    <span class="at">donor =</span> <span class="st">&#39;MVT1&#39;</span>,</span>
<span id="cb44-48"><a href="#cb44-48" tabindex="-1"></a>    <span class="at">comparison =</span> <span class="fu">gsub</span>(<span class="st">&quot; MVT1&quot;</span>, <span class="st">&quot;&quot;</span>, comparison)</span>
<span id="cb44-49"><a href="#cb44-49" tabindex="-1"></a>    )</span></code></pre></div>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" tabindex="-1"></a>min_counts <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb45-2"><a href="#cb45-2" tabindex="-1"></a>min_prop <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">3</span></span>
<span id="cb45-3"><a href="#cb45-3" tabindex="-1"></a>use_donor <span class="ot">&lt;-</span> <span class="st">&#39;py2t&#39;</span></span>
<span id="cb45-4"><a href="#cb45-4" tabindex="-1"></a>whole_tumor_sample <span class="ot">&lt;-</span> <span class="st">&#39;Py2T_In_vivo_Thy1&#39;</span></span>
<span id="cb45-5"><a href="#cb45-5" tabindex="-1"></a>whole_lung_sample <span class="ot">&lt;-</span> <span class="st">&#39;Py2T_In_vivo_Thy1_Lungs&#39;</span></span>
<span id="cb45-6"><a href="#cb45-6" tabindex="-1"></a></span>
<span id="cb45-7"><a href="#cb45-7" tabindex="-1"></a><span class="co"># Read RNA-seq data</span></span>
<span id="cb45-8"><a href="#cb45-8" tabindex="-1"></a>se_rna <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(params<span class="sc">$</span>se_rnaseq_facs)</span>
<span id="cb45-9"><a href="#cb45-9" tabindex="-1"></a></span>
<span id="cb45-10"><a href="#cb45-10" tabindex="-1"></a><span class="co"># Calculate CPM</span></span>
<span id="cb45-11"><a href="#cb45-11" tabindex="-1"></a><span class="fu">assay</span>(se_rna, <span class="st">&#39;cpm&#39;</span>) <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">cpm</span>(se_rna)</span>
<span id="cb45-12"><a href="#cb45-12" tabindex="-1"></a></span>
<span id="cb45-13"><a href="#cb45-13" tabindex="-1"></a><span class="co"># CRISPR library genes</span></span>
<span id="cb45-14"><a href="#cb45-14" tabindex="-1"></a>crispr_genes <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">rowData</span>(se)<span class="sc">$</span>Gene) <span class="sc">%&gt;%</span> <span class="fu">grep</span>(<span class="st">&#39;Non_Target&#39;</span>, ., <span class="at">invert =</span> T, <span class="at">value =</span> T)</span>
<span id="cb45-15"><a href="#cb45-15" tabindex="-1"></a></span>
<span id="cb45-16"><a href="#cb45-16" tabindex="-1"></a><span class="co"># Non-expressed genes</span></span>
<span id="cb45-17"><a href="#cb45-17" tabindex="-1"></a>non_expr_genes <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb45-18"><a href="#cb45-18" tabindex="-1"></a>expr_genes <span class="ot">&lt;-</span> <span class="fu">list</span>()</span></code></pre></div>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" tabindex="-1"></a>use_samples <span class="ot">&lt;-</span> whole_tumor_sample</span>
<span id="cb46-2"><a href="#cb46-2" tabindex="-1"></a></span>
<span id="cb46-3"><a href="#cb46-3" tabindex="-1"></a><span class="co"># RNA-seq : Select genes and samples</span></span>
<span id="cb46-4"><a href="#cb46-4" tabindex="-1"></a>use_se <span class="ot">&lt;-</span> se_rna[<span class="fu">rowData</span>(se_rna)<span class="sc">$</span>gene_name <span class="sc">%in%</span> crispr_genes, use_samples]</span>
<span id="cb46-5"><a href="#cb46-5" tabindex="-1"></a></span>
<span id="cb46-6"><a href="#cb46-6" tabindex="-1"></a><span class="co"># Change rownames</span></span>
<span id="cb46-7"><a href="#cb46-7" tabindex="-1"></a><span class="fu">rownames</span>(use_se) <span class="ot">&lt;-</span><span class="fu">make.names</span>(<span class="fu">rowData</span>(use_se)<span class="sc">$</span>gene_name, <span class="at">unique =</span> <span class="cn">TRUE</span>)</span>
<span id="cb46-8"><a href="#cb46-8" tabindex="-1"></a></span>
<span id="cb46-9"><a href="#cb46-9" tabindex="-1"></a><span class="co"># Proportion of samples with min_counts at each group</span></span>
<span id="cb46-10"><a href="#cb46-10" tabindex="-1"></a>use_group <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(use_se<span class="sc">$</span>donor)</span>
<span id="cb46-11"><a href="#cb46-11" tabindex="-1"></a>counts_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(use_se, <span class="st">&#39;counts&#39;</span>)</span>
<span id="cb46-12"><a href="#cb46-12" tabindex="-1"></a>counts_stats <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i=</span><span class="fu">levels</span>(use_group), <span class="at">.combine =</span> cbind) <span class="sc">%do%</span> {</span>
<span id="cb46-13"><a href="#cb46-13" tabindex="-1"></a>  use_cols <span class="ot">&lt;-</span> use_se<span class="sc">$</span>donor <span class="sc">==</span> i</span>
<span id="cb46-14"><a href="#cb46-14" tabindex="-1"></a>  <span class="fu">rowSums</span>(counts_mat[,use_cols, <span class="at">drop =</span> <span class="cn">FALSE</span>] <span class="sc">&gt;=</span> min_counts)  <span class="sc">/</span> <span class="fu">sum</span>(use_cols)</span>
<span id="cb46-15"><a href="#cb46-15" tabindex="-1"></a>} <span class="sc">%&gt;%</span> data.frame</span>
<span id="cb46-16"><a href="#cb46-16" tabindex="-1"></a><span class="fu">colnames</span>(counts_stats) <span class="ot">&lt;-</span> <span class="fu">levels</span>(use_group)</span>
<span id="cb46-17"><a href="#cb46-17" tabindex="-1"></a>counts_stats<span class="sc">$</span>all <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(counts_mat <span class="sc">&gt;=</span> min_counts)  <span class="sc">/</span> <span class="fu">ncol</span>(counts_mat)</span>
<span id="cb46-18"><a href="#cb46-18" tabindex="-1"></a>  </span>
<span id="cb46-19"><a href="#cb46-19" tabindex="-1"></a><span class="co"># Keep genes with min_counts threshold in &gt;=min_prop of samples in at least 1 group</span></span>
<span id="cb46-20"><a href="#cb46-20" tabindex="-1"></a>rows_to_keep <span class="ot">&lt;-</span> <span class="fu">rownames</span>(counts_stats)[counts_stats<span class="sc">$</span>all <span class="sc">&gt;=</span> min_prop]</span>
<span id="cb46-21"><a href="#cb46-21" tabindex="-1"></a>rows_to_remove <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(<span class="fu">rownames</span>(counts_stats), rows_to_keep)</span>
<span id="cb46-22"><a href="#cb46-22" tabindex="-1"></a></span>
<span id="cb46-23"><a href="#cb46-23" tabindex="-1"></a>expr_genes[[<span class="st">&#39;whole_tumor&#39;</span>]] <span class="ot">&lt;-</span> rows_to_keep</span>
<span id="cb46-24"><a href="#cb46-24" tabindex="-1"></a>non_expr_genes[[<span class="st">&#39;whole_tumor&#39;</span>]] <span class="ot">&lt;-</span> rows_to_remove</span></code></pre></div>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" tabindex="-1"></a>use_samples <span class="ot">&lt;-</span> whole_lung_sample</span>
<span id="cb47-2"><a href="#cb47-2" tabindex="-1"></a></span>
<span id="cb47-3"><a href="#cb47-3" tabindex="-1"></a><span class="co"># RNA-seq : Select genes and samples</span></span>
<span id="cb47-4"><a href="#cb47-4" tabindex="-1"></a>use_se <span class="ot">&lt;-</span> se_rna[<span class="fu">rowData</span>(se_rna)<span class="sc">$</span>gene_name <span class="sc">%in%</span> crispr_genes, use_samples]</span>
<span id="cb47-5"><a href="#cb47-5" tabindex="-1"></a></span>
<span id="cb47-6"><a href="#cb47-6" tabindex="-1"></a><span class="co"># Change rownames</span></span>
<span id="cb47-7"><a href="#cb47-7" tabindex="-1"></a><span class="fu">rownames</span>(use_se) <span class="ot">&lt;-</span><span class="fu">make.names</span>(<span class="fu">rowData</span>(use_se)<span class="sc">$</span>gene_name, <span class="at">unique =</span> <span class="cn">TRUE</span>)</span>
<span id="cb47-8"><a href="#cb47-8" tabindex="-1"></a></span>
<span id="cb47-9"><a href="#cb47-9" tabindex="-1"></a><span class="co"># Proportion of samples with min_counts at each group</span></span>
<span id="cb47-10"><a href="#cb47-10" tabindex="-1"></a>use_group <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(use_se<span class="sc">$</span>donor)</span>
<span id="cb47-11"><a href="#cb47-11" tabindex="-1"></a>counts_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(use_se, <span class="st">&#39;counts&#39;</span>)</span>
<span id="cb47-12"><a href="#cb47-12" tabindex="-1"></a>counts_stats <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i=</span><span class="fu">levels</span>(use_group), <span class="at">.combine =</span> cbind) <span class="sc">%do%</span> {</span>
<span id="cb47-13"><a href="#cb47-13" tabindex="-1"></a>  use_cols <span class="ot">&lt;-</span> use_se<span class="sc">$</span>donor <span class="sc">==</span> i</span>
<span id="cb47-14"><a href="#cb47-14" tabindex="-1"></a>  <span class="fu">rowSums</span>(counts_mat[,use_cols, <span class="at">drop =</span> <span class="cn">FALSE</span>] <span class="sc">&gt;=</span> min_counts)  <span class="sc">/</span> <span class="fu">sum</span>(use_cols)</span>
<span id="cb47-15"><a href="#cb47-15" tabindex="-1"></a>} <span class="sc">%&gt;%</span> data.frame</span>
<span id="cb47-16"><a href="#cb47-16" tabindex="-1"></a><span class="fu">colnames</span>(counts_stats) <span class="ot">&lt;-</span> <span class="fu">levels</span>(use_group)</span>
<span id="cb47-17"><a href="#cb47-17" tabindex="-1"></a>counts_stats<span class="sc">$</span>all <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(counts_mat <span class="sc">&gt;=</span> min_counts)  <span class="sc">/</span> <span class="fu">ncol</span>(counts_mat)</span>
<span id="cb47-18"><a href="#cb47-18" tabindex="-1"></a>  </span>
<span id="cb47-19"><a href="#cb47-19" tabindex="-1"></a><span class="co"># Keep genes with min_counts threshold in &gt;=min_prop of samples in at least 1 group</span></span>
<span id="cb47-20"><a href="#cb47-20" tabindex="-1"></a>rows_to_keep <span class="ot">&lt;-</span> <span class="fu">rownames</span>(counts_stats)[counts_stats<span class="sc">$</span>all <span class="sc">&gt;=</span> min_prop]</span>
<span id="cb47-21"><a href="#cb47-21" tabindex="-1"></a>rows_to_remove <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(<span class="fu">rownames</span>(counts_stats), rows_to_keep)</span>
<span id="cb47-22"><a href="#cb47-22" tabindex="-1"></a></span>
<span id="cb47-23"><a href="#cb47-23" tabindex="-1"></a>expr_genes[[<span class="st">&#39;whole_lung&#39;</span>]] <span class="ot">&lt;-</span> rows_to_keep</span>
<span id="cb47-24"><a href="#cb47-24" tabindex="-1"></a>non_expr_genes[[<span class="st">&#39;whole_lung&#39;</span>]] <span class="ot">&lt;-</span> rows_to_remove</span></code></pre></div>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" tabindex="-1"></a>use_annot <span class="ot">&lt;-</span> <span class="fu">colData</span>(se)</span>
<span id="cb48-2"><a href="#cb48-2" tabindex="-1"></a></span>
<span id="cb48-3"><a href="#cb48-3" tabindex="-1"></a>use_comp <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;StealTHY Py2T&quot;</span>, <span class="st">&quot;Puro Py2T&quot;</span>, <span class="st">&quot;StealTHY NSG Py2T&quot;</span>)</span>
<span id="cb48-4"><a href="#cb48-4" tabindex="-1"></a></span>
<span id="cb48-5"><a href="#cb48-5" tabindex="-1"></a>i <span class="ot">&lt;-</span> use_comp[<span class="dv">1</span>]</span>
<span id="cb48-6"><a href="#cb48-6" tabindex="-1"></a>hits_stats <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i =</span> use_comp, <span class="at">.combine =</span> rbind) <span class="sc">%do%</span> {</span>
<span id="cb48-7"><a href="#cb48-7" tabindex="-1"></a>  i_names <span class="ot">&lt;-</span> i <span class="sc">%&gt;%</span> <span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>, <span class="st">&quot;_&quot;</span>, .) <span class="sc">%&gt;%</span> tolower</span>
<span id="cb48-8"><a href="#cb48-8" tabindex="-1"></a>  st1 <span class="ot">&lt;-</span> <span class="fu">unique</span>(use_annot[ne_mageck_comp_list[[i]][[<span class="dv">1</span>]],]<span class="sc">$</span>sample_type)</span>
<span id="cb48-9"><a href="#cb48-9" tabindex="-1"></a>  st2 <span class="ot">&lt;-</span> <span class="fu">unique</span>(use_annot[ne_mageck_comp_list[[i]][[<span class="dv">2</span>]],]<span class="sc">$</span>sample_type)</span>
<span id="cb48-10"><a href="#cb48-10" tabindex="-1"></a>  </span>
<span id="cb48-11"><a href="#cb48-11" tabindex="-1"></a>  genes_expr <span class="ot">&lt;-</span> <span class="fu">intersect</span>(expr_genes[[st1]], expr_genes[[st2]])</span>
<span id="cb48-12"><a href="#cb48-12" tabindex="-1"></a> </span>
<span id="cb48-13"><a href="#cb48-13" tabindex="-1"></a>  da_res <span class="ot">&lt;-</span> ne_gene_summ[[i_names]]  <span class="sc">%&gt;%</span>  </span>
<span id="cb48-14"><a href="#cb48-14" tabindex="-1"></a>    <span class="fu">ReadRRA</span>(<span class="at">score =</span> <span class="st">&#39;lfc&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb48-15"><a href="#cb48-15" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">LFC =</span> Score) <span class="sc">%&gt;%</span> </span>
<span id="cb48-16"><a href="#cb48-16" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb48-17"><a href="#cb48-17" tabindex="-1"></a>      <span class="at">expressed =</span> id <span class="sc">%in%</span> genes_expr,</span>
<span id="cb48-18"><a href="#cb48-18" tabindex="-1"></a>      <span class="at">hit =</span> (<span class="fu">abs</span>(LFC) <span class="sc">&gt;=</span> <span class="dv">2</span>)</span>
<span id="cb48-19"><a href="#cb48-19" tabindex="-1"></a>      )</span>
<span id="cb48-20"><a href="#cb48-20" tabindex="-1"></a>  </span>
<span id="cb48-21"><a href="#cb48-21" tabindex="-1"></a>   prop_non_expr_genes <span class="ot">&lt;-</span> (<span class="fu">nrow</span>(da_res)<span class="sc">-</span><span class="fu">length</span>(genes_expr)) <span class="sc">/</span> <span class="fu">nrow</span>(da_res)</span>
<span id="cb48-22"><a href="#cb48-22" tabindex="-1"></a>   prop_non_expr_hits <span class="ot">&lt;-</span> <span class="fu">sum</span>(da_res<span class="sc">$</span>hit <span class="sc">&amp;</span> <span class="sc">!</span>da_res<span class="sc">$</span>expressed) <span class="sc">/</span> <span class="fu">sum</span>(da_res<span class="sc">$</span>hit)</span>
<span id="cb48-23"><a href="#cb48-23" tabindex="-1"></a>   <span class="fu">data.frame</span>(</span>
<span id="cb48-24"><a href="#cb48-24" tabindex="-1"></a>     <span class="at">comparison =</span> i,</span>
<span id="cb48-25"><a href="#cb48-25" tabindex="-1"></a>     <span class="at">total_genes =</span> <span class="fu">nrow</span>(da_res),</span>
<span id="cb48-26"><a href="#cb48-26" tabindex="-1"></a>     <span class="at">n_nonexpr_genes =</span> <span class="fu">nrow</span>(da_res)<span class="sc">-</span><span class="fu">length</span>(genes_expr),</span>
<span id="cb48-27"><a href="#cb48-27" tabindex="-1"></a>     <span class="at">prop_non_expr_genes =</span> (<span class="fu">nrow</span>(da_res)<span class="sc">-</span><span class="fu">length</span>(genes_expr)) <span class="sc">/</span> <span class="fu">nrow</span>(da_res),</span>
<span id="cb48-28"><a href="#cb48-28" tabindex="-1"></a>     <span class="at">n_hits =</span> <span class="fu">sum</span>(da_res<span class="sc">$</span>hit),</span>
<span id="cb48-29"><a href="#cb48-29" tabindex="-1"></a>     <span class="at">non_expr_hits =</span> <span class="fu">sum</span>(da_res<span class="sc">$</span>hit <span class="sc">&amp;</span> <span class="sc">!</span>da_res<span class="sc">$</span>expressed),</span>
<span id="cb48-30"><a href="#cb48-30" tabindex="-1"></a>     <span class="at">n_nohits =</span> <span class="fu">sum</span>(<span class="sc">!</span>da_res<span class="sc">$</span>hit)</span>
<span id="cb48-31"><a href="#cb48-31" tabindex="-1"></a>   ) <span class="sc">%&gt;%</span> </span>
<span id="cb48-32"><a href="#cb48-32" tabindex="-1"></a>     <span class="fu">mutate</span>(</span>
<span id="cb48-33"><a href="#cb48-33" tabindex="-1"></a>       <span class="at">prop_non_expr_hits =</span> non_expr_hits <span class="sc">/</span> n_hits,</span>
<span id="cb48-34"><a href="#cb48-34" tabindex="-1"></a>       <span class="at">prop_non_expr_hits_over_non_expr =</span> non_expr_hits <span class="sc">/</span> n_nohits,</span>
<span id="cb48-35"><a href="#cb48-35" tabindex="-1"></a>       <span class="at">lfc =</span> <span class="fu">log2</span>(prop_non_expr_hits <span class="sc">/</span> prop_non_expr_genes)</span>
<span id="cb48-36"><a href="#cb48-36" tabindex="-1"></a>     )</span>
<span id="cb48-37"><a href="#cb48-37" tabindex="-1"></a>  </span>
<span id="cb48-38"><a href="#cb48-38" tabindex="-1"></a>}</span>
<span id="cb48-39"><a href="#cb48-39" tabindex="-1"></a></span>
<span id="cb48-40"><a href="#cb48-40" tabindex="-1"></a>hits_stats <span class="ot">&lt;-</span> hits_stats <span class="sc">%&gt;%</span> </span>
<span id="cb48-41"><a href="#cb48-41" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb48-42"><a href="#cb48-42" tabindex="-1"></a>    <span class="at">comparison =</span> <span class="fu">factor</span>(comparison,<span class="at">levels =</span> <span class="fu">names</span>(ne_mageck_comp_list) <span class="sc">%&gt;%</span> rev)</span>
<span id="cb48-43"><a href="#cb48-43" tabindex="-1"></a>  )</span>
<span id="cb48-44"><a href="#cb48-44" tabindex="-1"></a></span>
<span id="cb48-45"><a href="#cb48-45" tabindex="-1"></a>hits_stats_py2t <span class="ot">&lt;-</span> hits_stats <span class="sc">%&gt;%</span> </span>
<span id="cb48-46"><a href="#cb48-46" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb48-47"><a href="#cb48-47" tabindex="-1"></a>    <span class="at">donor =</span> <span class="st">&#39;Py2T&#39;</span>,</span>
<span id="cb48-48"><a href="#cb48-48" tabindex="-1"></a>    <span class="at">comparison =</span> <span class="fu">gsub</span>(<span class="st">&quot; Py2T&quot;</span>, <span class="st">&quot;&quot;</span>, comparison)</span>
<span id="cb48-49"><a href="#cb48-49" tabindex="-1"></a>  )</span></code></pre></div>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" tabindex="-1"></a>use_annot <span class="ot">&lt;-</span> <span class="fu">colData</span>(se)</span>
<span id="cb49-2"><a href="#cb49-2" tabindex="-1"></a></span>
<span id="cb49-3"><a href="#cb49-3" tabindex="-1"></a>use_comp <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;StealTHY Py2T&quot;</span>, <span class="st">&quot;Puro Py2T&quot;</span>, <span class="st">&quot;StealTHY NSG Py2T&quot;</span>)</span>
<span id="cb49-4"><a href="#cb49-4" tabindex="-1"></a></span>
<span id="cb49-5"><a href="#cb49-5" tabindex="-1"></a>i <span class="ot">&lt;-</span> use_comp[<span class="dv">1</span>]</span>
<span id="cb49-6"><a href="#cb49-6" tabindex="-1"></a>hits_stats <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i =</span> use_comp, <span class="at">.combine =</span> rbind) <span class="sc">%do%</span> {</span>
<span id="cb49-7"><a href="#cb49-7" tabindex="-1"></a>  i_names <span class="ot">&lt;-</span> i <span class="sc">%&gt;%</span> <span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>, <span class="st">&quot;_&quot;</span>, .) <span class="sc">%&gt;%</span> tolower</span>
<span id="cb49-8"><a href="#cb49-8" tabindex="-1"></a>  st1 <span class="ot">&lt;-</span> <span class="fu">unique</span>(use_annot[ne_mageck_comp_list[[i]][[<span class="dv">1</span>]],]<span class="sc">$</span>sample_type)</span>
<span id="cb49-9"><a href="#cb49-9" tabindex="-1"></a>  st2 <span class="ot">&lt;-</span> <span class="fu">unique</span>(use_annot[ne_mageck_comp_list[[i]][[<span class="dv">2</span>]],]<span class="sc">$</span>sample_type)</span>
<span id="cb49-10"><a href="#cb49-10" tabindex="-1"></a>  </span>
<span id="cb49-11"><a href="#cb49-11" tabindex="-1"></a>  genes_expr <span class="ot">&lt;-</span> <span class="fu">intersect</span>(expr_genes_combined[[st1]], expr_genes_combined[[st2]])</span>
<span id="cb49-12"><a href="#cb49-12" tabindex="-1"></a> </span>
<span id="cb49-13"><a href="#cb49-13" tabindex="-1"></a>  da_res <span class="ot">&lt;-</span> ne_gene_summ[[i_names]]  <span class="sc">%&gt;%</span>  </span>
<span id="cb49-14"><a href="#cb49-14" tabindex="-1"></a>    <span class="fu">ReadRRA</span>(<span class="at">score =</span> <span class="st">&#39;lfc&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb49-15"><a href="#cb49-15" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">LFC =</span> Score) <span class="sc">%&gt;%</span> </span>
<span id="cb49-16"><a href="#cb49-16" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb49-17"><a href="#cb49-17" tabindex="-1"></a>      <span class="at">expressed =</span> id <span class="sc">%in%</span> genes_expr,</span>
<span id="cb49-18"><a href="#cb49-18" tabindex="-1"></a>      <span class="at">hit =</span> (<span class="fu">abs</span>(LFC) <span class="sc">&gt;=</span> <span class="dv">2</span>)</span>
<span id="cb49-19"><a href="#cb49-19" tabindex="-1"></a>      )</span>
<span id="cb49-20"><a href="#cb49-20" tabindex="-1"></a>  </span>
<span id="cb49-21"><a href="#cb49-21" tabindex="-1"></a>   prop_non_expr_genes <span class="ot">&lt;-</span> (<span class="fu">nrow</span>(da_res)<span class="sc">-</span><span class="fu">length</span>(genes_expr)) <span class="sc">/</span> <span class="fu">nrow</span>(da_res)</span>
<span id="cb49-22"><a href="#cb49-22" tabindex="-1"></a>   prop_non_expr_hits <span class="ot">&lt;-</span> <span class="fu">sum</span>(da_res<span class="sc">$</span>hit <span class="sc">&amp;</span> <span class="sc">!</span>da_res<span class="sc">$</span>expressed) <span class="sc">/</span> <span class="fu">sum</span>(da_res<span class="sc">$</span>hit)</span>
<span id="cb49-23"><a href="#cb49-23" tabindex="-1"></a>   <span class="fu">data.frame</span>(</span>
<span id="cb49-24"><a href="#cb49-24" tabindex="-1"></a>     <span class="at">comparison =</span> i,</span>
<span id="cb49-25"><a href="#cb49-25" tabindex="-1"></a>     <span class="at">total_genes =</span> <span class="fu">nrow</span>(da_res),</span>
<span id="cb49-26"><a href="#cb49-26" tabindex="-1"></a>     <span class="at">n_nonexpr_genes =</span> <span class="fu">nrow</span>(da_res)<span class="sc">-</span><span class="fu">length</span>(genes_expr),</span>
<span id="cb49-27"><a href="#cb49-27" tabindex="-1"></a>     <span class="at">prop_non_expr_genes =</span> (<span class="fu">nrow</span>(da_res)<span class="sc">-</span><span class="fu">length</span>(genes_expr)) <span class="sc">/</span> <span class="fu">nrow</span>(da_res),</span>
<span id="cb49-28"><a href="#cb49-28" tabindex="-1"></a>     <span class="at">n_hits =</span> <span class="fu">sum</span>(da_res<span class="sc">$</span>hit),</span>
<span id="cb49-29"><a href="#cb49-29" tabindex="-1"></a>     <span class="at">non_expr_hits =</span> <span class="fu">sum</span>(da_res<span class="sc">$</span>hit <span class="sc">&amp;</span> <span class="sc">!</span>da_res<span class="sc">$</span>expressed),</span>
<span id="cb49-30"><a href="#cb49-30" tabindex="-1"></a>     <span class="at">n_nohits =</span> <span class="fu">sum</span>(<span class="sc">!</span>da_res<span class="sc">$</span>hit)</span>
<span id="cb49-31"><a href="#cb49-31" tabindex="-1"></a>   ) <span class="sc">%&gt;%</span> </span>
<span id="cb49-32"><a href="#cb49-32" tabindex="-1"></a>     <span class="fu">mutate</span>(</span>
<span id="cb49-33"><a href="#cb49-33" tabindex="-1"></a>       <span class="at">prop_non_expr_hits =</span> non_expr_hits <span class="sc">/</span> n_hits,</span>
<span id="cb49-34"><a href="#cb49-34" tabindex="-1"></a>       <span class="at">prop_non_expr_hits_over_non_expr =</span> non_expr_hits <span class="sc">/</span> n_nohits,</span>
<span id="cb49-35"><a href="#cb49-35" tabindex="-1"></a>       <span class="at">lfc =</span> <span class="fu">log2</span>(prop_non_expr_hits <span class="sc">/</span> prop_non_expr_genes)</span>
<span id="cb49-36"><a href="#cb49-36" tabindex="-1"></a>     )</span>
<span id="cb49-37"><a href="#cb49-37" tabindex="-1"></a>  </span>
<span id="cb49-38"><a href="#cb49-38" tabindex="-1"></a>}</span>
<span id="cb49-39"><a href="#cb49-39" tabindex="-1"></a></span>
<span id="cb49-40"><a href="#cb49-40" tabindex="-1"></a>hits_stats <span class="ot">&lt;-</span> hits_stats <span class="sc">%&gt;%</span> </span>
<span id="cb49-41"><a href="#cb49-41" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb49-42"><a href="#cb49-42" tabindex="-1"></a>    <span class="at">comparison =</span> <span class="fu">factor</span>(comparison,<span class="at">levels =</span> <span class="fu">names</span>(ne_mageck_comp_list) <span class="sc">%&gt;%</span> rev)</span>
<span id="cb49-43"><a href="#cb49-43" tabindex="-1"></a>  )</span>
<span id="cb49-44"><a href="#cb49-44" tabindex="-1"></a></span>
<span id="cb49-45"><a href="#cb49-45" tabindex="-1"></a>hits_stats_py2t_comb <span class="ot">&lt;-</span> hits_stats <span class="sc">%&gt;%</span> </span>
<span id="cb49-46"><a href="#cb49-46" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb49-47"><a href="#cb49-47" tabindex="-1"></a>    <span class="at">donor =</span> <span class="st">&#39;Py2T&#39;</span>,</span>
<span id="cb49-48"><a href="#cb49-48" tabindex="-1"></a>    <span class="at">comparison =</span> <span class="fu">gsub</span>(<span class="st">&quot; Py2T&quot;</span>, <span class="st">&quot;&quot;</span>, comparison)</span>
<span id="cb49-49"><a href="#cb49-49" tabindex="-1"></a>  )</span></code></pre></div>
<div id="figure-4d-proportion-of-non-expressed-hits"
class="section level3">
<h3>Figure 4D: Proportion of non-expressed hits</h3>
<p>Bar graph presenting the proportion of hits (log2 fold-change ≥ 2 or
≤ −2) not expressed (read counts &lt;1 in ≥ 2/3 of samples) in MVT1, 4T1
and Py2T models (n=5-7), assessed using RNA-seq of Thy(A)-sorted
carcinoma cells from primary tumors (n=3) and metastatic lungs
(n=3).</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" tabindex="-1"></a>chunk_data <span class="ot">&lt;-</span> hits_stats_global</span>
<span id="cb50-2"><a href="#cb50-2" tabindex="-1"></a></span>
<span id="cb50-3"><a href="#cb50-3" tabindex="-1"></a>chunk_data <span class="sc">%&gt;%</span></span>
<span id="cb50-4"><a href="#cb50-4" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(comparison, prop_non_expr_hits, <span class="at">fill =</span> comparison)) <span class="sc">+</span></span>
<span id="cb50-5"><a href="#cb50-5" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb50-6"><a href="#cb50-6" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">&#39;solid&#39;</span>,</span>
<span id="cb50-7"><a href="#cb50-7" tabindex="-1"></a>             <span class="at">linewidth =</span> one_pt<span class="sc">/</span><span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb50-8"><a href="#cb50-8" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb50-9"><a href="#cb50-9" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(</span>
<span id="cb50-10"><a href="#cb50-10" tabindex="-1"></a>    <span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.05</span>))</span>
<span id="cb50-11"><a href="#cb50-11" tabindex="-1"></a>    )  <span class="sc">+</span></span>
<span id="cb50-12"><a href="#cb50-12" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> use_color_palette) <span class="sc">+</span></span>
<span id="cb50-13"><a href="#cb50-13" tabindex="-1"></a>  theme_facet <span class="sc">+</span></span>
<span id="cb50-14"><a href="#cb50-14" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb50-15"><a href="#cb50-15" tabindex="-1"></a>    <span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">linewidth =</span> one_pt<span class="sc">/</span><span class="dv">4</span>, <span class="at">color =</span> <span class="st">&#39;black&#39;</span>),</span>
<span id="cb50-16"><a href="#cb50-16" tabindex="-1"></a>    <span class="at">axis.ticks =</span> <span class="fu">element_line</span>(<span class="at">linewidth =</span> one_pt<span class="sc">/</span><span class="dv">4</span>, <span class="at">color =</span> <span class="st">&#39;black&#39;</span>),</span>
<span id="cb50-17"><a href="#cb50-17" tabindex="-1"></a>    <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb50-18"><a href="#cb50-18" tabindex="-1"></a>    <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">5</span>), </span>
<span id="cb50-19"><a href="#cb50-19" tabindex="-1"></a>    <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">5</span>),</span>
<span id="cb50-20"><a href="#cb50-20" tabindex="-1"></a>    <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">5</span>),</span>
<span id="cb50-21"><a href="#cb50-21" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">5</span>),</span>
<span id="cb50-22"><a href="#cb50-22" tabindex="-1"></a>    <span class="at">strip.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">4</span>, <span class="at">angle =</span> <span class="dv">0</span>)</span>
<span id="cb50-23"><a href="#cb50-23" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb50-24"><a href="#cb50-24" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill =</span> <span class="st">&#39;none&#39;</span>) <span class="sc">+</span></span>
<span id="cb50-25"><a href="#cb50-25" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb50-26"><a href="#cb50-26" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">&#39;&#39;</span>,</span>
<span id="cb50-27"><a href="#cb50-27" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">&#39;Proportion of</span><span class="sc">\n</span><span class="st">non-expressed hits&#39;</span></span>
<span id="cb50-28"><a href="#cb50-28" tabindex="-1"></a>  )</span></code></pre></div>
<p><img src="figure/crispr-mm_2215_sgRNA-StealTHY.Rmd/non-expressed-hits-barplot-prop-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-non-expressed-hits-barplot-prop-1">
Past versions of non-expressed-hits-barplot-prop-1.png
</button>
</p>
<div id="fig-non-expressed-hits-barplot-prop-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/32aca023d6d8209f1aa3df3f177fc383c7569844/docs/figure/crispr-mm_2215_sgRNA-StealTHY.Rmd/non-expressed-hits-barplot-prop-1.png" target="_blank">32aca02</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-07-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="supplementary-figure-4g-proportion-of-non-expressed-hits"
class="section level3">
<h3>Supplementary figure 4G: Proportion of non-expressed hits</h3>
<p>Detailed analysis of non-expressed screen hits sub-divided per
individual mammary carcinoma model. Top panel: bar graph presenting the
proportion of hits that are not expressed but are still identified by
CRISPR KO screening. Genes were classified as “hit” if their absolute
log2 fold-change was ≥2 in each screening condition (minimum n=5 mice
per model), and as “non-expressed” if read counts were &lt;1 in at least
2/3 of the samples, according to RNA-seq of Thy(A)-sorted carcinoma
cells from metastatic lungs (n=3) in intersection with RNA-seq data from
primary tumors (n=3); all RNA-seq data have been acquired in syngeneic
hosts.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" tabindex="-1"></a>chunk_data <span class="ot">&lt;-</span> <span class="fu">rbind</span>(hits_stats_4t1, hits_stats_mvt1, hits_stats_py2t) <span class="sc">%&gt;%</span> </span>
<span id="cb51-2"><a href="#cb51-2" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb51-3"><a href="#cb51-3" tabindex="-1"></a>    <span class="at">comparison =</span> <span class="fu">factor</span>(comparison, <span class="at">levels =</span> <span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">&#39;StealTHY&#39;</span>, <span class="st">&#39;Puro&#39;</span>, <span class="st">&#39;Cas9+Puro&#39;</span>, <span class="st">&#39;StealTHY NSG&#39;</span>)))</span>
<span id="cb51-4"><a href="#cb51-4" tabindex="-1"></a>  )</span>
<span id="cb51-5"><a href="#cb51-5" tabindex="-1"></a>chunk_data <span class="sc">%&gt;%</span></span>
<span id="cb51-6"><a href="#cb51-6" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(comparison, prop_non_expr_hits, <span class="at">fill =</span> comparison)) <span class="sc">+</span></span>
<span id="cb51-7"><a href="#cb51-7" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb51-8"><a href="#cb51-8" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb51-9"><a href="#cb51-9" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(</span>
<span id="cb51-10"><a href="#cb51-10" tabindex="-1"></a>    <span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.05</span>))</span>
<span id="cb51-11"><a href="#cb51-11" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb51-12"><a href="#cb51-12" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> use_color_palette) <span class="sc">+</span></span>
<span id="cb51-13"><a href="#cb51-13" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="at">rows =</span> <span class="fu">vars</span>(donor),</span>
<span id="cb51-14"><a href="#cb51-14" tabindex="-1"></a>             <span class="at">scales =</span> <span class="st">&#39;free_y&#39;</span>,</span>
<span id="cb51-15"><a href="#cb51-15" tabindex="-1"></a>             <span class="at">space =</span> <span class="st">&#39;free_y&#39;</span>) <span class="sc">+</span></span>
<span id="cb51-16"><a href="#cb51-16" tabindex="-1"></a>  theme_facet <span class="sc">+</span></span>
<span id="cb51-17"><a href="#cb51-17" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb51-18"><a href="#cb51-18" tabindex="-1"></a>    <span class="at">axis.ticks =</span> <span class="fu">element_line</span>(<span class="at">linewidth =</span> one_pt<span class="sc">/</span><span class="dv">2</span>, <span class="at">color =</span> <span class="st">&#39;black&#39;</span>),</span>
<span id="cb51-19"><a href="#cb51-19" tabindex="-1"></a>    <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">linewidth =</span> one_pt<span class="sc">/</span><span class="dv">2</span>, <span class="at">color =</span> <span class="st">&#39;black&#39;</span>),</span>
<span id="cb51-20"><a href="#cb51-20" tabindex="-1"></a>    <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">4</span>), </span>
<span id="cb51-21"><a href="#cb51-21" tabindex="-1"></a>    <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">4</span>),</span>
<span id="cb51-22"><a href="#cb51-22" tabindex="-1"></a>    <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">4</span>),</span>
<span id="cb51-23"><a href="#cb51-23" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">4</span>),</span>
<span id="cb51-24"><a href="#cb51-24" tabindex="-1"></a>    <span class="at">strip.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">4</span>, <span class="at">angle =</span> <span class="dv">0</span>)</span>
<span id="cb51-25"><a href="#cb51-25" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb51-26"><a href="#cb51-26" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill =</span> <span class="st">&#39;none&#39;</span>) <span class="sc">+</span></span>
<span id="cb51-27"><a href="#cb51-27" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb51-28"><a href="#cb51-28" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">&#39;&#39;</span>,</span>
<span id="cb51-29"><a href="#cb51-29" tabindex="-1"></a>    <span class="at">y =</span>  <span class="st">&#39;Proportion of</span><span class="sc">\n</span><span class="st">non-expressed hits&#39;</span></span>
<span id="cb51-30"><a href="#cb51-30" tabindex="-1"></a>  )</span></code></pre></div>
<p><img src="figure/crispr-mm_2215_sgRNA-StealTHY.Rmd/non-expressed-hits-barplot-prop-bymodel-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-non-expressed-hits-barplot-prop-bymodel-1">
Past versions of non-expressed-hits-barplot-prop-bymodel-1.png
</button>
</p>
<div id="fig-non-expressed-hits-barplot-prop-bymodel-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/32aca023d6d8209f1aa3df3f177fc383c7569844/docs/figure/crispr-mm_2215_sgRNA-StealTHY.Rmd/non-expressed-hits-barplot-prop-bymodel-1.png" target="_blank">32aca02</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-07-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<p>R version 4.4.3 (2025-02-28) Platform: aarch64-apple-darwin20 Running
under: macOS 26.0.1</p>
<p>Matrix products: default BLAS:
/Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib
LAPACK:
/Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;
LAPACK version 3.12.0</p>
<p>locale: [1]
en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</p>
<p>time zone: Europe/Zurich tzcode source: internal</p>
<p>attached base packages: [1] stats4 stats graphics grDevices utils
datasets methods<br />
[8] base</p>
<p>other attached packages: [1] cowplot_1.2.0 MAGeCKFlute_2.9.0<br />
[3] ggpubr_0.6.1 ggbeeswarm_0.7.2<br />
[5] colorblindr_0.1.0 colorspace_2.1-1<br />
[7] patchwork_1.3.1 ggh4x_0.3.1<br />
[9] ggridges_0.5.6 SummarizedExperiment_1.36.0 [11] Biobase_2.66.0
GenomicRanges_1.58.0<br />
[13] GenomeInfoDb_1.42.3 IRanges_2.40.1<br />
[15] S4Vectors_0.44.0 BiocGenerics_0.52.0<br />
[17] MatrixGenerics_1.18.1 matrixStats_1.5.0<br />
[19] diptest_0.77-1 kableExtra_1.4.0<br />
[21] DT_0.34.0 magrittr_2.0.3<br />
[23] foreach_1.5.2 knitr_1.50<br />
[25] lubridate_1.9.4 forcats_1.0.0<br />
[27] stringr_1.5.1 dplyr_1.1.4<br />
[29] purrr_1.1.0 readr_2.1.5<br />
[31] tidyr_1.3.1 tibble_3.3.0<br />
[33] ggplot2_3.5.2 tidyverse_2.0.0<br />
[35] workflowr_1.7.1</p>
<p>loaded via a namespace (and not attached): [1] splines_4.4.3
later_1.4.4 ggplotify_0.1.2<br />
[4] bitops_1.0-9 filelock_1.0.3 cellranger_1.1.0<br />
[7] R.oo_1.27.1 graph_1.84.1 XML_3.99-0.19<br />
[10] lifecycle_1.0.4 httr2_1.2.1 rstatix_0.7.2<br />
[13] edgeR_4.4.2 rprojroot_2.1.0 processx_3.8.6<br />
[16] lattice_0.22-7 crosstalk_1.2.2 backports_1.5.0<br />
[19] limma_3.62.2 sass_0.4.10 rmarkdown_2.29<br />
[22] jquerylib_0.1.4 yaml_2.3.10 ggtangle_0.0.7<br />
[25] httpuv_1.6.16 depmap_1.20.0 DBI_1.2.3<br />
[28] RColorBrewer_1.1-3 abind_1.4-8 zlibbioc_1.52.0<br />
[31] R.utils_2.13.0 RCurl_1.98-1.17 yulab.utils_0.2.0<br />
[34] rappdirs_0.3.3 git2r_0.36.2 GenomeInfoDbData_1.2.13 [37]
enrichplot_1.26.6 ggrepel_0.9.6 tidytree_0.4.6<br />
[40] svglite_2.2.1 codetools_0.2-20 DelayedArray_0.32.0<br />
[43] DOSE_4.0.1 xml2_1.3.8 tidyselect_1.2.1<br />
[46] aplot_0.2.8 UCSC.utils_1.2.0 farver_2.1.2<br />
[49] BiocFileCache_2.14.0 pathview_1.46.0 jsonlite_2.0.0<br />
[52] Formula_1.2-5 iterators_1.0.14 systemfonts_1.2.3<br />
[55] tools_4.4.3 treeio_1.30.0 Rcpp_1.1.0<br />
[58] glue_1.8.0 gridExtra_2.3 SparseArray_1.6.2<br />
[61] xfun_0.53 qvalue_2.38.0 withr_3.0.2<br />
[64] BiocManager_1.30.26 fastmap_1.2.0 callr_3.7.6<br />
[67] digest_0.6.37 gridGraphics_0.5-1 timechange_0.3.0<br />
[70] R6_2.6.1 textshaping_1.0.1 GO.db_3.20.0<br />
[73] dichromat_2.0-0.1 RSQLite_2.4.3 R.methodsS3_1.8.2<br />
[76] generics_0.1.4 data.table_1.17.8 httr_1.4.7<br />
[79] htmlwidgets_1.6.4 S4Arrays_1.6.0 whisker_0.4.1<br />
[82] pkgconfig_2.0.3 gtable_0.3.6 blob_1.2.4<br />
[85] XVector_0.46.0 clusterProfiler_4.14.6 htmltools_0.5.8.1<br />
[88] carData_3.0-5 fgsea_1.32.4 scales_1.4.0<br />
[91] png_0.1-8 ggfun_0.2.0 rstudioapi_0.17.1<br />
[94] tzdb_0.5.0 reshape2_1.4.4 nlme_3.1-168<br />
[97] curl_7.0.0 org.Hs.eg.db_3.20.0 cachem_1.1.0<br />
[100] BiocVersion_3.20.0 parallel_4.4.3 vipor_0.4.7<br />
[103] AnnotationDbi_1.68.0 pillar_1.11.0 grid_4.4.3<br />
[106] vctrs_0.6.5 promises_1.3.3 car_3.1-3<br />
[109] dbplyr_2.5.0 beeswarm_0.4.0 Rgraphviz_2.50.0<br />
[112] evaluate_1.0.5 KEGGgraph_1.66.0 locfit_1.5-9.12<br />
[115] cli_3.6.5 compiler_4.4.3 rlang_1.1.6<br />
[118] crayon_1.5.3 ggsignif_0.6.4 labeling_0.4.3<br />
[121] ps_1.9.1 getPass_0.2-4 plyr_1.8.9<br />
[124] fs_1.6.6 stringi_1.8.7 viridisLite_0.4.2<br />
[127] BiocParallel_1.40.2 Biostrings_2.74.1 lazyeval_0.2.2<br />
[130] GOSemSim_2.32.0 Matrix_1.7-3 ExperimentHub_2.14.0<br />
[133] hms_1.1.3 bit64_4.6.0-1 statmod_1.5.0<br />
[136] KEGGREST_1.46.0 AnnotationHub_3.14.0 igraph_2.1.4<br />
[139] broom_1.0.9 memoise_2.0.1 bslib_0.9.0<br />
[142] ggtree_3.14.0 fastmatch_1.1-6 bit_4.6.0<br />
[145] readxl_1.4.5 gson_0.1.0 ape_5.8-1</p>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
