<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Francesc Castro-Giner" />


<title>StealTHY eludes CRISPR immunogenicity to expose concealed metastatic regulators</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>







<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">StealTHY</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="about.html">
    <span class="fa fa-info"></span>
     
    About
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-bar-chart"></span>
     
    Analyses
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Main results</li>
    <li>
      <a href="crispr-mm_2215_sgRNA-clonality.html">In-vivo CRISPR screening in mouse: sgRNA distribution</a>
    </li>
    <li>
      <a href="rnaseq-tumor-bulk.html">RNA-seq of primary tumor transcriptomes from syngeneic hosts based on Cas9 delivery TrCD8a status</a>
    </li>
    <li>
      <a href="crispr-mm_2215_sgRNA-StealTHY.html">Restoration of metastatic gene discovery by StealTHY CRISPR screening</a>
    </li>
    <li>
      <a href="crispr-mm_2215_sgRNA-CMT167-LLC1-CT26.html">CRISPR screening transplantation in immunocompetent hosts of metastatic lung carcinoma cells (LLC1, C167) and colon carcinoma cells (CT26) previously subjected to StealTHY KO</a>
    </li>
    <li>
      <a href="crispr-hsapiens_2180_sgRNA.html">CRISPR screening in humanized mice by StealTHY</a>
    </li>
    <li>
      <a href="tcga_survival_crispr_hits_signatures.html">Survival analysis of gene-signatures identified in the CRISPR screening in humanized mice by StealTHY</a>
    </li>
    <li>
      <a href="rnaseq-amhr2_ko-deg.html">RNA-sequencing of primary tumors from syngeneic FVB/N orthotopically transplanted with Py2T carrying StealTHY KO of Amhr2 or a non-targeting sgRNA</a>
    </li>
    <li>
      <a href="rnaseq-tumor-facs-immune.html">RNA-seq of FACs sorted CD3+ TIL cells comparing Cas9-Thy1(A) versus Thy1(A) transduction</a>
    </li>
    <li>
      <a href="rnaseq-tumor-facs-cancer.html">RNA-seq of FACs sorted carcinoma cells comparing Cas9-Thy1(A) versus Thy1(A) transduction</a>
    </li>
    <li>
      <a href="tcga_immune_infiltrate_amhr2_ko_signatures.html">Correlation of AMH signatures with immune cell fractions in TCGA</a>
    </li>
    <li>
      <a href="tcga_survival_amhr2_ko_signatures.html">Survival analysis of AMH signatures in TCGA cohorts</a>
    </li>
    <li>
      <a href="tcga_survival_crispr_hits_genes.html">Survival analysis of AMH, AMHR2 and GDF9 expression in TCGA cohorts</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="https://github.com/TheAcetoLab/saini-stealTHY">
    <span class="fa fa-github"></span>
     
  </a>
</li>
<li>
  <a href="https://www.theacetolab.com">
    <span class="fa fa-flask"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">StealTHY eludes CRISPR immunogenicity to
expose concealed metastatic regulators</h1>
<h4 class="author">Francesc Castro-Giner</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-10-31
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>saini-stealTHY/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomTheAcetoLabsainistealTHYtreead11dd11969017f0c40dfc9ebaca78941299ddebtargetblankad11dd1a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/tree/ad11dd11969017f0c40dfc9ebaca78941299ddeb" target="_blank">ad11dd1</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomTheAcetoLabsainistealTHYtreead11dd11969017f0c40dfc9ebaca78941299ddebtargetblankad11dd1a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/TheAcetoLab/saini-stealTHY/tree/ad11dd11969017f0c40dfc9ebaca78941299ddeb" target="_blank">ad11dd1</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/templates/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    configuration/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/crispr/
    Ignored:    data/resources/
    Ignored:    data/rnaseq/
    Ignored:    output/.DS_Store
    Ignored:    output/clinical/
    Ignored:    output/crispr/
    Ignored:    output/rnaseq/

Untracked files:
    Untracked:  data.tar.gz
    Untracked:  output.tar.gz

Unstaged changes:
    Modified:   .gitignore
    Modified:   README.md
    Deleted:    update_workflowr.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML
(<code>docs/index.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/ad11dd11969017f0c40dfc9ebaca78941299ddeb/analysis/index.Rmd" target="_blank">ad11dd1</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-10-31
</td>
<td>
Update index
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheAcetoLab/saini-stealTHY/30468d6732c3368c2e61d57795cca8801e717d2e/docs/index.html" target="_blank">30468d6</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-10-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/54b3c4df86314709e556aa3999fc0bbf4226f47d/analysis/index.Rmd" target="_blank">54b3c4d</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-10-31
</td>
<td>
Update index, license and about
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheAcetoLab/saini-stealTHY/5e831e20de8357cadc402a20a98aa789b537a814/docs/index.html" target="_blank">5e831e2</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-10-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/c44a44f06465066cb175aacc9cfe2cbe6376900a/analysis/index.Rmd" target="_blank">c44a44f</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-10-31
</td>
<td>
Update index, license and about
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheAcetoLab/saini-stealTHY/595c4b0d5d16f835cd86f5fe463b47ce59cd3cea/docs/index.html" target="_blank">595c4b0</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-10-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/aa04e35a434c6815631b0107424f4798d3fd697b/analysis/index.Rmd" target="_blank">aa04e35</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-10-31
</td>
<td>
Update index, license and about
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheAcetoLab/saini-stealTHY/4921153ca8c84dd525d308e4183929b6d9909a0f/docs/index.html" target="_blank">4921153</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-10-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/455fa9121fcca51363cf639f7ee170fb108e34c1/analysis/index.Rmd" target="_blank">455fa91</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-10-31
</td>
<td>
Prepare files for publication
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheAcetoLab/saini-stealTHY/13b89b011934eb3bd6ea30828c0fe8a0c0cd5a92/docs/index.html" target="_blank">13b89b0</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2024-05-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/5c3fb91479e6eb57971b4b2e04b5d7c95d632e4a/analysis/index.Rmd" target="_blank">5c3fb91</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2024-05-17
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="publication" class="section level2">
<h2>Publication</h2>
<p>Saini, M., Castro-Giner, F., Hotz, A., Sznurkowska, M.K., Nüesch, M.,
Cuenot, F.M., Budinjas, S., Bilfeld, G., Ildız, E.S., Strittmatter, K.,
Krol, I., Diamantopoulou, Z., Paasinen-Sohns, A., Waldmeier, M., Cássio,
R., Kreutzer, S., Kontarakis, Z., Gvozdenovic, A., Aceto, N. StealTHY
eludes CRISPR immunogenicity to expose concealed metastatic regulators
(2025)</p>
</div>
<div id="abstract" class="section level2">
<h2>Abstract</h2>
<p>CRISPR screens have become standard gene discovery platforms in
various contexts, including cancer. Yet, commonly available CRISPR-Cas9
tools are increasingly recognized as unfit for in vivo investigations in
immunocompetent contexts, due to broad immunogenicity of bacterial
nucleases and reporters. Here, we show how conventional CRISPR screens
in tumor grafts are systematically jeopardized by immunoediting in
syngeneic and humanized immunocompetent hosts, resulting in iatrogenic
clonal dropouts and ultimately compromising target identification. To
resolve this, we present StealTHY, an immunogen-free CRISPR platform
compatible with virtually all immunocompetent designs, enabling
preservation of clonal architecture and exposing previously concealed
cancer vulnerabilities. Among these, we identify the AMH-AMHR2 axis as a
formerly unappreciated metastasis target. Thus, with StealTHY, we
provide a new resource to expand the applicability of CRISPR screens to
immunocompetent models including humanized tumor grafts, revealing
metastasis regulators of therapeutic relevance.</p>
</div>
<div id="data-pre-processing" class="section level2">
<h2>Data pre-processing</h2>
<p>Raw data is available at Gene Expression Omnibus (GEO, NCBI), under
the accession codes <a
href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE268044">GSE268044</a>,
<a
href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE268045">GSE268045</a>
and <a
href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE268588">GSE268588</a>.
Data processing is computationally expensive and is not covered in this
repository. We provide description of the data pre-processing workflow
together with software version in the original publication. Processed
data, large result files, additional functions, references and metadata
are were archived at <a
href="https://doi.org/10.5281/zenodo.11208099"><img
src="https://zenodo.org/doi/10.5281/zenodo.11208099.svg"
alt="DOI" /></a></p>
</div>
<div id="data-and-code-availability" class="section level2">
<h2>Data and code availability</h2>
<p>To reproduce our analysis, first clone source code from the <a
href="https://github.com/TheAcetoLab/saini-stealTHY">GitHub
repository</a>. This repository is also archived at <a
href="https://doi.org/10.5281/11208099"><img
src="https://zenodo.org/badge/DOI/10.5281/zenodo.11208099.svg"
alt="DOI" /></a></p>
<pre><code>git clone https://github.com/TheAcetoLab/saini-stealTHY</code></pre>
<p>Next, download processed data and output data deposited in <a
href="https://doi.org/10.5281/zenodo.11374563">Zenodo</a> into the
cloned project folder and untar the files.</p>
<pre><code>for file in *.tar.gz; do tar xzvf &quot;${file}&quot; &amp;&amp; rm &quot;${file}&quot;; done</code></pre>
</div>
<div id="reproducibility" class="section level2">
<h2>Reproducibility</h2>
<p>The results form our analyses are listed below in webpage format.
They were generated from R Markdown documents deposited in the <a
href="https://github.com/TheAcetoLab/saini-stealTHY">GitHub
repository</a>. The workflow of the analysis was created using the <a
href="https://cran.r-project.org/web/packages/workflowr/index.html">workflowr</a>
R package and can be reproduced in its totality using <a
href="https://cran.r-project.org/web/packages/workflowr/index.html">workflowr</a>
<a
href="https://jdblischak.github.io/workflowrBeta/reference/wflow_build.html">wflow_build</a>
command after the installation of the proper R packages. Session info,
including R and package versions, was automatically included at the end
of each analysis file.</p>
<p>Files containing pre-computed results from differential expression or
gene-set enrichment analyses were deposited in <a
href="https://doi.org/10.5281/zenodo.11374563"><img
src="https://zenodo.org/badge/DOI/10.5281/zenodo.11374563.svg"
alt="DOI" /></a>. In order to generate those files again change the
option <code>eval = FALSE</code> to <code>eval = TRUE</code> in the
specific code chunk from the R Markdown file.</p>
</div>
<div id="analyses" class="section level2">
<h2>Analyses</h2>
<ul>
<li><a href="crispr-mm_2215_sgRNA-clonality.html">In-vivo CRISPR
screening in mouse: sgRNA distribution</a></li>
<li><a href="rnaseq-tumor-bulk.html">RNA-seq of primary tumor
transcriptomes from syngeneic hosts based on Cas9 delivery TrCD8a
status</a></li>
<li><a href="crispr-mm_2215_sgRNA-StealTHY.html">Restoration of
metastatic gene discovery by StealTHY CRISPR screening</a></li>
<li><a href="crispr-mm_2215_sgRNA-CMT167-LLC1-CT26.html">CRISPR
screening transplantation in immunocompetent hosts of metastatic lung
carcinoma cells (LLC1, C167) and colon carcinoma cells (CT26) previously
subjected to StealTHY KO</a></li>
<li><a href="crispr-hsapiens_2180_sgRNA.html">CRISPR screening in
humanized mice by StealTHY</a></li>
<li><a href="tcga_survival_crispr_hits_signatures.html">Survival
analysis of gene-signatures identified in the CRISPR screening in
humanized mice by StealTHY</a></li>
<li><a href="rnaseq-amhr2_ko-deg.html">RNA-sequencing of primary tumors
from syngeneic FVB/N orthotopically transplanted with Py2T carrying
StealTHY KO of Amhr2 or a non-targeting sgRNA</a></li>
<li><a href="rnaseq-tumor-facs-immune.html">RNA-seq of FACs sorted CD3+
TIL cells comparing Cas9-Thy1(A) versus Thy1(A) transduction</a></li>
<li><a href="rnaseq-tumor-facs-cancer.html">RNA-seq of FACs sorted
carcinoma cells comparing Cas9-Thy1(A) versus Thy1(A)
transduction</a></li>
<li><a
href="tcga_immune_infiltrate_amhr2_ko_signatures.html">Correlation of
<em>AMH</em> signatures with immune cell fractions in TCGA</a></li>
<li><a href="tcga_survival_amhr2_ko_signatures.html">Survival analysis
of <em>AMH</em> signatures in TCGA cohorts</a></li>
<li><a href="tcga_survival_crispr_hits_genes.html">Survival analysis of
<em>AMH</em>, <em>AMHR2</em> and <em>GDF9</em> expression in TCGA
cohorts</a></li>
</ul>
</div>
<div id="main-figures" class="section level2">
<h2>Main figures</h2>
<div id="figure-2" class="section level3">
<h3>Figure 2</h3>
<ul>
<li><a
href="crispr-mm_2215_sgRNA-clonality.html#31_Figure_2B:_sgRNA_distribution_in_mice_transplanted_with_4T1_cells">Figure
2B</a></li>
<li><a
href="crispr-mm_2215_sgRNA-clonality.html#5_Figure_2C:_Representative_M%C3%BCller_plots_of_4T1_in_syngeneic_recipients">Figure
2C</a></li>
</ul>
</div>
<div id="figure-3" class="section level3">
<h3>Figure 3</h3>
<ul>
<li><a
href="rnaseq-tumor-bulk.html#Figure_3D_:_qRT-PCR_in_TrCD8a+_and_TrCD8a-">Figure
3D</a></li>
<li><a
href="rnaseq-tumor-bulk.html#Figure_3H:_Sample_distances_in_MVT1_models">Figure
3H</a></li>
</ul>
</div>
<div id="figure-4" class="section level3">
<h3>Figure 4</h3>
<ul>
<li><a
href="crispr-mm_2215_sgRNA-StealTHY.html#sgRNA_distribution2">Figure
4D</a></li>
<li><a
href="crispr-mm_2215_sgRNA-StealTHY.html#Figure_4C_and_Supplementary_Figure_4F">Figure
4C</a></li>
<li><a
href="crispr-mm_2215_sgRNA-StealTHY.html#Figure_4D:_Proportion_of_non-expressed_hits">Figure
4D</a></li>
</ul>
</div>
<div id="figure-5" class="section level3">
<h3>Figure 5</h3>
<ul>
<li><a
href="crispr-hsapiens_2180_sgRNA.html#31_Figure_5H:_Muller_plots_for_huNSGs">Figure
5H</a></li>
</ul>
</div>
<div id="figure-6" class="section level3">
<h3>Figure 6</h3>
<ul>
<li><a
href="crispr-hsapiens_2180_sgRNA.html#sgRNA_distribution_by_condition_Primary_tumor">Figure
6B</a></li>
<li><a
href="crispr-hsapiens_2180_sgRNA.html#Figure_6C:_Volcano_plots_showing_genes_Lungs_over_Primary_Tumor_in_StealTHY_KO">Figure
6C</a></li>
<li><a
href="crispr-hsapiens_2180_sgRNA.html#Figure_6D:_Distribution_of_the_fold-change_of_Lungs_over_Primary_Tumor_in_huNSG_and_NSG_host">Figure
6D</a></li>
<li><a
href="tcga_survival_crispr_hits_signatures.html#Figure_6F:_Forest_plot_by_study">Figure
6E</a></li>
<li><a
href="tcga_survival_crispr_hits_signatures.html#Figure_6E:_Kaplan-Meier_plots">Figure
6F</a></li>
</ul>
</div>
<div id="figure-7" class="section level3">
<h3>Figure 7</h3>
<ul>
<li><a
href="rnaseq-amhr2_ko-deg.html#__531_Comparison_InVivo_AMHR2_KO_+_AMHR2_KO_culture%E2%80%93over%E2%80%93InVivo_Control_sgRNA_+_AMHR2_Control_culture_">Figure
7F</a></li>
<li><a
href="tcga_immune_infiltrate_amhr2_ko_signatures.html#4_Figure_7G:_Correlation_of_AMH_signatures_with_immune_cell_fractions_across_TCGA_cohorts_(Heatmaps)">Figure
7G</a></li>
<li><a
href="tcga_survival_amhr2_ko_signatures.html#Figure_7H_Right:_Pooled_Kaplan-Meier_plots_-_Optimal_cut-off">Figure
7H: Left panel</a></li>
<li><a
href="tcga_survival_amhr2_ko_signatures.html#Figure_7H_Left:_Forest_plot_by_study">Figure
7H: Right panel</a></li>
</ul>
</div>
</div>
<div id="supplementary-figures" class="section level2">
<h2>Supplementary figures</h2>
<div id="supplementary-figure-2" class="section level3">
<h3>Supplementary Figure 2</h3>
<ul>
<li><a
href="crispr-mm_2215_sgRNA-clonality.html#32_Supplementary_Figure_2B:_sgRNA_distribution_in_mice_transplanted_with_Py2T_and_MVT1_cells">Supplementary
Figure 2B</a></li>
<li><a
href="crispr-mm_2215_sgRNA-clonality.html#41_Supplementary_Figure_2C_left_:_K-S_test_table">Supplementary
Figure 2C-left</a></li>
<li><a
href="crispr-mm_2215_sgRNA-clonality.html#42_Supplementary_Figure_2C_right_:_Empirical_Cumulative_Density_Function_(ECDF)_plots">Supplementary
Figure 2C-right</a></li>
<li><a
href="crispr-mm_2215_sgRNA-clonality.html#21_Supplementary_Figure_2D_(left_upper)">Supplementary
Figure 2D: Lung cancer in dCas9 + Puro (left-upper panel)</a></li>
<li><a
href="crispr-mm_2215_sgRNA-clonality.html#22_Supplementary_Figure_2D_(right_upper)">Supplementary
Figure 2D: Lung cancer in Thy1 (right-upper panel)</a></li>
<li><a
href="crispr-mm_2215_sgRNA-clonality.html#Supplementary_Figure_2D:_Metastasis_-_Puro_(left_lower)">Supplementary
Figure 2D: Mammary cancer in Puro (left-lower panel)</a></li>
<li><a
href="crispr-mm_2215_sgRNA-clonality.html#Supplementary_Figure_2D:_Metastasis_-_Thy1_(right_lower)">Supplementary
Figure 2D: Mammary cancer in Thy1 (right-lower panel)</a></li>
<li><a
href="crispr-mm_2215_sgRNA-clonality.html#23_Supplementary_Figure_2F">Supplementary
Figure 2F: Right panel</a></li>
<li><a
href="rnaseq-tumor-bulk.html#Supplementary_Figure_2G_top:_Differential_expressed_genes_in_dCas9-Thy1_over_Thy1_transduction">Supplementary
Figure 2G: Upper panel</a></li>
<li><a
href="rnaseq-tumor-bulk.html#Supplementary_Figure_2G_bottom:_Representative_gene-sets_enriched_in_dCas9-Thy1_over_Thy1_transduction">Supplementary
Figure 2G: Lower panel</a></li>
<li><a
href="rnaseq-tumor-facs-immune.html#41_Supplementary_Figure_2H_top_left:_Differential_expressed_genes_in_dCas9-Thy1_over_Thy1_transduction_in_sorted_CD3-positive_TILs">Supplementary
Figure 2H: CD3+ TIL upper panel</a></li>
<li><a
href="rnaseq-tumor-facs-immune.html#51_Supplementary_Figure_2H_bottom_left:_Representative_gene-sets_enriched_in_dCas9-Thy1_over_Thy1_transduction">Supplementary
Figure 2H: CD3+ TIL lower panel</a></li>
<li><a
href="rnaseq-tumor-facs-cancer.html#41_Supplementary_Figure_2H_top_right:_Differential_expressed_genes_in_dCas9-Thy1_over_Thy1_transduction_in_sorted_carcinoma_cells">Supplementary
Figure 2H: Carcinoma cells upper panel</a></li>
<li><a
href="rnaseq-tumor-facs-cancer.html#51_Supplementary_Figure_2H_bottom_right:_Representative_gene-sets_enriched_in_dCas9-Thy1_over_Thy1_transduction">Supplementary
Figure 2H: Carcinoma cells lower panel</a></li>
</ul>
</div>
<div id="supplementary-figure-3" class="section level3">
<h3>Supplementary Figure 3</h3>
<ul>
<li><a
href="rnaseq-tumor-bulk.html#Supplementary_figure_3E_left:_Sample_distances_in_Py2T_model">Supplementary
Figure 3E: Py2T1 (left panel)</a></li>
<li><a
href="rnaseq-tumor-bulk.html#Supplementary_figure_3E_right:_Sample_distances_in_4T1_model">Supplementary
Figure 3E: 4T1 (right panel)</a></li>
</ul>
</div>
<div id="supplementary-figure-4" class="section level3">
<h3>Supplementary Figure 4</h3>
<ul>
<li><a
href="crispr-mm_2215_sgRNA-StealTHY.html#Supplementary_figure_4B:_sgRNA_distribution_in_mice_transplanted_with_4T1_cells_previously_subjected_to_StealTHY_KO">Supplementary
Figure 4B</a></li>
<li><a
href="crispr-mm_2215_sgRNA-clonality.html#7_Supplementary_Figure_4C">Supplementary
Figure 4C</a></li>
<li><a
href="crispr-mm_2215_sgRNA-StealTHY.html#Supplementary_Figure_4D_:_K-S_test_table">Supplementary
Figure 4D</a></li>
<li><a
href="crispr-mm_2215_sgRNA-StealTHY.html#Supplementary_Figure_4E">Supplementary
Figure 4E</a></li>
<li><a
href="crispr-mm_2215_sgRNA-StealTHY.html#Figure_4C_and_Supplementary_Figure_4F">Supplementary
Figure 4F</a></li>
<li><a
href="crispr-mm_2215_sgRNA-StealTHY.html#Supplementary_figure_4G:_Proportion_of_non-expressed_hits">Supplementary
Figure 4G</a></li>
<li><a
href="crispr-mm_2215_sgRNA-clonality.html#24_Supplementary_Figure_4L">Supplementary
Figure 4L</a></li>
<li><a
href="crispr-mm_2215_sgRNA-CMT167-LLC1-CT26.html#Supplementary_figure_4M:_Differential_abundance_of_selected_genes">Supplementary
Figure 4M</a></li>
<li><a
href="crispr-mm_2215_sgRNA-CMT167-LLC1-CT26.html#Supplementary_figure_4N:_Muller_plots_of_selected_genes">Supplementary
Figure 4N</a></li>
</ul>
</div>
<div id="supplementary-figure-6" class="section level3">
<h3>Supplementary Figure 6</h3>
<ul>
<li><a
href="crispr-hsapiens_2180_sgRNA.html#Supplementary_Figure_6D:_Fold-change_of_Lungs_over_Primary_Tumor_in_huNSG_and_NSG_host">Supplementary
Figure 6D</a></li>
<li><a
href="tcga_survival_crispr_hits_genes.html#Supplementary_Figure_6L:_Forest_plot_by_study">Supplementary
Figure 6L</a></li>
<li><a
href="tcga_survival_crispr_hits_genes.html#Supplementary_Figure_6E:_Kaplan-Meier_plots">Supplementary
Figure 6E</a></li>
</ul>
</div>
<div id="supplementary-figure-7" class="section level3">
<h3>Supplementary Figure 7</h3>
<ul>
<li><a
href="rnaseq-amhr2_ko-deg.html#Supplementary_Figure_7L">Supplementary
Figure 7L</a></li>
<li><a
href="rnaseq-amhr2_ko-deg.html#Supplementary_Figure_7M_(Galunisertib_positive)">Supplementary
Figure 7M: Positive Galunisertib</a></li>
<li><a
href="rnaseq-amhr2_ko-deg.html#Mammary_epithelium_proliferation30">Supplementary
Figure 7M: Negative Galunisertib</a></li>
<li><a
href="rnaseq-amhr2_ko-deg.html#EMT_regulators_and_targets">Supplementary
Figure 7P: Epithelial genes</a></li>
<li><a
href="rnaseq-amhr2_ko-deg.html#Mammary_epithelium_proliferation">Supplementary
Figure 7P: Proliferative genes</a></li>
<li><a
href="rnaseq-amhr2_ko-deg.html#Supplementary_Figure_7U_left">Supplementary
Figure 7U: Negative Galunisertib</a></li>
<li><a
href="rnaseq-amhr2_ko-deg.html#Supplementary_Figure_7U_right_(Galunisertib_positive)">Supplementary
Figure 7U: Positive Galunisertib</a></li>
</ul>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
