<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Francesc Castro-Giner" />

<meta name="date" content="2025-07-01" />

<title>Analysis of RNA-seq FACS sorted cells</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">StealTHY</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/TheAcetoLab/saini-stealTHY">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Analysis of RNA-seq FACS sorted cells</h1>
<h3 class="subtitle">Sorted carcinoma cells</h3>
<h4 class="author">Francesc Castro-Giner</h4>
<h4 class="date">July 01, 2025</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-07-01
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>saini-stealTHY/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240517code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240517)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240517code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240517)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomTheAcetoLabsainistealTHYtreebc402e6695932824d0587de6c03eb192be28dfcatargetblankbc402e6a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/tree/bc402e6695932824d0587de6c03eb192be28dfca" target="_blank">bc402e6</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomTheAcetoLabsainistealTHYtreebc402e6695932824d0587de6c03eb192be28dfcatargetblankbc402e6a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/TheAcetoLab/saini-stealTHY/tree/bc402e6695932824d0587de6c03eb192be28dfca" target="_blank">bc402e6</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    code/raw_data_processing/rnaseq/p27851_o32062/pipelines/
    Ignored:    configuration/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/crispr/
    Ignored:    data/resources/
    Ignored:    data/rnaseq/
    Ignored:    output/.DS_Store
    Ignored:    output/clinical/
    Ignored:    output/crispr/
    Ignored:    output/rnaseq/

Untracked files:
    Untracked:  analysis/crispr-hsapiens_2180_sgRNA_r1.Rmd
    Untracked:  analysis/crispr-mm_2215_sgRNA-CMT167-LLC1-CT26.Rmd
    Untracked:  analysis/crispr-mm_2215_sgRNA-clonality_r1.Rmd
    Untracked:  analysis/crispr-mm_2215_sgRNA-r2oD_r1.Rmd
    Untracked:  analysis/crispr-muller-mm_2215_sgRNA.Rmd
    Untracked:  analysis/templates/
    Untracked:  code/R-functions/subchunkify.R

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/crispr-hsapiens_2180_sgRNA.Rmd
    Modified:   analysis/crispr-mm_2215_sgRNA-StealTHY.Rmd
    Modified:   analysis/crispr-mm_2215_sgRNA-clonality.Rmd
    Modified:   analysis/crispr-mm_2215_sgRNA-r2oC.Rmd
    Modified:   code/R-functions/gse_report.r
    Modified:   configuration/rmarkdown/ggplot_theme.R
    Modified:   update_workflowr.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/rnaseq-tumor-facs-cancer.Rmd</code>) and HTML
(<code>docs/rnaseq-tumor-facs-cancer.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/bc402e6695932824d0587de6c03eb192be28dfca/analysis/rnaseq-tumor-facs-cancer.Rmd" target="_blank">bc402e6</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-07-01
</td>
<td>
uodate rnaseq analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/628f8ec8600fd6671cb7c7f57c8e717199ff7f75/analysis/rnaseq-tumor-facs-cancer.Rmd" target="_blank">628f8ec</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2024-05-24
</td>
<td>
added F4
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/abf695c6a645a3aa42ea9e2fb622b5f22e723b11/analysis/rnaseq-tumor-facs-cancer.Rmd" target="_blank">abf695c</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2024-05-23
</td>
<td>
done f2 anf f3
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries-additional-functions-and-data"
class="section level2" number="1">
<h2><span class="header-section-number">1</span> Load libraries,
additional functions and data</h2>
<p>Setup environment</p>
<pre class="r"><code>knitr::opts_chunk$set(results=&#39;asis&#39;, echo=TRUE, message=FALSE, warning=FALSE, error=FALSE, fig.align = &#39;center&#39;, fig.width = 3.5, fig.asp = 0.618, dpi = 600, dev = c(&quot;png&quot;, &quot;pdf&quot;), fig.showtext = FALSE, engine.opts = list(bash = &quot;-l&quot;))

options(stringsAsFactors = FALSE)

use_seed &lt;- 1100101
set.seed(use_seed)

dir.create(params$output_dir, recursive = TRUE, showWarnings = FALSE)</code></pre>
<p>Load packages</p>
<pre class="r"><code>library(tidyverse)
library(knitr)
library(foreach)
library(magrittr)
library(DT)
library(kableExtra)
library(SummarizedExperiment)

library(edgeR)
library(scuttle)
library(ComplexHeatmap)
library(circlize)
library(RColorBrewer)
library(org.Mm.eg.db)</code></pre>
<p>Load ggplot theme</p>
<pre class="r"><code>source(&quot;./configuration/rmarkdown/ggplot_theme.R&quot;)</code></pre>
<p>Load ggplot theme</p>
<pre class="r"><code>source(&quot;./configuration/rmarkdown/color_palettes.R&quot;)</code></pre>
<p>Load custom functions</p>
<pre class="r"><code>source(&#39;./code/R-functions/dge_wrappers.r&#39;)
source(&#39;./code/R-functions/dge_report.r&#39;)
source(&#39;./code/R-functions/gse_omnibus.r&#39;)
source(&#39;./code/R-functions/gse_report.r&#39;)
clean_msigdb_names &lt;- function(x) x %&gt;% gsub(&#39;REACTOME_&#39;, &#39;&#39;, .) %&gt;% gsub(&#39;WP_&#39;, &#39;&#39;, .) %&gt;% gsub(&#39;BIOCARTA_&#39;, &#39;&#39;, .) %&gt;% gsub(&#39;KEGG_&#39;, &#39;&#39;, .) %&gt;% gsub(&#39;PID_&#39;, &#39;&#39;, .) %&gt;% gsub(&#39;GOBP_&#39;, &#39;&#39;, .) %&gt;% gsub(&#39;_&#39;, &#39; &#39;, .)</code></pre>
<p>Load MSigDB files</p>
<pre class="r"><code>gmt_files_symbols &lt;- list(
  msigdb.m2.cp.m5.bp.custom.signatures = &#39;./data/resources/custom_signatures/msigdb_m2_cp-m5_bp-signatures_cancer_cells.gmt&#39;
)</code></pre>
<p>Clean files generated in previous runs</p>
<pre class="r"><code>rmd_file &lt;- current_input()
if(!is.null(rmd_file)) {
  figures_dir &lt;- file.path(&#39;./docs/figure&#39;,rmd_file)
  if(dir.exists(figures_dir)) {
    unlink(file.path(figures_dir, &quot;*&quot;))
  }
}</code></pre>
<p>Load Ensembl biomaRt mmusculus data</p>
<pre class="r"><code>if(file.exists(&#39;./data/resources/ensembl_biomart/mart_mmu_genes.rds&#39;)){
  ensembl_mart_mmu_genes &lt;- readRDS(&#39;./data/resources/ensembl_biomart/mart_mmu_genes.rds&#39;)
} else {
  # httr::set_config(httr::config(ssl_verifypeer = FALSE))
  ensembl_mart_mmu_genes &lt;-  biomaRt::useEnsembl(
    biomart=&quot;genes&quot;, dataset=&quot;mmusculus_gene_ensembl&quot;,
    mirror = &quot;useast&quot;
    )
  saveRDS(
    ensembl_mart_mmu_genes, 
    file = &#39;./data/resources/ensembl_biomart/mart_mmu_genes.rds&#39;
    )
}</code></pre>
</div>
<div id="data-wrangling" class="section level2" number="2">
<h2><span class="header-section-number">2</span> Data wrangling</h2>
<p>Load data</p>
<pre class="r"><code>se &lt;- readRDS(file.path(params$data_dir, params$se_file))
rowData(se[rownames(se) == &#39;CAS9&#39;,])$gene_name &lt;- &#39;SpCas9&#39;
rownames(se)[rownames(se) == &#39;CAS9&#39;] &lt;- &#39;SpCas9&#39;
se &lt;- se[,se$cell_type == &#39;tumor&#39;]
se &lt;- se[,se$sample_type == &#39;primary_tumor&#39;]</code></pre>
<p>Modify sample annotation</p>
<pre class="r"><code>colData(se) %&lt;&gt;% data.frame %&gt;%
  mutate(
    condition = NA,
    condition = ifelse(grepl(&#39;Thy1&#39;, sample_alias), &#39;Thy1&#39;, condition),
    condition = ifelse(grepl(&#39;Cas9&#39;, sample_alias), &#39;Cas9&#39;, condition)
  ) %&gt;%
  DataFrame</code></pre>
<p>Filtering out low-abundance features. Keeping those features with at
least 1 count in at least 30% of samples of the smallest group.</p>
<pre class="r"><code>use_rows &lt;- filterByExpr(se, group=paste(se$donor, se$condition), min.count = params$min_counts, min.prop = params$min_present_prop)
se &lt;- se[use_rows,]

# Update PerCellQCMetrics
colData(se) &lt;- colData(se) %&gt;% data.frame %&gt;% dplyr::select(-(sum:subsets_Ribo_percent)) %&gt;% DataFrame
se &lt;- addPerCellQCMetrics(
  se,
  subsets=list(
    Mito=which(rowData(se)$is.mito),
    Ribo=which(rowData(se)$is.ribo)
  ),
  percent.top = c(1,5,10,20, 50, 100),
  threshold = 3
  )</code></pre>
<p>TMM normalization</p>
<pre class="r"><code>norm_method &lt;- &quot;TMM&quot;
exprs_mat_for_norm &lt;- assay(se, &#39;counts&#39;)
norm_factors &lt;- calcNormFactors(exprs_mat_for_norm, method = norm_method)
lib_size &lt;- colSums(exprs_mat_for_norm)/10^6
se$size_factor &lt;- (norm_factors * lib_size)
se &lt;- logNormCounts(se, size.factors =se$size_factor)
se &lt;- logNormCounts(se, size.factors =se$size_factor, log = FALSE)</code></pre>
</div>
<div id="data-generation" class="section level2" number="3">
<h2><span class="header-section-number">3</span> Data generation</h2>
<p>Configure comparisons and arguments for differential expression</p>
<pre class="r"><code>x &lt;- colData(se) %&gt;% data.frame
comp_list &lt;- list(
  `dCas9_Thy1--over--Thy1` = list(
    `dCas9-Thy1` = x %&gt;% 
      filter(condition == &#39;Cas9&#39; &amp; treatment == &#39;no&#39;) %&gt;% 
      pull(sample_alias) %&gt;% 
      sort,
    Thy1 = x %&gt;% 
      filter(condition == &#39;Thy1&#39;) %&gt;% 
      pull(sample_alias) %&gt;% 
      sort,
    batch = &#39;donor&#39;
  )
)</code></pre>
<div id="differential-expression-analysis" class="section level3"
number="3.1">
<h3><span class="header-section-number">3.1</span> Differential
expression analysis</h3>
<p>Run differential expression analysis using EdgeR QLF for
multidonor-comparisons</p>
<pre class="r"><code>use_comp &lt;-  comp_list[[1]]
dge_list &lt;- foreach(use_comp = comp_list) %do%{
  se_cols &lt;- se$sample_alias %in% unlist(use_comp)
  use_se &lt;- se[,se_cols]
  use_se$group &lt;- ifelse(
    use_se$sample_alias %in% use_comp[[1]],
    names(use_comp)[1],
    names(use_comp)[2]
  )
  
  if(is.null(use_comp$batch)) {
    dge &lt;- edgeR_dge(
      use_se,
      # Design configuration for differential expression
      group_var =  &#39;group&#39;,
      group_sample = names(use_comp)[1],
      group_ref = names(use_comp)[2],
      design_formula = &quot;~ group&quot;,
      coef = &#39;last&#39;,
      # Conversion from SingleCellExperiment/Summarized Experiment to DGEList
      assay_to_DGEList = &#39;counts&#39;,
      assay_to_row_filter = &quot;counts&quot;,
      # Feature filtering parameters
      use_filterByExpr = TRUE,
      min_counts = params$min_counts,
      min_present_prop = params$min_present_prop,
      # EdgeR workflow configuration
      run_calcNormFactors = &#39;TMM&#39;,
      estimateDisp_robust = TRUE,
      estimateDisp_trend.method = &quot;locfit&quot;,
      glm_approach = &quot;QLF&quot;,
      # Output configuration
      adjust_method = &#39;BH&#39;
      )
    } else if(use_comp$batch == &#39;donor&#39;) {
      dge &lt;- edgeR_dge(
        use_se,
        # Design configuration for differential expression
        group_var =  &#39;group&#39;,
        group_sample = names(use_comp)[1],
        group_ref = names(use_comp)[2],
        batch_vars = &#39;donor&#39;,
        design_formula = &quot;~ donor + group&quot;,
        coef = &#39;last&#39;,
        # Conversion from SingleCellExperiment/Summarized Experiment to DGEList
        assay_to_DGEList = &#39;counts&#39;,
        assay_to_row_filter = &quot;counts&quot;,
        # Feature filtering parameters
        use_filterByExpr = TRUE,
        min_counts = params$min_counts,
        min_present_prop = params$min_present_prop,
        # EdgeR workflow configuration
        run_calcNormFactors = &#39;TMM&#39;,
        estimateDisp_robust = TRUE,
        estimateDisp_trend.method = &quot;locfit&quot;,
        glm_approach = &quot;QLF&quot;,
        # Output configuration
        adjust_method = &#39;BH&#39;
        )
    }
  
  
  # Add gene description
  gene_desc &lt;- biomaRt::getBM(attributes=c(&#39;external_gene_name&#39;,&#39;description&#39;),
                              filters = &#39;external_gene_name&#39;,
                              values = dge$results$gene_name, mart =ensembl_mart_mmu_genes) %&gt;%
    dplyr::rename(&#39;gene_name&#39; = &#39;external_gene_name&#39;) %&gt;%
    unique
  
  
  use_res &lt;- dge$results %&gt;%  left_join(., gene_desc)
  dge$results &lt;- use_res %&gt;% 
    filter(!duplicated(feature)) %&gt;% 
    mutate(rownames = feature) %&gt;% 
    column_to_rownames(&#39;rownames&#39;)
  
  return(dge)
}

names(dge_list) &lt;- names(comp_list)
detach(&quot;package:biomaRt&quot;, unload=TRUE)

# Save objects
saveRDS(dge_list, file = file.path(params$output_dir, &#39;dge-edgeR_QLF.rds&#39;))</code></pre>
</div>
<div id="functional-enrichment" class="section level3" number="3.2">
<h3><span class="header-section-number">3.2</span> Functional
enrichment</h3>
<pre class="r"><code>dge_list &lt;- readRDS(file.path(params$output_dir, &#39;dge-edgeR_QLF.rds&#39;))
dge &lt;-  dge_list[[1]]
gse_list &lt;- list()
gse_list &lt;- foreach(dge = dge_list) %do%{
  gse_res &lt;- gse_omnibus(
    feature_names = dge$results$gene_name,
    p = dge$results$FDR,
    fc = dge$results$logFC,
    fc_thrs = 0.5,
    gmt_files = gmt_files_symbols,
    
    run_GSEA = TRUE,
    
    args_gse = list(minGSSize = 10, maxGSSize = 500, pvalueCutoff = 1)
    )
  
   return(gse_res)

}
names(gse_list) &lt;- names(dge_list)

# Save objects
saveRDS(gse_list, file = file.path(params$output_dir,&#39;gse_omnibus-edgeR_QLF.rds&#39;))</code></pre>
</div>
<div id="load-data" class="section level3" number="3.3">
<h3><span class="header-section-number">3.3</span> Load data</h3>
<pre class="r"><code>dge_list &lt;- readRDS(file.path(params$output_dir, &#39;dge-edgeR_QLF.rds&#39;))
gse_list &lt;- readRDS(file.path(params$output_dir, &#39;gse_omnibus-edgeR_QLF.rds&#39;))</code></pre>
</div>
</div>
<div id="differential-expression-analysis-1" class="section level2"
number="4">
<h2><span class="header-section-number">4</span> Differential expression
analysis</h2>
<div
id="supplementary-figure-2h-top-right-differential-expressed-genes-in-dcas9-thy1-over-thy1-transduction-in-sorted-carcinoma-cells"
class="section level3" number="4.1">
<h3><span class="header-section-number">4.1</span> Supplementary Figure
2H top right: Differential expressed genes in dCas9-Thy1 over Thy1
transduction in sorted carcinoma cells</h3>
<p>Showing selected genes and genes with a FDR &lt; 0.05 and a absolute
log2 fold-change &gt; 1. To calculate the fold-change by model, log
normalized counts were aggregated at model and condition using their
average (function aggregateAcrossCells). Then the aggregated expression
in dCas9-Thy1 was divided by the aggregated expression in Thy1.</p>
<pre class="r"><code># Heatmap parameters
p_thrs &lt;- 0.05
lfc_thrs &lt;- 1
# topn &lt;- 50
max_color_range &lt;- 3 # put the color range to -3 to 3

# Select comparison
i &lt;- &quot;dCas9_Thy1--over--Thy1&quot;
use_comp &lt;- comp_list[[i]]

# Select genes to highlight
use_labels &lt;- c(&quot;Acta2&quot;,&quot;Vim&quot;,&quot;H2-Ab1&quot;,&quot;H2-K1&quot;,&quot;H2-K2&quot;,&quot;Cd74&quot;,&quot;B2m&quot;,
                &quot;Tap1&quot;,&quot;Tap2&quot;,&quot;SpCas9&quot;,&quot;H2-Q5&quot;,&quot;H2-Q10&quot;,&quot;Ifitm1&quot;,&quot;Oas1g&quot;,
                &quot;Irf1&quot;,&quot;Eif3a&quot;,&quot;Eif2b3&quot;,&quot;Dicer1&quot;,&quot;Drosha&quot;)


# Extract samples from comparisons
use_samples &lt;- unlist(use_comp[1:2])

# Add dge_groups annotation
colData(se) &lt;- colData(se) %&gt;% data.frame %&gt;% 
  mutate(
    condition_new = ifelse(sample_alias %in% use_comp[[1]], 
                           names(use_comp)[1], 
                           NA),
    condition_new = ifelse(sample_alias %in% use_comp[[2]], 
                           names(use_comp)[2], 
                           condition_new),
    condition_new = factor(condition_new, levels = names(use_comp)[2:1]),
    condition_num = as.numeric(condition_new)
  ) %&gt;% DataFrame

# Average SE by condition
group_id &lt;- paste(se$donor, se$condition_new)

use_se &lt;- scuttle::aggregateAcrossCells(
  se, 
  statistics = &quot;mean&quot;, 
  subset.col = use_samples,
  ids = group_id,
  use.assay.type = &#39;logcounts&#39;
  )


# Arrange samples by condition
use_annot &lt;- colData(use_se) %&gt;% data.frame %&gt;% 
  arrange(condition_num, donor)
use_samples &lt;- use_annot$ids

# Extract DGE results
use_dge &lt;- dge_list[[i]]$results %&gt;% 
  mutate(
    gene_name = ifelse(feature == &#39;CAS9&#39;, &#39;CAS9&#39;, gene_name),
    significant = (FDR &lt; p_thrs &amp; abs(logFC) &gt; lfc_thrs),
    selected = ifelse(gene_name %in% use_labels, TRUE, significant),
    direction = ifelse(logFC &gt; 0, &#39;Up&#39;, &#39;Down&#39;)
    ) %&gt;% 
  filter(selected) %&gt;% 
  arrange(FDR)

# Extract matrix of expression
use_se &lt;- use_se[rownames(use_dge), use_samples]
mat &lt;- assay(use_se, &#39;logcounts&#39;)
rownames(mat) &lt;- use_dge$gene_name

# Calculate fold-change by donor
donor_levels &lt;- unique(use_annot$donor)
fcmat &lt;- foreach(use_donor = donor_levels, .combine = cbind) %do% {
  s1 &lt;- use_annot %&gt;% 
    filter(donor == use_donor) %&gt;% 
    filter(condition_new == names(use_comp)[[1]]) %&gt;% 
    pull(ids)
  s2 &lt;- use_annot %&gt;% 
    filter(donor == use_donor) %&gt;% 
    filter(condition_new == names(use_comp)[[2]]) %&gt;% 
    pull(ids)
  return(mat[,s1]/mat[,s2])
}
colnames(fcmat) &lt;- donor_levels
fcmat[is.na(fcmat)] &lt;- 1
fcmat[fcmat == Inf] &lt;- max(fcmat[fcmat &lt; Inf])
fcmat[fcmat == 0] &lt;- min(fcmat[fcmat &gt; 0])
fcmat &lt;- log2(fcmat)


# Row annotation for selected labels
row_annot_selected_labels &lt;- rowAnnotation(
  selected = anno_mark(
    at = match(use_labels, rownames(fcmat)), 
    labels =use_labels,
    labels_gp = gpar(fontsize = 6),
    lines_gp = gpar(lwd = 0.3),
    padding = unit(0.01, &quot;inches&quot;),
    link_width = unit(0.1, &quot;inches&quot;)
    )
  )


# Heatmap color
zmax &lt;- fcmat %&gt;% abs %&gt;% max
if(!is.null(max_color_range))
  zmax &lt;- max_color_range
col_fun &lt;-  colorRamp2(
  seq(-zmax, zmax, length.out = 11),
  rev(brewer.pal(11, &quot;RdBu&quot;))
)


# Generate Heatmap
htz &lt;- Heatmap(
  fcmat,
  name= &#39;log2(fold change)&#39;,
  col = col_fun,
  show_row_names = FALSE,
  show_column_names = TRUE,
  # row_names_gp = gpar(fontsize = 8),
  column_names_gp = gpar(fontsize = 3),
  column_names_rot = 0,
  column_names_side = &#39;top&#39;,
  cluster_column_slices = FALSE,
  
  # top_annotation = column_ha,
  column_split = colnames(fcmat),
  column_gap = unit(0.039/3, &quot;inches&quot;),
  column_title = NULL,
  cluster_row_slices = FALSE,
  
  right_annotation = row_annot_selected_labels,
  row_split = use_dge$direction,
  row_gap = unit(0.039/2, &quot;inches&quot;),
  # left_annotation = row_annot_left,
  row_title = NULL,
  
  cluster_columns = FALSE,
  show_row_dend = FALSE,
  
  # show_heatmap_legend = FALSE,
  heatmap_legend_param = list(
    direction = &quot;horizontal&quot;,
    title_gp = gpar(fontsize = 3, fontface = &quot;bold&quot;),
    labels_gp = gpar(fontsize = 3),
    legend_width = unit(0.5, &quot;inches&quot;),
    grid_height = unit(0.03, &quot;inches&quot;),
    title_position = &#39;lefttop&#39;
  )
)

draw(htz,  heatmap_legend_side = &quot;bottom&quot;)</code></pre>
<p><img src="figure/rnaseq-tumor-facs-cancer.Rmd/dge-heatmap-fc-dc9thy1--over--thy1-1.png" width="600" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cat(&quot;\n\n&quot;)</code></pre>
</div>
</div>
<div id="functional-enrichment-1" class="section level2" number="5">
<h2><span class="header-section-number">5</span> Functional
Enrichment</h2>
<div
id="supplementary-figure-2h-bottom-right-representative-gene-sets-enriched-in-dcas9-thy1-over-thy1-transduction"
class="section level3" number="5.1">
<h3><span class="header-section-number">5.1</span> Supplementary Figure
2H bottom right: Representative gene-sets enriched in dCas9-Thy1 over
Thy1 transduction</h3>
<pre class="r"><code>i &lt;- &quot;dCas9_Thy1--over--Thy1&quot;
x &lt;- gse_list[[i]]$GSEA[[&quot;msigdb.m2.cp.m5.bp.custom.signatures&quot;]]@result
use_sets &lt;- c(
  `Mesenchymal differentiation` = &#39;GOBP_MESENCHYMAL_CELL_DIFFERENTIATION&#39;,
  `Peptide antigen assembly via MHC protein complex` = &#39;GOBP_PEPTIDE_ANTIGEN_ASSEMBLY_WITH_MHC_PROTEIN_COMPLEX&#39;,
  `Response to type I interferon` = &#39;GOBP_RESPONSE_TO_INTERFERON_BETA&#39;,
  `Defense response to bacterium` = &#39;GOBP_DEFENSE_RESPONSE_TO_BACTERIUM&#39;,
  `Antigen processing and presentation` = &#39;GOBP_ANTIGEN_PROCESSING_AND_PRESENTATION&#39;,
  `Crosslinking of collagen` = &#39;REACTOME_CROSSLINKING_OF_COLLAGEN_FIBRILS&#39;,
  `Cytoplasmic translation` = &#39;GOBP_CYTOPLASMIC_TRANSLATIONAL_INITIATION&#39;,
  `Translation initiation` = &#39;REACTOME_EUKARYOTIC_TRANSLATION_INITIATION&#39;,
  `Ribosome subunit assembly` = &#39;GOBP_RIBOSOMAL_LARGE_SUBUNIT_ASSEMBLY&#39;,
  `Ribosome large subunit biogenesis` = &#39;GOBP_RIBOSOME_BIOGENESIS&#39;
)


use_gse_res &lt;- gse_list[[i]]$GSEA[[&quot;msigdb.m2.cp.m5.bp.custom.signatures&quot;]]@result[use_sets,]
use_gse_res$ID &lt;- factor(use_gse_res$ID, levels = rev(use_sets))
levels(use_gse_res$ID) &lt;- rev(names(use_sets))


label_func &lt;- default_labeller(30)
x_max &lt;- range(use_gse_res$NES) %&gt;% abs %&gt;% max
size_max &lt;- max(-log10(use_gse_res$p.adjust))
res &lt;- use_gse_res %&gt;% 
  ggplot(aes(NES, ID, color = NES, size = -log10(p.adjust))) +
  geom_point() +
  geom_vline(xintercept = 0, linetype = &#39;dotted&#39;, linewidth = one_pt/4) +
  scale_y_discrete(labels = label_func) +
  scale_x_continuous(
    expand = expansion(mult = c(.1, .1)),
    limits = c(-x_max, x_max)
    )+
  scale_size(range = c(0.1, 2), 
             limits = c(0,size_max)) +
  scale_color_distiller(
    palette = &quot;RdBu&quot;,
    limits = c(-x_max, x_max)
  ) +
  labs(
    x = &#39;NES&#39;,
    y = &#39;&#39;,
    size =  bquote(&quot;-log&quot;[10] ~ .(paste0(&quot;(&quot;, &#39;Adjusted P&#39;, &quot;)&quot;))),
    color = &#39;NES&#39;
  ) +
    theme(
      strip.background = element_blank(),
      axis.line = element_line(linewidth = one_pt/4, color = &#39;black&#39;),
      axis.ticks = element_line(linewidth = one_pt/4, color = &#39;black&#39;),
      axis.text = element_text(size=8), 
      axis.title = element_text(size = 8),
      legend.title = element_text(size=8),
      legend.text = element_text(size=8),
      legend.position=&quot;bottom&quot;,
      legend.box = &#39;vertical&#39;
      # legend.key.size = unit(0.2, &#39;inches&#39;),
      # legend.key.width = unit(0.05, &#39;inches&#39;),
      # legend.key.height = unit(0.5, &#39;inches&#39;)
    )
res_legend &lt;- get_legend(res)
print(res + theme(legend.position = &#39;none&#39;))</code></pre>
<p><img src="figure/rnaseq-tumor-facs-cancer.Rmd/gsea-curated-dot-plot-dc9thy1--over--thy1-1.png" width="1422" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cat(&quot;\n\n&quot;)</code></pre>
<pre class="r"><code>plot_grid(res_legend, ncol = 1, nrow = 1)</code></pre>
<p><img src="figure/rnaseq-tumor-facs-cancer.Rmd/gsea-curated-dot-plot-legend-dc9thy1--over--thy1-1.png" width="1422" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cat(&quot;\n\n&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<p>R version 4.4.3 (2025-02-28) Platform: aarch64-apple-darwin20 Running
under: macOS Sequoia 15.5</p>
<p>Matrix products: default BLAS:
/Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib
LAPACK:
/Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;
LAPACK version 3.12.0</p>
<p>locale: [1]
en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</p>
<p>time zone: Europe/Zurich tzcode source: internal</p>
<p>attached base packages: [1] grid stats4 stats graphics grDevices
utils datasets [8] methods base</p>
<p>other attached packages: [1] cowplot_1.1.3.9000
org.Mm.eg.db_3.20.0<br />
[3] AnnotationDbi_1.68.0 RColorBrewer_1.1-3<br />
[5] circlize_0.4.16 ComplexHeatmap_2.22.0<br />
[7] scuttle_1.16.0 SingleCellExperiment_1.28.1 [9] edgeR_4.4.2
limma_3.62.2<br />
[11] SummarizedExperiment_1.36.0 Biobase_2.66.0<br />
[13] GenomicRanges_1.58.0 GenomeInfoDb_1.42.3<br />
[15] IRanges_2.40.1 S4Vectors_0.44.0<br />
[17] BiocGenerics_0.52.0 MatrixGenerics_1.18.1<br />
[19] matrixStats_1.5.0 kableExtra_1.4.0<br />
[21] DT_0.33 magrittr_2.0.3<br />
[23] foreach_1.5.2 knitr_1.50<br />
[25] lubridate_1.9.4 forcats_1.0.0<br />
[27] stringr_1.5.1 dplyr_1.1.4<br />
[29] purrr_1.0.4 readr_2.1.5<br />
[31] tidyr_1.3.1 tibble_3.2.1<br />
[33] ggplot2_3.5.2 tidyverse_2.0.0<br />
[35] workflowr_1.7.1</p>
<p>loaded via a namespace (and not attached): [1] DBI_1.2.3 rlang_1.1.6
clue_0.3-66<br />
[4] GetoptLong_1.0.5 git2r_0.36.2 RSQLite_2.3.11<br />
[7] compiler_4.4.3 getPass_0.2-4 png_0.1-8<br />
[10] systemfonts_1.2.3 callr_3.7.6 vctrs_0.6.5<br />
[13] shape_1.4.6.1 pkgconfig_2.0.3 crayon_1.5.3<br />
[16] fastmap_1.2.0 magick_2.8.6 XVector_0.46.0<br />
[19] labeling_0.4.3 promises_1.3.2 rmarkdown_2.29<br />
[22] tzdb_0.5.0 UCSC.utils_1.2.0 ps_1.9.1<br />
[25] bit_4.6.0 xfun_0.52 zlibbioc_1.52.0<br />
[28] cachem_1.1.0 beachmat_2.22.0 jsonlite_2.0.0<br />
[31] blob_1.2.4 later_1.4.2 DelayedArray_0.32.0<br />
[34] BiocParallel_1.40.2 cluster_2.1.8.1 parallel_4.4.3<br />
[37] R6_2.6.1 bslib_0.9.0 stringi_1.8.7<br />
[40] jquerylib_0.1.4 Rcpp_1.0.14 iterators_1.0.14<br />
[43] httpuv_1.6.16 Matrix_1.7-3 timechange_0.3.0<br />
[46] tidyselect_1.2.1 rstudioapi_0.17.1 dichromat_2.0-0.1<br />
[49] abind_1.4-8 yaml_2.3.10 doParallel_1.0.17<br />
[52] codetools_0.2-20 processx_3.8.6 lattice_0.22-7<br />
[55] KEGGREST_1.46.0 withr_3.0.2 evaluate_1.0.3<br />
[58] xml2_1.3.8 Biostrings_2.74.1 pillar_1.10.2<br />
[61] whisker_0.4.1 generics_0.1.4 rprojroot_2.0.4<br />
[64] hms_1.1.3 scales_1.4.0 glue_1.8.0<br />
[67] tools_4.4.3 locfit_1.5-9.12 fs_1.6.6<br />
[70] Cairo_1.6-2 colorspace_2.1-1 GenomeInfoDbData_1.2.13 [73] cli_3.6.5
textshaping_1.0.1 S4Arrays_1.6.0<br />
[76] viridisLite_0.4.2 svglite_2.2.1 gtable_0.3.6<br />
[79] sass_0.4.10 digest_0.6.37 SparseArray_1.6.2<br />
[82] rjson_0.2.23 htmlwidgets_1.6.4 farver_2.1.2<br />
[85] memoise_2.0.1 htmltools_0.5.8.1 lifecycle_1.0.4<br />
[88] httr_1.4.7 GlobalOptions_0.1.2 statmod_1.5.0<br />
[91] bit64_4.6.0-1</p>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4,h5",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
