<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Francesc Castro-Giner" />

<meta name="date" content="2025-10-31" />

<title>Analysis of AMHR2 KO RNA-seq data</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.34.0/datatables.js"></script>
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<script src="site_libs/jszip-1.13.6/jszip.min.js"></script>
<link href="site_libs/dt-ext-buttons-1.13.6/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.13.6/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.6/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.6/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.6/js/buttons.print.min.js"></script>
<link href="site_libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet" />
<script src="site_libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="site_libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
<script src="site_libs/selectize-0.12.0/selectize.min.js"></script>
<link href="site_libs/crosstalk-1.2.2/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.2/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">StealTHY</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="about.html">
    <span class="fa fa-info"></span>
     
    About
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-bar-chart"></span>
     
    Analyses
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Data generation</li>
    <li>
      <a href="0_differential_expression_gsea_gsva.html">Run differential gene expression analysis, GSEA and GSVA</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Main results</li>
    <li>
      <a href="patients_ctc_counts_distribution.html">CTC counts distribution in breast cancer patients</a>
    </li>
    <li>
      <a href="br16_pca.html">PCA analysis of CTC gene expression in NSG-CDX-BR16 mice</a>
    </li>
    <li>
      <a href="br16_dge.html">Differential gene expression report for NSG-CDX-BR16 data</a>
    </li>
    <li>
      <a href="gsea_across_models.html">Report of GSEA and GSVA analysis in NSG-CDX-BR16, NSG-LM2, Patient and NSG-LM2 time kinetics</a>
    </li>
    <li>
      <a href="core_gene_sets.html">Analysis of genes from candidate pathways</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="https://github.com/TheAcetoLab/saini-stealTHY">
    <span class="fa fa-github"></span>
     
  </a>
</li>
<li>
  <a href="https://www.theacetolab.com">
    <span class="fa fa-flask"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Analysis of AMHR2 KO RNA-seq data</h1>
<h4 class="author">Francesc Castro-Giner</h4>
<h4 class="date">October 31, 2025</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-10-31
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>saini-stealTHY/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240517code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240517)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240517code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240517)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomTheAcetoLabsainistealTHYtree455fa9121fcca51363cf639f7ee170fb108e34c1targetblank455fa91a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/tree/455fa9121fcca51363cf639f7ee170fb108e34c1" target="_blank">455fa91</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomTheAcetoLabsainistealTHYtree455fa9121fcca51363cf639f7ee170fb108e34c1targetblank455fa91a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/TheAcetoLab/saini-stealTHY/tree/455fa9121fcca51363cf639f7ee170fb108e34c1" target="_blank">455fa91</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/templates/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    code/raw_data_processing/rnaseq/p27851_o32062/pipelines/
    Ignored:    configuration/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/crispr/
    Ignored:    data/resources/
    Ignored:    data/rnaseq/
    Ignored:    output/.DS_Store
    Ignored:    output/clinical/
    Ignored:    output/crispr/
    Ignored:    output/rnaseq/

Untracked files:
    Untracked:  analysis/crispr-hsapiens_2180_sgRNA.md
    Untracked:  analysis/crispr-mm_2215_sgRNA-CMT167-LLC1-CT26.md
    Untracked:  analysis/crispr-mm_2215_sgRNA-StealTHY.md
    Untracked:  analysis/crispr-mm_2215_sgRNA-clonality.md
    Untracked:  analysis/index.md
    Untracked:  analysis/rnaseq-tumor-bulk.md
    Untracked:  analysis/style.css
    Untracked:  analysis/tcga_survival_crispr_hits_signatures.md
    Untracked:  code/R-functions/subchunkify.R

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/_site.yml
    Deleted:    analysis/crispr-hsapiens_2180_sgRNA_r1.Rmd
    Deleted:    analysis/crispr-mm_2215_sgRNA-clonality_r1.Rmd
    Deleted:    analysis/crispr-mm_2215_sgRNA-r2oC.Rmd
    Deleted:    analysis/crispr-mm_2215_sgRNA-r2oD.Rmd
    Deleted:    analysis/crispr-mm_2215_sgRNA-r2oD_r1.Rmd
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_ko_culture--over--amhr2_control_culture_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_ko_culture--over--amhr2_control_culture_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_ko_notgfbeta--over--amhr2_ko_culture_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_ko_notgfbeta--over--amhr2_ko_culture_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_overexpression--over--amhr2_control_culture_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_overexpression--over--amhr2_control_culture_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_overexpression--over--amhr2_ko_culture_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_overexpression--over--amhr2_ko_culture_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_overexpression--over--amhr2_ko_notgfbeta_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_overexpression--over--amhr2_ko_notgfbeta_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/figure/gse-dotplot-invivo_amhr2_ko-+-amhr2_ko_culture--over--invivo_control_sgrna-+-amhr2_control_culture_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-invivo_amhr2_ko-+-amhr2_ko_culture--over--invivo_control_sgrna-+-amhr2_control_culture_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/figure/gse-dotplot-invivo_amhr2_ko--over--invivo_control_sgrna_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-invivo_amhr2_ko--over--invivo_control_sgrna_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/file/tmp/dge.xlsx
    Deleted:    analysis/templates/docs/file/tmp/gse_gsea_msigdb.h.xlsx
    Deleted:    analysis/templates/docs/file/tmp/gse_gsego_bp.xlsx
    Deleted:    analysis/templates/docs/file/tmp/gse_gsego_mf.xlsx
    Modified:   analysis/templates/rnaseq-deg-edger.Rmd
    Modified:   code/R-functions/gse_report.r
    Modified:   configuration/rmarkdown/color_palettes.R
    Modified:   configuration/rmarkdown/ggplot_theme.R
    Modified:   update_workflowr.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/rnaseq-amhr2_ko-deg.Rmd</code>)
and HTML (<code>docs/rnaseq-amhr2_ko-deg.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/455fa9121fcca51363cf639f7ee170fb108e34c1/analysis/rnaseq-amhr2_ko-deg.Rmd" target="_blank">455fa91</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-10-31
</td>
<td>
Prepare files for publication
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheAcetoLab/saini-stealTHY/8632845b1c6b024330836a1a798537363e31980e/docs/rnaseq-amhr2_ko-deg.html" target="_blank">8632845</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-07-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/bc402e6695932824d0587de6c03eb192be28dfca/analysis/rnaseq-amhr2_ko-deg.Rmd" target="_blank">bc402e6</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-07-01
</td>
<td>
uodate rnaseq analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries-additional-functions-and-data"
class="section level2">
<h2>Load libraries, additional functions and data</h2>
<p>Setup environment</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">results=</span><span class="st">&#39;asis&#39;</span>, <span class="at">echo=</span><span class="cn">TRUE</span>, <span class="at">message=</span><span class="cn">FALSE</span>, <span class="at">warning=</span><span class="cn">FALSE</span>, <span class="at">error=</span><span class="cn">FALSE</span>, <span class="at">fig.align =</span> <span class="st">&#39;center&#39;</span>, <span class="at">fig.width =</span> <span class="fl">3.5</span>, <span class="at">fig.asp =</span> <span class="fl">0.618</span>, <span class="at">dpi =</span> <span class="dv">600</span>, <span class="at">dev =</span> <span class="fu">c</span>(<span class="st">&quot;png&quot;</span>, <span class="st">&quot;pdf&quot;</span>), <span class="at">engine.opts =</span> <span class="fu">list</span>(<span class="at">bash =</span> <span class="st">&quot;-l&quot;</span>))</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">options</span>(<span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>use_seed <span class="ot">&lt;-</span> <span class="dv">1100101</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="fu">set.seed</span>(use_seed)</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="fu">dir.create</span>(params<span class="sc">$</span>output_dir, <span class="at">recursive =</span> <span class="cn">TRUE</span>, <span class="at">showWarnings =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p>Load packages</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">library</span>(showtext)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="fu">library</span>(foreach)</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="fu">library</span>(DT)</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="fu">library</span>(colorblindr)</span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a><span class="fu">library</span>(ggbeeswarm)</span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a><span class="fu">library</span>(magrittr)</span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a><span class="fu">library</span>(openxlsx)</span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a></span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a><span class="fu">library</span>(edgeR)</span>
<span id="cb2-17"><a href="#cb2-17" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb2-18"><a href="#cb2-18" tabindex="-1"></a><span class="fu">library</span>(scuttle)</span>
<span id="cb2-19"><a href="#cb2-19" tabindex="-1"></a><span class="fu">library</span>(scran)</span>
<span id="cb2-20"><a href="#cb2-20" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb2-21"><a href="#cb2-21" tabindex="-1"></a><span class="fu">library</span>(PCAtools)</span>
<span id="cb2-22"><a href="#cb2-22" tabindex="-1"></a><span class="fu">library</span>(org.Mm.eg.db)</span>
<span id="cb2-23"><a href="#cb2-23" tabindex="-1"></a><span class="fu">library</span>(ComplexHeatmap)</span>
<span id="cb2-24"><a href="#cb2-24" tabindex="-1"></a><span class="fu">library</span>(circlize)</span></code></pre></div>
<p>Load ggplot theme</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;./configuration/rmarkdown/ggplot_theme.R&quot;</span>)</span></code></pre></div>
<p>Set color palettes</p>
<p>Load custom functions</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&#39;./code/R-functions/dge_wrappers.r&#39;</span>)</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&#39;./code/R-functions/dge_report.r&#39;</span>)</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&#39;./code/R-functions/gse_omnibus.r&#39;</span>)</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&#39;./code/R-functions/gse_report.r&#39;</span>)</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&#39;./code/R-functions/subchunkify.R&#39;</span>)</span></code></pre></div>
<p>Load MSigDB gene sets</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>gmt_files_symbols <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>  <span class="at">msigdb.h =</span> <span class="fu">paste0</span>(params<span class="sc">$</span>msigdb_dir, <span class="st">&#39;/&#39;</span>, params<span class="sc">$</span>msigdb_vs, <span class="st">&#39;/mh.all.&#39;</span>, params<span class="sc">$</span>msigdb_vs, <span class="st">&#39;.symbols.gmt&#39;</span>)</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>)</span></code></pre></div>
<p>Clean files generated in previous runs</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>rmd_file <span class="ot">&lt;-</span> <span class="fu">current_input</span>()</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(rmd_file)) {</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>  figures_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">&#39;./docs/figure&#39;</span>,rmd_file)</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">dir.exists</span>(figures_dir)) {</span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a>    <span class="fu">unlink</span>(<span class="fu">file.path</span>(figures_dir, <span class="st">&quot;*&quot;</span>))</span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a>  }</span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="configure-analyses" class="section level2">
<h2>Configure analyses</h2>
<div id="data-wrangling" class="section level3">
<h3>Data wrangling</h3>
<p>Load SE object</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(params<span class="sc">$</span>se_path)</span></code></pre></div>
<p>Modify sample annotation</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="fu">colData</span>(se) <span class="sc">%&lt;&gt;%</span> </span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>  data.frame <span class="sc">%&gt;%</span> </span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>    <span class="at">condition =</span> sample_alias <span class="sc">%&gt;%</span> <span class="fu">gsub</span>(<span class="st">&#39;_rep.&#39;</span>, <span class="st">&#39;&#39;</span>, .),</span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>    <span class="at">condition_short =</span> <span class="fu">case_match</span>(</span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a>      condition,</span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a>      <span class="st">&quot;AMHR2_Control_culture&quot;</span> <span class="sc">~</span> <span class="st">&quot;c_AMHR2_ctrl&quot;</span>,</span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a>      <span class="st">&quot;AMHR2_KO_culture&quot;</span> <span class="sc">~</span> <span class="st">&quot;c_AMHR2_KO&quot;</span>,</span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a>      <span class="st">&quot;AMHR2_KO_noTGFbeta&quot;</span> <span class="sc">~</span> <span class="st">&quot;AMHR2_KO_noTGFb&quot;</span>,</span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a>      <span class="st">&quot;AMHR2_Overexpression&quot;</span> <span class="sc">~</span> <span class="st">&quot;AMHR2_over&quot;</span>,</span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a>      <span class="st">&quot;InVivo_AMHR2_KO&quot;</span> <span class="sc">~</span> <span class="st">&quot;i_AMHR2_KO&quot;</span>,</span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a>      <span class="st">&quot;InVivo_Control_sgRNA&quot;</span> <span class="sc">~</span> <span class="st">&quot;i_ctrl_sgRNA&quot;</span></span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a>    ),</span>
<span id="cb8-14"><a href="#cb8-14" tabindex="-1"></a>    <span class="at">sample_type =</span> condition_short,</span>
<span id="cb8-15"><a href="#cb8-15" tabindex="-1"></a>    <span class="at">condition =</span> <span class="fu">factor</span>(condition),</span>
<span id="cb8-16"><a href="#cb8-16" tabindex="-1"></a>    <span class="at">condition_short =</span> <span class="fu">factor</span>(condition_short)</span>
<span id="cb8-17"><a href="#cb8-17" tabindex="-1"></a></span>
<span id="cb8-18"><a href="#cb8-18" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> DataFrame</span></code></pre></div>
<p>Filtering out low-abundance features. Keeping those features with at
least 10 counts in at least 50% of samples of the smallest group.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>use_rows <span class="ot">&lt;-</span> <span class="fu">filterByExpr</span>(se, <span class="at">group=</span>se<span class="sc">$</span>condition, <span class="at">min.count =</span> <span class="dv">10</span>, <span class="at">min.total.count =</span> <span class="dv">15</span>, <span class="at">min.prop =</span> <span class="fl">0.5</span>)</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>se <span class="ot">&lt;-</span> se[use_rows,]</span></code></pre></div>
<p>TMM normalization</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>norm_method <span class="ot">&lt;-</span> <span class="st">&quot;TMM&quot;</span></span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>exprs_mat_for_norm <span class="ot">&lt;-</span> <span class="fu">assay</span>(se, <span class="st">&#39;counts&#39;</span>)</span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>norm_factors <span class="ot">&lt;-</span> <span class="fu">calcNormFactors</span>(exprs_mat_for_norm, <span class="at">method =</span> norm_method)</span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a>lib_size <span class="ot">&lt;-</span> <span class="fu">colSums</span>(exprs_mat_for_norm)<span class="sc">/</span><span class="dv">10</span><span class="sc">^</span><span class="dv">6</span></span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a>se<span class="sc">$</span>size_factor <span class="ot">&lt;-</span> (norm_factors <span class="sc">*</span> lib_size)</span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">logNormCounts</span>(se, <span class="at">size.factors =</span>se<span class="sc">$</span>size_factor)</span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">logNormCounts</span>(se, <span class="at">size.factors =</span>se<span class="sc">$</span>size_factor, <span class="at">log =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p>CPM calculation</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="fu">assay</span>(se, <span class="st">&#39;cpm&#39;</span>) <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">cpm</span>(<span class="fu">assay</span>(se, <span class="st">&#39;normcounts&#39;</span>))</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a><span class="fu">assay</span>(se, <span class="st">&#39;logcpm&#39;</span>) <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">cpm</span>(<span class="fu">assay</span>(se, <span class="st">&#39;normcounts&#39;</span>), <span class="at">log =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="define-of-comparisons" class="section level3">
<h3>Define of comparisons</h3>
<div id="process-data" class="section level4">
<h4>Process data</h4>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">colData</span>(se) <span class="sc">%&gt;%</span> data.frame</span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a></span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a><span class="co"># Define groups</span></span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a>use_levels <span class="ot">&lt;-</span> <span class="fu">levels</span>(se<span class="sc">$</span>condition_short)</span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a>i <span class="ot">&lt;-</span> use_levels[<span class="dv">1</span>]</span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a>group_samples <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i =</span> use_levels) <span class="sc">%do%</span> {</span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a>  use_cols <span class="ot">&lt;-</span> se<span class="sc">$</span>condition_short <span class="sc">==</span> i</span>
<span id="cb12-8"><a href="#cb12-8" tabindex="-1"></a>  <span class="fu">colnames</span>(se)[use_cols]</span>
<span id="cb12-9"><a href="#cb12-9" tabindex="-1"></a>}</span>
<span id="cb12-10"><a href="#cb12-10" tabindex="-1"></a><span class="fu">names</span>(group_samples) <span class="ot">&lt;-</span> use_levels</span>
<span id="cb12-11"><a href="#cb12-11" tabindex="-1"></a></span>
<span id="cb12-12"><a href="#cb12-12" tabindex="-1"></a></span>
<span id="cb12-13"><a href="#cb12-13" tabindex="-1"></a><span class="co"># Define all potential pairwise comparisons</span></span>
<span id="cb12-14"><a href="#cb12-14" tabindex="-1"></a>group_combn <span class="ot">&lt;-</span> <span class="fu">combn</span>(<span class="fu">names</span>(group_samples) , <span class="dv">2</span>, <span class="at">simplify =</span> <span class="cn">FALSE</span>)</span>
<span id="cb12-15"><a href="#cb12-15" tabindex="-1"></a></span>
<span id="cb12-16"><a href="#cb12-16" tabindex="-1"></a>i <span class="ot">&lt;-</span> group_combn[[<span class="dv">1</span>]]</span>
<span id="cb12-17"><a href="#cb12-17" tabindex="-1"></a>comp_list <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i =</span> group_combn) <span class="sc">%do%</span>{</span>
<span id="cb12-18"><a href="#cb12-18" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb12-19"><a href="#cb12-19" tabindex="-1"></a>    group_samples[[i[<span class="dv">1</span>]]],</span>
<span id="cb12-20"><a href="#cb12-20" tabindex="-1"></a>    group_samples[[i[<span class="dv">2</span>]]]</span>
<span id="cb12-21"><a href="#cb12-21" tabindex="-1"></a>  )</span>
<span id="cb12-22"><a href="#cb12-22" tabindex="-1"></a>  <span class="fu">names</span>(res) <span class="ot">&lt;-</span> i</span>
<span id="cb12-23"><a href="#cb12-23" tabindex="-1"></a>  res</span>
<span id="cb12-24"><a href="#cb12-24" tabindex="-1"></a>}</span>
<span id="cb12-25"><a href="#cb12-25" tabindex="-1"></a><span class="fu">names</span>(comp_list) <span class="ot">&lt;-</span> <span class="fu">lapply</span>(group_combn, <span class="cf">function</span>(x) <span class="fu">paste0</span>(x[<span class="dv">1</span>], <span class="st">&#39;-o-&#39;</span>, x[<span class="dv">2</span>]))</span>
<span id="cb12-26"><a href="#cb12-26" tabindex="-1"></a></span>
<span id="cb12-27"><a href="#cb12-27" tabindex="-1"></a></span>
<span id="cb12-28"><a href="#cb12-28" tabindex="-1"></a><span class="co"># Add additional comparison</span></span>
<span id="cb12-29"><a href="#cb12-29" tabindex="-1"></a>comp_list[[<span class="st">&#39;AMHR2_KO-o-ctrl&#39;</span>]] <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb12-30"><a href="#cb12-30" tabindex="-1"></a>  <span class="st">`</span><span class="at">AMHR2_KO</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb12-31"><a href="#cb12-31" tabindex="-1"></a>    <span class="fu">filter</span>(condition <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&#39;InVivo_AMHR2_KO&#39;</span>, <span class="st">&#39;AMHR2_KO_culture&#39;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb12-32"><a href="#cb12-32" tabindex="-1"></a>     <span class="fu">pull</span>(sample_alias),</span>
<span id="cb12-33"><a href="#cb12-33" tabindex="-1"></a>  <span class="st">`</span><span class="at">control</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb12-34"><a href="#cb12-34" tabindex="-1"></a>    <span class="fu">filter</span>(condition <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&#39;InVivo_Control_sgRNA&#39;</span>, <span class="st">&#39;AMHR2_Control_culture&#39;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb12-35"><a href="#cb12-35" tabindex="-1"></a>     <span class="fu">pull</span>(sample_alias)</span>
<span id="cb12-36"><a href="#cb12-36" tabindex="-1"></a>)</span>
<span id="cb12-37"><a href="#cb12-37" tabindex="-1"></a></span>
<span id="cb12-38"><a href="#cb12-38" tabindex="-1"></a>keep_comp <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb12-39"><a href="#cb12-39" tabindex="-1"></a>  <span class="st">`</span><span class="at">InVivo_AMHR2_KO + AMHR2_KO_culture--over--InVivo_Control_sgRNA + AMHR2_Control_culture</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;AMHR2_KO-o-ctrl&#39;</span>,</span>
<span id="cb12-40"><a href="#cb12-40" tabindex="-1"></a>  <span class="st">`</span><span class="at">AMHR2_KO_noTGFbeta--over--AMHR2_Control_culture</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;AMHR2_KO_noTGFb-o-c_AMHR2_ctrl&#39;</span>,</span>
<span id="cb12-41"><a href="#cb12-41" tabindex="-1"></a>  <span class="st">`</span><span class="at">InVivo_AMHR2_KO–over–InVivo_Control_sgRNA</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;i_AMHR2_KO-o-i_ctrl_sgRNA&#39;</span></span>
<span id="cb12-42"><a href="#cb12-42" tabindex="-1"></a>)</span>
<span id="cb12-43"><a href="#cb12-43" tabindex="-1"></a>comp_list <span class="ot">&lt;-</span> comp_list[keep_comp]</span>
<span id="cb12-44"><a href="#cb12-44" tabindex="-1"></a></span>
<span id="cb12-45"><a href="#cb12-45" tabindex="-1"></a></span>
<span id="cb12-46"><a href="#cb12-46" tabindex="-1"></a><span class="co"># Generate DF with name description</span></span>
<span id="cb12-47"><a href="#cb12-47" tabindex="-1"></a>comp_list_description <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb12-48"><a href="#cb12-48" tabindex="-1"></a>  <span class="at">comparison =</span> <span class="fu">names</span>(comp_list),</span>
<span id="cb12-49"><a href="#cb12-49" tabindex="-1"></a>  <span class="at">comparison_long =</span> <span class="fu">names</span>(keep_comp)</span>
<span id="cb12-50"><a href="#cb12-50" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="table-of-comparisons" class="section level4">
<h4>Table of comparisons</h4>
<p>We filter comparisons with n &lt; 3 samples in each group</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a>i <span class="ot">=</span> <span class="fu">names</span>(comp_list)[<span class="dv">1</span>]</span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>comp_df <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i =</span> <span class="fu">names</span>(comp_list), <span class="at">.combine =</span> rbind) <span class="sc">%do%</span> {</span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a>  icomp <span class="ot">&lt;-</span> comp_list[[i]]</span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb13-5"><a href="#cb13-5" tabindex="-1"></a>    <span class="at">comparison =</span> i,</span>
<span id="cb13-6"><a href="#cb13-6" tabindex="-1"></a>    <span class="st">`</span><span class="at">Sample Group (n)</span><span class="st">`</span> <span class="ot">=</span>  <span class="fu">length</span>(icomp[[<span class="dv">1</span>]]),</span>
<span id="cb13-7"><a href="#cb13-7" tabindex="-1"></a>    <span class="st">`</span><span class="at">Reference Group (n)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">length</span>(icomp[[<span class="dv">2</span>]]),</span>
<span id="cb13-8"><a href="#cb13-8" tabindex="-1"></a>    <span class="at">check.names =</span> <span class="cn">FALSE</span></span>
<span id="cb13-9"><a href="#cb13-9" tabindex="-1"></a>  )</span>
<span id="cb13-10"><a href="#cb13-10" tabindex="-1"></a>}</span>
<span id="cb13-11"><a href="#cb13-11" tabindex="-1"></a></span>
<span id="cb13-12"><a href="#cb13-12" tabindex="-1"></a><span class="co"># filter comparisons were we have at least 3 samples in each group</span></span>
<span id="cb13-13"><a href="#cb13-13" tabindex="-1"></a>comp_df <span class="sc">%&lt;&gt;%</span> </span>
<span id="cb13-14"><a href="#cb13-14" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Evaluated =</span> <span class="st">`</span><span class="at">Sample Group (n)</span><span class="st">`</span> <span class="sc">&gt;=</span> <span class="dv">3</span> <span class="sc">&amp;</span> <span class="st">`</span><span class="at">Reference Group (n)</span><span class="st">`</span> <span class="sc">&gt;=</span> <span class="dv">3</span>)</span>
<span id="cb13-15"><a href="#cb13-15" tabindex="-1"></a>kepp_comp <span class="ot">&lt;-</span> comp_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Evaluated) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(comparison)</span>
<span id="cb13-16"><a href="#cb13-16" tabindex="-1"></a>comp_list <span class="ot">&lt;-</span> comp_list[kepp_comp]</span>
<span id="cb13-17"><a href="#cb13-17" tabindex="-1"></a></span>
<span id="cb13-18"><a href="#cb13-18" tabindex="-1"></a><span class="co"># Add long names</span></span>
<span id="cb13-19"><a href="#cb13-19" tabindex="-1"></a>comp_df <span class="ot">&lt;-</span> comp_df <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(comp_list_description) <span class="sc">%&gt;%</span> </span>
<span id="cb13-20"><a href="#cb13-20" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">comparison =</span> comparison_long) <span class="sc">%&gt;%</span></span>
<span id="cb13-21"><a href="#cb13-21" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>comparison_long)</span>
<span id="cb13-22"><a href="#cb13-22" tabindex="-1"></a></span>
<span id="cb13-23"><a href="#cb13-23" tabindex="-1"></a>comp_df <span class="sc">%&gt;%</span></span>
<span id="cb13-24"><a href="#cb13-24" tabindex="-1"></a>  <span class="fu">datatable</span>(., </span>
<span id="cb13-25"><a href="#cb13-25" tabindex="-1"></a>            <span class="at">rownames =</span> <span class="cn">FALSE</span>, </span>
<span id="cb13-26"><a href="#cb13-26" tabindex="-1"></a>            <span class="at">filter =</span> <span class="st">&#39;top&#39;</span>, </span>
<span id="cb13-27"><a href="#cb13-27" tabindex="-1"></a>            <span class="at">caption =</span> <span class="st">&#39;List of comparisons for differential expression. Comparisons with n &lt; 3 samples in each group won</span><span class="sc">\&#39;</span><span class="st">t be evaluated&#39;</span>,</span>
<span id="cb13-28"><a href="#cb13-28" tabindex="-1"></a>            <span class="at">extensions =</span> <span class="st">&#39;Buttons&#39;</span>, </span>
<span id="cb13-29"><a href="#cb13-29" tabindex="-1"></a>            <span class="at">options =</span> <span class="fu">list</span>(</span>
<span id="cb13-30"><a href="#cb13-30" tabindex="-1"></a>              <span class="at">dom =</span> <span class="st">&#39;Blfrtip&#39;</span>,</span>
<span id="cb13-31"><a href="#cb13-31" tabindex="-1"></a>              <span class="at">buttons =</span> <span class="fu">c</span>(<span class="st">&#39;csv&#39;</span>, <span class="st">&#39;excel&#39;</span>)</span>
<span id="cb13-32"><a href="#cb13-32" tabindex="-1"></a>            ))</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-0e41329c1963262c826a" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-0e41329c1963262c826a">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"integer\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"3\" data-max=\"6\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"integer\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"3\" data-max=\"6\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"logical\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\" disabled=\"\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;true&quot;,&quot;false&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption>List of comparisons for differential expression. Comparisons with n &lt; 3 samples in each group won't be evaluated<\/caption>","data":[["InVivo_AMHR2_KO + AMHR2_KO_culture--over--InVivo_Control_sgRNA + AMHR2_Control_culture","AMHR2_KO_noTGFbeta--over--AMHR2_Control_culture","InVivo_AMHR2_KO–over–InVivo_Control_sgRNA"],[6,3,3],[6,3,3],[true,true,true]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>comparison<\/th>\n      <th>Sample Group (n)<\/th>\n      <th>Reference Group (n)<\/th>\n      <th>Evaluated<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["csv","excel"],"columnDefs":[{"className":"dt-right","targets":[1,2]},{"name":"comparison","targets":0},{"name":"Sample Group (n)","targets":1},{"name":"Reference Group (n)","targets":2},{"name":"Evaluated","targets":3}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</div>
<div id="data-generation" class="section level2">
<h2>Data generation</h2>
<div id="run-differential-expression-analysis" class="section level3">
<h3>Run differential expression analysis</h3>
<p>Get Ensembl gene info</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a>ensembl <span class="ot">&lt;-</span>  biomaRt<span class="sc">::</span><span class="fu">useEnsembl</span>(<span class="at">biomart=</span><span class="st">&quot;genes&quot;</span>, <span class="at">dataset=</span><span class="st">&quot;mmusculus_gene_ensembl&quot;</span>,</span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a>                                <span class="at">mirror =</span> <span class="st">&#39;www&#39;</span>)</span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a></span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a>gene_desc <span class="ot">&lt;-</span> biomaRt<span class="sc">::</span><span class="fu">getBM</span>(<span class="at">attributes=</span><span class="fu">c</span>(<span class="st">&#39;external_gene_name&#39;</span>,<span class="st">&#39;description&#39;</span>),</span>
<span id="cb14-5"><a href="#cb14-5" tabindex="-1"></a>                              <span class="at">filters =</span> <span class="st">&#39;external_gene_name&#39;</span>,</span>
<span id="cb14-6"><a href="#cb14-6" tabindex="-1"></a>                              <span class="at">values =</span> <span class="fu">rowData</span>(se)<span class="sc">$</span>gene_name, <span class="at">mart =</span>ensembl) <span class="sc">%&gt;%</span></span>
<span id="cb14-7"><a href="#cb14-7" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&#39;gene_name&#39;</span> <span class="ot">=</span> <span class="st">&#39;external_gene_name&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb14-8"><a href="#cb14-8" tabindex="-1"></a>    unique</span>
<span id="cb14-9"><a href="#cb14-9" tabindex="-1"></a></span>
<span id="cb14-10"><a href="#cb14-10" tabindex="-1"></a></span>
<span id="cb14-11"><a href="#cb14-11" tabindex="-1"></a>gene_desc <span class="ot">&lt;-</span> gene_desc[<span class="sc">!</span><span class="fu">duplicated</span>(gene_desc<span class="sc">$</span>gene_name),]</span>
<span id="cb14-12"><a href="#cb14-12" tabindex="-1"></a><span class="fu">saveRDS</span>(gene_desc, <span class="at">file =</span> <span class="fu">file.path</span>(<span class="fu">dirname</span>(params<span class="sc">$</span>se_path), <span class="st">&#39;ensembl_gene_desc.rds&#39;</span>))</span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a>gene_desc <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="fu">file.path</span>(<span class="fu">dirname</span>(params<span class="sc">$</span>se_path), <span class="st">&#39;ensembl_gene_desc.rds&#39;</span>))</span></code></pre></div>
<p>Run differential expression analysis using EdgeR QLF</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a>use_comp <span class="ot">&lt;-</span>  comp_list[[<span class="dv">1</span>]]</span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a>dge_list <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">use_comp =</span> comp_list) <span class="sc">%do%</span>{</span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a>  se_cols <span class="ot">&lt;-</span> se<span class="sc">$</span>sample_alias <span class="sc">%in%</span> <span class="fu">unlist</span>(use_comp)</span>
<span id="cb16-4"><a href="#cb16-4" tabindex="-1"></a>  use_se <span class="ot">&lt;-</span> se[,se_cols]</span>
<span id="cb16-5"><a href="#cb16-5" tabindex="-1"></a>  use_se<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(</span>
<span id="cb16-6"><a href="#cb16-6" tabindex="-1"></a>    use_se<span class="sc">$</span>sample_alias <span class="sc">%in%</span> use_comp[[<span class="dv">1</span>]],</span>
<span id="cb16-7"><a href="#cb16-7" tabindex="-1"></a>    <span class="fu">names</span>(use_comp)[<span class="dv">1</span>],</span>
<span id="cb16-8"><a href="#cb16-8" tabindex="-1"></a>    <span class="fu">names</span>(use_comp)[<span class="dv">2</span>]</span>
<span id="cb16-9"><a href="#cb16-9" tabindex="-1"></a>  )</span>
<span id="cb16-10"><a href="#cb16-10" tabindex="-1"></a>  </span>
<span id="cb16-11"><a href="#cb16-11" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">is.null</span>(use_comp<span class="sc">$</span>batch)) {</span>
<span id="cb16-12"><a href="#cb16-12" tabindex="-1"></a>    dge <span class="ot">&lt;-</span> <span class="fu">edgeR_dge</span>(</span>
<span id="cb16-13"><a href="#cb16-13" tabindex="-1"></a>      use_se,</span>
<span id="cb16-14"><a href="#cb16-14" tabindex="-1"></a>      <span class="co"># Design configuration for differential expression</span></span>
<span id="cb16-15"><a href="#cb16-15" tabindex="-1"></a>      <span class="at">group_var =</span>  <span class="st">&#39;group&#39;</span>,</span>
<span id="cb16-16"><a href="#cb16-16" tabindex="-1"></a>      <span class="at">group_sample =</span> <span class="fu">names</span>(use_comp)[<span class="dv">1</span>],</span>
<span id="cb16-17"><a href="#cb16-17" tabindex="-1"></a>      <span class="at">group_ref =</span> <span class="fu">names</span>(use_comp)[<span class="dv">2</span>],</span>
<span id="cb16-18"><a href="#cb16-18" tabindex="-1"></a>      <span class="at">design_formula =</span> <span class="st">&quot;~ group&quot;</span>,</span>
<span id="cb16-19"><a href="#cb16-19" tabindex="-1"></a>      <span class="at">coef =</span> <span class="st">&#39;last&#39;</span>,</span>
<span id="cb16-20"><a href="#cb16-20" tabindex="-1"></a>      <span class="co"># Conversion from SingleCellExperiment/Summarized Experiment to DGEList</span></span>
<span id="cb16-21"><a href="#cb16-21" tabindex="-1"></a>      <span class="at">assay_to_DGEList =</span> <span class="st">&#39;counts&#39;</span>,</span>
<span id="cb16-22"><a href="#cb16-22" tabindex="-1"></a>      <span class="at">assay_to_row_filter =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb16-23"><a href="#cb16-23" tabindex="-1"></a>      <span class="co"># Feature filtering parameters</span></span>
<span id="cb16-24"><a href="#cb16-24" tabindex="-1"></a>      <span class="at">use_filterByExpr =</span> <span class="cn">TRUE</span>,</span>
<span id="cb16-25"><a href="#cb16-25" tabindex="-1"></a>      <span class="at">min_counts =</span> params<span class="sc">$</span>min_counts,</span>
<span id="cb16-26"><a href="#cb16-26" tabindex="-1"></a>      <span class="at">min_present_prop =</span> params<span class="sc">$</span>min_present_prop,</span>
<span id="cb16-27"><a href="#cb16-27" tabindex="-1"></a>      <span class="co"># EdgeR workflow configuration</span></span>
<span id="cb16-28"><a href="#cb16-28" tabindex="-1"></a>      <span class="at">run_calcNormFactors =</span> <span class="st">&#39;TMM&#39;</span>,</span>
<span id="cb16-29"><a href="#cb16-29" tabindex="-1"></a>      <span class="at">estimateDisp_robust =</span> <span class="cn">TRUE</span>,</span>
<span id="cb16-30"><a href="#cb16-30" tabindex="-1"></a>      <span class="at">estimateDisp_trend.method =</span> <span class="st">&quot;locfit&quot;</span>,</span>
<span id="cb16-31"><a href="#cb16-31" tabindex="-1"></a>      <span class="at">glm_approach =</span> <span class="st">&quot;QLF&quot;</span>,</span>
<span id="cb16-32"><a href="#cb16-32" tabindex="-1"></a>      <span class="co"># Output configuration</span></span>
<span id="cb16-33"><a href="#cb16-33" tabindex="-1"></a>      <span class="at">adjust_method =</span> <span class="st">&#39;BH&#39;</span></span>
<span id="cb16-34"><a href="#cb16-34" tabindex="-1"></a>      )</span>
<span id="cb16-35"><a href="#cb16-35" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb16-36"><a href="#cb16-36" tabindex="-1"></a>      design_formula <span class="ot">=</span> <span class="fu">paste0</span>(<span class="st">&quot;~ &quot;</span>, use_comp<span class="sc">$</span>batch,<span class="st">&quot; + group&quot;</span>)</span>
<span id="cb16-37"><a href="#cb16-37" tabindex="-1"></a>      dge <span class="ot">&lt;-</span> <span class="fu">edgeR_dge</span>(</span>
<span id="cb16-38"><a href="#cb16-38" tabindex="-1"></a>        use_se,</span>
<span id="cb16-39"><a href="#cb16-39" tabindex="-1"></a>        <span class="co"># Design configuration for differential expression</span></span>
<span id="cb16-40"><a href="#cb16-40" tabindex="-1"></a>        <span class="at">group_var =</span>  <span class="st">&#39;group&#39;</span>,</span>
<span id="cb16-41"><a href="#cb16-41" tabindex="-1"></a>        <span class="at">group_sample =</span> <span class="fu">names</span>(use_comp)[<span class="dv">1</span>],</span>
<span id="cb16-42"><a href="#cb16-42" tabindex="-1"></a>        <span class="at">group_ref =</span> <span class="fu">names</span>(use_comp)[<span class="dv">2</span>],</span>
<span id="cb16-43"><a href="#cb16-43" tabindex="-1"></a>        <span class="at">batch_vars =</span> use_comp<span class="sc">$</span>batch,</span>
<span id="cb16-44"><a href="#cb16-44" tabindex="-1"></a>        <span class="at">design_formula =</span> design_formula,</span>
<span id="cb16-45"><a href="#cb16-45" tabindex="-1"></a>        <span class="at">coef =</span> <span class="st">&#39;last&#39;</span>,</span>
<span id="cb16-46"><a href="#cb16-46" tabindex="-1"></a>        <span class="co"># Conversion from SingleCellExperiment/Summarized Experiment to DGEList</span></span>
<span id="cb16-47"><a href="#cb16-47" tabindex="-1"></a>        <span class="at">assay_to_DGEList =</span> <span class="st">&#39;counts&#39;</span>,</span>
<span id="cb16-48"><a href="#cb16-48" tabindex="-1"></a>        <span class="at">assay_to_row_filter =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb16-49"><a href="#cb16-49" tabindex="-1"></a>        <span class="co"># Feature filtering parameters</span></span>
<span id="cb16-50"><a href="#cb16-50" tabindex="-1"></a>        <span class="at">use_filterByExpr =</span> <span class="cn">TRUE</span>,</span>
<span id="cb16-51"><a href="#cb16-51" tabindex="-1"></a>        <span class="at">min_counts =</span> params<span class="sc">$</span>min_counts,</span>
<span id="cb16-52"><a href="#cb16-52" tabindex="-1"></a>        <span class="at">min_present_prop =</span> params<span class="sc">$</span>min_present_prop,</span>
<span id="cb16-53"><a href="#cb16-53" tabindex="-1"></a>        <span class="co"># EdgeR workflow configuration</span></span>
<span id="cb16-54"><a href="#cb16-54" tabindex="-1"></a>        <span class="at">run_calcNormFactors =</span> <span class="st">&#39;TMM&#39;</span>,</span>
<span id="cb16-55"><a href="#cb16-55" tabindex="-1"></a>        <span class="at">estimateDisp_robust =</span> <span class="cn">TRUE</span>,</span>
<span id="cb16-56"><a href="#cb16-56" tabindex="-1"></a>        <span class="at">estimateDisp_trend.method =</span> <span class="st">&quot;locfit&quot;</span>,</span>
<span id="cb16-57"><a href="#cb16-57" tabindex="-1"></a>        <span class="at">glm_approach =</span> <span class="st">&quot;QLF&quot;</span>,</span>
<span id="cb16-58"><a href="#cb16-58" tabindex="-1"></a>        <span class="co"># Output configuration</span></span>
<span id="cb16-59"><a href="#cb16-59" tabindex="-1"></a>        <span class="at">adjust_method =</span> <span class="st">&#39;BH&#39;</span></span>
<span id="cb16-60"><a href="#cb16-60" tabindex="-1"></a>        )</span>
<span id="cb16-61"><a href="#cb16-61" tabindex="-1"></a>    }</span>
<span id="cb16-62"><a href="#cb16-62" tabindex="-1"></a>  </span>
<span id="cb16-63"><a href="#cb16-63" tabindex="-1"></a>  <span class="co"># Add gene description</span></span>
<span id="cb16-64"><a href="#cb16-64" tabindex="-1"></a>  use_res <span class="ot">&lt;-</span> dge<span class="sc">$</span>results <span class="sc">%&gt;%</span>  <span class="fu">left_join</span>(., gene_desc, <span class="at">relationship =</span> <span class="st">&quot;many-to-many&quot;</span>)</span>
<span id="cb16-65"><a href="#cb16-65" tabindex="-1"></a></span>
<span id="cb16-66"><a href="#cb16-66" tabindex="-1"></a>  dge<span class="sc">$</span>results <span class="ot">&lt;-</span> use_res <span class="sc">%&gt;%</span> </span>
<span id="cb16-67"><a href="#cb16-67" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(feature)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-68"><a href="#cb16-68" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">rownames =</span> feature) <span class="sc">%&gt;%</span> </span>
<span id="cb16-69"><a href="#cb16-69" tabindex="-1"></a>    <span class="fu">column_to_rownames</span>(<span class="st">&#39;rownames&#39;</span>)</span>
<span id="cb16-70"><a href="#cb16-70" tabindex="-1"></a>  </span>
<span id="cb16-71"><a href="#cb16-71" tabindex="-1"></a>   <span class="fu">return</span>(dge)</span>
<span id="cb16-72"><a href="#cb16-72" tabindex="-1"></a>}</span>
<span id="cb16-73"><a href="#cb16-73" tabindex="-1"></a></span>
<span id="cb16-74"><a href="#cb16-74" tabindex="-1"></a><span class="fu">names</span>(dge_list) <span class="ot">&lt;-</span> <span class="fu">names</span>(comp_list)</span>
<span id="cb16-75"><a href="#cb16-75" tabindex="-1"></a></span>
<span id="cb16-76"><a href="#cb16-76" tabindex="-1"></a><span class="co"># Save objects</span></span>
<span id="cb16-77"><a href="#cb16-77" tabindex="-1"></a><span class="fu">saveRDS</span>(dge_list, <span class="at">file =</span> <span class="fu">file.path</span>(params<span class="sc">$</span>output_dir, <span class="st">&#39;dge-edgeR_QLF.rds&#39;</span>))</span></code></pre></div>
</div>
<div id="run-functional-enrichment" class="section level3">
<h3>Run functional enrichment</h3>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a>dge_list <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">file.path</span>(params<span class="sc">$</span>output_dir, <span class="st">&#39;dge-edgeR_QLF.rds&#39;</span>))</span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a>dge <span class="ot">&lt;-</span>  dge_list[[<span class="dv">2</span>]]</span>
<span id="cb17-3"><a href="#cb17-3" tabindex="-1"></a></span>
<span id="cb17-4"><a href="#cb17-4" tabindex="-1"></a>gse_list <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">dge =</span> dge_list) <span class="sc">%do%</span>{</span>
<span id="cb17-5"><a href="#cb17-5" tabindex="-1"></a>  gse_res <span class="ot">&lt;-</span> <span class="fu">gse_omnibus</span>(</span>
<span id="cb17-6"><a href="#cb17-6" tabindex="-1"></a>    <span class="at">feature_names =</span> dge<span class="sc">$</span>results<span class="sc">$</span>gene_name,</span>
<span id="cb17-7"><a href="#cb17-7" tabindex="-1"></a>    <span class="at">p =</span> dge<span class="sc">$</span>results<span class="sc">$</span>FDR,</span>
<span id="cb17-8"><a href="#cb17-8" tabindex="-1"></a>    <span class="at">fc =</span> dge<span class="sc">$</span>results<span class="sc">$</span>logFC,</span>
<span id="cb17-9"><a href="#cb17-9" tabindex="-1"></a>    <span class="at">fc_thrs =</span> <span class="fl">0.5</span>,</span>
<span id="cb17-10"><a href="#cb17-10" tabindex="-1"></a>    <span class="at">gmt_files =</span> gmt_files_symbols,</span>
<span id="cb17-11"><a href="#cb17-11" tabindex="-1"></a>    </span>
<span id="cb17-12"><a href="#cb17-12" tabindex="-1"></a>    <span class="at">annot_db =</span> <span class="st">&#39;org.Mm.eg.db&#39;</span>,</span>
<span id="cb17-13"><a href="#cb17-13" tabindex="-1"></a>    <span class="at">organism =</span> <span class="st">&#39;mmusculus&#39;</span>,</span>
<span id="cb17-14"><a href="#cb17-14" tabindex="-1"></a>    <span class="at">kegg_organism =</span> <span class="st">&#39;mmu&#39;</span>,</span>
<span id="cb17-15"><a href="#cb17-15" tabindex="-1"></a>    <span class="at">reactome_organism =</span> <span class="st">&#39;mouse&#39;</span>,</span>
<span id="cb17-16"><a href="#cb17-16" tabindex="-1"></a>    </span>
<span id="cb17-17"><a href="#cb17-17" tabindex="-1"></a>    <span class="at">run_enricher =</span> <span class="cn">FALSE</span>,</span>
<span id="cb17-18"><a href="#cb17-18" tabindex="-1"></a>    <span class="at">run_enrichGO =</span> <span class="cn">FALSE</span>,</span>
<span id="cb17-19"><a href="#cb17-19" tabindex="-1"></a>    <span class="at">run_enrichKEGG =</span> <span class="cn">FALSE</span>,</span>
<span id="cb17-20"><a href="#cb17-20" tabindex="-1"></a>    <span class="at">run_enrichReactome =</span> <span class="cn">FALSE</span>,</span>
<span id="cb17-21"><a href="#cb17-21" tabindex="-1"></a>    </span>
<span id="cb17-22"><a href="#cb17-22" tabindex="-1"></a>    <span class="at">run_GSEA =</span> <span class="cn">TRUE</span>,</span>
<span id="cb17-23"><a href="#cb17-23" tabindex="-1"></a>    <span class="at">run_gseGO =</span> <span class="cn">FALSE</span>,</span>
<span id="cb17-24"><a href="#cb17-24" tabindex="-1"></a>    <span class="at">run_gseKEGG =</span> <span class="cn">FALSE</span>,</span>
<span id="cb17-25"><a href="#cb17-25" tabindex="-1"></a>    <span class="at">run_gseReactome =</span> <span class="cn">FALSE</span>,</span>
<span id="cb17-26"><a href="#cb17-26" tabindex="-1"></a>    </span>
<span id="cb17-27"><a href="#cb17-27" tabindex="-1"></a>    <span class="at">go_ontologies =</span> <span class="fu">c</span>(<span class="st">&#39;BP&#39;</span>, <span class="st">&#39;MF&#39;</span>),</span>
<span id="cb17-28"><a href="#cb17-28" tabindex="-1"></a>    <span class="at">go_simplify_p =</span> <span class="fl">0.05</span>,</span>
<span id="cb17-29"><a href="#cb17-29" tabindex="-1"></a>    <span class="at">args_gse =</span> <span class="fu">list</span>(<span class="at">minGSSize =</span> <span class="dv">10</span>, <span class="at">maxGSSize =</span> <span class="dv">250</span>, </span>
<span id="cb17-30"><a href="#cb17-30" tabindex="-1"></a>                    <span class="at">pvalueCutoff =</span> <span class="dv">1</span>, </span>
<span id="cb17-31"><a href="#cb17-31" tabindex="-1"></a>                    <span class="at">BPPARAM =</span> BiocParallel<span class="sc">::</span><span class="fu">MulticoreParam</span>(<span class="dv">1</span>))</span>
<span id="cb17-32"><a href="#cb17-32" tabindex="-1"></a>    )</span>
<span id="cb17-33"><a href="#cb17-33" tabindex="-1"></a>  </span>
<span id="cb17-34"><a href="#cb17-34" tabindex="-1"></a>   <span class="fu">return</span>(gse_res)</span>
<span id="cb17-35"><a href="#cb17-35" tabindex="-1"></a></span>
<span id="cb17-36"><a href="#cb17-36" tabindex="-1"></a>}</span>
<span id="cb17-37"><a href="#cb17-37" tabindex="-1"></a><span class="fu">names</span>(gse_list) <span class="ot">&lt;-</span> <span class="fu">names</span>(dge_list)</span>
<span id="cb17-38"><a href="#cb17-38" tabindex="-1"></a></span>
<span id="cb17-39"><a href="#cb17-39" tabindex="-1"></a><span class="co"># Save objects</span></span>
<span id="cb17-40"><a href="#cb17-40" tabindex="-1"></a><span class="fu">saveRDS</span>(gse_list, <span class="at">file =</span> <span class="fu">file.path</span>(params<span class="sc">$</span>output_dir,<span class="st">&#39;gse_omnibus-edgeR_QLF.rds&#39;</span>))</span></code></pre></div>
</div>
<div id="load-data" class="section level3">
<h3>Load data</h3>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a>dge_list <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="fu">file.path</span>(params<span class="sc">$</span>output_dir, <span class="st">&#39;dge-edgeR_QLF.rds&#39;</span>))</span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a>gse_list <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="fu">file.path</span>(params<span class="sc">$</span>output_dir, <span class="st">&#39;gse_omnibus-edgeR_QLF.rds&#39;</span>))</span></code></pre></div>
</div>
<div id="knitrtemplate" class="section level3">
<h3>Knitr:template</h3>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a>selected_labels_volcano <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;Amhr2&#39;</span>)</span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a><span class="co"># knit template: https://workflowr.github.io/workflowr/articles/wflow-07-common-code.html</span></span>
<span id="cb19-3"><a href="#cb19-3" tabindex="-1"></a>src <span class="ot">&lt;-</span> <span class="fu">knit_expand</span>(</span>
<span id="cb19-4"><a href="#cb19-4" tabindex="-1"></a>  <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;analysis/templates/rnaseq-deg-edger.Rmd&quot;</span>)</span>
<span id="cb19-5"><a href="#cb19-5" tabindex="-1"></a>  )</span></code></pre></div>
<!-- To be compatible with workflowr, we need to specify the absolute directory under which the plots are generated, otherwise they will be saved into figure/<basename>.Rmd/<chunk-name>.<extension. and not visible in the html document: https://github.com/workflowr/workflowr/issues/111 -->
</div>
</div>
<div id="download-results" class="section level2">
<h2>Download results</h2>
<!-- Write DGE results to xlsx file -->
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a><span class="co"># File name summary</span></span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a>rmd_file <span class="ot">&lt;-</span> <span class="fu">current_input</span>()</span>
<span id="cb20-3"><a href="#cb20-3" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">is.null</span>(rmd_file))</span>
<span id="cb20-4"><a href="#cb20-4" tabindex="-1"></a>  rmd_file <span class="ot">&lt;-</span> <span class="st">&#39;tmp&#39;</span></span>
<span id="cb20-5"><a href="#cb20-5" tabindex="-1"></a>file_xlsx <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">&#39;./docs/file&#39;</span>,rmd_file, <span class="st">&#39;dge.xlsx&#39;</span>)</span>
<span id="cb20-6"><a href="#cb20-6" tabindex="-1"></a></span>
<span id="cb20-7"><a href="#cb20-7" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="fu">dirname</span>(file_xlsx), <span class="at">recursive =</span> <span class="cn">TRUE</span>, <span class="at">showWarnings =</span> <span class="cn">FALSE</span>)</span>
<span id="cb20-8"><a href="#cb20-8" tabindex="-1"></a></span>
<span id="cb20-9"><a href="#cb20-9" tabindex="-1"></a><span class="co"># Generate workbook</span></span>
<span id="cb20-10"><a href="#cb20-10" tabindex="-1"></a>wb <span class="ot">&lt;-</span> <span class="fu">createWorkbook</span>()</span>
<span id="cb20-11"><a href="#cb20-11" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">names</span>(dge_list)) {</span>
<span id="cb20-12"><a href="#cb20-12" tabindex="-1"></a>  <span class="fu">addWorksheet</span>(wb, i)</span>
<span id="cb20-13"><a href="#cb20-13" tabindex="-1"></a>  res <span class="ot">&lt;-</span> dge_list[[i]]<span class="sc">$</span>results <span class="sc">%&gt;%</span> </span>
<span id="cb20-14"><a href="#cb20-14" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(feature, gene_name, mean, detected, logFC, PValue, FDR)</span>
<span id="cb20-15"><a href="#cb20-15" tabindex="-1"></a>  <span class="fu">writeData</span>(wb, i, res)</span>
<span id="cb20-16"><a href="#cb20-16" tabindex="-1"></a>}</span>
<span id="cb20-17"><a href="#cb20-17" tabindex="-1"></a><span class="fu">saveWorkbook</span>(wb, file_xlsx, <span class="cn">TRUE</span>)</span>
<span id="cb20-18"><a href="#cb20-18" tabindex="-1"></a></span>
<span id="cb20-19"><a href="#cb20-19" tabindex="-1"></a>dge_file_xlsx <span class="ot">&lt;-</span> file_xlsx</span></code></pre></div>
<!-- Write GSE results to xlsx file -->
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a>gse_collections <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a>  <span class="at">GSEA =</span> <span class="fu">names</span>(gmt_files_symbols),</span>
<span id="cb21-3"><a href="#cb21-3" tabindex="-1"></a>  <span class="at">gseGO =</span> <span class="fu">c</span>(<span class="st">&quot;BP&quot;</span>, <span class="st">&quot;MF&quot;</span>)</span>
<span id="cb21-4"><a href="#cb21-4" tabindex="-1"></a>)</span>
<span id="cb21-5"><a href="#cb21-5" tabindex="-1"></a></span>
<span id="cb21-6"><a href="#cb21-6" tabindex="-1"></a><span class="co"># File name summary</span></span>
<span id="cb21-7"><a href="#cb21-7" tabindex="-1"></a>rmd_file <span class="ot">&lt;-</span> <span class="fu">current_input</span>()</span>
<span id="cb21-8"><a href="#cb21-8" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">is.null</span>(rmd_file))</span>
<span id="cb21-9"><a href="#cb21-9" tabindex="-1"></a>  rmd_file <span class="ot">&lt;-</span> <span class="st">&#39;tmp&#39;</span></span>
<span id="cb21-10"><a href="#cb21-10" tabindex="-1"></a></span>
<span id="cb21-11"><a href="#cb21-11" tabindex="-1"></a>file_xlsx_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb21-12"><a href="#cb21-12" tabindex="-1"></a><span class="cf">for</span>(sr <span class="cf">in</span> <span class="fu">names</span>(gse_collections)) {</span>
<span id="cb21-13"><a href="#cb21-13" tabindex="-1"></a>  file_xlsx_list[[sr]] <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb21-14"><a href="#cb21-14" tabindex="-1"></a>  <span class="cf">for</span>(gc <span class="cf">in</span> gse_collections[[sr]]) {</span>
<span id="cb21-15"><a href="#cb21-15" tabindex="-1"></a>    <span class="co"># Generate file</span></span>
<span id="cb21-16"><a href="#cb21-16" tabindex="-1"></a>    file_xlsx <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">&#39;./docs/file&#39;</span>, </span>
<span id="cb21-17"><a href="#cb21-17" tabindex="-1"></a>                           rmd_file, </span>
<span id="cb21-18"><a href="#cb21-18" tabindex="-1"></a>                           <span class="fu">paste0</span>(<span class="st">&#39;gse_&#39;</span>, <span class="fu">tolower</span>(sr), <span class="st">&#39;_&#39;</span>, <span class="fu">tolower</span>(gc), <span class="st">&#39;.xlsx&#39;</span>)</span>
<span id="cb21-19"><a href="#cb21-19" tabindex="-1"></a>                           )</span>
<span id="cb21-20"><a href="#cb21-20" tabindex="-1"></a>    file_xlsx_list[[sr]][[gc]] <span class="ot">&lt;-</span> file_xlsx</span>
<span id="cb21-21"><a href="#cb21-21" tabindex="-1"></a>    <span class="fu">dir.create</span>(<span class="fu">dirname</span>(file_xlsx), <span class="at">recursive =</span> <span class="cn">TRUE</span>, <span class="at">showWarnings =</span> <span class="cn">FALSE</span>)</span>
<span id="cb21-22"><a href="#cb21-22" tabindex="-1"></a>    </span>
<span id="cb21-23"><a href="#cb21-23" tabindex="-1"></a>    <span class="co"># Generate workbook</span></span>
<span id="cb21-24"><a href="#cb21-24" tabindex="-1"></a>    wb <span class="ot">&lt;-</span> <span class="fu">createWorkbook</span>()</span>
<span id="cb21-25"><a href="#cb21-25" tabindex="-1"></a>    <span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">names</span>(gse_list)) {</span>
<span id="cb21-26"><a href="#cb21-26" tabindex="-1"></a>      <span class="fu">addWorksheet</span>(wb, i)</span>
<span id="cb21-27"><a href="#cb21-27" tabindex="-1"></a>      <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(gse_list[[i]][[sr]][[gc]])) {</span>
<span id="cb21-28"><a href="#cb21-28" tabindex="-1"></a>        res <span class="ot">&lt;-</span> gse_list[[i]][[sr]][[gc]]<span class="sc">@</span>result</span>
<span id="cb21-29"><a href="#cb21-29" tabindex="-1"></a>        <span class="fu">writeData</span>(wb, i, res)</span>
<span id="cb21-30"><a href="#cb21-30" tabindex="-1"></a>      }</span>
<span id="cb21-31"><a href="#cb21-31" tabindex="-1"></a>    }</span>
<span id="cb21-32"><a href="#cb21-32" tabindex="-1"></a>    <span class="fu">saveWorkbook</span>(wb, file_xlsx, <span class="cn">TRUE</span>)</span>
<span id="cb21-33"><a href="#cb21-33" tabindex="-1"></a>  }</span>
<span id="cb21-34"><a href="#cb21-34" tabindex="-1"></a>}</span></code></pre></div>
<!-- Change dge_list names after writing to xlsx -->
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a>comp_names <span class="ot">&lt;-</span> comp_list_description<span class="sc">$</span>comparison_long <span class="sc">%&gt;%</span> </span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a>  <span class="fu">set_names</span>(comp_list_description<span class="sc">$</span>comparison)</span>
<span id="cb22-3"><a href="#cb22-3" tabindex="-1"></a><span class="fu">names</span>(comp_list) <span class="ot">&lt;-</span> comp_names[<span class="fu">names</span>(comp_list)]</span>
<span id="cb22-4"><a href="#cb22-4" tabindex="-1"></a><span class="fu">names</span>(dge_list) <span class="ot">&lt;-</span> comp_names[<span class="fu">names</span>(dge_list)]</span>
<span id="cb22-5"><a href="#cb22-5" tabindex="-1"></a><span class="fu">names</span>(gse_list) <span class="ot">&lt;-</span> comp_names[<span class="fu">names</span>(gse_list)]</span></code></pre></div>
<p>The tables of results can be downloaded using the following
links:</p>
<ul>
<li><p><a
href="./file/rnaseq-amhr2_ko-deg.Rmd/dge.xlsx"><strong>Differential gene
expression</strong></a>
<!-- Write links to download GSE results --></p></li>
<li><p><a
href="./file/rnaseq-amhr2_ko-deg.Rmd/gse_gsea_msigdb.h.xlsx"><strong>Gene
set enrichment analysis: msigdb.h</strong></a></p></li>
<li><p><a
href="./file/rnaseq-amhr2_ko-deg.Rmd/gse_gsego_bp.xlsx"><strong>Gene set
enrichment analysis: GO:BP</strong></a></p></li>
<li><p><a
href="./file/rnaseq-amhr2_ko-deg.Rmd/gse_gsego_mf.xlsx"><strong>Gene set
enrichment analysis: GO:MF</strong></a></p></li>
</ul>
</div>
<div id="functional-enrichment-analysis" class="section level2">
<h2>Functional enrichment analysis</h2>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a>gse_collections <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a>  <span class="at">GSEA =</span> <span class="fu">names</span>(gmt_files_symbols)</span>
<span id="cb23-3"><a href="#cb23-3" tabindex="-1"></a>)</span></code></pre></div>
<div id="download-results-1" class="section level3">
<h3>Download results</h3>
<!-- Write links to download GSE results -->
<ul>
<li><p><a
href="./file/rnaseq-amhr2_ko-deg.Rmd/gse_gsea_msigdb.h.xlsx"><strong>Gene
set enrichment analysis: msigdb.h</strong></a></p></li>
<li><p><a
href="./file/rnaseq-amhr2_ko-deg.Rmd/gse_gsego_bp.xlsx"><strong>Gene set
enrichment analysis: GO:BP</strong></a></p></li>
<li><p><a
href="./file/rnaseq-amhr2_ko-deg.Rmd/gse_gsego_mf.xlsx"><strong>Gene set
enrichment analysis: GO:MF</strong></a></p></li>
</ul>
</div>
<div id="summary-of-results" class="section level3">
<h3>Summary of results</h3>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i =</span> <span class="fu">names</span>(gse_list), <span class="at">.combine =</span> rbind) <span class="sc">%do%</span> {</span>
<span id="cb24-2"><a href="#cb24-2" tabindex="-1"></a>  <span class="fu">foreach</span>(<span class="at">sr =</span> <span class="fu">names</span>(gse_collections), <span class="at">.combine =</span> rbind) <span class="sc">%do%</span> {</span>
<span id="cb24-3"><a href="#cb24-3" tabindex="-1"></a>    <span class="fu">foreach</span>(<span class="at">gc =</span> gse_collections[[sr]], <span class="at">.combine =</span> rbind) <span class="sc">%do%</span> {</span>
<span id="cb24-4"><a href="#cb24-4" tabindex="-1"></a>      <span class="cf">if</span>(sr <span class="sc">==</span> <span class="st">&#39;GSEA&#39;</span>) {</span>
<span id="cb24-5"><a href="#cb24-5" tabindex="-1"></a>        x <span class="ot">&lt;-</span> gse_list[[i]][[sr]][[gc]]<span class="sc">@</span>result <span class="sc">%&gt;%</span></span>
<span id="cb24-6"><a href="#cb24-6" tabindex="-1"></a>          <span class="fu">filter</span>(p.adjust <span class="sc">&lt;=</span> <span class="fl">0.05</span>)</span>
<span id="cb24-7"><a href="#cb24-7" tabindex="-1"></a>        <span class="cf">if</span>(gc <span class="sc">==</span> <span class="st">&#39;msigdb.c2.cp&#39;</span>) {</span>
<span id="cb24-8"><a href="#cb24-8" tabindex="-1"></a>          x <span class="sc">%&lt;&gt;%</span></span>
<span id="cb24-9"><a href="#cb24-9" tabindex="-1"></a>            <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">&#39;REACTOME|KEGG|WP&#39;</span>, ID))</span>
<span id="cb24-10"><a href="#cb24-10" tabindex="-1"></a>        }</span>
<span id="cb24-11"><a href="#cb24-11" tabindex="-1"></a>      }</span>
<span id="cb24-12"><a href="#cb24-12" tabindex="-1"></a>      <span class="cf">if</span>(sr <span class="sc">==</span> <span class="st">&#39;gseGO&#39;</span>) {</span>
<span id="cb24-13"><a href="#cb24-13" tabindex="-1"></a>        x <span class="ot">&lt;-</span> gse_list[[i]][[sr]][[gc]]<span class="sc">@</span>result <span class="sc">%&gt;%</span></span>
<span id="cb24-14"><a href="#cb24-14" tabindex="-1"></a>          <span class="fu">filter</span>(p.adjust <span class="sc">&lt;=</span> <span class="fl">0.05</span>)</span>
<span id="cb24-15"><a href="#cb24-15" tabindex="-1"></a>      }</span>
<span id="cb24-16"><a href="#cb24-16" tabindex="-1"></a>      <span class="fu">c</span>(</span>
<span id="cb24-17"><a href="#cb24-17" tabindex="-1"></a>        <span class="at">Comparison =</span> i,</span>
<span id="cb24-18"><a href="#cb24-18" tabindex="-1"></a>        <span class="st">`</span><span class="at">Collection</span><span class="st">`</span> <span class="ot">=</span> gc,</span>
<span id="cb24-19"><a href="#cb24-19" tabindex="-1"></a>        <span class="st">`</span><span class="at">Total</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">nrow</span>(x),</span>
<span id="cb24-20"><a href="#cb24-20" tabindex="-1"></a>        <span class="st">`</span><span class="at">Supressed</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(NES <span class="sc">&lt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> nrow,</span>
<span id="cb24-21"><a href="#cb24-21" tabindex="-1"></a>        <span class="st">`</span><span class="at">Activated</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> <span class="fu">filter</span>(NES <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> nrow</span>
<span id="cb24-22"><a href="#cb24-22" tabindex="-1"></a>      )</span>
<span id="cb24-23"><a href="#cb24-23" tabindex="-1"></a>    }</span>
<span id="cb24-24"><a href="#cb24-24" tabindex="-1"></a>  }</span>
<span id="cb24-25"><a href="#cb24-25" tabindex="-1"></a>}</span>
<span id="cb24-26"><a href="#cb24-26" tabindex="-1"></a>res <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>(<span class="at">check.names =</span> F)</span></code></pre></div>
</div>
<div id="dotplots" class="section level3">
<h3>Dotplots</h3>
<!-- # Subchunkify dotplots -->
<!-- ## chunk options need to be set to echo=FALSE, results='asis' -->
<!-- ## Cause results='asis',we must use raw HTML to print text and headings -->
<h4>
<span class="header-section-number"> 5.3.1 </span>Comparison
InVivo_AMHR2_KO + AMHR2_KO_culture–over–InVivo_Control_sgRNA +
AMHR2_Control_culture
</h4>
<p><img src="figure/rnaseq-amhr2_ko-deg.Rmd/gse-dotplot-invivo_amhr2_ko-+-amhr2_ko_culture--over--invivo_control_sgrna-+-amhr2_control_culture_GSEA_msigdb.h-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-gse-dotplot-invivo_amhr2_ko-+-amhr2_ko_culture--over--invivo_control_sgrna-+-amhr2_control_culture_GSEA_msigdb.h-1">
Past versions of
gse-dotplot-invivo_amhr2_ko-+-amhr2_ko_culture–over–invivo_control_sgrna-+-amhr2_control_culture_GSEA_msigdb.h-1.png
</button>
</p>
<div
id="fig-gse-dotplot-invivo_amhr2_ko-+-amhr2_ko_culture--over--invivo_control_sgrna-+-amhr2_control_culture_GSEA_msigdb.h-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/8632845b1c6b024330836a1a798537363e31980e/docs/figure/rnaseq-amhr2_ko-deg.Rmd/gse-dotplot-invivo_amhr2_ko-+-amhr2_ko_culture--over--invivo_control_sgrna-+-amhr2_control_culture_GSEA_msigdb.h-1.png" target="_blank">8632845</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-07-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<h4>
<span class="header-section-number"> 5.3.2 </span>Comparison
AMHR2_KO_noTGFbeta–over–AMHR2_Control_culture
</h4>
<h4>
<span class="header-section-number"> 5.3.3 </span>Comparison
InVivo_AMHR2_KO–over–InVivo_Control_sgRNA
</h4>
<p><img src="figure/rnaseq-amhr2_ko-deg.Rmd/gse-dotplot-invivo_amhr2_ko–over–invivo_control_sgrna_GSEA_msigdb.h-1.png" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="gsea-plots-for-selected-signatures" class="section level2">
<h2>GSEA plots for selected signatures</h2>
<p>Define signatures</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a>manual_gene_sets <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb25-2"><a href="#cb25-2" tabindex="-1"></a>   <span class="co"># Signature 4</span></span>
<span id="cb25-3"><a href="#cb25-3" tabindex="-1"></a>  <span class="st">`</span><span class="at">STAT1 target genes and interactors</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb25-4"><a href="#cb25-4" tabindex="-1"></a>    <span class="st">&#39;Ciita&#39;</span>,</span>
<span id="cb25-5"><a href="#cb25-5" tabindex="-1"></a>    <span class="st">&#39;Socs1&#39;</span>,</span>
<span id="cb25-6"><a href="#cb25-6" tabindex="-1"></a>    <span class="st">&#39;Cxcl10&#39;</span>,</span>
<span id="cb25-7"><a href="#cb25-7" tabindex="-1"></a>    <span class="st">&#39;Irf1&#39;</span>,</span>
<span id="cb25-8"><a href="#cb25-8" tabindex="-1"></a>    <span class="st">&#39;Irf7&#39;</span>,</span>
<span id="cb25-9"><a href="#cb25-9" tabindex="-1"></a>    <span class="st">&#39;Irf9&#39;</span>,</span>
<span id="cb25-10"><a href="#cb25-10" tabindex="-1"></a>    <span class="st">&#39;Stat1&#39;</span>,</span>
<span id="cb25-11"><a href="#cb25-11" tabindex="-1"></a>    <span class="st">&#39;Stat2&#39;</span>,</span>
<span id="cb25-12"><a href="#cb25-12" tabindex="-1"></a>    <span class="st">&#39;Stat3&#39;</span>,</span>
<span id="cb25-13"><a href="#cb25-13" tabindex="-1"></a>    <span class="st">&#39;Stat5a&#39;</span>,</span>
<span id="cb25-14"><a href="#cb25-14" tabindex="-1"></a>    <span class="st">&#39;Stat5b&#39;</span>,</span>
<span id="cb25-15"><a href="#cb25-15" tabindex="-1"></a>    <span class="st">&#39;Sting1&#39;</span>,</span>
<span id="cb25-16"><a href="#cb25-16" tabindex="-1"></a>    <span class="st">&#39;Jak1&#39;</span>,</span>
<span id="cb25-17"><a href="#cb25-17" tabindex="-1"></a>    <span class="st">&#39;Jak2&#39;</span>,</span>
<span id="cb25-18"><a href="#cb25-18" tabindex="-1"></a>    <span class="st">&#39;Tap1&#39;</span>,</span>
<span id="cb25-19"><a href="#cb25-19" tabindex="-1"></a>    <span class="st">&#39;Tap2&#39;</span>,</span>
<span id="cb25-20"><a href="#cb25-20" tabindex="-1"></a>    <span class="st">&#39;Usp18&#39;</span>,</span>
<span id="cb25-21"><a href="#cb25-21" tabindex="-1"></a>    <span class="st">&#39;Cyp1b1&#39;</span>,</span>
<span id="cb25-22"><a href="#cb25-22" tabindex="-1"></a>    <span class="st">&#39;Mx1&#39;</span>,</span>
<span id="cb25-23"><a href="#cb25-23" tabindex="-1"></a>    <span class="st">&#39;Ido1&#39;</span>,</span>
<span id="cb25-24"><a href="#cb25-24" tabindex="-1"></a>    <span class="st">&#39;Bcl6&#39;</span>,</span>
<span id="cb25-25"><a href="#cb25-25" tabindex="-1"></a>    <span class="st">&#39;Gbp5&#39;</span>,</span>
<span id="cb25-26"><a href="#cb25-26" tabindex="-1"></a>    <span class="st">&#39;Egr1&#39;</span>,</span>
<span id="cb25-27"><a href="#cb25-27" tabindex="-1"></a>    <span class="st">&#39;Fos&#39;</span>,</span>
<span id="cb25-28"><a href="#cb25-28" tabindex="-1"></a>    <span class="st">&#39;Lyn&#39;</span></span>
<span id="cb25-29"><a href="#cb25-29" tabindex="-1"></a>  ),</span>
<span id="cb25-30"><a href="#cb25-30" tabindex="-1"></a>  <span class="co"># Signature 2</span></span>
<span id="cb25-31"><a href="#cb25-31" tabindex="-1"></a>  <span class="st">`</span><span class="at">EMT regulators and targets</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb25-32"><a href="#cb25-32" tabindex="-1"></a>    <span class="st">&#39;Epcam&#39;</span>,</span>
<span id="cb25-33"><a href="#cb25-33" tabindex="-1"></a>    <span class="st">&#39;Cdh1&#39;</span>,</span>
<span id="cb25-34"><a href="#cb25-34" tabindex="-1"></a>    <span class="st">&#39;Cdh2&#39;</span>,</span>
<span id="cb25-35"><a href="#cb25-35" tabindex="-1"></a>    <span class="st">&#39;Cldn1&#39;</span>,</span>
<span id="cb25-36"><a href="#cb25-36" tabindex="-1"></a>    <span class="st">&#39;Cldn3&#39;</span>,</span>
<span id="cb25-37"><a href="#cb25-37" tabindex="-1"></a>    <span class="st">&#39;Cldn4&#39;</span>,</span>
<span id="cb25-38"><a href="#cb25-38" tabindex="-1"></a>    <span class="st">&#39;Cldn7&#39;</span>,</span>
<span id="cb25-39"><a href="#cb25-39" tabindex="-1"></a>    <span class="st">&#39;Vim&#39;</span>,</span>
<span id="cb25-40"><a href="#cb25-40" tabindex="-1"></a>    <span class="st">&#39;Acta2&#39;</span>,</span>
<span id="cb25-41"><a href="#cb25-41" tabindex="-1"></a>    <span class="st">&#39;Snai2&#39;</span>,</span>
<span id="cb25-42"><a href="#cb25-42" tabindex="-1"></a>    <span class="st">&#39;Twist1&#39;</span>,</span>
<span id="cb25-43"><a href="#cb25-43" tabindex="-1"></a>    <span class="st">&#39;Zeb1&#39;</span>,</span>
<span id="cb25-44"><a href="#cb25-44" tabindex="-1"></a>    <span class="st">&#39;Zeb2&#39;</span>,</span>
<span id="cb25-45"><a href="#cb25-45" tabindex="-1"></a>    <span class="st">&#39;Grhl2&#39;</span>,</span>
<span id="cb25-46"><a href="#cb25-46" tabindex="-1"></a>    <span class="st">&#39;Esrp1&#39;</span>,</span>
<span id="cb25-47"><a href="#cb25-47" tabindex="-1"></a>    <span class="st">&#39;Esrp2&#39;</span>,</span>
<span id="cb25-48"><a href="#cb25-48" tabindex="-1"></a>    <span class="st">&#39;Marveld2&#39;</span>,</span>
<span id="cb25-49"><a href="#cb25-49" tabindex="-1"></a>    <span class="st">&#39;Rab25&#39;</span>,</span>
<span id="cb25-50"><a href="#cb25-50" tabindex="-1"></a>    <span class="st">&#39;Myo5C&#39;</span>,</span>
<span id="cb25-51"><a href="#cb25-51" tabindex="-1"></a>    <span class="st">&#39;Pkp1&#39;</span>,</span>
<span id="cb25-52"><a href="#cb25-52" tabindex="-1"></a>    <span class="st">&#39;Pkp3&#39;</span>,</span>
<span id="cb25-53"><a href="#cb25-53" tabindex="-1"></a>    <span class="st">&#39;Jup&#39;</span>,</span>
<span id="cb25-54"><a href="#cb25-54" tabindex="-1"></a>    <span class="st">&#39;Dsp&#39;</span>,</span>
<span id="cb25-55"><a href="#cb25-55" tabindex="-1"></a>    <span class="st">&#39;Ppl&#39;</span>,</span>
<span id="cb25-56"><a href="#cb25-56" tabindex="-1"></a>    <span class="st">&#39;Dsg3&#39;</span></span>
<span id="cb25-57"><a href="#cb25-57" tabindex="-1"></a>  ),</span>
<span id="cb25-58"><a href="#cb25-58" tabindex="-1"></a>  <span class="co"># Signature 1</span></span>
<span id="cb25-59"><a href="#cb25-59" tabindex="-1"></a>  <span class="st">`</span><span class="at">Mammary epithelium proliferation</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb25-60"><a href="#cb25-60" tabindex="-1"></a>    <span class="st">&#39;Aurka&#39;</span>,</span>
<span id="cb25-61"><a href="#cb25-61" tabindex="-1"></a>    <span class="st">&#39;Aurkb&#39;</span>,</span>
<span id="cb25-62"><a href="#cb25-62" tabindex="-1"></a>    <span class="st">&#39;Brca1&#39;</span>,</span>
<span id="cb25-63"><a href="#cb25-63" tabindex="-1"></a>    <span class="st">&#39;Brca2&#39;</span>,</span>
<span id="cb25-64"><a href="#cb25-64" tabindex="-1"></a>    <span class="st">&#39;Ccdn1&#39;</span>,</span>
<span id="cb25-65"><a href="#cb25-65" tabindex="-1"></a>    <span class="st">&#39;Cdkn2a&#39;</span>,</span>
<span id="cb25-66"><a href="#cb25-66" tabindex="-1"></a>    <span class="st">&#39;Esr1&#39;</span>,</span>
<span id="cb25-67"><a href="#cb25-67" tabindex="-1"></a>    <span class="st">&#39;Gata3&#39;</span>,</span>
<span id="cb25-68"><a href="#cb25-68" tabindex="-1"></a>    <span class="st">&#39;Hoxa5&#39;</span>,</span>
<span id="cb25-69"><a href="#cb25-69" tabindex="-1"></a>    <span class="st">&#39;Mki67&#39;</span>,</span>
<span id="cb25-70"><a href="#cb25-70" tabindex="-1"></a>    <span class="st">&#39;Plk1&#39;</span>,</span>
<span id="cb25-71"><a href="#cb25-71" tabindex="-1"></a>    <span class="st">&#39;Wnt5A&#39;</span>,</span>
<span id="cb25-72"><a href="#cb25-72" tabindex="-1"></a>    <span class="st">&#39;Agap2&#39;</span>,</span>
<span id="cb25-73"><a href="#cb25-73" tabindex="-1"></a>    <span class="st">&#39;Areg&#39;</span>,</span>
<span id="cb25-74"><a href="#cb25-74" tabindex="-1"></a>    <span class="st">&#39;Bax&#39;</span>,</span>
<span id="cb25-75"><a href="#cb25-75" tabindex="-1"></a>    <span class="st">&#39;Deaf1&#39;</span>,</span>
<span id="cb25-76"><a href="#cb25-76" tabindex="-1"></a>    <span class="st">&#39;Epha2&#39;</span>,</span>
<span id="cb25-77"><a href="#cb25-77" tabindex="-1"></a>    <span class="st">&#39;Etv4&#39;</span>,</span>
<span id="cb25-78"><a href="#cb25-78" tabindex="-1"></a>    <span class="st">&#39;Id2&#39;</span>,</span>
<span id="cb25-79"><a href="#cb25-79" tabindex="-1"></a>    <span class="st">&#39;Kdm5b&#39;</span>,</span>
<span id="cb25-80"><a href="#cb25-80" tabindex="-1"></a>    <span class="st">&#39;Robo1&#39;</span>,</span>
<span id="cb25-81"><a href="#cb25-81" tabindex="-1"></a>    <span class="st">&#39;Mapk1&#39;</span>,</span>
<span id="cb25-82"><a href="#cb25-82" tabindex="-1"></a>    <span class="st">&#39;Med1&#39;</span>,</span>
<span id="cb25-83"><a href="#cb25-83" tabindex="-1"></a>    <span class="st">&#39;Pygo2&#39;</span>,</span>
<span id="cb25-84"><a href="#cb25-84" tabindex="-1"></a>    <span class="st">&#39;Tfap2c&#39;</span></span>
<span id="cb25-85"><a href="#cb25-85" tabindex="-1"></a>  ),</span>
<span id="cb25-86"><a href="#cb25-86" tabindex="-1"></a>  <span class="co"># Signature 3</span></span>
<span id="cb25-87"><a href="#cb25-87" tabindex="-1"></a>  <span class="st">`</span><span class="at">SMAD target genes and interactors</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb25-88"><a href="#cb25-88" tabindex="-1"></a>    <span class="st">&#39;Wnt4&#39;</span>,</span>
<span id="cb25-89"><a href="#cb25-89" tabindex="-1"></a>    <span class="st">&#39;Cited2&#39;</span>,</span>
<span id="cb25-90"><a href="#cb25-90" tabindex="-1"></a>    <span class="st">&#39;Tob1&#39;</span>,</span>
<span id="cb25-91"><a href="#cb25-91" tabindex="-1"></a>    <span class="st">&#39;Acvr1&#39;</span>,</span>
<span id="cb25-92"><a href="#cb25-92" tabindex="-1"></a>    <span class="st">&#39;Hipk2&#39;</span>,</span>
<span id="cb25-93"><a href="#cb25-93" tabindex="-1"></a>    <span class="st">&#39;Ppm1a&#39;</span>,</span>
<span id="cb25-94"><a href="#cb25-94" tabindex="-1"></a>    <span class="st">&#39;Crebbp&#39;</span>,</span>
<span id="cb25-95"><a href="#cb25-95" tabindex="-1"></a>    <span class="st">&#39;Zfyve9&#39;</span>,</span>
<span id="cb25-96"><a href="#cb25-96" tabindex="-1"></a>    <span class="st">&#39;Col5a2&#39;</span>,</span>
<span id="cb25-97"><a href="#cb25-97" tabindex="-1"></a>    <span class="st">&#39;Tgif1&#39;</span>,</span>
<span id="cb25-98"><a href="#cb25-98" tabindex="-1"></a>    <span class="st">&#39;Ankrd1&#39;</span>,</span>
<span id="cb25-99"><a href="#cb25-99" tabindex="-1"></a>    <span class="st">&#39;Smad4&#39;</span>,</span>
<span id="cb25-100"><a href="#cb25-100" tabindex="-1"></a>    <span class="st">&#39;Smurf1&#39;</span>,</span>
<span id="cb25-101"><a href="#cb25-101" tabindex="-1"></a>    <span class="st">&#39;Magi2&#39;</span>,</span>
<span id="cb25-102"><a href="#cb25-102" tabindex="-1"></a>    <span class="st">&#39;Zmiz1&#39;</span>,</span>
<span id="cb25-103"><a href="#cb25-103" tabindex="-1"></a>    <span class="st">&#39;Trim33&#39;</span>,</span>
<span id="cb25-104"><a href="#cb25-104" tabindex="-1"></a>    <span class="st">&#39;Skil&#39;</span>,</span>
<span id="cb25-105"><a href="#cb25-105" tabindex="-1"></a>    <span class="st">&#39;Ipo7&#39;</span>,</span>
<span id="cb25-106"><a href="#cb25-106" tabindex="-1"></a>    <span class="st">&#39;Mef2a&#39;</span>,</span>
<span id="cb25-107"><a href="#cb25-107" tabindex="-1"></a>    <span class="st">&#39;Smad9&#39;</span>,</span>
<span id="cb25-108"><a href="#cb25-108" tabindex="-1"></a>    <span class="st">&#39;Usp9x&#39;</span>,</span>
<span id="cb25-109"><a href="#cb25-109" tabindex="-1"></a>    <span class="st">&#39;Rnf111&#39;</span>,</span>
<span id="cb25-110"><a href="#cb25-110" tabindex="-1"></a>    <span class="st">&#39;Tcf12&#39;</span>,</span>
<span id="cb25-111"><a href="#cb25-111" tabindex="-1"></a>    <span class="st">&#39;Hmga2&#39;</span>,</span>
<span id="cb25-112"><a href="#cb25-112" tabindex="-1"></a>    <span class="st">&#39;Foxa2&#39;</span></span>
<span id="cb25-113"><a href="#cb25-113" tabindex="-1"></a>  )</span>
<span id="cb25-114"><a href="#cb25-114" tabindex="-1"></a>)</span>
<span id="cb25-115"><a href="#cb25-115" tabindex="-1"></a></span>
<span id="cb25-116"><a href="#cb25-116" tabindex="-1"></a></span>
<span id="cb25-117"><a href="#cb25-117" tabindex="-1"></a>i <span class="ot">&lt;-</span> <span class="fu">names</span>(manual_gene_sets)[<span class="dv">1</span>]</span>
<span id="cb25-118"><a href="#cb25-118" tabindex="-1"></a>manual_term2gene <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i =</span> <span class="fu">names</span>(manual_gene_sets), <span class="at">.combine =</span> rbind) <span class="sc">%do%</span> {</span>
<span id="cb25-119"><a href="#cb25-119" tabindex="-1"></a>  <span class="fu">cbind</span>(</span>
<span id="cb25-120"><a href="#cb25-120" tabindex="-1"></a>    <span class="at">term =</span> <span class="fu">rep</span>(i, <span class="fu">length</span>(manual_gene_sets[[i]])), </span>
<span id="cb25-121"><a href="#cb25-121" tabindex="-1"></a>    <span class="at">gene =</span> manual_gene_sets[[i]]</span>
<span id="cb25-122"><a href="#cb25-122" tabindex="-1"></a>    )</span>
<span id="cb25-123"><a href="#cb25-123" tabindex="-1"></a>}</span></code></pre></div>
<p>Run functional enrichment</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a>dge <span class="ot">&lt;-</span>  dge_list[[<span class="dv">2</span>]]</span>
<span id="cb26-3"><a href="#cb26-3" tabindex="-1"></a></span>
<span id="cb26-4"><a href="#cb26-4" tabindex="-1"></a>gse_manual_list <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">dge =</span> dge_list) <span class="sc">%do%</span>{</span>
<span id="cb26-5"><a href="#cb26-5" tabindex="-1"></a>  feature_fc_list <span class="ot">&lt;-</span> dge<span class="sc">$</span>results<span class="sc">$</span>logFC <span class="sc">%&gt;%</span> </span>
<span id="cb26-6"><a href="#cb26-6" tabindex="-1"></a>    <span class="fu">set_names</span>(<span class="fu">make.names</span>(dge<span class="sc">$</span>results<span class="sc">$</span>gene_name, <span class="at">unique =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-7"><a href="#cb26-7" tabindex="-1"></a>    <span class="fu">sort</span>(<span class="at">decreasing =</span> <span class="cn">TRUE</span>)</span>
<span id="cb26-8"><a href="#cb26-8" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">GSEA</span>(feature_fc_list, </span>
<span id="cb26-9"><a href="#cb26-9" tabindex="-1"></a>              <span class="at">TERM2GENE=</span>manual_term2gene,</span>
<span id="cb26-10"><a href="#cb26-10" tabindex="-1"></a>              <span class="at">minGSSize =</span> <span class="dv">0</span>,</span>
<span id="cb26-11"><a href="#cb26-11" tabindex="-1"></a>              <span class="at">maxGSSize =</span> <span class="cn">Inf</span>,</span>
<span id="cb26-12"><a href="#cb26-12" tabindex="-1"></a>              <span class="at">pvalueCutoff =</span> <span class="dv">1</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb26-13"><a href="#cb26-13" tabindex="-1"></a>  res</span>
<span id="cb26-14"><a href="#cb26-14" tabindex="-1"></a>}</span>
<span id="cb26-15"><a href="#cb26-15" tabindex="-1"></a></span>
<span id="cb26-16"><a href="#cb26-16" tabindex="-1"></a><span class="fu">names</span>(gse_manual_list) <span class="ot">&lt;-</span> <span class="fu">names</span>(dge_list)</span>
<span id="cb26-17"><a href="#cb26-17" tabindex="-1"></a></span>
<span id="cb26-18"><a href="#cb26-18" tabindex="-1"></a><span class="co"># Save objects</span></span>
<span id="cb26-19"><a href="#cb26-19" tabindex="-1"></a><span class="fu">saveRDS</span>(gse_manual_list, <span class="at">file =</span> <span class="fu">file.path</span>(params<span class="sc">$</span>output_dir,<span class="st">&#39;gsea-edgeR_QLF-signatures.rds&#39;</span>))</span></code></pre></div>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a>gse_manual_list <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="fu">file.path</span>(params<span class="sc">$</span>output_dir,<span class="st">&#39;gsea-edgeR_QLF-signatures.rds&#39;</span>))</span>
<span id="cb27-2"><a href="#cb27-2" tabindex="-1"></a>comp_names <span class="ot">&lt;-</span> comp_list_description<span class="sc">$</span>comparison_long <span class="sc">%&gt;%</span> </span>
<span id="cb27-3"><a href="#cb27-3" tabindex="-1"></a>  <span class="fu">set_names</span>(comp_list_description<span class="sc">$</span>comparison)</span>
<span id="cb27-4"><a href="#cb27-4" tabindex="-1"></a><span class="fu">names</span>(gse_manual_list) <span class="ot">&lt;-</span> comp_names[<span class="fu">names</span>(gse_manual_list)]</span></code></pre></div>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a>gseaScores <span class="ot">&lt;-</span> <span class="fu">getFromNamespace</span>(<span class="st">&quot;gseaScores&quot;</span>, <span class="st">&quot;DOSE&quot;</span>)</span>
<span id="cb28-2"><a href="#cb28-2" tabindex="-1"></a>gsInfo <span class="ot">&lt;-</span> <span class="fu">getFromNamespace</span>(<span class="st">&quot;gsInfo&quot;</span>, <span class="st">&quot;enrichplot&quot;</span>)</span>
<span id="cb28-3"><a href="#cb28-3" tabindex="-1"></a>gseaplot2_custom <span class="ot">&lt;-</span> <span class="cf">function</span> (</span>
<span id="cb28-4"><a href="#cb28-4" tabindex="-1"></a>    x, </span>
<span id="cb28-5"><a href="#cb28-5" tabindex="-1"></a>    geneSetID, </span>
<span id="cb28-6"><a href="#cb28-6" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">&quot;&quot;</span>, </span>
<span id="cb28-7"><a href="#cb28-7" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">&quot;green&quot;</span>, </span>
<span id="cb28-8"><a href="#cb28-8" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">11</span>, </span>
<span id="cb28-9"><a href="#cb28-9" tabindex="-1"></a>    <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="fl">1.5</span>, <span class="fl">0.5</span>, <span class="dv">1</span>), <span class="at">subplots =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="at">pvalue_table =</span> <span class="cn">FALSE</span>, </span>
<span id="cb28-10"><a href="#cb28-10" tabindex="-1"></a>    <span class="at">ES_geom =</span> <span class="st">&quot;line&quot;</span>,</span>
<span id="cb28-11"><a href="#cb28-11" tabindex="-1"></a>    <span class="at">axis.linewidth =</span> (<span class="dv">1</span><span class="sc">/</span><span class="fl">2.141959</span>)<span class="sc">/</span><span class="dv">4</span>,</span>
<span id="cb28-12"><a href="#cb28-12" tabindex="-1"></a>    <span class="at">pres.linewidth =</span> <span class="dv">1</span>,</span>
<span id="cb28-13"><a href="#cb28-13" tabindex="-1"></a>    <span class="at">p2.fill =</span> <span class="st">&#39;transparent&#39;</span>,</span>
<span id="cb28-14"><a href="#cb28-14" tabindex="-1"></a>    <span class="at">p2.linerange.linewidth =</span> <span class="fl">0.5</span>,</span>
<span id="cb28-15"><a href="#cb28-15" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="fu">c</span>(<span class="fl">0.8</span>, <span class="fl">0.8</span>),</span>
<span id="cb28-16"><a href="#cb28-16" tabindex="-1"></a>    <span class="at">p.pos.ylab =</span> <span class="st">&#39;Ranked List Metric&#39;</span>,</span>
<span id="cb28-17"><a href="#cb28-17" tabindex="-1"></a>    <span class="at">p2.labels =</span> <span class="cn">FALSE</span>, <span class="co"># Only works when there are two geneSetID</span></span>
<span id="cb28-18"><a href="#cb28-18" tabindex="-1"></a>    <span class="at">reverse.rank =</span> <span class="cn">FALSE</span>,</span>
<span id="cb28-19"><a href="#cb28-19" tabindex="-1"></a>    <span class="at">p2.add.intervals.colors =</span> <span class="cn">FALSE</span>,</span>
<span id="cb28-20"><a href="#cb28-20" tabindex="-1"></a>    <span class="at">p2.segment.color.var =</span> <span class="st">&#39;Description&#39;</span>, <span class="co"># Either Description or geneList (log2FC)</span></span>
<span id="cb28-21"><a href="#cb28-21" tabindex="-1"></a>    <span class="at">geneList.limit =</span> <span class="cn">NULL</span>,</span>
<span id="cb28-22"><a href="#cb28-22" tabindex="-1"></a>    <span class="at">fontface =</span> <span class="st">&quot;italic&quot;</span> </span>
<span id="cb28-23"><a href="#cb28-23" tabindex="-1"></a>    )</span>
<span id="cb28-24"><a href="#cb28-24" tabindex="-1"></a>{</span>
<span id="cb28-25"><a href="#cb28-25" tabindex="-1"></a>    <span class="do">##</span></span>
<span id="cb28-26"><a href="#cb28-26" tabindex="-1"></a>    <span class="do">## Enrichmment score plot</span></span>
<span id="cb28-27"><a href="#cb28-27" tabindex="-1"></a>    <span class="do">##</span></span>
<span id="cb28-28"><a href="#cb28-28" tabindex="-1"></a>    ES_geom <span class="ot">&lt;-</span> <span class="fu">match.arg</span>(ES_geom, <span class="fu">c</span>(<span class="st">&quot;line&quot;</span>, <span class="st">&quot;dot&quot;</span>))</span>
<span id="cb28-29"><a href="#cb28-29" tabindex="-1"></a>    geneList <span class="ot">&lt;-</span> position <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb28-30"><a href="#cb28-30" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">length</span>(geneSetID) <span class="sc">==</span> <span class="dv">1</span>) {</span>
<span id="cb28-31"><a href="#cb28-31" tabindex="-1"></a>        gsdata <span class="ot">&lt;-</span> <span class="fu">gsInfo</span>(x, geneSetID)</span>
<span id="cb28-32"><a href="#cb28-32" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb28-33"><a href="#cb28-33" tabindex="-1"></a>        gsdata <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, <span class="fu">lapply</span>(geneSetID, gsInfo, <span class="at">object =</span> x))</span>
<span id="cb28-34"><a href="#cb28-34" tabindex="-1"></a>    }</span>
<span id="cb28-35"><a href="#cb28-35" tabindex="-1"></a>    </span>
<span id="cb28-36"><a href="#cb28-36" tabindex="-1"></a>    <span class="co"># Reverse rank</span></span>
<span id="cb28-37"><a href="#cb28-37" tabindex="-1"></a>    <span class="cf">if</span>(reverse.rank) {</span>
<span id="cb28-38"><a href="#cb28-38" tabindex="-1"></a>      gsdata<span class="sc">$</span>x <span class="ot">&lt;-</span> <span class="fu">rev</span>(gsdata<span class="sc">$</span>x)</span>
<span id="cb28-39"><a href="#cb28-39" tabindex="-1"></a>    }</span>
<span id="cb28-40"><a href="#cb28-40" tabindex="-1"></a>    </span>
<span id="cb28-41"><a href="#cb28-41" tabindex="-1"></a>    <span class="co"># Cap the geneList values</span></span>
<span id="cb28-42"><a href="#cb28-42" tabindex="-1"></a>    <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(geneList.limit)) {</span>
<span id="cb28-43"><a href="#cb28-43" tabindex="-1"></a>      gsdata<span class="sc">$</span>geneListCapped <span class="ot">&lt;-</span> gsdata<span class="sc">$</span>geneList</span>
<span id="cb28-44"><a href="#cb28-44" tabindex="-1"></a>      gsdata<span class="sc">$</span>geneListCapped[gsdata<span class="sc">$</span>geneListCapped <span class="sc">&gt;</span> geneList.limit] <span class="ot">&lt;-</span> geneList.limit</span>
<span id="cb28-45"><a href="#cb28-45" tabindex="-1"></a>      gsdata<span class="sc">$</span>geneListCapped[gsdata<span class="sc">$</span>geneListCapped <span class="sc">&lt;</span> <span class="sc">-</span>geneList.limit] <span class="ot">&lt;-</span> <span class="sc">-</span>geneList.limit</span>
<span id="cb28-46"><a href="#cb28-46" tabindex="-1"></a>      p2.segment.color.var <span class="ot">&lt;-</span> <span class="st">&#39;geneListCapped&#39;</span></span>
<span id="cb28-47"><a href="#cb28-47" tabindex="-1"></a>    }</span>
<span id="cb28-48"><a href="#cb28-48" tabindex="-1"></a>    </span>
<span id="cb28-49"><a href="#cb28-49" tabindex="-1"></a>    p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(gsdata, <span class="fu">aes_</span>(<span class="at">x =</span> <span class="sc">~</span>x)) <span class="sc">+</span> </span>
<span id="cb28-50"><a href="#cb28-50" tabindex="-1"></a>      <span class="fu">xlab</span>(<span class="cn">NULL</span>) <span class="sc">+</span> </span>
<span id="cb28-51"><a href="#cb28-51" tabindex="-1"></a>      <span class="fu">theme_classic</span>(base_size) <span class="sc">+</span> </span>
<span id="cb28-52"><a href="#cb28-52" tabindex="-1"></a>      <span class="fu">theme</span>(</span>
<span id="cb28-53"><a href="#cb28-53" tabindex="-1"></a>        <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb28-54"><a href="#cb28-54" tabindex="-1"></a>        <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb28-55"><a href="#cb28-55" tabindex="-1"></a>        <span class="at">panel.grid.major.y =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb28-56"><a href="#cb28-56" tabindex="-1"></a>        <span class="at">panel.grid.minor.y =</span> <span class="fu">element_blank</span>()</span>
<span id="cb28-57"><a href="#cb28-57" tabindex="-1"></a>      ) <span class="sc">+</span> </span>
<span id="cb28-58"><a href="#cb28-58" tabindex="-1"></a>      <span class="fu">scale_x_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>))</span>
<span id="cb28-59"><a href="#cb28-59" tabindex="-1"></a>    </span>
<span id="cb28-60"><a href="#cb28-60" tabindex="-1"></a>    <span class="cf">if</span> (ES_geom <span class="sc">==</span> <span class="st">&quot;line&quot;</span>) {</span>
<span id="cb28-61"><a href="#cb28-61" tabindex="-1"></a>        es_layer <span class="ot">&lt;-</span> <span class="fu">geom_line</span>(<span class="fu">aes_</span>(<span class="at">y =</span> <span class="sc">~</span>runningScore, <span class="at">color =</span> <span class="sc">~</span>Description), </span>
<span id="cb28-62"><a href="#cb28-62" tabindex="-1"></a>            <span class="at">linewidth =</span> <span class="dv">1</span>)</span>
<span id="cb28-63"><a href="#cb28-63" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb28-64"><a href="#cb28-64" tabindex="-1"></a>        es_layer <span class="ot">&lt;-</span> <span class="fu">geom_point</span>(<span class="fu">aes_</span>(<span class="at">y =</span> <span class="sc">~</span>runningScore, <span class="at">color =</span> <span class="sc">~</span>Description), </span>
<span id="cb28-65"><a href="#cb28-65" tabindex="-1"></a>            <span class="at">linewidth =</span> <span class="dv">1</span>, <span class="at">data =</span> <span class="fu">subset</span>(gsdata, position <span class="sc">==</span> <span class="dv">1</span>))</span>
<span id="cb28-66"><a href="#cb28-66" tabindex="-1"></a>    }</span>
<span id="cb28-67"><a href="#cb28-67" tabindex="-1"></a>    p.res <span class="ot">&lt;-</span> p <span class="sc">+</span> </span>
<span id="cb28-68"><a href="#cb28-68" tabindex="-1"></a>      es_layer <span class="sc">+</span> </span>
<span id="cb28-69"><a href="#cb28-69" tabindex="-1"></a>      <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="at">color =</span> <span class="st">&quot;grey&quot;</span>) <span class="sc">+</span></span>
<span id="cb28-70"><a href="#cb28-70" tabindex="-1"></a>      <span class="fu">theme</span>(</span>
<span id="cb28-71"><a href="#cb28-71" tabindex="-1"></a>        <span class="at">legend.position =</span> legend.position,</span>
<span id="cb28-72"><a href="#cb28-72" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb28-73"><a href="#cb28-73" tabindex="-1"></a>        <span class="at">legend.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">&quot;transparent&quot;</span>)</span>
<span id="cb28-74"><a href="#cb28-74" tabindex="-1"></a>        ) <span class="sc">+</span> </span>
<span id="cb28-75"><a href="#cb28-75" tabindex="-1"></a>      <span class="fu">ylab</span>(<span class="st">&quot;Running Enrichment Score&quot;</span>) <span class="sc">+</span> </span>
<span id="cb28-76"><a href="#cb28-76" tabindex="-1"></a>      <span class="fu">theme</span>(</span>
<span id="cb28-77"><a href="#cb28-77" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb28-78"><a href="#cb28-78" tabindex="-1"></a>        <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb28-79"><a href="#cb28-79" tabindex="-1"></a>        <span class="at">axis.line.x =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb28-80"><a href="#cb28-80" tabindex="-1"></a>        <span class="at">axis.line.y =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb28-81"><a href="#cb28-81" tabindex="-1"></a>        <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span><span class="st">&quot;transparent&quot;</span>,<span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">linewidth =</span> axis.linewidth)</span>
<span id="cb28-82"><a href="#cb28-82" tabindex="-1"></a>        )</span>
<span id="cb28-83"><a href="#cb28-83" tabindex="-1"></a>    <span class="do">##</span></span>
<span id="cb28-84"><a href="#cb28-84" tabindex="-1"></a>    <span class="do">## Plot gene position in rank</span></span>
<span id="cb28-85"><a href="#cb28-85" tabindex="-1"></a>    <span class="do">##</span></span>
<span id="cb28-86"><a href="#cb28-86" tabindex="-1"></a>    i <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb28-87"><a href="#cb28-87" tabindex="-1"></a>    <span class="cf">for</span> (term <span class="cf">in</span> <span class="fu">unique</span>(gsdata<span class="sc">$</span>Description)) {</span>
<span id="cb28-88"><a href="#cb28-88" tabindex="-1"></a>        idx <span class="ot">&lt;-</span> <span class="fu">which</span>(gsdata<span class="sc">$</span>ymin <span class="sc">!=</span> <span class="dv">0</span> <span class="sc">&amp;</span> gsdata<span class="sc">$</span>Description <span class="sc">==</span> </span>
<span id="cb28-89"><a href="#cb28-89" tabindex="-1"></a>            term)</span>
<span id="cb28-90"><a href="#cb28-90" tabindex="-1"></a>        gsdata[idx, <span class="st">&quot;ymin&quot;</span>] <span class="ot">&lt;-</span> i</span>
<span id="cb28-91"><a href="#cb28-91" tabindex="-1"></a>        gsdata[idx, <span class="st">&quot;ymax&quot;</span>] <span class="ot">&lt;-</span> i <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb28-92"><a href="#cb28-92" tabindex="-1"></a>        i <span class="ot">&lt;-</span> i <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb28-93"><a href="#cb28-93" tabindex="-1"></a>    }</span>
<span id="cb28-94"><a href="#cb28-94" tabindex="-1"></a>    </span>
<span id="cb28-95"><a href="#cb28-95" tabindex="-1"></a>    p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(gsdata, <span class="fu">aes_</span>(<span class="at">x =</span> <span class="sc">~</span>x)) <span class="sc">+</span> </span>
<span id="cb28-96"><a href="#cb28-96" tabindex="-1"></a>      <span class="fu">annotate</span>(<span class="st">&quot;rect&quot;</span>, <span class="at">xmin=</span><span class="dv">0</span>, <span class="at">xmax=</span><span class="dv">100</span><span class="sc">+</span><span class="fu">max</span>(gsdata<span class="sc">$</span>x), <span class="at">ymin=</span><span class="dv">0</span> , <span class="at">ymax=</span><span class="dv">2</span>, <span class="at">alpha=</span><span class="fl">0.8</span>, <span class="at">color=</span>p2.fill, <span class="at">fill=</span>p2.fill) <span class="sc">+</span></span>
<span id="cb28-97"><a href="#cb28-97" tabindex="-1"></a>      <span class="fu">geom_linerange</span>(<span class="fu">aes_string</span>(<span class="at">ymin =</span> <span class="st">&#39;ymin&#39;</span>, <span class="at">ymax =</span> <span class="st">&#39;ymax&#39;</span>, <span class="at">color =</span> p2.segment.color.var), <span class="at">linewidth =</span> p2.linerange.linewidth) <span class="sc">+</span> </span>
<span id="cb28-98"><a href="#cb28-98" tabindex="-1"></a>      <span class="fu">xlab</span>(<span class="cn">NULL</span>) <span class="sc">+</span> </span>
<span id="cb28-99"><a href="#cb28-99" tabindex="-1"></a>      <span class="fu">ylab</span>(<span class="cn">NULL</span>) <span class="sc">+</span> </span>
<span id="cb28-100"><a href="#cb28-100" tabindex="-1"></a>      <span class="fu">theme_classic</span>(base_size) <span class="sc">+</span> </span>
<span id="cb28-101"><a href="#cb28-101" tabindex="-1"></a>      <span class="fu">theme</span>(</span>
<span id="cb28-102"><a href="#cb28-102" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>, </span>
<span id="cb28-103"><a href="#cb28-103" tabindex="-1"></a>        <span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="at">t =</span> <span class="sc">-</span><span class="fl">0.1</span>, <span class="at">b =</span> <span class="dv">0</span>, <span class="at">unit =</span> <span class="st">&quot;cm&quot;</span>), </span>
<span id="cb28-104"><a href="#cb28-104" tabindex="-1"></a>        <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb28-105"><a href="#cb28-105" tabindex="-1"></a>        <span class="at">axis.text =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb28-106"><a href="#cb28-106" tabindex="-1"></a>        <span class="at">axis.line.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb28-107"><a href="#cb28-107" tabindex="-1"></a>        <span class="at">axis.line.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb28-108"><a href="#cb28-108" tabindex="-1"></a>        <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span><span class="st">&quot;transparent&quot;</span>,<span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">linewidth =</span> axis.linewidth)</span>
<span id="cb28-109"><a href="#cb28-109" tabindex="-1"></a>        ) <span class="sc">+</span> </span>
<span id="cb28-110"><a href="#cb28-110" tabindex="-1"></a>      <span class="fu">scale_x_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span> </span>
<span id="cb28-111"><a href="#cb28-111" tabindex="-1"></a>      <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>))</span>
<span id="cb28-112"><a href="#cb28-112" tabindex="-1"></a>    </span>
<span id="cb28-113"><a href="#cb28-113" tabindex="-1"></a>    <span class="cf">if</span>(p2.labels <span class="sc">==</span> <span class="cn">TRUE</span> <span class="sc">&amp;</span> <span class="fu">length</span>(geneSetID) <span class="sc">==</span> <span class="dv">2</span>) {</span>
<span id="cb28-114"><a href="#cb28-114" tabindex="-1"></a>      gsdata_labels <span class="ot">&lt;-</span> gsdata <span class="sc">%&gt;%</span> <span class="fu">filter</span>(ymax <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb28-115"><a href="#cb28-115" tabindex="-1"></a>      max_y <span class="ot">&lt;-</span> <span class="fu">max</span>(gsdata_labels<span class="sc">$</span>ymax)</span>
<span id="cb28-116"><a href="#cb28-116" tabindex="-1"></a>      </span>
<span id="cb28-117"><a href="#cb28-117" tabindex="-1"></a>      description_labels <span class="ot">&lt;-</span> <span class="fu">unique</span>(gsdata_labels<span class="sc">$</span>Description) </span>
<span id="cb28-118"><a href="#cb28-118" tabindex="-1"></a>      description_cut_point <span class="ot">&lt;-</span> <span class="fu">ceiling</span>(<span class="fu">length</span>(description_labels) <span class="sc">/</span> <span class="dv">2</span>)</span>
<span id="cb28-119"><a href="#cb28-119" tabindex="-1"></a>      description_bottom <span class="ot">&lt;-</span> description_labels[<span class="dv">1</span><span class="sc">:</span>description_cut_point]</span>
<span id="cb28-120"><a href="#cb28-120" tabindex="-1"></a>      description_top <span class="ot">&lt;-</span> description_labels[(description_cut_point<span class="sc">+</span><span class="dv">1</span>)<span class="sc">:</span><span class="fu">length</span>(description_labels)]</span>
<span id="cb28-121"><a href="#cb28-121" tabindex="-1"></a>      </span>
<span id="cb28-122"><a href="#cb28-122" tabindex="-1"></a>      gsdata_labels_bottom <span class="ot">&lt;-</span> gsdata_labels <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Description <span class="sc">%in%</span> description_bottom)</span>
<span id="cb28-123"><a href="#cb28-123" tabindex="-1"></a>      gsdata_labels_top <span class="ot">&lt;-</span> gsdata_labels <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Description <span class="sc">%in%</span> description_top)</span>
<span id="cb28-124"><a href="#cb28-124" tabindex="-1"></a>      max_y <span class="ot">&lt;-</span> <span class="fu">max</span>(gsdata_labels<span class="sc">$</span>ymax)</span>
<span id="cb28-125"><a href="#cb28-125" tabindex="-1"></a>      p2 <span class="ot">&lt;-</span> p2 <span class="sc">+</span> </span>
<span id="cb28-126"><a href="#cb28-126" tabindex="-1"></a>        <span class="fu">geom_text_repel</span>(<span class="at">data =</span> gsdata_labels_top,</span>
<span id="cb28-127"><a href="#cb28-127" tabindex="-1"></a>                        <span class="fu">aes</span>(<span class="at">y =</span> max_y, <span class="at">label =</span> gene),</span>
<span id="cb28-128"><a href="#cb28-128" tabindex="-1"></a>                        <span class="at">force_pull   =</span> <span class="dv">0</span>, <span class="co"># do not pull toward data points</span></span>
<span id="cb28-129"><a href="#cb28-129" tabindex="-1"></a>                        <span class="at">nudge_y      =</span> max_y <span class="sc">*</span> <span class="fl">0.15</span>,</span>
<span id="cb28-130"><a href="#cb28-130" tabindex="-1"></a>                        <span class="at">direction    =</span> <span class="st">&quot;x&quot;</span>,</span>
<span id="cb28-131"><a href="#cb28-131" tabindex="-1"></a>                        <span class="at">ylim =</span> <span class="fu">c</span>(<span class="fl">2.5</span>, <span class="dv">4</span>),</span>
<span id="cb28-132"><a href="#cb28-132" tabindex="-1"></a>                        <span class="at">angle        =</span> <span class="dv">90</span>,</span>
<span id="cb28-133"><a href="#cb28-133" tabindex="-1"></a>                        <span class="at">hjust        =</span> <span class="dv">0</span>,</span>
<span id="cb28-134"><a href="#cb28-134" tabindex="-1"></a>                        <span class="at">max.time =</span> <span class="dv">3</span>,</span>
<span id="cb28-135"><a href="#cb28-135" tabindex="-1"></a>                        <span class="at">max.overlaps =</span> <span class="cn">Inf</span>,</span>
<span id="cb28-136"><a href="#cb28-136" tabindex="-1"></a>                        <span class="at">min.segment.length =</span> <span class="dv">0</span>,</span>
<span id="cb28-137"><a href="#cb28-137" tabindex="-1"></a>                        <span class="at">segment.size =</span> <span class="fl">0.2</span>,</span>
<span id="cb28-138"><a href="#cb28-138" tabindex="-1"></a>                        <span class="at">segment.square =</span> <span class="cn">TRUE</span>,</span>
<span id="cb28-139"><a href="#cb28-139" tabindex="-1"></a>                        <span class="at">segment.inflect =</span> <span class="cn">FALSE</span>,</span>
<span id="cb28-140"><a href="#cb28-140" tabindex="-1"></a>                        <span class="at">segment.curvature =</span> <span class="dv">1</span>,</span>
<span id="cb28-141"><a href="#cb28-141" tabindex="-1"></a>                        <span class="at">segment.angle =</span> <span class="sc">-</span><span class="dv">1</span>,</span>
<span id="cb28-142"><a href="#cb28-142" tabindex="-1"></a>                        <span class="at">box.padding =</span> <span class="fl">0.075</span>,</span>
<span id="cb28-143"><a href="#cb28-143" tabindex="-1"></a>                        <span class="at">fontface =</span> <span class="st">&quot;italic&quot;</span></span>
<span id="cb28-144"><a href="#cb28-144" tabindex="-1"></a>        ) <span class="sc">+</span></span>
<span id="cb28-145"><a href="#cb28-145" tabindex="-1"></a>        <span class="fu">geom_text_repel</span>(<span class="at">data =</span> gsdata_labels_bottom,</span>
<span id="cb28-146"><a href="#cb28-146" tabindex="-1"></a>                        <span class="fu">aes</span>(<span class="at">y =</span> <span class="dv">0</span>, <span class="at">label =</span> gene),</span>
<span id="cb28-147"><a href="#cb28-147" tabindex="-1"></a>                        <span class="at">force_pull   =</span> <span class="dv">0</span>, </span>
<span id="cb28-148"><a href="#cb28-148" tabindex="-1"></a>                        <span class="at">nudge_y      =</span> <span class="sc">-</span>(max_y <span class="sc">*</span> <span class="fl">0.15</span>),</span>
<span id="cb28-149"><a href="#cb28-149" tabindex="-1"></a>                        <span class="at">direction    =</span> <span class="st">&quot;x&quot;</span>,</span>
<span id="cb28-150"><a href="#cb28-150" tabindex="-1"></a>                        <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>, <span class="sc">-</span><span class="fl">0.5</span>),</span>
<span id="cb28-151"><a href="#cb28-151" tabindex="-1"></a>                        <span class="at">angle        =</span> <span class="dv">90</span>,</span>
<span id="cb28-152"><a href="#cb28-152" tabindex="-1"></a>                        <span class="at">hjust        =</span> <span class="dv">1</span>,</span>
<span id="cb28-153"><a href="#cb28-153" tabindex="-1"></a>                        <span class="at">max.time =</span> <span class="dv">3</span>,</span>
<span id="cb28-154"><a href="#cb28-154" tabindex="-1"></a>                        <span class="at">max.overlaps =</span> <span class="cn">Inf</span>,</span>
<span id="cb28-155"><a href="#cb28-155" tabindex="-1"></a>                        <span class="at">min.segment.length =</span> <span class="dv">0</span>,</span>
<span id="cb28-156"><a href="#cb28-156" tabindex="-1"></a>                        <span class="at">segment.size =</span> <span class="fl">0.2</span>,</span>
<span id="cb28-157"><a href="#cb28-157" tabindex="-1"></a>                        <span class="at">segment.square =</span> <span class="cn">TRUE</span>,</span>
<span id="cb28-158"><a href="#cb28-158" tabindex="-1"></a>                        <span class="at">segment.inflect =</span> <span class="cn">FALSE</span>,</span>
<span id="cb28-159"><a href="#cb28-159" tabindex="-1"></a>                        <span class="at">segment.curvature =</span> <span class="dv">1</span>,</span>
<span id="cb28-160"><a href="#cb28-160" tabindex="-1"></a>                        <span class="at">segment.angle =</span> <span class="sc">-</span><span class="dv">1</span>,</span>
<span id="cb28-161"><a href="#cb28-161" tabindex="-1"></a>                        <span class="at">box.padding =</span> <span class="fl">0.075</span>,</span>
<span id="cb28-162"><a href="#cb28-162" tabindex="-1"></a>                        <span class="at">fontface =</span> <span class="st">&quot;italic&quot;</span></span>
<span id="cb28-163"><a href="#cb28-163" tabindex="-1"></a>        ) <span class="sc">+</span></span>
<span id="cb28-164"><a href="#cb28-164" tabindex="-1"></a>        <span class="fu">coord_cartesian</span>(<span class="at">expand =</span> T, <span class="at">clip =</span> <span class="st">&quot;off&quot;</span>) <span class="sc">+</span></span>
<span id="cb28-165"><a href="#cb28-165" tabindex="-1"></a>        <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">3</span>)</span>
<span id="cb28-166"><a href="#cb28-166" tabindex="-1"></a>        </span>
<span id="cb28-167"><a href="#cb28-167" tabindex="-1"></a>      </span>
<span id="cb28-168"><a href="#cb28-168" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (p2.labels <span class="sc">==</span> <span class="cn">TRUE</span> <span class="sc">&amp;</span> <span class="fu">length</span>(geneSetID) <span class="sc">==</span> <span class="dv">1</span>) {</span>
<span id="cb28-169"><a href="#cb28-169" tabindex="-1"></a>      gsdata_labels <span class="ot">&lt;-</span> gsdata <span class="sc">%&gt;%</span> <span class="fu">filter</span>(ymax <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb28-170"><a href="#cb28-170" tabindex="-1"></a>      max_y <span class="ot">&lt;-</span> <span class="fu">max</span>(gsdata_labels<span class="sc">$</span>ymax)</span>
<span id="cb28-171"><a href="#cb28-171" tabindex="-1"></a>      </span>
<span id="cb28-172"><a href="#cb28-172" tabindex="-1"></a>      description_labels <span class="ot">&lt;-</span> <span class="fu">unique</span>(gsdata_labels<span class="sc">$</span>Description) </span>
<span id="cb28-173"><a href="#cb28-173" tabindex="-1"></a>      description_cut_point <span class="ot">&lt;-</span> <span class="fu">ceiling</span>(<span class="fu">length</span>(description_labels) <span class="sc">/</span> <span class="dv">2</span>)</span>
<span id="cb28-174"><a href="#cb28-174" tabindex="-1"></a>      description_top <span class="ot">&lt;-</span> description_labels[(description_cut_point<span class="sc">+</span><span class="dv">1</span>)<span class="sc">:</span><span class="fu">length</span>(description_labels)]</span>
<span id="cb28-175"><a href="#cb28-175" tabindex="-1"></a>      </span>
<span id="cb28-176"><a href="#cb28-176" tabindex="-1"></a>      gsdata_labels_top <span class="ot">&lt;-</span> gsdata_labels <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Description <span class="sc">%in%</span> description_top)</span>
<span id="cb28-177"><a href="#cb28-177" tabindex="-1"></a>      max_y <span class="ot">&lt;-</span> <span class="fu">max</span>(gsdata_labels<span class="sc">$</span>ymax)</span>
<span id="cb28-178"><a href="#cb28-178" tabindex="-1"></a>      p2 <span class="ot">&lt;-</span> p2 <span class="sc">+</span> </span>
<span id="cb28-179"><a href="#cb28-179" tabindex="-1"></a>        <span class="fu">geom_text_repel</span>(<span class="at">data =</span> gsdata_labels_top,</span>
<span id="cb28-180"><a href="#cb28-180" tabindex="-1"></a>                        <span class="fu">aes</span>(<span class="at">y =</span> max_y, <span class="at">label =</span> gene),</span>
<span id="cb28-181"><a href="#cb28-181" tabindex="-1"></a>                        <span class="at">force_pull   =</span> <span class="dv">0</span>, <span class="co"># do not pull toward data points</span></span>
<span id="cb28-182"><a href="#cb28-182" tabindex="-1"></a>                        <span class="at">nudge_y      =</span> max_y <span class="sc">*</span> <span class="fl">0.05</span>,</span>
<span id="cb28-183"><a href="#cb28-183" tabindex="-1"></a>                        <span class="at">direction    =</span> <span class="st">&quot;x&quot;</span>,</span>
<span id="cb28-184"><a href="#cb28-184" tabindex="-1"></a>                        <span class="at">ylim =</span> <span class="fu">c</span>(<span class="fl">1.4</span>, <span class="dv">2</span>),</span>
<span id="cb28-185"><a href="#cb28-185" tabindex="-1"></a>                        <span class="at">angle        =</span> <span class="dv">90</span>,</span>
<span id="cb28-186"><a href="#cb28-186" tabindex="-1"></a>                        <span class="at">hjust        =</span> <span class="dv">0</span>,</span>
<span id="cb28-187"><a href="#cb28-187" tabindex="-1"></a>                        <span class="at">max.time =</span> <span class="dv">3</span>,</span>
<span id="cb28-188"><a href="#cb28-188" tabindex="-1"></a>                        <span class="at">max.overlaps =</span> <span class="cn">Inf</span>,</span>
<span id="cb28-189"><a href="#cb28-189" tabindex="-1"></a>                        <span class="at">min.segment.length =</span> <span class="dv">0</span>,</span>
<span id="cb28-190"><a href="#cb28-190" tabindex="-1"></a>                        <span class="at">segment.size =</span> <span class="fl">0.2</span>,</span>
<span id="cb28-191"><a href="#cb28-191" tabindex="-1"></a>                        <span class="at">segment.square =</span> <span class="cn">TRUE</span>,</span>
<span id="cb28-192"><a href="#cb28-192" tabindex="-1"></a>                        <span class="at">segment.inflect =</span> <span class="cn">FALSE</span>,</span>
<span id="cb28-193"><a href="#cb28-193" tabindex="-1"></a>                        <span class="at">segment.curvature =</span> <span class="dv">1</span>,</span>
<span id="cb28-194"><a href="#cb28-194" tabindex="-1"></a>                        <span class="at">segment.angle =</span> <span class="sc">-</span><span class="dv">1</span>,</span>
<span id="cb28-195"><a href="#cb28-195" tabindex="-1"></a>                        <span class="at">box.padding =</span> <span class="fl">0.075</span>,</span>
<span id="cb28-196"><a href="#cb28-196" tabindex="-1"></a>                        <span class="at">fontface =</span> <span class="st">&quot;italic&quot;</span></span>
<span id="cb28-197"><a href="#cb28-197" tabindex="-1"></a>        )</span>
<span id="cb28-198"><a href="#cb28-198" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb28-199"><a href="#cb28-199" tabindex="-1"></a>      p2 <span class="ot">&lt;-</span> p2 <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> p2.fill))</span>
<span id="cb28-200"><a href="#cb28-200" tabindex="-1"></a>    }</span>
<span id="cb28-201"><a href="#cb28-201" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">length</span>(geneSetID) <span class="sc">==</span> <span class="dv">1</span> <span class="sc">&amp;</span> p2.add.intervals.colors) {</span>
<span id="cb28-202"><a href="#cb28-202" tabindex="-1"></a>        v <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">sum</span>(gsdata<span class="sc">$</span>position), <span class="at">length.out =</span> <span class="dv">9</span>)</span>
<span id="cb28-203"><a href="#cb28-203" tabindex="-1"></a>        inv <span class="ot">&lt;-</span> <span class="fu">findInterval</span>(<span class="fu">rev</span>(<span class="fu">cumsum</span>(gsdata<span class="sc">$</span>position)), v)</span>
<span id="cb28-204"><a href="#cb28-204" tabindex="-1"></a>        <span class="cf">if</span> (<span class="fu">min</span>(inv) <span class="sc">==</span> <span class="dv">0</span>) </span>
<span id="cb28-205"><a href="#cb28-205" tabindex="-1"></a>            inv <span class="ot">&lt;-</span> inv <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb28-206"><a href="#cb28-206" tabindex="-1"></a>        col <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rev</span>(<span class="fu">brewer.pal</span>(<span class="dv">5</span>, <span class="st">&quot;Blues&quot;</span>)), <span class="fu">brewer.pal</span>(<span class="dv">5</span>, <span class="st">&quot;Reds&quot;</span>))</span>
<span id="cb28-207"><a href="#cb28-207" tabindex="-1"></a>        ymin <span class="ot">&lt;-</span> <span class="fu">min</span>(p2<span class="sc">$</span>data<span class="sc">$</span>ymin)</span>
<span id="cb28-208"><a href="#cb28-208" tabindex="-1"></a>        yy <span class="ot">&lt;-</span> <span class="fu">max</span>(p2<span class="sc">$</span>data<span class="sc">$</span>ymax <span class="sc">-</span> p2<span class="sc">$</span>data<span class="sc">$</span>ymin) <span class="sc">*</span> <span class="fl">0.3</span></span>
<span id="cb28-209"><a href="#cb28-209" tabindex="-1"></a>        xmin <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="sc">!</span><span class="fu">duplicated</span>(inv))</span>
<span id="cb28-210"><a href="#cb28-210" tabindex="-1"></a>        xmax <span class="ot">&lt;-</span> xmin <span class="sc">+</span> <span class="fu">as.numeric</span>(<span class="fu">table</span>(inv)[<span class="fu">as.character</span>(<span class="fu">unique</span>(inv))])</span>
<span id="cb28-211"><a href="#cb28-211" tabindex="-1"></a>        d <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ymin =</span> ymin, <span class="at">ymax =</span> yy, <span class="at">xmin =</span> xmin, </span>
<span id="cb28-212"><a href="#cb28-212" tabindex="-1"></a>            <span class="at">xmax =</span> xmax, <span class="at">col =</span> col[<span class="fu">unique</span>(inv)])</span>
<span id="cb28-213"><a href="#cb28-213" tabindex="-1"></a>        p2 <span class="ot">&lt;-</span> p2 <span class="sc">+</span> <span class="fu">geom_rect</span>(<span class="fu">aes_</span>(<span class="at">xmin =</span> <span class="sc">~</span>xmin, <span class="at">xmax =</span> <span class="sc">~</span>xmax, </span>
<span id="cb28-214"><a href="#cb28-214" tabindex="-1"></a>            <span class="at">ymin =</span> <span class="sc">~</span>ymin, <span class="at">ymax =</span> <span class="sc">~</span>ymax, <span class="at">fill =</span> <span class="sc">~</span><span class="fu">I</span>(col)), <span class="at">data =</span> d, </span>
<span id="cb28-215"><a href="#cb28-215" tabindex="-1"></a>            <span class="at">alpha =</span> <span class="fl">0.9</span>, <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>)</span>
<span id="cb28-216"><a href="#cb28-216" tabindex="-1"></a>    }</span>
<span id="cb28-217"><a href="#cb28-217" tabindex="-1"></a>    </span>
<span id="cb28-218"><a href="#cb28-218" tabindex="-1"></a>    <span class="do">##</span></span>
<span id="cb28-219"><a href="#cb28-219" tabindex="-1"></a>    <span class="do">## Rank plot</span></span>
<span id="cb28-220"><a href="#cb28-220" tabindex="-1"></a>    <span class="do">##</span></span>
<span id="cb28-221"><a href="#cb28-221" tabindex="-1"></a>    df2 <span class="ot">&lt;-</span> p<span class="sc">$</span>data</span>
<span id="cb28-222"><a href="#cb28-222" tabindex="-1"></a>    df2<span class="sc">$</span>y <span class="ot">&lt;-</span> p<span class="sc">$</span>data<span class="sc">$</span>geneList[df2<span class="sc">$</span>x]</span>
<span id="cb28-223"><a href="#cb28-223" tabindex="-1"></a>    p.pos <span class="ot">&lt;-</span> p <span class="sc">+</span> </span>
<span id="cb28-224"><a href="#cb28-224" tabindex="-1"></a>      <span class="fu">geom_segment</span>(<span class="at">data =</span> df2, <span class="fu">aes_</span>(<span class="at">x =</span> <span class="sc">~</span>x, <span class="at">xend =</span> <span class="sc">~</span>x, <span class="at">y =</span> <span class="sc">~</span>y, <span class="at">yend =</span> <span class="dv">0</span>), <span class="at">color =</span> <span class="st">&quot;grey&quot;</span>)</span>
<span id="cb28-225"><a href="#cb28-225" tabindex="-1"></a>    p.pos <span class="ot">&lt;-</span> p.pos <span class="sc">+</span> </span>
<span id="cb28-226"><a href="#cb28-226" tabindex="-1"></a>      <span class="fu">ylab</span>(p.pos.ylab) <span class="sc">+</span> </span>
<span id="cb28-227"><a href="#cb28-227" tabindex="-1"></a>      <span class="fu">xlab</span>(<span class="st">&quot;Rank in Ordered Dataset&quot;</span>) <span class="sc">+</span> </span>
<span id="cb28-228"><a href="#cb28-228" tabindex="-1"></a>      <span class="fu">theme</span>(</span>
<span id="cb28-229"><a href="#cb28-229" tabindex="-1"></a>        <span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="at">t =</span> <span class="sc">-</span><span class="fl">0.1</span>, <span class="at">r =</span> <span class="fl">0.2</span>, <span class="at">b =</span> <span class="fl">0.2</span>, <span class="at">l =</span> <span class="fl">0.2</span>, <span class="at">unit =</span> <span class="st">&quot;cm&quot;</span>),</span>
<span id="cb28-230"><a href="#cb28-230" tabindex="-1"></a>        <span class="at">axis.line.x =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb28-231"><a href="#cb28-231" tabindex="-1"></a>        <span class="at">axis.line.y =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb28-232"><a href="#cb28-232" tabindex="-1"></a>        <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb28-233"><a href="#cb28-233" tabindex="-1"></a>        <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb28-234"><a href="#cb28-234" tabindex="-1"></a>        <span class="at">panel.grid.major.y =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb28-235"><a href="#cb28-235" tabindex="-1"></a>        <span class="at">panel.grid.minor.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb28-236"><a href="#cb28-236" tabindex="-1"></a>        <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span><span class="st">&quot;transparent&quot;</span>,<span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">linewidth =</span> axis.linewidth)</span>
<span id="cb28-237"><a href="#cb28-237" tabindex="-1"></a>        )</span>
<span id="cb28-238"><a href="#cb28-238" tabindex="-1"></a>    </span>
<span id="cb28-239"><a href="#cb28-239" tabindex="-1"></a>    <span class="do">##</span></span>
<span id="cb28-240"><a href="#cb28-240" tabindex="-1"></a>    <span class="do">## Combine plots</span></span>
<span id="cb28-241"><a href="#cb28-241" tabindex="-1"></a>    <span class="do">##</span></span>
<span id="cb28-242"><a href="#cb28-242" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(title) <span class="sc">&amp;&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(title) <span class="sc">&amp;&amp;</span> title <span class="sc">!=</span> <span class="st">&quot;&quot;</span>) </span>
<span id="cb28-243"><a href="#cb28-243" tabindex="-1"></a>        p.res <span class="ot">&lt;-</span> p.res <span class="sc">+</span> <span class="fu">ggtitle</span>(title)</span>
<span id="cb28-244"><a href="#cb28-244" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">length</span>(color) <span class="sc">==</span> <span class="fu">length</span>(geneSetID)) {</span>
<span id="cb28-245"><a href="#cb28-245" tabindex="-1"></a>        p.res <span class="ot">&lt;-</span> p.res <span class="sc">+</span> <span class="fu">scale_color_manual</span>(<span class="at">values =</span> color)</span>
<span id="cb28-246"><a href="#cb28-246" tabindex="-1"></a>        <span class="cf">if</span> (<span class="fu">length</span>(color) <span class="sc">==</span> <span class="dv">1</span>) {</span>
<span id="cb28-247"><a href="#cb28-247" tabindex="-1"></a>            p.res <span class="ot">&lt;-</span> p.res <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb28-248"><a href="#cb28-248" tabindex="-1"></a>            p2 <span class="ot">&lt;-</span> p2 <span class="sc">+</span> <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="st">&quot;black&quot;</span>)</span>
<span id="cb28-249"><a href="#cb28-249" tabindex="-1"></a>        }</span>
<span id="cb28-250"><a href="#cb28-250" tabindex="-1"></a>        <span class="cf">else</span> <span class="cf">if</span> (p2.segment.color.var <span class="sc">==</span> <span class="st">&#39;Description&#39;</span>) {</span>
<span id="cb28-251"><a href="#cb28-251" tabindex="-1"></a>            p2 <span class="ot">&lt;-</span> p2 <span class="sc">+</span> <span class="fu">scale_color_manual</span>(<span class="at">values =</span> color)</span>
<span id="cb28-252"><a href="#cb28-252" tabindex="-1"></a>        }</span>
<span id="cb28-253"><a href="#cb28-253" tabindex="-1"></a>    }</span>
<span id="cb28-254"><a href="#cb28-254" tabindex="-1"></a>    <span class="cf">if</span> (p2.segment.color.var <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&#39;geneList&#39;</span>,<span class="st">&#39;geneListCapped&#39;</span>) ) {</span>
<span id="cb28-255"><a href="#cb28-255" tabindex="-1"></a>      p2 <span class="ot">&lt;-</span> p2 <span class="sc">+</span> <span class="fu">scale_color_gradient2</span>(<span class="at">low =</span> <span class="st">&quot;#377eb8&quot;</span>, <span class="at">mid =</span> <span class="st">&quot;white&quot;</span>, <span class="at">high =</span> <span class="st">&quot;#e41a1c&quot;</span>, <span class="at">midpoint =</span> <span class="dv">0</span>)</span>
<span id="cb28-256"><a href="#cb28-256" tabindex="-1"></a>    }</span>
<span id="cb28-257"><a href="#cb28-257" tabindex="-1"></a>    <span class="cf">if</span> (pvalue_table) {</span>
<span id="cb28-258"><a href="#cb28-258" tabindex="-1"></a>        pd <span class="ot">&lt;-</span> x[geneSetID, <span class="fu">c</span>(<span class="st">&quot;Description&quot;</span>, <span class="st">&quot;pvalue&quot;</span>, <span class="st">&quot;p.adjust&quot;</span>)]</span>
<span id="cb28-259"><a href="#cb28-259" tabindex="-1"></a>        <span class="fu">rownames</span>(pd) <span class="ot">&lt;-</span> pd<span class="sc">$</span>Description</span>
<span id="cb28-260"><a href="#cb28-260" tabindex="-1"></a>        pd <span class="ot">&lt;-</span> pd[, <span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb28-261"><a href="#cb28-261" tabindex="-1"></a>        <span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq_len</span>(<span class="fu">ncol</span>(pd))) {</span>
<span id="cb28-262"><a href="#cb28-262" tabindex="-1"></a>            pd[, i] <span class="ot">&lt;-</span> <span class="fu">format</span>(pd[, i], <span class="at">digits =</span> <span class="dv">4</span>)</span>
<span id="cb28-263"><a href="#cb28-263" tabindex="-1"></a>        }</span>
<span id="cb28-264"><a href="#cb28-264" tabindex="-1"></a>        tp <span class="ot">&lt;-</span> <span class="fu">tableGrob2</span>(pd, p.res)</span>
<span id="cb28-265"><a href="#cb28-265" tabindex="-1"></a>        p.res <span class="ot">&lt;-</span> p.res <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">+</span> <span class="fu">annotation_custom</span>(tp, </span>
<span id="cb28-266"><a href="#cb28-266" tabindex="-1"></a>            <span class="at">xmin =</span> <span class="fu">quantile</span>(p.res<span class="sc">$</span>data<span class="sc">$</span>x, <span class="fl">0.5</span>), <span class="at">xmax =</span> <span class="fu">quantile</span>(p.res<span class="sc">$</span>data<span class="sc">$</span>x, </span>
<span id="cb28-267"><a href="#cb28-267" tabindex="-1"></a>                <span class="fl">0.95</span>), <span class="at">ymin =</span> <span class="fu">quantile</span>(p.res<span class="sc">$</span>data<span class="sc">$</span>runningScore, </span>
<span id="cb28-268"><a href="#cb28-268" tabindex="-1"></a>                <span class="fl">0.75</span>), <span class="at">ymax =</span> <span class="fu">quantile</span>(p.res<span class="sc">$</span>data<span class="sc">$</span>runningScore, </span>
<span id="cb28-269"><a href="#cb28-269" tabindex="-1"></a>                <span class="fl">0.9</span>))</span>
<span id="cb28-270"><a href="#cb28-270" tabindex="-1"></a>    }</span>
<span id="cb28-271"><a href="#cb28-271" tabindex="-1"></a>    plotlist <span class="ot">&lt;-</span> <span class="fu">list</span>(p.res, p2, p.pos)[subplots]</span>
<span id="cb28-272"><a href="#cb28-272" tabindex="-1"></a>    n <span class="ot">&lt;-</span> <span class="fu">length</span>(plotlist)</span>
<span id="cb28-273"><a href="#cb28-273" tabindex="-1"></a>    plotlist[[n]] <span class="ot">&lt;-</span> plotlist[[n]] <span class="sc">+</span> </span>
<span id="cb28-274"><a href="#cb28-274" tabindex="-1"></a>      <span class="fu">theme</span>(</span>
<span id="cb28-275"><a href="#cb28-275" tabindex="-1"></a>        <span class="at">axis.ticks.x =</span> <span class="fu">element_line</span>(),</span>
<span id="cb28-276"><a href="#cb28-276" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>()</span>
<span id="cb28-277"><a href="#cb28-277" tabindex="-1"></a>        )</span>
<span id="cb28-278"><a href="#cb28-278" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">length</span>(subplots) <span class="sc">==</span> <span class="dv">1</span>) </span>
<span id="cb28-279"><a href="#cb28-279" tabindex="-1"></a>        <span class="fu">return</span>(plotlist[[<span class="dv">1</span>]] <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="at">t =</span> <span class="fl">0.2</span>, </span>
<span id="cb28-280"><a href="#cb28-280" tabindex="-1"></a>            <span class="at">r =</span> <span class="fl">0.2</span>, <span class="at">b =</span> <span class="fl">0.2</span>, <span class="at">l =</span> <span class="fl">0.2</span>, <span class="at">unit =</span> <span class="st">&quot;cm&quot;</span>)))</span>
<span id="cb28-281"><a href="#cb28-281" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">length</span>(rel_heights) <span class="sc">&gt;</span> <span class="fu">length</span>(subplots)) </span>
<span id="cb28-282"><a href="#cb28-282" tabindex="-1"></a>        rel_heights <span class="ot">&lt;-</span> rel_heights[subplots]</span>
<span id="cb28-283"><a href="#cb28-283" tabindex="-1"></a>    aplot<span class="sc">::</span><span class="fu">gglist</span>(<span class="at">gglist =</span> plotlist, <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">heights =</span> rel_heights)</span>
<span id="cb28-284"><a href="#cb28-284" tabindex="-1"></a>}</span></code></pre></div>
<div id="supplementary-figure-7l" class="section level3">
<h3>Supplementary Figure 7L</h3>
<p>Gene set enrichment analysis of Amhr2 KO versus control RNA-seq
datasets (as reported in Fig. 7F) interrogating the enrichment of a
mammary cancer proliferative signature, comparing Py2T cells treated as
indicated; NES, normalized enrichment score.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" tabindex="-1"></a>i <span class="ot">&lt;-</span> <span class="st">&quot;InVivo_AMHR2_KO–over–InVivo_Control_sgRNA&quot;</span></span>
<span id="cb29-2"><a href="#cb29-2" tabindex="-1"></a>use_gsets <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;STAT1 target genes and interactors&#39;</span>, <span class="st">&#39;SMAD target genes and interactors&#39;</span>)</span>
<span id="cb29-3"><a href="#cb29-3" tabindex="-1"></a></span>
<span id="cb29-4"><a href="#cb29-4" tabindex="-1"></a>gse_manual_list[[i]]<span class="sc">@</span>result <span class="sc">%&gt;%</span> </span>
<span id="cb29-5"><a href="#cb29-5" tabindex="-1"></a>  <span class="fu">filter</span>(ID <span class="sc">%in%</span> use_gsets) <span class="sc">%&gt;%</span> </span>
<span id="cb29-6"><a href="#cb29-6" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb29-7"><a href="#cb29-7" tabindex="-1"></a>    <span class="at">NES =</span> <span class="fu">round</span>(NES, <span class="dv">2</span>),</span>
<span id="cb29-8"><a href="#cb29-8" tabindex="-1"></a>    <span class="at">pvalue =</span> <span class="fu">format.pval</span>(pvalue, <span class="at">digits =</span> <span class="dv">2</span>),</span>
<span id="cb29-9"><a href="#cb29-9" tabindex="-1"></a>    <span class="at">p.adjust =</span> <span class="fu">format.pval</span>(p.adjust, <span class="at">digits =</span> <span class="dv">2</span>)</span>
<span id="cb29-10"><a href="#cb29-10" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> </span>
<span id="cb29-11"><a href="#cb29-11" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Description, setSize, NES, pvalue, p.adjust) <span class="sc">%&gt;%</span> </span>
<span id="cb29-12"><a href="#cb29-12" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">caption =</span> <span class="st">&#39;GSEA results&#39;</span>,</span>
<span id="cb29-13"><a href="#cb29-13" tabindex="-1"></a>      <span class="at">row.names =</span> F) <span class="sc">%&gt;%</span></span>
<span id="cb29-14"><a href="#cb29-14" tabindex="-1"></a>  <span class="fu">kable_paper</span>(<span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">&quot;striped&quot;</span>, <span class="st">&quot;hover&quot;</span>, <span class="st">&quot;condensed&quot;</span>), </span>
<span id="cb29-15"><a href="#cb29-15" tabindex="-1"></a>              <span class="at">full_width =</span> F) <span class="sc">%&gt;%</span></span>
<span id="cb29-16"><a href="#cb29-16" tabindex="-1"></a>  <span class="fu">column_spec</span>(<span class="dv">1</span>, <span class="at">bold =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb29-17"><a href="#cb29-17" tabindex="-1"></a>  print</span></code></pre></div>
<table class=" lightable-paper" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
GSEA results
</caption>
<thead>
<tr>
<th style="text-align:left;">
Description
</th>
<th style="text-align:right;">
setSize
</th>
<th style="text-align:right;">
NES
</th>
<th style="text-align:left;">
pvalue
</th>
<th style="text-align:left;">
p.adjust
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;font-weight: bold;">
STAT1 target genes and interactors
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
2.23
</td>
<td style="text-align:left;">
3.5e-06
</td>
<td style="text-align:left;">
7.0e-06
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
SMAD target genes and interactors
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
-2.13
</td>
<td style="text-align:left;">
5.5e-06
</td>
<td style="text-align:left;">
7.4e-06
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" tabindex="-1"></a><span class="co"># n_gsets &lt;- seq_along(gse_manual_list[[i]]$Description)</span></span>
<span id="cb31-2"><a href="#cb31-2" tabindex="-1"></a>n_gsets <span class="ot">&lt;-</span> <span class="fu">which</span>(gse_manual_list[[i]]<span class="sc">$</span>Description <span class="sc">%in%</span> use_gsets)</span>
<span id="cb31-3"><a href="#cb31-3" tabindex="-1"></a><span class="cf">for</span>(j <span class="cf">in</span> n_gsets) {</span>
<span id="cb31-4"><a href="#cb31-4" tabindex="-1"></a>  es_color <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(gse_manual_list[[i]]<span class="sc">$</span>NES[j] <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">&quot;#e41a1c&quot;</span>, <span class="st">&quot;#377eb8&quot;</span>)</span>
<span id="cb31-5"><a href="#cb31-5" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;####&quot;</span>, gse_manual_list[[i]]<span class="sc">$</span>Description[j], <span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb31-6"><a href="#cb31-6" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">gseaplot2_custom</span>(</span>
<span id="cb31-7"><a href="#cb31-7" tabindex="-1"></a>    gse_manual_list[[i]], </span>
<span id="cb31-8"><a href="#cb31-8" tabindex="-1"></a>    <span class="at">geneSetID =</span> j, </span>
<span id="cb31-9"><a href="#cb31-9" tabindex="-1"></a>    <span class="at">title =</span> i,</span>
<span id="cb31-10"><a href="#cb31-10" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">8</span>,</span>
<span id="cb31-11"><a href="#cb31-11" tabindex="-1"></a>    <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">1.5</span>, <span class="dv">0</span>),</span>
<span id="cb31-12"><a href="#cb31-12" tabindex="-1"></a>    <span class="at">color =</span>es_color,</span>
<span id="cb31-13"><a href="#cb31-13" tabindex="-1"></a>    <span class="at">p2.fill =</span> <span class="st">&#39;grey90&#39;</span>,</span>
<span id="cb31-14"><a href="#cb31-14" tabindex="-1"></a>    <span class="at">p2.linerange.linewidth =</span> <span class="fl">0.80</span>,</span>
<span id="cb31-15"><a href="#cb31-15" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">&#39;top&#39;</span>,</span>
<span id="cb31-16"><a href="#cb31-16" tabindex="-1"></a>    <span class="at">p.pos.ylab =</span> <span class="st">&#39;Log2 Fold Change&#39;</span>,</span>
<span id="cb31-17"><a href="#cb31-17" tabindex="-1"></a>    <span class="at">p2.labels =</span> <span class="cn">TRUE</span>,</span>
<span id="cb31-18"><a href="#cb31-18" tabindex="-1"></a>    <span class="at">reverse.rank =</span> <span class="cn">FALSE</span>,</span>
<span id="cb31-19"><a href="#cb31-19" tabindex="-1"></a>    <span class="at">p2.segment.color.var =</span> <span class="st">&#39;geneList&#39;</span>,</span>
<span id="cb31-20"><a href="#cb31-20" tabindex="-1"></a>    <span class="at">geneList.limit =</span> <span class="dv">5</span></span>
<span id="cb31-21"><a href="#cb31-21" tabindex="-1"></a>  )</span>
<span id="cb31-22"><a href="#cb31-22" tabindex="-1"></a>  <span class="fu">print</span>(res)</span>
<span id="cb31-23"><a href="#cb31-23" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb31-24"><a href="#cb31-24" tabindex="-1"></a>}</span></code></pre></div>
<div id="stat1-target-genes-and-interactors" class="section level4">
<h4>STAT1 target genes and interactors</h4>
<p><img src="figure/rnaseq-amhr2_ko-deg.Rmd/gseaplot2-custom-labels-individual-plot-sf7L-1.png" style="display: block; margin: auto;" /></p>
</div>
<div id="smad-target-genes-and-interactors" class="section level4">
<h4>SMAD target genes and interactors</h4>
<p><img src="figure/rnaseq-amhr2_ko-deg.Rmd/gseaplot2-custom-labels-individual-plot-sf7L-2.png" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="supplementary-figure-7m-galunisertib-positive"
class="section level3">
<h3>Supplementary Figure 7M (Galunisertib positive)</h3>
<p>Gene set enrichment analysis of Amhr2 KO versus control RNA-seq
datasets (as reported in Fig. 7F) interrogating the enrichment of a
mammary cancer proliferative signature, comparing Py2T cells treated in
vitro with Galunisertib as indicated.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" tabindex="-1"></a>i <span class="ot">&lt;-</span> <span class="st">&quot;AMHR2_KO_noTGFbeta--over--AMHR2_Control_culture&quot;</span></span>
<span id="cb32-2"><a href="#cb32-2" tabindex="-1"></a>use_gsets <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;Mammary epithelium proliferation&#39;</span>)</span>
<span id="cb32-3"><a href="#cb32-3" tabindex="-1"></a></span>
<span id="cb32-4"><a href="#cb32-4" tabindex="-1"></a>gse_manual_list[[i]]<span class="sc">@</span>result <span class="sc">%&gt;%</span> </span>
<span id="cb32-5"><a href="#cb32-5" tabindex="-1"></a>  <span class="fu">filter</span>(ID <span class="sc">%in%</span> use_gsets) <span class="sc">%&gt;%</span> </span>
<span id="cb32-6"><a href="#cb32-6" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb32-7"><a href="#cb32-7" tabindex="-1"></a>    <span class="at">NES =</span> <span class="fu">round</span>(NES, <span class="dv">2</span>),</span>
<span id="cb32-8"><a href="#cb32-8" tabindex="-1"></a>    <span class="at">pvalue =</span> <span class="fu">format.pval</span>(pvalue, <span class="at">digits =</span> <span class="dv">2</span>),</span>
<span id="cb32-9"><a href="#cb32-9" tabindex="-1"></a>    <span class="at">p.adjust =</span> <span class="fu">format.pval</span>(p.adjust, <span class="at">digits =</span> <span class="dv">2</span>)</span>
<span id="cb32-10"><a href="#cb32-10" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> </span>
<span id="cb32-11"><a href="#cb32-11" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Description, setSize, NES, pvalue, p.adjust) <span class="sc">%&gt;%</span> </span>
<span id="cb32-12"><a href="#cb32-12" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">caption =</span> <span class="st">&#39;GSEA results&#39;</span>,</span>
<span id="cb32-13"><a href="#cb32-13" tabindex="-1"></a>      <span class="at">row.names =</span> F) <span class="sc">%&gt;%</span></span>
<span id="cb32-14"><a href="#cb32-14" tabindex="-1"></a>  <span class="fu">kable_paper</span>(<span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">&quot;striped&quot;</span>, <span class="st">&quot;hover&quot;</span>, <span class="st">&quot;condensed&quot;</span>), </span>
<span id="cb32-15"><a href="#cb32-15" tabindex="-1"></a>              <span class="at">full_width =</span> F) <span class="sc">%&gt;%</span></span>
<span id="cb32-16"><a href="#cb32-16" tabindex="-1"></a>  <span class="fu">column_spec</span>(<span class="dv">1</span>, <span class="at">bold =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb32-17"><a href="#cb32-17" tabindex="-1"></a>  print</span></code></pre></div>
<table class=" lightable-paper" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
GSEA results
</caption>
<thead>
<tr>
<th style="text-align:left;">
Description
</th>
<th style="text-align:right;">
setSize
</th>
<th style="text-align:right;">
NES
</th>
<th style="text-align:left;">
pvalue
</th>
<th style="text-align:left;">
p.adjust
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;font-weight: bold;">
Mammary epithelium proliferation
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
-0.65
</td>
<td style="text-align:left;">
0.9
</td>
<td style="text-align:left;">
0.99
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" tabindex="-1"></a>n_gsets <span class="ot">&lt;-</span> <span class="fu">which</span>(gse_manual_list[[i]]<span class="sc">$</span>Description <span class="sc">%in%</span> use_gsets)</span>
<span id="cb34-2"><a href="#cb34-2" tabindex="-1"></a><span class="cf">for</span>(j <span class="cf">in</span> n_gsets) {</span>
<span id="cb34-3"><a href="#cb34-3" tabindex="-1"></a>  es_color <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(gse_manual_list[[i]]<span class="sc">$</span>NES[j] <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">&quot;#e41a1c&quot;</span>, <span class="st">&quot;#377eb8&quot;</span>)</span>
<span id="cb34-4"><a href="#cb34-4" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;####&quot;</span>, gse_manual_list[[i]]<span class="sc">$</span>Description[j], <span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb34-5"><a href="#cb34-5" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">gseaplot2_custom</span>(</span>
<span id="cb34-6"><a href="#cb34-6" tabindex="-1"></a>    gse_manual_list[[i]], </span>
<span id="cb34-7"><a href="#cb34-7" tabindex="-1"></a>    <span class="at">geneSetID =</span> j, </span>
<span id="cb34-8"><a href="#cb34-8" tabindex="-1"></a>    <span class="at">title =</span> i,</span>
<span id="cb34-9"><a href="#cb34-9" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">8</span>,</span>
<span id="cb34-10"><a href="#cb34-10" tabindex="-1"></a>    <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">1.5</span>, <span class="dv">0</span>),</span>
<span id="cb34-11"><a href="#cb34-11" tabindex="-1"></a>    <span class="at">color =</span>es_color,</span>
<span id="cb34-12"><a href="#cb34-12" tabindex="-1"></a>    <span class="at">p2.fill =</span> <span class="st">&#39;grey90&#39;</span>,</span>
<span id="cb34-13"><a href="#cb34-13" tabindex="-1"></a>    <span class="at">p2.linerange.linewidth =</span> <span class="fl">0.80</span>,</span>
<span id="cb34-14"><a href="#cb34-14" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">&#39;top&#39;</span>,</span>
<span id="cb34-15"><a href="#cb34-15" tabindex="-1"></a>    <span class="at">p.pos.ylab =</span> <span class="st">&#39;Log2 Fold Change&#39;</span>,</span>
<span id="cb34-16"><a href="#cb34-16" tabindex="-1"></a>    <span class="at">p2.labels =</span> <span class="cn">TRUE</span>,</span>
<span id="cb34-17"><a href="#cb34-17" tabindex="-1"></a>    <span class="at">reverse.rank =</span> <span class="cn">FALSE</span>,</span>
<span id="cb34-18"><a href="#cb34-18" tabindex="-1"></a>    <span class="at">p2.segment.color.var =</span> <span class="st">&#39;geneList&#39;</span>,</span>
<span id="cb34-19"><a href="#cb34-19" tabindex="-1"></a>    <span class="at">geneList.limit =</span> <span class="dv">5</span></span>
<span id="cb34-20"><a href="#cb34-20" tabindex="-1"></a>  )</span>
<span id="cb34-21"><a href="#cb34-21" tabindex="-1"></a>  <span class="fu">print</span>(res)</span>
<span id="cb34-22"><a href="#cb34-22" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb34-23"><a href="#cb34-23" tabindex="-1"></a>}</span></code></pre></div>
<div id="mammary-epithelium-proliferation" class="section level4">
<h4>Mammary epithelium proliferation</h4>
<p><img src="figure/rnaseq-amhr2_ko-deg.Rmd/gseaplot2-custom-labels-individual-plot-sf7Mtop-1.png" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="supplementary-figure-7p" class="section level3">
<h3>Supplementary Figure 7P</h3>
<p>Gene set enrichment analysis of Amhr2 KO versus control RNA-seq
datasets (as reported in Fig. 7F) interrogating the enrichment of a
mammary cancer proliferative signature or the enrichment of an EMT
signature including epithelial and mesenchymal genes, showing data from
orthotopic Py2T tumors in FVB/N mice.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" tabindex="-1"></a>i <span class="ot">&lt;-</span> <span class="st">&quot;InVivo_AMHR2_KO + AMHR2_KO_culture--over--InVivo_Control_sgRNA + AMHR2_Control_culture&quot;</span></span>
<span id="cb35-2"><a href="#cb35-2" tabindex="-1"></a>use_gsets <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;Mammary epithelium proliferation&#39;</span>, <span class="st">&#39;EMT regulators and targets&#39;</span>)</span>
<span id="cb35-3"><a href="#cb35-3" tabindex="-1"></a></span>
<span id="cb35-4"><a href="#cb35-4" tabindex="-1"></a>gse_manual_list[[i]]<span class="sc">@</span>result <span class="sc">%&gt;%</span> </span>
<span id="cb35-5"><a href="#cb35-5" tabindex="-1"></a>  <span class="fu">filter</span>(ID <span class="sc">%in%</span> use_gsets) <span class="sc">%&gt;%</span> </span>
<span id="cb35-6"><a href="#cb35-6" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb35-7"><a href="#cb35-7" tabindex="-1"></a>    <span class="at">NES =</span> <span class="fu">round</span>(NES, <span class="dv">2</span>),</span>
<span id="cb35-8"><a href="#cb35-8" tabindex="-1"></a>    <span class="at">pvalue =</span> <span class="fu">format.pval</span>(pvalue, <span class="at">digits =</span> <span class="dv">2</span>),</span>
<span id="cb35-9"><a href="#cb35-9" tabindex="-1"></a>    <span class="at">p.adjust =</span> <span class="fu">format.pval</span>(p.adjust, <span class="at">digits =</span> <span class="dv">2</span>)</span>
<span id="cb35-10"><a href="#cb35-10" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> </span>
<span id="cb35-11"><a href="#cb35-11" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Description, setSize, NES, pvalue, p.adjust) <span class="sc">%&gt;%</span> </span>
<span id="cb35-12"><a href="#cb35-12" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">caption =</span> <span class="st">&#39;GSEA results&#39;</span>,</span>
<span id="cb35-13"><a href="#cb35-13" tabindex="-1"></a>      <span class="at">row.names =</span> F) <span class="sc">%&gt;%</span></span>
<span id="cb35-14"><a href="#cb35-14" tabindex="-1"></a>  <span class="fu">kable_paper</span>(<span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">&quot;striped&quot;</span>, <span class="st">&quot;hover&quot;</span>, <span class="st">&quot;condensed&quot;</span>), </span>
<span id="cb35-15"><a href="#cb35-15" tabindex="-1"></a>              <span class="at">full_width =</span> F) <span class="sc">%&gt;%</span></span>
<span id="cb35-16"><a href="#cb35-16" tabindex="-1"></a>  <span class="fu">column_spec</span>(<span class="dv">1</span>, <span class="at">bold =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb35-17"><a href="#cb35-17" tabindex="-1"></a>  print</span></code></pre></div>
<table class=" lightable-paper" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
GSEA results
</caption>
<thead>
<tr>
<th style="text-align:left;">
Description
</th>
<th style="text-align:right;">
setSize
</th>
<th style="text-align:right;">
NES
</th>
<th style="text-align:left;">
pvalue
</th>
<th style="text-align:left;">
p.adjust
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;font-weight: bold;">
EMT regulators and targets
</td>
<td style="text-align:right;">
24
</td>
<td style="text-align:right;">
-2.38
</td>
<td style="text-align:left;">
2.1e-08
</td>
<td style="text-align:left;">
8.6e-08
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Mammary epithelium proliferation
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:right;">
-1.35
</td>
<td style="text-align:left;">
0.12
</td>
<td style="text-align:left;">
0.12
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" tabindex="-1"></a>n_gsets <span class="ot">&lt;-</span> <span class="fu">which</span>(gse_manual_list[[i]]<span class="sc">$</span>Description <span class="sc">%in%</span> use_gsets)</span>
<span id="cb37-2"><a href="#cb37-2" tabindex="-1"></a><span class="cf">for</span>(j <span class="cf">in</span> n_gsets) {</span>
<span id="cb37-3"><a href="#cb37-3" tabindex="-1"></a>  es_color <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(gse_manual_list[[i]]<span class="sc">$</span>NES[j] <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">&quot;#e41a1c&quot;</span>, <span class="st">&quot;#377eb8&quot;</span>)</span>
<span id="cb37-4"><a href="#cb37-4" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;####&quot;</span>, gse_manual_list[[i]]<span class="sc">$</span>Description[j], <span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb37-5"><a href="#cb37-5" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">gseaplot2_custom</span>(</span>
<span id="cb37-6"><a href="#cb37-6" tabindex="-1"></a>    gse_manual_list[[i]], </span>
<span id="cb37-7"><a href="#cb37-7" tabindex="-1"></a>    <span class="at">geneSetID =</span> j, </span>
<span id="cb37-8"><a href="#cb37-8" tabindex="-1"></a>    <span class="at">title =</span> i,</span>
<span id="cb37-9"><a href="#cb37-9" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">8</span>,</span>
<span id="cb37-10"><a href="#cb37-10" tabindex="-1"></a>    <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">1.5</span>, <span class="dv">0</span>),</span>
<span id="cb37-11"><a href="#cb37-11" tabindex="-1"></a>    <span class="at">color =</span>es_color,</span>
<span id="cb37-12"><a href="#cb37-12" tabindex="-1"></a>    <span class="at">p2.fill =</span> <span class="st">&#39;grey90&#39;</span>,</span>
<span id="cb37-13"><a href="#cb37-13" tabindex="-1"></a>    <span class="at">p2.linerange.linewidth =</span> <span class="fl">0.80</span>,</span>
<span id="cb37-14"><a href="#cb37-14" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">&#39;top&#39;</span>,</span>
<span id="cb37-15"><a href="#cb37-15" tabindex="-1"></a>    <span class="at">p.pos.ylab =</span> <span class="st">&#39;Log2 Fold Change&#39;</span>,</span>
<span id="cb37-16"><a href="#cb37-16" tabindex="-1"></a>    <span class="at">p2.labels =</span> <span class="cn">TRUE</span>,</span>
<span id="cb37-17"><a href="#cb37-17" tabindex="-1"></a>    <span class="at">reverse.rank =</span> <span class="cn">FALSE</span>,</span>
<span id="cb37-18"><a href="#cb37-18" tabindex="-1"></a>    <span class="at">p2.segment.color.var =</span> <span class="st">&#39;geneList&#39;</span>,</span>
<span id="cb37-19"><a href="#cb37-19" tabindex="-1"></a>    <span class="at">geneList.limit =</span> <span class="dv">5</span></span>
<span id="cb37-20"><a href="#cb37-20" tabindex="-1"></a>  )</span>
<span id="cb37-21"><a href="#cb37-21" tabindex="-1"></a>  <span class="fu">print</span>(res)</span>
<span id="cb37-22"><a href="#cb37-22" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb37-23"><a href="#cb37-23" tabindex="-1"></a>}</span></code></pre></div>
<div id="emt-regulators-and-targets" class="section level4">
<h4>EMT regulators and targets</h4>
<p><img src="figure/rnaseq-amhr2_ko-deg.Rmd/gseaplot2-custom-labels-individual-plot-sf7p-1.png" style="display: block; margin: auto;" /></p>
</div>
<div id="mammary-epithelium-proliferation-1" class="section level4">
<h4>Mammary epithelium proliferation</h4>
<p><img src="figure/rnaseq-amhr2_ko-deg.Rmd/gseaplot2-custom-labels-individual-plot-sf7p-2.png" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="supplementary-figure-7u-left-galunisertib-negative"
class="section level3">
<h3>Supplementary Figure 7U left (Galunisertib negative)</h3>
<p>Gene set enrichment analysis of Amhr2 KO versus control RNA-seq
datasets (as reported in Fig. 7F) interrogating the enrichment of an
IFN-α transcriptional target signature, comparing Py2T cells treated in
vitro with Galunisertib as indicated.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" tabindex="-1"></a>i <span class="ot">&lt;-</span> <span class="st">&quot;InVivo_AMHR2_KO + AMHR2_KO_culture--over--InVivo_Control_sgRNA + AMHR2_Control_culture&quot;</span></span>
<span id="cb38-2"><a href="#cb38-2" tabindex="-1"></a>use_gsets <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;STAT1 target genes and interactors&#39;</span>)</span>
<span id="cb38-3"><a href="#cb38-3" tabindex="-1"></a></span>
<span id="cb38-4"><a href="#cb38-4" tabindex="-1"></a>gse_manual_list[[i]]<span class="sc">@</span>result <span class="sc">%&gt;%</span> </span>
<span id="cb38-5"><a href="#cb38-5" tabindex="-1"></a>  <span class="fu">filter</span>(ID <span class="sc">%in%</span> use_gsets) <span class="sc">%&gt;%</span> </span>
<span id="cb38-6"><a href="#cb38-6" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb38-7"><a href="#cb38-7" tabindex="-1"></a>    <span class="at">NES =</span> <span class="fu">round</span>(NES, <span class="dv">2</span>),</span>
<span id="cb38-8"><a href="#cb38-8" tabindex="-1"></a>    <span class="at">pvalue =</span> <span class="fu">format.pval</span>(pvalue, <span class="at">digits =</span> <span class="dv">2</span>),</span>
<span id="cb38-9"><a href="#cb38-9" tabindex="-1"></a>    <span class="at">p.adjust =</span> <span class="fu">format.pval</span>(p.adjust, <span class="at">digits =</span> <span class="dv">2</span>)</span>
<span id="cb38-10"><a href="#cb38-10" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> </span>
<span id="cb38-11"><a href="#cb38-11" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Description, setSize, NES, pvalue, p.adjust) <span class="sc">%&gt;%</span> </span>
<span id="cb38-12"><a href="#cb38-12" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">caption =</span> <span class="st">&#39;GSEA results&#39;</span>,</span>
<span id="cb38-13"><a href="#cb38-13" tabindex="-1"></a>      <span class="at">row.names =</span> F) <span class="sc">%&gt;%</span></span>
<span id="cb38-14"><a href="#cb38-14" tabindex="-1"></a>  <span class="fu">kable_paper</span>(<span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">&quot;striped&quot;</span>, <span class="st">&quot;hover&quot;</span>, <span class="st">&quot;condensed&quot;</span>), </span>
<span id="cb38-15"><a href="#cb38-15" tabindex="-1"></a>              <span class="at">full_width =</span> F) <span class="sc">%&gt;%</span></span>
<span id="cb38-16"><a href="#cb38-16" tabindex="-1"></a>  <span class="fu">column_spec</span>(<span class="dv">1</span>, <span class="at">bold =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb38-17"><a href="#cb38-17" tabindex="-1"></a>  print</span></code></pre></div>
<table class=" lightable-paper" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
GSEA results
</caption>
<thead>
<tr>
<th style="text-align:left;">
Description
</th>
<th style="text-align:right;">
setSize
</th>
<th style="text-align:right;">
NES
</th>
<th style="text-align:left;">
pvalue
</th>
<th style="text-align:left;">
p.adjust
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;font-weight: bold;">
STAT1 target genes and interactors
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
1.97
</td>
<td style="text-align:left;">
0.00014
</td>
<td style="text-align:left;">
0.00027
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" tabindex="-1"></a>n_gsets <span class="ot">&lt;-</span> <span class="fu">which</span>(gse_manual_list[[i]]<span class="sc">$</span>Description <span class="sc">%in%</span> use_gsets)</span>
<span id="cb40-2"><a href="#cb40-2" tabindex="-1"></a><span class="cf">for</span>(j <span class="cf">in</span> n_gsets) {</span>
<span id="cb40-3"><a href="#cb40-3" tabindex="-1"></a>  es_color <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(gse_manual_list[[i]]<span class="sc">$</span>NES[j] <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">&quot;#e41a1c&quot;</span>, <span class="st">&quot;#377eb8&quot;</span>)</span>
<span id="cb40-4"><a href="#cb40-4" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;####&quot;</span>, gse_manual_list[[i]]<span class="sc">$</span>Description[j], <span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb40-5"><a href="#cb40-5" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">gseaplot2_custom</span>(</span>
<span id="cb40-6"><a href="#cb40-6" tabindex="-1"></a>    gse_manual_list[[i]], </span>
<span id="cb40-7"><a href="#cb40-7" tabindex="-1"></a>    <span class="at">geneSetID =</span> j, </span>
<span id="cb40-8"><a href="#cb40-8" tabindex="-1"></a>    <span class="at">title =</span> i,</span>
<span id="cb40-9"><a href="#cb40-9" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">8</span>,</span>
<span id="cb40-10"><a href="#cb40-10" tabindex="-1"></a>    <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">1.5</span>, <span class="dv">0</span>),</span>
<span id="cb40-11"><a href="#cb40-11" tabindex="-1"></a>    <span class="at">color =</span>es_color,</span>
<span id="cb40-12"><a href="#cb40-12" tabindex="-1"></a>    <span class="at">p2.fill =</span> <span class="st">&#39;grey90&#39;</span>,</span>
<span id="cb40-13"><a href="#cb40-13" tabindex="-1"></a>    <span class="at">p2.linerange.linewidth =</span> <span class="fl">0.80</span>,</span>
<span id="cb40-14"><a href="#cb40-14" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">&#39;top&#39;</span>,</span>
<span id="cb40-15"><a href="#cb40-15" tabindex="-1"></a>    <span class="at">p.pos.ylab =</span> <span class="st">&#39;Log2 Fold Change&#39;</span>,</span>
<span id="cb40-16"><a href="#cb40-16" tabindex="-1"></a>    <span class="at">p2.labels =</span> <span class="cn">TRUE</span>,</span>
<span id="cb40-17"><a href="#cb40-17" tabindex="-1"></a>    <span class="at">reverse.rank =</span> <span class="cn">FALSE</span>,</span>
<span id="cb40-18"><a href="#cb40-18" tabindex="-1"></a>    <span class="at">p2.segment.color.var =</span> <span class="st">&#39;geneList&#39;</span>,</span>
<span id="cb40-19"><a href="#cb40-19" tabindex="-1"></a>    <span class="at">geneList.limit =</span> <span class="dv">5</span></span>
<span id="cb40-20"><a href="#cb40-20" tabindex="-1"></a>  )</span>
<span id="cb40-21"><a href="#cb40-21" tabindex="-1"></a>  <span class="fu">print</span>(res)</span>
<span id="cb40-22"><a href="#cb40-22" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb40-23"><a href="#cb40-23" tabindex="-1"></a>}</span></code></pre></div>
<div id="stat1-target-genes-and-interactors-1" class="section level4">
<h4>STAT1 target genes and interactors</h4>
<p><img src="figure/rnaseq-amhr2_ko-deg.Rmd/gseaplot2-custom-labels-individual-plot-sf7Uleft-1.png" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="supplementary-figure-7u-right-galunisertib-positive"
class="section level3">
<h3>Supplementary Figure 7U right (Galunisertib positive)</h3>
<p>Gene set enrichment analysis of Amhr2 KO versus control RNA-seq
datasets (as reported in Fig. 7F) interrogating the enrichment of an
IFN-α transcriptional target signature, comparing Py2T cells treated in
vitro with Galunisertib as indicated.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" tabindex="-1"></a>i <span class="ot">&lt;-</span> <span class="st">&quot;AMHR2_KO_noTGFbeta--over--AMHR2_Control_culture&quot;</span></span>
<span id="cb41-2"><a href="#cb41-2" tabindex="-1"></a>use_gsets <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;STAT1 target genes and interactors&#39;</span>)</span>
<span id="cb41-3"><a href="#cb41-3" tabindex="-1"></a></span>
<span id="cb41-4"><a href="#cb41-4" tabindex="-1"></a>gse_manual_list[[i]]<span class="sc">@</span>result <span class="sc">%&gt;%</span> </span>
<span id="cb41-5"><a href="#cb41-5" tabindex="-1"></a>  <span class="fu">filter</span>(ID <span class="sc">%in%</span> use_gsets) <span class="sc">%&gt;%</span> </span>
<span id="cb41-6"><a href="#cb41-6" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb41-7"><a href="#cb41-7" tabindex="-1"></a>    <span class="at">NES =</span> <span class="fu">round</span>(NES, <span class="dv">2</span>),</span>
<span id="cb41-8"><a href="#cb41-8" tabindex="-1"></a>    <span class="at">pvalue =</span> <span class="fu">format.pval</span>(pvalue, <span class="at">digits =</span> <span class="dv">2</span>),</span>
<span id="cb41-9"><a href="#cb41-9" tabindex="-1"></a>    <span class="at">p.adjust =</span> <span class="fu">format.pval</span>(p.adjust, <span class="at">digits =</span> <span class="dv">2</span>)</span>
<span id="cb41-10"><a href="#cb41-10" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> </span>
<span id="cb41-11"><a href="#cb41-11" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Description, setSize, NES, pvalue, p.adjust) <span class="sc">%&gt;%</span> </span>
<span id="cb41-12"><a href="#cb41-12" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">caption =</span> <span class="st">&#39;GSEA results&#39;</span>,</span>
<span id="cb41-13"><a href="#cb41-13" tabindex="-1"></a>      <span class="at">row.names =</span> F) <span class="sc">%&gt;%</span></span>
<span id="cb41-14"><a href="#cb41-14" tabindex="-1"></a>  <span class="fu">kable_paper</span>(<span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">&quot;striped&quot;</span>, <span class="st">&quot;hover&quot;</span>, <span class="st">&quot;condensed&quot;</span>), </span>
<span id="cb41-15"><a href="#cb41-15" tabindex="-1"></a>              <span class="at">full_width =</span> F) <span class="sc">%&gt;%</span></span>
<span id="cb41-16"><a href="#cb41-16" tabindex="-1"></a>  <span class="fu">column_spec</span>(<span class="dv">1</span>, <span class="at">bold =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb41-17"><a href="#cb41-17" tabindex="-1"></a>  print</span></code></pre></div>
<table class=" lightable-paper" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
GSEA results
</caption>
<thead>
<tr>
<th style="text-align:left;">
Description
</th>
<th style="text-align:right;">
setSize
</th>
<th style="text-align:right;">
NES
</th>
<th style="text-align:left;">
pvalue
</th>
<th style="text-align:left;">
p.adjust
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;font-weight: bold;">
STAT1 target genes and interactors
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
1.4
</td>
<td style="text-align:left;">
0.079
</td>
<td style="text-align:left;">
0.32
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" tabindex="-1"></a>n_gsets <span class="ot">&lt;-</span> <span class="fu">which</span>(gse_manual_list[[i]]<span class="sc">$</span>Description <span class="sc">%in%</span> use_gsets)</span>
<span id="cb43-2"><a href="#cb43-2" tabindex="-1"></a><span class="cf">for</span>(j <span class="cf">in</span> n_gsets) {</span>
<span id="cb43-3"><a href="#cb43-3" tabindex="-1"></a>  es_color <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(gse_manual_list[[i]]<span class="sc">$</span>NES[j] <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">&quot;#e41a1c&quot;</span>, <span class="st">&quot;#377eb8&quot;</span>)</span>
<span id="cb43-4"><a href="#cb43-4" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;####&quot;</span>, gse_manual_list[[i]]<span class="sc">$</span>Description[j], <span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb43-5"><a href="#cb43-5" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">gseaplot2_custom</span>(</span>
<span id="cb43-6"><a href="#cb43-6" tabindex="-1"></a>    gse_manual_list[[i]], </span>
<span id="cb43-7"><a href="#cb43-7" tabindex="-1"></a>    <span class="at">geneSetID =</span> j, </span>
<span id="cb43-8"><a href="#cb43-8" tabindex="-1"></a>    <span class="at">title =</span> i,</span>
<span id="cb43-9"><a href="#cb43-9" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">8</span>,</span>
<span id="cb43-10"><a href="#cb43-10" tabindex="-1"></a>    <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">1.5</span>, <span class="dv">0</span>),</span>
<span id="cb43-11"><a href="#cb43-11" tabindex="-1"></a>    <span class="at">color =</span>es_color,</span>
<span id="cb43-12"><a href="#cb43-12" tabindex="-1"></a>    <span class="at">p2.fill =</span> <span class="st">&#39;grey90&#39;</span>,</span>
<span id="cb43-13"><a href="#cb43-13" tabindex="-1"></a>    <span class="at">p2.linerange.linewidth =</span> <span class="fl">0.80</span>,</span>
<span id="cb43-14"><a href="#cb43-14" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">&#39;top&#39;</span>,</span>
<span id="cb43-15"><a href="#cb43-15" tabindex="-1"></a>    <span class="at">p.pos.ylab =</span> <span class="st">&#39;Log2 Fold Change&#39;</span>,</span>
<span id="cb43-16"><a href="#cb43-16" tabindex="-1"></a>    <span class="at">p2.labels =</span> <span class="cn">TRUE</span>,</span>
<span id="cb43-17"><a href="#cb43-17" tabindex="-1"></a>    <span class="at">reverse.rank =</span> <span class="cn">FALSE</span>,</span>
<span id="cb43-18"><a href="#cb43-18" tabindex="-1"></a>    <span class="at">p2.segment.color.var =</span> <span class="st">&#39;geneList&#39;</span>,</span>
<span id="cb43-19"><a href="#cb43-19" tabindex="-1"></a>    <span class="at">geneList.limit =</span> <span class="dv">5</span></span>
<span id="cb43-20"><a href="#cb43-20" tabindex="-1"></a>  )</span>
<span id="cb43-21"><a href="#cb43-21" tabindex="-1"></a>  <span class="fu">print</span>(res)</span>
<span id="cb43-22"><a href="#cb43-22" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb43-23"><a href="#cb43-23" tabindex="-1"></a>}</span></code></pre></div>
<div id="stat1-target-genes-and-interactors-2" class="section level4">
<h4>STAT1 target genes and interactors</h4>
<p><img src="figure/rnaseq-amhr2_ko-deg.Rmd/gseaplot2-custom-labels-individual-plot-sf7Uright-1.png" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="all-gsea-plots" class="section level2">
<h2>All GSEA plots</h2>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" tabindex="-1"></a>i <span class="ot">&lt;-</span> <span class="fu">names</span>(gse_manual_list)[<span class="dv">1</span>]</span>
<span id="cb44-2"><a href="#cb44-2" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">names</span>(gse_manual_list)) {</span>
<span id="cb44-3"><a href="#cb44-3" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;### Comparison&quot;</span>, i, <span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb44-4"><a href="#cb44-4" tabindex="-1"></a>  </span>
<span id="cb44-5"><a href="#cb44-5" tabindex="-1"></a>  gse_manual_list[[i]]<span class="sc">@</span>result <span class="sc">%&gt;%</span> </span>
<span id="cb44-6"><a href="#cb44-6" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb44-7"><a href="#cb44-7" tabindex="-1"></a>      <span class="at">NES =</span> <span class="fu">round</span>(NES, <span class="dv">2</span>),</span>
<span id="cb44-8"><a href="#cb44-8" tabindex="-1"></a>      <span class="at">pvalue =</span> <span class="fu">format.pval</span>(pvalue, <span class="at">digits =</span> <span class="dv">2</span>),</span>
<span id="cb44-9"><a href="#cb44-9" tabindex="-1"></a>      <span class="at">p.adjust =</span> <span class="fu">format.pval</span>(p.adjust, <span class="at">digits =</span> <span class="dv">2</span>)</span>
<span id="cb44-10"><a href="#cb44-10" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb44-11"><a href="#cb44-11" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(Description, setSize, NES, pvalue, p.adjust) <span class="sc">%&gt;%</span> </span>
<span id="cb44-12"><a href="#cb44-12" tabindex="-1"></a>    <span class="fu">kbl</span>(<span class="at">caption =</span> <span class="st">&#39;GSEA results&#39;</span>,</span>
<span id="cb44-13"><a href="#cb44-13" tabindex="-1"></a>        <span class="at">row.names =</span> F) <span class="sc">%&gt;%</span></span>
<span id="cb44-14"><a href="#cb44-14" tabindex="-1"></a>    <span class="fu">kable_paper</span>(<span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">&quot;striped&quot;</span>, <span class="st">&quot;hover&quot;</span>, <span class="st">&quot;condensed&quot;</span>), </span>
<span id="cb44-15"><a href="#cb44-15" tabindex="-1"></a>                <span class="at">full_width =</span> F) <span class="sc">%&gt;%</span></span>
<span id="cb44-16"><a href="#cb44-16" tabindex="-1"></a>    <span class="fu">column_spec</span>(<span class="dv">1</span>, <span class="at">bold =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb44-17"><a href="#cb44-17" tabindex="-1"></a>    print</span>
<span id="cb44-18"><a href="#cb44-18" tabindex="-1"></a>  </span>
<span id="cb44-19"><a href="#cb44-19" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb44-20"><a href="#cb44-20" tabindex="-1"></a>  </span>
<span id="cb44-21"><a href="#cb44-21" tabindex="-1"></a>  n_gsets <span class="ot">&lt;-</span> <span class="fu">seq_along</span>(gse_manual_list[[i]]<span class="sc">$</span>Description)</span>
<span id="cb44-22"><a href="#cb44-22" tabindex="-1"></a>  <span class="cf">for</span>(j <span class="cf">in</span> n_gsets) {</span>
<span id="cb44-23"><a href="#cb44-23" tabindex="-1"></a>    es_color <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(gse_manual_list[[i]]<span class="sc">$</span>NES[j] <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">&quot;#e41a1c&quot;</span>, <span class="st">&quot;#377eb8&quot;</span>)</span>
<span id="cb44-24"><a href="#cb44-24" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">&quot;####&quot;</span>, gse_manual_list[[i]]<span class="sc">$</span>Description[j], <span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb44-25"><a href="#cb44-25" tabindex="-1"></a>    res <span class="ot">&lt;-</span> <span class="fu">gseaplot2_custom</span>(</span>
<span id="cb44-26"><a href="#cb44-26" tabindex="-1"></a>      gse_manual_list[[i]], </span>
<span id="cb44-27"><a href="#cb44-27" tabindex="-1"></a>      <span class="at">geneSetID =</span> j, </span>
<span id="cb44-28"><a href="#cb44-28" tabindex="-1"></a>      <span class="at">title =</span> i,</span>
<span id="cb44-29"><a href="#cb44-29" tabindex="-1"></a>      <span class="at">base_size =</span> <span class="dv">8</span>,</span>
<span id="cb44-30"><a href="#cb44-30" tabindex="-1"></a>      <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">1.5</span>, <span class="dv">0</span>),</span>
<span id="cb44-31"><a href="#cb44-31" tabindex="-1"></a>      <span class="at">color =</span>es_color,</span>
<span id="cb44-32"><a href="#cb44-32" tabindex="-1"></a>      <span class="at">p2.fill =</span> <span class="st">&#39;grey90&#39;</span>,</span>
<span id="cb44-33"><a href="#cb44-33" tabindex="-1"></a>      <span class="at">p2.linerange.linewidth =</span> <span class="fl">0.80</span>,</span>
<span id="cb44-34"><a href="#cb44-34" tabindex="-1"></a>      <span class="at">legend.position =</span> <span class="st">&#39;top&#39;</span>,</span>
<span id="cb44-35"><a href="#cb44-35" tabindex="-1"></a>      <span class="at">p.pos.ylab =</span> <span class="st">&#39;Log2 Fold Change&#39;</span>,</span>
<span id="cb44-36"><a href="#cb44-36" tabindex="-1"></a>      <span class="at">p2.labels =</span> <span class="cn">TRUE</span>,</span>
<span id="cb44-37"><a href="#cb44-37" tabindex="-1"></a>      <span class="at">reverse.rank =</span> <span class="cn">FALSE</span>,</span>
<span id="cb44-38"><a href="#cb44-38" tabindex="-1"></a>      <span class="at">p2.segment.color.var =</span> <span class="st">&#39;geneList&#39;</span>,</span>
<span id="cb44-39"><a href="#cb44-39" tabindex="-1"></a>      <span class="at">geneList.limit =</span> <span class="dv">5</span></span>
<span id="cb44-40"><a href="#cb44-40" tabindex="-1"></a>    )</span>
<span id="cb44-41"><a href="#cb44-41" tabindex="-1"></a>    <span class="fu">print</span>(res)</span>
<span id="cb44-42"><a href="#cb44-42" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb44-43"><a href="#cb44-43" tabindex="-1"></a>  }</span>
<span id="cb44-44"><a href="#cb44-44" tabindex="-1"></a>}</span></code></pre></div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<p>R version 4.4.3 (2025-02-28) Platform: aarch64-apple-darwin20 Running
under: macOS 26.0.1</p>
<p>Matrix products: default BLAS:
/Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib
LAPACK:
/Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;
LAPACK version 3.12.0</p>
<p>locale: [1]
en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</p>
<p>time zone: Europe/Zurich tzcode source: internal</p>
<p>attached base packages: [1] grid stats4 stats graphics grDevices
utils datasets [8] methods base</p>
<p>other attached packages: [1] circlize_0.4.16
ComplexHeatmap_2.22.0<br />
[3] org.Mm.eg.db_3.20.0 AnnotationDbi_1.68.0<br />
[5] PCAtools_2.18.0 pheatmap_1.0.13<br />
[7] scran_1.34.0 scuttle_1.16.0<br />
[9] SingleCellExperiment_1.28.1 DESeq2_1.46.0<br />
[11] edgeR_4.4.2 limma_3.62.2<br />
[13] SummarizedExperiment_1.36.0 Biobase_2.66.0<br />
[15] GenomicRanges_1.58.0 GenomeInfoDb_1.42.3<br />
[17] IRanges_2.40.1 S4Vectors_0.44.0<br />
[19] BiocGenerics_0.52.0 MatrixGenerics_1.18.1<br />
[21] matrixStats_1.5.0 openxlsx_4.2.8<br />
[23] magrittr_2.0.3 ggrepel_0.9.6<br />
[25] ggbeeswarm_0.7.2 RColorBrewer_1.1-3<br />
[27] colorblindr_0.1.0 colorspace_2.1-1<br />
[29] cowplot_1.2.0 kableExtra_1.4.0<br />
[31] knitr_1.50 DT_0.34.0<br />
[33] foreach_1.5.2 showtext_0.9-7<br />
[35] showtextdb_3.0 sysfonts_0.8.9<br />
[37] lubridate_1.9.4 forcats_1.0.0<br />
[39] stringr_1.5.1 dplyr_1.1.4<br />
[41] purrr_1.1.0 readr_2.1.5<br />
[43] tidyr_1.3.1 tibble_3.3.0<br />
[45] ggplot2_3.5.2 tidyverse_2.0.0<br />
[47] workflowr_1.7.1</p>
<p>loaded via a namespace (and not attached): [1] splines_4.4.3
later_1.4.4<br />
[3] ggplotify_0.1.2 R.oo_1.27.1<br />
[5] lifecycle_1.0.4 doParallel_1.0.17<br />
[7] rprojroot_2.1.0 processx_3.8.6<br />
[9] lattice_0.22-7 crosstalk_1.2.2<br />
[11] sass_0.4.10 rmarkdown_2.29<br />
[13] jquerylib_0.1.4 yaml_2.3.10<br />
[15] ggtangle_0.0.7 metapod_1.14.0<br />
[17] httpuv_1.6.16 zip_2.3.3<br />
[19] DBI_1.2.3 abind_1.4-8<br />
[21] zlibbioc_1.52.0 R.utils_2.13.0<br />
[23] yulab.utils_0.2.0 git2r_0.36.2<br />
[25] GenomeInfoDbData_1.2.13 enrichplot_1.26.6<br />
[27] irlba_2.3.5.1 tidytree_0.4.6<br />
[29] dqrng_0.4.1 svglite_2.2.1<br />
[31] DelayedMatrixStats_1.28.1 codetools_0.2-20<br />
[33] DelayedArray_0.32.0 DOSE_4.0.1<br />
[35] xml2_1.3.8 tidyselect_1.2.1<br />
[37] shape_1.4.6.1 aplot_0.2.8<br />
[39] UCSC.utils_1.2.0 farver_2.1.2<br />
[41] ScaledMatrix_1.14.0 jsonlite_2.0.0<br />
[43] GetoptLong_1.0.5 BiocNeighbors_2.0.1<br />
[45] iterators_1.0.14 systemfonts_1.2.3<br />
[47] tools_4.4.3 treeio_1.30.0<br />
[49] Rcpp_1.1.0 glue_1.8.0<br />
[51] SparseArray_1.6.2 xfun_0.53<br />
[53] here_1.0.1 qvalue_2.38.0<br />
[55] withr_3.0.2 fastmap_1.2.0<br />
[57] bluster_1.16.0 callr_3.7.6<br />
[59] digest_0.6.37 rsvd_1.0.5<br />
[61] gridGraphics_0.5-1 timechange_0.3.0<br />
[63] R6_2.6.1 textshaping_1.0.1<br />
[65] GO.db_3.20.0 dichromat_2.0-0.1<br />
[67] RSQLite_2.4.3 R.methodsS3_1.8.2<br />
[69] generics_0.1.4 data.table_1.17.8<br />
[71] httr_1.4.7 htmlwidgets_1.6.4<br />
[73] S4Arrays_1.6.0 whisker_0.4.1<br />
[75] pkgconfig_2.0.3 gtable_0.3.6<br />
[77] blob_1.2.4 XVector_0.46.0<br />
[79] htmltools_0.5.8.1 fgsea_1.32.4<br />
[81] clue_0.3-66 scales_1.4.0<br />
[83] png_0.1-8 ggfun_0.2.0<br />
[85] rstudioapi_0.17.1 tzdb_0.5.0<br />
[87] reshape2_1.4.4 rjson_0.2.23<br />
[89] nlme_3.1-168 cachem_1.1.0<br />
[91] GlobalOptions_0.1.2 parallel_4.4.3<br />
[93] vipor_0.4.7 pillar_1.11.0<br />
[95] vctrs_0.6.5 promises_1.3.3<br />
[97] BiocSingular_1.22.0 beachmat_2.22.0<br />
[99] cluster_2.1.8.1 beeswarm_0.4.0<br />
[101] evaluate_1.0.5 cli_3.6.5<br />
[103] locfit_1.5-9.12 compiler_4.4.3<br />
[105] rlang_1.1.6 crayon_1.5.3<br />
[107] labeling_0.4.3 ps_1.9.1<br />
[109] getPass_0.2-4 plyr_1.8.9<br />
[111] fs_1.6.6 stringi_1.8.7<br />
[113] viridisLite_0.4.2 BiocParallel_1.40.2<br />
[115] Biostrings_2.74.1 lazyeval_0.2.2<br />
[117] GOSemSim_2.32.0 Matrix_1.7-3<br />
[119] patchwork_1.3.1 hms_1.1.3<br />
[121] sparseMatrixStats_1.18.0 bit64_4.6.0-1<br />
[123] KEGGREST_1.46.0 statmod_1.5.0<br />
[125] igraph_2.1.4 memoise_2.0.1<br />
[127] bslib_0.9.0 ggtree_3.14.0<br />
[129] fastmatch_1.1-6 bit_4.6.0<br />
[131] ape_5.8-1</p>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
