<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Francesc Castro-Giner" />

<meta name="date" content="2025-10-31" />

<title>Analysis of RNA-seq bulk data</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">StealTHY</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="about.html">
    <span class="fa fa-info"></span>
     
    About
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-bar-chart"></span>
     
    Analyses
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Data generation</li>
    <li>
      <a href="0_differential_expression_gsea_gsva.html">Run differential gene expression analysis, GSEA and GSVA</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Main results</li>
    <li>
      <a href="patients_ctc_counts_distribution.html">CTC counts distribution in breast cancer patients</a>
    </li>
    <li>
      <a href="br16_pca.html">PCA analysis of CTC gene expression in NSG-CDX-BR16 mice</a>
    </li>
    <li>
      <a href="br16_dge.html">Differential gene expression report for NSG-CDX-BR16 data</a>
    </li>
    <li>
      <a href="gsea_across_models.html">Report of GSEA and GSVA analysis in NSG-CDX-BR16, NSG-LM2, Patient and NSG-LM2 time kinetics</a>
    </li>
    <li>
      <a href="core_gene_sets.html">Analysis of genes from candidate pathways</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="https://github.com/TheAcetoLab/saini-stealTHY">
    <span class="fa fa-github"></span>
     
  </a>
</li>
<li>
  <a href="https://www.theacetolab.com">
    <span class="fa fa-flask"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Analysis of RNA-seq bulk data</h1>
<h4 class="author">Francesc Castro-Giner</h4>
<h4 class="date">October 31, 2025</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-10-31
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>saini-stealTHY/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240517code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240517)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240517code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240517)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomTheAcetoLabsainistealTHYtree455fa9121fcca51363cf639f7ee170fb108e34c1targetblank455fa91a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/tree/455fa9121fcca51363cf639f7ee170fb108e34c1" target="_blank">455fa91</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomTheAcetoLabsainistealTHYtree455fa9121fcca51363cf639f7ee170fb108e34c1targetblank455fa91a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/TheAcetoLab/saini-stealTHY/tree/455fa9121fcca51363cf639f7ee170fb108e34c1" target="_blank">455fa91</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/templates/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    code/raw_data_processing/rnaseq/p27851_o32062/pipelines/
    Ignored:    configuration/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/crispr/
    Ignored:    data/resources/
    Ignored:    data/rnaseq/
    Ignored:    output/.DS_Store
    Ignored:    output/clinical/
    Ignored:    output/crispr/
    Ignored:    output/rnaseq/

Untracked files:
    Untracked:  analysis/crispr-mm_2215_sgRNA-clonality.md
    Untracked:  analysis/index.md
    Untracked:  analysis/style.css
    Untracked:  code/R-functions/subchunkify.R

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/_site.yml
    Deleted:    analysis/crispr-hsapiens_2180_sgRNA_r1.Rmd
    Deleted:    analysis/crispr-mm_2215_sgRNA-clonality_r1.Rmd
    Deleted:    analysis/crispr-mm_2215_sgRNA-r2oC.Rmd
    Deleted:    analysis/crispr-mm_2215_sgRNA-r2oD.Rmd
    Deleted:    analysis/crispr-mm_2215_sgRNA-r2oD_r1.Rmd
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_ko_culture--over--amhr2_control_culture_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_ko_culture--over--amhr2_control_culture_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_ko_notgfbeta--over--amhr2_ko_culture_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_ko_notgfbeta--over--amhr2_ko_culture_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_overexpression--over--amhr2_control_culture_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_overexpression--over--amhr2_control_culture_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_overexpression--over--amhr2_ko_culture_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_overexpression--over--amhr2_ko_culture_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_overexpression--over--amhr2_ko_notgfbeta_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-amhr2_overexpression--over--amhr2_ko_notgfbeta_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/figure/gse-dotplot-invivo_amhr2_ko-+-amhr2_ko_culture--over--invivo_control_sgrna-+-amhr2_control_culture_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-invivo_amhr2_ko-+-amhr2_ko_culture--over--invivo_control_sgrna-+-amhr2_control_culture_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/figure/gse-dotplot-invivo_amhr2_ko--over--invivo_control_sgrna_GSEA_msigdb.h-1.pdf
    Deleted:    analysis/templates/docs/figure/gse-dotplot-invivo_amhr2_ko--over--invivo_control_sgrna_GSEA_msigdb.h-1.png
    Deleted:    analysis/templates/docs/file/tmp/dge.xlsx
    Deleted:    analysis/templates/docs/file/tmp/gse_gsea_msigdb.h.xlsx
    Deleted:    analysis/templates/docs/file/tmp/gse_gsego_bp.xlsx
    Deleted:    analysis/templates/docs/file/tmp/gse_gsego_mf.xlsx
    Modified:   analysis/templates/rnaseq-deg-edger.Rmd
    Modified:   code/R-functions/gse_report.r
    Modified:   configuration/rmarkdown/color_palettes.R
    Modified:   configuration/rmarkdown/ggplot_theme.R
    Modified:   update_workflowr.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/rnaseq-tumor-bulk.Rmd</code>) and
HTML (<code>docs/rnaseq-tumor-bulk.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/455fa9121fcca51363cf639f7ee170fb108e34c1/analysis/rnaseq-tumor-bulk.Rmd" target="_blank">455fa91</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-10-31
</td>
<td>
Prepare files for publication
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheAcetoLab/saini-stealTHY/8632845b1c6b024330836a1a798537363e31980e/docs/rnaseq-tumor-bulk.html" target="_blank">8632845</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-07-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/bc402e6695932824d0587de6c03eb192be28dfca/analysis/rnaseq-tumor-bulk.Rmd" target="_blank">bc402e6</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-07-01
</td>
<td>
uodate rnaseq analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/628f8ec8600fd6671cb7c7f57c8e717199ff7f75/analysis/rnaseq-tumor-bulk.Rmd" target="_blank">628f8ec</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2024-05-24
</td>
<td>
added F4
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/abf695c6a645a3aa42ea9e2fb622b5f22e723b11/analysis/rnaseq-tumor-bulk.Rmd" target="_blank">abf695c</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2024-05-23
</td>
<td>
done f2 anf f3
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries-additional-functions-and-data"
class="section level2">
<h2>Load libraries, additional functions and data</h2>
<p>Setup environment</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">results=</span><span class="st">&#39;asis&#39;</span>, <span class="at">echo=</span><span class="cn">TRUE</span>, <span class="at">message=</span><span class="cn">FALSE</span>, <span class="at">warning=</span><span class="cn">FALSE</span>, <span class="at">error=</span><span class="cn">FALSE</span>, <span class="at">fig.align =</span> <span class="st">&#39;center&#39;</span>, <span class="at">fig.width =</span> <span class="fl">3.5</span>, <span class="at">fig.asp =</span> <span class="fl">0.618</span>, <span class="at">dpi =</span> <span class="dv">600</span>, <span class="at">dev =</span> <span class="fu">c</span>(<span class="st">&quot;png&quot;</span>, <span class="st">&quot;pdf&quot;</span>), <span class="at">fig.showtext =</span> <span class="cn">FALSE</span>, <span class="at">engine.opts =</span> <span class="fu">list</span>(<span class="at">bash =</span> <span class="st">&quot;-l&quot;</span>))</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">options</span>(<span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>use_seed <span class="ot">&lt;-</span> <span class="dv">1100101</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="fu">set.seed</span>(use_seed)</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="fu">dir.create</span>(params<span class="sc">$</span>output_dir, <span class="at">recursive =</span> <span class="cn">TRUE</span>, <span class="at">showWarnings =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p>Load packages</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="fu">library</span>(foreach)</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="fu">library</span>(magrittr)</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="fu">library</span>(DT)</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a><span class="fu">library</span>(edgeR)</span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a><span class="fu">library</span>(scuttle)</span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a><span class="fu">library</span>(scran)</span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a><span class="fu">library</span>(ComplexHeatmap)</span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a><span class="fu">library</span>(circlize)</span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a><span class="fu">library</span>(org.Mm.eg.db)</span></code></pre></div>
<p>Load ggplot theme</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;./configuration/rmarkdown/ggplot_theme.R&quot;</span>)</span></code></pre></div>
<p>Load ggplot theme</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;./configuration/rmarkdown/color_palettes.R&quot;</span>)</span></code></pre></div>
<p>Load custom functions</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&#39;./code/R-functions/dge_wrappers.r&#39;</span>)</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&#39;./code/R-functions/dge_report.r&#39;</span>)</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&#39;./code/R-functions/gse_omnibus.r&#39;</span>)</span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&#39;./code/R-functions/gse_report.r&#39;</span>)</span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>clean_msigdb_names <span class="ot">&lt;-</span> <span class="cf">function</span>(x) x <span class="sc">%&gt;%</span> <span class="fu">gsub</span>(<span class="st">&#39;REACTOME_&#39;</span>, <span class="st">&#39;&#39;</span>, .) <span class="sc">%&gt;%</span> <span class="fu">gsub</span>(<span class="st">&#39;WP_&#39;</span>, <span class="st">&#39;&#39;</span>, .) <span class="sc">%&gt;%</span> <span class="fu">gsub</span>(<span class="st">&#39;BIOCARTA_&#39;</span>, <span class="st">&#39;&#39;</span>, .) <span class="sc">%&gt;%</span> <span class="fu">gsub</span>(<span class="st">&#39;KEGG_&#39;</span>, <span class="st">&#39;&#39;</span>, .) <span class="sc">%&gt;%</span> <span class="fu">gsub</span>(<span class="st">&#39;PID_&#39;</span>, <span class="st">&#39;&#39;</span>, .) <span class="sc">%&gt;%</span> <span class="fu">gsub</span>(<span class="st">&#39;GOBP_&#39;</span>, <span class="st">&#39;&#39;</span>, .) <span class="sc">%&gt;%</span> <span class="fu">gsub</span>(<span class="st">&#39;_&#39;</span>, <span class="st">&#39; &#39;</span>, .)</span></code></pre></div>
<p>Load MSigDB files</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>gmt_files_symbols <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>  <span class="at">msigdb.m2.cp.m5.bp.custom.signatures =</span> <span class="st">&#39;./data/resources/custom_signatures/msigdb_m2_cp-m5_bp-signatures_custom.gmt&#39;</span></span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>)</span></code></pre></div>
<p>Clean files generated in previous runs</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>rmd_file <span class="ot">&lt;-</span> <span class="fu">current_input</span>()</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(rmd_file)) {</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>  figures_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">&#39;./docs/figure&#39;</span>,rmd_file)</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">dir.exists</span>(figures_dir)) {</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>    <span class="fu">unlink</span>(<span class="fu">file.path</span>(figures_dir, <span class="st">&quot;*&quot;</span>))</span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>  }</span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>}</span></code></pre></div>
<p>Load Ensembl biomaRt mmusculus data</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">file.exists</span>(<span class="st">&#39;./data/resources/ensembl_biomart/mart_mmu_genes.rds&#39;</span>)){</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>  ensembl_mart_mmu_genes <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&#39;./data/resources/ensembl_biomart/mart_mmu_genes.rds&#39;</span>)</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>  <span class="co"># httr::set_config(httr::config(ssl_verifypeer = FALSE))</span></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>  ensembl_mart_mmu_genes <span class="ot">&lt;-</span>  biomaRt<span class="sc">::</span><span class="fu">useEnsembl</span>(</span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a>    <span class="at">biomart=</span><span class="st">&quot;genes&quot;</span>, <span class="at">dataset=</span><span class="st">&quot;mmusculus_gene_ensembl&quot;</span>,</span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a>    <span class="at">mirror =</span> <span class="st">&quot;useast&quot;</span></span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a>    )</span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a>  <span class="fu">saveRDS</span>(</span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a>    ensembl_mart_mmu_genes, </span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a>    <span class="at">file =</span> <span class="st">&#39;./data/resources/ensembl_biomart/mart_mmu_genes.rds&#39;</span></span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a>    )</span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="data-wrangling" class="section level2">
<h2>Data wrangling</h2>
<p>Load data</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">file.path</span>(params<span class="sc">$</span>data_dir, params<span class="sc">$</span>se_file))</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a><span class="fu">rowData</span>(se[<span class="fu">rownames</span>(se) <span class="sc">==</span> <span class="st">&#39;CAS9&#39;</span>,])<span class="sc">$</span>gene_name <span class="ot">&lt;-</span> <span class="st">&#39;SpCas9&#39;</span></span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a><span class="fu">rownames</span>(se)[<span class="fu">rownames</span>(se) <span class="sc">==</span> <span class="st">&#39;CAS9&#39;</span>] <span class="ot">&lt;-</span> <span class="st">&#39;SpCas9&#39;</span></span></code></pre></div>
<p>Modify sample annotation</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="fu">colData</span>(se) <span class="sc">%&lt;&gt;%</span> data.frame <span class="sc">%&gt;%</span> </span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>    <span class="at">transduction =</span> <span class="fu">gsub</span>(<span class="st">&#39;Thy.&#39;</span>, <span class="st">&#39;Thy1&#39;</span>, transduction),</span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a>    <span class="at">condition =</span> <span class="fu">gsub</span>(<span class="st">&#39; Mammary fat pad tumor| 4T1| MVT1| Py2T&#39;</span>, <span class="st">&#39;&#39;</span>, sample_title),</span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a>    <span class="at">condition =</span> <span class="fu">gsub</span>(<span class="st">&#39; *$&#39;</span>, <span class="st">&#39;&#39;</span>, condition)</span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> </span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a>  DataFrame</span></code></pre></div>
<p>Filtering out low-abundance features. Keeping those features with at
least 1 count in at least 30% of samples of the smallest group.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>use_rows <span class="ot">&lt;-</span> <span class="fu">filterByExpr</span>(se, <span class="at">group=</span><span class="fu">paste</span>(se<span class="sc">$</span>donor, se<span class="sc">$</span>condition), <span class="at">min.count =</span> params<span class="sc">$</span>min_counts, <span class="at">min.prop =</span> params<span class="sc">$</span>min_present_prop)</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>se <span class="ot">&lt;-</span> se[use_rows,]</span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a></span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a><span class="co"># Update PerCellQCMetrics</span></span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a><span class="fu">colData</span>(se) <span class="ot">&lt;-</span> <span class="fu">colData</span>(se) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>(sum<span class="sc">:</span>subsets_Ribo_percent)) <span class="sc">%&gt;%</span> DataFrame</span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">addPerCellQCMetrics</span>(</span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a>  se,</span>
<span id="cb11-8"><a href="#cb11-8" tabindex="-1"></a>  <span class="at">subsets=</span><span class="fu">list</span>(</span>
<span id="cb11-9"><a href="#cb11-9" tabindex="-1"></a>    <span class="at">Mito=</span><span class="fu">which</span>(<span class="fu">rowData</span>(se)<span class="sc">$</span>is.mito),</span>
<span id="cb11-10"><a href="#cb11-10" tabindex="-1"></a>    <span class="at">Ribo=</span><span class="fu">which</span>(<span class="fu">rowData</span>(se)<span class="sc">$</span>is.ribo)</span>
<span id="cb11-11"><a href="#cb11-11" tabindex="-1"></a>  ),</span>
<span id="cb11-12"><a href="#cb11-12" tabindex="-1"></a>  <span class="at">percent.top =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">5</span>,<span class="dv">10</span>,<span class="dv">20</span>, <span class="dv">50</span>, <span class="dv">100</span>),</span>
<span id="cb11-13"><a href="#cb11-13" tabindex="-1"></a>  <span class="at">threshold =</span> <span class="dv">3</span></span>
<span id="cb11-14"><a href="#cb11-14" tabindex="-1"></a>  )</span></code></pre></div>
<p>TMM normalization</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a>norm_method <span class="ot">&lt;-</span> <span class="st">&quot;TMM&quot;</span></span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a>exprs_mat_for_norm <span class="ot">&lt;-</span> <span class="fu">assay</span>(se, <span class="st">&#39;counts&#39;</span>)</span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a>norm_factors <span class="ot">&lt;-</span> <span class="fu">calcNormFactors</span>(exprs_mat_for_norm, <span class="at">method =</span> norm_method)</span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a>lib_size <span class="ot">&lt;-</span> <span class="fu">colSums</span>(exprs_mat_for_norm)<span class="sc">/</span><span class="dv">10</span><span class="sc">^</span><span class="dv">6</span></span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a>se<span class="sc">$</span>size_factor <span class="ot">&lt;-</span> (norm_factors <span class="sc">*</span> lib_size)</span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">logNormCounts</span>(se, <span class="at">size.factors =</span>se<span class="sc">$</span>size_factor)</span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">logNormCounts</span>(se, <span class="at">size.factors =</span>se<span class="sc">$</span>size_factor, <span class="at">log =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<div id="data-generation" class="section level2">
<h2>Data generation</h2>
<p>Configure comparisons and arguments for differential expression</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">colData</span>(se) <span class="sc">%&gt;%</span> data.frame</span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>comp_list <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a>   <span class="st">`</span><span class="at">dCas9_Thy1--over--Thy1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a>    <span class="st">`</span><span class="at">dCas9-Thy1</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb13-5"><a href="#cb13-5" tabindex="-1"></a>      <span class="fu">filter</span>(condition <span class="sc">==</span> <span class="st">&#39;Rejected Cas9-BROAD&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-6"><a href="#cb13-6" tabindex="-1"></a>      <span class="fu">pull</span>(sample_alias) <span class="sc">%&gt;%</span> </span>
<span id="cb13-7"><a href="#cb13-7" tabindex="-1"></a>      sort,</span>
<span id="cb13-8"><a href="#cb13-8" tabindex="-1"></a>    <span class="st">`</span><span class="at">Thy1</span><span class="st">`</span> <span class="ot">=</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb13-9"><a href="#cb13-9" tabindex="-1"></a>      <span class="fu">filter</span>(transduction <span class="sc">==</span> <span class="st">&#39;Thy1-transduced&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-10"><a href="#cb13-10" tabindex="-1"></a>      <span class="fu">pull</span>(sample_alias) <span class="sc">%&gt;%</span> </span>
<span id="cb13-11"><a href="#cb13-11" tabindex="-1"></a>      sort,</span>
<span id="cb13-12"><a href="#cb13-12" tabindex="-1"></a>    <span class="at">batch =</span> <span class="st">&#39;donor&#39;</span></span>
<span id="cb13-13"><a href="#cb13-13" tabindex="-1"></a>  )</span>
<span id="cb13-14"><a href="#cb13-14" tabindex="-1"></a>)</span></code></pre></div>
<div id="differential-expression-analysis" class="section level3">
<h3>Differential expression analysis</h3>
<p>Run differential expression analysis using EdgeR QLF for
multidonor-comparisons</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a>use_comp <span class="ot">&lt;-</span>  comp_list[[<span class="dv">1</span>]]</span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a>dge_list <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">use_comp =</span> comp_list) <span class="sc">%do%</span>{</span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a>  se_cols <span class="ot">&lt;-</span> se<span class="sc">$</span>sample_alias <span class="sc">%in%</span> <span class="fu">unlist</span>(use_comp)</span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a>  use_se <span class="ot">&lt;-</span> se[,se_cols]</span>
<span id="cb14-5"><a href="#cb14-5" tabindex="-1"></a>  use_se<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(</span>
<span id="cb14-6"><a href="#cb14-6" tabindex="-1"></a>    use_se<span class="sc">$</span>sample_alias <span class="sc">%in%</span> use_comp[[<span class="dv">1</span>]],</span>
<span id="cb14-7"><a href="#cb14-7" tabindex="-1"></a>    <span class="fu">names</span>(use_comp)[<span class="dv">1</span>],</span>
<span id="cb14-8"><a href="#cb14-8" tabindex="-1"></a>    <span class="fu">names</span>(use_comp)[<span class="dv">2</span>]</span>
<span id="cb14-9"><a href="#cb14-9" tabindex="-1"></a>  )</span>
<span id="cb14-10"><a href="#cb14-10" tabindex="-1"></a>  </span>
<span id="cb14-11"><a href="#cb14-11" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">is.null</span>(use_comp<span class="sc">$</span>batch)) {</span>
<span id="cb14-12"><a href="#cb14-12" tabindex="-1"></a>    dge <span class="ot">&lt;-</span> <span class="fu">edgeR_dge</span>(</span>
<span id="cb14-13"><a href="#cb14-13" tabindex="-1"></a>      use_se,</span>
<span id="cb14-14"><a href="#cb14-14" tabindex="-1"></a>      <span class="co"># Design configuration for differential expression</span></span>
<span id="cb14-15"><a href="#cb14-15" tabindex="-1"></a>      <span class="at">group_var =</span>  <span class="st">&#39;group&#39;</span>,</span>
<span id="cb14-16"><a href="#cb14-16" tabindex="-1"></a>      <span class="at">group_sample =</span> <span class="fu">names</span>(use_comp)[<span class="dv">1</span>],</span>
<span id="cb14-17"><a href="#cb14-17" tabindex="-1"></a>      <span class="at">group_ref =</span> <span class="fu">names</span>(use_comp)[<span class="dv">2</span>],</span>
<span id="cb14-18"><a href="#cb14-18" tabindex="-1"></a>      <span class="at">design_formula =</span> <span class="st">&quot;~ group&quot;</span>,</span>
<span id="cb14-19"><a href="#cb14-19" tabindex="-1"></a>      <span class="at">coef =</span> <span class="st">&#39;last&#39;</span>,</span>
<span id="cb14-20"><a href="#cb14-20" tabindex="-1"></a>      <span class="co"># Conversion from SingleCellExperiment/Summarized Experiment to DGEList</span></span>
<span id="cb14-21"><a href="#cb14-21" tabindex="-1"></a>      <span class="at">assay_to_DGEList =</span> <span class="st">&#39;counts&#39;</span>,</span>
<span id="cb14-22"><a href="#cb14-22" tabindex="-1"></a>      <span class="at">assay_to_row_filter =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb14-23"><a href="#cb14-23" tabindex="-1"></a>      <span class="co"># Feature filtering parameters</span></span>
<span id="cb14-24"><a href="#cb14-24" tabindex="-1"></a>      <span class="at">use_filterByExpr =</span> <span class="cn">TRUE</span>,</span>
<span id="cb14-25"><a href="#cb14-25" tabindex="-1"></a>      <span class="at">min_counts =</span> params<span class="sc">$</span>min_counts,</span>
<span id="cb14-26"><a href="#cb14-26" tabindex="-1"></a>      <span class="at">min_present_prop =</span> params<span class="sc">$</span>min_present_prop,</span>
<span id="cb14-27"><a href="#cb14-27" tabindex="-1"></a>      <span class="co"># EdgeR workflow configuration</span></span>
<span id="cb14-28"><a href="#cb14-28" tabindex="-1"></a>      <span class="at">run_calcNormFactors =</span> <span class="st">&#39;TMM&#39;</span>,</span>
<span id="cb14-29"><a href="#cb14-29" tabindex="-1"></a>      <span class="at">estimateDisp_robust =</span> <span class="cn">TRUE</span>,</span>
<span id="cb14-30"><a href="#cb14-30" tabindex="-1"></a>      <span class="at">estimateDisp_trend.method =</span> <span class="st">&quot;locfit&quot;</span>,</span>
<span id="cb14-31"><a href="#cb14-31" tabindex="-1"></a>      <span class="at">glm_approach =</span> <span class="st">&quot;QLF&quot;</span>,</span>
<span id="cb14-32"><a href="#cb14-32" tabindex="-1"></a>      <span class="co"># Output configuration</span></span>
<span id="cb14-33"><a href="#cb14-33" tabindex="-1"></a>      <span class="at">adjust_method =</span> <span class="st">&#39;BH&#39;</span></span>
<span id="cb14-34"><a href="#cb14-34" tabindex="-1"></a>      )</span>
<span id="cb14-35"><a href="#cb14-35" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span>(use_comp<span class="sc">$</span>batch <span class="sc">==</span> <span class="st">&#39;donor&#39;</span>) {</span>
<span id="cb14-36"><a href="#cb14-36" tabindex="-1"></a>      dge <span class="ot">&lt;-</span> <span class="fu">edgeR_dge</span>(</span>
<span id="cb14-37"><a href="#cb14-37" tabindex="-1"></a>        use_se,</span>
<span id="cb14-38"><a href="#cb14-38" tabindex="-1"></a>        <span class="co"># Design configuration for differential expression</span></span>
<span id="cb14-39"><a href="#cb14-39" tabindex="-1"></a>        <span class="at">group_var =</span>  <span class="st">&#39;group&#39;</span>,</span>
<span id="cb14-40"><a href="#cb14-40" tabindex="-1"></a>        <span class="at">group_sample =</span> <span class="fu">names</span>(use_comp)[<span class="dv">1</span>],</span>
<span id="cb14-41"><a href="#cb14-41" tabindex="-1"></a>        <span class="at">group_ref =</span> <span class="fu">names</span>(use_comp)[<span class="dv">2</span>],</span>
<span id="cb14-42"><a href="#cb14-42" tabindex="-1"></a>        <span class="at">batch_vars =</span> <span class="st">&#39;donor&#39;</span>,</span>
<span id="cb14-43"><a href="#cb14-43" tabindex="-1"></a>        <span class="at">design_formula =</span> <span class="st">&quot;~ donor + group&quot;</span>,</span>
<span id="cb14-44"><a href="#cb14-44" tabindex="-1"></a>        <span class="at">coef =</span> <span class="st">&#39;last&#39;</span>,</span>
<span id="cb14-45"><a href="#cb14-45" tabindex="-1"></a>        <span class="co"># Conversion from SingleCellExperiment/Summarized Experiment to DGEList</span></span>
<span id="cb14-46"><a href="#cb14-46" tabindex="-1"></a>        <span class="at">assay_to_DGEList =</span> <span class="st">&#39;counts&#39;</span>,</span>
<span id="cb14-47"><a href="#cb14-47" tabindex="-1"></a>        <span class="at">assay_to_row_filter =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb14-48"><a href="#cb14-48" tabindex="-1"></a>        <span class="co"># Feature filtering parameters</span></span>
<span id="cb14-49"><a href="#cb14-49" tabindex="-1"></a>        <span class="at">use_filterByExpr =</span> <span class="cn">TRUE</span>,</span>
<span id="cb14-50"><a href="#cb14-50" tabindex="-1"></a>        <span class="at">min_counts =</span> params<span class="sc">$</span>min_counts,</span>
<span id="cb14-51"><a href="#cb14-51" tabindex="-1"></a>        <span class="at">min_present_prop =</span> params<span class="sc">$</span>min_present_prop,</span>
<span id="cb14-52"><a href="#cb14-52" tabindex="-1"></a>        <span class="co"># EdgeR workflow configuration</span></span>
<span id="cb14-53"><a href="#cb14-53" tabindex="-1"></a>        <span class="at">run_calcNormFactors =</span> <span class="st">&#39;TMM&#39;</span>,</span>
<span id="cb14-54"><a href="#cb14-54" tabindex="-1"></a>        <span class="at">estimateDisp_robust =</span> <span class="cn">TRUE</span>,</span>
<span id="cb14-55"><a href="#cb14-55" tabindex="-1"></a>        <span class="at">estimateDisp_trend.method =</span> <span class="st">&quot;locfit&quot;</span>,</span>
<span id="cb14-56"><a href="#cb14-56" tabindex="-1"></a>        <span class="at">glm_approach =</span> <span class="st">&quot;QLF&quot;</span>,</span>
<span id="cb14-57"><a href="#cb14-57" tabindex="-1"></a>        <span class="co"># Output configuration</span></span>
<span id="cb14-58"><a href="#cb14-58" tabindex="-1"></a>        <span class="at">adjust_method =</span> <span class="st">&#39;BH&#39;</span></span>
<span id="cb14-59"><a href="#cb14-59" tabindex="-1"></a>        )</span>
<span id="cb14-60"><a href="#cb14-60" tabindex="-1"></a>    }</span>
<span id="cb14-61"><a href="#cb14-61" tabindex="-1"></a>  </span>
<span id="cb14-62"><a href="#cb14-62" tabindex="-1"></a>  </span>
<span id="cb14-63"><a href="#cb14-63" tabindex="-1"></a>  <span class="co"># Add gene description</span></span>
<span id="cb14-64"><a href="#cb14-64" tabindex="-1"></a>  gene_desc <span class="ot">&lt;-</span> biomaRt<span class="sc">::</span><span class="fu">getBM</span>(<span class="at">attributes=</span><span class="fu">c</span>(<span class="st">&#39;external_gene_name&#39;</span>,<span class="st">&#39;description&#39;</span>),</span>
<span id="cb14-65"><a href="#cb14-65" tabindex="-1"></a>                              <span class="at">filters =</span> <span class="st">&#39;external_gene_name&#39;</span>,</span>
<span id="cb14-66"><a href="#cb14-66" tabindex="-1"></a>                              <span class="at">values =</span> dge<span class="sc">$</span>results<span class="sc">$</span>gene_name, <span class="at">mart =</span>ensembl_mart_mmu_genes) <span class="sc">%&gt;%</span></span>
<span id="cb14-67"><a href="#cb14-67" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&#39;gene_name&#39;</span> <span class="ot">=</span> <span class="st">&#39;external_gene_name&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb14-68"><a href="#cb14-68" tabindex="-1"></a>    unique</span>
<span id="cb14-69"><a href="#cb14-69" tabindex="-1"></a>  </span>
<span id="cb14-70"><a href="#cb14-70" tabindex="-1"></a>  </span>
<span id="cb14-71"><a href="#cb14-71" tabindex="-1"></a>  use_res <span class="ot">&lt;-</span> dge<span class="sc">$</span>results <span class="sc">%&gt;%</span>  <span class="fu">left_join</span>(., gene_desc)</span>
<span id="cb14-72"><a href="#cb14-72" tabindex="-1"></a>  dge<span class="sc">$</span>results <span class="ot">&lt;-</span> use_res <span class="sc">%&gt;%</span> </span>
<span id="cb14-73"><a href="#cb14-73" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(feature)) <span class="sc">%&gt;%</span> </span>
<span id="cb14-74"><a href="#cb14-74" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">rownames =</span> feature) <span class="sc">%&gt;%</span> </span>
<span id="cb14-75"><a href="#cb14-75" tabindex="-1"></a>    <span class="fu">column_to_rownames</span>(<span class="st">&#39;rownames&#39;</span>)</span>
<span id="cb14-76"><a href="#cb14-76" tabindex="-1"></a>  </span>
<span id="cb14-77"><a href="#cb14-77" tabindex="-1"></a>  <span class="fu">return</span>(dge)</span>
<span id="cb14-78"><a href="#cb14-78" tabindex="-1"></a>}</span>
<span id="cb14-79"><a href="#cb14-79" tabindex="-1"></a></span>
<span id="cb14-80"><a href="#cb14-80" tabindex="-1"></a><span class="fu">names</span>(dge_list) <span class="ot">&lt;-</span> <span class="fu">names</span>(comp_list)</span>
<span id="cb14-81"><a href="#cb14-81" tabindex="-1"></a><span class="fu">detach</span>(<span class="st">&quot;package:biomaRt&quot;</span>, <span class="at">unload=</span><span class="cn">TRUE</span>)</span>
<span id="cb14-82"><a href="#cb14-82" tabindex="-1"></a></span>
<span id="cb14-83"><a href="#cb14-83" tabindex="-1"></a><span class="co"># Save objects</span></span>
<span id="cb14-84"><a href="#cb14-84" tabindex="-1"></a><span class="fu">saveRDS</span>(dge_list, <span class="at">file =</span> <span class="fu">file.path</span>(params<span class="sc">$</span>output_dir, <span class="st">&#39;dge-edgeR_QLF.rds&#39;</span>))</span></code></pre></div>
</div>
<div id="functional-enrichment" class="section level3">
<h3>Functional enrichment</h3>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a>dge_list <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">file.path</span>(params<span class="sc">$</span>output_dir, <span class="st">&#39;dge-edgeR_QLF.rds&#39;</span>))</span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a>dge <span class="ot">&lt;-</span>  dge_list[[<span class="dv">1</span>]]</span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a>gse_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a>gse_list <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">dge =</span> dge_list) <span class="sc">%do%</span>{</span>
<span id="cb15-5"><a href="#cb15-5" tabindex="-1"></a>  gse_res <span class="ot">&lt;-</span> <span class="fu">gse_omnibus</span>(</span>
<span id="cb15-6"><a href="#cb15-6" tabindex="-1"></a>    <span class="at">feature_names =</span> dge<span class="sc">$</span>results<span class="sc">$</span>gene_name,</span>
<span id="cb15-7"><a href="#cb15-7" tabindex="-1"></a>    <span class="at">p =</span> dge<span class="sc">$</span>results<span class="sc">$</span>FDR,</span>
<span id="cb15-8"><a href="#cb15-8" tabindex="-1"></a>    <span class="at">fc =</span> dge<span class="sc">$</span>results<span class="sc">$</span>logFC,</span>
<span id="cb15-9"><a href="#cb15-9" tabindex="-1"></a>    <span class="at">fc_thrs =</span> <span class="fl">0.5</span>,</span>
<span id="cb15-10"><a href="#cb15-10" tabindex="-1"></a>    <span class="at">gmt_files =</span> gmt_files_symbols,</span>
<span id="cb15-11"><a href="#cb15-11" tabindex="-1"></a>    </span>
<span id="cb15-12"><a href="#cb15-12" tabindex="-1"></a>    <span class="at">run_gseGO =</span> <span class="cn">TRUE</span>,</span>
<span id="cb15-13"><a href="#cb15-13" tabindex="-1"></a>    </span>
<span id="cb15-14"><a href="#cb15-14" tabindex="-1"></a>    <span class="at">go_ontologies =</span> <span class="st">&#39;BP&#39;</span>,</span>
<span id="cb15-15"><a href="#cb15-15" tabindex="-1"></a>    <span class="at">go_simplify_p =</span> <span class="fl">0.05</span>,</span>
<span id="cb15-16"><a href="#cb15-16" tabindex="-1"></a>    <span class="at">args_gse =</span> <span class="fu">list</span>(<span class="at">minGSSize =</span> <span class="dv">10</span>, <span class="at">maxGSSize =</span> <span class="dv">500</span>, <span class="at">pvalueCutoff =</span> <span class="dv">1</span>),</span>
<span id="cb15-17"><a href="#cb15-17" tabindex="-1"></a>    <span class="at">annot_db =</span> <span class="st">&#39;org.Mm.eg.db&#39;</span>,</span>
<span id="cb15-18"><a href="#cb15-18" tabindex="-1"></a>    <span class="at">organism =</span> <span class="st">&#39;mmusculus&#39;</span></span>
<span id="cb15-19"><a href="#cb15-19" tabindex="-1"></a>    )</span>
<span id="cb15-20"><a href="#cb15-20" tabindex="-1"></a>  </span>
<span id="cb15-21"><a href="#cb15-21" tabindex="-1"></a>   <span class="fu">return</span>(gse_res)</span>
<span id="cb15-22"><a href="#cb15-22" tabindex="-1"></a></span>
<span id="cb15-23"><a href="#cb15-23" tabindex="-1"></a>}</span>
<span id="cb15-24"><a href="#cb15-24" tabindex="-1"></a><span class="fu">names</span>(gse_list) <span class="ot">&lt;-</span> <span class="fu">names</span>(dge_list)</span>
<span id="cb15-25"><a href="#cb15-25" tabindex="-1"></a></span>
<span id="cb15-26"><a href="#cb15-26" tabindex="-1"></a><span class="co"># Save objects</span></span>
<span id="cb15-27"><a href="#cb15-27" tabindex="-1"></a><span class="fu">saveRDS</span>(gse_list, <span class="at">file =</span> <span class="fu">file.path</span>(params<span class="sc">$</span>output_dir,<span class="st">&#39;gse_omnibus-edgeR_QLF.rds&#39;</span>))</span></code></pre></div>
</div>
<div id="load-data" class="section level3">
<h3>Load data</h3>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a>dge_list <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">file.path</span>(params<span class="sc">$</span>output_dir, <span class="st">&#39;dge-edgeR_QLF.rds&#39;</span>))</span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a>gse_list <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">file.path</span>(params<span class="sc">$</span>output_dir, <span class="st">&#39;gse_omnibus-edgeR_QLF.rds&#39;</span>))</span></code></pre></div>
</div>
</div>
<div id="differential-expression-analysis-1" class="section level2">
<h2>Differential expression analysis</h2>
<div
id="supplementary-figure-2g-top-differential-expressed-genes-in-dcas9-thy1-over-thy1-transduction"
class="section level3">
<h3>Supplementary Figure 2G top: Differential expressed genes in
dCas9-Thy1 over Thy1 transduction</h3>
<p>Transcriptomic analysis of primary tumors harvested from syngeneic
mice transplanted orthotopically with three mammary carcinoma cell
models (MVT1, 4T1, Py2T) at the timepoint indicated in A, untreated; n=3
each. Top: annotated heat map of hierarchically clustered genes,
highlighting transcripts differentially expressed in dCas9-Thy1(B)
versus Thy1(B) transduction conditions. Bottom: dot plots highlighting
representative gene signatures enriched in dCas9-Thy1(B) versus Thy1(B)
transduction conditions; NES, normalized enrichment score.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a><span class="co"># Heatmap parameters</span></span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a>p_thrs <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb17-3"><a href="#cb17-3" tabindex="-1"></a>lfc_thrs <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb17-4"><a href="#cb17-4" tabindex="-1"></a><span class="co"># topn &lt;- 50</span></span>
<span id="cb17-5"><a href="#cb17-5" tabindex="-1"></a>max_color_range <span class="ot">&lt;-</span> <span class="dv">3</span> <span class="co"># put the color range to -3 to 3</span></span>
<span id="cb17-6"><a href="#cb17-6" tabindex="-1"></a></span>
<span id="cb17-7"><a href="#cb17-7" tabindex="-1"></a><span class="co"># Select comparison</span></span>
<span id="cb17-8"><a href="#cb17-8" tabindex="-1"></a>i <span class="ot">&lt;-</span> <span class="st">&quot;dCas9_Thy1--over--Thy1&quot;</span></span>
<span id="cb17-9"><a href="#cb17-9" tabindex="-1"></a>use_comp <span class="ot">&lt;-</span> comp_list[[i]]</span>
<span id="cb17-10"><a href="#cb17-10" tabindex="-1"></a></span>
<span id="cb17-11"><a href="#cb17-11" tabindex="-1"></a><span class="co"># Select genes to highlight</span></span>
<span id="cb17-12"><a href="#cb17-12" tabindex="-1"></a>use_labels <span class="ot">&lt;-</span>  <span class="fu">c</span>(<span class="st">&#39;SpCas9&#39;</span>, <span class="st">&#39;H2-Q5&#39;</span>, <span class="st">&#39;H2-Q10&#39;</span>, <span class="st">&#39;Cd3e&#39;</span>, <span class="st">&#39;Cd3g&#39;</span>, <span class="st">&#39;H2-Ab1&#39;</span>, <span class="st">&#39;H2-Aa&#39;</span>, <span class="st">&#39;Tap1&#39;</span>, </span>
<span id="cb17-13"><a href="#cb17-13" tabindex="-1"></a>                 <span class="st">&#39;Tap2&#39;</span>, <span class="st">&#39;B2m&#39;</span>, <span class="st">&#39;Cldn3&#39;</span>, <span class="st">&#39;Cdh1&#39;</span>, <span class="st">&#39;Epcam&#39;</span>, <span class="st">&#39;Krt14&#39;</span>, <span class="st">&#39;Cldn7&#39;</span>)</span>
<span id="cb17-14"><a href="#cb17-14" tabindex="-1"></a>                 </span>
<span id="cb17-15"><a href="#cb17-15" tabindex="-1"></a><span class="co"># Extract samples from comparisons</span></span>
<span id="cb17-16"><a href="#cb17-16" tabindex="-1"></a>use_samples <span class="ot">&lt;-</span> <span class="fu">unlist</span>(use_comp[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>])</span>
<span id="cb17-17"><a href="#cb17-17" tabindex="-1"></a></span>
<span id="cb17-18"><a href="#cb17-18" tabindex="-1"></a><span class="co"># Add dge_groups annotation</span></span>
<span id="cb17-19"><a href="#cb17-19" tabindex="-1"></a><span class="fu">colData</span>(se) <span class="ot">&lt;-</span> <span class="fu">colData</span>(se) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span> </span>
<span id="cb17-20"><a href="#cb17-20" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb17-21"><a href="#cb17-21" tabindex="-1"></a>    <span class="at">condition_new =</span> <span class="fu">ifelse</span>(sample_alias <span class="sc">%in%</span> use_comp[[<span class="dv">1</span>]], </span>
<span id="cb17-22"><a href="#cb17-22" tabindex="-1"></a>                           <span class="fu">names</span>(use_comp)[<span class="dv">1</span>], </span>
<span id="cb17-23"><a href="#cb17-23" tabindex="-1"></a>                           <span class="cn">NA</span>),</span>
<span id="cb17-24"><a href="#cb17-24" tabindex="-1"></a>    <span class="at">condition_new =</span> <span class="fu">ifelse</span>(sample_alias <span class="sc">%in%</span> use_comp[[<span class="dv">2</span>]], </span>
<span id="cb17-25"><a href="#cb17-25" tabindex="-1"></a>                           <span class="fu">names</span>(use_comp)[<span class="dv">2</span>], </span>
<span id="cb17-26"><a href="#cb17-26" tabindex="-1"></a>                           condition_new),</span>
<span id="cb17-27"><a href="#cb17-27" tabindex="-1"></a>    <span class="at">condition_new =</span> <span class="fu">factor</span>(condition_new, <span class="at">levels =</span> <span class="fu">names</span>(use_comp)[<span class="dv">2</span><span class="sc">:</span><span class="dv">1</span>]),</span>
<span id="cb17-28"><a href="#cb17-28" tabindex="-1"></a>    <span class="at">condition_num =</span> <span class="fu">as.numeric</span>(condition_new)</span>
<span id="cb17-29"><a href="#cb17-29" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> DataFrame</span>
<span id="cb17-30"><a href="#cb17-30" tabindex="-1"></a></span>
<span id="cb17-31"><a href="#cb17-31" tabindex="-1"></a><span class="co"># Average SE by condition</span></span>
<span id="cb17-32"><a href="#cb17-32" tabindex="-1"></a>group_id <span class="ot">&lt;-</span> <span class="fu">paste</span>(se<span class="sc">$</span>donor, se<span class="sc">$</span>condition_new) <span class="sc">%&gt;%</span> </span>
<span id="cb17-33"><a href="#cb17-33" tabindex="-1"></a>  <span class="fu">gsub</span>(<span class="st">&quot;CD8-knockout&quot;</span>, <span class="st">&quot;CD8-KO&quot;</span>, .) <span class="sc">%&gt;%</span> </span>
<span id="cb17-34"><a href="#cb17-34" tabindex="-1"></a>  <span class="fu">gsub</span>(<span class="st">&quot;CD8-transduced&quot;</span>, <span class="st">&quot;CD8&quot;</span>, .) <span class="sc">%&gt;%</span> </span>
<span id="cb17-35"><a href="#cb17-35" tabindex="-1"></a>  <span class="fu">gsub</span>(<span class="st">&quot;Rejected Cas9-BROAD&quot;</span>, <span class="st">&quot;Cas9-BROAD&quot;</span>, .)</span>
<span id="cb17-36"><a href="#cb17-36" tabindex="-1"></a></span>
<span id="cb17-37"><a href="#cb17-37" tabindex="-1"></a>use_se <span class="ot">&lt;-</span> <span class="fu">aggregateAcrossCells</span>(</span>
<span id="cb17-38"><a href="#cb17-38" tabindex="-1"></a>  se, </span>
<span id="cb17-39"><a href="#cb17-39" tabindex="-1"></a>  <span class="at">statistics =</span> <span class="st">&quot;mean&quot;</span>, </span>
<span id="cb17-40"><a href="#cb17-40" tabindex="-1"></a>  <span class="at">subset.col =</span> use_samples,</span>
<span id="cb17-41"><a href="#cb17-41" tabindex="-1"></a>  <span class="at">ids =</span> group_id,</span>
<span id="cb17-42"><a href="#cb17-42" tabindex="-1"></a>  <span class="at">use.assay.type =</span> <span class="st">&#39;logcounts&#39;</span></span>
<span id="cb17-43"><a href="#cb17-43" tabindex="-1"></a>  )</span>
<span id="cb17-44"><a href="#cb17-44" tabindex="-1"></a></span>
<span id="cb17-45"><a href="#cb17-45" tabindex="-1"></a></span>
<span id="cb17-46"><a href="#cb17-46" tabindex="-1"></a><span class="co"># Arrange samples by condition</span></span>
<span id="cb17-47"><a href="#cb17-47" tabindex="-1"></a>use_annot <span class="ot">&lt;-</span> <span class="fu">colData</span>(use_se) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span> </span>
<span id="cb17-48"><a href="#cb17-48" tabindex="-1"></a>  <span class="fu">arrange</span>(condition_num, donor)</span>
<span id="cb17-49"><a href="#cb17-49" tabindex="-1"></a>use_samples <span class="ot">&lt;-</span> use_annot<span class="sc">$</span>ids</span>
<span id="cb17-50"><a href="#cb17-50" tabindex="-1"></a></span>
<span id="cb17-51"><a href="#cb17-51" tabindex="-1"></a><span class="co"># Extract DGE results</span></span>
<span id="cb17-52"><a href="#cb17-52" tabindex="-1"></a>use_dge <span class="ot">&lt;-</span> dge_list[[i]]<span class="sc">$</span>results <span class="sc">%&gt;%</span> </span>
<span id="cb17-53"><a href="#cb17-53" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb17-54"><a href="#cb17-54" tabindex="-1"></a>    <span class="at">significant =</span> (FDR <span class="sc">&lt;</span> p_thrs <span class="sc">&amp;</span> <span class="fu">abs</span>(logFC) <span class="sc">&gt;</span> lfc_thrs),</span>
<span id="cb17-55"><a href="#cb17-55" tabindex="-1"></a>    <span class="at">selected =</span> <span class="fu">ifelse</span>(gene_name <span class="sc">%in%</span> use_labels, <span class="cn">TRUE</span>, significant),</span>
<span id="cb17-56"><a href="#cb17-56" tabindex="-1"></a>    <span class="at">direction =</span> <span class="fu">ifelse</span>(logFC <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">&#39;Up&#39;</span>, <span class="st">&#39;Down&#39;</span>)</span>
<span id="cb17-57"><a href="#cb17-57" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb17-58"><a href="#cb17-58" tabindex="-1"></a>  <span class="fu">filter</span>(selected) <span class="sc">%&gt;%</span> </span>
<span id="cb17-59"><a href="#cb17-59" tabindex="-1"></a>  <span class="fu">arrange</span>(FDR)</span>
<span id="cb17-60"><a href="#cb17-60" tabindex="-1"></a></span>
<span id="cb17-61"><a href="#cb17-61" tabindex="-1"></a><span class="co"># Extract matrix of expression</span></span>
<span id="cb17-62"><a href="#cb17-62" tabindex="-1"></a>use_se <span class="ot">&lt;-</span> use_se[<span class="fu">rownames</span>(use_dge), use_samples]</span>
<span id="cb17-63"><a href="#cb17-63" tabindex="-1"></a>mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(use_se, <span class="st">&#39;logcounts&#39;</span>)</span>
<span id="cb17-64"><a href="#cb17-64" tabindex="-1"></a><span class="fu">rownames</span>(mat) <span class="ot">&lt;-</span> use_dge<span class="sc">$</span>gene_name</span>
<span id="cb17-65"><a href="#cb17-65" tabindex="-1"></a></span>
<span id="cb17-66"><a href="#cb17-66" tabindex="-1"></a><span class="co"># Calculate fold-change by donor</span></span>
<span id="cb17-67"><a href="#cb17-67" tabindex="-1"></a>donor_levels <span class="ot">&lt;-</span> <span class="fu">unique</span>(use_annot<span class="sc">$</span>donor)</span>
<span id="cb17-68"><a href="#cb17-68" tabindex="-1"></a>fcmat <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">use_donor =</span> donor_levels, <span class="at">.combine =</span> cbind) <span class="sc">%do%</span> {</span>
<span id="cb17-69"><a href="#cb17-69" tabindex="-1"></a>  s1 <span class="ot">&lt;-</span> use_annot <span class="sc">%&gt;%</span> </span>
<span id="cb17-70"><a href="#cb17-70" tabindex="-1"></a>    <span class="fu">filter</span>(donor <span class="sc">==</span> use_donor) <span class="sc">%&gt;%</span> </span>
<span id="cb17-71"><a href="#cb17-71" tabindex="-1"></a>    <span class="fu">filter</span>(condition_new <span class="sc">==</span> <span class="fu">names</span>(use_comp)[[<span class="dv">1</span>]]) <span class="sc">%&gt;%</span> </span>
<span id="cb17-72"><a href="#cb17-72" tabindex="-1"></a>    <span class="fu">pull</span>(ids)</span>
<span id="cb17-73"><a href="#cb17-73" tabindex="-1"></a>  s2 <span class="ot">&lt;-</span> use_annot <span class="sc">%&gt;%</span> </span>
<span id="cb17-74"><a href="#cb17-74" tabindex="-1"></a>    <span class="fu">filter</span>(donor <span class="sc">==</span> use_donor) <span class="sc">%&gt;%</span> </span>
<span id="cb17-75"><a href="#cb17-75" tabindex="-1"></a>    <span class="fu">filter</span>(condition_new <span class="sc">==</span> <span class="fu">names</span>(use_comp)[[<span class="dv">2</span>]]) <span class="sc">%&gt;%</span> </span>
<span id="cb17-76"><a href="#cb17-76" tabindex="-1"></a>    <span class="fu">pull</span>(ids)</span>
<span id="cb17-77"><a href="#cb17-77" tabindex="-1"></a>  <span class="fu">return</span>(mat[,s1]<span class="sc">/</span>mat[,s2])</span>
<span id="cb17-78"><a href="#cb17-78" tabindex="-1"></a>}</span>
<span id="cb17-79"><a href="#cb17-79" tabindex="-1"></a><span class="fu">colnames</span>(fcmat) <span class="ot">&lt;-</span> donor_levels</span>
<span id="cb17-80"><a href="#cb17-80" tabindex="-1"></a>fcmat[<span class="fu">is.na</span>(fcmat)] <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb17-81"><a href="#cb17-81" tabindex="-1"></a>fcmat[fcmat <span class="sc">==</span> <span class="cn">Inf</span>] <span class="ot">&lt;-</span> <span class="fu">max</span>(fcmat[fcmat <span class="sc">&lt;</span> <span class="cn">Inf</span>])</span>
<span id="cb17-82"><a href="#cb17-82" tabindex="-1"></a>fcmat[fcmat <span class="sc">==</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="fu">min</span>(fcmat[fcmat <span class="sc">&gt;</span> <span class="dv">0</span>])</span>
<span id="cb17-83"><a href="#cb17-83" tabindex="-1"></a>fcmat <span class="ot">&lt;-</span> <span class="fu">log2</span>(fcmat)</span>
<span id="cb17-84"><a href="#cb17-84" tabindex="-1"></a></span>
<span id="cb17-85"><a href="#cb17-85" tabindex="-1"></a></span>
<span id="cb17-86"><a href="#cb17-86" tabindex="-1"></a><span class="co"># Row annotation for selected labels</span></span>
<span id="cb17-87"><a href="#cb17-87" tabindex="-1"></a>row_annot_selected_labels <span class="ot">&lt;-</span> <span class="fu">rowAnnotation</span>(</span>
<span id="cb17-88"><a href="#cb17-88" tabindex="-1"></a>  <span class="at">selected =</span> <span class="fu">anno_mark</span>(</span>
<span id="cb17-89"><a href="#cb17-89" tabindex="-1"></a>    <span class="at">at =</span> <span class="fu">match</span>(use_labels, <span class="fu">rownames</span>(fcmat)), </span>
<span id="cb17-90"><a href="#cb17-90" tabindex="-1"></a>    <span class="at">labels =</span>use_labels,</span>
<span id="cb17-91"><a href="#cb17-91" tabindex="-1"></a>    <span class="at">labels_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">6</span>),</span>
<span id="cb17-92"><a href="#cb17-92" tabindex="-1"></a>    <span class="at">lines_gp =</span> <span class="fu">gpar</span>(<span class="at">lwd =</span> <span class="fl">0.3</span>),</span>
<span id="cb17-93"><a href="#cb17-93" tabindex="-1"></a>    <span class="at">padding =</span> <span class="fu">unit</span>(<span class="fl">0.01</span>, <span class="st">&quot;inches&quot;</span>),</span>
<span id="cb17-94"><a href="#cb17-94" tabindex="-1"></a>    <span class="at">link_width =</span> <span class="fu">unit</span>(<span class="fl">0.1</span>, <span class="st">&quot;inches&quot;</span>)</span>
<span id="cb17-95"><a href="#cb17-95" tabindex="-1"></a>    )</span>
<span id="cb17-96"><a href="#cb17-96" tabindex="-1"></a>  )</span>
<span id="cb17-97"><a href="#cb17-97" tabindex="-1"></a></span>
<span id="cb17-98"><a href="#cb17-98" tabindex="-1"></a></span>
<span id="cb17-99"><a href="#cb17-99" tabindex="-1"></a><span class="co"># Heatmap color</span></span>
<span id="cb17-100"><a href="#cb17-100" tabindex="-1"></a>zmax <span class="ot">&lt;-</span> fcmat <span class="sc">%&gt;%</span> abs <span class="sc">%&gt;%</span> max</span>
<span id="cb17-101"><a href="#cb17-101" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(max_color_range))</span>
<span id="cb17-102"><a href="#cb17-102" tabindex="-1"></a>  zmax <span class="ot">&lt;-</span> max_color_range</span>
<span id="cb17-103"><a href="#cb17-103" tabindex="-1"></a>col_fun <span class="ot">&lt;-</span>  <span class="fu">colorRamp2</span>(</span>
<span id="cb17-104"><a href="#cb17-104" tabindex="-1"></a>  <span class="fu">seq</span>(<span class="sc">-</span>zmax, zmax, <span class="at">length.out =</span> <span class="dv">11</span>),</span>
<span id="cb17-105"><a href="#cb17-105" tabindex="-1"></a>  <span class="fu">rev</span>(<span class="fu">brewer.pal</span>(<span class="dv">11</span>, <span class="st">&quot;RdBu&quot;</span>))</span>
<span id="cb17-106"><a href="#cb17-106" tabindex="-1"></a>)</span>
<span id="cb17-107"><a href="#cb17-107" tabindex="-1"></a></span>
<span id="cb17-108"><a href="#cb17-108" tabindex="-1"></a></span>
<span id="cb17-109"><a href="#cb17-109" tabindex="-1"></a><span class="co"># Generate Heatmap</span></span>
<span id="cb17-110"><a href="#cb17-110" tabindex="-1"></a>htz <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(</span>
<span id="cb17-111"><a href="#cb17-111" tabindex="-1"></a>  fcmat,</span>
<span id="cb17-112"><a href="#cb17-112" tabindex="-1"></a>  <span class="at">name=</span> <span class="st">&#39;log2(fold change)&#39;</span>,</span>
<span id="cb17-113"><a href="#cb17-113" tabindex="-1"></a>  <span class="at">col =</span> col_fun,</span>
<span id="cb17-114"><a href="#cb17-114" tabindex="-1"></a>  <span class="at">show_row_names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb17-115"><a href="#cb17-115" tabindex="-1"></a>  <span class="at">show_column_names =</span> <span class="cn">TRUE</span>,</span>
<span id="cb17-116"><a href="#cb17-116" tabindex="-1"></a>  <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">3</span>),</span>
<span id="cb17-117"><a href="#cb17-117" tabindex="-1"></a>  <span class="at">column_names_rot =</span> <span class="dv">0</span>,</span>
<span id="cb17-118"><a href="#cb17-118" tabindex="-1"></a>  <span class="at">column_names_side =</span> <span class="st">&#39;top&#39;</span>,</span>
<span id="cb17-119"><a href="#cb17-119" tabindex="-1"></a>  <span class="at">cluster_column_slices =</span> <span class="cn">FALSE</span>,</span>
<span id="cb17-120"><a href="#cb17-120" tabindex="-1"></a>  </span>
<span id="cb17-121"><a href="#cb17-121" tabindex="-1"></a>  <span class="at">column_split =</span> <span class="fu">colnames</span>(fcmat),</span>
<span id="cb17-122"><a href="#cb17-122" tabindex="-1"></a>  <span class="at">column_gap =</span> <span class="fu">unit</span>(<span class="fl">0.039</span><span class="sc">/</span><span class="dv">3</span>, <span class="st">&quot;inches&quot;</span>),</span>
<span id="cb17-123"><a href="#cb17-123" tabindex="-1"></a>  <span class="at">column_title =</span> <span class="cn">NULL</span>,</span>
<span id="cb17-124"><a href="#cb17-124" tabindex="-1"></a>  <span class="at">cluster_row_slices =</span> <span class="cn">FALSE</span>,</span>
<span id="cb17-125"><a href="#cb17-125" tabindex="-1"></a>  </span>
<span id="cb17-126"><a href="#cb17-126" tabindex="-1"></a>  <span class="at">right_annotation =</span> row_annot_selected_labels,</span>
<span id="cb17-127"><a href="#cb17-127" tabindex="-1"></a>  <span class="at">row_split =</span> use_dge<span class="sc">$</span>direction,</span>
<span id="cb17-128"><a href="#cb17-128" tabindex="-1"></a>  <span class="at">row_gap =</span> <span class="fu">unit</span>(<span class="fl">0.039</span><span class="sc">/</span><span class="dv">2</span>, <span class="st">&quot;inches&quot;</span>),</span>
<span id="cb17-129"><a href="#cb17-129" tabindex="-1"></a>  <span class="at">row_title =</span> <span class="cn">NULL</span>,</span>
<span id="cb17-130"><a href="#cb17-130" tabindex="-1"></a>  </span>
<span id="cb17-131"><a href="#cb17-131" tabindex="-1"></a>  <span class="at">cluster_columns =</span> <span class="cn">FALSE</span>,</span>
<span id="cb17-132"><a href="#cb17-132" tabindex="-1"></a>  <span class="at">show_row_dend =</span> <span class="cn">FALSE</span>,</span>
<span id="cb17-133"><a href="#cb17-133" tabindex="-1"></a>  </span>
<span id="cb17-134"><a href="#cb17-134" tabindex="-1"></a>  <span class="at">heatmap_legend_param =</span> <span class="fu">list</span>(</span>
<span id="cb17-135"><a href="#cb17-135" tabindex="-1"></a>    <span class="at">direction =</span> <span class="st">&quot;horizontal&quot;</span>,</span>
<span id="cb17-136"><a href="#cb17-136" tabindex="-1"></a>    <span class="at">title_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">3</span>, <span class="at">fontface =</span> <span class="st">&quot;bold&quot;</span>),</span>
<span id="cb17-137"><a href="#cb17-137" tabindex="-1"></a>    <span class="at">labels_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">3</span>),</span>
<span id="cb17-138"><a href="#cb17-138" tabindex="-1"></a>    <span class="at">legend_width =</span> <span class="fu">unit</span>(<span class="fl">0.5</span>, <span class="st">&quot;inches&quot;</span>),</span>
<span id="cb17-139"><a href="#cb17-139" tabindex="-1"></a>    <span class="at">grid_height =</span> <span class="fu">unit</span>(<span class="fl">0.03</span>, <span class="st">&quot;inches&quot;</span>),</span>
<span id="cb17-140"><a href="#cb17-140" tabindex="-1"></a>    <span class="at">title_position =</span> <span class="st">&#39;lefttop&#39;</span></span>
<span id="cb17-141"><a href="#cb17-141" tabindex="-1"></a>  )</span>
<span id="cb17-142"><a href="#cb17-142" tabindex="-1"></a>)</span>
<span id="cb17-143"><a href="#cb17-143" tabindex="-1"></a></span>
<span id="cb17-144"><a href="#cb17-144" tabindex="-1"></a><span class="fu">draw</span>(htz,  <span class="at">heatmap_legend_side =</span> <span class="st">&quot;bottom&quot;</span>)</span></code></pre></div>
<p><img src="figure/rnaseq-tumor-bulk.Rmd/dge-heatmap-fc-dc9thy1--over--thy1-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-dge-heatmap-fc-dc9thy1--over--thy1-1">
Past versions of dge-heatmap-fc-dc9thy1–over–thy1-1.png
</button>
</p>
<div id="fig-dge-heatmap-fc-dc9thy1--over--thy1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/8632845b1c6b024330836a1a798537363e31980e/docs/figure/rnaseq-tumor-bulk.Rmd/dge-heatmap-fc-dc9thy1--over--thy1-1.png" target="_blank">8632845</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-07-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="functional-enrichment-1" class="section level2">
<h2>Functional Enrichment</h2>
<div
id="supplementary-figure-2g-bottom-representative-gene-sets-enriched-in-dcas9-thy1-over-thy1-transduction"
class="section level3">
<h3>Supplementary Figure 2G bottom: Representative gene-sets enriched in
dCas9-Thy1 over Thy1 transduction</h3>
<p>Transcriptomic analysis of primary tumors harvested from syngeneic
mice transplanted orthotopically with three mammary carcinoma cell
models (MVT1, 4T1, Py2T) at the timepoint indicated in A, untreated; n=3
each. Top: annotated heat map of hierarchically clustered genes,
highlighting transcripts differentially expressed in dCas9-Thy1(B)
versus Thy1(B) transduction conditions. Bottom: dot plots highlighting
representative gene signatures enriched in dCas9-Thy1(B) versus Thy1(B)
transduction conditions; NES, normalized enrichment score.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a>i <span class="ot">&lt;-</span> <span class="st">&quot;dCas9_Thy1--over--Thy1&quot;</span></span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a>use_sets <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;GO:0019882&quot;</span>,<span class="st">&quot;GO:0002250&quot;</span>,<span class="st">&quot;GO:0050870&quot;</span>,<span class="st">&quot;GO:0001819&quot;</span>,<span class="st">&quot;GO:0001906&quot;</span>,</span>
<span id="cb19-3"><a href="#cb19-3" tabindex="-1"></a>              <span class="st">&quot;GO:0006909&quot;</span>,<span class="st">&quot;GO:0001910&quot;</span>,<span class="st">&quot;GO:1903037&quot;</span>,<span class="st">&quot;GO:0050670&quot;</span>,<span class="st">&quot;GO:0002697&quot;</span></span>
<span id="cb19-4"><a href="#cb19-4" tabindex="-1"></a>              )</span>
<span id="cb19-5"><a href="#cb19-5" tabindex="-1"></a>use_gse_res <span class="ot">&lt;-</span> gse_list[[i]]<span class="sc">$</span>gseGO<span class="sc">$</span>BP<span class="sc">@</span>result[use_sets,]</span>
<span id="cb19-6"><a href="#cb19-6" tabindex="-1"></a>use_gse_res<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">factor</span>(use_gse_res<span class="sc">$</span>ID, <span class="at">levels =</span> <span class="fu">rev</span>(use_sets))</span>
<span id="cb19-7"><a href="#cb19-7" tabindex="-1"></a></span>
<span id="cb19-8"><a href="#cb19-8" tabindex="-1"></a>use_gse_res<span class="sc">$</span>Description_short <span class="ot">&lt;-</span> <span class="fu">recode</span>(use_gse_res<span class="sc">$</span>Description,</span>
<span id="cb19-9"><a href="#cb19-9" tabindex="-1"></a>                                               <span class="st">`</span><span class="at">antigen processing and presentation</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Antigen processing and presentation&quot;</span>,</span>
<span id="cb19-10"><a href="#cb19-10" tabindex="-1"></a>                                               <span class="st">`</span><span class="at">adaptive immune response</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Adaptive immune response&quot;</span>,</span>
<span id="cb19-11"><a href="#cb19-11" tabindex="-1"></a>                                               <span class="st">`</span><span class="at">positive regulation of T cell activation</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;T cell activation&quot;</span>,</span>
<span id="cb19-12"><a href="#cb19-12" tabindex="-1"></a>                                               <span class="st">`</span><span class="at">positive regulation of cytokine production</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Cytokine production&quot;</span>,</span>
<span id="cb19-13"><a href="#cb19-13" tabindex="-1"></a>                                               <span class="st">`</span><span class="at">cell killing</span><span class="st">`</span>   <span class="ot">=</span> <span class="st">&quot;Cell killing&quot;</span>,</span>
<span id="cb19-14"><a href="#cb19-14" tabindex="-1"></a>                                               <span class="st">`</span><span class="at">phagocytosis</span><span class="st">`</span>   <span class="ot">=</span> <span class="st">&quot;Phagocytosis&quot;</span>,</span>
<span id="cb19-15"><a href="#cb19-15" tabindex="-1"></a>                                               <span class="st">`</span><span class="at">regulation of leukocyte mediated cytotoxicity</span><span class="st">`</span>  <span class="ot">=</span> <span class="st">&quot;Leukocyte cytotoxicity&quot;</span>,</span>
<span id="cb19-16"><a href="#cb19-16" tabindex="-1"></a>                                               <span class="st">`</span><span class="at">regulation of leukocyte cell-cell adhesion</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Leukocyte adhesion&quot;</span>,</span>
<span id="cb19-17"><a href="#cb19-17" tabindex="-1"></a>                                               <span class="st">`</span><span class="at">regulation of lymphocyte proliferation</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Lymphocyte proliferation&quot;</span>,</span>
<span id="cb19-18"><a href="#cb19-18" tabindex="-1"></a>                                               <span class="st">`</span><span class="at">regulation of immune effector process</span><span class="st">`</span>  <span class="ot">=</span> <span class="st">&quot;Immune effector process&quot;</span></span>
<span id="cb19-19"><a href="#cb19-19" tabindex="-1"></a>)</span>
<span id="cb19-20"><a href="#cb19-20" tabindex="-1"></a></span>
<span id="cb19-21"><a href="#cb19-21" tabindex="-1"></a>use_gse_res<span class="sc">$</span>Description <span class="ot">&lt;-</span> <span class="fu">factor</span>(use_gse_res<span class="sc">$</span>Description, <span class="at">levels =</span> <span class="fu">rev</span>(use_gse_res<span class="sc">$</span>Description))</span>
<span id="cb19-22"><a href="#cb19-22" tabindex="-1"></a>use_gse_res<span class="sc">$</span>Description_short <span class="ot">&lt;-</span> <span class="fu">factor</span>(use_gse_res<span class="sc">$</span>Description_short, <span class="at">levels =</span> <span class="fu">rev</span>(use_gse_res<span class="sc">$</span>Description_short))</span>
<span id="cb19-23"><a href="#cb19-23" tabindex="-1"></a></span>
<span id="cb19-24"><a href="#cb19-24" tabindex="-1"></a></span>
<span id="cb19-25"><a href="#cb19-25" tabindex="-1"></a></span>
<span id="cb19-26"><a href="#cb19-26" tabindex="-1"></a>label_func <span class="ot">&lt;-</span> <span class="fu">default_labeller</span>(<span class="dv">30</span>)</span>
<span id="cb19-27"><a href="#cb19-27" tabindex="-1"></a>x_max <span class="ot">&lt;-</span> <span class="fu">range</span>(use_gse_res<span class="sc">$</span>NES) <span class="sc">%&gt;%</span> abs <span class="sc">%&gt;%</span> max</span>
<span id="cb19-28"><a href="#cb19-28" tabindex="-1"></a>size_max <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="sc">-</span><span class="fu">log10</span>(use_gse_res<span class="sc">$</span>p.adjust))</span>
<span id="cb19-29"><a href="#cb19-29" tabindex="-1"></a>res <span class="ot">&lt;-</span> use_gse_res <span class="sc">%&gt;%</span> </span>
<span id="cb19-30"><a href="#cb19-30" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(NES, Description_short, <span class="at">color =</span> NES, <span class="at">size =</span> <span class="sc">-</span><span class="fu">log10</span>(p.adjust))) <span class="sc">+</span></span>
<span id="cb19-31"><a href="#cb19-31" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb19-32"><a href="#cb19-32" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">&#39;dotted&#39;</span>, <span class="at">linewidth =</span> one_pt<span class="sc">/</span><span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb19-33"><a href="#cb19-33" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">labels =</span> label_func) <span class="sc">+</span></span>
<span id="cb19-34"><a href="#cb19-34" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(</span>
<span id="cb19-35"><a href="#cb19-35" tabindex="-1"></a>    <span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(.<span class="dv">1</span>, .<span class="dv">1</span>)),</span>
<span id="cb19-36"><a href="#cb19-36" tabindex="-1"></a>    <span class="at">limits =</span> <span class="fu">c</span>(<span class="sc">-</span>x_max, x_max)</span>
<span id="cb19-37"><a href="#cb19-37" tabindex="-1"></a>    )<span class="sc">+</span></span>
<span id="cb19-38"><a href="#cb19-38" tabindex="-1"></a>  <span class="fu">scale_size</span>(<span class="at">range =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="dv">2</span>), </span>
<span id="cb19-39"><a href="#cb19-39" tabindex="-1"></a>             <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,size_max)) <span class="sc">+</span></span>
<span id="cb19-40"><a href="#cb19-40" tabindex="-1"></a>  <span class="fu">scale_color_distiller</span>(</span>
<span id="cb19-41"><a href="#cb19-41" tabindex="-1"></a>    <span class="at">palette =</span> <span class="st">&quot;RdBu&quot;</span>,</span>
<span id="cb19-42"><a href="#cb19-42" tabindex="-1"></a>    <span class="at">limits =</span> <span class="fu">c</span>(<span class="sc">-</span>x_max, x_max)</span>
<span id="cb19-43"><a href="#cb19-43" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb19-44"><a href="#cb19-44" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb19-45"><a href="#cb19-45" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">&#39;NES&#39;</span>,</span>
<span id="cb19-46"><a href="#cb19-46" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">&#39;&#39;</span>,</span>
<span id="cb19-47"><a href="#cb19-47" tabindex="-1"></a>    <span class="at">size =</span>  <span class="fu">bquote</span>(<span class="st">&quot;-log&quot;</span>[<span class="dv">10</span>] <span class="sc">~</span> .(<span class="fu">paste0</span>(<span class="st">&quot;(&quot;</span>, <span class="st">&#39;Adjusted P&#39;</span>, <span class="st">&quot;)&quot;</span>))),</span>
<span id="cb19-48"><a href="#cb19-48" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">&#39;NES&#39;</span></span>
<span id="cb19-49"><a href="#cb19-49" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb19-50"><a href="#cb19-50" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb19-51"><a href="#cb19-51" tabindex="-1"></a>      <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb19-52"><a href="#cb19-52" tabindex="-1"></a>      <span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">linewidth =</span> one_pt<span class="sc">/</span><span class="dv">4</span>, <span class="at">color =</span> <span class="st">&#39;black&#39;</span>),</span>
<span id="cb19-53"><a href="#cb19-53" tabindex="-1"></a>      <span class="at">axis.ticks =</span> <span class="fu">element_line</span>(<span class="at">linewidth =</span> one_pt<span class="sc">/</span><span class="dv">4</span>, <span class="at">color =</span> <span class="st">&#39;black&#39;</span>),</span>
<span id="cb19-54"><a href="#cb19-54" tabindex="-1"></a>      <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">8</span>), </span>
<span id="cb19-55"><a href="#cb19-55" tabindex="-1"></a>      <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>),</span>
<span id="cb19-56"><a href="#cb19-56" tabindex="-1"></a>      <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">8</span>),</span>
<span id="cb19-57"><a href="#cb19-57" tabindex="-1"></a>      <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">8</span>),</span>
<span id="cb19-58"><a href="#cb19-58" tabindex="-1"></a>      <span class="at">legend.position=</span><span class="st">&quot;bottom&quot;</span>,</span>
<span id="cb19-59"><a href="#cb19-59" tabindex="-1"></a>      <span class="at">legend.box =</span> <span class="st">&#39;vertical&#39;</span></span>
<span id="cb19-60"><a href="#cb19-60" tabindex="-1"></a>    )</span>
<span id="cb19-61"><a href="#cb19-61" tabindex="-1"></a>res_legend <span class="ot">&lt;-</span> <span class="fu">get_legend</span>(res)</span>
<span id="cb19-62"><a href="#cb19-62" tabindex="-1"></a><span class="fu">print</span>(res <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&#39;none&#39;</span>))</span></code></pre></div>
<p><img src="figure/rnaseq-tumor-bulk.Rmd/gsea-curated-dot-plot-dc9thy1--over--thy1-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-gsea-curated-dot-plot-dc9thy1--over--thy1-1">
Past versions of gsea-curated-dot-plot-dc9thy1–over–thy1-1.png
</button>
</p>
<div id="fig-gsea-curated-dot-plot-dc9thy1--over--thy1-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/8632845b1c6b024330836a1a798537363e31980e/docs/figure/rnaseq-tumor-bulk.Rmd/gsea-curated-dot-plot-dc9thy1--over--thy1-1.png" target="_blank">8632845</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-07-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a><span class="fu">plot_grid</span>(res_legend, <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">nrow =</span> <span class="dv">1</span>)</span></code></pre></div>
<p><img src="figure/rnaseq-tumor-bulk.Rmd/gsea-curated-dot-plot-legend-dc9thy1--over--thy1-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-gsea-curated-dot-plot-legend-dc9thy1--over--thy1-1">
Past versions of gsea-curated-dot-plot-legend-dc9thy1–over–thy1-1.png
</button>
</p>
<div id="fig-gsea-curated-dot-plot-legend-dc9thy1--over--thy1-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/8632845b1c6b024330836a1a798537363e31980e/docs/figure/rnaseq-tumor-bulk.Rmd/gsea-curated-dot-plot-legend-dc9thy1--over--thy1-1.png" target="_blank">8632845</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-07-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="sample-clustering" class="section level2">
<h2>Sample clustering</h2>
<p>Define groups of samples to analyse and feature to include</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a><span class="co"># Groups of samples</span></span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">colData</span>(se) <span class="sc">%&gt;%</span> data.frame</span>
<span id="cb23-3"><a href="#cb23-3" tabindex="-1"></a>clustering_groups <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb23-4"><a href="#cb23-4" tabindex="-1"></a>  <span class="st">`</span><span class="at">MVT1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">colData</span>(se) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span> <span class="fu">filter</span>(donor <span class="sc">==</span> <span class="st">&#39;MVT1&#39;</span> <span class="sc">&amp;</span> condition <span class="sc">!=</span> <span class="st">&#39;Rejection control&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(<span class="st">&#39;sample_alias&#39;</span>),</span>
<span id="cb23-5"><a href="#cb23-5" tabindex="-1"></a>  <span class="st">`</span><span class="at">4T1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">colData</span>(se) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span> <span class="fu">filter</span>(donor <span class="sc">==</span> <span class="st">&#39;4T1&#39;</span> <span class="sc">&amp;</span> condition <span class="sc">!=</span> <span class="st">&#39;Rejection control&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(<span class="st">&#39;sample_alias&#39;</span>),</span>
<span id="cb23-6"><a href="#cb23-6" tabindex="-1"></a>  <span class="st">`</span><span class="at">Py2T</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">colData</span>(se) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span> <span class="fu">filter</span>(donor <span class="sc">==</span> <span class="st">&#39;Py2T&#39;</span> <span class="sc">&amp;</span> condition <span class="sc">!=</span> <span class="st">&#39;Rejection control&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(<span class="st">&#39;sample_alias&#39;</span>)</span>
<span id="cb23-7"><a href="#cb23-7" tabindex="-1"></a>)</span>
<span id="cb23-8"><a href="#cb23-8" tabindex="-1"></a></span>
<span id="cb23-9"><a href="#cb23-9" tabindex="-1"></a><span class="co"># Define features</span></span>
<span id="cb23-10"><a href="#cb23-10" tabindex="-1"></a>use_features <span class="ot">&lt;-</span> <span class="fu">rowData</span>(se)<span class="sc">$</span>gene_type <span class="sc">==</span> <span class="st">&#39;protein_coding&#39;</span> <span class="sc">|</span> <span class="fu">rowData</span>(se)<span class="sc">$</span>gene_name <span class="sc">==</span> <span class="st">&#39;SpCas9&#39;</span></span>
<span id="cb23-11"><a href="#cb23-11" tabindex="-1"></a></span>
<span id="cb23-12"><a href="#cb23-12" tabindex="-1"></a><span class="co"># Heatmap colors</span></span>
<span id="cb23-13"><a href="#cb23-13" tabindex="-1"></a>ph_colors <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">rev</span>(<span class="fu">brewer.pal</span>(<span class="at">n =</span> <span class="dv">7</span>, <span class="at">name =</span> <span class="st">&quot;Purples&quot;</span>)))(<span class="dv">255</span>)</span>
<span id="cb23-14"><a href="#cb23-14" tabindex="-1"></a></span>
<span id="cb23-15"><a href="#cb23-15" tabindex="-1"></a><span class="co"># Set annotation colors</span></span>
<span id="cb23-16"><a href="#cb23-16" tabindex="-1"></a>condition_levels <span class="ot">&lt;-</span> se<span class="sc">$</span>condition <span class="sc">%&gt;%</span> unique</span>
<span id="cb23-17"><a href="#cb23-17" tabindex="-1"></a>condition_palette <span class="ot">&lt;-</span> <span class="fu">get_scater_palette</span>(<span class="st">&#39;tableau10medium&#39;</span>)[<span class="fu">seq_along</span>(condition_levels)] <span class="sc">%&gt;%</span></span>
<span id="cb23-18"><a href="#cb23-18" tabindex="-1"></a>  <span class="fu">set_names</span>(condition_levels)</span>
<span id="cb23-19"><a href="#cb23-19" tabindex="-1"></a></span>
<span id="cb23-20"><a href="#cb23-20" tabindex="-1"></a><span class="co"># Define annotation</span></span>
<span id="cb23-21"><a href="#cb23-21" tabindex="-1"></a>ann_list <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">comp =</span> clustering_groups) <span class="sc">%do%</span> {</span>
<span id="cb23-22"><a href="#cb23-22" tabindex="-1"></a>  <span class="fu">colData</span>(se) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span> </span>
<span id="cb23-23"><a href="#cb23-23" tabindex="-1"></a>    <span class="fu">filter</span>(sample_alias <span class="sc">%in%</span> comp) <span class="sc">%&gt;%</span> </span>
<span id="cb23-24"><a href="#cb23-24" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(condition)</span>
<span id="cb23-25"><a href="#cb23-25" tabindex="-1"></a>}</span>
<span id="cb23-26"><a href="#cb23-26" tabindex="-1"></a><span class="fu">names</span>(ann_list) <span class="ot">&lt;-</span> <span class="fu">names</span>(clustering_groups)</span>
<span id="cb23-27"><a href="#cb23-27" tabindex="-1"></a></span>
<span id="cb23-28"><a href="#cb23-28" tabindex="-1"></a><span class="co"># Define annotation colors</span></span>
<span id="cb23-29"><a href="#cb23-29" tabindex="-1"></a>ann_colors <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb23-30"><a href="#cb23-30" tabindex="-1"></a>  <span class="at">condition =</span> condition_palette</span>
<span id="cb23-31"><a href="#cb23-31" tabindex="-1"></a>  )</span></code></pre></div>
<p>Generate a list of HVG for each group</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a>hvg_list <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">comp =</span> clustering_groups) <span class="sc">%do%</span> {</span>
<span id="cb24-2"><a href="#cb24-2" tabindex="-1"></a>  use_se <span class="ot">&lt;-</span> se[use_features, comp]</span>
<span id="cb24-3"><a href="#cb24-3" tabindex="-1"></a>  <span class="co"># Variance modelling</span></span>
<span id="cb24-4"><a href="#cb24-4" tabindex="-1"></a>  fit_res <span class="ot">&lt;-</span> <span class="fu">modelGeneVar</span>(use_se)</span>
<span id="cb24-5"><a href="#cb24-5" tabindex="-1"></a>  fit_md <span class="ot">&lt;-</span> <span class="fu">metadata</span>(fit_res)</span>
<span id="cb24-6"><a href="#cb24-6" tabindex="-1"></a>  <span class="co"># Define highly variable genes</span></span>
<span id="cb24-7"><a href="#cb24-7" tabindex="-1"></a>  hvg_genes <span class="ot">&lt;-</span> <span class="fu">getTopHVGs</span>(fit_res, <span class="at">prop =</span> <span class="fl">0.1</span>)</span>
<span id="cb24-8"><a href="#cb24-8" tabindex="-1"></a>  hvg_genes</span>
<span id="cb24-9"><a href="#cb24-9" tabindex="-1"></a>}</span>
<span id="cb24-10"><a href="#cb24-10" tabindex="-1"></a><span class="fu">names</span>(hvg_list) <span class="ot">&lt;-</span> <span class="fu">names</span>(clustering_groups)</span></code></pre></div>
<p>Generate heatmaps of sample distances</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a>comp <span class="ot">&lt;-</span> <span class="fu">names</span>(clustering_groups)[<span class="dv">1</span>]</span>
<span id="cb25-2"><a href="#cb25-2" tabindex="-1"></a>clustering_heatmaps <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">comp =</span> <span class="fu">names</span>(clustering_groups)) <span class="sc">%do%</span> {</span>
<span id="cb25-3"><a href="#cb25-3" tabindex="-1"></a>  use_samples <span class="ot">&lt;-</span> clustering_groups[[comp]]</span>
<span id="cb25-4"><a href="#cb25-4" tabindex="-1"></a>  ann_df <span class="ot">&lt;-</span> ann_list[[comp]]</span>
<span id="cb25-5"><a href="#cb25-5" tabindex="-1"></a>  hvg_genes <span class="ot">&lt;-</span> hvg_list[[comp]]</span>
<span id="cb25-6"><a href="#cb25-6" tabindex="-1"></a>  use_donors <span class="ot">&lt;-</span> use_se<span class="sc">$</span>donor <span class="sc">%&gt;%</span> unique</span>
<span id="cb25-7"><a href="#cb25-7" tabindex="-1"></a>  use_se <span class="ot">&lt;-</span> se[hvg_genes, use_samples]</span>
<span id="cb25-8"><a href="#cb25-8" tabindex="-1"></a></span>
<span id="cb25-9"><a href="#cb25-9" tabindex="-1"></a>  <span class="co"># Calculate Sample distances</span></span>
<span id="cb25-10"><a href="#cb25-10" tabindex="-1"></a>  sample_dist <span class="ot">&lt;-</span> <span class="fu">dist</span>(<span class="fu">t</span>(<span class="fu">assay</span>(use_se, <span class="st">&#39;logcounts&#39;</span>)))</span>
<span id="cb25-11"><a href="#cb25-11" tabindex="-1"></a>  sample_dist_hvg_matrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(sample_dist)</span>
<span id="cb25-12"><a href="#cb25-12" tabindex="-1"></a>  </span>
<span id="cb25-13"><a href="#cb25-13" tabindex="-1"></a>  <span class="fu">colnames</span>(sample_dist_hvg_matrix) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(sample_dist_hvg_matrix) <span class="sc">%&gt;%</span> </span>
<span id="cb25-14"><a href="#cb25-14" tabindex="-1"></a>    <span class="fu">gsub</span>(<span class="st">&#39;MVT1_|4T1_|Py2T_&#39;</span>, <span class="st">&#39;&#39;</span>, .) <span class="sc">%&gt;%</span> </span>
<span id="cb25-15"><a href="#cb25-15" tabindex="-1"></a>    <span class="fu">gsub</span>(<span class="st">&#39;MVT|4T1|PyT&#39;</span>, <span class="st">&#39;&#39;</span>, .) <span class="sc">%&gt;%</span> </span>
<span id="cb25-16"><a href="#cb25-16" tabindex="-1"></a>    <span class="fu">gsub</span>(<span class="st">&#39;Cas9BROAD&#39;</span>, <span class="st">&#39;Integrating Cas9 LV&#39;</span>, .) <span class="sc">%&gt;%</span> </span>
<span id="cb25-17"><a href="#cb25-17" tabindex="-1"></a>    <span class="fu">gsub</span>(<span class="st">&#39;CD8&#39;</span>, <span class="st">&#39;Apo Cas9 TrCD8(+)&#39;</span>, .) <span class="sc">%&gt;%</span> </span>
<span id="cb25-18"><a href="#cb25-18" tabindex="-1"></a>    <span class="fu">gsub</span>(<span class="st">&#39;KO&#39;</span>, <span class="st">&#39;Apo-Cas9 TrCD8(−)&#39;</span>, .) <span class="sc">%&gt;%</span> </span>
<span id="cb25-19"><a href="#cb25-19" tabindex="-1"></a>    <span class="fu">gsub</span>(<span class="st">&#39;UT&#39;</span>, <span class="st">&#39;Untransduced&#39;</span>, .) <span class="sc">%&gt;%</span> </span>
<span id="cb25-20"><a href="#cb25-20" tabindex="-1"></a>    <span class="fu">gsub</span>(<span class="st">&#39;_FVB&#39;</span>, <span class="st">&#39; &#39;</span>, .) <span class="sc">%&gt;%</span> </span>
<span id="cb25-21"><a href="#cb25-21" tabindex="-1"></a>    <span class="fu">gsub</span>(<span class="st">&#39;_BLB&#39;</span>, <span class="st">&#39; &#39;</span>, .) <span class="sc">%&gt;%</span> </span>
<span id="cb25-22"><a href="#cb25-22" tabindex="-1"></a>    <span class="fu">gsub</span>(<span class="st">&#39;_rep&#39;</span>, <span class="st">&#39; &#39;</span>, .)</span>
<span id="cb25-23"><a href="#cb25-23" tabindex="-1"></a>  </span>
<span id="cb25-24"><a href="#cb25-24" tabindex="-1"></a>  ht <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(</span>
<span id="cb25-25"><a href="#cb25-25" tabindex="-1"></a>    sample_dist_hvg_matrix,</span>
<span id="cb25-26"><a href="#cb25-26" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">&quot;Euclidean</span><span class="sc">\n</span><span class="st">distance&quot;</span>,</span>
<span id="cb25-27"><a href="#cb25-27" tabindex="-1"></a>    <span class="at">column_title =</span> comp,</span>
<span id="cb25-28"><a href="#cb25-28" tabindex="-1"></a>    <span class="at">col =</span> ph_colors,</span>
<span id="cb25-29"><a href="#cb25-29" tabindex="-1"></a>    <span class="at">show_row_names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb25-30"><a href="#cb25-30" tabindex="-1"></a>    <span class="at">show_column_dend =</span> <span class="cn">FALSE</span>,</span>
<span id="cb25-31"><a href="#cb25-31" tabindex="-1"></a>    <span class="at">row_dend_width =</span> <span class="fu">unit</span>(<span class="fl">0.2</span>, <span class="st">&quot;inches&quot;</span>),</span>
<span id="cb25-32"><a href="#cb25-32" tabindex="-1"></a>    <span class="at">row_dend_gp =</span> <span class="fu">gpar</span>(<span class="at">lwd =</span>  one_pt<span class="sc">/</span><span class="dv">2</span>),</span>
<span id="cb25-33"><a href="#cb25-33" tabindex="-1"></a>    <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">2</span>),</span>
<span id="cb25-34"><a href="#cb25-34" tabindex="-1"></a>    <span class="at">column_title_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">6</span>),</span>
<span id="cb25-35"><a href="#cb25-35" tabindex="-1"></a>    <span class="at">heatmap_legend_param =</span> <span class="fu">list</span>(</span>
<span id="cb25-36"><a href="#cb25-36" tabindex="-1"></a>      <span class="at">direction =</span> <span class="st">&quot;horizontal&quot;</span>,</span>
<span id="cb25-37"><a href="#cb25-37" tabindex="-1"></a>      <span class="at">title_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">3</span>, <span class="at">fontface =</span> <span class="st">&quot;bold&quot;</span>),</span>
<span id="cb25-38"><a href="#cb25-38" tabindex="-1"></a>      <span class="at">labels_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">3</span>),</span>
<span id="cb25-39"><a href="#cb25-39" tabindex="-1"></a>      <span class="at">legend_width =</span> <span class="fu">unit</span>(<span class="fl">0.5</span>, <span class="st">&quot;inches&quot;</span>),</span>
<span id="cb25-40"><a href="#cb25-40" tabindex="-1"></a>      <span class="at">grid_height =</span> <span class="fu">unit</span>(<span class="fl">0.08</span>, <span class="st">&quot;inches&quot;</span>),</span>
<span id="cb25-41"><a href="#cb25-41" tabindex="-1"></a>      <span class="at">at =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">25</span>, <span class="dv">50</span>, <span class="dv">75</span>, <span class="dv">100</span>)</span>
<span id="cb25-42"><a href="#cb25-42" tabindex="-1"></a>      )</span>
<span id="cb25-43"><a href="#cb25-43" tabindex="-1"></a>  )</span>
<span id="cb25-44"><a href="#cb25-44" tabindex="-1"></a>}</span>
<span id="cb25-45"><a href="#cb25-45" tabindex="-1"></a><span class="fu">names</span>(clustering_heatmaps) <span class="ot">&lt;-</span> <span class="fu">names</span>(clustering_groups)</span></code></pre></div>
<div id="figure-3h-sample-distances-in-mvt1-models"
class="section level3">
<h3>Figure 3H: Sample distances in MVT1 models</h3>
<p>Unsupervised hierarchical clustering of highly variable genes (HVGs)
from MVT1 transcriptomes from FVB/N (n=3) based on Cas9 delivery TrCD8a
status.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a><span class="fu">draw</span>(clustering_heatmaps[[<span class="st">&#39;MVT1&#39;</span>]],  <span class="at">heatmap_legend_side =</span> <span class="st">&quot;bottom&quot;</span>)</span></code></pre></div>
<p><img src="figure/rnaseq-tumor-bulk.Rmd/sample-clustering-heatmap-mvt1-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-sample-clustering-heatmap-mvt1-1">
Past versions of sample-clustering-heatmap-mvt1-1.png
</button>
</p>
<div id="fig-sample-clustering-heatmap-mvt1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/8632845b1c6b024330836a1a798537363e31980e/docs/figure/rnaseq-tumor-bulk.Rmd/sample-clustering-heatmap-mvt1-1.png" target="_blank">8632845</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-07-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span></code></pre></div>
</div>
<div id="supplementary-figure-3e-left-sample-distances-in-py2t-model"
class="section level3">
<h3>Supplementary figure 3E left: Sample distances in Py2T model</h3>
<p>Heatmaps showing unsupervised hierarchical clustering of highly
variable genes (HVGs) from primary tumor transcriptomes obtained from
syngeneic hosts, following transplantation of cells transduced with
integrating Cas9-Thy1 (LV Cas9) or transduced with hara-kiri (HK) vector
and transfected with Apo-Cas9, then MACS-sorted as TrCD8a-positive and
TrCD8a-negative cells; n=3 each; 4T1 tumors from BALB/c mice and Py2T
tumors from FVB/N mice are shown.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a><span class="fu">draw</span>(clustering_heatmaps[[<span class="st">&#39;Py2T&#39;</span>]],  <span class="at">heatmap_legend_side =</span> <span class="st">&quot;bottom&quot;</span>)</span></code></pre></div>
<p><img src="figure/rnaseq-tumor-bulk.Rmd/sample-clustering-heatmap-py2t-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-sample-clustering-heatmap-py2t-1">
Past versions of sample-clustering-heatmap-py2t-1.png
</button>
</p>
<div id="fig-sample-clustering-heatmap-py2t-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/8632845b1c6b024330836a1a798537363e31980e/docs/figure/rnaseq-tumor-bulk.Rmd/sample-clustering-heatmap-py2t-1.png" target="_blank">8632845</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-07-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span></code></pre></div>
</div>
<div id="supplementary-figure-3e-right-sample-distances-in-4t1-model"
class="section level3">
<h3>Supplementary figure 3E right: Sample distances in 4T1 model</h3>
<p>Heatmaps showing unsupervised hierarchical clustering of highly
variable genes (HVGs) from primary tumor transcriptomes obtained from
syngeneic hosts, following transplantation of cells transduced with
integrating Cas9-Thy1 (LV Cas9) or transduced with hara-kiri (HK) vector
and transfected with Apo-Cas9, then MACS-sorted as TrCD8a-positive and
TrCD8a-negative cells; n=3 each; 4T1 tumors from BALB/c mice and Py2T
tumors from FVB/N mice are shown.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a><span class="fu">draw</span>(clustering_heatmaps[[<span class="st">&#39;4T1&#39;</span>]],  <span class="at">heatmap_legend_side =</span> <span class="st">&quot;bottom&quot;</span>)</span></code></pre></div>
<p><img src="figure/rnaseq-tumor-bulk.Rmd/sample-clustering-heatmap-4t1-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-sample-clustering-heatmap-4t1-1">
Past versions of sample-clustering-heatmap-4t1-1.png
</button>
</p>
<div id="fig-sample-clustering-heatmap-4t1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/8632845b1c6b024330836a1a798537363e31980e/docs/figure/rnaseq-tumor-bulk.Rmd/sample-clustering-heatmap-4t1-1.png" target="_blank">8632845</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-07-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="qrt-pcr-validation" class="section level2">
<h2>qRT-PCR validation</h2>
<p>Plots configuration</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" tabindex="-1"></a>ph_colors <span class="ot">&lt;-</span><span class="fu">colorRamp2</span>(</span>
<span id="cb32-2"><a href="#cb32-2" tabindex="-1"></a>  <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">2</span>, <span class="dv">5</span>, <span class="at">length.out =</span> <span class="dv">7</span>), </span>
<span id="cb32-3"><a href="#cb32-3" tabindex="-1"></a>  <span class="fu">rev</span>(<span class="fu">brewer.pal</span>(<span class="at">n =</span> <span class="dv">7</span>, <span class="at">name =</span> <span class="st">&quot;Purples&quot;</span>))</span>
<span id="cb32-4"><a href="#cb32-4" tabindex="-1"></a>  )</span></code></pre></div>
<p>Generate heatmap for TrCD8a-positive clones</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a>mat <span class="ot">&lt;-</span> <span class="fu">matrix</span>(</span>
<span id="cb33-2"><a href="#cb33-2" tabindex="-1"></a>  <span class="fu">c</span>(<span class="fl">1.9</span>, <span class="fl">2.02</span>, <span class="fl">4.37</span>, <span class="fl">2.79</span>, <span class="fl">1.48</span>, <span class="fl">0.74</span>, <span class="fl">4.07</span>, <span class="fl">4.33</span>, <span class="fl">4.72</span>, <span class="fl">2.88</span>, <span class="fl">1.33</span>, <span class="dv">2</span>, <span class="fl">3.39</span>, <span class="fl">3.63</span>, <span class="fl">3.61</span>, <span class="fl">0.72</span>),</span>
<span id="cb33-3"><a href="#cb33-3" tabindex="-1"></a>  <span class="at">ncol =</span> <span class="dv">4</span>,</span>
<span id="cb33-4"><a href="#cb33-4" tabindex="-1"></a>  <span class="at">byrow =</span> <span class="cn">TRUE</span></span>
<span id="cb33-5"><a href="#cb33-5" tabindex="-1"></a>)</span>
<span id="cb33-6"><a href="#cb33-6" tabindex="-1"></a></span>
<span id="cb33-7"><a href="#cb33-7" tabindex="-1"></a><span class="fu">rownames</span>(mat) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;sgCcl2&#39;</span>, <span class="st">&#39;sgFgf7&#39;</span>, <span class="st">&#39;sgEgfr&#39;</span>, <span class="st">&#39;sgFgfr3&#39;</span>)</span>
<span id="cb33-8"><a href="#cb33-8" tabindex="-1"></a><span class="fu">colnames</span>(mat) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;Fgf7&#39;</span>, <span class="st">&#39;Ccl2&#39;</span>, <span class="st">&#39;Egfr&#39;</span>, <span class="st">&#39;Fgfr3&#39;</span>)</span>
<span id="cb33-9"><a href="#cb33-9" tabindex="-1"></a></span>
<span id="cb33-10"><a href="#cb33-10" tabindex="-1"></a>mat <span class="ot">&lt;-</span> mat[<span class="fu">c</span>(<span class="st">&#39;sgFgf7&#39;</span>, <span class="st">&#39;sgCcl2&#39;</span>, <span class="st">&#39;sgEgfr&#39;</span>, <span class="st">&#39;sgFgfr3&#39;</span>),]</span>
<span id="cb33-11"><a href="#cb33-11" tabindex="-1"></a></span>
<span id="cb33-12"><a href="#cb33-12" tabindex="-1"></a>ht_TrCD8a_pos <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(</span>
<span id="cb33-13"><a href="#cb33-13" tabindex="-1"></a>  mat,</span>
<span id="cb33-14"><a href="#cb33-14" tabindex="-1"></a>  <span class="at">column_title =</span> <span class="st">&#39;TrCD8(+)</span><span class="sc">\n</span><span class="st">64 hpt&#39;</span>,</span>
<span id="cb33-15"><a href="#cb33-15" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">&quot;Gene expression&quot;</span>,</span>
<span id="cb33-16"><a href="#cb33-16" tabindex="-1"></a>  <span class="at">col =</span> ph_colors,</span>
<span id="cb33-17"><a href="#cb33-17" tabindex="-1"></a>  <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>,</span>
<span id="cb33-18"><a href="#cb33-18" tabindex="-1"></a>  <span class="at">cluster_columns =</span> <span class="cn">FALSE</span>,</span>
<span id="cb33-19"><a href="#cb33-19" tabindex="-1"></a>  <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">2</span>),</span>
<span id="cb33-20"><a href="#cb33-20" tabindex="-1"></a>  <span class="at">row_names_side =</span> <span class="st">&#39;left&#39;</span>,</span>
<span id="cb33-21"><a href="#cb33-21" tabindex="-1"></a>  <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">2</span>),</span>
<span id="cb33-22"><a href="#cb33-22" tabindex="-1"></a>  <span class="at">column_names_rot =</span> <span class="dv">45</span>,</span>
<span id="cb33-23"><a href="#cb33-23" tabindex="-1"></a>  <span class="at">column_title_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">3</span>),</span>
<span id="cb33-24"><a href="#cb33-24" tabindex="-1"></a>  <span class="at">heatmap_legend_param =</span> <span class="fu">list</span>(</span>
<span id="cb33-25"><a href="#cb33-25" tabindex="-1"></a>    <span class="at">direction =</span> <span class="st">&quot;horizontal&quot;</span>,</span>
<span id="cb33-26"><a href="#cb33-26" tabindex="-1"></a>    <span class="at">title_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">2</span>, <span class="at">fontface =</span> <span class="st">&quot;bold&quot;</span>),</span>
<span id="cb33-27"><a href="#cb33-27" tabindex="-1"></a>    <span class="at">labels_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">2</span>),</span>
<span id="cb33-28"><a href="#cb33-28" tabindex="-1"></a>    <span class="at">legend_width =</span> <span class="fu">unit</span>(<span class="fl">0.4</span>, <span class="st">&quot;inches&quot;</span>),</span>
<span id="cb33-29"><a href="#cb33-29" tabindex="-1"></a>    <span class="at">grid_height =</span> <span class="fu">unit</span>(<span class="fl">0.03</span>, <span class="st">&quot;inches&quot;</span>)</span>
<span id="cb33-30"><a href="#cb33-30" tabindex="-1"></a>  )</span>
<span id="cb33-31"><a href="#cb33-31" tabindex="-1"></a>)</span></code></pre></div>
<p>Generate heatmap for TrCD8a-negative clones</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" tabindex="-1"></a>mat <span class="ot">&lt;-</span> <span class="fu">matrix</span>(</span>
<span id="cb34-2"><a href="#cb34-2" tabindex="-1"></a>  <span class="fu">c</span>(<span class="sc">-</span><span class="fl">1.08</span>, <span class="fl">2.07</span>, <span class="fl">4.93</span>, <span class="fl">3.01</span>, <span class="fl">1.81</span>, <span class="sc">-</span><span class="fl">1.65</span>, <span class="fl">2.77</span>, <span class="fl">1.02</span>, <span class="fl">1.68</span>, <span class="fl">0.06</span>, <span class="sc">-</span><span class="fl">1.67</span>, <span class="fl">0.87</span>, <span class="fl">3.14</span>, <span class="fl">4.17</span>, <span class="fl">3.21</span>, <span class="sc">-</span><span class="fl">1.58</span>),</span>
<span id="cb34-3"><a href="#cb34-3" tabindex="-1"></a>  <span class="at">ncol =</span> <span class="dv">4</span>,</span>
<span id="cb34-4"><a href="#cb34-4" tabindex="-1"></a>  <span class="at">byrow =</span> <span class="cn">TRUE</span></span>
<span id="cb34-5"><a href="#cb34-5" tabindex="-1"></a>)</span>
<span id="cb34-6"><a href="#cb34-6" tabindex="-1"></a><span class="fu">rownames</span>(mat) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;sgFgf7&#39;</span>, <span class="st">&#39;sgCcl2&#39;</span>, <span class="st">&#39;sgEgfr&#39;</span>, <span class="st">&#39;sgFgfr3&#39;</span>)</span>
<span id="cb34-7"><a href="#cb34-7" tabindex="-1"></a><span class="fu">colnames</span>(mat) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;Fgf7&#39;</span>, <span class="st">&#39;Ccl2&#39;</span>, <span class="st">&#39;Egfr&#39;</span>, <span class="st">&#39;Fgfr3&#39;</span>)</span>
<span id="cb34-8"><a href="#cb34-8" tabindex="-1"></a></span>
<span id="cb34-9"><a href="#cb34-9" tabindex="-1"></a>mat <span class="ot">&lt;-</span> mat[<span class="fu">c</span>(<span class="st">&#39;sgFgf7&#39;</span>, <span class="st">&#39;sgCcl2&#39;</span>, <span class="st">&#39;sgEgfr&#39;</span>, <span class="st">&#39;sgFgfr3&#39;</span>),]</span>
<span id="cb34-10"><a href="#cb34-10" tabindex="-1"></a></span>
<span id="cb34-11"><a href="#cb34-11" tabindex="-1"></a>ht_TrCD8a_neg <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(</span>
<span id="cb34-12"><a href="#cb34-12" tabindex="-1"></a>  mat,</span>
<span id="cb34-13"><a href="#cb34-13" tabindex="-1"></a>  <span class="at">column_title =</span> <span class="st">&#39;TrCD8(-)</span><span class="sc">\n</span><span class="st">64 hpt&#39;</span>,</span>
<span id="cb34-14"><a href="#cb34-14" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">&quot;Gene expression&quot;</span>,</span>
<span id="cb34-15"><a href="#cb34-15" tabindex="-1"></a>  <span class="at">col =</span> ph_colors,</span>
<span id="cb34-16"><a href="#cb34-16" tabindex="-1"></a>  <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>,</span>
<span id="cb34-17"><a href="#cb34-17" tabindex="-1"></a>  <span class="at">cluster_columns =</span> <span class="cn">FALSE</span>,</span>
<span id="cb34-18"><a href="#cb34-18" tabindex="-1"></a>  <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">2</span>),</span>
<span id="cb34-19"><a href="#cb34-19" tabindex="-1"></a>  <span class="at">row_names_side =</span> <span class="st">&#39;left&#39;</span>,</span>
<span id="cb34-20"><a href="#cb34-20" tabindex="-1"></a>  <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">2</span>),</span>
<span id="cb34-21"><a href="#cb34-21" tabindex="-1"></a>  <span class="co"># column_names_rot = 0,</span></span>
<span id="cb34-22"><a href="#cb34-22" tabindex="-1"></a>  <span class="at">column_names_rot =</span> <span class="dv">45</span>,</span>
<span id="cb34-23"><a href="#cb34-23" tabindex="-1"></a>  <span class="at">column_title_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">3</span>),</span>
<span id="cb34-24"><a href="#cb34-24" tabindex="-1"></a>  <span class="at">heatmap_legend_param =</span> <span class="fu">list</span>(</span>
<span id="cb34-25"><a href="#cb34-25" tabindex="-1"></a>    <span class="at">direction =</span> <span class="st">&quot;horizontal&quot;</span>,</span>
<span id="cb34-26"><a href="#cb34-26" tabindex="-1"></a>    <span class="at">title_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">2</span>, <span class="at">fontface =</span> <span class="st">&quot;bold&quot;</span>),</span>
<span id="cb34-27"><a href="#cb34-27" tabindex="-1"></a>    <span class="at">labels_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">2</span>),</span>
<span id="cb34-28"><a href="#cb34-28" tabindex="-1"></a>    <span class="at">legend_width =</span> <span class="fu">unit</span>(<span class="fl">0.4</span>, <span class="st">&quot;inches&quot;</span>),</span>
<span id="cb34-29"><a href="#cb34-29" tabindex="-1"></a>    <span class="at">grid_height =</span> <span class="fu">unit</span>(<span class="fl">0.03</span>, <span class="st">&quot;inches&quot;</span>)</span>
<span id="cb34-30"><a href="#cb34-30" tabindex="-1"></a>  )</span>
<span id="cb34-31"><a href="#cb34-31" tabindex="-1"></a>)</span></code></pre></div>
<div id="figure-3d-qrt-pcr-in-trcd8a-and-trcd8a-"
class="section level3">
<h3>Figure 3D : qRT-PCR in TrCD8a+ and TrCD8a-</h3>
<p>qRT-PCR (2-dCt) heatmaps in MACS-sorted TrCD8a-positive and
TrCD8a-negative single-cell clones after sgRNA delivery.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" tabindex="-1"></a>ht_list <span class="ot">&lt;-</span>  ht_TrCD8a_pos <span class="sc">+</span> ht_TrCD8a_neg</span>
<span id="cb35-2"><a href="#cb35-2" tabindex="-1"></a><span class="fu">draw</span>(ht_list, <span class="at">heatmap_legend_side =</span> <span class="st">&quot;bottom&quot;</span>, <span class="at">merge_legend =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="figure/rnaseq-tumor-bulk.Rmd/qrt-pcr-heatmap-combined2-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-qrt-pcr-heatmap-combined2-1">
Past versions of qrt-pcr-heatmap-combined2-1.png
</button>
</p>
<div id="fig-qrt-pcr-heatmap-combined2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/8632845b1c6b024330836a1a798537363e31980e/docs/figure/rnaseq-tumor-bulk.Rmd/qrt-pcr-heatmap-combined2-1.png" target="_blank">8632845</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-07-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<p>R version 4.4.3 (2025-02-28) Platform: aarch64-apple-darwin20 Running
under: macOS 26.0.1</p>
<p>Matrix products: default BLAS:
/Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib
LAPACK:
/Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;
LAPACK version 3.12.0</p>
<p>locale: [1]
en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</p>
<p>time zone: Europe/Zurich tzcode source: internal</p>
<p>attached base packages: [1] grid stats4 stats graphics grDevices
utils datasets [8] methods base</p>
<p>other attached packages: [1] cowplot_1.2.0 org.Mm.eg.db_3.20.0<br />
[3] AnnotationDbi_1.68.0 RColorBrewer_1.1-3<br />
[5] circlize_0.4.16 ComplexHeatmap_2.22.0<br />
[7] scran_1.34.0 scuttle_1.16.0<br />
[9] SingleCellExperiment_1.28.1 edgeR_4.4.2<br />
[11] limma_3.62.2 SummarizedExperiment_1.36.0 [13] Biobase_2.66.0
GenomicRanges_1.58.0<br />
[15] GenomeInfoDb_1.42.3 IRanges_2.40.1<br />
[17] S4Vectors_0.44.0 BiocGenerics_0.52.0<br />
[19] MatrixGenerics_1.18.1 matrixStats_1.5.0<br />
[21] kableExtra_1.4.0 DT_0.34.0<br />
[23] magrittr_2.0.3 foreach_1.5.2<br />
[25] knitr_1.50 lubridate_1.9.4<br />
[27] forcats_1.0.0 stringr_1.5.1<br />
[29] dplyr_1.1.4 purrr_1.1.0<br />
[31] readr_2.1.5 tidyr_1.3.1<br />
[33] tibble_3.3.0 ggplot2_3.5.2<br />
[35] tidyverse_2.0.0 workflowr_1.7.1</p>
<p>loaded via a namespace (and not attached): [1] rstudioapi_0.17.1
jsonlite_2.0.0 shape_1.4.6.1<br />
[4] magick_2.8.7 farver_2.1.2 rmarkdown_2.29<br />
[7] GlobalOptions_0.1.2 fs_1.6.6 zlibbioc_1.52.0<br />
[10] vctrs_0.6.5 Cairo_1.6-2 memoise_2.0.1<br />
[13] htmltools_0.5.8.1 S4Arrays_1.6.0 BiocNeighbors_2.0.1<br />
[16] SparseArray_1.6.2 sass_0.4.10 bslib_0.9.0<br />
[19] htmlwidgets_1.6.4 cachem_1.1.0 whisker_0.4.1<br />
[22] igraph_2.1.4 lifecycle_1.0.4 iterators_1.0.14<br />
[25] pkgconfig_2.0.3 rsvd_1.0.5 Matrix_1.7-3<br />
[28] R6_2.6.1 fastmap_1.2.0 GenomeInfoDbData_1.2.13 [31] clue_0.3-66
digest_0.6.37 colorspace_2.1-1<br />
[34] ps_1.9.1 rprojroot_2.1.0 dqrng_0.4.1<br />
[37] irlba_2.3.5.1 textshaping_1.0.1 RSQLite_2.4.3<br />
[40] beachmat_2.22.0 labeling_0.4.3 timechange_0.3.0<br />
[43] httr_1.4.7 abind_1.4-8 compiler_4.4.3<br />
[46] bit64_4.6.0-1 withr_3.0.2 doParallel_1.0.17<br />
[49] BiocParallel_1.40.2 DBI_1.2.3 DelayedArray_0.32.0<br />
[52] rjson_0.2.23 bluster_1.16.0 tools_4.4.3<br />
[55] httpuv_1.6.16 glue_1.8.0 callr_3.7.6<br />
[58] promises_1.3.3 getPass_0.2-4 cluster_2.1.8.1<br />
[61] generics_0.1.4 gtable_0.3.6 tzdb_0.5.0<br />
[64] hms_1.1.3 BiocSingular_1.22.0 ScaledMatrix_1.14.0<br />
[67] metapod_1.14.0 xml2_1.3.8 XVector_0.46.0<br />
[70] pillar_1.11.0 later_1.4.4 lattice_0.22-7<br />
[73] bit_4.6.0 tidyselect_1.2.1 locfit_1.5-9.12<br />
[76] Biostrings_2.74.1 git2r_0.36.2 svglite_2.2.1<br />
[79] xfun_0.53 statmod_1.5.0 stringi_1.8.7<br />
[82] UCSC.utils_1.2.0 yaml_2.3.10 evaluate_1.0.5<br />
[85] codetools_0.2-20 cli_3.6.5 systemfonts_1.2.3<br />
[88] processx_3.8.6 jquerylib_0.1.4 dichromat_2.0-0.1<br />
[91] Rcpp_1.1.0 png_0.1-8 parallel_4.4.3<br />
[94] blob_1.2.4 viridisLite_0.4.2 scales_1.4.0<br />
[97] crayon_1.5.3 GetoptLong_1.0.5 rlang_1.1.6<br />
[100] KEGGREST_1.46.0</p>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
