<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Francesc Castro-Giner" />

<meta name="date" content="2025-07-01" />

<title>Survival analysis in TCGA</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">StealTHY</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/TheAcetoLab/saini-stealTHY">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Survival analysis in TCGA</h1>
<h3 class="subtitle">Amhr2 KO signatures</h3>
<h4 class="author">Francesc Castro-Giner</h4>
<h4 class="date">July 01, 2025</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-07-01
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>saini-stealTHY/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240517code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240517)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240517code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240517)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomTheAcetoLabsainistealTHYtree2d1b5a6827bac6db4bf98f99c9267fd81f37ecc9targetblank2d1b5a6a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/tree/2d1b5a6827bac6db4bf98f99c9267fd81f37ecc9" target="_blank">2d1b5a6</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomTheAcetoLabsainistealTHYtree2d1b5a6827bac6db4bf98f99c9267fd81f37ecc9targetblank2d1b5a6a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/TheAcetoLab/saini-stealTHY/tree/2d1b5a6827bac6db4bf98f99c9267fd81f37ecc9" target="_blank">2d1b5a6</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    code/raw_data_processing/rnaseq/p27851_o32062/pipelines/
    Ignored:    configuration/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/crispr/
    Ignored:    data/resources/
    Ignored:    data/rnaseq/
    Ignored:    output/.DS_Store
    Ignored:    output/clinical/
    Ignored:    output/crispr/
    Ignored:    output/rnaseq/

Untracked files:
    Untracked:  analysis/crispr-hsapiens_2180_sgRNA_r1.Rmd
    Untracked:  analysis/crispr-mm_2215_sgRNA-CMT167-LLC1-CT26.Rmd
    Untracked:  analysis/crispr-mm_2215_sgRNA-clonality_r1.Rmd
    Untracked:  analysis/crispr-mm_2215_sgRNA-r2oD_r1.Rmd
    Untracked:  analysis/crispr-muller-mm_2215_sgRNA.Rmd
    Untracked:  analysis/rnaseq-amhr2_ko-deg.Rmd
    Untracked:  analysis/templates/
    Untracked:  code/R-functions/subchunkify.R

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/crispr-hsapiens_2180_sgRNA.Rmd
    Modified:   analysis/crispr-mm_2215_sgRNA-StealTHY.Rmd
    Modified:   analysis/crispr-mm_2215_sgRNA-clonality.Rmd
    Modified:   analysis/crispr-mm_2215_sgRNA-r2oC.Rmd
    Modified:   analysis/rnaseq-tumor-bulk.Rmd
    Modified:   analysis/rnaseq-tumor-facs-cancer.Rmd
    Modified:   analysis/rnaseq-tumor-facs-immune.Rmd
    Modified:   code/R-functions/gse_report.r
    Modified:   configuration/rmarkdown/ggplot_theme.R
    Modified:   update_workflowr.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/tcga_survival_amhr2_ko_signatures.Rmd</code>) and HTML
(<code>docs/tcga_survival_amhr2_ko_signatures.html</code>) files. If
you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/saini-stealTHY/blob/2d1b5a6827bac6db4bf98f99c9267fd81f37ecc9/analysis/tcga_survival_amhr2_ko_signatures.Rmd" target="_blank">2d1b5a6</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-07-01
</td>
<td>
add tcga analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries-additional-functions-and-data"
class="section level2" number="1">
<h2><span class="header-section-number">1</span> Load libraries,
additional functions and data</h2>
<p>Setup environment</p>
<pre class="r"><code>knitr::opts_chunk$set(results=&#39;asis&#39;, echo=TRUE, message=FALSE, warning=FALSE, error=FALSE, fig.align = &#39;center&#39;, fig.width = 3.5, fig.asp = 0.618, dpi = 600, dev = c(&quot;png&quot;, &quot;pdf&quot;), engine.opts = list(bash = &quot;-l&quot;))

options(stringsAsFactors = FALSE)

use_seed &lt;- 1100101

set.seed(use_seed)

dir.create(params$output_dir, recursive = TRUE, showWarnings = FALSE)</code></pre>
<p>Load packages</p>
<pre class="r"><code>library(tidyverse)
library(knitr)
# library(showtext)
library(foreach)
library(SummarizedExperiment)
library(survival)
library(gtsummary)
library(ggsurvfit)
library(survminer)
library(meta)
library(magrittr)
# library(DT)
# library(colorblindr)
# library(ggdendro)
# library(RColorBrewer)
# library(circlize)
# library(Hmisc)
# library(ComplexHeatmap)
# library(ineq)
library(kableExtra)
library(openxlsx)
# library(magrittr)
# library(ggrepel)
# library(ggpubr)
# library(ggbeeswarm)
# library(ggridges)
# library(openxlsx)
# library(MAGeCKFlute)
library(GSVA)</code></pre>
<p>Load ggplot theme</p>
<pre class="r"><code>source(&quot;./configuration/rmarkdown/ggplot_theme.R&quot;)</code></pre>
<p>Clean files generated in previous runs</p>
<pre class="r"><code>rmd_file &lt;- current_input()
if(!is.null(rmd_file)) {
  figures_dir &lt;- file.path(&#39;./docs/figure&#39;,rmd_file)
  if(dir.exists(figures_dir)) {
    unlink(file.path(figures_dir, &quot;*&quot;))
  }
}</code></pre>
</div>
<div id="generate-signatures" class="section level2" number="2">
<h2><span class="header-section-number">2</span> Generate
signatures</h2>
<p>Load data</p>
<pre class="r"><code># Load DGE results
dge_list &lt;- readRDS(&#39;./output/rnaseq/amhr2_ko/dge-edgeR_QLF.rds&#39;)

# Select comparisons
use_comp &lt;- c(&quot;i_AMHR2_KO-o-i_ctrl_sgRNA&quot;, &quot;c_AMHR2_KO-o-c_AMHR2_ctrl&quot;, &quot;AMHR2_KO-o-ctrl&quot;)
dge_list &lt;- dge_list[use_comp]</code></pre>
<p>Extract top-100 genes for each comparison and direction (up, down)
based on FDR and FC</p>
<pre class="r"><code>i &lt;- dge_list[[1]]
dge_up_fdr &lt;- foreach(i = dge_list) %do% {
  i$results %&gt;% 
    filter(FDR &lt; 0.001) %&gt;%
    filter(logFC &gt; 1) %&gt;% 
    arrange(FDR) %&gt;% 
    head(100) %&gt;% 
    pull(gene_name)
}
names(dge_up_fdr) &lt;- names(dge_list) %&gt;% gsub(&quot;.o.*&quot;, &quot;&quot;, .) %&gt;% paste0(&quot;up_fdr_&quot;, .)

i &lt;- dge_list[[1]]
dge_down_fdr &lt;- foreach(i = dge_list) %do% {
  i$results %&gt;% 
    filter(FDR &lt; 0.001) %&gt;%
    filter(logFC &lt; -1) %&gt;% 
    arrange(FDR) %&gt;% 
    head(100) %&gt;% 
    pull(gene_name)
}
names(dge_down_fdr) &lt;- names(dge_list) %&gt;% gsub(&quot;.o.*&quot;, &quot;&quot;, .) %&gt;% paste0(&quot;down_fdr_&quot;, .)


i &lt;- dge_list[[1]]
dge_up_fc &lt;- foreach(i = dge_list) %do% {
  i$results %&gt;% 
    filter(FDR &lt; 0.001) %&gt;%
    filter(logFC &gt; 1) %&gt;% 
    arrange(desc(logFC)) %&gt;% 
    head(100) %&gt;% 
    pull(gene_name)
}
names(dge_up_fc) &lt;- names(dge_list) %&gt;% gsub(&quot;.o.*&quot;, &quot;&quot;, .) %&gt;% paste0(&quot;up_fc_&quot;, .)

i &lt;- dge_list[[1]]
dge_down_fc &lt;- foreach(i = dge_list) %do% {
  i$results %&gt;% 
    filter(FDR &lt; 0.001) %&gt;%
    filter(logFC &lt; -1) %&gt;% 
    arrange(logFC) %&gt;% 
    head(100) %&gt;% 
    pull(gene_name)
}
names(dge_down_fc) &lt;- names(dge_list) %&gt;% gsub(&quot;.o.*&quot;, &quot;&quot;, .) %&gt;% paste0(&quot;down_fc_&quot;, .)

dge_signatures_mm &lt;- c(dge_up_fdr, dge_down_fdr, dge_up_fc, dge_down_fc)</code></pre>
<p>Convert to genes to from mouse to human orthologs</p>
<pre class="r"><code>library(biomaRt)
use_mart &lt;- useMart(&quot;ensembl&quot;, dataset = &quot;mmusculus_gene_ensembl&quot;)

attributes&lt;-  c(&quot;external_gene_name&quot;,
                &quot;hsapiens_homolog_ensembl_gene&quot;, 
                &quot;hsapiens_homolog_associated_gene_name&quot;,
                &quot;hsapiens_homolog_orthology_type&quot;,
                &quot;hsapiens_homolog_perc_id_r1&quot;)

orth_info &lt;-  getBM(attributes, filters=&quot;with_hsapiens_homolog&quot;,
                    values=TRUE, mart = use_mart, uniqueRows=TRUE)
i &lt;-  dge_signatures_mm[[1]]
dge_signatures &lt;- foreach(i = dge_signatures_mm) %do% {
  orth_info %&gt;% filter(external_gene_name %in% i) %&gt;% pull(hsapiens_homolog_associated_gene_name)
}
names(dge_signatures) &lt;- names(dge_signatures_mm)
saveRDS(dge_signatures, file = file.path(params$output_dir, &#39;dge_signatures.rds&#39;))</code></pre>
<p>Load signatures</p>
<pre class="r"><code>dge_signatures &lt;- readRDS(file.path(params$output_dir, &#39;dge_signatures.rds&#39;))</code></pre>
</div>
<div id="gene-signatures" class="section level2" number="3">
<h2><span class="header-section-number">3</span> Gene signatures</h2>
<!-- Write signatures to xlsx file -->
<pre class="r"><code># File name summary
rmd_file &lt;- current_input()
if(is.null(rmd_file))
  rmd_file &lt;- &#39;tmp&#39;
file_xlsx &lt;- file.path(&#39;./docs/file&#39;,rmd_file, &#39;dge_signatures.xlsx&#39;)

dir.create(dirname(file_xlsx), recursive = TRUE, showWarnings = FALSE)

# Generate workbook
wb &lt;- createWorkbook()
for(i in names(dge_signatures)) {
  addWorksheet(wb, i)
  res &lt;- dge_signatures[[i]] %&gt;% data.frame %&gt;% 
    set_names(&#39;Gene&#39;) %&gt;%
    filter(Gene != &#39;&#39;)
  writeData(wb, i, res)
}
saveWorkbook(wb, file_xlsx, TRUE)

dge_file_xlsx &lt;- file_xlsx</code></pre>
<p>You can download the list of genes per signature using the following
link:</p>
<ul>
<li><a
href="./file/tcga_survival_amhr2_ko_signatures.Rmd/dge_signatures.xlsx"><strong>Differential
gene expression signatures</strong></a></li>
</ul>
<p>Below you will find a summary table with the number of genes for each
signature.</p>
<pre class="r"><code>lapply(dge_signatures, function(x) length(x[x!=&#39;&#39;])) %&gt;% 
  data.frame(check.names = FALSE) %&gt;%
  t() %&gt;% 
  data.frame(check.names = FALSE) %&gt;% 
  set_names(&#39;Number of genes&#39;) %&gt;%
  rownames_to_column(&#39;signature&#39;)  %&gt;%
  kbl(caption = &#39;Number of genes by signaure&#39;) %&gt;%
  kable_paper(bootstrap_options = c(&quot;striped&quot;, &quot;hover&quot;, &quot;condensed&quot;), full_width = F)</code></pre>
<table class=" lightable-paper" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
Number of genes by signaure
</caption>
<thead>
<tr>
<th style="text-align:left;">
signature
</th>
<th style="text-align:right;">
Number of genes
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
up_fdr_i_AMHR2_KO
</td>
<td style="text-align:right;">
94
</td>
</tr>
<tr>
<td style="text-align:left;">
up_fdr_c_AMHR2_KO
</td>
<td style="text-align:right;">
100
</td>
</tr>
<tr>
<td style="text-align:left;">
up_fdr_AMHR2_KO
</td>
<td style="text-align:right;">
89
</td>
</tr>
<tr>
<td style="text-align:left;">
down_fdr_i_AMHR2_KO
</td>
<td style="text-align:right;">
94
</td>
</tr>
<tr>
<td style="text-align:left;">
down_fdr_c_AMHR2_KO
</td>
<td style="text-align:right;">
96
</td>
</tr>
<tr>
<td style="text-align:left;">
down_fdr_AMHR2_KO
</td>
<td style="text-align:right;">
91
</td>
</tr>
<tr>
<td style="text-align:left;">
up_fc_i_AMHR2_KO
</td>
<td style="text-align:right;">
73
</td>
</tr>
<tr>
<td style="text-align:left;">
up_fc_c_AMHR2_KO
</td>
<td style="text-align:right;">
81
</td>
</tr>
<tr>
<td style="text-align:left;">
up_fc_AMHR2_KO
</td>
<td style="text-align:right;">
73
</td>
</tr>
<tr>
<td style="text-align:left;">
down_fc_i_AMHR2_KO
</td>
<td style="text-align:right;">
74
</td>
</tr>
<tr>
<td style="text-align:left;">
down_fc_c_AMHR2_KO
</td>
<td style="text-align:right;">
92
</td>
</tr>
<tr>
<td style="text-align:left;">
down_fc_AMHR2_KO
</td>
<td style="text-align:right;">
107
</td>
</tr>
</tbody>
</table>
</div>
<div id="run-survival-analysis" class="section level2" number="4">
<h2><span class="header-section-number">4</span> Run survival
analysis</h2>
<pre class="r"><code>project_id &lt;- list.files(
  path = params$data_dir, 
  pattern = &#39;TCGA-[A-Z]+.rds$&#39;
  ) %&gt;%
  gsub(&quot;.rds&quot;, &quot;&quot;, .)

tcga_selected &lt;- c(&#39;BRCA&#39;, &#39;LUAD&#39;, &#39;COAD&#39;) %&gt;% 
  paste(&#39;TCGA&#39;, ., sep = &#39;-&#39;)

project_id &lt;- intersect(project_id, tcga_selected)


surv_endpoints &lt;- list(
  DSS = list(
    time_var = &#39;DSS.time&#39;,
    event_var = &#39;DSS&#39;
  ),
  DFI = list(
    time_var = &#39;DFI.time&#39;,
    event_var = &#39;DFI&#39;
  )
)</code></pre>
<p>Load raw data, and create SummarizedExperiments with signatures for
each project.</p>
<pre class="r"><code>i &lt;- project_id[1]
data_se &lt;- foreach(i = project_id) %do% {
  cat(&quot;### &quot;, i, &quot;\n\n&quot;)
  se &lt;- readRDS(file.path(params$data_dir, paste0(i, &#39;.rds&#39;)))
  
  # Remove duplicated samples: https://ucsc-xena.gitbook.io/project/how-do-i/how-do-i-remove-duplicate-samples-from-a-km-plot
  sample_annot &lt;- colData(se) %&gt;% data.frame %&gt;% 
    filter(sample_type == &#39;Primary Tumor&#39;)
  
  # Filter samples
  se &lt;- se[, rownames(sample_annot)]
  
  # Run GSVA
  use_assay &lt;- edgeR::cpm(assay(se, &#39;normcounts&#39;), log = TRUE)
  gsvapar &lt;- gsvaParam(
    use_assay,
    dge_signatures,
    kcdf = &quot;Gaussian&quot;,
    maxDiff = TRUE
  )
  gsva_res &lt;- gsva(gsvapar, 
       verbose = FALSE)

  # Generate SummarizedExperiment object using GSVA matrix
  se &lt;- SummarizedExperiment(
    assays = list(gsva = gsva_res), 
    colData = colData(se)
    )
  
  se
}
names(data_se) &lt;- project_id

# Remove NULL elements
data_se &lt;- compact(data_se)
data_se &lt;- data_se[!sapply(data_se, is.null)]

saveRDS(data_se, file = file.path(params$output_dir, &#39;data_se_signatures.rds&#39;))</code></pre>
<p>Pool data from multiple studies</p>
<pre class="r"><code>data_se &lt;- readRDS(file.path(params$output_dir, &#39;data_se_signatures.rds&#39;))
# Combine projects for pooled survival analysis
se1 &lt;- data_se[[&quot;TCGA-BRCA&quot;]]
se2 &lt;- data_se[[&quot;TCGA-LUAD&quot;]]
se3 &lt;- data_se[[&quot;TCGA-COAD&quot;]]
use_cols &lt;- intersect(colnames(colData(se1)), colnames(colData(se2))) %&gt;% 
  intersect(., colnames(colData(se3)))
colData(se1) &lt;- colData(se1)[,use_cols]
colData(se2) &lt;- colData(se2)[,use_cols]
colData(se3) &lt;- colData(se3)[,use_cols]
data_se[[&quot;TCGA-BRCA-LUAD-COAD&quot;]] &lt;- cbind(se1, se2, se3)
saveRDS(data_se, file = file.path(params$output_dir, &#39;data_se_signatures.rds&#39;))</code></pre>
<p>For each SummarizedExperiments, we set up the follow-up limit and
configure survival variable. Remove studies with less than 3 events and
20 patients with survival data</p>
<pre class="r"><code>data_se_signatures &lt;- readRDS(file.path(params$output_dir, &#39;data_se_signatures.rds&#39;))

data_se &lt;- foreach(surv_i = surv_endpoints) %do% {
  res &lt;- foreach(se = data_se_signatures) %do% {
    # Set survival variables and follow up limit
    sample_annot &lt;- colData(se) %&gt;% data.frame()
    sample_annot$status &lt;- sample_annot[[surv_i$event_var]]
    sample_annot$time &lt;- sample_annot[[surv_i$time_var]]/365.25
    
    sample_annot &lt;- sample_annot %&gt;% 
      mutate(
        status = ifelse(time &gt;= params$max_time_years, 0, status),
        time = ifelse(time &gt;= params$max_time_years, params$max_time_years, time)
      )
    
    colData(se) &lt;- DataFrame(sample_annot)
    
    # Number of events
    n_surv_data &lt;- sum(!is.na(se$status))
    n_events &lt;- sum(se$status, na.rm = TRUE)
    
    if(ncol(se) &gt; 0 &amp; n_events &gt;= params$min_event_n &amp; n_surv_data &gt;= params$min_surv_n) {
      se
    } else {
      NULL
    }
  }
  names(res) &lt;- names(data_se_signatures)
  res
}
names(data_se) &lt;- names(surv_endpoints)

# Remove NULL elements
data_se &lt;- compact(data_se)
data_se &lt;- data_se[!sapply(data_se, is.null)]


saveRDS(data_se, file = file.path(params$output_dir, &#39;data_se_surv.rds&#39;))</code></pre>
<p>Generate data using optimal cutoff point. Cutoff determined using the
maximally selected rank statistics from the ‘maxstat’ R package. This is
an outcome-oriented methods providing a value of a cutpoint that
correspond to the most significant relation with outcome (here,
survival).</p>
<pre class="r"><code>surv_i &lt;- names(data_se)[1]
surv_maxstat &lt;- foreach(surv_i = names(data_se), .combine = rbind) %do% {
  res &lt;- foreach(i = names(data_se[[surv_i]]), .combine = rbind) %do% {
    se &lt;- data_se[[surv_i]][[i]]
    gene_exp &lt;- assay(se, &#39;gsva&#39;) %&gt;% t()
    
    colnames(gene_exp) &lt;-rownames(se)
    
    sample_annot &lt;- colData(se) %&gt;% data.frame()
    # sample_annot$status &lt;- sample_annot[[params$event_var]]
    # sample_annot$time &lt;- sample_annot[[params$time_var]]/365.25
    
    # Set follow up limit
    # sample_annot &lt;- sample_annot %&gt;% 
    #   mutate(
    #     status = ifelse(time &gt;= params$max_time_years, 0, status),
    #     time = ifelse(time &gt;= params$max_time_years, params$max_time_years, time)
    #   )
    # # Keep only genes with expression in at least 20% of samples
    # gene_exp_prop &lt;- colSums(gene_exp &gt; 0) / nrow(gene_exp)
    # gene_exp &lt;- gene_exp[,gene_exp_prop &gt;= 0.2]
    # 
    
    res_cut &lt;- surv_cutpoint(cbind(sample_annot, gene_exp),
                             time = &quot;time&quot;, event = &quot;status&quot;,
                             variables = colnames(gene_exp))
    res_cat &lt;- surv_categorize(res_cut, labels = c(&#39;Low&#39;, &#39;High&#39;))
    sample_annot &lt;- cbind(sample_annot, res_cat)
    
    g &lt;- colnames(gene_exp)[1]
    foreach(g = colnames(gene_exp), .combine = rbind) %do% {
      use_formula &lt;- as.formula(paste0(&quot;Surv(time, status) ~ &quot;, g))
      cox.res &lt;- coxph(use_formula, data = sample_annot) %&gt;%
        tbl_regression(exp = TRUE)
      data.frame(
        surv_var = surv_i,
        project_id = i,
        gene_name = g,
        HR = cox.res$table_body$estimate[3],
        p.value = cox.res$table_body$p.value[3],
        std.error = cox.res$table_body$std.error[3],
        conf.low = cox.res$table_body$conf.low[3],
        conf.high = cox.res$table_body$conf.high[3],
        n.low = sum(sample_annot[[g]] == &quot;Low&quot;),
        n.high = sum(sample_annot[[g]] == &quot;High&quot;)
      )
    }
  }
  res
}
saveRDS(surv_maxstat, file = file.path(params$output_dir, &#39;surv_maxstat.rds&#39;))

survfit_km_maxstat &lt;- foreach(surv_i = names(data_se)) %do% {
  res &lt;- foreach(i = names(data_se[[surv_i]])) %do% {
    se &lt;- data_se[[surv_i]][[i]]
    gene_exp &lt;- assay(se, &#39;gsva&#39;) %&gt;% t()
    
    sample_annot &lt;- colData(se) %&gt;% data.frame()
    # sample_annot$status &lt;- sample_annot[[params$event_var]]
    # sample_annot$time &lt;- sample_annot[[params$time_var]]/365.25
    
    # Set follow up limit
    # sample_annot &lt;- sample_annot %&gt;% 
    #   mutate(
    #     status = ifelse(time &gt;= params$max_time_years, 0, status),
    #     time = ifelse(time &gt;= params$max_time_years, params$max_time_years, time)
    #   )
    # # Keep only genes with expression in at least 10% of samples
    # gene_exp_prop &lt;- colSums(gene_exp &gt; 0) / nrow(gene_exp)
    # gene_exp &lt;- gene_exp[,gene_exp_prop &gt;= 0.2]
    
    res_cut &lt;- surv_cutpoint(cbind(sample_annot, gene_exp),
                             time = &quot;time&quot;, event = &quot;status&quot;,
                             variables = colnames(gene_exp))
    res_cat &lt;- surv_categorize(res_cut, labels = c(&#39;Low&#39;, &#39;High&#39;))
    sample_annot &lt;- cbind(sample_annot, res_cat)
    
    g &lt;- colnames(gene_exp)[1]
    res_by_genes &lt;- foreach(g = colnames(gene_exp)) %do% {
      use_formula &lt;- as.formula(paste0(&quot;Surv(time, status) ~ &quot;, g))
      # KM plot
      survfit2(use_formula, data = sample_annot) %&gt;% 
        ggsurvfit() +
        add_risktable() +
        labs(
          x = &quot;Time (years)&quot;,
          title = i
        ) +
        theme_project + 
        theme(legend.position=&quot;bottom&quot;)
      
    }
    names(res_by_genes) &lt;- colnames(gene_exp)
    res_by_genes
  }
  names(res) &lt;- names(data_se[[surv_i]])
  res
}

names(survfit_km_maxstat) &lt;- names(data_se)
saveRDS(survfit_km_maxstat, file = file.path(params$output_dir, &#39;survfit_km_maxstat.rds&#39;))</code></pre>
<p>Load survival data</p>
<pre class="r"><code>data_se &lt;- readRDS(file.path(params$output_dir, &#39;data_se_signatures.rds&#39;))
surv_maxstat &lt;- readRDS(file.path(params$output_dir, &#39;surv_maxstat.rds&#39;))
survfit_km_maxstat &lt;- readRDS(file.path(params$output_dir, &#39;survfit_km_maxstat.rds&#39;))
survfit_km_maxstat_list &lt;- survfit_km_maxstat</code></pre>
<p>Change signature names to more readable format</p>
<pre class="r"><code>new_signature_names &lt;- c(
  up_fc_i_AMHR2_KO = &#39;AMH loss signature (TME)&#39;,
  down_fc_c_AMHR2_KO = &#39;AMH fully active signature&#39;
)

# Select signatures and change names
dge_signatures &lt;- dge_signatures[names(new_signature_names)]
names(dge_signatures) &lt;- new_signature_names

# Modify data
new_data_se &lt;- foreach(i = data_se) %do% {
  res &lt;- i[names(new_signature_names),]
  rownames(res) &lt;- new_signature_names
  res
}
names(new_data_se) &lt;- names(data_se)
data_se &lt;- new_data_se
rm(new_data_se)

res &lt;- surv_maxstat %&gt;% 
  filter(gene_name %in% names(new_signature_names))
for(i in names(new_signature_names)){
  res$gene_name[res$gene_name == i] &lt;- new_signature_names[i]
}
surv_maxstat &lt;- res


new_survfit_km_maxstat&lt;- foreach(surv_i = names(survfit_km_maxstat)) %do% {
  res &lt;- foreach(i = names(survfit_km_maxstat[[surv_i]])) %do% {
    x &lt;- survfit_km_maxstat[[surv_i]][[i]][names(new_signature_names)]
    names(x) &lt;- new_signature_names
    x
  }
  names(res) &lt;- names(survfit_km_maxstat[[surv_i]])
  res
}
names(new_survfit_km_maxstat) &lt;- names(survfit_km_maxstat)
survfit_km_maxstat &lt;- new_survfit_km_maxstat
survfit_km_maxstat_list &lt;- survfit_km_maxstat</code></pre>
<p>Run metanalysis for survival analysis
<!-- https://bookdown.org/MathiasHarrer/Doing_Meta_Analysis_in_R/ --></p>
<pre class="r"><code>## Survival with maxtat
meta_surv_maxstat &lt;- foreach(sv = names(surv_endpoints)) %do% {
  res &lt;- foreach(g = unique(surv_maxstat$gene_name)) %do% {
    use_surv &lt;- surv_maxstat %&gt;% filter(gene_name == g &amp; surv_var == sv &amp; project_id != &#39;TCGA-BRCA-LUAD-COAD&#39;)
    metagen(
      HR = log(use_surv$HR), 
      lower = log(use_surv$conf.low), 
      upper = log(use_surv$conf.high),
      sm = &quot;HR&quot;, common=F, random=T,
      studlab = use_surv$project_id,
      method.tau = &quot;DL&quot;, ## method to calculate Tau
      method.random.ci = &quot;classic&quot; ## method to calculate estimator&#39;s CI
    )
  }
  names(res) &lt;- unique(surv_maxstat$gene_name)
  res
}
names(meta_surv_maxstat) &lt;- names(surv_endpoints)

meta_surv_maxstat_summary &lt;- foreach(sv = names(surv_endpoints), .combine = rbind) %do% {
  foreach(g = unique(surv_maxstat$gene_name), .combine = rbind) %do% {
    x &lt;- summary(meta_surv_maxstat[[sv]][[g]])
    data.frame(
      surv_var = sv,
      project_id = &#39;Summary&#39;,
      gene_name = g,
      HR = exp(x$random$TE),
      p.value = x$random$p,
      std.error = x$random$seTE,
      conf.low = exp(x$random$lower),
      conf.high = exp(x$random$upper)#,
      # w.random = round(100*x$w.random / sum(x$w.random), 1)
    )
  }
}

meta_surv_maxstat_weight &lt;- foreach(sv = names(surv_endpoints), .combine = rbind) %do% {
  foreach(g = unique(surv_maxstat$gene_name), .combine = rbind) %do% {
    x &lt;- summary(meta_surv_maxstat[[sv]][[g]])
    data.frame(
      surv_var = sv,
      project_id = x$studlab,
      gene_name = g,
      w.random = round(100*x$w.random / sum(x$w.random), 1)
    )
  }
}

use_meta_surv &lt;- meta_surv_maxstat</code></pre>
<pre class="r"><code>use_surv &lt;- surv_maxstat
use_meta_surv &lt;- meta_surv_maxstat</code></pre>
</div>
<div id="table-of-results" class="section level2" number="5">
<h2><span class="header-section-number">5</span> Table of results</h2>
<!-- Write results to xlsx file -->
<pre class="r"><code># File name summary
rmd_file &lt;- current_input()
if(is.null(rmd_file))
  rmd_file &lt;- &#39;tmp&#39;
file_xlsx &lt;- file.path(&#39;./docs/file&#39;,rmd_file, &#39;amhr2_signature_surv_maxstat.xlsx&#39;)

dir.create(dirname(file_xlsx), recursive = TRUE, showWarnings = FALSE)

# Generate workbook
wb &lt;- createWorkbook()
for(i in unique(use_surv$surv_var)) {
  addWorksheet(wb, i)
  res &lt;- use_surv %&gt;% 
    filter(surv_var == i) %&gt;% 
    dplyr::rename(signature = gene_name) %&gt;%
    select(surv_var, everything())
  
  res_meta &lt;- foreach(g = unique(res$signature), .combine = rbind) %do% {
    use_meta_sum &lt;- summary(use_meta_surv[[i]][[g]])
    data.frame(
      project_id = &#39;Meta&#39;,
      signature = g,
      HR = exp(use_meta_sum$random$TE),
      p.value = use_meta_sum$random$p,
      std.error = use_meta_sum$random$seTE,
      conf.low = exp(use_meta_sum$random$lower),
      conf.high = exp(use_meta_sum$random$upper),
      surv_var = i
    )
  }
  res &lt;- bind_rows(res, res_meta)
  writeData(wb, i, res)
}
saveWorkbook(wb, file_xlsx, TRUE)

dge_file_xlsx &lt;- file_xlsx</code></pre>
<p>You can download the results of the survival analysis using the
following link:</p>
<ul>
<li><a
href="./file/tcga_survival_amhr2_ko_signatures.Rmd/amhr2_signature_surv_maxstat.xlsx"><strong>Survival
analysis using optimal cut-off</strong></a></li>
</ul>
</div>
<div id="figure-7h-left-forest-plot-by-study" class="section level2"
number="6">
<h2><span class="header-section-number">6</span> Figure 7H Left: Forest
plot by study</h2>
<p>Forest plot showing HR and 95%CI for each study. The size of the
circle point is determined by the weight of the effect size in the
meta-analysis. The diamond shape at the bottom of the plot represents
the average effect calculated using meta-analysis.</p>
<pre class="r"><code>chunk_data &lt;- use_surv

meta_weight &lt;- foreach(i = names(use_meta_surv), .combine = rbind) %do% {
  foreach(j = names(use_meta_surv[[i]]), .combine = rbind) %do% {
    use_meta_sum &lt;- summary(use_meta_surv[[i]][[j]])
    data.frame(
      surv_var = i,
      gene_name = j,
      project_id = use_meta_sum$studlab,
      w.fixed = use_meta_sum$w.fixed,
      w.pct = 100* use_meta_sum$w.fixed / sum(use_meta_sum$w.fixed)
    )
  }
}


# use_meta_sum &lt;- summary(use_meta)
chunk_data %&lt;&gt;% left_join(meta_weight, by = c(&#39;surv_var&#39;, &#39;gene_name&#39;, &#39;project_id&#39;))

cox_meta_add &lt;- foreach(i = names(use_meta_surv), .combine = rbind) %do% {
  foreach(g = names(use_meta_surv[[i]]), .combine = rbind) %do% {
    use_meta_sum &lt;- summary(use_meta_surv[[i]][[g]])
    cox_meta &lt;- data.frame(
      project_id = &#39;Meta&#39;,
      gene_name = g,
      HR = exp(use_meta_sum$random$TE),
      p.value = use_meta_sum$random$p,
      std.error = use_meta_sum$random$seTE,
      conf.low = exp(use_meta_sum$random$lower),
      conf.high = exp(use_meta_sum$random$upper),
      surv_var = i,
      w.fixed = NA,
      w.pct = 2*max(chunk_data %&gt;% filter(gene_name == g) %&gt;% pull(w.pct), na.rm = TRUE)
    )
    cox_empty &lt;- data.frame(
      project_id = &#39;&#39;,
      gene_name = g,
      HR = NA,
      p.value = 1,
      std.error = 0,
      conf.low = 1,
      conf.high = 1,
      surv_var = i,
      w.fixed = NA,
      w.pct = 0
    )
    rbind(cox_meta, cox_empty)
  }
}

chunk_data %&lt;&gt;%
  bind_rows(cox_meta_add) %&gt;%
  mutate(
    project_id = factor(project_id, levels= c(&#39;Meta&#39;, &#39;&#39;, rev(tcga_selected))),
    project_type = ifelse(project_id == &#39;Meta&#39;, &#39;Meta&#39;, &#39;Project&#39;),
    FDR = p.adjust(p.value, method = &#39;BH&#39;),
    logP = -log10(p.value),
    logP = ifelse(logP &gt; 10, 10 , logP),
    logFDR = -log10(FDR),
    logFDR = ifelse(logFDR &gt; 10, 10 , logFDR),
    root_signature_name = gsub(&quot; loss| fully active&quot;, &quot;&quot;, gene_name) %&gt;% 
      gsub(&quot; active&quot;, &quot;&quot;, .)
  )</code></pre>
<p>Add <strong>AMH loss signature (TME)</strong> as a loss signature for
<strong>AMH fully active signature</strong></p>
<pre class="r"><code>chunk_data %&lt;&gt;% 
  # filter(gene_name == &#39;AMH loss signature (TME)&#39;) %&gt;% 
  mutate(
    root_signature_name = ifelse(gene_name == &#39;AMH loss signature (TME)&#39;, &#39;AMH signature&#39;, root_signature_name)
  )</code></pre>
<pre class="r"><code>fc_palette &lt;- c(Active = &quot;#e41a1c&quot;, Loss = &quot;#377eb8&quot;)
shape_scale &lt;- c(DSS = 16, DFI = 15, PFI = 0)
use_postition_dodge &lt;- 0.8
use_lwd &lt;- (1/2.141959)/4

signature_levels &lt;- chunk_data$root_signature_name %&gt;% unique

for(i in signature_levels) {
  # cat(&quot;###&quot;, i, &quot;\n\n&quot;)
  res &lt;- chunk_data %&gt;%
    filter(surv_var %in% c(&#39;DSS&#39;, &#39;DFI&#39;)) %&gt;% 
    filter(!project_id %in% c(&#39;&#39;))  %&gt;% 
    filter(!is.na(project_id)) %&gt;% 
    filter(root_signature_name == i) %&gt;% 
    mutate(
      gene_name = ifelse(grepl(&#39;loss&#39;, gene_name), &#39;Loss&#39;, &#39;Active&#39;)
    ) %&gt;% 
    filter(std.error &lt;= 10) %&gt;%
    ggplot(aes(HR, project_id,
               size = w.pct,
               shape = surv_var,
               color = gene_name
    )
    ) +
    geom_point(position = position_dodge(use_postition_dodge)) +
    geom_linerange(aes(xmin=conf.low, xmax=conf.high), 
                   linewidth = use_lwd,
                   show.legend = FALSE,
                   position = position_dodge(use_postition_dodge)
    ) +
    geom_vline(xintercept = 1, linetype = &#39;dotted&#39;, linewidth = use_lwd) +
    facet_grid(rows = vars(gene_name)) +
    scale_shape_manual(values = shape_scale) +
    scale_color_manual(values = fc_palette) +
    scale_size_continuous(range = c(0.6, 2)
                          # breaks = c(0.1, 0.5, 1, 2, 5, 10),
    ) +
    scale_x_continuous(trans = &#39;log2&#39;,
                       breaks = c(0.1, 0.2, 0.5, 1, 2, 4, 10),
                       labels = as.character(c(0.1, 0.2, 0.5, 1, 2, 4, 10))
    ) +
    theme(
      axis.line.y = element_blank(),
      axis.ticks.y = element_blank(),
      axis.line.x = element_line(linewidth = use_lwd),
      axis.ticks.x = element_line(linewidth = use_lwd)
    ) +
    labs(
      x = &#39;HR&#39;,
      y = &#39;&#39;,
      shape = &#39;&#39;,
      size = &#39;Weight (%)&#39;,
      caption = &#39;Estimates with a standard error &gt; 10 were removed from the plot&#39;,
      title = i,
      color = &#39;Signature&#39;
    ) #+
  # guides(
  #   shape = &#39;none&#39;
  # )
  print(res)
  cat(&quot;\n\n&quot;)
}</code></pre>
<p><img src="figure/tcga_survival_amhr2_ko_signatures.Rmd/signature-maxstat-coxph-ggplot-forestplot-dss-dfi-vertical-1.png" width="1680" style="display: block; margin: auto;" /></p>
</div>
<div id="figure-7h-right-pooled-kaplan-meier-plots---optimal-cut-off"
class="section level2" number="7">
<h2><span class="header-section-number">7</span> Figure 7H Right: Pooled
Kaplan-Meier plots - Optimal cut-off</h2>
<pre class="r"><code>use_survfit_km &lt;- foreach(surv_var_i = names(survfit_km_maxstat_list)) %do% {
  res &lt;- foreach(p_i = names(survfit_km_maxstat_list[[surv_var_i]])) %do% {
    survfit_km_maxstat_list[[surv_var_i]][[p_i]][[&#39;TCGA-BRCA-LUAD-COAD&#39;]]
  }
  names(res) &lt;- names(survfit_km_maxstat_list[[surv_var_i]])
}
names(use_survfit_km) &lt;- names(survfit_km_maxstat_list)

use_survfit_km &lt;- survfit_km_maxstat_list[&#39;DFI&#39;]</code></pre>
<pre class="r"><code>use_lwd &lt;- (1/2.141959)/4
surv_var_i &lt;- names(use_survfit_km)[1]
gene_i &lt;- names(use_survfit_km[[1]][[1]])[1]
p_i &lt;- names(use_survfit_km[[1]])[1]
for(surv_var_i in names(use_survfit_km)) {
  for(gene_i in names(use_survfit_km[[surv_var_i]][[1]])) {
    cat(&quot;###&quot;, surv_var_i,&quot;-&quot;, gene_i, &quot;\n\n&quot;)
    for(p_i in names(use_survfit_km[[surv_var_i]])) {
      top_color &lt;- ifelse(grepl(&quot;active&quot;, gene_i), 
                          &#39;#e41a1c&#39;, 
                          &#39;#377eb8&#39;)
      use_palette &lt;- c(Low = &#39;grey60&#39;, High = top_color)
      hr_p &lt;- chunk_data %&gt;% 
        filter(surv_var == surv_var_i) %&gt;% 
        filter(project_id == &quot;Meta&quot;) %&gt;% 
        filter(gene_name == gene_i)
      cox_res &lt;- paste0(&#39;Meta-analysis HR(95%CI)=&#39;, round(hr_p$HR, 2), 
                        &#39;(&#39;, round(hr_p$conf.low,2),&#39;-&#39;,round(hr_p$conf.high,2),&#39;)&#39;,
                        &#39;;P=&#39;,format.pval(hr_p$p.value,2))
      
      res &lt;- use_survfit_km[[surv_var_i]][[p_i]][[gene_i]] +
        labs(
          title = paste(surv_var_i, p_i, gene_i),
          caption = cox_res
        ) +
        scale_color_manual(values = use_palette) +
        theme(
          axis.line = element_line(linewidth = use_lwd),
          axis.ticks = element_line(linewidth = use_lwd)
        )
      print(res)
      cat(&quot;\n\n&quot;)
      

    } 
  }
}</code></pre>
<div id="dfi---amh-loss-signature-tme" class="section level3"
number="7.1">
<h3><span class="header-section-number">7.1</span> DFI - AMH loss
signature (TME)</h3>
<p><img src="figure/tcga_survival_amhr2_ko_signatures.Rmd/signature-km-pooled-1.png" width="1920" style="display: block; margin: auto;" /></p>
<p><img src="figure/tcga_survival_amhr2_ko_signatures.Rmd/signature-km-pooled-2.png" width="1920" style="display: block; margin: auto;" /></p>
<p><img src="figure/tcga_survival_amhr2_ko_signatures.Rmd/signature-km-pooled-3.png" width="1920" style="display: block; margin: auto;" /></p>
<p><img src="figure/tcga_survival_amhr2_ko_signatures.Rmd/signature-km-pooled-4.png" width="1920" style="display: block; margin: auto;" /></p>
</div>
<div id="dfi---amh-fully-active-signature" class="section level3"
number="7.2">
<h3><span class="header-section-number">7.2</span> DFI - AMH fully
active signature</h3>
<p><img src="figure/tcga_survival_amhr2_ko_signatures.Rmd/signature-km-pooled-5.png" width="1920" style="display: block; margin: auto;" /></p>
<p><img src="figure/tcga_survival_amhr2_ko_signatures.Rmd/signature-km-pooled-6.png" width="1920" style="display: block; margin: auto;" /></p>
<p><img src="figure/tcga_survival_amhr2_ko_signatures.Rmd/signature-km-pooled-7.png" width="1920" style="display: block; margin: auto;" /></p>
<p><img src="figure/tcga_survival_amhr2_ko_signatures.Rmd/signature-km-pooled-8.png" width="1920" style="display: block; margin: auto;" /></p>
<pre class="r"><code># c(Up = &quot;#e41a1c&quot;, Down = &quot;#377eb8&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<p>R version 4.4.3 (2025-02-28) Platform: aarch64-apple-darwin20 Running
under: macOS Sequoia 15.5</p>
<p>Matrix products: default BLAS:
/Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib
LAPACK:
/Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;
LAPACK version 3.12.0</p>
<p>locale: [1]
en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</p>
<p>time zone: Europe/Zurich tzcode source: internal</p>
<p>attached base packages: [1] stats4 stats graphics grDevices utils
datasets methods<br />
[8] base</p>
<p>other attached packages: [1] cowplot_1.1.3.9000 GSVA_2.0.7<br />
[3] openxlsx_4.2.8 kableExtra_1.4.0<br />
[5] magrittr_2.0.3 meta_8.1-0<br />
[7] metadat_1.4-0 survminer_0.5.0<br />
[9] ggpubr_0.6.0 ggsurvfit_1.1.0<br />
[11] gtsummary_2.2.0 survival_3.8-3<br />
[13] SummarizedExperiment_1.36.0 Biobase_2.66.0<br />
[15] GenomicRanges_1.58.0 GenomeInfoDb_1.42.3<br />
[17] IRanges_2.40.1 S4Vectors_0.44.0<br />
[19] BiocGenerics_0.52.0 MatrixGenerics_1.18.1<br />
[21] matrixStats_1.5.0 foreach_1.5.2<br />
[23] knitr_1.50 lubridate_1.9.4<br />
[25] forcats_1.0.0 stringr_1.5.1<br />
[27] dplyr_1.1.4 purrr_1.0.4<br />
[29] readr_2.1.5 tidyr_1.3.1<br />
[31] tibble_3.2.1 ggplot2_3.5.2<br />
[33] tidyverse_2.0.0 workflowr_1.7.1</p>
<p>loaded via a namespace (and not attached): [1] splines_4.4.3
later_1.4.2<br />
[3] graph_1.84.1 XML_3.99-0.18<br />
[5] lifecycle_1.0.4 Rdpack_2.6.4<br />
[7] rstatix_0.7.2 rprojroot_2.0.4<br />
[9] processx_3.8.6 lattice_0.22-7<br />
[11] MASS_7.3-65 backports_1.5.0<br />
[13] metafor_4.8-0 sass_0.4.10<br />
[15] rmarkdown_2.29 jquerylib_0.1.4<br />
[17] yaml_2.3.10 httpuv_1.6.16<br />
[19] zip_2.3.3 DBI_1.2.3<br />
[21] minqa_1.2.8 RColorBrewer_1.1-3<br />
[23] abind_1.4-8 zlibbioc_1.52.0<br />
[25] git2r_0.36.2 GenomeInfoDbData_1.2.13<br />
[27] KMsurv_0.1-5 irlba_2.3.5.1<br />
[29] annotate_1.84.0 svglite_2.2.1<br />
[31] codetools_0.2-20 DelayedArray_0.32.0<br />
[33] xml2_1.3.8 tidyselect_1.2.1<br />
[35] UCSC.utils_1.2.0 farver_2.1.2<br />
[37] lme4_1.1-37 ScaledMatrix_1.14.0<br />
[39] mathjaxr_1.8-0 jsonlite_2.0.0<br />
[41] Formula_1.2-5 iterators_1.0.14<br />
[43] systemfonts_1.2.3 tools_4.4.3<br />
[45] Rcpp_1.0.14 glue_1.8.0<br />
[47] gridExtra_2.3 SparseArray_1.6.2<br />
[49] xfun_0.52 HDF5Array_1.34.0<br />
[51] withr_3.0.2 numDeriv_2016.8-1.1<br />
[53] fastmap_1.2.0 boot_1.3-31<br />
[55] rhdf5filters_1.18.1 callr_3.7.6<br />
[57] digest_0.6.37 rsvd_1.0.5<br />
[59] timechange_0.3.0 R6_2.6.1<br />
[61] textshaping_1.0.1 dichromat_2.0-0.1<br />
[63] RSQLite_2.3.11 generics_0.1.4<br />
[65] data.table_1.17.2 httr_1.4.7<br />
[67] S4Arrays_1.6.0 whisker_0.4.1<br />
[69] pkgconfig_2.0.3 gtable_0.3.6<br />
[71] blob_1.2.4 SingleCellExperiment_1.28.1 [73] XVector_0.46.0
survMisc_0.5.6<br />
[75] htmltools_0.5.8.1 carData_3.0-5<br />
[77] GSEABase_1.68.0 scales_1.4.0<br />
[79] png_0.1-8 SpatialExperiment_1.16.0<br />
[81] reformulas_0.4.1 km.ci_0.5-6<br />
[83] rstudioapi_0.17.1 tzdb_0.5.0<br />
[85] rjson_0.2.23 nlme_3.1-168<br />
[87] nloptr_2.2.1 cachem_1.1.0<br />
[89] zoo_1.8-14 rhdf5_2.50.2<br />
[91] parallel_4.4.3 AnnotationDbi_1.68.0<br />
[93] pillar_1.10.2 grid_4.4.3<br />
[95] vctrs_0.6.5 promises_1.3.2<br />
[97] car_3.1-3 BiocSingular_1.22.0<br />
[99] beachmat_2.22.0 xtable_1.8-4<br />
[101] evaluate_1.0.3 magick_2.8.6<br />
[103] cli_3.6.5 compiler_4.4.3<br />
[105] rlang_1.1.6 crayon_1.5.3<br />
[107] ggsignif_0.6.4 labeling_0.4.3<br />
[109] ps_1.9.1 getPass_0.2-4<br />
[111] fs_1.6.6 stringi_1.8.7<br />
[113] viridisLite_0.4.2 BiocParallel_1.40.2<br />
[115] Biostrings_2.74.1 CompQuadForm_1.4.3<br />
[117] Matrix_1.7-3 patchwork_1.3.0<br />
[119] hms_1.1.3 sparseMatrixStats_1.18.0<br />
[121] bit64_4.6.0-1 Rhdf5lib_1.28.0<br />
[123] KEGGREST_1.46.0 rbibutils_2.3<br />
[125] broom_1.0.8 memoise_2.0.1<br />
[127] bslib_0.9.0 bit_4.6.0</p>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4,h5",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
